<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Credits</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="#fFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics">
    </TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../js/jquery.js'></script>
<script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../cgi-bin/hgMenubar"--></TD></TR>
<!-- Intro Tables====================================================== -->
    <TR><TD COLSPAN=2>	
    	<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	    <TR><TD>
		<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; UCSC Genome Browser Acknowledgments</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		    <TR><TD WIDTH=10></TD>
			<TD>
			<P>
			The data and software displayed on this site are the result of
			a large collaborative effort among many individuals at UCSC and
			at research institutions around the world. We'd like to 
			acknowledge the scientists and engineers who have contributed
			to the UCSC Genome Browser and its data, and the organizations 
			who have made this project possible through their generous 
			funding.
			<P>
			<B>Mammals</B>
			<UL>
	        	    <LI><A HREF="#human_credits">Human Genome</A>
	        	    <LI><A HREF="#alpaca_credits">Alpaca Genome</A>
	        	    <LI><A HREF="#armadillo_credits">Armadillo Genome</A>
	        	    <LI><A HREF="#baboon_credits">Baboon Genome</A>
	        	    <LI><A HREF="#bonobo_credits">Bonobo Genome</A>
	        	    <LI><A HREF="#bushbaby_credits">Bushbaby Genome</A>
	        	    <LI><A HREF="#cat_credits">Cat Genome</A>
	        	    <LI><A HREF="#chimp_credits">Chimpanzee Genome</A>
                            <LI><A HREF="#chinese_hamster_credits">Chinese Hamster Genome</A>
	        	    <LI><A HREF="#cow_credits">Cow Genome</A>
	        	    <LI><A HREF="#dog_credits">Dog Genome</A>
	        	    <LI><A HREF="#dolphin_credits">Dolphin Genome</A>
	        	    <LI><A HREF="#elephant_credits">Elephant Genome</A>
	        	    <LI><A HREF="#ferret_credits">Ferret Genome</A>
	        	    <LI><A HREF="#gibbon_credits">Gibbon Genome</A>
	        	    <LI><A HREF="#gorilla_credits">Gorilla Genome</A>
	        	    <LI><A HREF="#guinea_pig_credits">Guinea Pig Genome</A>
	        	    <LI><A HREF="#hedgehog_credits">Hedgehog Genome</A>
	        	    <LI><A HREF="#horse_credits">Horse Genome</A>
	        	    <LI><A HREF="#kangaroo_rat_credits">Kangaroo Rat Genome</A>
	        	    <LI><A HREF="#manatee_credits">Manatee Genome</A>
	        	    <LI><A HREF="#marmoset_credits">Marmoset Genome</A>
	        	    <LI><A HREF="#megabat_credits">Megabat Genome</A>
	        	    <LI><A HREF="#microbat_credits">Microbat Genome</A>
			    <LI><A HREF="#minke_whale_credits">Minke Whale Genome</A>
	        	    <LI><A HREF="#mouse_credits">Mouse Genome</A>
	        	    <LI><A HREF="#mouse_lemur_credits">Mouse Lemur Genome</A>
      			    <LI><A HREF="#naked_mole-rat_credits">Naked Mole-rat Genome</A>
	        	    <LI><A HREF="#opossum_credits">Opossum Genome</A>
	        	    <LI><A HREF="#orangutan_credits">Orangutan Genome</A>
	        	    <LI><A HREF="#panda_credits">Panda Genome</A>
	        	    <LI><A HREF="#pig_credits">Pig Genome</A>
	        	    <LI><A HREF="#pika_credits">Pika Genome</A>
	        	    <LI><A HREF="#platypus_credits">Platypus Genome</A>
	        	    <LI><A HREF="#rabbit_credits">Rabbit Genome</A>
	        	    <LI><A HREF="#rat_credits">Rat Genome</A>
	        	    <LI><A HREF="#rhesus_credits">Rhesus Genome</A>
	        	    <LI><A HREF="#rock_hyrax_credits">Rock Hyrax Genome</A>
	        	    <LI><A HREF="#sheep_credits">Sheep Genome</A>
	        	    <LI><A HREF="#shrew_credits">Shrew Genome</A>
	        	    <LI><A HREF="#sloth_credits">Sloth Genome</A>
			    <LI><A HREF="#squirrel_credits">Squirrel Genome</A>
			    <LI><A HREF="#squirrel_monkey_credits">Squirrel Monkey Genome</A>
	        	    <LI><A HREF="#tarsier_credits">Tarsier Genome</A>
			    <LI><A HREF="#tasmanian_devil_credits">Tasmanian Devil Genome</A>
                            <LI><A HREF="#tenrec_credits">Tenrec Genome</A>
	        	    <LI><A HREF="#tree_shrew_credits">Tree Shrew Genome</A>
			    <LI><A HREF="#wallaby_credits">Wallaby Genome</A>
			    <LI><A HREF="#white_rhinoceros_credits">White Rhinoceros Genome</A>
			</UL>
			<P>
			<B>Other Vertebrates</B>
			<UL>
			<LI><A HREF="#american_alligator_credits">American Alligator Genome</A>
			<LI><A HREF="#atlantic_cod_credits">Atlantic Cod Genome</A>
			<LI><A HREF="#budgerigar_credits">Budgerigar Genome</A>
			<LI><A HREF="#chicken_credits">Chicken Genome</A>
			<LI><A HREF="#coelacanth_credits">Coelacanth Genome</A>
			<LI><A HREF="#elephant_shark_credits">Elephant Shark Genome</A>
			<LI><A HREF="#fugu_credits">Fugu Genome</A>
			<LI><A HREF="#lamprey_credits">Lamprey Genome</A>
			<LI><A HREF="#lizard_credits">Lizard Genome</A>
			<LI><A HREF="#medaka_credits">Medaka Genome</A>
			<LI><A HREF="#medium_ground_finch_credits">Medium Ground Finch Genome</A>
			<LI><A HREF="#nile_tilapia_credits">Nile Tilapia Genome</A>
			<LI><A HREF="#painted_turtle_credits">Painted Turtle Genome</A>
			<LI><A HREF="#stickleback_credits">Stickleback Genome</A>
			<LI><A HREF="#tetraodon_credits">Tetraodon Genome</A>
			<LI><A HREF="#turkey_credits">Turkey Genome</A>
			<LI><A HREF="#xentro_credits"><em>X. tropicalis</em> Genome</A>
			<LI><A HREF="#zebra_finch_credits">Zebra Finch Genome</A>
			<LI><A HREF="#zebrafish_credits">Zebrafish Genome</A>
		    </UL>
		    <P>
		    <B>Deuterostomes</B>
		    <UL>
			<LI><A HREF="#ciona_credits"><em>C. intestinalis</em> Genome</A>
			<LI><A HREF="#lancelet_credits">Lancelet Genome</A>
			<LI><A HREF="#urchin_credits"><em>S. purpuratus</em> Genome</A>
		    </UL>
		    <P>
		    <B>Insects</B>
		    <UL>
			<LI><A HREF="#anoGam_credits"><em>A. gambiae</em> Genome</A>
			<LI><A HREF="#apiMel_credits"><em>A. mellifera</em> Genome</A>
			<LI><A HREF="#droAna_credits"><em>D. ananassae</em> Genome</A>
			<LI><A HREF="#droEre_credits"><em>D. erecta</em> Genome</A>
			<LI><A HREF="#droGri_credits"><em>D. grimshawi</em> Genome</A>
			<LI><A HREF="#dm_credits"><em>D. melanogaster</em> Genome</A>
			<LI><A HREF="#droMoj_credits"><em>D. mojavensis</em> Genome</A>
			<LI><A HREF="#droPer_credits"><em>D. persimilis</em> Genome</A>
			<LI><A HREF="#dp_credits"><em>D. pseudoobscura</em> Genome</A>
			<LI><A HREF="#droSec_credits"><em>D. sechellia</em> Genome</A>
			<LI><A HREF="#droSim_credits"><em>D. simulans</em> Genome</A>
			<LI><A HREF="#droVir_credits"><em>D. virilis</em> Genome</A>
			<LI><A HREF="#droYak_credits"><em>D. yakuba</em> Genome</A>
		    </UL>
		    <P>
		    <B>Nematodes</B>
		    <UL>
			<LI><A HREF="#worm_credits"><em>C. brenneri</em> Genome</A>
			<LI><A HREF="#worm_credits"><em>C. briggsae</em> Genome</A>
			<LI><A HREF="#ce_credits"><em>C. elegans</em> Genome</A>
			<LI><A HREF="#worm_credits"><em>C. japonica</em> Genome</A>
			<LI><A HREF="#worm_credits"><em>C. remanei</em> Genome</A>
			<LI><A HREF="#worm_credits"><em>P. pacificus</em> Genome</A>
		    </UL>
		    <P>
		    <B>Other</B>
		    <UL>
			<LI><A HREF="#yeast_credits">Yeast Genome</A>
			<LI><A HREF="#seahare_credits">California Sea Hare Genome</A>
		    </UL>
		    <B>Viruses</B>
		    <UL>
			<LI><A HREF="#ebola_credits">Ebola Genome</A>
		    </UL>
		    <UL>
			<LI><A HREF="#other_credits">Other scientists featured on this site</A>
			<LI><A HREF="#early_credits">Acknowledgment of early support</A>
			<LI><A HREF="../staff.html">UCSC Genome Browser project team</A>
		    </UL>
		    </P>
		    </TD>
		    <TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Human Credits ====================================================== -->
<A NAME="human_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; Human Genome Browser: Acknowledgments </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		    <TD>
			<TABLE BORDER=0 WIDTH=435 ALIGN="right" CELLPADDING=2 CELLSPACING=2>
			    <TR><TD><IMG SRC="../images/DOT.gif" WIDTH=10 HEIGHT=1></TD>
			    <TD>
				<TABLE BORDER=1 CELLPADDING=0 CELLSPACING=0 ALIGN="right">
				    <TR><TD>
					<TABLE BORDER=1 CELLPADDING=3 CELLSPACING=3>
					    <TR><TD ROWSPAN=2 BGCOLOR="CECE9C" WIDTH="10%">
						<A HREF="http://www.hhmi.org" TARGET="_blank">
						    <IMG SRC="../images/HHMI.gif" BORDER=0 ALT="HHMI"></A></TD>
					    <TD BGCOLOR="002D59" HEIGHT="38">	
						<A HREF="http://www.ucop.edu/california-institutes/" TARGET="_blank">
						    <IMG SRC="../images/CISI.gif" BORDER=0 ALT="CISI"></A></TD></TR>
					    <TR><TD BGCOLOR="D9E4F8" >			
						<A HREF="http://www.genome.gov/" TARGET="_blank">
						    <IMG SRC="../images/nhgriLogo.gif" BORDER=0 ALT="NHGRI"></A></TD>
					    </TR>
				       </TABLE>			
				   </TD></TR>
				</TABLE>		
			    </TD></TR>
			</TABLE>	
		    <P>
		    The UCSC Human Genome Browser is generated by the 
		    <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A> in 
		    collaboration with the <A HREF="http://www.genome.gov/10001772" 
		    TARGET="_blank">International Human Genome Project</A>. 
		    The browser project is funded by grants from the 
		    <A HREF="http://www.genome.gov" 
		    TARGET="_blank">National Human Genome Research Institute</A>,
		    and generous support from the 
		    <A HREF="http://www.hhmi.org" 
		    TARGET="_blank">Howard Hughes Medical Institute</A> and the 
		    <A HREF="http://www.ucop.edu/california-institutes/" 
		    TARGET="_blank">California Institutes for Science and Innovation</A>.</P>
		    We work with collaborators at many other institutions
		    to produce and annotate this reference sequence of the human genome.
		    To view complete acknowledgments for a specific annotation, see the
		    &quot;Credits&quot; section on the track's description page 
		    in the Genome Browser. For a list of browser-related publications,
		    see our <A HREF="pubs.html">publications page</A>.
		    <P>
		    <B>UCSC Human Genome Browsers:</B> 
		    <UL>
		      <LI><a href="../../cgi-bin/hgGateway?db=hg38"
		    target="_blank">hg38</a> (GRCh38):
		      Hiram Clawson, Brian Raney, Robert Kuhn, Donna Karolchik, Steve Heitner
		      <LI><a href="../../cgi-bin/hgGateway?db=hg19"
		      target="_blank">hg19</a> (GRCh37): Hiram Clawson, Brooke Rhead, 
		      Pauline Fujita, Ann Zweig, Katrina Learned, Donna Karolchik and Robert Kuhn
		      <LI><a href="../../cgi-bin/hgGateway?db=hg18" target="_blank">hg18</a>:
		      Engineering effort led by Fan Hsu; QA effort led by Ann Zweig
		      <LI><a href="../../cgi-bin/hgGateway?db=hg17" target="_blank">hg17</a>:
		      Hiram Clawson, Terry Furey, Heather Trumbower, Robert Kuhn, Donna Karolchik,
		      Kate Rosenbloom, Angie Hinrichs, Rachel Harte, and Jim Kent
		      <LI><a href="../../cgi-bin/hgGateway?db=hg16" target="_blank">hg16</a>:
		      Terry Furey, Hiram Clawson, 
		      Heather Trumbower, Mark Diekhans, Robert Baertsch, Donna Karolchik, Jim Kent
		      <LI>hg15 (archived): Jim Kent, Terry Furey, Matt Schwartz, 
		      Heather Trumbower, Angie Hinrichs, Fan Hsu, Donna Karolchik, Jorge Garcia, 
		      Patrick Gavin, Chuck Sugnet, Yontao Lu, Mark Diekhans, Ryan Weber, 
		      Robert Baertsch, Krishna Roskin, and many other students in the UCSC 
		      Genome Bioinformatics group
		      <LI>Earlier browsers (archived): UCSC Genome Bioinformatics staff and
		      students
		    </UL>
		    </TD>
		    <TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Human Data Sources================================================== -->
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
	    <FONT SIZE="4"><B>&nbsp; Human Genome: Data Sources</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		Starting with the <a href="../../cgi-bin/agGateway?db=hg19" 
		target="_blank">hg19</a> (GRCh37) assembly, the human genome sequence is 
		provided by
		the <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/"
		TARGET="_blank">Genome Reference Consortium</A> (GRC), whose goal is to correct 
		the small number of regions in the reference that are currently misrepresented, 
		to close as many remaining gaps as possible and to produce alternative 
		assemblies of structurally variant loci when necessary. </P>
		<P>
		Prior to <a href="../../cgi-bin/hgGateway?db=hg19" target="_blank">hg19</a>, the
		human genome sequence data used by the browser were generated by laboratories 
		belonging to the <A HREF="labs.html">Human Genome Sequencing Consortium</A>. 
		Periodic freezes of this data were generated by 
		<A HREF="http://www.ncbi.nlm.nih.gov/" TARGET="_blank">NCBI</A>, and EST, mRNA, 
		BACend and other types of data used for the assembly and annotation were 
		obtained from <A HREF="http://www.ncbi.nlm.nih.gov/genbank/"
		TARGET="_blank">GenBank</A>, <A HREF="http://www.embl.org/"
		TARGET="_blank">European Molecular Biology Lab</A> (EMBL), and 
		<A HREF="http://www.ddbj.nig.ac.jp/" 
		TARGET="_blank">DNA Data Bank of Japan</A> (DDBJ).
		Clone maps were created by the individual sequencing centers. </P>
		<P>
		The original chromosome files were produced by the institutions and 
		individuals listed in the table below.</P>
		<P>

    <!-- finished chromosome table-->
		<TABLE BORDER=1 CELLPADDING=0 WIDTH="100%">
		    <TR><TD>
		    <TABLE BORDER=1 CELLPADDING=4 WIDTH="100%">
			<TR><TD COLSPAN=4><B>Finished Chromosomes</B></TD></TR>
			<TR><TD nowrap><FONT color="#006666"><B>Chr. #</B></FONT></TD>
			    <TD nowrap><FONT color="#006666"><B>Sequencing Centers</B></FONT></TD>
			    <TD nowrap><FONT color="#006666"><B>AGP File</B></FONT></TD></TR>
			<TR valign="top"><TD align="center">1</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Sanger Institute</A>, Cambridge, UK
				    <BR>	
				    <A HREF="http://www.gs.washington.edu/" 
				    TARGET="_blank">University of Washington, Department of Genome Sciences</A>, 
				    Seattle, WA, USA
			    </TD>
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top">
			    <TD align="center">2</TD>
			    <TD>
				    <A HREF="http://genome.wustl.edu/genomes/detail/homo-sapiens/"
				    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>, 
				    St. Louis, MO, USA
				</TD> 
			    <TD nowrap>Rick Wilson</TD>
			</TR>
			<TR valign="top">
			    <TD align="center">3</TD>
			    <TD>
				    <A HREF="https://www.hgsc.bcm.edu/" 
				    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>, 
				    Houston, TX, USA
				    <BR>
				    <A HREF="http://www.genomics.cn/en/index"
				    TARGET="_blank">Beijing Genomics Institute/Human Genome Center</A>, 
				    Beijing, China 
				    <BR>
				    <A HREF="http://www.gs.washington.edu/" 
				    TARGET="_blank">University of Washington, Department of Genome Sciences</A>, 
				    Seattle, WA, USA
			    </TD> 
			    <TD>Steve Scherer</TD>
			</TR>
			<TR valign="top">
			    <TD align="center">4</TD>
			    <TD>
				    <A HREF="http://genome.wustl.edu/genomes/detail/homo-sapiens/"
				    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>, 
				    St. Louis, MO, USA
				    <BR>
				    Stanford Human Genome Center, Palo Alto, CA, USA
			    </TD>
			    <TD nowrap>Rick Wilson</TD>
			</TR>
			<TR valign="top">
			    <TD align="center">5</TD>
			    <TD>
				    <A HREF="http://genome.jgi-psf.org/chr5/chr5.home.html" 
				    TARGET="_blank"> Joint Genome Institute</A>, 
				    U.S. Department of Energy, Walnut Creek, CA, USA
				    <BR>
				    Stanford Human Genome Center, Palo Alto, CA, USA
			    </TD>
			    <TD>Jeremy Schmutz</TD>
			</TR>
			<TR valign="top"><TD align="center">6</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
			    </TD> 
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top"><TD align="center">7</TD>
			    <TD>
				    <A HREF="http://genome.wustl.edu/genomes/detail/homo-sapiens/"
				    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>, 
				    St. Louis, MO USA
				    <BR>
				    <A HREF="http://www.gs.washington.edu/" 
				    TARGET="_blank">University of Washington, Department of Genome Sciences</A>, 
				    Seattle, WA, USA
			    </TD> 
			    <TD nowrap>Rick Wilson</TD>
			</TR>
			<TR valign="top"><TD align="center">8</TD>
			    <TD>
				    <A HREF="http://www.broadinstitute.org/"
				    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
				    <BR>
				    <A HREF="http://www.keio.ac.jp" 
				    TARGET="_blank">Keio University</A>, Tokyo, Japan
				    <BR>
				    <A HREF="http://genome.imb-jena.de/" 
				    TARGET="_blank">Institute of Molecular Biology (IMB)</A>, 
				    Jena, Germany
			    </TD> 
			    <TD>Chad Nusbaum</TD>
			</TR>
			<TR valign="top"><TD align="center">9</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
				    <BR>
				    <A HREF="http://www.helmholtz-hzi.de/en/" 
				    TARGET="_blank">Gesellschaft fuer Biotechnologische Forschung (GBF)</A>, 
				    Braunschweig, Germany 
			    </TD>
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top"><TD align="center">10</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
				    <BR>
				    Genome Therapeutics Corporation (GTC), Waltham, MA, USA
			    </TD> 
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top"><TD align="center">11</TD>
			    <TD>
				    <A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
				    TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
				    <BR>
				    <A HREF="http://www.broadinstitute.org/"
				    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
			    </TD>
			    <TD>Todd Taylor</TD>
			</TR>
			<TR valign="top"><TD align="center">12</TD>
			    <TD>
				    <A HREF="https://www.hgsc.bcm.edu/" 
				    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>, 
				    Houston, TX, USA
			    </TD>
			    <TD>Steve Scherer</TD>
			</TR>
			<TR valign="top"><TD align="center">13</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
			    </TD> 
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top"><TD align="center">14</TD>
			    <TD>
				    <A HREF="http://www.genoscope.cns.fr/spip/spip.php?lang=en" 
				    TARGET="_blank">Genoscope National Sequencing Centre</A>, 
				    Evry, France
				    <BR>
				    <A HREF="http://www.systemsbiology.org/" 
				    TARGET="_blank">Institute for Systems Biology (ISB)</A>, 
				    Seattle, WA
				    <BR>
				    <A HREF="http://genome.wustl.edu/genomes/detail/homo-sapiens/"
				    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>, 
				    St. Louis, MO USA
				    <BR>
				    <A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
				    TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
			    </TD> 
			    <TD>Jean Weissenbach</TD>
			</TR>
			<TR valign="top"><TD align="center">15</TD>
			    <TD>
				    <A HREF="http://www.broadinstitute.org/"
				    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
				    <BR>
				    <A HREF="http://genome.imb-jena.de/" 
				    TARGET="_blank">Institute of Molecular Biology (IMB)</A>, 
				    <BR>
				    <A HREF="http://www.systemsbiology.org/" 
				    TARGET="_blank">Institute for Systems Biology (ISB)</A>, 
				    Seattle, WA
			    </TD> 
			    <TD>Chad Nusbaum</TD>
			</TR>
			<TR valign="top"><TD align="center">16</TD>
			    <TD>
				    <A HREF="http://jgi.doe.gov/" 
				    TARGET="_blank">Joint Genome Institute (JGI)</A>, 
				    U.S. Department of Energy, Walnut Creek, CA, USA
				    <BR>
				    Stanford Human Genome Center, Palo Alto, CA, USA
				    <BR>
				    <A HREF="http://www.jcvi.org/cms/home/"
				    TARGET="_blank">The Institute for Genomic Research (TIGR)</A>, 
				    Rockville, MD, USA
				    <BR>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
			    </TD>
			    <TD>Jeremy Schmutz</TD>
			</TR>
			<TR valign="top"><TD align="center">17</TD>
			    <TD>
				    <A HREF="http://www.broadinstitute.org/"
				    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
			    </TD>
			    <TD>Chad Nusbaum</TD>
			</TR>
			<TR valign="top"><TD align="center">18</TD>
			    <TD>
				    <A HREF="http://www.broadinstitute.org/"
				    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
				    <BR>
				    <A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
				    TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
			    </TD>
			    <TD>Chad Nusbaum</TD>
			</TR>
			<TR valign="top"><TD align="center">19</TD>
			    <TD>
				    <A HREF="http://genome.jgi-psf.org/Chr19ncbi34/Chr19ncbi34.home.html" 
				    TARGET="_blank"> Joint Genome Institute (JGI)</A>, 
				    U.S. Department of Energy, Walnut Creek, CA, USA		
				    <BR>
				    Stanford Human Genome Center, Palo Alto, CA, USA
			    </TD> 
			    <TD>Jeremy Schmutz</TD>
			</TR>
			<TR valign="top"><TD align="center">20</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
			    </TD>
			    <TD>James Gilbert</TD>
			 </TR>
			 <TR valign="top"><TD align="center">21</TD>
			    <TD>
				    <A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
				    TARGET="_blank">RIKEN Human Genome Research Group</A>, Japan
				    <BR>
				    <A HREF="http://www.mpg.de/en" 
				    TARGET="_blank">Max Planck Institute for Molecular Genetics</A>, 
				    Berlin, Germany  
				    <BR>
				    <A HREF="http://www.helmholtz-hzi.de/en/" 
				    TARGET="_blank">Gesellschaft fuer Biotechnologische Forschung (GBF)</A>, 
				    Braunschweig, Germany 
				    <BR>
				    <A HREF="http://www.med.keio.ac.jp/index-e.html" 
				    TARGET="_blank">Keio University School of Medicine</A>, 
				    Tokyo, Japan
				    <BR>
				    <A HREF="http://chr21.molgen.mpg.de/index.html" 
				    TARGET="_blank">Chromosome 21 Consortium</A>
			    </TD>
			    <TD nowrap>Todd Taylor</TD>
			</TR>
			<TR valign="top"><TD align="center">22</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
				    <BR>
				    <A HREF="http://www.genome.ou.edu/" 
				    TARGET="_blank">The University of Oklahoma</A>, 
				    Norman, OK USA
				    <BR>
				    <A HREF="http://www.med.keio.ac.jp/index-e.html" 
				    TARGET="_blank">Keio University School of Medicine</A>, 
				    Tokyo, Japan
			    </TD> 
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top"><TD align="center">X</TD>
			    <TD>
				    <A HREF="http://www.sanger.ac.uk/" 
				    TARGET="_blank">The Wellcome Trust Sanger Institute</A>, 
				    Cambridge, UK
				    <BR>
				    <A HREF="https://www.hgsc.bcm.edu/" 
				    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>, 
				    Houston, TX, USA
				    <BR>
				    <A HREF="http://genome.imb-jena.de/" 
				    TARGET="_blank">Institute of Molecular Biology (IMB)</A>, 
				    Jena, Germany
				    <BR>
				    <A HREF="http://genome.wustl.edu/genomes/detail/homo-sapiens/"
				    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>, 
				    St. Louis, MO USA
				    <BR>
				    <A HREF="http://www.mpg.de/en" 
				    TARGET="_blank">Max Planck Institute for Molecular Genetics</A>, 
				    Berlin, Germany  
			    </TD>
			    <TD>Jane Rogers</TD>
			</TR>
			<TR valign="top"><TD align="center">Y</TD>
			    <TD>
				    <A HREF="http://genome.wustl.edu/genomes/detail/homo-sapiens/"
				    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
				    St. Louis, MO USA
			    </TD>
			    <TD nowrap>Rick Wilson</TD>
			</TR>
		    </TABLE>
		    </TD></TR>
		</TABLE>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    <BR>
    </TD></TR>
</TABLE>
<BR>

<!-- Alpaca Credits ======================================================= -->
<A NAME="alpaca_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Alpaca Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Alpaca Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Alpaca Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI> <a href="../../cgi-bin/hgGateway?db=vicPac2" target="_blank">vicPac2</a>:
		    Hiram Clawson and Luvina Guruvadoo
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=vicPac1" target="_blank">vicPac1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="alpaca_use"></A>
		The alpaca sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The alpaca data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Armadillo Credits ======================================================= -->
<A NAME="armadillo_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Armadillo Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Armadillo Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Armadillo Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI><a href="../../cgi-bin/hgGateway?db=dasNov3" target="_blank">dasNov3</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="armadillo_use"></A>
		The armadillo sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The armadillo data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Baboon Credits ======================================================= -->
<A NAME="baboon_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Baboon Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Baboon Genome Browser is provided by the Baboon Genome Sequencing Consortium
		with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Funding: </B>
		  <A HREF="http://www.genome.gov/"
		  TARGET="_blank">National Human Genome Research Institute</A>
		  (NHGRI), USA
		  <LI>
		  <B>Sequencing/Assembly: </B>
		  <A HREF="https://www.hgsc.bcm.edu/non-human-primates/baboon-genome-project"
		  TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>,
		  Houston, TX, USA
		  <LI>
		  <B>UCSC Baboon Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=papAnu2" target="_blank">papAnu2</a>:
		    Chin Li and Luvina Guruvadoo
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=papHam1" target="_blank">papHam1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="baboon_use"></A>
		The baboon sequence is made freely available for public use by the
		Baboon Genome Sequencing Consortium. Please review the BCM-HGSC
		<A HREF="https://www.hgsc.bcm.edu/bcm-hgsc-conditions-use"
		TARGET="_blank">conditions for use</A> guidelines before using this data.
		</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Bonobo Credits ======================================================= -->

<A NAME="bonobo_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Bonobo Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Bonobo Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Funding: </B>
		  <A HREF="http://erc.europa.eu/" TARGET="_blank">European Research Council</A>
      (grant 233297, TWOPAN) and the
      <a href="http://www.mpg.de/" target="_blank">Max Planck Society</a>
		  <LI>
		  <B>Sequencing/Assembly: </B>
		  <A HREF="http://www.eva.mpg.de/bonobo-genome/index.html"
		  TARGET="_blank">Max-Planck Institute for Evolutionary Anthropology</A>,
		  Germany
		  <LI>
		  <B>UCSC Baboon Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=panPan1" target="_blank">panPan1</a>:
		    Chin Li, Hiram Clawson, and Jonathan Casper
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="bonobo_use"></A>
		The bonobo sequence is made freely available by the
		Max-Planck Institute for Evolutionary Anthropology.
		</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>
<!-- Bushbaby Credits ======================================================= -->
<A NAME="bushbaby_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Bushbaby Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Bushbaby Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B>
		  <A HREF="http://www.genome.gov"
		  TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Bushbaby Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=otoGar3" target="_blank">otoGar3</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="bushbaby_use"></A>
		The bushbaby sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The bushbaby data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Cat Credits ====================================================== -->
<A NAME="cat_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Cat Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC <a href="../../cgi-bin/hgGateway?db=felCat5" 
		target="_blank">felCat5</a> Genome Browser displays data from the
		Sep. 2011 Felis_catus-6.2 assembly of the domestic cat produced by the
		International Cat Genome Sequencing Consortium.  This assembly was provided
		with the following acknowledgments:
		<UL>
		    <LI><B>Funding:</B> <A HREF="http://www.genome.gov/"
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI),
		    Bethesda, MD
		    <BR>
		    <LI><B>454 Production sequencing: </B>
		    <A HREF="http://genome.wustl.edu/genomes/detail/felis-catus/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		    St. Louis, MO, USA
		    <LI><B>BAC sequencing:</B> The Genome Institute at WUSTL
		    <LI><B>Plasmid and fosmid sequencing:</B>
		    <A HREF="http://www.broadinstitute.org/" TARGET="_blank">The Broad Institute</A>,
		    Cambridge, MA, USA and <A HREF="http://www.beckmangenomics.com/"
		    TARGET="_blank">Agencourt Bioscience</A>
		    <LI><B>Sequence assembly and data integration for creation of chromosomal
		    AGP files:</B> The Genome Institute at WUSTL
		    <LI><B>Whole genome alignments:</B> Brian Raney,
		    <A HREF="../index.html">UCSC Genome Browser Group</A>, University of
		    California, Santa Cruz, CA, and The Genome Institute at WUSTL
		    <LI><B>Cat RH map:</B> Bill Murphy, <A HREF="http://genetics.tamu.edu/"
		    TARGET=_blank>Texas A&M University</A>, College Station, TX.
		    <LI><B>DNA source:</B> Dr. Kristina Narfstrom,
		    <A HREF="http://www.vmth.missouri.edu/"
		    TARGET=_blank>University of Missouri</A>, Columbia, MO.
		    <LI><B>UCSC Cat Genome Browser/Initial Annotations (<a href="../../cgi-bin/hgGateway?db=felCat5" target="_blank">felCat5</a>):</B>
		    <A HREF="../staff.html"
		    TARGET="_blank">UCSC Genome Bioinformatics Group</A>, Santa Cruz, CA, USA -
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina Guruvadoo,
		    Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		</UL>
		<P>
		The UCSC <a href="../../cgi-bin/hgGateway?db=felCat4" target="_blank">felCat4</a> Genome Browser displays data from the
		Dec. 2008 catChrV17e draft assembly, based on sequence
		reads from six domestic cats and one wild cat. The
		browser is provided  with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly:</B>
		    <UL>
		      <LI>
			<A HREF="http://www.nisc.nih.gov/"
			TARGET="_blank">NHGRI Genome Technology Branch and NIH
			Intramural Sequencing Center</A>, Bethesda,
			MD, USA
		      <LI>
			<A HREF="http://www.beckmangenomics.com/"
			TARGET="_blank">Agencourt Bioscience</A>,
			Danvers, MA, USA
		      <LI>
			<A HREF="http://www.cancer.gov/"
			TARGET="_blank">NCI Laboratory of Genomic Diversity</A>,
			Frederick, MD, USA
		      <LI>
			<A HREF="http://www.hillspet.com"
			TARGET="_blank">Hill's Pet Nutrition, Inc.</A>,
			Topeka, KS, USA
		    </UL>
		   <LI><B>UCSC Cat Genome Browser/Initial Annotations
		   (<a href="../../cgi-bin/hgGateway?db=felCat4" target="_blank">felCat4</a>):</B>
		     <A HREF="../staff.html"
		     TARGET="_blank">UCSC Genome Bioinformatics Group</A>, Santa Cruz, CA, USA -
		     Chin Li, Hiram Clawson, Antonio Coelho, Mary
		     Goldman, and Donna Karolchik
		</UL>
		<P>
		The UCSC <a href="../../cgi-bin/hgGateway?db=felCat3" target="_blank">felCat3</a> Genome Browser displays data from the
		March 2006 v3 draft assembly of the domestic cat (<em>Felis catus</em>).
		This assembly was provided with the following acknowledgments:
		<UL>
		    <LI><B>Funding:</B> <A HREF="http://www.genome.gov/"
		    TARGET="_blank">National Human Genome Research Institute</A> (NHGRI),
		    Bethesda, MD
		    <BR>
		    <LI><B>Project Leadership:</B>
		    <A HREF="http://www.broadinstitute.org/science/projects/mammals-models/data-release-mammalian-genome-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI><B>Sequencing/Assembly:</B>
		    The Broad Institute - team led by by Kerstin Lindblad-Toh, Ph.D.
		    <LI><B>Plasmid and fosmid libraries:</B> The Broad Institute
		    <LI><B>BAC end sequencing:</B>
		    <A HREF="http://www.beckmangenomics.com/"
		    TARGET="_blank">Agencourt Bioscience</A>, Danvers, MA, USA
		    <BR>
		    <LI><B>UCSC Cat Genome Browser/Initial Annotations (<a href="../../cgi-bin/hgGateway?db=felCat3" target="_blank">felCat3</a>):</B>
		    <A HREF="../staff.html"
		    TARGET="_blank">UCSC Genome Bioinformatics Group</A>, Santa Cruz, CA, USA -
		    Heather Trumbower, Angie Hinrichs, Mark Diekhans,
		    Brooke Rhead, and Archana Thakkapallayil
		</UL>
		<P>
		<HR>
		<A NAME="cat_use"></A>
		The catChrV17e cat assembly was initially published in
		Pontius JU <em>et al</em>.
		<A HREF="http://genome.cshlp.org/content/17/11/1675.long"
		TARGET=_blank>Initial sequence and comparative analysis of the cat genome</A>.
		<em>Genome Res</em>. 2007 Nov;17(11):1675-89. PMID 17975172; PMCID: PMC2045150
		<P>
		The Broad Institute released the initial 2X (v3) assembly of the cat genome as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET="_blank">A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357
		<P>
		The cat sequence is made freely available with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Chimp Credits ====================================================== -->
<A NAME="chimp_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Chimpanzee Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC Chimpanzee Genome Browsers display draft assembly 
		data produced by the Chimpanzee Genome Sequencing 
		Consortium: 
		<UL>
		  <LI>Feb. 2011 <a href="../../cgi-bin/hgGateway?db=panTro4" 
		  target="_blank">panTro4</a> browser: v2.1.4 
		  chromosome-based 6X draft assembly 
		  <LI>Oct. 2010 <a href="../../cgi-bin/hgGateway?db=panTro3" 
		  target="_blank">panTro3</a> browser: v2.1.3 
		  chromosome-based 6X draft assembly 
		  <LI>Mar. 2006 <a href="../../cgi-bin/hgGateway?db=panTro2" 
		  target="_blank">panTro2</a> browser: v2.1 chromosome-based
		  6X draft assembly (Oct. 2005) 
		  <LI>Nov. 2003 <a href="../../cgi-bin/hgGateway?db=panTro1" 
		  target="_blank">panTro1</a> browser: v1.1 Arachne draft assembly 
		</UL>
		<P>
		The Chimpanzee Genome Browsers have the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B> <A HREF="http://www.genome.gov/"
		  TARGET="_blank">National Human Genome Research Institute</A> 
		  (NHGRI)
		  <BR>
		  <LI><B>Sequencing/Assembly (panTro2, panTro3, panTro4):</B>
		  <UL>
		    <LI>
		    Whole genome shotgun (WGS) sequence data assembly and
		    organization:
		    <A HREF="http://genome.wustl.edu/genomes/detail/pan-troglodytes/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		    St. Louis, MO, USA
		    <LI>
		    Underlying WGS data: The Genome Institute at WUSTL and
		    <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/chimpanzee/chimpanzee-genome-project" 
		    TARGET =_blank>The Broad Institute</A>, Cambridge, MA USA
		    <LI>
		    5-Mb region of chromosome 7 at chr7:84674857-89461887): finished by The
		    Genome Institute at WUSTL
		    <LI>
		    Chromosome Y sequence: finished by The Genome Institute at WUSTL,
		    with detailed mapping and extensive collaboration with 
		    David Page's group at the <A HREF="http://wi.mit.edu/"
		    TARGET="_blank">Whitehead Institute</A>. For more information, see: 
		    <UL>
		      <LI>
		      Hughes JF <em>et al</em>. 
		      <A HREF="http://www.nature.com/nature/journal/v437/n7055/abs/nature04101.html"
		      TARGET="_blank">Conservation of Y-linked genes during human 
		      evolution revealed by comparative sequencing in chimpanzee</A>. 
		      <em>Nature</em>. 2005 Sep 1;437(7055):100-3. PMID: 16136134
		      <LI>
		      &quot;The DNA Sequence of Chimpanzee Chromosome Y&quot;, 
		      unpublished.
		    </UL>
		    <LI>
		    Chromosome 21 sequence: Todd Taylor and the 
		    <A HREF="http://www.gsc.riken.go.jp/eng/index.html" 
		    TARGET="_blank">RIKEN Genomic Sciences Center</A>. 
		    See Watanabe <em>et al.</em> 
		    <A HREF="http://www.nature.com/nature/journal/v429/n6990/abs/nature02564.html"
		    TARGET="_blank">DNA sequence and comparative analysis of 
		    chimpanzee chromosome 22</A>. 
		    <em>Nature</em>. 2004 May 27;429(6990):382-8. PMID: 15164055
		  </UL>
		<LI><B>Sequencing/Assembly (panTro1):</B>
		Sequence assembled by teams led by Eric 
		Lander, Ph.D. at The Broad Institute and Richard K. Wilson, 
		Ph.D. at The Genome Institute at WUSTL
		<BR>
		<LI><B>Alignments:</B> 
		LaDeana Hillier, The Genome Institute at WUSTL and The Broad Institute
		<BR>
		<LI><B>Annotations:</B> 
		<UL>
		  <LI>
		  The Genome Institute at WUSTL, St. Louis, MO, USA
		  <LI>
		  <A HREF="http://www.imim.cat/programesrecerca/informaticabiomedica/en_index.html"
		  TARGET="_blank">Research Group on Biomedical Informatics of IMIM/UPF</A> (GRIB),
		  Barcelona, Spain
		  <LI>
		  Riken Genomic Sciences Center, Yokohama City, Japan 
		  <LI>
		  <A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>, 
		  Santa Cruz, CA, USA
		</UL>
		<LI><B>UCSC Chimp Genome Browser:</B> 
		<UL>
		  <LI>
		  <a href="../../cgi-bin/hgGateway?db=panTro4" target="_blank">panTro4</a>:
		  Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		  Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  <LI>
		  <a href="../../cgi-bin/hgGateway?db=panTro3" target="_blank">panTro3</a>:
		  Hiram Clawson, Greg Roe, and Donna Karolchik
		  <LI>
		  <a href="../../cgi-bin/hgGateway?db=panTro2" target="_blank">panTro2</a>:
		  Kate Rosenbloom, Brian Raney, Hiram Clawson, Ann Zweig, Archana 
		  Thakkapallayil, and Donna Karolchik 
		  <LI>
		  <a href="../../cgi-bin/hgGateway?db=panTro1" target="_blank">panTro1</a>:
		  Kate Rosenbloom, Jim Kent, Hiram Clawson, Heather Trumbower, Robert Kuhn, and
		  Donna Karolchik
		</UL>
	      </UL>
		<P>
		<HR>
		<A NAME="chimp_use"></A>
		The initial sequencing and analysis of the chimpanzee genome
		was published by the Chimpanzee Sequencing and Analysis Consortium. 
		The chimpanzee sequence data may be freely downloaded, used in analyses, and 
		repackaged in databases if the provider is properly acknowledged. Please cite 
		this publication when using these data:</P>
		<P>
		The Chimpanzee Sequencing and Analysis Consortium.
		<a href="http://www.nature.com/nature/journal/v437/n7055/abs/nature04072.html"
		target="_blank">Initial sequence of the chimpanzee genome and comparison
		with the human genome</a>. 
		<em>Nature</em>. 2005 Sep 1;437(7055):69-87. PMID: 16136131 </P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Chinese Hamster Credits ====================================================== -->
<A NAME="chinese_hamster_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Chinese Hamster Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Chinese hamster assembly was provided by the Beijing
		Genomics Institution-Shenzhen with the following
		acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly:</B>
		  <a href="http://www.genomics.cn/en/index"
		  target="_blank">Beijing Genomics Institution-Shenzhen</a>
		  <LI>
		  <B>UCSC Chinese Hamster Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=criGri1" target="_blank">criGri1</a>:
		    Hiram Clawson, Brian Raney, and Steve Heitner
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="chinese_hamster_use"></A>
		The Chinese hamster sequence data may be freely downloaded, used in analyses, 
		and repackaged in databases if the provider is properly acknowledged. Please 
		cite this publication when using these data:</P>
		<P>
		Lewis NE, Liu X, Li Y, Nagarajan H, Yerganian G, O'Brien E, Bordbar A, Roth AM, 
		Rosenbloom J, Bian C <em>et al.</em>
		<a href="http://www.nature.com/nbt/journal/v31/n8/full/nbt.2624.html"
		target="_blank">Genomic landscapes of Chinese hamster ovary cell lines as 
		revealed by the Cricetulus griseus draft genome</a>.
		<em>Nat Biotechnol</em>. 2013 Aug;31(8):759-65. </P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Cow Credits====================================================== -->
<A NAME="cow_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Cow Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC Cow Genome Browser displays two different
		assemblies of the bovine genome: one provided by the
		University of Maryland Center for Bioinformatics and
		Computational Biology (UMD CBCB) and one from the Baylor
		College of Medicine Human Genome Sequencing Center (BCM-HGSC).
		The assemblies available in the browser include:
		<UL>
		  <LI>Jun. 2014 (<a href="../../cgi-bin/hgGateway?db=bosTau6" 
		  target="_blank">bosTau8</a>) assembly - UMD CBCB version UMD_3.1.1
		  <LI>Oct. 2011 (<a href="../../cgi-bin/hgGateway?db=bosTau7" 
		  target="_blank">bosTau7</a>) assembly - Bovine Genome Sequencing Consortium 
		  (BCM-HGSC) version Btau_4.6.1
		  <LI>Nov. 2009 (<a href="../../cgi-bin/hgGateway?db=bosTau6" 
		  target="_blank">bosTau6</a>) assembly - UMD CBCB version UMD_3.1
		  <LI>Oct. 2007 (<a href="../../cgi-bin/hgGateway?db=bosTau4" 
		  target="_blank">bosTau4</a>) assembly - BCM-HGSC version Btau_4.0
		  <LI>Aug. 2006 (<a href="../../cgi-bin/hgGateway?db=bosTau3" 
		  target="_blank">bosTau3</a>) assembly - BCM-HGSC version Btau_3.1
		  <LI>Mar. 2005 (<a href="../../cgi-bin/hgGateway?db=bosTau2" 
		  target="_blank">bosTau2</a>) assembly - BCM-HGSC version Btau_2.0
		  <LI>Sep. 2004 (bosTau1) assembly - BCM-HGSC version Btau_1.0
		</UL></P>
		<P>
		The UMD CBCB genome assemblies are provided with the following acknowledgments:
		<UL>
		  <LI><B>Genome sequence:</B> NCBI Trace Archive
		  <LI><B>Genome assembly:</B> <A HREF="http://www.cbcb.umd.edu/"
		  TARGET="_blank">University of Maryland Center for Bioinformatics and 
		  Computational Biology</A>, College Park, MD, USA
		  <LI><B>UCSC Cow Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=bosTau8" target="_blank">bosTau8</a>: 
		    Steve Heitner and Luvina Guruvadoo
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=bosTau6" target="_blank">bosTau6</a>: 
		    Hiram Clawson, Chin Li, Brooke Rhead, Greg Roe, Steve Heitner, and Donna 
		    Karolchik
		  </UL>
		</UL>
		<P>
		The BCM-HGSC genome assemblies are provided with the following acknowledgments:
		<UL>
		  <LI><B>Funding</B> - For a list of the many agencies that
		  funded the cow sequencing project, see the BCM-HGSC
		  <A HREF="https://www.hgsc.bcm.edu/other-mammals/bovine-genome-project"
		  TARGET="_blank">Bovine Genome Project</A> page.
		  <LI><B>BAC library DNA: </B>
		    Dr. Michael MacNeil's laboratory at the
		    <A HREF="http://www.ars.usda.gov/main/main.htm"
		    TARGET="_blank">USDA Agricultural Research Service</A>,
		    Miles City, MT, USA
		  <LI><B>Whole genome shotgun sequence DNA: </B>
		    Dr. Timothy Smith's laboratory at the
		    <A HREF="http://www.ars.usda.gov/Main/docs.htm?docid=2340"
		    TARGET="_blank">U.S. Meat Animal Research Center</A>,
		    Clay Center, NE, USA
		  <LI><B>BAC-based fingerprint map: </B>
		    <A HREF="http://www.bcgsc.ca/"
		    TARGET=-blank>Genome Sciences Centre</A>, Vancouver,
		    B.C.
		  <LI><B>UCSC Cow Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI><a href="../../cgi-bin/hgGateway?db=bosTau7" target="_blank">bosTau7</a>:
		    Chin Li, Hiram Clawson, Brian Raney and Steve Heitner
		    <LI><a href="../../cgi-bin/hgGateway?db=bosTau4" target="_blank">bosTau4</a>: 
		    Hiram Clawson, Brian Raney, and Ann Zweig
		    <LI><a href="../../cgi-bin/hgGateway?db=bosTau3" target="_blank">bosTau3</a>: 
		    Heather Trumbower, Angie Hinrichs, Kayla Smith and Donna Karolchik
		    <LI><a href="../../cgi-bin/hgGateway?db=bosTau2" target="_blank">bosTau2</a>: 
		    Galt Barber, Brian Raney, Mark Diekhans, Jennifer Jackson, and Donna 
		    Karolchik
		    <LI>bosTau1 (archived):
		    Heather Trumbower, Jim Kent, Hiram Clawson, Angie Hinrichs, Brian Raney, 
		    Mark Diekhans, Robert Kuhn, Ali Sultan-Qurraie and Donna Karolchik
		  </UL>
		</UL>
		<HR>
		<A NAME="cow_use"></A>
		The cow sequence data may be freely downloaded, used in analyses, and 
		repackaged in databases if the provider is properly acknowledged. Please cite 
		the appropriate publication when using these data.</P>
		<P>
		<B>UMD cow assembly:</B></P>
		<P>
		Zimin AV, Delcher AL, Florea L, Kelley DR, Schatz MC, Puiu D, Hanrahan F, 
		Pertea G, Van Tassell CP, Sonstegard TS <em>et al</em>.
		<a href="http://genomebiology.com/content/10/4/R42"
		target="_blank">A whole-genome assembly of the domestic cow, Bos taurus</a>.
		<em>Genome Biol</em>. 2009;10(4):R42. </P>
		<P>
		<B>Baylor cow assembly:</B></P>
		<P>
		Bovine Genome Sequencing and Analysis Consortium. 
		<a href="http://www.sciencemag.org/content/324/5926/522.abs"
		target="_blank">The genome sequence of taurine cattle: a window to ruminant 
		biology and evolution</a>.
		<em>Science</em>. 2009 Apr 24;324(5926):522-8. </P>
		For more information see the 
		<a href="http://www.cbcb.umd.edu/" target="_blank">UMD CBCB</a>
		<!--Commenting out until link is fixed
		<A HREF="http://www.cbcb.umd.edu/research/bos_taurus_assembly.shtml"
		TARGET="_blank">Bos taurus assembly</A> page--> and the BCM-HGSC
		<A HREF="https://www.hgsc.bcm.edu/other-mammals/bovine-genome-project"
		TARGET="_blank">Bovine Genome Project</A> pages.
		</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Dog Credits ====================================================== -->
<A NAME="dog_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Dog Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC Dog Genome Browser displays assemblies of the domestic dog 
		(<em>Canis familiaris</em>) with the following acknowledgments:
		<UL>
		    <LI><B>Funding:</B> The
		    <A HREF="http://www.genome.gov/"
		    TARGET="_blank">National Human Genome Research Institute</A>
		    <BR>
		    <LI><B>Project Leadership:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI><B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A> -
		    team led by by Kerstin Lindblad-Toh, Ph.D.
		    <LI><B>Plasmid and fosmid libraries:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>
		    <LI><B>BAC libraries:</B>
		    <A HREF="http://www.chori.org/"
		    TARGET="_blank">Children's Hospital Oakland Research Institute</A>
		    (CHORI), Oakland, CA, USA
		    <LI><B>BAC end sequencing:</B>
		    <A HREF="http://www.beckmangenomics.com/"
		    TARGET="_blank">Agencourt Bioscience</A>, Danvers, MA, USA
		    <LI><B>RH map:</B>
		    <A HREF="http://www.cnrs.fr/index.php"
		    TARGET="_blank">Centre National de la Recherche Scientifique</A>
		    (CNRS), France
		    <LI><B>FISH mapping:</B> <A HREF="https://www.ncsu.edu/"
		    TARGET="_blank">North Carolina State University</A> (NCSU),
		    Raleigh, NC, USA
		    <BR>
		    <LI><B>Dog Selection:</B>
		    <A HREF="http://www.fredhutch.org/en/labs.html"
		    TARGET="_blank">Fred Hutchinson Cancer Research Center</A>
		    (FHCRC), Seattle, WA, USA
		    <LI><B>UCSC Dog Genome Browser/Initial Annotations:</B>
		    <UL>
		      <LI><a href="../../cgi-bin/hgGateway?db=canFam3" 
		      target="_blank">canFam3</a>: Hiram Clawson and Luvina Guruvadoo
		      <LI><a href="../../cgi-bin/hgGateway?db=canFam2" 
		      target="_blank">canFam2</a>: Angie Hinrichs, Jennifer Jackson, and Donna 
		      Karolchik
		      <LI><a href="../../cgi-bin/hgGateway?db=canFam1" 
		      target="_blank">canFam1</a>: Angie Hinrichs, Robert Kuhn and Donna 
		      Karolchik
		    </UL>
		</UL>
		<P>
		<HR>
		<A NAME="dog_use"></A>
		The dog genome sequence is made freely available by the
		<A HREF="http://www.broadinstitute.org/mammals/dog"
		TARGET="_blank">Dog Genome Sequencing Project</A>.
		Please cite this publication when using these data:</P>
		<P>
		Lindblad-Toh K, Wade CM, Mikkelsen TS, Karlsson EK, Jaffe DB, Kamal M,
		Clamp M, Chang JL, Kulbokas EJ 3rd, Zody MC <em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v438/n7069/abs/nature04338.html"
		target="_blank">Genome sequence, comparative analysis and haplotype structure
		of the domestic dog</a>.
		<em>Nature</em>. 2005 Dec 8;438(7069):803-19. PMID: 16341006</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Dolphin Credits ================================================= -->
<A NAME="dolphin_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Dolphin Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Dolphin Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing: </B>
		  <A HREF="https://www.hgsc.bcm.edu/marine-mammals/dolphin-genome-project"
		  TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center 
		  (BCM-HGSC)</A>, Houston, TX, USA
		  <LI>
		  <B>Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Dolphin Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI><a href="../../cgi-bin/hgGateway?db=turTru2" 
		    target="_blank">turTru2</a>: Hiram Clawson, Chin Li, Brian Raney, Pauline 
		    Fujita, Luvina Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna 
		    Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="dolphin_use"></A>
		The dolphin sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET="_blank">A high-resolution map of human evolutionary constraint using
		29 mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The dolphin data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Elephant Credits ====================================================== -->
<A NAME="elephant_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Elephant Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The elephant assembly was provided by the Broad Institute with the following 
		acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly:</B>
		  <A HREF="http://www.broadinstitute.org/"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Elephant Genome Browser/Initial Annotations:</B> 
		    <UL>
		      <LI><a href="../../cgi-bin/hgGateway?db=loxAfr3" 
		      target="_blank">loxAfr3</a>: Hiram Clawson, Pauline Fujita, Vanessa Swing,
		      Antonio Coelho, and Donna Karolchik
		    </UL>
		</UL>
		<P>
		<HR>
		<A NAME="elephant_use"></A>
		The elephant sequence is made freely available before
		scientific publication with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The center producing the data reserves the right to
		    publish the initial large-scale analyses of the data
		    set, including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more).
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Ferret Credits ====================================================== -->
<A NAME="ferret_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Ferret Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The ferret assembly was provided by the Ferret
		Genome Sequencing Consortium  with the following acknowledgments:
		<UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    Ferret Genome Sequencing Consortium and
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI>
		    <B>UCSC Ferret Genome Browser/Initial Annotations:</B> 
		    <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=musFur1" 
			target="_blank">musFur1</a>: Hiram Clawson, Brian Lee, and Brooke Rhead
		    </UL>
		</UL>
		<P>
		<HR>
		<A NAME="ferret_use"></A>
		The ferret sequence data may be freely downloaded, used in analyses, and 
		repackaged in databases if the provider is properly acknowledged. Please cite 
		this publication when using these data:</P>
		<P>
		Peng X, Alf&ouml;ldi J, Gori K, Eisfeld AJ, Tyler SR, Tisoncik-Go J, Brawand D, 
		Law GL, Skunca N, Hatta M <em>et al</em>.
		<a href="http://www.nature.com/nbt/journal/v32/n12/full/nbt.3079.html" 
		target="_blank">The draft genome sequence of the ferret (Mustela putorius furo) 
		facilitates study of human respiratory disease</A>.
		<em>Nat Biotechnol</em>. 2014 Dec;32(12):1250-5. </P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Gibbon Credits=================================================== -->
<A NAME="gibbon_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Gibbon Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC Gibbon Genome Browser displays data from the
		draft assemblies of the northern white-cheeked gibbon
		(<em>Nomascus leucogenys</em>) provided by the Gibbon Genome Sequencing
		Consortium with the following acknowledgments:
		<UL>
		  <LI><B>Funding: </B> <A HREF="http://www.genome.gov"
		  TARGET=_blank>National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly:</B>
		  <UL>
		    <LI><A HREF="http://genome.wustl.edu/genomes/detail/nomascus-leucogenys/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		    St. Louis, MO, USA
		    <LI>
		    <A HREF="https://www.hgsc.bcm.edu/non-human-primates/gibbon-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center 
		    (BCM-HGSC)</A>, Houston, TX, USA
		  </UL>
		  <LI><B>UCSC Gibbon Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI><a href="../../cgi-bin/hgGateway?db=nomLeu3" 
		    target="_blank">nomLeu3</a>: Pauline Fujita and Hiram Clawson
		    <LI><a href="../../cgi-bin/hgGateway?db=nomLeu2" 
		    target="_blank">nomLeu2</a>: Hiram Clawson, Chin Li, Brian Raney, Pauline 
		    Fujita, Luvina Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna 
		    Karolchik
		    <LI><a href="../../cgi-bin/hgGateway?db=nomLeu1" 
		    target="_blank">nomLeu1</a>: Chin Li, Brooke Rhead, Steve Heitner, Lauren 
		    Scott and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="gibbon_use"></A>
		The gibbon sequence data may be freely downloaded, used in analyses, and 
		repackaged in databases if the provider is properly acknowledged. Please cite 
		this publication when using these data:</P>
		<P>
		Carbone L, Harris RA, Gnerre S, Veeramah KR, Lorente-Galdos B, Huddleston J, 
		Meyer TJ, Herrero J, Roos C, Aken B <em>et al</em>.
		<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4249732/"
		target="_blank">Gibbon genome and the fast karyotype evolution of small 
		apes</a>.
		<em>Nature</em>. 2014 Sep 11;513(7517):195-201. </P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Gorilla Credits=================================================== -->
<A NAME="gorilla_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Gorilla Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The gorilla genome assembly (<em>Gorilla gorilla gorilla</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Sequencing:</B>
		  The <A HREF="http://www.sanger.ac.uk/resources/downloads/gorilla/"
		  TARGET="_blank">Wellcome Trust Sanger Institute</A>, 
		  Hinxton, Cambridge, UK
		  <LI><B>Assembly:</B>
		  <UL>
		    <LI>The <A HREF="http://www.sanger.ac.uk/resources/downloads/gorilla/"
		    TARGET="_blank">Wellcome Trust Sanger Institute</A>, 
		    Hinxton, Cambridge, UK
		    <LI>The <A HREF="http://www.ebi.ac.uk/"
		    TARGET="_blank">European Bioinformatics Institute</A> (EBI), 
		    Hinxton, Cambridge, UK
		  </UL>
		  <LI><B>UCSC Gorilla Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI><a href="../../cgi-bin/hgGateway?db=gorGor3" 
		    target="_blank">gorGor3</a>: Hiram Clawson, Mark Diekhans, Brian Raney, 
		    Brooke Rhead, Steve Heitner, Lauren Scott and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="gorilla_use"></A>
		The gorilla sequence data may be freely downloaded, used in analyses, and 
		repackaged in databases if the provider is properly acknowledged. Please cite 
		this publication when using these data:</P>
		<P>
		Scally A, Dutheil JY, Hillier LW, Jordan GE, Goodhead I, Herrero J, Hobolth A, 
		Lappalainen T, Mailund T, Marques-Bonet T <em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v483/n7388/abs/nature10842.html"
		target="_blank">Insights into hominid evolution from the gorilla genome 
		sequence</a>.
		<em>Nature</em>. 2012 Mar 7;483(7388):169-75. 
		</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Guinea Pig Credits ====================================================== -->
<A NAME="guinea_pig_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Guinea Pig Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The guinea pig  assembly (<em>Cavia porcellus</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B> 
		  <A HREF="http://www.genome.gov/" 
		  TARGET="_blank">National Human Genome Research Institute</A>
		  <LI><B>Sequencing/Assembly:</B>
		  <A HREF="http://www.broadinstitute.org/"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
<!-- 
		  <LI><B>BAC End Reads:</B>
		  <A HREF="http://www.tiho-hannover.de/index_e.htm"
		  TARGET="_blank">University of Veterinary Medicine 
		  Hanover</A>, Germany, and the 
		  <A HREF="http://www.helmholtz-hzi.de/en/"
		  TARGET="_blank">Helmholtz Centre for Infection 
		  Research</A>, Braunschweig, Germany
-->
		  <LI><B>UCSC Guinea Pig Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=cavPor3" 
		    target="_blank">cavPor3</a>: Tim Dreszer, Kate Rosenbloom, Hiram Clawson, 
		    Kayla Smith, Robert Kuhn, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="guinea_pig_use"></A>
		The guinea pig sequence is made freely available before 
		scientific publication with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The center producing the data reserves the right to 
		    publish the initial large-scale analyses of the data 
		    set, including large-scale identification of regions of 
		    evolutionary conservation and large-scale genomic 
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more). 
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Hedgehog Credits =================================================== -->
<A NAME="hedgehog_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Hedgehog Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Hedgehog Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A> at MIT and Harvard, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Hedgehog Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=eriEur2" target="_blank">eriEur2</a>:
		    Hiram Clawson, Brian Raney and Steve Heitner
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=eriEur1" target="_blank">eriEur1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="hedgehog_use"></A>
		The hedgehog sequence has been made freely available as 
		part of the Mammalian Genome Project (29 Mammals Project). 
		The initial analysis of this dataset can be found in 
		Lindblad-Toh K <em>et al</em>. 
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET="_blank">A high-resolution map of human evolutionary constraint using 
		29 mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The hedgehog data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Horse Credits ====================================================== -->
<A NAME="horse_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Horse Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC Horse Genome Browser displays the draft assembly 
		of the horse (<em>Equus caballus</em>) genome from The 
		Broad Institute. This assembly is provided with the 
		following acknowledgments:
		<UL>
		    <LI><B>Funding:</B> The 
		    <A HREF="http://www.genome.gov/" 
		    TARGET="_blank">National Human Genome Research Institute</A>
		    <LI><B>Project Leadership:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI><B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>
		    <LI><B>BAC End Reads:</B>
		    <A HREF="http://www.tiho-hannover.de/index_e.htm"
		    TARGET="_blank">University of Veterinary Medicine 
		    Hanover</A>, Germany, and the 
		    <A HREF="http://www.helmholtz-hzi.de/en/"
		    TARGET="_blank">Helmholtz Centre for Infection 
		    Research</A>, Braunschweig, Germany
		    <LI><B>UCSC Horse Genome Browser/Initial Annotations:</B> 
		    <UL>
		      <LI>
		      <a href="../../cgi-bin/hgGateway?db=equCab2" target="_blank">equCab2</a>: 
		      Larry Meyer, Hiram Clawson, Pauline Fujita, Brooke Rhead, and Donna 
		      Karolchik
		      <LI>
		      <a href="../../cgi-bin/hgGateway?db=equCab1" target="_blank">equCab1</a>:
		      Fan Hsu, Brooke Rhead, Robert Kuhn, Hiram Clawson, Angie Hinrichs, Kate 
		      Rosenbloom, and Donna Karolchik
		    </UL>
		</UL>
		<P>
		<HR>
		<A NAME="horse_use"></A>
		The horse sequence data may be freely downloaded, used in analyses, and 
		repackaged in databases if the provider is properly acknowledged. Please cite 
		this publication when using these data:</P>
		<P>
		Wade CM, Giulotto E, Sigurdsson S, Zoli M, Gnerre S, Imsland F, Lear TL, 
		Adelson DL, Bailey E, Bellone RR <em>et al</em>.
		<a href="http://www.sciencemag.org/content/326/5954/865.abs"
		target="_blank">Genome sequence, comparative analysis, and population genetics 
		of the domestic horse</a>.
		<em>Science</em>. 2009 Nov 6;326(5954):865-7.  </P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Kangaroo Rat Credits ================================================= -->
<A NAME="kangaroo_rat_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Kangaroo Rat Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Kangaroo Rat Genome Browser is provided with the 
		following acknowledgments:
		<UL>
		  <LI><B>Sequencing: </B>
		  <A HREF="https://www.hgsc.bcm.edu/others-mammals/kangaroo-rat-genome-project"
		  TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>, 
		  Houston, TX, USA 
		  <LI><B>Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		  <LI>
		  <B>UCSC Kangaroo Rat Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=dipOrd1" target="_blank">dipOrd1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="kangaroo_rat_use"></A>
		The kangaroo rat sequence has been made freely available as 
		part of the Mammalian Genome Project (29 Mammals Project). 
		The initial analysis of this dataset can be found in 
		Lindblad-Toh K <em>et al</em>. 
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET="_blank">A high-resolution map of human evolutionary constraint using 
		29 mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The kangaroo rat data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Manatee Credits ======================================================= -->
<A NAME="manatee_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Manatee Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Manatee Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Manatee Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=triMan1" target="_blank">triMan1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="manatee_use"></A>
		The manatee genome sequence data are made available to the
		public with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The center producing the data reserves the right to
		    publish the initial large-scale analyses of the data set,
		    including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more).
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Marmoset Credits ====================================================== -->
<A NAME="marmoset_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Marmoset Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>The Marmoset Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing and assembly:</B>
		  <A HREF="http://genome.wustl.edu/genomes/detail/callithrix-jacchus/"
		  TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		  St. Louis, MO, USA and
		  <a href="https://www.hgsc.bcm.edu/"
		  TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</a>,
		  Houston, TX, USA
		  <LI>
		  <B>DNA Source:</B>
		  Dr. Suzette Tardif,
		  <A HREF="http://www.txbiomed.org/primate-research-center/"
		  TARGET="_blank">Southwest National Primate Research Center</A>,
		  San Antonio, TX, USA
		  <LI>
		  <B>CHORI-259 BAC library:</B>
		  Dr. Pieter DeJong,
		  <A HREF="http://www.chori.org/"
		  TARGET="_blank">Children's Hospital Oakland Research Institute</A>,
		  Oakland, CA, USA
		  <LI>
		  <B>UCSC Marmoset Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=calJac3" target="_blank">calJac3</a>:
		    Hiram Clawson, Chin Li, Mark Diekhans, Mary Goldman, Vanessa Swing, and 
		    Donna Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=calJac1" target="_blank">calJac1</a>:
		    Hiram Clawson, Robert Kuhn, Pauline Fujita, Brooke Rhead, and Donna 
		    Karolchik
		  </UL>
		</UL>
		<P>
		<A NAME="marmoset_use"></A>
		<HR>
		These data were produced by The Genome Institute at WUSTL and the BCM-HGSC
		and can be downloaded
		<A HREF="http://genome.wustl.edu/pub/organism/Primates/Callithrix_jacchus/assembly/"
		TARGET="_blank">here</A>.</P>
		<P>
		The marmoset sequence is made freely available before
		scientific publication with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The centers producing the data reserve the right to
		    publish the initial large-scale analyses of the data set,
		    including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome.
		    <LI>
		    This is in accordance with, and with the understandings
		    in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A>
		    discussing Community Resource Projects and the resulting
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET="_blank">NHGRI policy statement</A>.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Megabat Credits ======================================================= -->
<A NAME="megabat_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Megabat Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Megabat (Fruitbat) Genome Browser is provided with the following
		acknowledgments:
		<UL>
		  <LI><B>Funding:</B>
		  <A HREF="http://www.genome.gov"
		  TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly: </B>
		  <UL>
		    <LI><A HREF="https://www.hgsc.bcm.edu/other-mammals/megabat-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>,
		    Houston, TX, USA
		    <LI><A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  </UL>
		  <LI>
		  <B>UCSC Megabat Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=pteVam1" target="_blank">pteVam1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="megabat_use"></A>
		The megabat sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The megabat data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Microbat Credits ====================================================== -->
<A NAME="microbat_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Microbat Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The microbat (little brown bat) genome is provided by
		The Broad Institute with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly: </B><A HREF="http://www.broadinstitute.org/"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Microbat Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=myoLuc2" target="_blank">myoLuc2</a>:
		    Chin Li, Brian Raney, Robert Kuhn, Steve Heitner, Brooke Rhead, Greg Roe and
		    Donna Karolchik
		  </UL>
		</UL>
		<P>
		<A NAME="microbat_use"></A>
		<HR>
		The microbat genome sequence data are made available to the
		public with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The center producing the data reserves the right to
		    publish the initial large-scale analyses of the data set,
		    including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more).
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
		</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>


<!-- Minke whale Credits ====================================================== -->
<A NAME="minke_whale_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Minke whale Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The minke whale genome is provided by the Korea Institute of Ocean Science and 
		Technology with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly: </B><A HREF="http://eng.kiost.ac/kordi_eng/main/"
		  TARGET="_blank">Korea Institute of Ocean Science and Technology</A>, Ansan, 
		  Republic of Korea
		  <LI>
		  <B>UCSC Minke whale Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=balAcu1" target="_blank">balAcu1</a>:
		    Hiram Clawsom and Matthew Speir
		  </UL>
		</UL>
		<P>
		<hr>
		<A NAME="minke_whale_use"></A>
		The minke whale genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</P>
		<P>
		Yim HS, Cho YS, Guang X, Kang SG, Jeong JY, Cha SS, Oh HM, Lee JH, Yang EC, 
		Kwon KK <em>et al</em>.
		<a href="http://www.nature.com/ng/journal/v46/n1/full/ng.2835.html" 
		target="_blank">Minke whale genome and aquatic adaptation in cetaceans</a>.
		<em>Nat Genet</em>. 2014 Jan;46(1):88-92.
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24270359" 
		target="_blank">24270359</a>
		</p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Mouse Credits ====================================================== -->
<A NAME="mouse_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Mouse Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
	    <P>
	    Starting with <A HREF="../../cgi-bin/hgGateway?db=mm10" TARGET="_blank">mm10</A> 
	    (GRCm38), the mouse genome assembly is now provided by the 
	    <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/"
	    TARGET="_blank">Genome Reference Consortium</A> (GRC). See the 
	    <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/"
	    TARGET="blank">GRC Mouse Genome</A> web pages for acknowledgments.
	    </P>
	    <P>
	    Prior to mm10/GRCm38, mouse assemblies were produced by the Mouse Genome
	    Sequencing Consortium, with these acknowledgments:
	    <UL>
	      <LI><B>Funding:</B>
	      <UL>
		<LI><A HREF="http://www.nih.gov/" TARGET="_blank">National
		    Institutes of Health</A>
		<LI><A HREF="http://www.wellcome.ac.uk/"
		    TARGET="_blank">The Wellcome Trust</A>
		<LI><A HREF="http://www.gsk.com/"
		    TARGET="_blank">GlaxoSmithKline</A>
		<LI><A HREF="http://www.merck.com/" TARGET="_blank">The
		    Merck Genome Research Institute</A>
		<LI><A HREF="http://www.affymetrix.com/estore/"
		    TARGET="_blank">Affymetrix</A>
	      </UL>

	      <LI><B>Sequencing:</B>
	      <UL>
		<LI><A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute's Arachne Group</A>
		<LI><A HREF="http://www.sanger.ac.uk"
		    TARGET="_blank">The Wellcome Trust Sanger Institute</A>
		<LI><A HREF="http://genome.wustl.edu/genomes/detail/mus-musculus/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>
		<LI><A HREF="https://www.hgsc.bcm.edu/other-mammals/mouse-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome
		    Sequencing Center (BCM-HGSC)</A>
		<LI><A HREF="http://www.ucl.ac.uk/wibr"
		    TARGET="_blank">Wolfson Institute for Biomedical
		    Research</A>
		<LI><A HREF="http://www.genome.ou.edu/"
		    TARGET="_blank">University of Oklahoma, Advanced Center
		    for Genome Technology</A>
		<LI><A HREF="http://www.har.mrc.ac.uk/"
		    TARGET="_blank">UK MRC Mouse Genome Centre and Mammalian
		    Genetics Unit
		<LI><A HREF="http://www.nisc.nih.gov/"
		    TARGET="_blank">National Institutes of Health Intramural
		    Sequencing Center</A>
		<LI><A HREF="http://personalizedmedicine.partners.org/"
		    TARGET="_blank">Harvard Partners Genome Center</A>
		<LI><A HREF="http://www.cshl.edu/"
		    TARGET="_blank">Cold Springs Harbor Laboratory</A>
	      </UL>
	      <LI><B>Physical Map:</B>
	      <UL>
		<LI>Mouse BAC libraries: <A HREF="http://bacpac.chori.org/"
		    TARGET="_blank">Children's Hospital Oakland Research
		    Institute</A> (CHORI)
		<LI>BAC fingerprint map:
		    <A HREF="http://www.bcgsc.ca/"
		    TARGET="_blank">Genome Sciences Centre</A>
		<LI>BAC end sequences:
		    <A HREF="http://www.jcvi.org/cms/home/"
		    TARGET="_blank">The Institute of Genome Research</A>
		    (TIGR)
		<LI>TPF production - Tina Graves, April Cook, Christine
		    Burton, Darren Graham, Shannon Dugan, Donna Muzny
	      </UL>
		<LI><B>Assembly:</B> Deanna Church, Richa Agarwala, Josh
		    Cherry, Mike DiCuccio, Hsiu-Chuan Chen, Paul Kitts,
		    Victor Sapojnikov

		<LI><B>cDNA Sequencing:</B>
		<UL>
		<LI><A HREF="http://genome.gsc.riken.go.jp/"
		    TARGET="_blank">Riken Institute</A>
		<LI><A HREF="http://mgc.nci.nih.gov/"
		    TARGET="_blank">The Mammalian Gene Collection</A>
	      </UL>
	      <LI><B>UCSC Mouse Genome Browser:</B> 
	      <UL>
		<LI>
		<a href="../../cgi-bin/hgGateway?db=mm10" target="_blank">mm10</a>: 
		Hiram Clawson and Luvina Guruvadoo
		<LI>
		<a href="../../cgi-bin/hgGateway?db=mm9" target="_blank">mm9</a>: 
		Hiram Clawson, Archana Thakkapallayil, Robert Kuhn and Donna Karolchik
		<LI>
		<a href="../../cgi-bin/hgGateway?db=mm8" target="_blank">mm8</a>:
		Hiram Clawson, Fan Hsu, Kayla Smith, Ann Zweig, Robert Kuhn, Brooke Rhead, 
		Archana Thakkapallayil, and Donna Karolchik
		<LI>
		<a href="../../cgi-bin/hgGateway?db=mm7" target="_blank">mm7</a>: 
		Hiram Clawson, Fan Hsu, Ann Zweig, Kayla Smith, Robert Kuhn and Donna Karolchik
		<LI>
		mm6 (archived): Hiram Clawson, Fan Hsu, Jennifer Jackson, Robert Kuhn, Ali 
		Sultan-Qurraie, Heather Trumbower and Donna Karolchik
	      </UL>
	      <LI><B>Annotations:</B>
	      <UL>
		<LI><A HREF="../staff.html" TARGET="_blank">UCSC Genome Bioinformatics Group</A>
		<LI><A HREF="http://www.jax.org/" TARGET="_blank">The Jackson Laboratory</A>
		<LI><A HREF="http://www.ncbi.nlm.nih.gov/"
		    TARGET="_blank">NCBI</A>
		<LI><A HREF="http://www.ensembl.org/"
		    TARGET="_blank">Ensembl</A>
		<LI><A HREF="http://www.bx.psu.edu/miller_lab/"
		    TARGET="_blank">Pennsylvania State University</A>
		<LI><A HREF="http://www.lbl.gov/"
		    TARGET="_blank">University of California Berkeley/LBNL</A>
	      </UL>
	    </UL>
	    <P>
	    The initial 3X coverage of the mouse genome was produced by
	    the Mouse Sequencing Consortium, with the following acknowledgments:
	    <UL>
	      <LI><B>Project Oversight:</B> Arthur Holden and Francis Collins
	      <LI><B>Funding:</B>
	      <UL>
		<LI><A HREF="http://www.gsk.com/" TARGET="_blank">GlaxoSmithKline</A>
		<LI><A HREF="http://www.mgri.org/home.html"
		  TARGET="_blank">Merck Genome Research Institute</A>
		<LI><A HREF="http://www.affymetrix.com/estore/"
		  TARGET="_blank">Affymetrix, Inc.</A>
		<LI><A HREF="http://www.wellcome.ac.uk/" TARGET="_blank">The Wellcome Trust</A>
		<LI><A HREF="http://www.nih.gov/" 
		  TARGET="_blank">The National Institutes of Health</A>
	      </UL>
	      <LI><B>Sequencing:</B>
	      <UL>
		<LI><A HREF="http://www.broadinstitute.org/"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		<LI><A HREF="http://genome.wustl.edu/genomes/detail/mus-musculus/"
		  TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>
		<LI><A HREF="http://www.sanger.ac.uk"
		  TARGET="_blank">The Wellcome Trust Sanger Institute</A>
	      </UL>
	    </UL></P>
	    <P>
	    <HR>
	    <A NAME="mouse_use"></A>
	    The mouse genome data may be freely downloaded, used in analyses, and 
	    repackaged in databases if the data provider is properly acknowledged. Please 
	    cite this publication when using these data:</P>
	    <P>
	    Mouse Genome Sequencing Consortium.
	    <A HREF="http://www.nature.com/nature/mousegenome/"
	    TARGET="_blank">Initial sequencing and comparative analysis of the mouse genome</A>.
	    <em>Nature</em>. 2002 Dec 5;420(6915):520-62. PMID: 12466850</P>
	    <P>
	    For more information about mouse assemblies provided by the GRC, see the 
	    <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/"
	    TARGET = "_blank">GRC mouse genome</A> web pages. For information about assemblies 
	    prior to mm10/GRCm38, see the Wellcome Trust Sanger Institute
	    <A HREF="https://www.sanger.ac.uk/resources/mouse/"
	    TARGET="_blank">Mouse Resources Portal</A>.</P>
	    </TD>
	    <TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Mouse Lemur Credits =================================================== -->
<A NAME="mouse_lemur_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Mouse Lemur Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Mouse Lemur Genome Browser is provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A> at MIT
		  and Harvard, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Mouse Lemur Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=micMur1" target="_blank">micMur1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="mouse_lemur_use"></A>
		The mouse lemur sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET="_blank">A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>The mouse lemur data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Naked Mole-rat Credits====================================================== -->

<A NAME="naked_mole-rat_credits"></A>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	    <TR><TD><FONT SIZE="4"></A><B>&nbsp; Naked Mole-rat Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The naked mole-rat genome was provided with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly: </B> <A HREF="http://www.genomics.cn/en/index"
		  target="_blank">Beijing Genomics Institute</A>, Shenzhen, China
		  <LI>
		  <B>UCSC Naked Mole-rat Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=hetGla2" target="_blank">hetGla2</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=hetGla1" target="_blank">hetGla1</a>:
		    Chin Li and Luvina Guruvadoo
		  </UL>
		</UL>
		<P>
		<A NAME="naked_mole-rat_use"></A>
		<HR>
		The naked mole-rat genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</P>
		Keane M, Craig T, Alf&ouml;ldi J, Berlin AM, Johnson J, Seluanov A, Gorbunova V,
		Di Palma F, Lindblad-Toh K, Church GM <em>et al</em>.
		<a href="http://bioinformatics.oxfordjournals.org/content/30/24/3558.long"
		target="_blank">The Naked Mole Rat Genome Resource: facilitating analyses of 
		cancer and longevity-related adaptations</a>.
		<em>Bioinformatics</em>. 2014 Dec 15;30(24):3558-60. </p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Opossum Credits=================================================== -->
<A NAME="opossum_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Opossum Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The opossum genome assembly (<em>Monodelphis domestica</em>) is provided
		by The Broad Institute with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/mammals/opossum"
		    TARGET="_blank">The Broad Institute</A>,
		    Cambridge, MA, USA
		  <LI><B>FISH Mapping: </B>
		    <A HREF="https://www.ncsu.edu/" TARGET="_blank">North
		    Carolina State University</A> (NCSU), Raleigh, NC, USA
		  <LI><B>UCSC Opossum Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=monDom5" target="_blank">monDom5</a>:
		    Andy Pohl, Hiram Clawson, Brian Raney, Mark Diekhans, Brooke Rhead and 
		    Katrina Learned
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=monDom4" target="_blank">monDom4</a>:
		    Hiram Clawson, Archana Thakkapallayil, Ann Zweig, Kayla Smith and Donna 
		    Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=monDom1" target="_blank">monDom1</a>:
		    Hiram Clawson, Galt Barber, Ali Sultan-Qurraie, Brian Raney and Donna 
		    Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="opossum_use"></A>
		The opossum genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</p>
		<p>
		Mikkelsen TS, Wakefield MJ, Aken B, Amemiya CT, Chang JL, Duke S, Garber M, 
		Gentles AJ, Goodstadt L, Heger A <em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v447/n7141/full/nature05805.html"
		target="_blank">Genome of the marsupial Monodelphis domestica reveals innovation
		in non-coding sequences</a>.
		<em>Nature</em>. 2007 May 10;447(7141):167-77.</p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Orangutan Credits=================================================== -->
<A NAME="orangutan_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Orangutan Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The Sumatran orangutan genome assembly (<em>Pongo pygmaeus abelii</em>) is 
		provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing:</B>
		  <UL>
		    <LI><A HREF="http://genome.wustl.edu/genomes/detail/pongo-abelii/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		    St. Louis, MO, USA
		    <LI>
		    <A HREF="https://www.hgsc.bcm.edu/non-human-primates/orangutan-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing
		    Center</A> (BCM-HGSC), Houston, TX, USA
		  </UL>
		  <LI><B>Sequence Assembly and Chromosomal Sequence/AGP Construction:</B> 
		    The Genome Institute at WUSTL
		  <LI><B>DNA Source:</B>
		    Dr. Greely Stones, <A HREF="http://www.gpz.org/"
		    TARGET="_blank">Gladys Porter Zoo</A>, Brownsville, TX, USA
		  <LI><B>BAC Library:</B>
		    Yuko Yoshonaga in Pieter de Jong's laboratory, 
		    <A HREF="http://bacpac.chori.org" 
		    TARGET="_blank">Children's Hospital Oakland Research Institute</A>, 
		    Oakland, CA, USA
		  <LI><B>Fosmid Library:</B> The Genome Institute at WUSTL
		  <LI><B>Fingerprint Map:</B> The Genome Institute at WUSTL
		  <LI><B>Cytogenetic Mapping and Human/Orangutan Breakpoint Analyses:</B> 
		    Mariano Rocchi, <A HREF="http://www.biologia.uniba.it/DIGEMI/"
		    TARGET="_blank">Department of Genetics and Microbiology</A>,
		    University of Bari, Bari, Italy
		  <LI><B>Fosmid End Placement against the Human Genome</B>
		    (used for breakpoint/inversion analyses during AGP
		    construction): Lin Chen and Evan Eichler, Department of
		    Genome Sciences, University of Washington, Seattle, WA, USA
		    <LI><B>Funding: </B> National Human Genome Research
		    Institute (NHGRI), National Institutes of Health (NIH)
		  <LI><B>UCSC Orangutan Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=ponAbe2" target="_blank">ponAbe2</a>:
		    Hiram Clawson, Kayla Smith, Robert Kuhn, Ann Zweig and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="orangutan_use"></A>
		The orangutan genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</p>
		<p>
		Locke DP, Hillier LW, Warren WC, Worley KC, Nazareth LV, Muzny DM, Yang SP, 
		Wang Z, Chinwalla AT, Minx P <em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v469/n7331/full/nature09687.html"
		target="_blank">Comparative and demographic analysis of orang-utan genomes</a>.
		<em>Nature</em>. 2011 Jan 27;469(7331):529-33. </p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Panda Credits ====================================================== -->
<A NAME="panda_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Panda Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The panda (<em>Ailuropoda melanoleuca</em>) assembly was provided by 
		BGI-Shenzhen with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly:</B>
		  <A HREF="http://www.genomics.cn/en/index"
		  TARGET="_blank">The Beijing Genomics Institute at Shenzhen</A>
		  <LI>
		  <B>UCSC Panda Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=ailMel1" target="_blank">ailMel1</a>:
		    Hiram Clawson, Ann Zweig, and Antonio Coelho
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="panda_use"></A>
		The panda genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</p>
		<p>
		Li R, Fan W, Tian G, Zhu H, He L, Cai J, Huang Q, Cai Q, Li B, Bai Y, Zhang Z 
		<em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v463/n7279/full/nature08696.html"
		target="_blank">The sequence and de novo assembly of the giant panda genome</a>.
		<em>Nature</em>. 2010 Jan 21;463(7279):311-7.  
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20010809" 
		target="_blank">20010809</a></p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Pig Credits ====================================================== -->
<A NAME="pig_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Pig Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The pig (<em>Sus scrofa</em>) assembly was provided with the following 
		acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly:</B>
		  <UL>
		    <LI><A HREF="http://piggenome.org/" 
		    TARGET="_blank">The International Swine Genome Sequencing Consortium</A> 
		    <LI><A HREF="http://www.sanger.ac.uk/resources/downloads/othervertebrates/pig.html"
		    TARGET="_blank">Wellcome Trust Sanger Institute</A>
		  </UL>
		  <LI>
		  <B>UCSC Pig Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=susScr3" target="_blank">susScr3</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=susScr2" target="_blank">susScr2</a>:
		    Hiram Clawson, Brian Raney, Mark Diekhans, and Mary Goldman
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="pig_use"></A>
		The International Swine Genome Sequencing Consortium genome-wide analyses of 
		the pig genome is described in:</p>
		<p> 
		Uenishi H, Morozumi T, Toki D, Eguchi-Ogawa T, Rund LA, Schook LB.
		<a href="http://www.biomedcentral.com/1471-2164/13/581"
		target=_blank">Large-scale sequencing based on full-length-enriched cDNA 
		libraries in pigs: contribution to annotation of the pig genome draft
		sequence</a>.
		<em>BMC Genomics</em>. 2012 Nov 15;13:581. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23150988" 
		target="_blank">23150988</a></p>
		<p>
		The pig genome data is made freely available with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The centers producing the data reserve the right to 
		    publish the initial large-scale analyses of the data 
		    set, including large-scale identification of regions of 
		    evolutionary conservation and large-scale genomic 
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more). 
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Pika Credits ======================================================= -->
<A NAME="pika_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Pika Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The pika genome assembly (<em>Ochotona princeps</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		  <LI>
		  <B>UCSC Pika Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=ochPri3" target="_blank">ochPri3</a>:
		    Hiram Clawson, Brian Raney, and Steve Heitner
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=ochPri2" target="_blank">ochPri2</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="pika_use"></A>
		The pika sequence has been made freely available as 
		part of the Mammalian Genome Project (29 Mammals Project). 
		The initial analysis of this dataset can be found in 
		Lindblad-Toh K <em>et al</em>. 
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29 
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The pika data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Platypus Credits =================================================== -->
<A NAME="platypus_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Platypus Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The platypus genome assembly (<em>Ornithorhynchus anatinus</em>) is provided 			with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://genome.wustl.edu/genomes/detail/ornithorhynchus-anatinus/"
		  TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		  St. Louis, MO, USA
		  <LI>
		  <B>UCSC Platypus Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=ornAna1" target="_blank">ornAna1</a>:
		    Angie Hinrichs, Kayla Smith, Robert Kuhn, Brian Raney, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="platypus_use"></A>
		The platypus genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</p>
		<p>
		Warren WC, Hillier LW, Marshall Graves JA, Birney E, Ponting CP, Grützner F, 
		Belov K, Miller W, Clarke L, Chinwalla AT <em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v453/n7192/full/nature06936.html"
		target="_blank">Genome analysis of the platypus reveals unique signatures of 
		evolution</a>.
		<em>Nature</em>. 2008 May 8;453(7192):175-83. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18464734" 
		target="_blank">18464734</a></p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Rabbit Credits =================================================== -->
<A NAME="rabbit_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Rabbit Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The rabbit genome assembly (<em>Oryctolagus cuniculus</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/science/projects/mammals-models/rabbit/rabbit-genome-sequencing-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		  <LI>
		  <B>UCSC Rabbit Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=oryCun2" target="_blank">oryCun2</a>:
		    Hiram Clawson and Antonio Coelho
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="rabbit_use"></A>
		The rabbit genome data are made freely available before 
		scientific publication with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers 
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The centers producing the data reserve the right to 
		    publish the initial large-scale analyses of the data 
		    set, including large-scale identification of regions of 
		    evolutionary conservation and large-scale genomic 
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more). 
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Rat Credits ====================================================== -->
<A NAME="rat_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Rat Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<p>
		The rat genome assembly (<em>Rattus norvegicus</em>) is provided by the
		Rat Genome Sequencing Consortium with the following acknowledgments:
		<UL>
		  <LI><B>Project Coordination:</B>
		    <A HREF="https://www.hgsc.bcm.edu/"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing
		    Center</A> (BCM-HGSC)
		  <LI><B>Funding:</B>
		  <UL>
		    <LI><A HREF="http://www.genome.gov/"
			TARGET="_blank">National Human Genome Research Institute</A> (NIH)
		    <LI><A HREF="http://www.nhlbi.nih.gov/"
			TARGET="_blank">National Heart, Lung and Blood Institute</A> (NIH)
		  </UL>
		  <LI><B>Sequencing:</B>
		  <UL>
		    <LI>BCM-HGSC
		    <LI><A HREF="https://www.celera.com/"
			TARGET="_blank">Celera Genomics Group of Applera Corporation</A>
		    <LI><A HREF="http://www.genomecorp.com/"
			TARGET="_blank">Genome Therapeutics Corporation</A>
		    <LI><A HREF="http://www.utah.edu/" TARGET="_blank">University of Utah</A>
		  </UL>
		  <LI><B>Assembly:</B> BCM-HGSC ATLAS group - Rui Chen, James Durbin, Paul 
		    Havlak
		  <LI><B>Fingerprint Mapping:</B>
		    <A HREF="http://www.bcgsc.ca/" TARGET="_blank">The British Columbia Cancer 
		    Agency Genome Sciences Center</A> - Marco Marra, Jackie Schein
		  <LI><B>BAC End Sequencing:</B>
		    <A HREF="http://www.jcvi.org/cms/home/"
		    TARGET="_blank">The Institute for Genomic Research</A> - Shaying Zhao
		  <LI><B>BAC Library Construction and Distribution:</B>
		    <A HREF="http://bacpac.chori.org/"
		    TARGET="_blank">Children's Hospital Oakland Research Institute</A> (CHORI) 
		    - Pieter deJong
		  <LI><B><A HREF="http://rgd.mcw.edu" 
		    TARGET="_blank">Rat Genome Database</A>:</B> Medical College of Wisconsin
		  <LI><B>UCSC Rat Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=rn6" target="_blank">rn6</a>: Hiram
		    Clawson, Brian Raney and Matthew Speir
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=rn5" target="_blank">rn5</a>: Hiram
		    Clawson, Brian Raney and Steve Heitner
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=rn4" target="_blank">rn4</a>: Angie
		    Hinrichs, Fan Hsu, Brooke Rhead, Archana Thakkapallayil,
		    Kayla Smith, Ann Zweig, Robert Kuhn, and Donna Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=rn3" target="_blank">rn3</a>: Hiram
		    Clawson, Heather Trumbower, Robert Kuhn, Yontao Lu, Terry Furey, Mark 
		    Diekhans, Robert Baertsch, Donna Karolchik, and Jim Kent
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="rat_use"></A>
		The rat genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite these publications when using these data:</p>
		<p>
		Havlak P, Chen R, Durbin KJ, Egan A, Ren Y, Song XZ, Weinstock GM, Gibbs RA.
		<a href="http://genome.cshlp.org/content/14/4/721.long" 
		target="_blank">The Atlas genome assembly system</a>.
		<em>Genome Res</em>. 2004 Apr;14(4):721-32. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15060016" 
		target="_blank">15060016</a></p>
		<p>
		Rat Genome Sequencing Project Consortium.
		<a href="http://www.nature.com/nature/journal/v428/n6982/abs/nature02426.html"
		target="_blank">Genome sequence of the Brown Norway rat yields insights into
		mammalian evolution</a>.
		<em>Nature</em>. 2004 Apr 1;428(6982):493-521. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15057822" 
		target="_blank">15057822</a></p>
		<p>
		See the Baylor <A HREF="https://www.hgsc.bcm.edu/bcm-hgsc-conditions-use"
		TARGET="_blank">conditions of use</A> web page for
		restrictions on the use of these data.</p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Rhesus Credits=================================================== -->
<A NAME="rhesus_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Rhesus Macaque Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The UCSC Genome Browser displays multiple assemblies of the Rhesus macaque
		genome produced by different institutions:
		<UL>
		<LI>
		<B><a href="../../cgi-bin/hgGateway?db=rheMac3" target="_blank">rheMac3</a> Genome Browser:</B> Oct. 2010 draft assembly of the rhesus monkey,
		version CR_1.0 from <A HREF="http://www.genomics.cn/en/index"
		TARGET=_blank>Beijing Genomics Institute-Shenzhen</A> (BGI), Shenzhen, China
		<LI>
		<B><a href="../../cgi-bin/hgGateway?db=rheMac2" target="_blank">rheMac2</a> Genome Browser:</B> Jan. 2006 draft assembly of the rhesus monkey,
		version Mmul_051212 from the Macaca mulatta Genome Sequencing Consortium (MMGSC)
		led by the
		<A HREF="https://www.hgsc.bcm.edu/rhesus-monkey-genome-project"
		TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>,
		in collaboration with the <A HREF="http://www.jcvi.org/cms/home/"
		TARGET="_blank">J. Craig Venter Science Foundation Joint Technology Center</A>
		and <A HREF="http://genome.wustl.edu/genomes/detail/macaca-mulatta/"
		TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>
		<LI>
		<B>rheMac1 Genome Browser:</B> Jan. 2005 draft assembly of the rhesus monkey,
		version Mmul_0.1 from MMGSC (see above)
		</UL></P>
		<P>
		The BGI assembly is provided with the following acknowledgments:
		<UL>
		<LI>
		<B>Funding:</B> Southern China Center for Innovative Pharmaceuticals (SCCIP),
		the National Natural Science Foundation of China, the Shenzhen Municipal
		Government of China, the Major State Basic Research Development Program of
		China, the National Basic Research Program of China, and the National Science
		and Technology Major Project of Key Drug Innovation and Development
		<LI>
		<B>Sequencing/Assembly:</B> BGI, Shenzhen, China
		</UL>
		<P>
		The MMGSC assembly is provided with the following acknowledgments:
		<UL>
		<LI>
		<B>Funding: </B>
		<A HREF="http://www.genome.gov/"
		TARGET="_blank">National Human Genome Research Institute</A> (NHGRI), USA
		<LI>
		<B>Sequencing/Assembly: </B>
		BCM-HGSC, Houston, TX, USA in collaboration with:
		  <UL>
		  <LI>
		  J. Craig Venter Science Foundation Joint Technology Center, Rockville, MD, USA
		  <LI>
		  The Genome Institute at WUSTL, St. Louis, MO, USA
		  </UL>
		<LI>
		<B>BAC resources: </B>
		<A HREF="http://bacpac.chori.org/rhesus250.htm"
		TARGET="_blank">Children's Hospital Oakland Research Institute</A> (CHORI),
		Oakland, CA, USA
		<LI>
		<B>BAC-based fingerprint map: </B>
		<A HREF="http://www.bcgsc.ca/"
		TARGET=-blank>Genome Sciences Centre</A>, Vancouver, B.C.
		</ul>
		<P>
		For more information on the MMGSC rhesus macaque genome project, see the BCM
		HGSC
		<A HREF="https://www.hgsc.bcm.edu/rhesus-monkey-genome-project"
		TARGET="_blank">Rhesus Monkey Genome Project</A> web page.
		<p>
		<UL>
		  <LI><B>UCSC Rhesus Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=rheMac3" target="_blank">rheMac3</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina Guruvadoo, Steve 
		    Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=rheMac2" target="_blank">rheMac2</a>:
		    Robert Baertsch, Kayla Smith, Ann Zweig, Robert Kuhn, and Donna Karolchik
		    <LI>
		    rheMac1 (archived): Robert Baertsch, Galt Barber, and Donna Karolchik
		  </UL>
		</UL>
		<p>
		<HR>
		<A NAME="rhesus_use"></A>
		The initial sequencing and analysis of the rhesus macaque genome
		produced by the MMGSC was published in: Rhesus Macaque Genome
		Sequencing and Analysis Consortium. 
		<A HREF="http://www.sciencemag.org/content/316/5822/222.long"
		TARGET="_blank">Evolutionary and Biomedical Insights from the Rhesus Macaque
		Genome</A>.  <em>Science</em>. 2007 Apr 13;316(5822):222-34. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17431167" 
		target="_blank">17431167</a></p>
		<p>
		The initial sequencing and analysis of the rhesus macaque genome produced by
		BGI was published in:
		Yan G, Zhang G, Fang X, Zhang Y, Li C, Ling F, Cooper DN, Li Q, Li Y,
		van Gool AJ <em>et al</em>
		<A HREF="http://www.nature.com/nbt/journal/v29/n11/full/nbt.1992.html"
		TARGET=_blank>Genome sequencing and comparison of two nonhuman primate animal
		models, the cynomolgus and Chinese rhesus macaques</A>. <em>Nat Biotechnol</em>.
		2011 Oct 16;29(11):1019-23. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22002653" 
		target="_blank">22002653</a></p>
		<p>
		The rhesus macaque sequence is made freely available for public use
		with the following understanding:
		<OL>
		<LI>The data may be freely downloaded, used in analyses,
		and repackaged in databases.
		<LI>Users are free to use the data in scientific papers
		analyzing particular genes and regions if the provider of
		these data is properly acknowledged. For the rheMac1 and <a href="../../cgi-bin/hgGateway?db=rheMac2" target="_blank">rheMac2</a> data, please
		cite the BCM-HGSC web site or publications from BCM-HGSC referring to
		the genome sequence.
		<LI>The BCM-HGSC and RMGSC plan to publish the assembly and
		genomic annotation of the dataset, including large-scale
		identification of regions of evolutionary conservation.
		<LI>This is in accordance with, and with the understandings
		in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A> discussing
		Community Resource Projects and the resulting
		<A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		TARGET="_blank">NHGRI policy statement</A>.
		<LI>Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Rock Hyrax Credits ======================================================= -->
<A NAME="rock_hyrax_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Rock Hyrax Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The rock hyrax genome assembly (<em>Procavia capensis</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B>
		  <A HREF="http://www.genome.gov"
		  TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly: </B>
		  <UL>
		    <LI><A HREF="https://www.hgsc.bcm.edu/hyrax-genome-project"
		    TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)</A>,
		    Houston, TX, USA
		    <LI><A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  </UL>
		  <LI>
		  <B>UCSC Rock Hyrax Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=proCap1" target="_blank">proCap1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="rock_hyrax_use"></A>
		The rock hyrax sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The rock hyrax data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!--  Sheep Credits ======================================================== -->
<a name="sheep_credits"></a>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <tr><td>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <tr><td><font size="4"><b>&nbsp; Sheep Genome </b></font>
	    </td></tr>
	</table>
	<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	    <th height=10></th>
	    <tr><td width=10></td><td>
		<p>
		The sheep genome assembly (<em>Ovis aries</em>) is provided by the International
		Sheep Genomics Consortium (ISGC) with the following acknowledgments:
		<ul>
		  <li><b>Sequencing/Assembly:</b>
		    <a href="http://www.sheephapmap.org/" 
		    target="_blank">International Sheep Genomics Consortium</a> (ISGC)</li>
		  <li><b>Mitochondrial Genome Sequencing:</b>
		    <a href="http://www.uni-giessen.de/cms?set_language=en"
		    target="_blank">Justus-Liebig-University</a>, Germany</li>
		  <li><b>UCSC Sheep Genome Browser/Initial Annotations:</B> 
		  <ul>
		    <li>
		    <a href="../../cgi-bin/hgGateway?db=oviAri3" target="_blank">oviAri3</a>:
		    Hiram Clawson, Brian Raney, Matthew Speir, Brian Lee, Brooke Rhead, Steve 
		    Heitner, and Donna Karolchik</li>
		    <li>
		    <a href="../../cgi-bin/hgGateway?db=oviAri1" target="_blank">oviAri1</a>:
		    Chin Li, Greg Roe, and Luvina Guruvadoo</li>
		 </ul>
		</ul>
		<p>
		<hr>
		<a name="sheep_use"></a>
		The sheep genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite these publications when using these data:</p>
		<p>
		Jiang Y, Xie M, Chen W, Talbot R, Maddox JF, Faraut T, Wu C, Muzny DM, Li Y, 
		Zhang W <em>et al</em>.
		<a href="http://www.sciencemag.org/content/344/6188/1168.abs"
		target="_blank">The sheep genome illuminates biology of the rumen and lipid 
		metabolism</a>.
		<em>Science</em>. 2014 Jun 6;344(6188):1168-73. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24904168" 
		target="_blank">24904168</a></p>
		<p>
		International Sheep Genomics Consortium.
		<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1365-2052.2010.02100.x/abstract;jsessionid=C33A2FD64BF95FF988F729F8C53421C8.f02t03"
		target="_blank">The sheep genome reference sequence: a work in progress</a>.
		<em>Anim Genet</em>. 2010 Oct;41(5):449-53. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20809919" 
		target="_blank">20809919</a></p>
		</p>
	    </td><td width=15></td></tr>
	</table>
    </td></tr>
</table>
<br>

<!-- Shrew Credits ======================================================= -->
<A NAME="shrew_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Shrew Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The common shrew genome assembly (<em>Sorex araneus</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B>
		  <A HREF="http://www.genome.gov"
		  TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Shrew Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=sorAra2" target="_blank">sorAra2</a>:
		    Hiram Clawson, Brian Raney, and Steve Heitner
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=sorAra1" target="_blank">sorAra1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="shrew_use"></A>
		The shrew sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The shrew data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Sloth Credits ======================================================= -->
<A NAME="sloth_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Sloth Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The sloth genome assembly (<em>Choloepus hoffmanni</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Sequencing:</B>
		  <A HREF="http://genome.wustl.edu/genomes/detail/choloepus-hoffmanni/"
		  TARGET="_blank">The Genome Institute at Washington University</A> (WUSTL),
		  St. Louis, MO, USA
		  <LI><B>Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Sloth Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=choHof1" target="_blank">choHof1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="sloth_use"></A>
		The sloth sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The sloth data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Squirrel Credits ======================================================= -->
<A NAME="squirrel_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Squirrel Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The squirrel genome assembly (<em>Spermophilus tridecemlineatus)</em>) is 
		provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Squirrel Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=speTri2" target="_blank">speTri2</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="squirrel_use"></A>
		The squirrel sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The squirrel data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Squirrel Monkey Credits ======================================================= -->
<A NAME="squirrel_monkey_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Squirrel Monkey Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The squirrel monkey  genome assembly (<em>Saimiri boliviensis</em>) is provided 
		with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Squirrel Monkey Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=saiBol1" target="_blank">saiBol1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="squirrel_monkey_use"></A>
		The squirrel monkey genome sequence data are made available to the
		public with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The center producing the data reserves the right to
		    publish the initial large-scale analyses of the data set,
		    including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more).
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Tarsier Credits ======================================================= -->
<A NAME="tarsier_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Tarsier Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The tarsier genome assembly (<em>Tarsius syrichta</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B>
		  <A HREF="http://www.genome.gov"
		  TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly: </B>
		  <UL>
		    <LI><A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI><A HREF="http://genome.wustl.edu/genomes/detail/tarsius-syrichta/"
		    TARGET="_blank">Washington University</A>, St Louis, MO, USA
		  </UL>
		  <LI>
		  <B>UCSC Tarsier Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=tarSyr2" target="_blank">tarSyr2</a>:
		    Hiram Clawson, Brian Raney, and Matthew Speir
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=tarSyr1" target="_blank">tarSyr1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="tarsier_use"></A>
		The tarsier sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The tarsier data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Tasmanian Devil Credits =================================================== -->
<A NAME="tasmanian_devil_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Tasmanian Devil Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Tasmanian devil genome assembly (<em>Sarcophilus harrisii</em>) is provided 
		with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly:</B>
		  <A HREF="http://www.sanger.ac.uk/"
		  TARGET="_blank">Wellcome Trust Sanger Institute</A>, Cambridge, UK
		  <LI>
		  <B>UCSC Tasmanian Devil Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=sarHar1" target="_blank">sarHar1</a>:
		    Chin Li and Luvina Guruvadoo
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="tasmanian_devil_use"></A>
		The Tasmanian devil genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</p>
		<p>
		Murchison EP, Schulz-Trieglaff OB, Ning Z, Alexandrov LB, Bauer MJ, Fu B, 
		Hims M, Ding Z, Ivakhno S, Stewart C <em>et al</em>.
		<a href="http://www.cell.com/cell/abstract/S0092-8674%2812%2900081-5"
		target="_blank">Genome sequencing and analysis of the Tasmanian devil and its 
		transmissible cancer</a>.
		<em>Cell</em>. 2012 Feb 17;148(4):780-91. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22341448" 
		target="_blank">22341448</a></p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Tenrec Credits====================================================== -->

<A NAME="tenrec_credits"></A>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	    <TR><TD><FONT SIZE="4"></A><B>&nbsp; Tenrec Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The tenrec genome assembly (<em>Echinops telfairi</em>) is provided by The 
		Broad Institute with the following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly: </B>
		  <a href="http://www.broadinstitute.org/"
		  target="_blank">The Broad Institute</a>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Tenrec Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=echTel2" target="_blank">echTel2</a>:
		    Hiram Clawson, Brian Raney and Steve Heitner
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=echTel1" target="_blank">echTel1</a>:
		    Hiram Clawson, Brian Raney and Steve Heitner
		  </UL>
		</UL>
		<P>
		<A NAME="tenrec_use"></A>
		<HR>
		The tenrec genome sequence data are made available to the
		public with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The center producing the data reserves the right to
		    publish the initial large-scale analyses of the data set,
		    including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more).
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
		</P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Tree Shrew Credits ======================================================= -->
<A NAME="tree_shrew_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Tree Shrew Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The tree shrew genome assembly (<em>Tupaia belangeri</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Funding:</B> <A HREF="http://www.genome.gov"
		  TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI>
		  <B>UCSC Tree Shrew Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=tupBel1" target="_blank">tupBel1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="tree_shrew_use"></A>
		The tree shrew sequence has been made freely available as
		part of the Mammalian Genome Project (29 Mammals Project).
		The initial analysis of this dataset can be found in
		Lindblad-Toh K <em>et al</em>.
		<A HREF="http://www.nature.com/nature/journal/v478/n7370/full/nature10530.html"
		TARGET=_blank>A high-resolution map of human evolutionary constraint using 29
		mammals</A>. <em>Nature</em>. 2011 Oct 12;478(7370):476-82. PMID: 21993624;
		PMCID: PMC3207357</p>
		<p>
		The tree shrew data are provided with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Wallaby Credits =================================================== -->
<A NAME="wallaby_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Wallaby Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The wallaby genome assembly (<em>Macropus eugenii</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  The Tammar Wallaby Genome Sequencing Consortium -
		  <A HREF="https://www.hgsc.bcm.edu/other-mammals/wallaby-genome-project"
		  TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A> 
		  (BCM-HGSC) and the
		  <A HREF="http://www.agrf.org.au/"
		  TARGET="_blank">Australian Genome Research Facility</A>
		  <LI>
		  <B>UCSC Wallaby Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=macEug2" target="_blank">macEug2</a>:
		    Chin Li and Luvina Guruvadoo
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="wallaby_use"></A>
		The wallaby genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite this publication when using these data:</p>
		<p>
		Renfree MB, Papenfuss AT, Deakin JE, Lindsay J, Heider T, Belov K, Rens W, 
		Waters PD, Pharo EA, Shaw G <em>et al.</em>
		<a href="http://genomebiology.com/content/12/8/R81"
		target="_blank">Genome sequence of an Australian kangaroo, Macropus eugenii, 
		provides insight into the evolution of mammalian reproduction and
		development</a>.
		<em>Genome Biol</em>. 2011 Aug 29;12(8):R81. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21854559" 
		target="_blank">21854559</a></p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- White Rhinoceros Credits =================================================== -->
<A NAME="white_rhinoceros_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; White Rhinoceros Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The southern white rhinoceros genome assembly (<em>Ceratotherium simum</em>) is 
		provided with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.broadinstitute.org/science/projects/projects"
		  TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		  <LI><B>Data Samples: </B>
		  <a href="http://biology.ucsd.edu/research/faculty/oryder"
		  target ="_blank">Dr. Oliver Ryder</a>, San Diego Zoo Institute for 
		  Conservation Research, San Diego, CA
		  <LI>
		  <B>UCSC White Rhinoceros Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=cerSim1" target="_blank">cerSim1</a>:
		    Hiram Clawson and Steve Heitner
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="white_rhinoceros_use"></A>
		The southern white rhinoceros sequence is made freely available before
		scientific publication with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The centers producing the data reserve the right to
		    publish the initial large-scale analyses of the data
		    set, including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome (that is, 30 Mb or more).
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- American Alligator Credits ====================================================== -->
<A NAME="american_alligator_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; American Alligator Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The American alligator assembly (<em>Alligator mississippiensis</em>) is 
		provided by the International Crocodilian Genomes Working Group with the 
		following acknowledgments:
		<UL>
		  <LI>
		  <B>Sequencing/Assembly:</B>
		   <a href="http://www.crocgenomes.org/"
		   target="_blank">International Crocodilian Genomes Working Group</a>
		  <LI>
		  <B>UCSC American Alligator Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=allMis1" target="_blank">allMis1</a>:
		    Hiram Clawson, Jonathan Casper, and Brian Lee
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="American_alligator_use"></A>
		The American alligator genome data may be freely downloaded, used in analyses, 
		and repackaged in databases if the data provider is properly acknowledged. 
		Please cite this publication when using these data:</p>
		<p>
		St John JA, Braun EL, Isberg SR, Miles LG, Chong AY, Gongora J, Dalzell P, 
		Moran C, Bed'hom B, Abzhanov A <em>et al</em>.
		<a href="http://genomebiology.com/content/13/1/415"
		target="_blank">Sequencing three crocodilian genomes to illuminate the evolution
		of archosaurs and amniotes</a>.  
		<em>Genome Biol</em>. 2012 Jan 31;13(1):415. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22293439" 
		target="_blank">22293439</a></p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Atlantic Cod Credits ======================================================= -->
<A NAME="atlantic_cod_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Atlantic Cod Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The Atlantic cod genome assembly (<em>Gadus morhua</em>) is provided with the 
		following acknowledgments:
		<UL>
		  <LI><B>Funding:</B>
		  <A HREF="http://www.forskningsradet.no/prognett-fuge/Programme_description/1226993493179"
		  TARGET=_blank>Research Council of Norway</A> (FUGE)
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://www.codgenome.no/index.php"
		  TARGET="_blank">GenoFisk Consortium</A>
		  <LI>
		  <B>UCSC Atlantic Cod Genome Browser/Initial Annotations:</B>
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=gadMor1" target="_blank">gadMor1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="atlantic_cod_use"></A>
		The Atlantic cod genome data may be freely downloaded, used in analyses, and 
		repackaged in databases if the data provider is properly acknowledged. Please 
		cite these publications when using these data:</p>
		<p>
		Johansen SD, Coucheron DH, Andreassen M, Karlsen BO, Furmanek T,
		J&oslash;rgensen TE, Emblem A, Breines R, Nordeide JT, Moum T <em>et al</em>.
		<A HREF="http://www.sciencedirect.com/science/article/pii/S1871678409000569"
		TARGET=_blank>Large-scale sequence analyses of Atlantic cod</A>.
		<em>N Biotechnol</em>. 2009 Jun;25(5):263-71. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19491044" 
		target="_blank">19491044</a></p>
		<p>
		Star B, Nederbragt AJ, Jentoft S, Grimholt U, Malmstr&oslash;m M, Gregers TF, 
		Rounge TB, Paulsen J, Solbakken MH, Sharma A <em>et al</em>.
		<a href="http://www.nature.com/nature/journal/v477/n7363/full/nature10342.html"
		target="_blank">The genome sequence of Atlantic cod reveals a unique immune 
		system</a>.
		<em>Nature</em>. 2011 Aug 10;477(7363):207-10. 
		PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21832995" 
		target="_blank">21832995</a></p>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Budgerigar Credits ======================================================= -->
<A NAME="budgerigar_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; Budgerigar Genome </B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD><TD>
		<P>
		The budgerigar genome assembly (<em>Melopsittacus undulatus</em>) is provided 
		with the following acknowledgments:
		<UL>
		  <LI><B>Sequencing/Assembly: </B>
		  <A HREF="http://genome.wustl.edu/genomes/"
		  TARGET="_blank">The Genome Institute at Washington University</A> (WUSTL),
		  St. Louis, MO, USA
		  <LI>
		  <B>UCSC Budgerigar Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=melUnd1" target="_blank">melUnd1</a>:
		    Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		    Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		  </UL>
		</UL>
		<P>
		<HR>
		<A NAME="budgerigar_use"></A>
		The budgerigar sequence is made freely available before
		scientific publication with the following understanding:
		<OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and
		    repackaged in databases.
		    <LI>
		    Users are free to use the data in scientific papers
		    analyzing particular genes and regions if the provider
		    of these data is properly acknowledged.
		    <LI>
		    The centers producing the data reserve the right to
		    publish the initial large-scale analyses of the data set,
		    including large-scale identification of regions of
		    evolutionary conservation and large-scale genomic
		    assembly. Large-scale refers to regions with size on the
		    order of a chromosome.
		    <LI>
		    This is in accordance with, and with the understandings
		    in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A>
		    discussing Community Resource Projects and the resulting
		    <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
		    TARGET="_blank">NHGRI policy statement</A>.
		    <LI>
		    Any redistribution of the data should carry this notice.
		</OL></P>
	    </TD><TD WIDTH=15></TD></TR>
	</TABLE>
    </TD></TR>
</TABLE>
<BR>

<!-- Chicken Credits ====================================================== -->
<A NAME="chicken_credits"></A>
<TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
    <TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
		<FONT SIZE="4"><B>&nbsp; Chicken Genome</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
	    <TD>
		<P>
		The chicken genome assembly (<em>Gallus gallus</em>) is provided with 
		the following acknowledgments:
		<UL>
		  <LI><B>Sequencing: </B>
		  <A HREF="http://genome.wustl.edu/genomes/detail/gallus-gallus/"
		  TARGET="_blank">The Genome Institute at Washington University</A> (WUSTL),
		  St. Louis, MO, USA
		  <LI><B>Physical Map:</B> The Genome Institute at WUSTL
		  <BR>
		  <LI><B>Genetic Mapping/Linkage Analysis: </B>The Chicken Mapping Consortium:
		  <UL>
		    <LI>
		    Wageningen Map based on Wageningen broilerxbroiler mapping population 
		    ("W" linkage maps; 460 F2 animals), coordinated by Martien Groenen, 
		    <A HREF="http://www.wageningenur.nl/en/Expertise-Services/Chair-groups/Animal-Sciences/Animal-Breeding-and-Genetics-Group.htm"
		    TARGET="_blank">Wageningen University</A>, Wageningen, The Netherlands 
		    <LI>
		    East Lansing Map based on the RJFxlayer cross ("E" linkage maps; 56 BC 
		    animals), coordinated by Hans Cheng,
		    <A HREF="https://www.ars.usda.gov/PandP/locations/cityPeopleList.cfm?modecode=50-50-00-00"
		    TARGET="_blank">USDA-ARS</A>, East Lansing, MI, USA
		    <LI>
		    Compton map based on layerxlayer cross ("C" linkage maps; 52 BC animals), 
		    coordinated by Nat Bumstead, <A HREF="http://www.pirbright.ac.uk/"
		    TARGET="_blank">Institute for Animal Health</A>
		    <LI>
		    Consensus linkage map: Martien Groenen and Richard Crooijmans, Wageningen 
		    University
		    <LI>
		    RH map: coordinated by Mireille Morisson and Alain Vignal, 
		    <A HREF="https://www-lgc.toulouse.inra.fr/internet/"
		    TARGET="_blank">French National Institute for Agricultural Research</A> 
		    (INRA), Toulouse 
		    <LI>
		    Linkage Mapping: White Leghorn X Red Junglefowl (800 F2 animals), 
		    coordinated by Susanne Kerje, Lina Jacobsson and Leif Andersson, 
		    <A HREF="http://www.uu.se/en/"
		    TARGET="_blank">Uppsala University</A>, Uppsala, Sweden
		    <LI>
		    Clone/Marker Pairs: coordinated by Jerry Dodgson, 
		    Michigan State University, East Lansing, MI, USA
		    <LI>
		    Marker Name/Sequence Resolution: coordinated by Hans 
		    Cheng, USDA-ARS, East Lansing, MI, USA
		    <LI>
		    Chicken FPC browser: 
		    <A HREF="http://www.bioinformatics.nl/gbrowse/cgi-bin/gbrowse/ChickFPC/"
		    TARGET="_blank">ChickFPC</A>, Martien Groenen and Richard 
		    Crooijmans (<a href="../../cgi-bin/hgGateway?db=galGal3" target="_blank">galGal3</a>), Jan Aerts (<a href="../../cgi-bin/hgGateway?db=galGal2" target="_blank">galGal2</a>), 
		    Wageningen University
		    <LI>
		    Additional Mapping Data Contributions: Winston Bellott, 
		    David Page Lab, 
		    <A HREF="http://wi.mit.edu/"
		    TARGET="_blank">Whitehead Institute for Biomedical Research</A> (WIBR-MIT), 
		    Cambridge, MA, USA 
		  </UL>
		  <LI>
		  <B>cDNA sequences:</B> <A HREF="http://www.ncbi.nlm.nih.gov/"
		  TARGET="_blank">National Institutes of Health</A> and The University of 
		  Manchester <A HREF="http://www.chick.manchester.ac.uk/"
		  TARGET="_blank">BBSRC ChickEST Database</A>
		  <LI>
		  <B>RJF finished clones:</B>
		  <UL>
		    <LI>
		    <A HREF="http://www.nisc.nih.gov/"
		    TARGET="_blank">NIH Intramural Sequencing Center</A> (NISC) (AC091708.2, 
		    AC094012.2, AC138565.3) <LI>S. Kerje and L. Andersson,
		    <A HREF="http://www.uu.se/en/" TARGET="_blank">Uppsala University</A> 
		    (AY636124.1)
		    <LI>
		    The Genome Institute at WUSTL (all others)
		  </UL>
		  <LI>
		  <B>Assembly, Assembly/Map Integration, Golden Path 
		  Creation:</B> The Genome Institute at WUSTL
		  <LI>
		  <B>SNPs and cross-referenced gene annotations:</B> 
		  <A HREF="http://www.genomics.cn/en/index"
		  TARGET="_blank">The Beijing Genomics Institute (BGI)</A>, 
		  Beijing, China
		  <LI>
		  <B>UCSC Chicken Genome Browser/Initial Annotations:</B> 
		  <UL>
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=galGal4" target="_blank">galGal4</a>:
		    Hiram Clawson, Brian Raney and Steve Heitner
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=galGal3" target="_blank">galGal3</a>:
		    Angie Hinrichs, Kayla Smith, Donna Karolchik
		    <LI>
		    <a href="../../cgi-bin/hgGateway?db=galGal2" target="_blank">galGal2</a>:
		    Angie Hinrichs, Heather Trumbower, Rachel Harte, Donna Karolchik
		  </UL>
		</UL>
		    <P>
		    <HR>
		    <A NAME="chicken_use"></A>
                    The chicken genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    International Chicken Genome Sequencing Consortium.
		    <a href="http://www.nature.com/nature/journal/v432/n7018/full/nature03154.html"
		    target="_blank">Sequence and comparative analysis of the chicken genome provide 
		    unique perspectives on vertebrate evolution</a>.
		    <em>Nature</em>. 2004 Dec 9;432(7018):695-716.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/15592404" 
		    target="_blank">15592404</a></p>
		    <P>
		    The BGI gene annotations and SNP data are made available with the following 
		    terms of use:
		    <OL>
		      <LI>
		      Users are free to use these data in scientific papers analyzing particular 
		      genes, if the Beijing Genomics Institute (BGI) is acknowledged.
		      <LI>
		      BGI and its collaborators reserve the right to publish the 
	  	      initial analyses of these data, including but not restricted 
		      to the large-scale identification of functional polymorphisms,
		      evolutionary patterns and signs of selection, correlations 
		      with known QTLs, utility of data for genetic mapping 
		      purposes, etc.
		      <LI>
		      Any redistribution of these data should carry this notice. 
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Coelacanth Credits ======================================================= -->
<A NAME="coelacanth_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Coelacanth Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The coelacanth genome assembly (<em>Latimeria chalumnae</em>) is provided with 
		    the following acknowledgments:
		    <UL>
	   	      <LI><B>Funding:</B>
		      <A HREF="http://www.genome.gov" 
		      TARGET="_blank">National Human Genome Research Institute</A> (NHGRI)
		      <LI><B>Sequencing/Assembly: </B>
	  	      <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/vertebrates-invertebrates/coelacanth/coelacanth"
		      TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA 
		      <LI>
		      <B>UCSC Coelacanth Genome Browser/Initial Annotations:</B> 
		      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=latCha1" target="_blank">latCha1</a>:
		      	Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
		      	Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
		      </UL>
		    </UL>
		    <P>
		    <HR>
		    <A NAME="coelacanth_use"></A>
                    The coelacanth genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Amemiya CT, Alf&ouml;ldi J, Lee AP, Fan S, Philippe H, Maccallum I, Braasch
		    I, Manousaki T, Schneider I, Rohner N <em>et al</em>.
		    <a href="http://www.nature.com/nature/journal/v496/n7445/full/nature12027.html"
		    target="_blank">The African coelacanth genome provides insights into tetrapod
		    evolution</a>.
		    <em>Nature</em>. 2013 Apr 18;496(7445):311-6. 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23598338" 
		    target="_blank">23598338</a></p>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Elephant Shark Credits ====================================================== -->
<A NAME="elephant_shark_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Elephant Shark Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The elephant shark genome assembly (<em>Callorhinchus milii</em>) is provided 
                    by the Singapore Institute of Molecular and Cell Biology with the following
                    acknowledgments:
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly:</B>
                      <a href="http://www.imcb.a-star.edu.sg/php/main.php"
	              target="_blank">Singapore Institute of Molecular and Cell Biology</a>
                      <LI>
                      <B>UCSC Elephant Shark Genome Browser/Initial Annotations:</B> 
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=calMil1" target="_blank">calMil1</a>:
                      	Hiram Clawson, Jonathan Casper, and Donna Karolchik
                      </UL>
                    </UL>
                    <P>
                    <HR>
                    <A NAME="elephant_shark_use"></A>
                    The elephant shark genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
                    Venkatesh B, Lee AP, Ravi V, Maurya AK, Lian MM, Swann JB, Ohta Y, Flajnik MF, 
		    Sutoh Y, Kasahara M <em>et al</em>.
                    <a href="http://www.nature.com/nature/journal/v505/n7482/full/nature12826.html"
                    target="_blank">Elephant shark genome provides unique insights into gnathostome 
		    evolution</a>. 
		    <em>Nature</em>. 2014 Jan 9;505(7482):174-9.
                    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24402279"
                    target="_blank">24402279</a> </p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Fugu Credits ====================================================== -->
<A NAME="fugu_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Fugu Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The Fugu genome assembly (<em>Takifugu rubripes</em>) is provided by the
                    <A HREF="http://genome.jgi-psf.org/fugu6/fugu6.info.html"
                    TARGET="_blank">U.S.  DOE Joint Genome Institute</A> (JGI) as
                    part of the International Fugu Genome Consortium, led by the JGI and the
                    <A HREF="http://www.fugu-sg.org"
                    TARGET="_blank">Singapore Institute of Molecular and Cell Biology</A> (IMCB). 
		    The assembly has the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly:</B>
                      Assembly was constructed with the JGI assembler, JAZZ, from paired end 
		      sequencing reads produced by JGI and the IMCB at JGI, Myriad Genetics, and 
		      Celera Genomics
                      <LI>
                      <B>UCSC Fugu Genome Browser/Initial Annotations:</B> 
                      <UL>
		     	<LI>
			<a href="../../cgi-bin/hgGateway?db=fr3" target="_blank">fr3</a>:
                      	Hiram Clawson and Greg Roe
                      	<LI>
                      	<a href="../../cgi-bin/hgGateway?db=fr2" target="_blank">fr2</a>:
                      	Cory McClean, Hiram Clawson, Ann Zweig, and Donna Karolchik
                      	<LI>
                      	<a href="../../cgi-bin/hgGateway?db=fr1" target="_blank">fr1</a>:
                      	Kate Rosenbloom, Heather Trumbower, Robert Kuhn, and Donna Karolchik
                      </UL>
		    </UL>
                    <P>
                    <A NAME="fugu_use"></A>
                    <HR>
		    <p>
                    The Fugu genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Kai W, Kikuchi K, Tohari S, Chew AK, Tay A, Fujiwara A, Hosoya S, Suetake H, 
		    Naruse K, Brenner S <em>et al</em>.
		    <a href="http://gbe.oxfordjournals.org/content/3/424.long"
		    target="_blank">Integration of the genetic map and genome assembly of fugu 
		    facilitates insights into distinct features of genome evolution in teleosts and 
		    mammals</a>.
		    <em>Genome Biol Evol</em>. 2011;3:424-42. 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21551351" 
		    target="_blank">21551351</a></p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Lamprey Credits =================================================== -->
<A NAME="lamprey_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Lamprey Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
		    The lamprey genome assembly (<em>Petromyzon marinus</em>) is provided with 
		    the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Funding:</B>
                      <A HREF="http://www.genome.gov" TARGET="_blank">National Human
                      Genome Research Institute</A> (NHGRI)
                      <LI>
                      <B>Sequencing/Assembly:</B>
                      <UL>
                        <LI><A HREF="http://genome.wustl.edu/genomes/detail/petromyzon-marinus/"
                        TARGET="_blank">The Genome Institute at WUSTL</A>, St. Louis, MO, USA
                        <LI>Jeramiah Smith
                          (<a href="https://bio.as.uky.edu/users/jjsmit3" target="_blank">University
                          of Kentucky</a>, <a href="http://www.gs.washington.edu/" 
		  	  target="_blank">University of Washington Department of Genome 
			  Sciences</a>, and <a href="https://www.benaroyaresearch.org/" 
			  target="_blank">Benaroya Research Institute at Virginia Mason</a>)
                      </UL>
                      <LI>
                      <B>UCSC Lamprey Genome Browser/Initial Annotations:</B> 
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=petMar2" target="_blank">petMar2</a>:
                      	Hiram Clawson, Brian Raney, and Steve Heitner
                        <LI>
                      	<a href="../../cgi-bin/hgGateway?db=petMar1" target="_blank">petMar1</a>:
                      	Hiram Clawson, Ann Zweig, Pauline Fujita, and Donna Karolchik
                      </UL>
                    </UL>
                    <P>
                    <HR>
                    <A NAME="lamprey_use"></A>
                    The lamprey genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Smith JJ, Kuraku S, Holt C, Sauka-Spengler T, Jiang N, Campbell MS, Yandell MD, 
		    Manousaki T, Meyer A <em>et al</em>.
		    <a href="http://www.nature.com/ng/journal/v45/n4/full/ng.2568.html"
		    target="_blank">Sequencing of the sea lamprey (Petromyzon marinus) genome 
		    provides insights into vertebrate evolution</a>.
		    <em>Nat Genet</em>. 2013 Apr;45(4):415-21, 421e1-2.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23435085" 
		    target="blank">234350859</a></p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Lizard Credits =================================================== -->
<A NAME="lizard_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Lizard Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The lizard genome assembly (<em>Anolis carolinensis</em>) is provided by The 
		    Broad Institute with the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly:</B>
                      <A HREF="http://www.broadinstitute.org/"
                      TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
                      <LI>
                      <B>UCSC Lizard Genome Browser/Initial Annotations:</B>
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=anoCar2" target="_blank">anoCar2</a>:
                        Hiram Clawson, Brian Raney, Luvina Guruvadoo, and Donna Karolchik
                        <LI>
                      	<a href="../../cgi-bin/hgGateway?db=anoCar1" target="_blank">anoCar1</a>:
                        Hiram Clawson, Archana Thakkapallayil, Robert Kuhn, and Donna Karolchik
                      </UL>
                    </UL>
                    <P>
                    <HR>
                    <A NAME="lizard_use"></A>
                    The lizard genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Alf&ouml;ldi J, Di Palma F, Grabherr M, Williams C, Kong L, Mauceli E, 
		    Russell P, Lowe CB, Glor RE, Jaffe JD <em>et al</em>.
		    <a href="http://www.nature.com/nature/journal/v477/n7366/full/nature10390.html"
		    target="_blank">The genome of the green anole lizard and a comparative analysis 
		    with birds and mammals</a>.
		    <em>Nature</em>. 2011 Aug 31;477(7366):587-91. 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/21881562" 
		    target="_blank">21881562</a></p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Medaka Credits ====================================================== -->
<A NAME="medaka_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Medaka Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The medaka genome assembly (<em>Oryzias latipes</em>) is provided by
                    Japan's <A HREF="http://www.nig.ac.jp/english/index.html"
                    TARGET="_blank">National Institute of Genetics</A>
                    (NIG) and the <A HREF="http://utgenome.org/medaka/"
                    TARGET="_blank">University of Tokyo</A>.
                    The assembly was constructed using the RAMEN assembler.</P>
                    <P>
                    NOTE: The UCSC oryLat1 assembly was replaced by 
		    <a href="../../cgi-bin/hgGateway?db=oryLat2" target="_blank">oryLat2</a> in
                    Nov. 2008 to correct a UCSC assembly error with chrUn in
                    which the gap relationships between contigs within their
                    ultracontigs were broken. The <a href="../../cgi-bin/hgGateway?db=oryLat2" 
		    target="_blank">oryLat2</a> assembly is based on the same assembly as oryLat1: 
		    the v1.0 assembly from NIG and the University of Tokyo.
                    </P>
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly: </B> NIG and University of Tokyo, Japan
                      <LI>
                      <B>UCSC Medaka Genome Browser/Initial Annotations:</B> 
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=oryLat2" target="_blank">oryLat2</a>:
                      	Hiram Clawson, Pauline Fujita, and Donna Karolchik
                      </UL>
                    </UL>
                    <P>
		    <A NAME="medaka_use"></A>
                    <HR>
                    The medaka genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Kasahara M, Naruse K, Sasaki S, Nakatani Y, Qu W, Ahsan B, Yamada T, Nagayasu Y,
		    Doi K, Kasai Y <em>et al</em>.
		    <a href="http://www.nature.com/nature/journal/v447/n7145/full/nature05846.html"
		    target="_blank">The medaka draft genome and insights into vertebrate genome 
		    evolution</a>.
		    <em>Nature</em>. 2007 Jun 7;447(7145):714-9.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17554307" 
		    target="_blank">17554307</a></p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Medium Ground Finch Credits ====================================================== -->
<A NAME="medium_ground_finch_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Medium Ground Finch Genome</B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The medium ground finch genome assembly (<em>Geospiza fortis</em>) is the 
		    product of a collaboration between the <a href="http://genome10k.soe.ucsc.edu"
                    target="_blank">Genome 10K Project</a> and 
		    <a href="http://www.genomics.cn/en/index"
                    target="_blank">Beijing Genomics Institute</a> (BGI). The data are provided 
		    with the following acknowledgments:</P>
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly: </B> Genome 10K Project and Beijing Genomics 
			Institute (BGI)
                      <LI>
                      <B>UCSC Medium Ground Finch Genome Browser/Initial Annotations:</B>
		      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=geoFor1" target="_blank">geoFor1</a>:
                      	Hiram Clawson and Greg Roe
                      </UL>
                    </UL>
                    <P>
		    <A NAME="medium_ground_finch_use"></A>
                    <HR>
                    The medium ground finch sequence is made freely available before scientific
                    publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
		    </OL>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Nile Tilapia Credits ======================================================= -->
<A NAME="nile_tilapia_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Nile Tilapia Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The Nile tilapia  genome assembly (<em>Oreochromis niloticus</em>) is provided 
		    with the following acknowledgments:
                    <UL>
                      <LI><B>Sequencing/Assembly: </B>
                      <A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/vertebrates-invertebrates/tilapia/tilapia-genom"
                      TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
                      <LI>
                      <B>UCSC Nile Tilapia Genome Browser/Initial Annotations:</B> 
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=oreNil2" target="_blank">oreNil2</a>:
                      	Hiram Clawson, Chin Li, Brian Raney, Pauline Fujita, Luvina
                      	Guruvadoo, Steve Heitner, Brooke Rhead, Greg Roe, and Donna Karolchik
                      </UL>
                    </UL>
                    <P>
                    <A NAME="nile_tilapia_use"></A>
                    <HR>
                    The Nile Tilapia genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Painted Turtle Credits====================================================== -->

<A NAME="painted_turtle_credits"></A>
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDERCOLOR="888888" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD><FONT SIZE="4"></A><B>&nbsp; Painted Turtle Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The painted turtle genome assembly (<em>Chrysemys picta bellii</em>) is provided
		    with the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly: </B> International Painted Turtle Genome Sequencing
                      Consortium
                      <LI>
                      <B>UCSC Painted Turtle Genome Browser/Initial Annotations:</B> 
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=chrPic1" target="_blank">chrPic1</a>:
                      	Hiram Clawson, Brian Raney and Steve Heitner
                      </UL>
                    </UL>
                    <P>
                    <A NAME="painted_turtle_use"></A>
                    <HR>
                    The painted turtle genome sequence data are made available to the
                    public with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                    </P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
   </TABLE>
    <BR>

<!-- Stickleback Credits =================================================== -->
<A NAME="stickleback_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Stickleback Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The stickleback genome assembly (<em>Gasterosteus aculeatus</em>) is provided 
		    by The Broad Institute with the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly:</B>
                      <A HREF="http://www.broadinstitute.org/"
                      TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
                      <LI>
                      <B>UCSC Stickleback Genome Browser/Initial Annotations:</B>
                      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=gasAcu1" target="_blank">gasAcu1</a>:
                      	Angie Hinrichs, Hiram Clawson, Kayla Smith, and Donna Karolchik
                      </UL>
                    </UL>
                    <P>
                    <HR>
                    <A NAME="stickleback_use"></A>
                    The stickleback sequence is made freely available before
                    scientific publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data
                        set, including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Tetraodon Credits =================================================== -->
<A NAME="tetraodon_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; Tetraodon Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
		    The Tetraodon genome assembly (<em>Tetraodon nigroviridis</em>) is provided with
		    the following acknowledgments:
                    <P>
                    <UL>
                      <LI>
                      <B>Sequencing/Assembly:</B>
                      <UL>
                        <LI>
                        <A HREF="http://www.genoscope.cns.fr/spip/spip.php?lang=en"
                        TARGET="_blank">Genoscope</A>, Evry, France
                        <LI>
                        <A HREF="http://www.broadinstitute.org/annotation/tetraodon/background.html"
                        TARGET="_blank">The Broad Institute</A>,MA, USA
                      </UL>
                      <LI>
                      <B>Project Funding:</B>
                      <UL>
                        <LI>
                        <A HREF="http://www.cng.fr/en/"
                        TARGET="_blank">Consortium National de Recherche en Genomique</A>, France
                        <LI>
                        <A HREF="http://www.genome.gov" 
			TARGET="_blank">National Human Genome Research Institute</A> (NHGRI), USA
                      </UL>
                      <LI>
                      <B>Initial Genome Browser Annotations:</B>
                      <UL>
                        <LI>
                        Genoscope - Genscan gene predictions, GAZE gene predictions
                        <LI>
                        <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>, University of 
			California, Santa Cruz, CA, USA  - remaining annotations (based on data from
                        Genoscope)
                      </UL>
                      <LI>
                      <B>UCSC Tetraodon Genome Browser:</B> 
		      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=tetNig2" target="_blank">tetNig2</a>:
			Hiram Clawson, Mark Diekhans, Brian Raney, Ann Zweig, Mary Goldman, Vanessa 
			Swing, Antonio Coelho, Robert Kuhn, and Donna Karolchik
                      	<LI>
                      	<a href="../../cgi-bin/hgGateway?db=tetNig1" target="_blank">tetNig1</a>:
			Rachel Harte, Robert Kuhn, Donna Karolchik
		      </UL>
                    </UL>
                    <P>
		    <A NAME="tetraodon_use"></A>
                    <HR>
                    The Tetraodon genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Yue GH, Lo LC, Zhu ZY, Lin G, Feng F.
		    The complete nucleotide sequence of the mitochondrial genome of Tetraodon 
		    nigroviridis.
		    <em>DNA Seq</em>. 2006 Apr;17(2):115-21.</p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Turkey Credits =================================================== -->
<A NAME="turkey_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Turkey Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The turkey genome assembly (<em>Meleagris gallopavo</em>) is provided with 
		    the following acknowledgments:
		    <UL>
		      <LI><B>Sequencing/Assembly: </B>
	  	      The Turkey Genome Consortium</A> - 
		      <A HREF="http://www.cals.vt.edu/"
		      TARGET="_blank">Virginia Tech College of Agriculture and Life 
		      Sciences</A> and
		      <A HREF="http://www.vbi.vt.edu/"
		      TARGET="_blank">Virginia Bioinformatics Institute</A>, 
		      Blacksburg, VA, USA
		      <LI>
		      <B>UCSC Turkey Genome Browser/Initial Annotations:</B> 
		      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=melGal1" target="_blank">melGal1</a>:
		        Chin Li, Mark Diekhans, Brian Raney, Brooke Rhead, Greg Roe,
		        Steve Heitner, and Donna Karolchik
		      </UL>
		    </UL>
		    <P>
		    <HR>
		    <A NAME="turkey_use"></A>
                    The turkey genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Dalloul RA, Long JA, Zimin AV, Aslam L, Beal K, Blomberg Le Ann,
		    Bouffard P, Burt DW, Crasta O, Crooijmans RP <em>et al.</em>
		    <a href="http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1000475"
		    target="_blank">Multi-platform next-generation sequencing of the domestic turkey
		    (Meleagris gallopavo): genome assembly and analysis</a>.
		    <em>PLoS Biol</em>. 2010 Sep 7;8(9). 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20838655" 
		    target="_blank">20838655</a></p>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- X. tropicalis Credits =============================================== -->
<A NAME="xentro_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>X. tropicalis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <em>Xenopus (Silurana) tropicalis</em> genome assembly is provided by the
		    <A HREF="http://genome.jgi-psf.org/Xentr4/Xentr4.home.html"
		    TARGET="_blank">U.S. DOE Joint Genome Institute</A> (JGI)
		    with the following acknowledgments:
		    <P>
		    <UL>
		      <LI>
		      <B>Sequencing/Assembly:</B> JGI, Walnut Creek, CA, USA
		      <LI>
		      <B>UCSC <em>X. tropicalis</em> Genome Browser/Initial Annotations:</B> 
		      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=xenTro3" target="_blank">xenTro3</a>:
		      	Hiram Clawson, Brian Raney, Brooke Rhead, Greg Roe, Steve 
		      	Heitner, and Donna Karolchik
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=xenTro2" target="_blank">xenTro2</a>:
		        Angie Hinrichs, Kayla Smith, Robert Kuhn, and Donna Karolchik
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=xenTro1" target="_blank">xenTro1</a>:
		        Fan Hsu, Jim Kent, Heather Trumbower, Hiram Clawson, Brian Raney,
		        Galt Barber, Mark Diekhans, Angie Hinrichs, and Donna Karolchik 
		      </UL>
		    </UL>
		    <P>
		    <HR>
                    The <em>X. tropicalis</em> genome data may be freely downloaded, used in 
		    analyses, and repackaged in databases if the data provider is properly 
		    acknowledged. Please cite this publication when using these data:</p>
		    <p>
		    Hellsten U, Harland RM, Gilchrist MJ, Hendrix D, Jurka J, Kapitonov V, 
		    Ovcharenko I, Putnam NH, Shu S, Taher L <em>et al</em>.
		    <a href="http://www.sciencemag.org/content/328/5978/633.abs"
		    target="_blank">The genome of the Western clawed frog Xenopus tropicalis</a>.
		    <em>Science</em>. 2010 Apr 30;328(5978):633-6. 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20431018" 
		    target="_blank">20431018</a></p>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Zebra Finch Credits================================================= -->
<A NAME="zebra_finch_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Zebra Finch Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The zebra finch genome assembly (<em>Taeniopygia guttata</em>) is
		    provided by the Genome Institute at Washington University (WUSTL)
		    with the following acknowledgments:
		    <P>
		    <UL>
		      <LI>
		      <B>Sequencing/Assembly:</B> 
                      <A HREF="http://genome.wustl.edu/genomes/detail/taeniopygia-guttata/"
		      TARGET="_blank">The Genome Institute at WUSTL</A>, St. Louis, MO, USA
		      <LI>
		      <B>Zebra finch DNA:</B> 
		      Arthur P. Arnold lab, Dept. of Physiological Science, 
		      UCLA, Los Angeles, CA, USA
  <!-- Commented out until we verify this info
		      <LI>
		      <B>BAC library:</B> 
		      Barbara Blackmon, Clemson University Genomics Institute, Clemson, SC, USA 
  -->
                      <LI>
                      <B>UCSC Zebra Finch Genome Browser/Initial Annotation:</B> 
		      <UL>
			<LI>
			<a href="../../cgi-bin/hgGateway?db=taeGut2" target="_blank">taeGut2</a>:
                      	Hiram Clawson, Brian Raney, and Steve Heitner
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=taeGut1" target="_blank">taeGut1</a>:
		      	Brian Raney, Kayla Smith, Pauline Fujita, and Donna Karolchik
		      </UL>
		    </UL>
		    <P>
		    <HR>
		    <A NAME="zebra_finch_use"></A>
                    The zebra finch genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Warren WC, Clayton DF, Ellegren H, Arnold AP, Hillier LW, K&uuml;nstner A, 
		    Searle S, White S, Vilella AJ, Fairley S <em>et al</em>.
		    <a href="http://www.nature.com/nature/journal/v464/n7289/full/nature08819.html"
		    target="_blank">The genome of a songbird</a>.
		    <em>Nature</em>. 2010 Apr 1;464(7289):757-62. 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20360741" 
		    target="_blank">20360741</a></p>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Zebrafish Credits ====================================================== -->
<A NAME="zebrafish_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Zebrafish Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    Starting with the <A HREF="../../cgi-bin/hgGateway?db=danRer10" 
		    TARGET="_blank">danRer10</A> (GRCz10) assembly, the zebrafish genome sequence is
		    now provided by the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/zebrafish/"
		    TARGET="_blank">Genome Reference Consortium</A> (GRC).</P>
		    <P>
		    Prior assemblies have the following acknowledgments:
		    <P>
		    <UL>
		      <LI>
		      <B>Sequencing/Assembly:</B> 
		      <A HREF="http://www.sanger.ac.uk/resources/zebrafish/genomeproject.html"
		      TARGET="_blank">The Wellcome Trust Sanger Institute</A>, UK. 
		      <LI>
		      <B>FPC map:</B> 
		      <UL>
		        <LI>
		        The Wellcome Trust Sanger Institute, UK
		        <LI>
		        <A HREF="http://www.hubrecht.eu/"
		        TARGET="_blank">Hubrecht Laboratory</A>, Netherlands Institute
		        for Developmental Biology, The Netherlands
		        <LI>
		        <A HREF="http://www.eb.tuebingen.mpg.de/" 
		        TARGET="_blank">Max-Planck Institute for Developmental 
		        Biology</A>, Germany
		      </UL>
		      <LI>
		      <B>BAC Clones, BAC End Pairs: </B> 
		      <UL>
		        <LI>
		        The Wellcome Trust Sanger Institute, UK - GenBank 
		        accessions for BAC clones
		        <LI>
		        Max-Planck Institute for Developmental 
		        Biology, Germany - BAC end pairs sequence data
		        <LI>
		        Hubrecht Laboratory</A>, Netherlands Institute
		        for Developmental Biology, The Netherlands - BAC end pairs 
		        sequence data
		      </UL>
		      <LI>
		      <B>Annotations:</B> 
		      <UL>
		        <LI>
		        Yi Zhou, Anthony DiBiase, Leonard Zon and the Zebrafish
		        Genome Initiative at <A HREF="http://zon.tchlab.org/" 
		        TARGET="_blank">Boston Children's Hospital</A>, Boston, MA, US -
		        collaboration on Zebrafish WZ EST clusters, BAC End Singles/Pairs, Radiation 
		        Hybrid, and Human tBLASTn annotation tracks
		        <LI>
		        <A HREF="http://www.genetics.wustl.edu/fish_lab/frank/cgi-bin/fish/"
		        TARGET="_blank">Washington University in St. Louis</A>, St. 
		        Louis, MO, US - Zebrafish WZ EST clusters data
		        <LI>
		        <A HREF="http://www.ensembl.org" TARGET="_blank">Ensembl</A>,
		        UK - Ensembl gene predictions
		        <LI>
		        <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>, University of 
		        California, Santa Cruz, CA, USA - Other annotations
		      </UL>
                      <LI>
                      <B>UCSC Zebrafish Genome Browsers:</B>
		      <UL>
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=danRer10" 
		        target="_blank">danRer10</a>: Hiram Clawson, Brian Raney, Steve Heitner
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=danRer7" 
		        target="_blank">danRer7</a>: Hiram Clawson, Mary Goldman, Greg Roe, Brian 
		        Raney, Donna Karolchik 
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=danRer6" 
		        target="_blank">danRer6</a>: Galt Barber, Robert Kuhn, Katrina Learned, 
		        Donna Karolchik 
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=danRer5" 
		        target="_blank">danRer5</a>: Rachel Harte, Ann Zweig, Donna Karolchik 
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=danRer4" 
		        target="_blank">danRer4</a>: Rachel Harte, Archana Thakkapallayil, Robert 
		        Kuhn, Ann Zweig, Donna Karolchik 
		        <LI>
		        <a href="../../cgi-bin/hgGateway?db=danRer3" 
		        target="_blank">danRer3</a>: Rachel Harte, Jennifer Jackson, Ann Zweig, 
		        Ali Sultan-Qurraie, Donna Karolchik 
		        <LI>
		        danRer2 (archived): Rachel Harte, 
		        Mark Diekhans, Heather Trumbower, Donna Karolchik, Jennifer Jackson 
		      </UL>
		    </UL>
		    <P>
		    <HR>
		    <A NAME="zebrafish_use"></A>
		    <P>
		    <p>
                    The zebrafish genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Howe K, Clark MD, Torroja CF, Torrance J, Berthelot C, Muffato M, Collins JE, 
		    Humphray S, McLaren K, Matthews L <em>et al</em>.
		    <a href="http://www.nature.com/nature/journal/v496/n7446/full/nature12111.html"
		    target="_blank">The zebrafish reference genome sequence and its relationship
		    to the human genome</a>.
		    <em>Nature</em>. 2013 Apr 25;496(7446):498-503. 
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/23594743" 
		    target="_blank">23594743</a></p>
		    <p>
		    For more information about zebrafish assemblies provided by the GRC, see the 
		    <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/zebrafish/"
		    TARGET = "_blank">GRC zebrafish</A> web pages. For information about assemblies 
	 	    prior to danRer10/GRCz10, see the Sanger Institute's  
		    <A HREF="http://www.sanger.ac.uk/resources/zebrafish/genomeproject.html"
		    TARGET="_blank"><em>Danio rerio</em> Sequencing Project</A> web page.
		    </p>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Ciona Credits====================================================== -->

<A NAME="ciona_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. intestinalis</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>
                    The <em>C. intestinalis</em> v1.0 and v2.0 draft assemblies are provided
                    by the <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.home.html"
                    TARGET="_blank">U.S. DOE Joint Genome Institute</A> (JGI) with the 
                    following acknowledgments:
                    <P>
                      <UL>
                        <LI>
                        <B>Project Funding</B>: Work performed under the auspices of
                        the US Department of Energy's Office of Science, Biological
                        and Environmental Research Program and the by the University
                        of California, Lawrence Livermore National Laboratory under
                        Contract No. W-7405-Eng-48, Lawrence Berkeley National
                        Laboratory under contract No. DE-AC03-76SF00098 and Los Alamos
                        National Laboratory under contract No. W-7405-ENG-36.
                        <LI>
                        <B>Project Collaborators</B>:
                        <UL>
                          <LI>
                          JGI, Walnut Creek, CA, USA
                          <LI>
                          <A HREF="http://www.zool.kyoto-u.ac.jp/zool-e.html"
                          TARGET="_blank">Department of Zoology, Graduate School of Science, Kyoto 
			  University</A>, Kyoto, Japan
                          <LI>
                          <A HREF="http://mcb.berkeley.edu/" TARGET="_blank">Department of Molecular
			  and Cellular Biology, University of California, Berkeley</A>, CA, USA
                          <LI>
                          <A HREF="http://www.nig.ac.jp/english/index.html"
                          TARGET="_blank">National Institute of Genetics</A>, Mishima,
                          Japan
                          <LI>
                          <A HREF="http://www.ornl.gov/" TARGET="_blank">Oak Ridge National 
			  Laboratory</A>, Oak Ridge, TN, USA
                          <LI>
                          <A HREF="http://www.lanl.gov/" TARGET="_blank">Los Alamos National 
			  Laboratory</A>, Los Alamos, NM, USA
                        </UL>
                      <LI>
                      <B>Annotations:</B>
                        <UL>
                          <LI>
                          Colin Dewey, University of California, Berkeley, CA, USA -
                          SNAP Gene Predictions, using the SNAP parser written by Ian Korf
                          <LI>
                    	  JGI, Walnut Creek, CA, USA - JGI Gene Predictions
                    	  <LI>
                    	  <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>, University 
			  of California, Santa Cruz, CA, USA - Other annotations
                        </UL>
                      <LI>
                      <B>UCSC <em>C. intestinalis</em>  Genome Browsers:</B>
		  	<UL>
		          <LI>
                          <a href="../../cgi-bin/hgGateway?db=ci2" target="_blank">ci2</a>: Brian 
			  Raney, Mark Diekhans, Ann Zweig, Kayla Smith, Robert Kuhn, Donna Karolchik
                    	  <LI>
                    	  <a href="../../cgi-bin/hgGateway?db=ci1" target="_blank">ci1</a>: Brian
                    	  Raney, Galt Barber, Heather Trumbower, Robert Kuhn, Donna Karolchik
                        </UL>
		      </UL>
                    <P>
                    <HR>
                    The <em>C. intestinalis</em> sequence data have been freely provided by the 
		    JGI and may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    Dehal P, Satou Y, Campbell RK, Chapman J, Degnan B, De Tomaso A, Davidson B, 
		    Di Gregorio A, Gelpke M, Goodstein DM <em>et al</em>. 
		    <a href="http://www.sciencemag.org/content/298/5601/2157.abtract"
		    target="_blank">The draft genome of Ciona intestinalis: insights into chordate 
		    and vertebrate origins</a>. <em>Science</em>. 2002 Dec 13;298(5601):2157-67.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/12481130" 
		    target="_blank">12481130</a></P>
		    <P>
                    For more information about the <em>C. intestinalis</em> assembly, see the JGI
                    <A HREF="http://genome.jgi-psf.org/ciona4/ciona4.info.html"
                    TARGET="_blank"><em>C. intestinalis</em> portal</A>.  </P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- Lancelet Credits =================================================== -->
<A NAME="lancelet_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Lancelet Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The lancelet (<em>Branchiostoma floridae</em>) genome assembly is provided 
		    by the DOE Joint Genome Institute (JGI) with the following acknowledgments:
		    <UL>
		      <LI>
		      <B>Funding:</B>
		      U.S. Department of Energy Office of Science, Biological 
		      and Environmental Research Program 
		      <LI>
		      <B>Sequencing/Assembly:</B>
		      <A HREF="http://genome.jgi-psf.org/Brafl1/Brafl1.home.html"
		      TARGET="_blank">JGI</A>
		      <LI>
		      <B>UCSC Lancelet Genome Browser/Initial Annotations:</B> 
		      <UL>
		        <LI>
			<a href="../../cgi-bin/hgGateway?db=braFlo1" target="_blank">braFlo1</a>:
		        Hiram Clawson, Ann Zweig, Pauline Fujita, Donna Karolchik </LI>
		      </UL>
		    </UL>
		    <P>
		    <HR>
		    <A NAME="lancelet_use"></A>
		    <p>
                    The lancelet genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Putnam NH, Butts T, Ferrier DE, Furlong RF, Hellsten U, Kawashima T,
		    Robinson-Rechavi M, Shoguchi E, Terry A, Yu JK <em>et al</em>.
		    <a href="http://www.nature.com/nature/journal/v453/n7198/full/nature06967.html"
		    target="_blank">The amphioxus genome and the evolution of the chordate
		    karyotype</a>.
		    <em>Nature</em>. 2008 Jun 19;453(7198):1064-71. doi: 10.1038/nature06967.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/18563158" 
		    target="_blank">18563158</a></p>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- S. purpuratus Credits============================================= -->
<A NAME="urchin_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD>
		    <FONT SIZE="4"><B>&nbsp; <em>S. purpuratus</em> Genome</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD>
		<TD>
		    <P>
		    The sea urchin (<em>Strongylocentrotus purpuratus</em>) genome assembly 
		    is provided by the Baylor College of Medicine Human Genome Sequencing Center 
		    (BCM-HGSC). The assemblies available in the browser include: 
		    <UL>
		      <LI>
		      Sep. 2006 strPur2 browser: Baylor version Spur 2.1 (NCBI v2.1)
		      <LI>
		      Apr. 2005 strPur1 browser: Baylor version Spur_0.5 (NCBI v1.1)
		    </UL>
		    <P>These data are provided with the following acknowledgments:
		    <UL>
	   	      <LI><B>Funding</B> - 
		      <A HREF="http://www.genome.gov" TARGET="_blank">National Human
		      Genome Research Institute</A> (NHGRI)
		      <LI><B>Sequencing/Assembly: </B>
			  <A HREF="https://www.hgsc.bcm.edu/" TARGET="_blank">BCM-HGSC</A>, Houston, TX, USA<BR>
                      <LI><B>UCSC <em>S. purpuratus</em> Genome Browsers/Initial Annotations:</B>
		      <UL>
		        <LI><a href="../../cgi-bin/hgGateway?db=strPur2" target="_blank">strPur2</a>: 
			Kord Kober, Heather Trumbower, Rachel Harte, Kayla Smith, Donna Karolchik
		     	
		        <LI><a href="../../cgi-bin/hgGateway?db=strPur1" target="_blank">strPur1</a>: 
			Andy Pohl, Kayla Smith, Donna Karolchik
		      </UL>
		    </UL>
		    <A NAME="urchin_use"></A>
		    <P>
                    The sea urchin genome data may be freely downloaded, used in analyses, and 
		    repackaged in databases if the data provider is properly acknowledged. Please 
		    cite this publication when using these data:</p>
		    <p>
		    Sea Urchin Genome Sequencing Consortium.
		    <a href="http://www.sciencemag.org/content/314/5801/941.long"
		    target=_blank>The genome of the sea urchin Strongylocentrotus purpuratus</a>.
		    <em>Science</em>. 2006 Nov 10;314(5801):941-52.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17095691" 
		    target="_blank">17095691</a></p>
		    <p>
		    For more information on the sea urchin genome project, see the BCM-HGSC 
		    <A HREF="https://www.hgsc.bcm.edu/other-invertebrates/sea-urchin-genome-project"
		    TARGET="_blank">Sea Urchin Genome Project</A> web page.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- A. gambiae Credits================================================== -->
<A NAME="anoGam_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>A. gambiae</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>The MOZ2 <em>Anopheles gambiae</em> data were provided by the
                    International Anopheles Genome Project and downloaded from Ensembl. The data 
		    have the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Funding:</B>
                      <A HREF="http://www.niaid.nih.gov/Pages/default.aspx"
                      TARGET="_blank">National Institute of Allergy and Infectious Diseases</A> 
		      (NIAID), <A HREF="http://www.enseignementsup-recherche.gouv.fr/"
                      TARGET="_blank">French Ministry of Research</A>, and the
                      <A HREF="http://www.who.int/en/" TARGET="_blank">World Health Organization</A>
		      (WHO). Celera Genomics and Genoscope
                      were funded respectively by <A HREF="http://www.nih.gov/"
                      TARGET="_blank">National Institutes of Health</A> (NIH) and the
                      French Ministry of Research.
                      <LI>
                      <B>Project collaboration:</B>
                      The International Anopheles Genome Project - a collaboration between
                      <A HREF="https://www.celera.com/" TARGET="_blank">Celera Genomics</A>,
                      <A HREF="http://www.genoscope.cns.fr/spip/spip.php?lang=en"
                      TARGET="_blank">Genoscope</A>,
                      <A HREF="http://www.nd.edu/" TARGET="_blank">University of Notre Dame</A>,
                      <A HREF="http://www.ebi.ac.uk/"
                      TARGET="_blank">EBI</A>/<A HREF="http://www.sanger.ac.uk/"
                      TARGET="_blank">Sanger Institute</A>,
                      <A HREF="http://www.embl.org/" TARGET="_blank">EMBL</A>,
                      <A HREF="http://www.pasteur.fr/en" TARGET="_blank">Institut Pasteur</A>, 
		      <A HREF="http://www.imbb.forth.gr/"
                      TARGET="_blank">The Institute of Molecular Biology and Biotechnology</A> 
		      (IMBB) and <A HREF="http://www.jcvi.org/cms/home/" 
		      TARGET="_blank">The Institute for Genomic Research</A> (TIGR).
                      <LI>
                      <B>Sequencing and Assembly:</B>
                      <A HREF="https://www.celera.com/" TARGET="_blank">Celera Genomics</A>, 
		      <A HREF="http://www.genoscope.cns.fr/spip/spip.php?lang=en"
                      TARGET="_blank">Genoscope</A>
                      <LI>
                      <B>Data Downloads:</B>
                      <A HREF="http://metazoa.ensembl.org/Anopheles_gambiae/Info/Index"
                      TARGET="_blank">Ensembl</A>
                      <LI>
                      <B>UCSC <em>A. gambaie</em> Genome Browser/Initial Annotations:</B>
		      <UL>
			<LI><a href="../../cgi-bin/hgGateway?db=anoGam1" target="_blank">anoGam1</a>:
                      Angie Hinrichs, Galt Barber, Donna Karolchik
		      </UL>
                      <LI>
                      <B>Additional Genome Browser Annotations</B>:
                      <A HREF="http://www.ensembl.org" TARGET="_blank">Ensembl</A> and
                      <A HREF="https://www.celera.com/" TARGET="_blank">Celera Genomics</A> 
		      (Ensembl Genes)
                    </UL>
                    <P>
                    <HR>
                    The <em>A. gambaie</em> sequence is made freely available
                    by the International Anopheles Genome Project. Please cite the following
		    publication when using these data:
		    <p>
		    Holt RA, Subramanian GM, Halpern A, Sutton GG, Charlab R, Nusskern DR, Wincker 
		    P, Clark AG, Ribeiro JM, Wides R <em>et al</em>. 
		    <a href="http://www.sciencemag.org/content/298/5591/129.abstract"
		    target=_blank>The genome sequence of the malaria mosquito Anopheles gambiae</a>.
		    <em>Science</em>. 2002 Oct 4;298(5591):129-49.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/12364791" 
		    target="_blank">12364791</a></p>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- A. mellifera Credits=============================================== -->
<A NAME="apiMel_credits"></A>
<A NAME="apiMel_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>A. mellifera</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>The Amel_1.2 data were provided by the
                    Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC)
                    with the following acknowledgments:
                    <UL>
                      <LI>
                      <B>Funding: </B>
                      <A HREF="http://www.genome.gov" TARGET="_blank">National Human
                      Genome Research Institute</A> (NHGRI) and the
                      <A HREF="http://www.usda.gov/wps/portal/usda/usdahome"
                      TARGET="_blank">U.S. Department of Agriculture</A> (USDA)
                      <LI>
                      <B>Sequencing and assembly:</B>
                      <A HREF="https://www.hgsc.bcm.edu/arthropods/honey-bee-genome-project"
		      TARGET="_blank">BCM-HGSC</A>, Houston, TX, USA
                      <LI>
                      <B>BAC library:</B>
                      Pieter de Jong and Katzutoyo Osoegawa at
                      <A HREF="http://bacpac.chori.org/"
                      TARGET="_blank">Children's Hospital Oakland Research
                      Institute</A> (CHORI), Oakland, CA, USA
                      <LI>
                      <B>UCSC <em>A. mellifera</em> Genome Browser/Initial Annotations:</B>
                      <UL>
			<LI><a href="../../cgi-bin/hgGateway?db=apiMel2" target="_blank">apiMel2</a>:
                        Andy Pohl, Angie Hinrichs, Jennifer Jackson, Donna Karolchik
			<LI><a href="../../cgi-bin/hgGateway?db=apiMel1" target="_blank">apiMel1</a>:
                        Angie Hinrichs, Brian Raney, Robert Kuhn, Donna Karolchik
		      </UL>
                    </UL>
                    <P>
                    <HR>
                    The <em>A. mellifera</em> sequence is made freely available
                    by the Honeybee Genome Sequencing Consortium. Please cite the following
		    publication when using these data: </p>
		    <p>
		    Honeybee Genome Sequencing Consortium.
		    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2048586/"
		    target="_blank">Insights into social insects from the genome of the honeybee 
		    Apis mellifera</a>.
		    <em>Nature</em>. 2006 Oct 26;443(7114):931-49.
		    PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17073008" 
		    target="_blank">17073008</a></p>
		    <p>
                    See the Baylor <A HREF="https://www.hgsc.bcm.edu/bcm-hgsc-conditions-use"
                    TARGET="_blank">conditions of use</A> statement for guidelines regarding the 
		    use of these data.
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. ananassae Credits================================================== -->
<A NAME="droAna_credits"></A>
<A NAME="droAna_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. ananassae</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>The droAna2 data were provided by Agencourt
                    Bioscience. The droAna1 data were produced by the Institute
                    for Genomic Research (TIGR). The <em>D. ananassae</em>
                    Genome Browser has the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing:</B>
                    <A HREF="http://www.beckmangenomics.com/"
                    TARGET="_blank">Agencourt Bioscience</A>,
                    Danvers, MA, USA
                    <LI>
                    <B>Assembly:</B>
                    <UL>
                    <LI>
                    <A HREF="http://www.beckmangenomics.com/"
                    TARGET="_blank">Agencourt Bioscience</A>,
                    Danvers, MA, USA, using the
                    <A HREF="http://www.broadinstitute.org/science/programs/genome-biology/computational-rd/computational-research-and-development"
                    TARGET="_blank">Arachne</A> assembler (droAna2)
                    <LI>
                    <A HREF="http://www.jcvi.org/cms/home/" TARGET="_blank">TIGR</A>,
                    Rockville, MD, USA, using an assembler produced by
                    <A HREF="https://www.celera.com/" TARGET="_blank">Celera</A>
                    (droAna1)
                    </UL>
                    <LI>
                    <B>UCSC <em>D. ananassae</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droAna2" target="_blank">droAna2</a>):</B>
                    Angie Hinrichs, Brian Raney, Ann Zweig, Kayla Smith, and
                    Donna Karolchik
                    <LI>
                    <B>UCSC <em>D. ananassae</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droAna1" target="_blank">droAna1</a>):</B>
                    Angie Hinrichs, Brian Raney, Heather Trumbower, Ali
                    Sultan-Qurraie, and Donna Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome
                    Bioinformatics Group</A>, University of California,
                    Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    These data have been freely provided by Agencourt
                    Bioscience before
                    publication for use in the UCSC Genome Browser with the
                    following understanding:
                    <OL>
                    <LI>
                    The data may be freely downloaded, used in analyses, and
                    repackaged in databases.
                    <LI>
                    Users are free to use the data in scientific papers
                    analyzing particular genes and regions if the
                    provider of these data is properly acknowledged.
                    <LI>
                    Agencourt Bioscience reserves the right to publish the initial
                    large-scale analyses of the dataset, including large-scale
                    identification of regions of evolutionary conservation and
                    large-scale genomic assembly.
                    <LI>
                    Any redistribution of the data should carry this notice.
                    </OL>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. erecta Credits================================================== -->
<A NAME="droEre_credits"></A>
<A NAME="droEre_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. erecta</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    The <a href="../../cgi-bin/hgGateway?db=droEre1" target="_blank">droEre1</a> data were provided by
                    Agencourt Bioscience. The
                    <em>D.  erecta</em> Genome Browser has the following
                    acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing and Assembly:</B>
                    <A HREF="http://www.beckmangenomics.com/"
                    TARGET="_blank">Agencourt Bioscience</A>,
                    Danvers, MA, USA, using the
                    <A HREF="http://www.broadinstitute.org/science/programs/genome-biology/computational-rd/computational-research-and-development"
                    TARGET="_blank">Arachne</A> assembler
                    <LI>
                    <B>UCSC <em>D. erecta</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droEre1" target="_blank">droEre1</a>):</B>
                    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna
                    Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome
                    Bioinformatics Group</A>, University of California,
                    Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    These data have been freely provided by Agencourt
                    Bioscience before
                    publication for use in the UCSC Genome Browser with the
                    following understanding:
                    <OL>
                    <LI>
                    The data may be freely downloaded, used in analyses, and
                    repackaged in databases.
                    <LI>
                    Users are free to use the data in scientific papers
                    analyzing particular genes and regions if the
                    provider of these data is properly acknowledged.
                    <LI>
                    Agencourt Bioscience reserves the right to publish the initial
                    large-scale analyses of the dataset, including large-scale
                    identification of regions of evolutionary conservation and
                    large-scale genomic assembly.
                    <LI>
                    Any redistribution of the data should carry this notice.
                    </OL>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. grimshawi Credits================================================== -->
<A NAME="droGri_credits"></A>
<A NAME="droGri_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. grimshawi</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    The droGri data were provided by
                    Agencourt Bioscience. The
                    <em>D.  grimshawi</em> Genome Browser has the following
                    acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing and Assembly:</B>
                    <A HREF="http://www.beckmangenomics.com/"
                    TARGET="_blank">Agencourt Bioscience</A>,
                    Danvers, MA, USA, using the
                    <A HREF="http://www.broadinstitute.org/science/programs/genome-biology/computational-rd/computational-research-and-development"
                    TARGET="_blank">Arachne</A> assembler
                    <LI>
                    <B>UCSC <em>D. grimshawi</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droGri1" target="_blank">droGri1</a>):</B>
                    Angie Hinrichs, Brian Raney, Jennifer Jackson, Kayla Smith
                    and Donna Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome
                    Bioinformatics Group</A>, University of California,
                    Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    These data have been freely provided by Agencourt
                    Bioscience before
                    publication for use in the UCSC Genome Browser with the
                    following understanding:
                    <OL>
                    <LI>
                    The data may be freely downloaded, used in analyses, and
                    repackaged in databases.
                    <LI>
                    Users are free to use the data in scientific papers
                    analyzing particular genes and regions if the
                    provider of these data is properly acknowledged.
                    <LI>
                    Agencourt Bioscience reserves the right to publish the initial
                    large-scale analyses of the dataset, including large-scale
                    identification of regions of evolutionary conservation and
                    large-scale genomic assembly.
                    <LI>
                    Any redistribution of the data should carry this notice.
                    </OL>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. melanogaster Credits=============================================== -->
<A NAME="dm_credits"></A>
<A NAME="dm_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. melanogaster</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>D. melanogaster</em> assemblies displayed in the 
		    UCSC Genome Browser were obtained from the 
		    <A HREF="http://www.fruitfly.org/" TARGET="_blank">Berkeley 
		    Drosophila Genome Project</A> (BDGP) with the following 
		    acknowledgments:
		    <UL>
		    <LI><B>Sequencing and Assembly:</B>
		    <UL>
		    <LI>Euchromatic sequence - 
		    BDGP, Berkeley, CA, USA
		    <LI>Heterochromatic sequence - 
		    <A HREF="http://www.fruitfly.org/" TARGET="_blank">Drosophila
		    Heterochromatin Genome Project</A>, Lawrence Berkeley
		    National Laboratory, Berkeley, CA, USA
		    </UL>
		    <LI><B>Annotations: </B>
		    <UL>
		    <LI>Euchromatic regions - 
		    <A HREF="http://flybase.org/"
		    TARGET="_blank">FlyBase</A> 
		    <LI>Heterochromatic regions - 
		    DHGP, Lawrence Berkeley
		    National Laboratory, Berkeley, CA, USA
		    </UL>
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser
		    and additional annotations (<a href="../../cgi-bin/hgGateway?db=dm6" target="_blank">dm6</a>): </B>
		    Hiram Clawson, Angie Hinrichs, and Matthew Speir
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser 
		    and additional annotations (<a href="../../cgi-bin/hgGateway?db=dm3" target="_blank">dm3</a>): </B>
		    Angie Hinrichs, Archana Thakkapallayil, Kayla Smith, and
		    Donna Karolchik - 
		    <A HREF="../staff.html">UCSC Genome 
		    Bioinformatics Group</A>, UCSC, Santa Cruz, CA, USA 
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser 
		    and additional annotations (<a href="../../cgi-bin/hgGateway?db=dm2" target="_blank">dm2</a>): </B>
		    Angie Hinrichs, Brian Raney, Galt Barber,
		    and Donna Karolchik - 
		    UCSC Genome Bioinformatics Group, UCSC, Santa Cruz, CA, USA 
		    <LI><B>UCSC <em>D. melanogaster</em> Genome Browser 
		    and additional annotations (<a href="../../cgi-bin/hgGateway?db=dm1" target="_blank">dm1</a>): </B>
		    Angie Hinrichs, Brian Raney, Heather Trumbower, Ali 
		    Sultan-Qurraie, and Donna Karolchik - 
		    UCSC Genome Bioinformatics Group, UCSC, Santa Cruz, CA, USA 
		    </UL>
		    </P>
		    <P>
		    <HR>
		    For additional information about these data, including 
		    citation guidelines, see the 
		    <A HREF="http://www.fruitfly.org" 
		    TARGET="_blank">BDGP web site</A>.
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. mojavensis Credits================================================== -->
<A NAME="droMoj_credits"></A>
<A NAME="droMoj_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. mojavensis</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>These data were produced by Agencourt
                    Bioscience. The <em>D. mojavensis</em> Genome Browser
                    has the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing and Assembly:</B>
                    <A HREF="http://www.beckmangenomics.com/"
                    TARGET="_blank">Agencourt Bioscience</A>,
                    Danvers, MA, USA, using the
                    <A HREF="http://www.broadinstitute.org/science/programs/genome-biology/computational-rd/computational-research-and-development"
                    TARGET="_blank">Arachne</A> assembler
                    <LI>
                    <B>Funding:</B>
                    <A HREF="http://www.genome.gov" TARGET="_blank">National Human
                    Genome Research Institute</A> (NHGRI), USA
                    <LI>
                    <B>UCSC <em>D. mojavensis</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droMoj2" target="_blank">droMoj2</a>):</B>
                    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna
                    Karolchik
                    <LI>
                    <B>UCSC <em>D. mojavensis</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droMoj1" target="_blank">droMoj1</a>):</B>
                    Angie Hinrichs, Brian Raney, Galt Barber and Donna
                    Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome
                    Bioinformatics Group</A>, University of California,
                    Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    These data have been freely provided by Agencourt
                    Bioscience before
                    publication for use in the UCSC Genome Browser with the
                    following understanding:
                    <OL>
                    <LI>
                    The data may be freely downloaded, used in analyses, and
                    repackaged in databases.
                    <LI>
                    Users are free to use the data in scientific papers
                    analyzing particular genes and regions if the
                    provider of these data is properly acknowledged.
                    <LI>
                    Agencourt Bioscience reserves the right to publish the initial
                    large-scale analyses of the dataset, including large-scale
                    identification of regions of evolutionary conservation and
                    large-scale genomic assembly.
                    <LI>
                    Any redistribution of the data should carry this notice.
                    </OL>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. persimilis Credits================================================== -->
<A NAME="droPer_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. persimilis</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <a href="../../cgi-bin/hgGateway?db=droPer1" target="_blank">droPer1</a> assembly was provided with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI>
		    <B>UCSC <em>D. persimilis</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droPer1" target="_blank">droPer1</a>):</B>
		    Angie Hinrichs, Kayla Smith, Robert Kuhn, Jennifer Jackson 
		    and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>, 
		    University of California, 
		    Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="droPer_use"></A>
		    The <em>D. persimilis</em> sequence is made freely available
		    before scientific
		    publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data set, 
			including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. pseudoobscura Credits================================================== -->
<A NAME="dp_credits"></A>
<A NAME="dp_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. pseudoobscura</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>These data were produced by the
                    Baylor College of Medicine Human Genome Sequencing Center (BCM-HGSC).
                    The <em>D. pseudoobscura</em> Genome Browser has the following
                    acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing and Assembly:</B>
                    <A HREF="https://www.hgsc.bcm.edu/arthropods/drosophila-genome-project"
		    TARGET="_blank">BCM-HGSC</A>, Houston, TX, USA
                    <LI>
                    <B>UCSC <em>D. pseudoobscura</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=dp3" target="_blank">dp3</a>):</B>
                    Angie Hinrichs, Brian Raney, Jennifer Jackson and Donna
                    Karolchik
                    <LI>
                    <B>UCSC <em>D. pseudoobscura</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=dp2" target="_blank">dp2</a>):</B>
                    Angie Hinrichs, Heather Trumbower, Robert Kuhn, Donna
                    Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome
                    Bioinformatics Group</A>, University of California,
                    Santa Cruz, CA, USA
                    </UL>
                    <P>The <em>D. pseudoobscura</em> data are made available with
                    specific <A HREF="https://www.hgsc.bcm.edu/bcm-hgsc-conditions-use"
                    TARGET="_blank">conditions for use</A>.
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. sechellia Credits================================================== -->
<A NAME="droSec_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. sechellia</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>The <a href="../../cgi-bin/hgGateway?db=droSec1" target="_blank">droSec1</a> assembly was provided with the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing/Assembly:</B>
                    <A HREF="http://www.broadinstitute.org/"
                    TARGET="_blank">The Broad Institute</A>, Cambridge, MA USA
                    <LI>
                    <B>UCSC <em>D. sechellia</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droSec1" target="_blank">droSec1</a>):</B>
                    Angie Hinrichs, Kayla Smith and Donna Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>,
                    University of California, Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    <A NAME="droSec_use"></A>
                    The <em>D. sechellia</em> sequence is made freely available
                    before scientific
                    publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The center producing the data reserves the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome (that is, 30 Mb or more).
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. simulans Credits================================================== -->
<A NAME="droSim_credits"></A>
<A NAME="droSim_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. simulans</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were provided with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing and assembly:</B>
		    <A HREF="http://genome.wustl.edu/genomes/detail/drosophila-simulans-white-501/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>,
		    St. Louis, MO, USA 
		    <LI>
		    <B>UCSC <em>D. simulans</em> Genome Browser
		    (<a href="../../cgi-bin/hgGateway?db=droSim1" target="_blank">droSim1</a>)</B>:
		    Angie Hinrichs, Jennifer Jackson, Ali Sultan-Qurraie,
		    Brian Raney and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
                    The <em>D. simulans</em> sequence is made freely available before
                    scientific publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The centers producing the data reserve the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome.
                        <LI>
                        This is in accordance with, and with the understandings
                        in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A>
                        discussing Community Resource Projects and the resulting
                        <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
                        TARGET="_blank">NHGRI policy statement</A>.
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- D. virilis Credits================================================== -->
<A NAME="droVir_credits"></A>
<A NAME="droVir_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
        <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. virilis</em> Genome </B></FONT>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
                <TR><TD WIDTH=10></TD><TD>
                    <P>These data were produced by Agencourt
                    Bioscience. The <em>D. virilis</em> Genome Browser
                    has the following acknowledgments:
                    <UL>
                    <LI>
                    <B>Sequencing and Assembly:</B>
                    <A HREF="http://www.beckmangenomics.com/"
                    TARGET="_blank">Agencourt Bioscience</A>,
                    Danvers, MA, USA, using the
                    <A HREF="http://www.broadinstitute.org/science/programs/genome-biology/computational-rd/computational-research-and-development"
                    TARGET="_blank">Arachne</A> assembler
                    <LI>
                    <B>Funding:</B>
                    <A HREF="http://www.genome.gov" TARGET="_blank">National Human
                    Genome Research Institute</A> (NHGRI), USA
                    <LI>
                    <B>UCSC <em>D. virilis</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droVir2" target="_blank">droVir2</a>):</B>
                    Angie Hinrichs, Brian Raney, Jennifer Jackson and
                    Donna Karolchik
                    <LI>
                    <B>UCSC <em>D. virilis</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droVir1" target="_blank">droVir1</a>):</B>
                    Angie Hinrichs, Brian Raney, Heather Trumbower,
                    Robert Kuhn, Jennifer Jackson, and Donna Karolchik
                    <LI>
                    <B>Initial Genome Browser Annotations:</B>
                    <A HREF="../staff.html">UCSC Genome
                    Bioinformatics Group</A>, University of California,
                    Santa Cruz, CA, USA
                    </UL>
                    <P>
                    <HR>
                    These data have been freely provided by Agencourt
                    Bioscience before
                    publication for use in the UCSC Genome Browser with the
                    following understanding:
                    <OL>
                    <LI>
                    The data may be freely downloaded, used in analyses, and
                    repackaged in databases.
                    <LI>
                    Users are free to use the data in scientific papers
                    analyzing particular genes and regions if the
                    provider of these data is properly acknowledged.
                    <LI>
                    Agencourt Bioscience reserves the right to publish the initial
                    large-scale analyses of the dataset, including large-scale
                    identification of regions of evolutionary conservation and
                    large-scale genomic assembly.
                    <LI>
                    Any redistribution of the data should carry this notice.
                    </OL>
                </TD><TD WIDTH=15></TD></TR>
            </TABLE>
        </TD></TR>
    </TABLE>
    <BR>

<!-- D. yakuba Credits================================================== -->
<A NAME="droYak_credits"></A>
<A NAME="droYak_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>D. yakuba</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>These data were provided by The Genome Institute at
		    Washington University (WUSTL) with the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Funding:</B>
		    <A HREF="http://www.genome.gov/" TARGET="_blank">National Human
		    Genome Research Institute</A> (NHGRI)
		    <LI>
		    <B>Sequencing, assembly and AGP creation:</B>
		    <A HREF="http://genome.wustl.edu/genomes/detail/drosophila-yakuba/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>
		    <LI>
		    <B>Initial Sequencing Proposal, Project Collaboration:</B>
		    David J. Begun and Charles H. Langley, 
		    <A HREF="http://www-eve.ucdavis.edu/" 
		    TARGET="_blank">University of California, Davis</A>, CA, USA
		    <LI>
		    <B>UCSC <em>D. yakuba</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droYak2" target="_blank">droYak2</a>)</B>:
		    Angie Hinrichs, Jennifer Jackson and Donna Karolchik
		    <LI>
		    <B>UCSC <em>D. yakuba</em> Genome Browser (<a href="../../cgi-bin/hgGateway?db=droYak1" target="_blank">droYak1</a>)</B>:
		    Angie Hinrichs, Michael Chalup and Donna Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations</B>:
		    <A HREF="../staff.html">UCSC Genome 
		    Bioinformatics Group</A>, University of California,
		    Santa Cruz, CA, USA
		    </UL>
		    <P>
		    <HR>
		    The <em>D. yakuba</em> sequence is made freely available 
		    before scientific publication by The Genome Institute at
		    WUSTL with the following understanding: 
		    <P>
		    <OL>
		    <LI>
		    The data may be freely downloaded, used in analyses, and 
		    repackaged in databases. 
		    <LI>
		    Users are free to use the data in scientific papers analyzing 
		    particular genes and regions if the provider of these data 
		    is properly acknowledged.
		    <LI>
		    The <em>Drosophila yakuba</em> analysis group is aiming to 
		    publish an initial analysis of the <em>D. yakuba</em> genome 
		    sequence in 2005 (submitted in early 2005) that will include 
		    descriptions of the assembly, genome landscape, comparative 
		    analysis and initial gene content. People who would like to 
		    coordinate other genome-wide analysis with this work should 
		    contact <A HREF="mailto:rwilson@watson.wustl.edu">Richard K.
		    Wilson</A>, Genome Sequencing Center Director, Washington 
		    University School of Medicine. We welcome a coordinated 
		    approach to describing this community resource. 
		    <LI>
		    Any redistribution of the data should carry this notice. 
		    </OL>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- C. elegans Credits================================================== -->
<A NAME="ce_credits"></A>
<A NAME="ce_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; <em>C. elegans</em> Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The <em>C. elegans</em> data was obtained from 
		    <A HREF="http://www.wormbase.org/" 
		    TARGET="_blank">WormBase</A>. We'd like to thank
		    <A HREF="http://genome.wustl.edu/" 
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A> and the 
		    <A HREF="http://www.sanger.ac.uk/" 
		    TARGET="_blank">Sanger Institute</A> for their collaborative 
		    work on sequencing the <em>C. elegans</em> genome, and the 
		    WormBase consortium for providing access to the current 
		    <em>C. elegans</em> sequence. </P>
		</TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Misc Worm Credits================================================== -->
<A NAME="worm_credits"></A>
<A NAME="worm_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Worm Genomes (other than <em>C. elegans</em>)</B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>The latest versions of the <em>C. brenneri</em>, 
		    <em>C. briggsae</em>, <em>C. japonica</em>, 
		    <em>C. remanei</em>, and <em>P. pacificus</em> sequences 
		    were obtained from the Genome Sequencing Center at 
		    <A HREF="http://genome.wustl.edu/genomes/category/invertebrates/"
		    TARGET="_blank">The Genome Institute at Washington University (WUSTL)</A>.
		    The cb1 browser data
		    were obtained from <A HREF="http://www.wormbase.org/" 
		    TARGET="_blank">WormBase</A>.</P>
		    <P>
		    We'd like to thank WUSTL for providing the sequence data
		    for these assemblies, as well as the Sanger Institute for
		    their collaborative work in sequencing the initial 
		    <em>C. briggsae</em> cb1 genome. Thanks also to the 
		    WormBase consortium for providing access to the cb1 
		    sequence. 
		    </P>
		    <HR>
                    The worm sequences are made freely available before
                    scientific publication with the following understanding:
                    <OL>
                        <LI>
                        The data may be freely downloaded, used in analyses, and
                        repackaged in databases.
                        <LI>
                        Users are free to use the data in scientific papers
                        analyzing particular genes and regions if the provider
                        of these data is properly acknowledged.
                        <LI>
                        The centers producing the data reserve the right to
                        publish the initial large-scale analyses of the data set,
                        including large-scale identification of regions of
                        evolutionary conservation and large-scale genomic
                        assembly. Large-scale refers to regions with size on the
                        order of a chromosome.
                        <LI>
                        This is in accordance with, and with the understandings
                        in the <A HREF="ftLauderdale.pdf">Fort Lauderdale meeting</A>
                        discussing Community Resource Projects and the resulting
                        <A HREF="http://www.genome.gov/page.cfm?pageID=10506537"
                        TARGET="_blank">NHGRI policy statement</A>.
                        <LI>
                        Any redistribution of the data should carry this notice.
                    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Sacchromyces cerevisiae Credits======================= -->
<A NAME="yeast_credits"></A>
<A NAME="yeast_use"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Yeast Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
			<P><b>UCSC <I>Sacchromyces cerevisiae</I> Genome Browser (<a href="../../cgi-bin/hgGateway?db=sacCer3" target="_blank">sacCer3</a>):</b> Hiram Clawson, 
			Greg Roe, Steve Heitner.<br>
		    The April 2011 <I>Sacchromyces cerevisiae</I> genome assembly 
		    is based on sequence dated April 2011 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET="_blank">Sacchromyces Genome
		    Database</A> (SGD). This genome sequence was downloaded from
			<a href="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/fungi/Saccharomyces_cerevisiae/R64-1-1/"
			TARGET="_blank">NCBI</a> (genbank/genomes/Eukaryotes/fungi/Saccharomy
			ces_cerevisiae/SacCer_Apr2011/). The SGD&#8482; gene annotations were downloaded 
			from the <A HREF="http://downloads.yeastgenome.org/curation/chromosomal_feature/" TARGET="_blank"
			>SGD downloads site</A>. The S288C strain was used in this sequencing project.
		    </P>
		    
		    <P>
		    <b>The June 2008 <I>Sacchromyces cerevisiae</I> genome assembly
		    (<a href="../../cgi-bin/hgGateway?db=sacCer2" target="_blank">sacCer2</a>)</b> 
		    is based on sequence dated June 2008 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET="_blank">Sacchromyces Genome
		    Database</A> (SGD) and was obtained from the site
		    http://downloads.yeastgenome.org/sequence/genomic_sequence/chromosomes/fasta/
		    The S288C strain was used in this sequencing project.
		    </P>
		    <P>
		    <b>The Oct. 2003 <I>Sacchromyces cerevisiae</I> genome assembly
		    (<a href="../../cgi-bin/hgGateway?db=sacCer1" target="_blank">sacCer1</a>)</b> 
		    is based on sequence dated 1 Oct. 2003 in the 
		    <A HREF="http://www.yeastgenome.org/" TARGET="_blank">Sacchromyces Genome
		    Database</A> (SGD). The sequence, open reading frame (ORF), 
		    and gene annotations were downloaded from the site
		    <A HREF="http://www.yeastgenome.org/download-data/sequence"
		    TARGET="_blank">http://www.yeastgenome.org/download-data/sequence</A>. 
		    The cellular localization and protein abundance data displayed
		    in the Gene Sorter are taken from the
		    <A HREF="http://yeastgfp.yeastgenome.org/" TARGET="_blank">Yeast GFP 
		    Fusion Localization Database</A>.
		    <P>See the SGD 
		    <A HREF="http://www.yeastgenome.org/cache/chromosomes.shtml"
		    TARGET="_blank">Systematic Sequencing Table</A> for credit and
		    contact information for each of the chromosomes in the 
		    assembly. We'd like to thank Stanford University, the SGD,
		    the University of California San Francisco (UCSF), 
		    Washington University in St. Louis, and The Broad 
		    Institute 
		    for providing the data and annotations for this release.
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- California Sea Hare Credits======================= -->
<A NAME="seahare_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; California Sea Hare Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <P>
		    The <I>Aplysia californica</I> genome assembly 
		    has the following acknowledgments:
		    <UL>
		    <LI>
		    <B>Sequencing/Assembly:</B>
		    <A HREF="http://www.broadinstitute.org/"
		    TARGET="_blank">The Broad Institute</A>, Cambridge, MA, USA
		    <LI>
		    <B>UCSC California Sea Hare (<a href="../../cgi-bin/hgGateway?db=aplCal1" target="_blank">aplCal1</a>) Genome Browser:</B>
		    Galt Barber, Brian Raney, Hiram Clawson, Katrina Learned and Donna 
		    Karolchik
		    <LI>
		    <B>Initial Genome Browser Annotations:</B> 
		    <A HREF="../staff.html">UCSC Genome Bioinformatics Group</A>, 
		    University of California, Santa Cruz, CA, USA 
		    </UL>
		    <P>
		    <HR>
		    <A NAME="seahare_use"></A>
		    The California sea hare sequence is made freely available 
		    before 
		    scientific publication with the following understanding:
		    <OL>
			<LI>
			The data may be freely downloaded, used in analyses, and
			repackaged in databases.
			<LI>
			Users are free to use the data in scientific papers 
			analyzing particular genes and regions if the provider
			of these data is properly acknowledged.
			<LI>
			The center producing the data reserves the right to 
			publish the initial large-scale analyses of the data 
			set, including large-scale identification of regions of 
			evolutionary conservation and large-scale genomic 
			assembly. Large-scale refers to regions with size on the
			order of a chromosome (that is, 30 Mb or more). 
			<LI>
			Any redistribution of the data should carry this notice.
		    </OL></P>
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Ebola Virus Credits======================= -->
<A NAME="ebola_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
  	<TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
		<TR><TD><FONT SIZE="4"><B>&nbsp; Ebola Virus Genome </B></FONT>
		</TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>
		<TR><TD WIDTH=10></TD><TD>
		    <p>
		    The Ebola virus genome assembly
		    has the following acknowledgments:
		    <ul>
		    <li>
		    <b>Sequencing/Assembly:</b>
                    <a href="http://www.broadinstitute.org/"
                    target="_blank">The Broad Institute</a>, Cambridge, MA, USA
		    <li>
		    <b>UCSC Ebola virus (<a href="../../cgi-bin/hgGateway?db=eboVir3" target="_blank">eboVir3</a>) Genome Browser:</b>
		    Hiram Clawson, Brian Raney, Maximillian Haeussler, Angie Hirichs, Kate
		    Rosenbloom, Chris Eisenhart, Pauline Fujita, Matthew Speir, Donna
		    Karolchik, Ann Zweig, David Haussler, and Jim Kent
		    <li>
		    <b>Initial Genome Browser Annotations:</b>
		    <a href="../staff.html">UCSC Genome Bioinformatics Group</a>,
		    University of California, Santa Cruz, CA, USA
		    </ul>
		    <b>Other Acknowledgements:</b>
		    <p>
		    Thank you to
		    <a href="http://cbse.soe.ucsc.edu/people/berman"
		    target="_blank">Phillip Berman</a> and his lab at UCSC for providing continued
		    feedback on the Ebola virus Genome Browser, and to the
		    <a href="http://www.sabetilab.org/" target="_blank">Pardis Sabeti Lab</a> at
		    the Broad Institute for their guidance.
		    </p>
		    
		    <p>
		    Thank you to the following institutions for proving additional annotations
		    on the Ebola virus genome:
		    <ul>
		    <li><a href="http://www.ncbi.nlm.nih.gov/" target="_blank">National Center
		    for Biotechnology Information</a> (NCBI)</li>
		    <li><a href="http://www.iedb.org/" target="_blank">Immune Epitope
		    Database</a> (IEDB)</li>
		    <li><a href="http://www.uniprot.org/" target="_blank">UniProt</a></li>
		    <li><a href="http://www.rcsb.org/pdb/home/home.do" target="_blank">RCSB
		    Protein Data Bank</a> (PDB)</li>
		    </ul></ul></p>
		    <p>
		    <hr>
		    <a name="ebola_use"></a>
		    The Ebola virus sequence may be freely downloaded, used in analyses,
		    and repackaged in databases. The data may be freely used in scientific
		    papers analyzing particular genes and regions if the provider of these
		    data is properly acknowledged.
		    </p>
		</TD><TD WIDTH=15></TD></TR>
	     </TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Other Credits ====================================================== -->
<A NAME="other_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD><FONT SIZE="4"><B>&nbsp; External Contributors
		to the UCSC Genome Browser Project</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	    <TH HEIGHT=10></TH>
		<TR><TD COLSPAN=2></TD></TR>
	 	<TR><TD COLSPAN=2>
		    <UL>
			<LI>
			<A HREF="http://www.affymetrix.com/estore/"
			TARGET="_blank">Affymetrix</A>: Consensus and exemplar
			sequences for probe selection on several chips, 
			exon array data, transcriptome data, Genie Gene 
			predictions, 
			design and analysis support in the early stages of the 
			UCSC Human Genome project, including software for the 
			synthetic chromosome 22 data we used to test our 
			assembly methods.
			<LI>
			<A HREF="http://www.genomics.agilent.com/en/home.jsp"
TARGET="_blank">Agilent Technologies</A> 
			(Shane Giles, Peter Webb, and Dione Bailey): Agilent 
			microarray probe sets.
			<LI>
			<A HREF="http://www.alleninstitute.org/"
TARGET="_blank">Allen Institute 
			for Brain Science</A>: Allen Brain Atlas probes.
			<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/genome/cyto/hbrc.shtml" TARGET="_blank">BAC Resource
			Consortium</A>: Cytogenetic markers (FISH-mapped clones annotation track).
			<LI>
			<A HREF="https://www.hgsc.bcm.edu/" TARGET="_blank">Baylor College of Medicine
			Human Genome Sequencing Center (BCM-HGSC)</A>
			(Richard Gibbs, John Bouck, Kim Worley and James Durbin): Rat Genome
			sequence data, <em>D. pseudoobscura</em> sequence for alignment tracks
			with <em>D. melanogaster</em> genome.
			<LI>
			<A HREF="http://www.fruitfly.org/" TARGET="_blank">Berkeley Drosophila Genome
			Project</A> (BDGP): Data for <em>D. melanogaster</em> gene annotations.
			<LI>
			<A HREF="http://www.biobase-international.com/"
			TARGET="_blank">BioBase</A>: Transfac database data used
			in HMR Conserved Transcription Factor Binding Sites
			track.
			<LI>
			<A HREF="http://www.bioinfor.com/" TARGET="_blank">
			Bioinformatics Solutions, Inc.</A>
			(Ming Li and colleagues): Pattern Hunter mouse homology track.
			<LI>
			<A HREF="http://biomarkersconsortium.org/"
			TARGET="_blank">Biomarkers Consortium</A>: I-SPY 
			collaboration on the UCSC Cancer Genome Browser
			project.
			<LI>
			<A HREF="http://h-invitational.jp/hinv/ahg-db/index.jsp"
			TARGET="_blank">Biomedicinal Information Research Center 
			(BIRC)</A>, 
			National Institute of Advanced Industrial Science and 
			Technology (AIST): H-InvDB Gene Predictions.
			<LI>
			<A HREF="http://www.bu.edu/chemistry/faculty/tullius/" 
		        TARGET="_blank">Boston University</A> 
			(Tullius Lab, in collaboration with MIT,
			NHGRI, and the La Jolla Institute for Allergy and 
			Immunology): ORChID Predicted DNA Cleavage 
			Sites (ENCODE).
			<LI>
			<A HREF="http://www.broadinstitute.org/"
			TARGET="_blank">The Broad Institute</A>
			<BR>--(Eric Lander, Mike Zody, Chad Nusbaum): Plasmid end reads. Human working
			draft sequence analysis project. Special thanks to Eric 
			for inviting us to join the Hardcore Analysis Group 
			for the draft sequence project and for 
			his direction of that group. 
			<BR>--(<A HREF="http://bernstein.mgh.harvard.edu/"
			TARGET="_blank">Bernstein Lab</A>): Histone modifications
			using ChIP-seq (ENCODE).
			<LI>
			<A HREF="http://woldlab.caltech.edu/" 
			TARGET="_blank">California Institute of Technology</A> 
			(Wold Lab): RNA-Seq data (ENCODE).
			<LI>
			<A HREF="http://www.bcgsc.ca/"
			TARGET="_blank">Canada's Michael Smith Genome Sciences
			Centre</A>: ORegAnno data, brain DNA methylation track 
			(ENCODE).
			<LI>
			<A HREF="http://cedars-sinai.edu"
			TARGET="_blank">Cedars-Sinai Medical Center</A> (CSMC):
			FISH-mapped clones annotation track. 
			<LI>
			<A HREF="http://projects.tcag.ca/variation/" 
			TARGET="_blank">Centre for Applied Genomics</A>: Database 
			of Genomic Variants (DGV) data.
			<LI>
			<A HREF="http://www.cephb.fr/en/index.php"
			TARGET="_blank">Centre d'Etude du Polymorphisme Humain</A>
			(CEPH): Data from Genotype database for SNP Population
			Allele Frequencies track.
			<LI>
			<A HREF="http://pasteur.crg.es/portal/page/portal/Internet/02_Research/01_Programmes/01_Bioinformatics_Genomics"
			TARGET="_blank">Centre for Genomic Regulation (CRG)</A> 
			(Guigo Lab): CRG mappability (alignability) data,
			RNA-seq data.
			<LI>
			<A HREF="https://www-lbme.biotoul.fr/" TARGET=blank>CNRS 
			Universit&eacute; Paul Sabatier</A>, Laboratoire de Biologie
			Mol&eacute;culaire Eucaryote, Toulouse, 
			France (Michel Weber): miRNA Data annotation track (in collaboration with Sam 
			Griffiths-Jones of The Wellcome Trust Sanger Institute).
			<LI>
			<A HREF="http://www.cshl.org/" TARGET=blank>Cold 
			Spring Harbor Laboratory</A> (CSHL)
			<BR>--(Stein Lab): HapMap data.
			<BR>--(<A HREF="http://gingeraslab.cshl.edu/" 
			TARGET="_blank">Gingeras Lab</A> and
			<A HREF="http://hannonlab.cshl.edu/"
			TARGET="_blank">Hannon Lab</A>): RNA-seq data
			tracks (ENCODE). 
			<BR>--(<A HREF="http://rulai.cshl.org/" 
			TARGET="_blank">Ivo Grosse and Michael Q. Zhang</A>): 
			Predictions for First Exon Finder annotation track.
			<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi"
			TARGET="_blank">Consensus Coding Sequence Project</A> 
			(EBI, NCBI, WTSI, UCSC): Consensus CDS Genes.
		    	<LI>
			<A HREF="http://compgen.bscb.cornell.edu/" 
			TARGET="_blank">Cornell University (Siepel Lab)</A>: 
			Exoniphy.
			<LI>
			<A HREF="http://www.genome.duke.edu/index.php" 
			TARGET="_blank">Duke University</A>
			(Institute for Genome Sciences and Policy): 
			Open chromatin track, Human tissue Affymetrix Exon 
			expression array data (ENCODE).
			<LI>
			<A HREF="http://www.embl.org/" TARGET="_blank">EMBL</A> (Peter Bork): Mouse pseudogene
			predictions used for Bork Pseudogenes custom annotation track.
			<LI>
			<A HREF="http://genetics.emory.edu/"
			TARGET="_blank">Emory University School of Medicine</A> 
			(D.  Ledbetter,
			C. Martin): Collaboration on CGH-ICSA grant. 
			<LI>
			<A HREF="http://www.ensembl.org/" 
			TARGET="_blank">Ensembl</A> (Paul Flicek,  Guy Slater, 
			Imre Vastrik, Arek Kasprzyk, Michele Clamp): Ensembl 
			Gene predictions, 
			data exchange formats, 
			TPF clone map coordination, 
			exonerate mouse homology track, early help placing FISH-mapped clones. 
			<LI>
			<A HREF="http://www.ieo.it/en/"
			TARGET="_blank">European Institute of Oncology</A>:
			Nuclease accessible sites (ENCODE).
			<LI>
			<A HREF="http://flybase.org/" TARGET="_blank">FlyBase</A>: Curated Drosophila
			gene annotations.
			<LI>
			<A HREF="http://www.fredhutch.org/en.html" TARGET="_blank">Fred Hutchison Cancer Research Center</A> (FHCRC)
			(Barbara Trask): FISH-mapped clones annotation track.
			<LI>
			<A HREF="http://geneontology.org/" TARGET="_blank">Gene Ontology (GO) Consortium</A>:
			Structured annotations for Gene Sorter.
			<LI>
			<A HREF="http://www.girinst.org/"
			TARGET="_blank">Genetic Information Research Institute</A>
			(GIRI): Repbase repeat library, repeatmasking tools.
			<LI>
			<A HREF="http://big.crg.cat/computational_biology_of_rna_processing" TARGET="_blank">Genome Informatics Research Lab</A> 
			(GIRL) within Grup de Recerca en Inform&agrave;tica M&egrave;dica (GRIB)
			(Francisco C&acirc;mara Ferreira, 
			Roderic Guig&oacute;): Geneid and SGP gene prediction annotation tracks. 
			<LI>
			<A HREF="http://www.gnf.org/" TARGET="_blank">Genomics Institute of the Novartis
			Research Foundation</A> (GNF): Expression data using Affymetrix chips for GNF Expression
			Atlas annotation tracks.
			<LI>
			<A HREF="http://www.gis.a-star.edu.sg/internet/site/" 
			TARGET="_blank">Genome Institute of Singapore</A>
			(GIS): DNA Paired-end Ditag data, RNA-seq data (ENCODE).
			<LI>
			<A HREF="http://www.genoscope.cns.fr/spip/spip.php?lang=en" TARGET="_blank">Genoscope</A>
			(Jean Weissenbach and Olivier Jaillon): Exofish program, <em>Tetraodon nigroviridis</em> 
			sequence for Tetraodon Blat alignments annotation tracks, data for Anopheles
			Evolutionary Conserved Regions annotation track.
			<LI>
			Helicos BioSciences: RNA-seq data 
			(ENCODE).
			<LI>
			<A HREF="http://www.sickkids.ca/"
			TARGET="_blank">Hospital for Sick Children</A> (Sick 
			Kids) (L. Feuk): Structural variation data.
			<LI>
			<A HREF="http://research.hudsonalpha.org/Myers/"
			TARGET="_blank">HudsonAlpha Institute for Biotechnology</A> (Myers Lab): 
		        RNA-seq data (ENCODE),
			Methyl-seq data (ENCODE), transcription factor binding 
			sites identified by ChIP-seq (ENCODE), copy number
			variation (CNV) data (ENCODE). 
			<LI>
			<A HREF="http://www.illumina.com/" 
			TARGET="_blank">Illumnia</A>: Probe set alignments.
			<LI>
			Institute for Genomic Research (TIGR): TIGR Gene Index 
			annotation track. 
			<LI>
			<A HREF="http://www.systemsbiology.org" TARGET="_blank">Institute for Systems Biology</A>
			(Arian Smit): <A HREF="http://www.repeatmasker.org" 
			TARGET="_blank">RepeatMasker</A> repeat identification and masking software and libraries.
			<LI>
			<A HREF="http://www.mousephenotype.org/martsearch_ikmc_project/about"
			TARGET="_blank">International Knockout Mouse 	
			Consortium</A>: Data for IKMC tracks.
			<LI>
			Iowa State University (J.  Murray): Collaboration on 
			<A HREF="https://www.facebase.org/node/68"
			TARGET="_blank">FaceBase</A> project.
			<LI>
			<A HREF="http://www.jcvi.org/cms/home/"
			TARGET="_blank">J. Craig Venter Institute</A>:
			Nuclease accessible sites (ENCODE).
			<LI>
			<A HREF="http://www.informatics.jax.org/" 
			TARGET="_blank">Jackson Laboratory</A>: 
			Mouse Genome Informatics tracks, 
			Mouse ortholog links from known genes track. 
			<LI>
			<A HREF="http://www.genome.jp/" TARGET="_blank">Kyoto University</A>, GenomeNet: 
			<A HREF="http://www.genome.jp/kegg/kegg2.html" TARGET="_blank">Kyoto Encyclopedia of 
			Genes and Genomes</A> (KEGG) - pathway diagrams for Known Genes annotation track.
			<LI>
			<A HREF="http://jgi.doe.gov/about-us/organization/genomic-technologies/len-pennacchio/"
			TARGET="_blank">Lawrence Berkeley National Laboratory</A> 
			(L.A. Pennacchio): Vista HMR-Conserved
			non-coding human enhancers track.
			<LI>
			<A HREF="http://www.lanl.gov/" TARGET="_blank">Los Alamos National Laboratory</A> (LANL):
			FISH-mapped clones annotation track. 
		 	<LI>
			<A HREF="http://genes.mit.edu/burgelab/" 
			TARGET="_blank">Massachusetts Institute of Technology
			(MIT)</A> (Chris Burge): RNA-Seq data, Genscan gene 
			prediction program. 
			<LI>
			<A HREF="http://rgd.mcw.edu/" TARGET="_blank">Medical
			College of Wisconsin</A>: Rat Genome Database (RGD).
		 	<LI>
			<A HREF="http://www2.mrc-lmb.cam.ac.uk/" 
			TARGET="_blank">Medical Research Council Laboratory 
			of Molecular Biology</A>: 
			<A HREF="http://supfam.cs.bris.ac.uk/SUPERFAMILY/"
			TARGET="_blank">Superfamily</A> database.
			<LI>
			<A HREF="http://www.mountsinai.org/" TARGET="_blank">
			Mount Sinai School of Medicine</A> 
			(<A HREF="http://tandem.bu.edu/benson.html" 
			TARGET="_blank">Gary Benson</A> - now at Boston 
			University): Tandem Repeats Finder.
			<LI>
			<A HREF="http://www.cancer.gov/"
			TARGET="_blank">National Cancer Institute</A> (NCI), NIH: 
			<BR>--<A HREF="http://cancergenome.nih.gov/"
			TARGET="_blank">The Cancer Genome Atlas</A> (TCGA).
			<BR>--<A HREF="http://mgc.nci.nih.gov" 
			TARGET="_blank">Mammalian Gene Collection (MGC) 
			Project</A>. 
			<BR>--<A HREF="http://cgap.nci.nih.gov/" 	
			TARGET="_blank">Cancer Genome Anatomy Project</A> (CGAP). 
			<BR>--SAGE project (in collaboration with NCBI).
			<BR>--FISH-mapped clones annotation track. 
		 	<LI>
			<A HREF="http://www.ncbi.nlm.nih.gov/index.html" 
			TARGET="_blank">National Center for Biotechnology
			Information</A> (NCBI), NIH 
			<BR>--<A HREF="http://www.ncbi.nlm.nih.gov/refseq/" 
			TARGET="_blank">RefSeq</A>, 
			<A HREF="http://www.ncbi.nlm.nih.gov/gene"
			TARGET="_blank">Entrez Gene</A>,
			<A HREF="http://www.ncbi.nlm.nih.gov/SNP/" 
			TARGET="_blank">dbSNP</A>, 
			<A HREF="http://hapmap.ncbi.nlm.nih.gov/index.html.en"
			TARGET="_blank">HapMap</A>,
			<A HREF="http://www.ncbi.nlm.nih.gov/omim"
			TARGET="_blank">OMIM</A>, and 
			<A HREF="http://www.ncbi.nlm.nih.gov/pubmed"
			TARGET="_blank">PubMed</A> databases.
			<BR>--<A HREF="http://www.ncbi.nlm.nih.gov/SAGE/" 
			TARGET="_blank">Serial Analysis of
			Gene Expression Tag to Gene Mapping</A> (SAGEmap) data resource.
			<BR>--(Deanna Church): Mouse synteny track, database of 
			whole genome shotgun mouse reads and assemblies used for
			mouse tracks.
			<BR>--(Greg Schuler): GenBank freezes of human clones 
			used in the assembly, NT contigs of multiple 
			finished clones, STS maps (in earlier freezes), 
			specially-formatted versions of the finished 
			chromosomes. 
			<BR>--(Steve Sherry): SNP tracks. 
			<BR>--(Wonhee Jang): FISH-mapped clones annotation 
			track.
			<BR>--(Danielle and Jean Thierry-Mieg): Acembly Genes 
			annotation track.
			<BR>--(Lukas Wagner): 3' EST track in the July 2000 
			browser.
			<LI>
			<A HREF="http://www.genome.gov" TARGET="_blank">National 
			Human Genome Research Institute</A> 
			(NHGRI), NIH
			<BR>--(<A HREF="http://www.genome.gov/12514761"
			TARGET="_blank">Elnitski Lab</A>): Bidirectional 
			promoters, negative regulatory elements (ENCODE).
			<BR>--(ENCODE group): Guidance for ENCODE DCC, 
			collaboration and contributions for various ENCODE 
			annotation tracks.
			<BR>--<A HREF="http://www.genome.gov/gwastudies/"
			TARGET="_blank">GWAS</A>: GWAS data from the Catalog 
			of Published Genome-Wide Association Studies.
			<BR>--(NISC group): Collaboration and annotations
			for <A HREF="http://www.nisc.nih.gov/index.html" 
			TARGET="_blank">NISC</A> browser, including 
			curated gene annotations, DNAse I Hypersensitive Sites 
			annotation, Multi-species Conserved Sequences (MCS) 
			tracks, Binomial_25 Conservation Scores,
			ZooSeq Blastz alignments tracks. 
			<LI>
			<A HREF="http://www.nia.nih.gov/" 
			TARGET="_blank">National Institute on Aging</A>, NIH:
			Data from the 
			<A HREF="http://geneticassociationdb.nih.gov/"
			TARGET="_blank">Genetic Association Database</A>.
			<LI>
			<A HREF="http://research.nki.nl/vansteensellab/"
			TARGET="_blank">Netherlands Cancer Institute</A> (van Steensel Lab):
			Nuclear lamina associated domains tracks.
			<LI>
			<A HREF="http://bmi.osu.edu/" TARGET="_blank">Ohio State
			University Bioinformatics & Computational Biology Group</A> (Ramana V. Davuluri): 
			Predictions for First Exon Finder annotation track.
			<LI>
			<A HREF="http://www.orfeomecollaboration.org/"
			TARGET="_blank">ORFeome Collaboration</A>: ORFeome
			Collaboration human gene clones.
			<LI>
			<A HREF="http://www.bx.psu.edu/" 
			TARGET="_blank">Pennsylvania State University 
			Center for Comparative Genomics and 
			Bioinformatics</A>: BLASTZ, LASTZ, and multiz alignment 
			programs, Genome Alignment and Annotation Database 
			(GALA), collaboration on multiple alignment annotation 
			tracks, ENCODE annotation tracks, Personal Genome 
			Variants track, Galaxy, regulatory potential scores 
			tracks.
			<LI>
			Perlegen Sciences: STS markers for 
			the Haplotype Blocks annotation track on chr 21,
			SNP data for the Affymetrix 120K SNP track, 
			TNG radiation hybrid map.
			<LI>
			<A HREF="http://www.gsc.riken.go.jp" 
			TARGET="_blank">RIKEN
			Genomic Sciences Center</A>: Chimpanzee BAC end sequence
			data, mouse annotations, human genome sequence 
			contributions, CAGE tags RNA subcellular localization
			data (ENCODE).
			<LI>
			<A HREF="https://www.roswellpark.org/" 
			TARGET="_blank">Roswell Park Cancer Institute</A> (RPCI):
			FISH-mapped clones annotation track.
			<LI>
			SNP Consortium: Data for SNP annotation tracks.
			<LI>
			Softberry Inc. (V. Solovyev):
			Fgenesh++ and Fgenesv+ gene predictions, TSSW promoters tracks.
			<LI>
			<A HREF="http://www.mad-cow.org/" 
			TARGET="_blank">Sperling Biomedical 
			Foundation</A> (Thomas Pringle): Technical consultant on
			Proteome Browser, avid Genome Browser user and the 
			source of myriad ideas for expanding its functionality, 
			early assistance and support for UCSC Genome Browser.
			<LI>
			<A HREF="http://bejerano.stanford.edu/"
			TARGET="_blank">Stanford University</A> 
			<BR>--(<A HREF="http://bejerano.stanford.edu/"
			TARGET="_blank">Bejerano Lab</A>): GREAT tool.
			<BR>--(
			<!-- Link is currently broken, can uncomment when it is fixed
			<A HREF="http://www.stanford.edu/group/morrinst/abi.html"
			TARGET="_blank">Morrison Institute for Population
			and Resource Studies</A>-->
			Morrison Institute for Population
                        and Resource Studies, Human Genome Diversity 
			Project): Data for SNP Population Allele Frequencies 
			track.
			<BR>--<A HREF="http://source.stanford.edu/"
			TARGET="_blank">SOURCE tool</A>: SOURCE links from UCSC
			Genes tracks.
			<LI>
			Stanford University School of 
			Medicine (Myers Lab - now at HudsonAlpha): Data set for 
			Human Promoters custom annotation track.
			<LI>
			<A HREF="http://switchgeargenomics.com/"
			TARGET="_blank">SwitchGear Genomics</A> (N. Trinklein 
			and S.F. Aldred): Transcription start sites (TSSs).
			<LI>
			<A HREF="http://www.uniprot.org/" 
			TARGET="_blank">UniProt</A>: Protein data from UniProt 
			database.
			<LI>
			<A HREF="http://bioinfo2.ugr.es/isochores/" 
			TARGET="_blank">Universidad de Granada</A>, 
			Grupo de Bioinform&aacute;tica, Spain (Jose L. Oliver): 
			Isochore custom annotation track.
			<LI>
			<A HREF="http://www.sunycnse.com/AboutUs/FacultyStaff/Faculty/ScottTenenbaum.aspx"
			TARGET="_blank">University at Albany - SUNY</A> 
			(TenenBaum Lab): Target RNA binding protein 
			expression measured by RIP-chip (ENCODE).
			<LI>
			<A HREF="http://qb3.berkeley.edu/ccb/"
			TARGET="_blank">University of CA, Berkeley</A> (Lior
			Pachter, Simon Cawley, Marina Alexandersson): 
			<A HREF="http://bio.math.berkeley.edu/slam/mouse/" 
			TARGET="_blank">SLAM</A> Gene predictions.
			<LI>
			<A HREF="http://bioinformatics.ucsd.edu/"
			TARGET="_blank">University of CA, San Diego</A> 
			<BR>--(<A HREF="http://www.snl.salk.edu/~geneyeo/"
			TARGET="_blank">G. Yeo</A>, now at Salk Institute): 
			FOX2 adaptor-trimmed CLIP-seq reads track.
			<BR>--(<A HREF="http://bioinformatics-renlab.ucsd.edu/rentrac/"
			TARGET="_blank">Ren Lab</A>, Ludwig Institute for Cancer
			Research): TAF1 binding in fibroblastoid cells assayed
			by ChIP-chip (ENCODE).
			<LI>
			<A HREF="http://www.ucsf.edu/" 
			TARGET="_blank">University of CA, San Francisco</A>:
			Collaboration on UCSC Cancer Browser, 	
			Chimera links from
			SNPs and UCSC Genes, FISH-mapped clones annotation 
			track.
			<BR>--(ENCODE group): Brain DNA methylation track 
			(ENCODE).
			<LI>
			<A HREF="http://bme.ucsc.edu/"
			TARGET="_blank">University of CA, Santa Cruz</A> 
			<BR>--(<A HREF="http://bme.ucsc.edu/people/ed"
			TARGET="_blank">R.E. Green</A>): Neanderthal annotation 
			tracks.
			<BR>--(<A HREF="http://lowelab.ucsc.edu/"
			TARGET="_blank">Lowe Lab</A>): tRNA Gene Predictions 
			using tRNAscan-SE.  
			<LI>
			<A HREF="http://www.oncology.cam.ac.uk/"
			TARGET="_blank">University of Cambridge, Department of
			Oncology</A>: Open chromatin track (ENCODE).
			<LI>
			<A HREF="http://pritchardlab.stanford.edu/home.html"
			TARGET="_blank">Stanford University</A> (Pritchard Lab):
			Collaboration on Human Genome Diversity Project tracks.
			<LI>
			<A HREF="http://genes.uchicago.edu/contents/faculty/lieb-jason-ph-d.html"
			TARGET="_blank">University of North Carolina, 
			Chapel Hill</A> (Lieb Lab, now at the University of Chicago): Open chromatin track 
			(ENCODE).
			<LI>
			<A HREF="http://www.allgenes.org/" TARGET="_blank">
			University of Pennsylvania</A> (Y. Thomas Gan):
			Database of Transcribed Sequences (DoTS) Genes custom annotation track.
			<LI>
			University of Pittsburgh (M. Marazita): 
			Collaboration on 
			<A HREF="https://www.facebase.org/node/68"
			TARGET="_blank">FaceBase</A> project.
			<LI>
			<A HREF="http://www.genenetwork.org/"
			TARGET="_blank">University of Tennessee GeneNetwork</A>: GeneNetwork 
			links from UCSC Genes annotations.
			<LI>
			<A HREF="http://ccri.uthscsa.edu/LPenalva.asp"
			TARGET="_blank">University of Texas Health Science
			Center</A>, Greehy Children's Cancer Research Institute 
			(L.O.F. Penalva): 
			<LI>
			<A HREF="http://microarray.icmb.utexas.edu/research.html"
			TARGET="_blank">University of Texas, Austin</A> (Iyer 
			Lab): Open chromatin track (ENCODE).
			<LI>
			<A HREF="http://www.tbi.univie.ac.at/index.html" 
			TARGET="_blank">University of Vienna</A>, Institute 
			for Theoretical Chemistry and Structural Biology, 
			Theoretical Biochemistry Group</A>: 
			Vienna RNA utility 
			<A HREF="http://www.tbi.univie.ac.at/RNA/"
			TARGET="_blank">RNAplot</A> - Known Genes annotation 
			track. 
			<LI>
			<A HREF="http://www.gs.washington.edu/"
			TARGET="_blank">University of Washington</A>
			<BR>--(<A HREF="http://eichlerlab.gs.washington.edu/" 
			TARGET="_blank">Eichler Lab</A>): Segmental duplications 
			data. 
			<BR>--(<A HREF="http://hgsv.washington.edu/"
			TARGET="_blank">Human Genome Structural Variation 
			Project</A>): Variation data for HGSV tracks.
			<BR>--(ENCODE group): Digital DNase genomic footprinting
			track, DNaseI sensitivity using digital DNaseI
			methodology, DNaseI hypersensitive sites, histone
			modifications using ChIP-seq (ENCODE).
			<BR>--(<A HREF="http://noble.gs.washington.edu/"
			TARGET="_blank">Noble Lab</A>): Predicted nucleosome 
			occupancy tracks.
			<BR>--(<A HREF="http://www.gs.washington.edu/faculty/nickerson.htm"
			TARGET="_blank">Nickerson Lab</A>): Tajima's D SNPs track.
			<LI>
			<A HREF="http://www.klinimm.uu.se/research/genetics_genomics/wadelius/"
			TARGET="_blank">Uppsala University</A> 
			(Wadelius Lab): ChIP-chip data (ENCODE).
			<LI>
			<A HREF="http://www.wustl.edu/" 
			TARGET="_blank">Washington University in St. Louis</A> and
			<A HREF="http://medicine.wustl.edu/"
			TARGET="_blank">Washington University in St. Louis
			School of Medicine</A> 
			<BR>--(Robert Waterston (now at UW), John McPhearson, Asif Chinwalla,
			Shiaw-Pyng Yang, John Wallis and LaDeana Hillier): Fingerprint-based map
			of BAC clones from the human genome and a corresponding layout of sequenced
			clones, plasmid end reads, BACend analysis, CpG islands.
			<BR>--(<A HREF="http://www.hhmi.org/scientists/sean-r-eddy"
			TARGET="_blank">Sean Eddy</A>
			and Todd Lowe (now at UCSC)): RNA Genes track.
			<BR>--(<A HREF="http://mblab.wustl.edu/" TARGET="_blank">Michael 
			Brent Lab</A>): N-SCAN and Twinscan Genes.
			<BR>--(ENCODE group): Brain DNA methylation track 
			(ENCODE).
			<LI>
			<A HREF="https://www.sanger.ac.uk/about/history/hgp/" TARGET="_blank">
			Wellcome Trust Sanger Institute</A>
			<BR>--(J. Harrow, J. Gilbert, J. Sulston, J. Rogers, 
			J. Mullikin, R. Durbin, S. Searle, I. Dunham and 
			the annotation teams): Vega Genes and Pseudogenes, 
			Sanger 20 and Sanger 22 Genes and Pseudogenes. Plasmid 
			end reads, processing of whole genome shotgun mouse 
			reads. Eponine program.
			<BR>--(S. Griffiths-Jones): miRNA Data annotation track 
			(in collaboration with 
			M. Weber of CNRS Universit&eacute; Paul Sabatier).
			<BR>--<A HREF="https://decipher.sanger.ac.uk/"
			TARGET="_blank">Decipher</A>
			<BR>--<A HREF="http://pfam.xfam.org/"
			TARGET="_blank">Protein Families 
			Database of Alignments and HMMs</A> (Pfam).
			<LI>
			<A HREF="http://jura.wi.mit.edu/bio/"
			TARGET="_blank">Whitehead Institute, Bioinformatics and
			Research Computing</A> (G. Bell): TargetScan miRNA 
			regulatory sites.
			<LI>
			<A HREF="http://www.wormbase.org/" TARGET="_blank">WormBase</A>
			(Lincoln Stein): Wormbase curated gene annotations for
			<em>C. elegans</em> genome.
			<LI>
			<A HREF="http://cbb.yale.edu/"
			TARGET="_blank">Yale University</A>: 
			<BR>--(<A HREF="http://cbb.yale.edu/"
			TARGET="_blank">Snyder, Gerstein, and Weissman Labs</A>
			in collaboration with NASA Ames Research Center and 
			Eloret Corporation): 
			Transcriptionally Active Regions (TARs) data (ENCODE).
			<BR>--(<A HREF="http://medicine.yale.edu/lab/sestan/"
			TARGET="_blank">Sestan Lab</A>): Exon microarray expression
			data from the late mid-fetal human brain.
		</UL></TD><TD WIDTH=15></TD></TR></TABLE>
	</TD></TR></TABLE>

	<BR>

<!-- Early Credits ====================================================== -->
<A NAME="early_credits"></A>
    <TABLE BGCOLOR="#fffee8" WIDTH="100%" BORDER=1>
	<TR><TD>
	<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	    <TR><TD>
 	    <FONT SIZE="4"><B>&nbsp; Acknowledgment of Early Support</B></FONT>
	    </TD></TR>
	</TABLE>
	<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	    <TH HEIGHT=10></TH>
	    <TR><TD WIDTH=10></TD>
		<TD>
		    <P>During the initial stage of this project, from January 2000 to June 2001,
		    we received no direct federal funding to develop the working draft other than
		    travel support. However, we gratefully acknowledge the support for our
		    personnel provided by our basic bioinformatics research grants, especially
		    DOE grant DE-FG03-99ER62849 (to develop gene-finding methods), but also NSF
		    grant DBI-9809007, UC BIOTECH grant 99-11, a grant from the Sloan Foundation,
		    the David and Lucille Packard Foundation, NIH grant GM-52848, and the Zahler
		    grant that provided support for Jim Kent in the early stages of this work. 
		    <P>
		    We are particularly thankful for the risks that Patrick Mantey (UCSC Dean of
		    Engineering), James Gill (Vice Chancellor for Research), John Simpson (UCSC
		    Provost), and M.R.C. Greenwood (UCSC Chancellor) were willing to take 
		    in advancing us the money in February 2000 needed to build the original
		    100-node LINUX cluster used in this work, and in the case of Dean Mantey,
		    additional personnel support. Phokion Kolaitis, Chair of Computer Science,
		    contributed as well. 
		    <P>We thank <A HREF="http://www.compaq.com/" TARGET="_blank">Compaq</A> for
		    providing an additional 100-CPU computational cluster in Cambridge, MA for use
		    by the public human genome effort. We thank <A HREF="http://www-01.ibm.com/software/info/ilog/"
		    TARGET="_blank">ILOG, Inc.</A> (now part of IBM) for providing discounted licenses to its CPLEX
		    linear programming software, which was essential for some of our early
		    experiments.</P>
		    <P>
		    In December 2000, <A HREF="http://www.hhmi.org" TARGET="_blank">HHMI</A> 
		    began providing support for equipment and personnel. In the summer of 2001,
		    HHMI provided the primary funding for a new cluster of 1000 CPUs running 
		    LINUX purchased from <A HREF="http://www.sgi.com/" TARGET="_blank">
		    Rackable</A> (now SGI), with additional support from
		    <A HREF="http://www.ucop.edu/california-institutes/" TARGET="_blank">CISI</A>
		    and <A HREF="http://www.genome.gov/" TARGET="_blank">NHGRI</A>. Dubbed the 
		    UCSC KiloKluster, this equipment became the computational backbone
		    for the initial years of the Genome Browser project.   
		    </P>
		</TD>
		<TD WIDTH=15></TD>
	    </TR>
	</TABLE>
	<BR>
	</TD></TR>
    </TABLE>
    </TD></TR>
</TABLE>
</BODY></HTML>
