<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=240>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.hgsc.bcm.tmc.edu/project-species-m-Bovine.hgsc?pageLocation=Bovine"
    TARGET=_blank>
    <IMG SRC="/images/Bos_taurus.jpg" WIDTH=225 HEIGHT=138 ALT="Cow"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT>
    <FONT SIZE=-1><em>Bos Taurus</em><BR>
    </FONT>
    <FONT SIZE=-2>
     Photo courtesy of Michael MacNeil, USDA<BR>
     (NHGRI <A HREF="http://www.genome.gov/10005141" 
     TARGET=_blank>Press Photos</A>)
    </FONT>
  </TD></TR>
</TABLE>



<P>
The Nov. 2009 <em>Bos taurus</em> draft assembly 
(UMD_3.1 (NCBI project 
<A
HREF="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=genomeprj&cmd=ShowDetailView&TermToSearch=10708"
TARGET=_blank>10708</A>,
<A
HREF="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Bos_taurus/Bos_taurus_UMD_3.1/" 
TARGET=_blank>GCA_000003055.3</A>)
was produced by the Center for Bioinformatics and Computational Biology, University of Maryland.
The mitochondrial genome from NCBI project 
<A
HREF="http://www.ncbi.nlm.nih.gov/nuccore/NC_006853"
TARGET=_blank>NC_006853</A> was included.
 
<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Cow genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.

<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chr28</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome 28</TD>
    </TR>
    <TR><TD VALIGN=Top>chr28:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chromosome 28</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr28:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 28 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>BC140604</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number BC140604</TD>
    </TR>
    <TR><TD VALIGN=Top>EE384456</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		EE384456 on chr X</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>
            BC122573; BC102169 <BR>
        <TD WIDTH=14></TD>
        <TD VALIGN=Top>Displays region between genome landmarks, such as
                      the mRNA markers BC122573 and BC102169.
                      This syntax may also be used for other range queries, such
                      as between uniquely determined ESTs, STSs,
                      refSeqs, etc.</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>zahler</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Zahler</TD>
    </TR>
    <TR><TD VALIGN=Top>Evans,J.E.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author J.E. Evans</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Evans JE</em>, internally it
	uses the format <em>Evans,J.E.</em>.</TD>
    </TR>
</TABLE>


<HR>
<H3>Assembly details</H3>
<P>



The genome has been sequenced to 9.5X coverage.
Approximately 91% of the sequence has been anchored to 30 chromosomes, 
chromosomes 1-29 and X.
There are 3286 scaffolds (named chrUn_GJnnnnnn) for the remaining unanchored contigs. 
The mitochondrial sequence (chrM) was obtained from Genbank
project NC_006853. 

The sequence length is 2,670,422,299 bases. There are 20,740,270 'N' bases in gaps, leaving 2,649,682,029 ACGT bases. 
</P>
<P>
For more information about this assembly, see the 
<A
HREF="http://www.ncbi.nlm.nih.gov/pubmed/19393038"
TARGET=_blank>Zimin AV et al.</A>,
 "A whole-genome assembly of the domestic cow, Bos taurus.", Genome Biol, 2009 Apr 24; 10 (4) :R42
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau6/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</A> page. 


The Cow browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#cow_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
