<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: axt Alignment Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
  <TR><TD COLSPAN=2 CELLPADDING=10>	
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
		<FONT SIZE="4"><b>&nbsp;  axt Alignment Format</b></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		<em>axt</em> alignment files are produced from Blastz,
		an alignment tool available from 
		<A HREF="http://www.bx.psu.edu/miller_lab/" 
		TARGET=_blank>Webb Miller's lab</A> at Penn State University.
		The axtNet and axtChain alignments are produced by 
		processing the alignment files with additional utilities 
		written by Jim Kent at UCSC.

		<P><B><em>Example:</em></B><BR>
		The following segment from an axt file shows the first 2 
		sets of alignments of the human assembly (the aligning 
		assembly) to mouse chromsome 19 (the primary assembly).
	    	<FONT color="#0000FF"><PRE><TT>  0 chr19 3001012 3001075 chr11 70568380 70568443 - 3500
  TCAGCTCATAAATCACCTCCTGCCACAAGCCTGGCCTGGTCCCAGGAGAGTGTCCAGGCTCAGA
  TCTGTTCATAAACCACCTGCCATGACAAGCCTGGCCTGTTCCCAAGACAATGTCCAGGCTCAGA

  1 chr19 3008279 3008357 chr11 70573976 70574054 - 3900
  CACAATCTTCACATTGAGATCCTGAGTTGCTGATCAGAATGGAAGGCTGAGCTAAGATGAGCGACGAGGCAATGTCACA
  CACAGTCTTCACATTGAGGTACCAAGTTGTGGATCAGAATGGAAAGCTAGGCTATGATGAGGGACAGTGCGCTGTCACA
		</TT></PRE></FONT>

		<P><B>Structure</B><BR>
		Each alignment block in an axt file contains three lines: 
		a summary line and 2 sequence lines. 
		Blocks are separated from one another by blank lines.

		<P><B>1. Summary line</B>
	    	<FONT color="#0000FF"><PRE><TT> 0 chr19 3001012 3001075 chr11 70568380 70568443 - 3500</TT></PRE></FONT>
		The summary line contains chromosomal position and size 
		information about the alignment. It consists of 9 required
		fields:
		<UL>
		<LI>
		<B><em>Alignment number</em></B> -- The alignment numbering 
		starts with
		0 and increments by 1, i.e. the first alignment in a file
    		is numbered 0,  the next 1, etc. 
		<LI>
		<B><em>Chromosome</em></B> (primary organism)
		<LI>
		<B><em>Alignment start</em></B> (primary organism) -- The 
		first base is numbered 1.
		<LI>
		<B><em>Alignment end</em></B> (primary organism) -- The end 
		base is included.
		<LI>
		<B><em>Chromosome</em></B> (aligning organism)
		<LI>
		<B><em>Alignment start</em></B> (aligning organism)
		<LI>
		<B><em>Alignment end</em></B> (aligning organism)
		<LI>
		<B><em>Strand</em></B> (aligning organism) -- If the strand 
		value is &quot;-&quot;, the values of the aligning organism's 
		start and end fields are relative to the reverse-complemented
    		coordinates of its chromosome.
		<LI>
		<B><em>Blastz score</em></B> -- Different blastz scoring 
		matrices are used for different organisms. See the README.txt 
		file in the alignments directory for scoring information 
		specific to a pair of alignments.
		</UL>
		<P><B>2. &amp; 3. Sequence lines</B>
	    	<FONT color="#0000FF"><PRE><TT>  TCAGCTCATAAATCACCTCCTGCCACAAGCCTGGCCTGGTCCCAGGAGAGTGTCCAGGCTCAGA
  TCTGTTCATAAACCACCTGCCATGACAAGCCTGGCCTGTTCCCAAGACAATGTCCAGGCTCAGA</TT></PRE></FONT>
		The sequence lines contain the sequence of the primary 
		assembly (line 2) and aligning assembly (line 3) with
		inserts. Repeats are indicated by lower-case letters.
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	    <BR>
	  </TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

