<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: BAM Track Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  <!--outer table is for border purposes-->
  <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
      <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><b>&nbsp;  BAM Track Format</b></FONT>
	</TD></TR></TABLE>
      <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>

<P>
BAM is the compressed binary version of the 
<A HREF="http://samtools.sourceforge.net/" 
TARGET=_BLANK>Sequence Alignment/Map (SAM)</A> format,
a compact and index-able representation of nucleotide sequence alignments.  
Many 
<A HREF="http://samtools.sourceforge.net/swlist.shtml" 
TARGET=_BLANK>next-generation sequencing and analysis tools</A> 
work with SAM/BAM.
For custom track display, the main advantage of indexed BAM over PSL
and other human-readable alignment formats is that only the portions
of the files needed to display a particular region are transferred to
UCSC.  This makes it possible to display alignments from files that
are so large that the connection to UCSC would time out when
attempting to upload the whole file to UCSC.
Both the BAM file and its associated index file remain on your
web-accessible server (http, https, or ftp), not on the UCSC server.
UCSC temporarily caches the accessed portions of the files to speed up
interactive display.
</P><P>
The typical workflow for generating a BAM custom track is this:

<OL>
  <LI>If you haven't done so already,
      <A HREF="http://sourceforge.net/projects/samtools/files/" TARGET=_BLANK>
      download</A> and build the 
      <A HREF="http://samtools.sourceforge.net/" TARGET=_BLANK>samtools</A>
      program.  Test your installation by running <TT>samtools</TT> with no
      command line arguments; it should print a brief usage message.
      For help with <TT>samtools</TT>, please contact the
      <A HREF="http://sourceforge.net/mail/?group_id=246254" TARGET=_BLANK>
      SAM tools mailing list</A>.
  <LI>Align sequences using a tool that outputs SAM directly, or outputs a 
      format that can be converted to SAM.  (See 
      <A HREF="http://samtools.sourceforge.net/swlist.shtml" TARGET=_BLANK>
      list of tools and converters</A>)</LI>
  <LI>Convert SAM to BAM using the samtools program:
<PRE><TT>samtools view -S -b -o my.bam my.sam</TT></PRE>
      If converting a SAM file that does not have a proper header, the 
      <TT>-t</TT> or <TT>-T</TT> 
      option is necessary.  For more information about the command, run 
      <TT>samtools view</TT> with no other arguments.</LI>
  <LI>Sort and create an index for the BAM:
<PRE><TT>samtools sort my.bam my.sorted
samtools index my.sorted.bam</TT></PRE>
      The sort command appends <TT>.bam</TT> to <TT>my.sorted</TT>, creating a 
      BAM file of alignments ordered by leftmost position on the reference assembly.
      The index command generates a new file, <TT>my.sorted.bam.bai</TT>, with which 
      genomic coordinates can quickly be translated into file offsets in <TT>my.sorted.bam</TT>.
  <LI>Move both the BAM file and index file (<TT>my.sorted.bam</TT> and 
      <TT>my.sorted.bam.bai</TT>) to an http, https, or ftp location.</LI>
  <LI>Construct a <A HREF="hgTracksHelp.html#CustomTracks">custom track</A>
      using a single 
      <A HREF="hgTracksHelp.html#TRACK">track line</A>.                            
      The most basic version of the &quot;track&quot; line will look something 
      like this:
<PRE><TT>track type=bam name="My BAM" bigDataUrl=<EM>http://myorg.edu/mylab/my.sorted.bam</EM></TT></PRE>
      Again, in addition to <EM>http://myorg.edu/mylab/my.sorted.bam</EM>, the 
      associated index file <EM>http://myorg.edu/mylab/my.sorted.bam.bai</EM> 
      must also be available at the same location.</LI>
  <LI>Paste the custom track line into the text box in the 
      <A HREF="../../cgi-bin/hgCustom" TARGET=_BLANK>custom track 
      management page</A>, click submit and view in the Genome Browser.</LI>
</OL>
</P>

<H3>Parameters for BAM custom track definition lines</H3>
<P>
All options are placed in a single line separated by spaces (lines are broken 
only for readability here):
<PRE>
  <B>track type=bam bigDataUrl=</B><EM>http://...</EM>
     <B>pairEndsByName=</B><EM>.</EM> <B>pairSearchRange=</B><EM>N</EM>
     <B>bamColorMode=</B><EM>strand|gray|tag|off</EM> 
     <B>bamGrayMode=</B><EM>aliQual|baseQual|unpaired</EM>
     <B>bamColorTag=</B><EM>XX</EM> <B>minAliQual=</B><EM>N</EM> <B>showNames=</B><EM>on|off</EM>
     <B>name=</B><EM>track_label</EM> <B>description=</B><EM>center_label</EM> 
     <B>visibility=</B><EM>display_mode</EM> <B>priority=</B><EM>priority</EM>
     <B>db=</B><EM>db</EM> <B>maxWindowToDraw=</B><EM>N</EM> 
     <B>chromosomes=</B><EM>chr1,chr2,...</EM>
</PRE></P>
<P>Note if you copy/paste the above example, you must remove the line breaks.
Click <A HREF="examples/bamExample.txt">here</A> for a text version that you can paste 
without editing.
</P>
<P>
The track type and bigDataUrl are REQUIRED:
<PRE>  <B>type=bam bigDataUrl=</B><EM>http://myorg.edu/mylab/my.sorted.bam</EM></B></PRE>

The remaining settings are OPTIONAL.  Some are specific to BAM:
<PRE>
  <B>pairEndsByName  </B><EM>any value                  </EM> # presence indicates paired-end alignments
  <B>pairSearchRange </B><EM>N                          </EM> # max distance between paired alignments, default 20,000 bases
  <B>bamColorMode    </B><EM>strand|gray|tag|off        </EM> # coloring method, default is strand
  <B>bamGrayMode     </B><EM>aliQual|baseQual|unpaired  </EM> # grayscale metric, default is aliQual
  <B>bamColorTag     </B><EM>XX                         </EM> # optional tag for RGB color, default is "YC"
  <B>minAliQual      </B><EM>N                          </EM> # display only items with alignment quality at least N, default 0
  <B>showNames       </B><EM>on|off                     </EM> # if off, don't display query names, default is on
</PRE>
Other optional settings are not specific to BAM, but relevant:
<PRE>
  <B>name            </B><EM>track label                </EM> # default is "User Track"
  <B>description     </B><EM>center label               </EM> # default is "User Supplied Track"
  <B>visibility      </B><EM>squish|pack|full|dense|hide</EM> # default is hide (will also take numeric values 4|3|2|1|0)
  <B>priority        </B><EM>N                          </EM> # default is 100
  <B>db              </B><EM>genome database            </EM> # e.g. hg18 for Human Mar. 2006
  <B>maxWindowToDraw </B><EM>N                          </EM> # don't display track when viewing more than N bases
  <B>chromosomes     </B><EM>chr1,chr2,...              </EM> # track contains data only on listed reference assembly sequences
</PRE>
The <A TARGET=_BLANK HREF="hgBamTrackHelp.html">BAM track configuration</A> help page
describes the BAM track configuration page options corresponding to <TT>pairEndsByName</TT>,
<TT>minAliQual</TT>, <TT>bamColorMode</TT>, <TT>bamGrayMode</TT> and <TT>bamColorTag</TT> 
in more detail.</P>
<P>
<TT>pairSearchRange</TT> applies only when <TT>pairEndsByName</TT> is given.  
It allows for a tradeoff of display speed vs. completeness of 
pairing the paired-end alignments.  When paired ends are split or separated 
by large gaps or introns, but one is viewing a small genomic region, it is 
necessary to search a large number of bases upstream and downstream of the 
viewed region in order to find mates of the alignments in the viewed region.  
However, searching a very large region can be slow, especially when the 
alignments have deep coverage of the genome.  To ensure that all properly 
paired mates will be found, <TT>pairSearchRange</TT> should be set to the 
largest genomic size of a mapped pair.  However, it can be set to a smaller 
size if necessary to speed up the display, at the cost of some items being 
displayed as unpaired when the mate is too far outside the viewed window.
</P>


<P></P>
<H3>Example One</H3>
<P>
In this example, you will create a custom track for an indexed BAM file that 
is already on a public server &mdash; alignments of sequence generated by the 
<A HREF="http://1000genomes.org/" TARGET=_BLANK>1000 Genomes Project</A>.  
The line breaks inserted here for readability must be removed before submitting
the track line:
<PRE><TT>track type=bam name="BAM Example One" description="Bam Ex. 1: 1000 Genomes read alignments (individual NA12878)"
    pairEndsByName=. pairSearchRange=10000 chromosomes=chr21 bamColorMode=gray maxWindowToDraw=200000
    db=hg18 visibility=pack
    bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bamExample.bam
</TT></PRE>
<P>Include the following &quot;browser&quot; line to view a small region of 
chromosome 21 with alignments from the .bam file:</P>

<TT>browser position chr21:33,038,946-33,039,092</TT>

<P>Note if you copy/paste the above example, you must remove the line breaks
(or, click <A HREF="examples/bamExampleOne.txt">here</A> for a text version that you 
can paste without editing).</P>
<P>Paste the &quot;browser&quot; line and &quot;track&quot; line into the 
<A HREF="../../cgi-bin/hgCustom" TARGET=_BLANK>custom track management page</A>
for the human assembly hg18 (May 2006), then press the <EM>submit</EM> button.
On the following page, press the <EM>chr21</EM> link in the custom track
listing to view the BAM track in the Genome Browser.</P>


<H3>Example Two</H3>
<P>
In this example, you will create indexed BAM from an existing SAM file.
First, save this SAM file <A HREF="examples/samExample.sam" TARGET=_BLANK>samExample.sam</A> 
to your machine.
Perform steps 1 and 3-7 in the workflow described above, but substituting
<TT>samExample.sam</TT> for <TT>my.sam</TT>.  On the 
<A HREF="../../cgi-bin/hgCustom" TARGET=_BLANK>custom track management page</A>, 
click the &quot;add custom tracks&quot; button if necessary and 
make sure that the genome is set to Human and the assembly is set to Mar. 
2006 (hg18) before pasting the track line and submitting.
This track line is a little nicer than the one shown in step 6, but remember 
to remove the line breaks that have been added to the track line for 
readability (or, click <A HREF="examples/bamExampleTwo.txt">here</A> for a text version 
that you can paste without editing):
<PRE><TT>track type=bam name="BAM Example Two" bigDataUrl=<EM>http://myorg.edu/mylab/my.sorted.bam</EM>
    description="Bam Ex. 2: Simulated RNA-seq read alignments" visibility=squish
    db=hg18 chromosomes=chr21

browser position chr21:33,037,317-33,038,137
browser pack mrna</TT></PRE>
</P>

<H3>Sharing Your Data with Others</H3>
<P>
If you would like to share your BAM data track with a colleague, learn
how to create a URL by looking at <B><EM>Example 11</EM></B> on
<A HREF="customTrack.html#SHARE">this</A> page.
</P>

      </TD><TD WIDTH=15></TD></TR></TABLE>
    <BR></TD></TR></TABLE>
  </TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>
