<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: bigBed Track Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  bigBed Track Format</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
The bigBed format stores annotation items that can either be simple, or a
linked collection of exons, much as 
<A HREF="../../FAQ/FAQformat.html#format1">BED</A> files do.  
BigBed files are created initially from BED type files, 
using the program <TT>bedToBigBed</TT>.   The
resulting bigBed files are in an indexed binary format.  The main advantage of
the bigBed files is that only the portions of the files needed to display a
particular region are transferred to UCSC, so for large data sets bigBed is
considerably faster than regular BED files.  The bigBed file remains on 
your web accessible server (http, https, or ftp), not on the UCSC server. 
Only the portion that is needed
for the chromosomal position you are currently viewing is locally cached as a 
&quot;sparse file&quot;.</P>

<P>
Additional indices can be created for the items in a bigBed file.  
These indices can be used to support item search in track hubs. See <A HREF="#Ex3">Example 3</A> for an example of how to build an additional index.
<P>
See 
<A HREF="http://genomewiki.ucsc.edu/index.php/Selecting_a_graphing_track_data_format"
TARGET=_blank>this page</A> for help in selecting a graphing track data format
that is most approriate for the type of data you have.
</P>

<P>Note that the <TT>bedToBigBed</TT> utility uses a substantial amount of 
memory; somewhere on the order of 1/4 times more RAM than the 
uncompressed BED input file.</P>

<P>
To create a bigBed track, follow these steps:<BR>

<OL>
 <LI> Create a BED format file following the directions 
  <A HREF="../../FAQ/FAQformat.html#format1">here</A>.  
   <UL>
    <LI> When converting a BED file to a bigBed file, you are limited to 
    one track of data in your input file; you must create a separate BED file
    for each data track. 
    <LI> Your BED file must be sorted by chrom then chromStart. You can use
    the UNIX sort command to do this: 
    <TT>sort -k1,1 -k2,2n unsorted.bed &gt; input.bed</TT>
    <LI> This is the file that is referred to as
  <TT>input.bed</TT> in <EM>step 5</EM> below.
   </UL>
 <LI> Remove any existing 'track' or 'browser' lines from your BED file
  so that it contains only data.
 <LI> Download the <TT>bedToBigBed</TT> program from the 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> 
  of binary utilities.
 <LI> Use the <TT>fetchChromSizes</TT> script from the same 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> 
  to create the chrom.sizes file for the UCSC database you are working with 
  (e.g. hg19).  Note that this is the file that is referred to as
  <TT>chrom.sizes</TT> in <EM>step 5</EM> below.
 <LI> Create the bigBed file from your sorted BED file using the <TT>bedToBigBed</TT>
  utility like so: 
  <TT><B>bedToBigBed</B> input.bed chrom.sizes myBigBed.bb</TT><BR>
 <LI> Move the newly created bigBed file (<TT>myBigBed.bb</TT>) to a http, 
  https, or ftp location.
 <LI> Construct a <A HREF="hgTracksHelp.html#CustomTracks">custom track</A> 
  using a single 
  <A HREF="hgTracksHelp.html#TRACK">track line</A>. 
  Note that any of the track attributes listed
  <A HREF="customTrack.html#TRACK">here</A> are applicable 
  to tracks of type bigBed. 
  The most basic version of the &quot;track&quot; line will look something 
  like this:<BR>
  <TT>track type=bigBed name="My Big Bed" description="A Graph of 
  Data from My Lab" bigDataUrl=http://myorg.edu/mylab/myBigBed.bb</TT>
 <LI> Paste this custom track line into the text box in the 
  <A HREF="../../cgi-bin/hgCustom">custom track</A> management page.
</OL>

The <tt>bedToBigBed</tt> program can also be run with several additional options.
Some of them, like the <tt>-as</tt> and <tt>-type</tt> options, are used in examples
below.  A full list of the available options can be seen by running
<tt>bedToBigBed</tt> by itself with no arguments to display the usage message.

<p>
<H3>Example One</H3>
</p>
<P>In this example, you will use an existing bigBed file to create a bigBed 
custom track.  A bigBed file that contains data on chromosome 21 on the hg19 
assembly has been placed on our http server.  
You can create a custom track using this bigBed file by constructing a 
&quot;track&quot; line that references this file like so:</P>

<TT>track type=bigBed name=&quot;bigBed Example One&quot; 
description=&quot;A bigBed file&quot; 
bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigBedExample.bb</TT>

<P>Include the following &quot;browser&quot; line to ensure that the custom
track opens at the correct position:</P>

<TT>browser position chr21:33,031,597-33,041,570</TT>

<P>Paste the &quot;browser&quot; line and &quot;track&quot; line into the 
<A HREF="../../cgi-bin/hgCustom">custom track</A> management page for the
human assembly hg19 (Feb. 2009), then
press the <EM>submit</EM> button.
On the following page, press the <EM>chr21</EM> link in the custom track
listing to view the bigBed track in the Genome Browser.</P>

<H3>Example Two</H3>
<P>In this example, you will create your own bigBed file from an existing 
bed file.

<UL>
 <LI> Save this <A HREF="examples/bedExample.txt">BED file</A> 
  to your machine 
  (this satisfies <EM>steps 1</EM> and <EM>2</EM> above).
 <LI> Save this <A HREF="hg19.chrom.sizes">text file</A> to your machine.
  It contains the chrom.sizes for the human (hg19) assembly 
  (this satisfies <EM>step 4</EM> above).
 <LI> Download the <TT>bedToBigBed</TT> utility (see <EM>step 3</EM>).
 <LI> Run the utility to create the bigBed output file 
  (see <EM>step 5</EM>):<BR>
  <TT><B>bedToBigBed</B> bedExample.txt hg19.chrom.sizes myBigBed.bb</TT>
 <LI> Place the bigBed file you just created (<TT>myBigBed.bb</TT>) on a 
  web-accessible server (see <EM>step 6</EM>).
 <LI> Construct a &quot;track&quot; line that points to your bigBed file
  (see <EM>step 7</EM>).
 <LI> Create the custom track on the human assembly hg19 (Feb. 2009), and 
  view it in the genome browser (see <EM>step 8</EM>). Note that the original
  BED file contains data on only chromsome 21.
</UL>

<A NAME=Ex3></A>
<H3>Example Three</H3>
<P>In this example, you will create your own bigBed file from a fully-featured
existing BED file that contains the standard BED fields up to and including 
the color field (field 9), plus two additional non-standard fields (two 
alternate names for each item in the file).  
BigBed files can store extra fields in addition to the 
<A HREF="../../FAQ/FAQformat.html#format1">predefined BED fields</A>.  
If you add extra fields to your bigBed file, you must include
a <TT>.as</TT> (AutoSQL) format file describing the fields.  See
<A HREF="http://www.linuxjournal.com/article/5949" TARGET=_blank>this paper</A>
for information on AutoSQL.  There are several sample <TT>.as</TT> files 
<A HREF="http://genome-test.cse.ucsc.edu/~kent/src/unzipped/hg/lib/">here</A>.
This example also demonstrates how to create extra indices on the name field,
and the first of the extra fields to be used for track item search.
</P>

<UL>
 <LI> Save this <A HREF="examples/bedExample2.bed">BED file</A> 
  to your machine 
  (this satisfies <EM>steps 1</EM> and <EM>2</EM> above).
 <LI> Save this <A HREF="hg18.chrom.sizes">text file</A> to your machine.
  It contains the chrom.sizes for the human (hg18) assembly 
  (this satisfies <EM>step 4</EM> above).
 <LI> Save this <A HREF="examples/bedExample2.as">.as file</A> to your machine.  
  The <TT>.as</TT> file contains a description of the fields in the BED file.  
  This is required when the BED file contains a field for color.
 <LI> Download the <TT>bedToBigBed</TT> utility (see <EM>step 3</EM>).
 <LI> Run the utility to create the bigBed output file with an index on the name field and the first extra field:
  (see <EM>step 5</EM>):<BR>
  <TT><B>bedToBigBed</B> -as=bedExample2.as -type=bed9+2 -extraIndex=name,geneSymbol bedExample2.bed
   hg18.chrom.sizes myBigBed2.bb</TT>
 <LI> Place the bigBed file you just created (<TT>myBigBed2.bb</TT>) on a
  web-accessible server (see <EM>step 6</EM>).
 <LI> Construct a &quot;track&quot; line that points to your bigBed file
  (see <EM>step 7</EM>).  Because this bigBed file includes a field for color,
  you must include the <A HREF="../../FAQ/FAQformat.html#format1"><TT>itemRgb</TT></A>
  attribute in the &quot;track&quot; line. It will look somewhat similar to
  this (note that you must insert the correct URL to your bigBed file):<BR>
   <TT>track type=bigBed name="bigBed Example Three"
   description="A bigBed File with Color and two Extra Fields" itemRgb="On"
   bigDataUrl=http://yourWebAddress/myBigBed2.bb</TT>
 <LI> Create the custom track on the human assembly hg18 (Mar. 2006), and
  view it in the genome browser (see <EM>step 8</EM>). Note that the original
  BED file contains data only on chromsome 7.
 <LI> If you are using the bigBed file in a track hub, then you can use the
additional indices for track item searches. See the setting "searchIndex" in the
<A href="trackDb/trackDbHub.html#searchIndex">Track Database Definition Document</A>
for more information. For example, if you ran your bedToBigBed with the
option of "-extraIndex=name", you will only be able to search on the name field by
adding the following line, "searchIndex name" to the stanza about your bigBed
in the hub's trackDb.txt file. 
</UL>

<H3>Sharing Your Data with Others</H3>
<P>
If you would like to share your bigBed data track with a colleague, learn
how to create a URL by looking at <B><EM>Example 11</EM></B> on
<A HREF="customTrack.html#SHARE">this</A> page.
</P>

<H3>Extracting Data from the bigBed Format</H3>
<P>
Because the bigBed files are indexed binary files, they can be difficult to
extract data from.  Consequently, we have developed the following two 
programs, both of which are available from the 
<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> of binary
utilities.<BR>
<UL>
 <LI><TT>bigBedToBed</TT> &mdash; this program converts a bigBed file 
      to ASCII BED format.
 <LI><TT>bigBedSummary</TT> &mdash; this program extracts summary information
      from a bigBed file.
 <LI><TT>bigBedInfo</TT> &mdash; this program prints out information about a 
      bigBed file.
</UL>
As with all UCSC Genome Browser programs, simply type the program name
at the command line with no parameters to see the usage statement.
</P>

<H3>Troubleshooting</H3>
<P>If you get an error when you run the <TT>bedToBigBed</TT> program,
it may be because your input BED file has data off the end of a chromosome.
In this case, use the <TT>bedClip</TT> program 
<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">here</A> before the
<TT>bedToBigBed</TT> program.  It will remove the row(s) in your input BED
file that are off the end of a chromosome.
</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>
