<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser bigGenePred Track Format" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>bigGenePred Track Format</h1> 
<p>
The bigGenePred format stores annotation items that are a linked collection of exons, much as 
<a href="../../FAQ/FAQformat.html#format1">BED</a> files indexed as bigBeds do, but bigGenePred has 
additional information about the coding frames and other gene specific information in eight 
additional fields.</p>
<p>
bigGenePred files are created using the program <code>bedToBigBed</code> with a special AutoSQL file
that defines the fields of the bigGenePred. The resulting bigBed files are in an indexed binary 
format. The main advantage of the bigBed files is that only portions of the files needed to display 
a particular region are transferred to UCSC. So for large data sets, bigBed is considerably faster 
than regular BED files. The bigBed file remains on your web-accessible server (http, https, or ftp),
not on the UCSC server. Only the portion that is needed for the chromosomal position you are 
currently viewing is locally cached as a &quot;sparse file&quot;.</p>

<a name="bigGenePred"></a>
<h3>bigGenePred File Definition</h3>
<p>
The following autoSQL definition is used for bigGenePred gene prediction files. This is the 
<a href="examples/bigGenePred.as"><code>bigGenePred.as</code></a> file defined by the
<code>-as=</code> option when using <code>bedToBigBed</code>. Click this 
<a href="examples/bigGenePred.txt"><code>bed12+8</code></a> file for an example of bigGenePred 
input. In alternative-splicing situations, each transcript has its own row.</p>
<pre><code>table bigGenePred
"bigGenePred gene models"
    (
    string chrom;       "Reference sequence chromosome or scaffold"
    uint   chromStart;  "Start position in chromosome" uint   chromEnd;    "End position in chromosome"
    string name;        "Name or ID of item, ideally both human readable and unique"
    uint score;         "Score (0-1000)"
    char[1] strand;     "+ or - for strand"
    uint thickStart;    "Start of where display should be thick (start codon)"
    uint thickEnd;      "End of where display should be thick (stop codon)"
    uint reserved;       "RGB value (use R,G,B string in input file)"
    int blockCount;     "Number of blocks"
    int[blockCount] blockSizes; "Comma separated list of block sizes"
    int[blockCount] chromStarts; "Start positions relative to chromStart"
    string name2;       "Alternative/human readable name"
    string cdsStartStat; "enum('none','unk','incmpl','cmpl')"
    string cdsEndStat;   "enum('none','unk','incmpl','cmpl')"
    int[blockCount] exonFrames; "Exon frame {0,1,2}, or -1 if no frame for exon"
    string type;        "Transcript type"
    string geneName;    "Primary identifier for gene"
    string geneName2;   "Alternative/human readable gene name"
    string geneType;    "Gene type"
    )  </code></pre>
<p>
Note that the <code>bedToBigBed</code> utility uses a substantial amount of memory: approximately 
25% more RAM than the uncompressed BED input file.</p>
<p>
To create a bigGenePred track, follow these steps:</p>
<ol>
  <li> 
  Create a <code>bed12+8</code> bigGenePred format file that has the first twelve fields
  described by a normal BED file as described <a href="../../FAQ/FAQformat.html#format1">here</a>.
  (You can also read about genePred <a href="../../FAQ/FAQformat.html#format9">here</a>.)<br><br>  
  Your bigGenePred file must have the extra eight fields described in the AutoSQL file above:
  <code>name2, cdsStartStat, cdsEndStat, exonFrames, type, geneName, geneName2, geneType.</code><br><br> 
  Your bigGenePred file must be sorted by chrom then chromStart. You can use the UNIX sort command 
  to do this: <code>sort -k1,1 -k2,2n unsorted.bed &gt; input.bed</code></li>
  <li> 
  Download the <code>bedToBigBed</code> program from the 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary utilities.</li>
  <li> 
  Use the <code>fetchChromSizes</code> script from the same 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> to create a chrom.sizes file for
  the UCSC database you are working with (e.g. hg38).<br><br>
  Alternatively, you can download the chrom.sizes file for any assembly hosted at UCSC from our 
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html">downloads</a> page (click on &quot;Full 
  data set&quot; for any assembly). For example, for the hg38 database, the hg38.chrom.sizes are 
  located at <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes"
  target="_blank">http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes</a>.</li>
  <li> 
  Create the bigBed file from your sorted bigGenePred input file using the <code>bedToBigBed</code>
  utility: 
  <pre><code><strong>bedToBigBed</strong> -as=bigGenePred.as -type=bed12+8 bigGenePred.txt chrom.sizes myBigGenePred.bb</code></pre></li>
  <li> 
  Move the newly created bigBed file (<code>myBigGenePred.bb</code>) to an http, https, or ftp 
  location.</li>
  <li> 
  Construct a <a href="hgTracksHelp.html#CustomTracks">custom track</a> using a single 
  <a href="hgTracksHelp.html#TRACK">track line</a>. Note that any of the track attributes listed
  <a href="customTrack.html#TRACK">here</a> are applicable to tracks of type bigBed. The basic 
  version of the &quot;track&quot; line will look something like this:
  <pre><code>track type=bigGenePred name="My Big GenePred" description="A Gene Set
  Built from Data from My Lab" bigDataUrl=http://myorg.edu/mylab/myBigGenePred.bb</code></pre></li>
  <li> 
  Paste this custom track line into the text box in the <a href="../../cgi-bin/hgCustom">custom 
  track</a> management page.</li>
</ol>
<p>
The <code>bedToBigBed</code> program can also be run with several additional options. A full list o the available options can be seen by running <code>bedToBigBed</code> with no arguments to display 
the usage message.</p>

<h3>Example &num;1</h3>
<p>
In this example, you will use an existing bigGenePred file to create a bigGenePred custom track. A 
bigGenePred file that contains data on the hg38 assembly has been placed on our http server. You can
create a custom track using this bigGenePred file by constructing a &quot;track&quot; line that 
references this file:</p>
<pre><code>track type=bigGenePred name=&quot;bigGenePred Example One&quot; 
  description=&quot;A bigGenePred file&quot; 
  bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb</code></pre>
<p>
Paste the above &quot;track&quot; line into the <a href="../../cgi-bin/hgCustom?db=hg38">custom 
track</a> management page for the human assembly hg38 (Dec. 2013), then press the <em>submit</em> 
button. Note that line breaks have been inserted into the above track line for readability;
they must be removed for the example to work correctly in the browser.</p>
<p>
Custom tracks can also be loaded via one URL line. The link below loads the same bigGenePred track, 
but includes parameters on the URL line (line break inserted for readability):</p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=track%20type=bigGenePred%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb"
target="_blank"><pre><code>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&hgct_customText=track%20type=bigGenePred
%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb</code></pre></a>
<p>
With this example bigGenePred loaded, click into a gene from the track. Note that the details page 
has a &quot;Links to sequence:&quot; section that includes &quot;Translated Protein&quot;, 
&quot;Predicted mRNA&quot;, and &quot;Genomic Sequence&quot; links. Click the &quot;Go to ... track 
controls&quot; link. There change the &quot;Color track by codons:&quot; option from &quot;OFF&quot;
too &quot;genomic codons&quot; and be sure &quot;Display mode:&quot is &quot;full&quot; then click 
&quot;Submit&quot;. Then zoom to a region where amino acids display, such as 
<code>chr9:133,255,650-133,255,700</code> and see how bigGenePred allows the display of codons. 
Click back into the track controls page and click the box next to &quot;Show codon numbering&quot;. 
Return to the browser to see amino acid numbering.</p>
<p>
You can also add a parameter in the custom track line, <code>baseColorDefault=genomicCodons</code>, 
to set the display of codons:</p>
<pre><code>browser position chr10:67,884,600-67,884,900
track type=bigGenePred baseColorDefault=genomicCodons name="bigGenePred Example Two" 
  description="A bigGenePred file" visibility=pack 
  bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigGenePred.bb </code></pre>
<p>
Paste the above into the hg38 <a href="../../cgi-bin/hgCustom?db=hg38">custom track</a> page 
(removing the line breaks) to see an example of bigGenePred amino acid display around the beginning 
of the gene SIRT1 on chromosome 10.</p>

<h3>Example &num;2</h3>
<p>
In this example, you will create your own bigGenePred file from an existing bigGenePred input 
file.</p>
<ul>
  <li> 
  Save this <code>bed12+8</code> <a href="examples/bigGenePred.txt">bigGenePred.txt</a>
  example input file to your machine (<em>satisfies above step 1</em>).</li>
  <li> 
  Download the <code>bedToBigBed</code> 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">utility</a> (<em>step 2</em>).</li>
  <li> 
  Save this <code>hg38.chrom.sizes</code> <a href="hg38.chrom.sizes">text file</a> to your machine.
  It contains the chrom.sizes for the human (hg38) assembly (<em>step 3</em>).</li>
  <li> 
  Save this <code>bigGenePred.as</code> <a href="examples/bigGenePred.as"> text file</a> to your 
  machine.</li>
  <li> 
  Run the utility to create the bigBed output file (<em>step 4</em>):
  <pre><code><B>bedToBigBed</B> -type=bed12+8 -tab -as=bigGenePred.as bigGenePred.txt hg38.chrom.sizes bigGenePred.bb</code></pre></li>
  <li> 
  Place the bigBed file you just created (<code>bigGenePred.bb</code>) on a web-accessible server 
  (<em>step 5</em>).</li>
  <li> 
  Construct a &quot;track&quot; line that points to your bigGenePred file
  (see <em>step 6</em>).</li>
  <li> 
  Create the custom track on the human assembly hg38 (Dec. 2013), and view it in the genome browser 
  (see <em>step 7</em>).</li>
</ul>
<p>
Note the above description in <strong>Example &num;1</strong> on how to view genomic codons, 
including numbering.</p> 

<h3>Sharing Your Data with Others</h3>
<p>
If you would like to share your bigGenePred data track with a colleague, learn how to create a URL 
by looking at <strong><em>Example 11</em></strong> on <a href="customTrack.html#SHARE">this</a> 
page.</p>

<h3>Extracting Data from bigBed Format</h3>
<p>
Since the bigGenePred files are an extension of bigBed files, which are indexed binary files, they 
can be difficult to extract data from.  We have developed the following programs, all of which are 
available from the <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary 
utilities.</p>
<ul>
  <li>
  <code>bigBedToBed</code> &mdash; this program converts a bigBed file to ASCII BED format.</li>
  <li>
  <code>bigBedSummary</code> &mdash; this program extracts summary information from a bigBed 
  file.</li>
  <li>
  <code>bigBedInfo</code> &mdash; this program prints out information about a bigBed file.</li>
</ul>
<p>
As with all UCSC Genome Browser programs, simply type the program name at the command line with no 
parameters to see the usage statement.</p>

<h3>Troubleshooting</h3>
<p>
If you encounter an error when you run the <code>bedToBigBed</code> program, it may be because your 
input bigGenePred file has data off the end of a chromosome. In this case, use the 
<code>bedClip</code> program <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">here</a> before the
<code>bedToBigBed</code> program. It will remove the row(s) in your input BED file that are off the 
end of a chromosome.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
