<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser bigPsl Track Format" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>bigPsl Track Format</h1> 
<p>
The bigPsl format stores alignments between two sequences, as 
<a href="../../FAQ/FAQformat.html#format2">PSL</a> files do, but they are compressed and indexed as 
bigBeds. bigPsl files are created using the program <code>bedToBigBed</code> with a special AutoSQL 
file that defines the fields of the bigPsl. The resulting bigPsl files are in an indexed binary 
format. The main advantage of the bigPsl files is that only portions of the files needed to display 
a particular region are transferred to UCSC. So for large data sets, bigPsl is considerably faster 
than regular PSL files.  The bigPsl file remains on your web accessible server (http, https, or 
ftp), not on the UCSC server. Only the portion that is needed for the chromosomal position you are 
currently viewing is locally cached as a &quot;sparse file&quot;.</p>

<a name=bigPsl></a>
<h2>Big PSL</h2>
<p>
The following AutoSQL definition is used for bigPsl gene prediction files. This is the 
<a href="examples/bigPsl.as"><code>bigPsl.as</code></a> file defined by the <code>-as</code> option 
when using <code>bedToBigBed</code>.
<!--Click this <a href="examples/bigPsl.txt"><code>bigPsl.txt</code></a> file for
an example of bigPsl input. -->
</p>
<pre><code>table bigPsl
"bigPsl pairwise alignment"  
    ( 
    string chrom;       "Reference sequence chromosome or scaffold"
    uint   chromStart;  "Start position in chromosome"
    uint   chromEnd;    "End position in chromosome"
    string name;        "Name or ID of item, ideally both human readable and unique"
    uint score;         "Score (0-1000)"
    char[1] strand;     "+ or - indicates whether the query aligns to the + or - strand on the reference"
    uint thickStart;    "Start of where display should be thick (start codon)"
    uint thickEnd;      "End of where display should be thick (stop codon)"
    uint reserved;       "RGB value (use R,G,B string in input file)"
    int blockCount;     "Number of blocks"
    int[blockCount] blockSizes; "Comma separated list of block sizes"
    int[blockCount] chromStarts; "Start positions relative to chromStart"

    uint    oChromStart;"Start position in other chromosome"
    uint    oChromEnd;  "End position in other chromosome"
    char[1] oStrand;    "+ or -, - means that psl was reversed into BED-compatible coordinates" 
    uint    oChromSize; "Size of other chromosome."
    int[blockCount] oChromStarts; "Start positions relative to oChromStart or from oChromStart+oChromSize depending on strand"
    
    lstring  oSequence;  "Sequence on other chrom (or edit list, or empty)"
    string   oCDS;       "CDS in NCBI format"
  
    uint    chromSize;"Size of target chromosome"
  
    uint match;        "Number of bases matched."
    uint misMatch; " Number of bases that don't match "
    uint repMatch; " Number of bases that match but are part of repeats "
    uint nCount;   " Number of 'N' bases "
    uint seqType;   " 0=empty, 1=nucleotide, 2=amino_acid"
    ) </code></pre>
<p>
Note that the oStrand field is an indicator of whether or not the stored psl needs to be reverse 
complemented before output or display. This is due to the need for the bigPsl file to have reference
coordinates on the positive strand which is a requirement of the BED format. The strand field 
indicates whether the positions in oChromStarts are listed from the beginning (+) or the end of the 
chromosome (-).</p>
<p>
Note that the <code>bedToBigBed</code> utility uses a substantial amount of memory; somewhere on the
order of 1.25 times more RAM than the uncompressed BED input file.</p>
<p>
To create a bigPsl track, follow these steps:</p>
<ol>
  <li>If you already have a PSL file, perhaps from using BLAT or other tools, skip to <em>step 2</em>, otherwise
       download the example <a href="examples/bigPsl.psl">PSL file</a> for the Human GRCh38/hg38 assembly.
  <ul>
    <li>
    You may also want to download the <a href="examples/bigPsl.fa">bigPsl.fa</a> and 
    <a href="examples/bigPsl.cds">bigPsl.cds</a> files if you would like to use the alternate 
    options in <em>step 4</em>.</li>
  </ul>
  <li>
  Download the <code>bedToBigBed</code> and <code>pslToBigPsl</code> programs from the
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary utilities.</li>
  <li>
  Use the <code>fetchChromSizes</code> script from the same 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> to create a chrom.sizes file for
  the UCSC database you are working with (e.g., hg38). Alternatively, you can download the 
  chrom.sizes file for any assembly hosted at UCSC from our 
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html">downloads</a> page (click on &quot;Full 
  data set&quot; for any assembly). For example, for the hg38 database, the hg38.chrom.sizes are 
  located at <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes" 
  target="_blank">http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.chrom.sizes</a>.</li> 
  <li>
  Create a <code>bed12+13</code> bigPsl format file that has the 25 fields described by a bigPsl 
  file as described <a href="#bigPsl">here</a>.</li>
  <ul>
    <li>
    Your bigPsl file must have the extra thirteen fields described in the AutoSQL file above: 
    <code>oChromStart, oChromEnd, oStrand, oChromSize, oChromStarts, oSequence, oCDS, chromSize, 
    match, misMatch, repMatch, nCount, seqType </code> 
    <li>
    Use the <code>pslToBigPsl</code> utility to create a correctly formatted <code>bed12+13</code> 
    file like so: <pre><code>pslToBigPsl bigPsl.psl stdout | sort -k1,1 -k2,2n &gt; bigPsl.txt</code></pre></li>
    <li>
    Note that if you have created your own PSL file you may have corresponding FASTA and CDS files 
    that accompany it. You may provide these files as input to <code>pslToBigPsl</code> to generate 
    a more informative final bigPsl file: 
    <pre><code>pslToBigPsl bigPsl.psl -cds=bigPsl.cds -fa=bigPsl.fa stdout | sort -k1,1 -k2,2n &gt; bigPsl.txt</code></pre></li>
  </ul>
  <li>
  Create the binary indexed bigPsl file from your sorted bigPsl input file using the 
  <code>bedToBigBed</code> utility like so: 
  <pre><code>bedToBigBed -as=bigPsl.as -type=bed12+13 -tab bigPsl.txt chrom.sizes bigPsl.bb</code></pre></li>
  <li>
  Move the newly created bigPsl file (<code>bigPsl.bb</code>) to an http, https, or ftp 
  location.</li>
  <li>
  Construct a <a href="hgTracksHelp.html#CustomTracks">custom track</a> using a single 
  <a href="hgTracksHelp.html#TRACK">track line</a>. Note that any of the track attributes listed 
  <a href="customTrack.html#TRACK">here</a> are applicable to tracks of type bigBed. The most basic 
  version of the &quot;track&quot; line will look something like this:
<pre><code>track type=bigPsl name="My Big Psl" description="Some mRNAs Discovered from Data from My Lab" bigDataUrl=http://myorg.edu/mylab/myBigPsl.bb</code></pre></li>
  <li>
  Paste this custom track line into the text box on the <a href="../../cgi-bin/hgCustom">custom 
  track</a> management page.</li>
</ol>
<p>
The <code>bedToBigBed</code> program can also be run with several additional options. Run 
<code>bedToBigBed</code> with no arguments to view a full list of the available options.</p>
<p>
<h2>Example #1</h2>
<p>
In this example, you will use an existing bigPsl file to create a bigPsl custom track. A bigPsl file
that contains data on the hg38 assembly has been placed on our http server. You can create a custom 
track using this bigPsl file by constructing a &quot;track&quot; line that references this file like
so:</p>
<pre><code>track type=bigPsl name=&quot;bigPsl Example One&quot; description=&quot;A bigPsl file&quot; bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigPsl.bb</code></pre>
<p>
Paste the above &quot;track&quot; line into the <a href="../../cgi-bin/hgCustom?db=hg38">custom 
track</a> management page for the human assembly hg38, then press the &quot;submit&quot;
button.</p>
<p>
Custom tracks can also be loaded via one URL line.  The below link loads the same bigPsl track, but 
includes parameters on the URL line:</p>
<pre><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&position=chr1:10000-200000&hgct_customText=track%20type=bigPsl%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigPsl.bb%20visibility=pack"
target="_blank"><code>http://genome.ucsc.edu/cgi-bin/hgTracks?db=hg38&position=chr1:10000-200000&hgct_customText=track%20type=bigPsl%20name=Example%20bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigPsl.bb%20visibility=pack</code></a></pre>
<p>
With this example bigPsl loaded, click into an item from the track. Note how the details page 
displays information about the alignment, similar to PSL tracks, as well as links to display the 
browser position of the alignment and more detailed information about the alignment.</p>

<h2>Example #2</h2>
<p>
In this example, you will create your own bigPsl file from an existing bigPsl input file.</p>
<ul>
  <li>
  Save this <code>bed12+13</code> <a href="examples/bigPsl.txt">bigPsl.txt</a> example input file to
  your machine (<em>satisfies above step 4</em>).</li>
  <li>
  Download the <code>bedToBigBed</code> 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">utility</a> (<em>step 2</em>).</li>
  <li>
  Save this <code>hg38.chrom.sizes</code> <a href="hg38.chrom.sizes">text file</a> to your machine. 
  It contains the chrom.sizes for the human (hg38) assembly (<em>step 3</em>).</li>
  <li>
  Save this <code>bigPsl.as</code> <a href="examples/bigPsl.as"> text file</a> to your machine.</li>
  <li>
  Run the utility to create the bigBed output file (<em>step 5</em>):
<pre><code>bedToBigBed -as=bigPsl.as -type=bed12+13 -tab  bigPsl.txt hg38.chrom.sizes bigPsl.bb</code></pre></li>
  <li>
  Place the bigPsl file you just created (<code>bigPsl.bb</code>) on a web-accessible server 
  (<em>step 6</em>).</li>
  <li>
  Construct a &quot;track&quot; line that points to your bigPsl file (see <em>step 7</em>).</li>
  <li>
  Create the custom track on the human assembly hg38 (Dec. 2013), and view it in the Genome Browser 
  (see <em>step 8</em>).</li>
</ul>

<h2>Sharing your data with others</h2>
<p>
If you would like to share your bigPsl data track with a colleague, learn how to create a URL by 
looking at <em>Example 11</em> on <a href="customTrack.html#SHARE">this</a> page.</p>

<h2>Extracting data from the bigPsl format</h2>
<p>
Since the bigPsl files are an extension of bigBed files, which are indexed binary files, they can 
be difficult to extract data from. We have developed the following programs, all of which are 
available from the <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary
utilities.</p>
<ul>
  <li>
  <code>bigBedToBed</code> &mdash; this program converts a bigBed file to ASCII BED format.</li>
  <li>
  <code>bigBedSummary</code> &mdash; this program extracts summary information from a bigBed 
  file.</li>
  <li>
  <code>bigBedInfo</code> &mdash; this program prints out information about a bigBed file.</li>
</ul>
<p>
As with all UCSC Genome Browser programs, simply type the program name at the command line with no 
parameters to see the usage statement.</p>

<h2>Troubleshooting</h2>
<p>
If you encounter an error when you run the <code>bedToBigBed</code> program, it may be because your 
input bigPsl file has data off the end of a chromosome. In this case, use the <code>bedClip</code> 
program <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">here</a> before the 
<code>bedToBigBed</code> program. It will remove the row(s) in your input BED file that are off the 
end of a chromosome.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
