<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser bigWig Format" -->
<!--#set var="ROOT" value="../.." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>bigWig Track Format</h1> 
<p>
The bigWig format is for display of dense, continuous data that will be displayed in the Genome 
Browser as a graph. BigWig files are created initially from <a href="wiggle.html">wiggle</a> (wig) 
type files, using the program <code>wigToBigWig</code>. The resulting bigWig files are in an 
indexed binary format. The main advantage of the bigWig files is that only the portions of the files
needed to display a particular region are transferred to UCSC, so for large data sets bigWig is 
considerably faster than regular wiggle files. The bigWig file remains on your web accessible server
(http, https, or ftp), not on the UCSC server. Only the portion that is needed for the chromosomal 
position you are currently viewing is locally cached as a &quot;sparse file&quot;.</p>
<p>
Wiggle data must be continuous and the elements must be equally sized. If your data is sparse or 
contains elements of varying size, use the <a href="bedgraph.html">bedGraph</a> format instead of 
the wiggle format. If you have a very large bedGraph data set, you can convert it to the bigWig 
format using the <code>bedGraphToBigWig</code> program. For details, see <a href="#Ex3">Example 
Three</a> below.</p>
<p>
See <a href="http://genomewiki.ucsc.edu/index.php/Selecting_a_graphing_track_data_format" 
target="_blank">this page</a> for help in selecting a graphing track data format that is most 
approriate for the type of data you have.</p>
<p>
Please note that the <code>wigToBigWig</code> utility uses a lot of memory; somewhere on the order 
of 1.5 times more memory than the uncompressed wiggle input file. We recommend that as you run the 
program you monitor your memory usage by using the <code>top</code> command. The 
<code>bedGraphToBigWig</code> utility uses somewhere on the order of 1/4 times more RAM than the 
uncompressed bedGraph input file.</p>
<p>
To create a bigWig track from a wiggle file, follow these steps:<p>
<ol>
  <li> 
  Create a wig format file following the directions <a href="wiggle.html">here</a>. Note that when 
  converting a wig file to a bigWig file, you are limited to one track of data in your input file; 
  you must create a separate wig file for each data track. Note that this is the file that is 
  referred to as <code>input.wig</code> in <em>step 5</em> below.</li>
  <li> 
  Remove any existing track or browser lines from your wig file so that it contains only data.</li>
  <li> 
  Download the <code>wigToBigWig</code> program from the 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary utilities.</li>
  <li> 
  Use the <code>fetchChromSizes</code> script from the same 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> to create the chrom.sizes file 
  for the UCSC database you are working with (e.g., hg19). Note that this is the file that is 
  referred to as <code>chrom.sizes</code> in <em>step 5</em> below.</li>
  <li> 
  Create the bigWig file from your wig file using the <code>wigToBigWig</code> utility like so: 
  <pre><code>  <strong>wigToBigWig</strong> input.wig chrom.sizes myBigWig.bw</code></pre><br>
  (Note that the wigToBigWig program also accepts a gzipped wig input file.)</li> 
  <li> 
  Move the newly created bigWig file (<code>myBigWig.bw</code>) to a http, https, or ftp
  location.</li>
  <li> 
  If the file URL ends with .bigWig or .bw, you can paste the URL directly into the 
  <a href="../../cgi-bin/hgCustom">custom track</a> management page, click submit and view in the 
  Genome Browser. The track name will then be the name of the file. If you want to configure the 
  track label or other visualisation options, you will need to create a 
  <a href="hgTracksHelp.html#TRACK">track line</a>, as shown below.</li>
  <li> 
  Construct a <a href="hgTracksHelp.html#CustomTracks">custom track</a> using a single 
  <a href="hgTracksHelp.html#TRACK">track line</a>. The most basic version of the track line will 
  look something like this:
  <pre><code>  track type=bigWig name="My Big Wig" description="A Graph of Data from My Lab"
  bigDataUrl=http://myorg.edu/mylab/myBigWig.bw</code></pre></li>
  <p>
  <a name="optional"></a>
  Optional values can include:
  <pre>
  <strong>autoScale         </strong><I>&lt;on|off&gt;                            </I> # default is on
  <strong>alwaysZero        </strong><I>&lt;on|off&gt;                            </I> # default is off
  <strong>gridDefault       </strong><I>&lt;on|off&gt;                            </I> # default is off
  <strong>maxHeightPixels   </strong><I>&lt;max:default:min&gt;                   </I> # default is 128:128:11
  <strong>graphType         </strong><I>&lt;bar|points&gt;                        </I> # default is bar
  <strong>viewLimits        </strong><I>&lt;lower:upper&gt;                       </I> # default is range found in data
  <strong>viewLimitsMax     </strong><I>&lt;lower:upper&gt;                       </I> # suggested bounds of viewLimits, but not enforced
  <strong>yLineMark         </strong><I>&lt;real-value&gt;                        </I> # default is 0.0
  <strong>yLineOnOff        </strong><I>&lt;on|off&gt;                            </I> # default is off
  <strong>windowingFunction </strong><I>&lt;mean+whiskers|maximum|mean|minimum&gt;</I> # default is maximum, mean+whiskers is recommended
  <strong>smoothingWindow   </strong><I>&lt;off|[2-16]&gt;                        </I> # default is off
  <strong>transformFunc     </strong><I>&lt;NONE|LOG&gt;                          </I> # default is NONE</pre>
  <p>
  For further information on custom bigWig track settings, see the 
  <a href="trackDb/trackDbDoc.html#wig_-_Signal_Graphing_Track_Settings">Track Database Definition 
  Document</a>. For further information on how bigWig settings are used in native Browser tracks, 
  see the <a href="hgWiggleTrackHelp.html">Configuring graph-based tracks</a> page.</li>
  <li> 
  Paste this custom track line into the text box in the <a href="../../cgi-bin/hgCustom">custom 
  track</a> management page.</li>
</ol>

<h2>Example #1</h2>
<p>
In this example, you will use an existing bigWig file to create a bigWig custom track. A bigWig file
that spans chromosome 21 on the hg19 assembly has been placed on our http server. You can create a 
custom track using this bigWig file by pasting the URL 
<code>http://genome.ucsc.edu/goldenPath/help/examples/bigWigExample.bw</code> into the 
<a href="../../cgi-bin/hgCustom">custom track</a> management page for the human assembly hg19 (Feb. 
2009), then press the <em>submit</em> button. On the following page, press the <em>chr21</em> link 
in the custom track listing to view the bigWig track at position chr21:33,031,597-33,041,570 in the 
Genome Browser.</p>
<p>
Alternatively, you can construct a &quot;track&quot; line that references this file like so:</p>
<pre><code>  track type=bigWig name=&quot;Example One&quot; 
  description=&quot;A bigWig file&quot; 
  bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigWigExample.bw</code></pre>
<p>
Include the following &quot;browser&quot; line to ensure that the custom track opens at the correct 
position:</p>
<pre><code>  browser position chr21:33,031,597-33,041,570</code></pre>
<p>
Paste the &quot;browser&quot; line and &quot;track&quot; line into the 
<a href="../../cgi-bin/hgCustom">custom track</a> management page for the human assembly hg19 (Feb. 
2009), then press the <em>submit</em> button. On the following page, press the <em>chr21</em> link 
in the custom track listing to view the bigWig track in the Genome Browser.</p>

<h2>Example #2</h2>
<p>
In this example, you will create your own bigWig file from an existing wiggle file.</p>
<ul>
  <li> 
  Save this <a href="examples/wigVarStepExample.gz">wiggle file</a> to your machine (this satisfies 
  <em>steps 1</em> and <em>2</em> above).</li>
  <li> 
  Save this <a href="hg19.chrom.sizes">text file</a> to your machine. It contains the chrom.sizes 
  for the human (hg19) assembly (this satisfies <em>step 4</em> above).</li>
  <li> 
  Download the <code>wigToBigWig</code> utility (see <em>step 3</em>).</li> 
  <li> 
  Run the utility to create the bigWig output file (see <em>step 5</em>):<br>
  <code><strong>  wigToBigWig</strong> wigVarStepExample.gz hg19.chrom.sizes
myBigWig.bw</code></li>
  <li> 
  Place the bigWig file you just created (<code>myBigWig.bw</code>) on a web-accessible server 
  (see <em>step 6</em>).</li>
  <li> 
  Paste the URL to the file into the custom track entry form or construct a &quot;track&quot; line 
  that points to your bigWig file (see <em>step 7</em>).</li>
  <li> 
  Create the custom track on the human assembly hg19 (Feb. 2009), and view it in the Genome Browser 
  (see <em>step 8</em>). Note that the original wiggle file spans only chromsome 21.</li>
</ul>

<a name=Ex3></a>
<h2>Example #3</h2>
<p>
To create a bigWig track from a bedGraph file, follow these steps:</p>
<ol>
  <li> 
  Create a bedGraph format file following the directions <a href="bedgraph.html">here</a>. Note that
  when converting a bedGraph file to a bigWig file, you are limited to one track of data in your 
  input file; you must create a separate bedGraph file for each data track. Note that this is the 
  file that is referred to as <code>in.bedGraph</code> in <em>step 5</em> below.</li>
  <li> 
  Remove any existing track or browser lines from your bedGraph file so that it contains only 
  data.</li>
  <li> 
  Download the <code>bedGraphToBigWig</code> program from the 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary utilities.</li>
  <li> 
  Use the <code>fetchChromSizes</code> script from the same 
  <a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> to create the chrom.sizes file 
  for the UCSC database you are working with (e.g., hg19). Note that this is the file that is 
  referred to as <code>chrom.sizes</code> in <em>step 5</em> below.</li>
  <li> 
  Create the bigWig file from your bedGraph file using the <code>bedGraphToBigWig</code> utility 
  like so:
  <pre><code><strong>  bedGraphToBigWig</strong> in.bedGraph chrom.sizes myBigWig.bw</code></pre><br>
  (Note that the bedGraphToBigWig program <strong>DOES NOT</strong> accept a gzipped bedGraph input 
  file.)</li>
  <li> 
  Move the newly created bigWig file (<code>myBigWig.bw</code>) to a http, https, or ftp 
  location.</li>
  <li> 
  Paste the URL into the custom track entry form or construct a 
  <a href="hgTracksHelp.html#CustomTracks">custom track</a> using a single 
  <a href="hgTracksHelp.html#TRACK">track line</a>.</li>
  <li> 
  Paste this custom track line into the text box in the <a href="../../cgi-bin/hgCustom">custom 
  track</a> management page.</li>
</ol>

<h2>Sharing Your Data with Others</h2>
<p>
If you would like to share your bigWig data track with a colleague, learn how to create a URL by 
looking at <em>Example 11</em> on <a href="customTrack.html#SHARE">this</a> page.</p>

<h2>Extracting Data from the bigWig Format</h2>
<p>
Because the bigWig files are indexed binary files, they can be difficult to extract data from. 
Consequently, we have developed the following two programs, both of which are available from the 
<a href="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</a> of binary utilities.
<ul>
  <li>
  <code>bigWigToBedGraph</code> &mdash; this program converts a bigWig file to ASCII 
  <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1.8">bedGraph</a> format.</li>
  <li>
  <code>bigWigToWig</code> &mdash; this program converts a bigWig file to 
  <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format6">wig</a> format.</li>
  <li>
  <code>bigWigSummary</code> &mdash; this program extracts summary information from a bigWig 
  file.</li>
  <li>
  <code>bigWigAverageOverBed</code> &mdash; this program computes the average score of a bigWig over
  each bed, which may have introns.</li>
  <li>
  <code>bigWigInfo</code> &mdash; this program prints out information about a bigWig file.</li>
</ul>
<p>
These utilities accept paths to files or alternatively URLs to file. As with all UCSC Genome Browserprograms, simply type the program name at the command line with no parameters to see the usage 
statement.</p>
<p>
In some cases, bigWigSummary and bigWigAverageOverBed will produce very similar results, but in 
other cases, the results may differ. This is due to differences in how the utilities handle data. 
Summary levels are used with bigWigSummary, so some rounding errors and border conditions are 
encountered when extracting data over relatively small regions. The bigWigAverageOverBed utility 
uses the actual data which ensures the highest level of accuracy.</p>

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
