<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: bigWig Track Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><b>&nbsp;  bigWig Track Format</b></FONT>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
		    <P>
The bigWig format is for display of dense, continuous data that will be
displayed in the Genome Browser as a graph.  BigWig files are
created initially from <A HREF="wiggle.html">wiggle</A> (wig) type  
files, using the program <TT>wigToBigWig</TT>.  The resulting bigWig files are 
in an indexed binary format.  The main advantage of the bigWig files 
is that only the portions of the files needed to display a particular 
region are transferred to UCSC, so for large data sets bigWig is 
considerably faster than regular wiggle files.
The bigWig file remains on your web accessible server (http, 
https, or ftp), not on the UCSC server. Only the portion that is needed
for the chromosomal position you are currently viewing is locally cached as a 
&quot;sparse file&quot;.</P>

<P>
Wiggle data must be continuous and the elements must be equally sized. If your
data is sparse or contains elements of varying size, use the 
<A HREF="bedgraph.html">bedGraph</A> format instead of the wiggle format.  
If you have a very large bedGraph data set, you can convert it to the bigWig
format using the <TT>bedGraphToBigWig</TT> program.  For details, see 
<A HREF="#Ex3">Example Three</A> below.
</P>

<P>
See 
<A HREF="http://genomewiki.ucsc.edu/index.php/Selecting_a_graphing_track_data_format"
TARGET=_blank>this page</A> for help in selecting a graphing track data format
that is most approriate for the type of data you have.
</P>

<P>Please note that the <TT>wigToBigWig</TT> utility uses a lot of memory;
somewhere on the order of 1.5 times more memory than the uncompressed 
wiggle input file.  We recommend that as you run the program you monitor your 
memory usage by using the <TT>top</TT> command.  The <TT>bedGraphToBigWig</TT> 
utility uses somewhere on the order of 1/4 times more RAM than the
uncompressed bedGraph input file.</P>

<P>
To create a bigWig track from a wiggle file, follow these steps:<BR>

<OL>
 <LI> Create a wig format file following the directions 
  <A HREF="wiggle.html">here</A>.  
  Note that when converting a wig file to a bigWig file, you are limited to 
  one track of data in your input file; you must create a separate wig file
  for each data track.  Note that this is the file that is referred to as
  <TT>input.wig</TT> in <EM>step 5</EM> below.
 <LI> Remove any existing 'track' or 'browser' lines from your wig file
  so that it contains only data.
 <LI> Download the <TT>wigToBigWig</TT> program from the 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> 
  of binary utilities.
 <LI> Use the <TT>fetchChromSizes</TT> script from the same 
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> 
  to create the chrom.sizes file for the UCSC database you are working with 
  (e.g. hg19).  Note that this is the file that is referred to as
  <TT>chrom.sizes</TT> in <EM>step 5</EM> below.
 <LI> Create the bigWig file from your wig file using the <TT>wigToBigWig</TT>
  utility like so: 
  <TT><B>wigToBigWig</B> input.wig chrom.sizes myBigWig.bw</TT><BR>
  (Note that the wigToBigWig program also accepts a gzipped wig input file.)
 <LI> Move the newly created bigWig file (<TT>myBigWig.bw</TT>) to a http, 
  https, or ftp location.
 <LI> Construct a <A HREF="hgTracksHelp.html#CustomTracks">custom track</A>
  using a single 
  <A HREF="hgTracksHelp.html#TRACK">track line</A>.                            
  The most basic version of the track line will look something 
  like this:<BR>
  <TT>track type=bigWig name="My Big Wig" description="A Graph of 
  Data from My Lab" bigDataUrl=http://myorg.edu/mylab/myBigWig.bw</TT>
  <p>
  <a name="optional"></a>
  Optional values can include:
  <PRE>
  <B>autoScale         </B><I>&lt;on|off&gt;                            </I> # default is on
  <B>alwaysZero        </B><I>&lt;on|off&gt;                            </I> # default is off
  <B>gridDefault       </B><I>&lt;on|off&gt;                            </I> # default is off
  <B>maxHeightPixels   </B><I>&lt;max:default:min&gt;                   </I> # default is 128:128:11
  <B>graphType         </B><I>&lt;bar|points&gt;                        </I> # default is bar
  <B>viewLimits        </B><I>&lt;lower:upper&gt;                       </I> # default is range found in data
  <B>viewLimitsMax     </B><I>&lt;lower:upper&gt;                       </I> # suggested bounds of viewLimits, but not enforced
  <B>yLineMark         </B><I>&lt;real-value&gt;                        </I> # default is 0.0
  <B>yLineOnOff        </B><I>&lt;on|off&gt;                            </I> # default is off
  <B>windowingFunction </B><I>&lt;mean+whiskers|maximum|mean|minimum&gt;</I> # default is maximum, mean+whiskers is recommended
  <B>smoothingWindow   </B><I>&lt;off|[2-16]&gt;                        </I> # default is off
  <B>transformFunc     </B><I>&lt;NONE|LOG&gt;                          </I> # default is NONE</PRE>

  <P>
  For further information on custom bigWig track settings, see the
  <a href="trackDb/trackDbDoc.html#wig_-_Signal_Graphing_Track_Settings">Track Database Definition Document</a>.
  For further information on how bigWig settings are used in native Browser tracks, see the
  <a href="hgWiggleTrackHelp.html">Configuring graph-based tracks</a> page.
 <LI> Paste this custom track line into the text box in the 
  <A HREF="../../cgi-bin/hgCustom">custom track</A> management page.
</OL>

<P><H3>Example One</H3></P>
<P>In this example, you will use an existing bigWig file to create a bigWig 
custom track.  A bigWig file that spans chromosome 21 on the hg19 assembly 
has been placed on our http server.  
You can create a custom track using this bigWig file by constructing a 
&quot;track&quot; line that references this file like so:</P>

<TT>track type=bigWig name=&quot;Example One&quot; 
description=&quot;A bigWig file&quot; 
bigDataUrl=http://genome.ucsc.edu/goldenPath/help/examples/bigWigExample.bw</TT>

<P>Include the following &quot;browser&quot; line to ensure that the custom
track opens at the correct position:</P>

<TT>browser position chr21:33,031,597-33,041,570</TT>

<P>Paste the &quot;browser&quot; line and &quot;track&quot; line into the 
<A HREF="../../cgi-bin/hgCustom">custom track</A> management page for the
human assembly hg19 (Feb. 2009), then
press the <EM>submit</EM> button.
On the following page, press the <EM>chr21</EM> link in the custom track
listing to view the bigWig track in the Genome Browser.</P>

<P><H3>Example Two</H3></P>
<P>In this example, you will create your own bigWig file from an existing 
wiggle file.

<UL>
 <LI> Save this <A HREF="examples/wigVarStepExample.gz">wiggle file</A> 
  to your machine 
  (this satisfies <EM>steps 1</EM> and <EM>2</EM> above).
 <LI> Save this <A HREF="hg19.chrom.sizes">text file</A> to your machine.
  It contains the chrom.sizes for the human (hg19) assembly 
  (this satisfies <EM>step 4</EM> above).
 <LI> Download the <TT>wigToBigWig</TT> utility (see <EM>step 3</EM>).
 <LI> Run the utility to create the bigWig output file 
  (see <EM>step 5</EM>):<BR>
  <TT><B>wigToBigWig</B> wigVarStepExample.gz hg19.chrom.sizes myBigWig.bw</TT>
 <LI> Place the bigWig file you just created (<TT>myBigWig.bw</TT>) on a 
  web-accessible server (see <EM>step 6</EM>).
 <LI> Construct a &quot;track&quot; line that points to your bigWig file
  (see <EM>step 7</EM>).
 <LI> Create the custom track on the human assembly hg19 (Feb. 2009), and 
  view it in the genome browser (see <EM>step 8</EM>). Note that the original
  wiggle file spans only chromsome 21.
</UL>

<A NAME=Ex3></A>
<P><H3>Example Three</H3></P>
<P>
To create a bigWig track from a bedGraph file, follow these steps:<BR>

<OL>
 <LI> Create a bedGraph format file following the directions
  <A HREF="bedgraph.html">here</A>.
  Note that when converting a bedGraph file to a bigWig file, you are limited to
  one track of data in your input file; you must create a separate bedGraph file
  for each data track.  Note that this is the file that is referred to as
  <TT>in.bedGraph</TT> in <EM>step 5</EM> below.
 <LI> Remove any existing 'track' or 'browser' lines from your bedGraph file
  so that it contains only data.
 <LI> Download the <TT>bedGraphToBigWig</TT> program from the
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A>
  of binary utilities.
 <LI> Use the <TT>fetchChromSizes</TT> script from the same
  <A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A>
  to create the chrom.sizes file for the UCSC database you are working with
  (e.g. hg19).  Note that this is the file that is referred to as
  <TT>chrom.sizes</TT> in <EM>step 5</EM> below.
 <LI> Create the bigWig file from your bedGraph file using the 
  <TT>bedGraphToBigWig</TT> utility like so:
  <TT><B>bedGraphToBigWig</B> in.bedGraph chrom.sizes myBigWig.bw</TT><BR>
  (Note that the bedGraphToBigWig program <b>DOES NOT</b> accept a gzipped bedGraph input file.)
 <LI> Move the newly created bigWig file (<TT>myBigWig.bw</TT>) to a http,
  https, or ftp location.
 <LI> Construct a <A HREF="hgTracksHelp.html#CustomTracks">custom track</A>
  using a single
  <A HREF="hgTracksHelp.html#TRACK">track line</A>.
 <LI> Paste this custom track line into the text box in the
  <A HREF="../../cgi-bin/hgCustom">custom track</A> management page.
</OL>

<H3>Sharing Your Data with Others</H3>
<P>
If you would like to share your bigWig data track with a colleague, learn
how to create a URL by looking at <B><EM>Example 11</EM></B> on
<A HREF="customTrack.html#SHARE">this</A> page.
</P>

<H3>Extracting Data from the bigWig Format</H3>
<P>
Because the bigWig files are indexed binary files, they can be difficult to
extract data from.  Consequently, we have developed the following two
programs, both of which are available from the
<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">directory</A> of binary
utilities.<BR>
<UL>
 <LI><TT>bigWigToBedGraph</TT> &mdash; this program converts a bigWig file
      to ASCII <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format1.8">bedGraph</a> format.
 <LI><TT>bigWigToWig</TT> &mdash; this program converts a bigWig file
      to <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format6">wig</a> format.
 <LI><TT>bigWigSummary</TT> &mdash; this program extracts summary information
      from a bigWig file.
 <LI><TT>bigWigAverageOverBed</TT> &mdash; this program computes the average score
      of a bigWig over each bed, which may have introns.
 <LI><TT>bigWigInfo</TT> &mdash; this program prints out information about a 
      bigWig file.
</UL>
As with all UCSC Genome Browser programs, simply type the program name
at the command line with no parameters to see the usage statement.
</P>

</P>
In some cases, bigWigSummary and bigWigAverageOverBed will produce very similar
results, but in other cases, the results may differ. This is due to differences
in how the utilities handle data.  Summary levels are used with
bigWigSummary, so some rounding errors and border conditions are encountered
when extracting data over relatively small regions. The bigWigAverageOverBed
utility uses the actual data which ensures the highest level of accuracy.
</P>

	</TD><TD WIDTH=15></TD></TR></TABLE>
	<BR></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
</BODY></HTML>
