<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!-----------Indel Display Help Page------------------------------>
<A NAME="Topic5"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
			    <TR><TD><FONT SIZE="4"><A
                            NAME="TOC"></A><B>&nbsp; Alignment Insertion/Deletion Display Options</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
				<p>
				The Genome Browser offers several ways to highlight gaps in alignments
				of query sequences (usually transcripts) to the genome.  Gaps
		      	        result from sequences in the genome or query (or both) that cannot
				be aligned. 
				</p><p>
				<b>Legend</b>
				<ul>
				<li>double horizontal line (=): both the genome and query have unalignable 
				sequence between regions of aligned sequence, a double-sided insertion. If 
				this option is not selected, it will display as a single horizontal line.</li>
				<li><font color="#FFA70F">orange lines</font> or 
				<font color="#9214FF">purple lines</font>: unalignable query sequence, orange 
				for the middle of a sequence and purple for the beginning or end.</li>
				<li><font color="#00E000">green lines</font>: poly-A tail or poly-T head that 
				does not align to the genome.</li>
				</ul>
				<B>Details</B><BR>
				When zoomed out past the base level, the browser chooses one color to 
				represent many bases. The priority of display, from most important to least 
				important, is: 
				<a href="hgCodonColoringMrna.html">different mRNA base/nonsynonymous codon coloring</a>
				(if enabled) 
			        <em>or </em>
				<a href="hgBaseLabel.html">different item bases</a>
				(if enabled),
				unalignable query sequence (orange or purple), an insertion in 
				both genome and query (double horizontal line), and a poly-A tail (green). 
				The browser will not display 
				<a href="hgCodonColoringMrna.html">genomic/mRNA codon coloring</a> when 
				viewing large regions of the genome.
				</P><P>
				<B>Interpretation of Display</B><BR>
				Gaps are usually due to a deletion or insertion in one or both sequences, 
				or, infrequently, a problem in the sequencing. Often, the genome will have a 
				large "insertion" relative to a query (single horizontal line) that is 
				actually an intron. 
				Double-sided insertions (double horizontal line) are unusual and may indicate 
				an assembly error, sequencing error, or polymorphism.
				</p><p>
 				Unalignable query sequence in the middle of a query (orange) implies extra 
				bases in the transcript sequence or missing bases in the genome sequence. 
				Insertions at the beginning or end of a query (purple), implies a partial 
				alignment of the query. For instance, a very short sequence next to large 
				intron gap may be incorrectly aligned. Unalignable query 
				sequence may also be due to polymorphisms. 
				</p><p>
				Poly-T heads result from queries that are the reverse complement of the genomic
				sequence. Poly-A tails and poly-T heads (green) of mRNAs usually can not be 
				aligned to the genome; this is a special case of an unalignable query sequence.
				</p><p>
				<b>For information about mRNA codon and base coloring, click
				<a href="hgCodonColoringMrna.html">here</a>.</b></p>
				<b>For information about EST base coloring, click
				<a href="hgBaseLabel.html">here</a>.</b></p>
				</p>
			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
    </TABLE>
</BODY></HTML>
