<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="description" content="UCSC Genome Browser">
	<META name="keywords" content="genome, genome data, table browser, genome browser, genome assembly, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Browser: Table Browser User's Guide</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">

</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
<TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables-->  
<TR><TD CELLPADDING=10 ROWSPAN=2 VALIGN="top">
  	<!--outer table is for border purposes-->
  	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%><TR><TD>
		<FONT SIZE="4"><B>&nbsp;Table Browser User's Guide </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
	<TR><TD WIDTH=10>&nbsp;</TD><TD>
			
<H3> Table of Contents: </H3>
<UL><B>
  <LI><A CLASS="toc" HREF="#Introduction">
      Introduction</A>
      <UL>
        <LI><A CLASS="toc" HREF="#Purpose">
        What can the Table Browser do?</A>
        <LI><A CLASS="toc" HREF="#Tables">
        About the Genome Browser databases and tables</A>
      </UL>
  <LI><A CLASS="toc" HREF="#Pages">
      Table Browser Forms</A>
      <UL>
        <LI><A CLASS="toc" HREF="#Gateway">
        Choose Organism &amp; Assembly</A>
        <LI><A CLASS="toc" HREF="#ChooseTable">
        Choose a Table</A>
        <LI><A CLASS="toc" HREF="#PasteNames">
        Paste in Names/Accessions</A>
        <LI><A CLASS="toc" HREF="#UploadNames">
        Upload a File of Names/Accessions</A>
        <LI><A CLASS="toc" HREF="#DescTable">
        Describe Table</A>
        <LI><A CLASS="toc" HREF="#AdvancedQuery">
        Advanced Query</A>
        <LI><A CLASS="toc" HREF="#IntersectOptions">
        Intersect Results</A>
        <LI><A CLASS="toc" HREF="#BEDOptions">
        BED/Custom Track Options</A>
        <LI><A CLASS="toc" HREF="#FASTAOptions">
        FASTA (DNA sequence) Options</A>
        <LI><A CLASS="toc" HREF="#ChooseFields">
        Choose Fields</A>
      </UL>
  <LI><A CLASS="toc" HREF="#Formats">
      Output Data Formats</A>
      <UL>
        <LI><A CLASS="toc" HREF="#BED">
        BED/Custom Track</A>
        <LI><A CLASS="toc" HREF="#FASTA">
        FASTA (DNA Sequence)</A>
        <LI><A CLASS="toc" HREF="#GTF">
        GTF</A>
        <LI><A CLASS="toc" HREF="#Hyperlinks">
        Hyperlinks to Genome Browser</A>
        <LI><A CLASS="toc" HREF="#TabSep">
        Tab Separated</A>
        <LI><A CLASS="toc" HREF="#Stats">
        Summary/Statistics</A>
      </UL>
  <LI><A CLASS="toc" HREF="#Constraints">
      Filtering Results by Constraining Field Values</A>
      <UL>
        <LI><A CLASS="toc" HREF="#StringConstraints">
        Constraints on strings</A>
        <LI><A CLASS="toc" HREF="#NumericConstraints">
        Constraints on numbers</A>
        <LI><A CLASS="toc" HREF="#FreeFormConstraints">
        Free-Form Query</A>
      </UL>
  <LI><A CLASS="toc" HREF="#Intersection">
      Filtering Results by Comparing Two Tables</A>
      <UL>
        <LI><A CLASS="toc" HREF="#Preserving">
        Comparisons that preserve primary table structure</A>
        <LI><A CLASS="toc" HREF="#BaseByBase">
        Base-by-base comparisons</A>
      </UL>
  <LI><A CLASS="toc" HREF="#FeatureBits">
      Retrieving Subregions of Features</A>
      <UL>
        <LI><A CLASS="toc" HREF="#FBGenePred">
	Subregions of gene predictions</A>
        <LI><A CLASS="toc" HREF="#FBPSL">
	Subregions of alignments</A>
        <LI><A CLASS="toc" HREF="#FBBed8">
	Subregions of "thick/thin" records</A>
        <LI><A CLASS="toc" HREF="#FBBed4">
	Subregions of simple records</A>
      </UL>
  <LI><A CLASS="toc" HREF="#Examples">
      Example Uses of the Table Browser</A>
      <UL>
        <LI><A CLASS="toc" HREF="#getDna">
        Get DNA sequence for RefSeq genes</A>
        <LI><A CLASS="toc" HREF="#matchGeneName">
        Get locations of all RefSeq genes that start with HOX</A>
        <LI><A CLASS="toc" HREF="#blastz">
        Get human alignments to mouse at ESPN</A>
        <LI><A CLASS="toc" HREF="#repeatRefGene">
        Get locations of repeats identified on exons of RefSeq genes</A>
        <LI><A CLASS="toc" HREF="#tabSep">
        Download the contents of a table for local processing</A>
      </UL>
</B></UL>
<P>
The UCSC Table Browser User's Guide was last updated on 23 October 2003. 
<A HREF="../../contacts.html">Questions and feedback are welcome</A>.
</TD><TD WIDTH=15></TD></TR></TABLE>
<BR></TD></TR></TABLE>
</TD></TR></TABLE>

<!--outer table is for border purposes-->
<P>
<A NAME="Introduction"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>

<FONT SIZE="4"><B>&nbsp; Introduction </B></FONT>

</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>

The Table Browser provides a query interface to the Genome Browser 
database tables.  
Because the Table Browser uses the same database as the Genome Browser, 
the two views are always consistent.  
(Chromosomal coordinates usually change with each assembly because of
newly filled gaps or assembly procedure refinements.) 

<A NAME="Purpose"></A>
<H4> What can the Table Browser do? </H4>

The Table Browser can retrieve DNA sequence data or annotation data, 
for Genome Browser tracks or for intersections/unions of tracks, over 
a given position range, over a given set of accessions, 
or over the whole genome.  
It can display some basic statistics calculated over selected data.  
It can generate a 
<A HREF="/goldenPath/help/customTrack.html">custom track</A> and 
automatically add it to the user's session, so that it can be graphically 
displayed in the Genome Browser.  It allows 
<A HREF="#Intersection">combined queries</A> on two tables, or on a 
user custom track and a table, so compound queries are possible.  

<P>
One goal of the Table Browser is to provide an extremely flexible 
query interface.  The user can specify constraints on the values 
of fields, to narrow down the set of results.  The output format 
is also highly configurable.  

<P>
Some things that the Table Browser cannot do (yet) include 
joining together relational tables in a unified interface.  
The Table Browser is not as powerful as running MySQL on a 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">local installation of the database itself</A>.  


<A NAME="Tables"></A>
<H4> About the Genome Browser databases and tables </H4>

The 
<A HREF="/goldenPath/gbdDescriptions.html">
Genome Browser Database</A> actually consists of several separate 
databases, one for each genome assembly.  

<A NAME="Positional"></A>
<P>
The Table Browser divides the tables in a database into two categories:
<B>positional</B> and <B>non-positional</B>.  
Some output formats and query options are only applicable to positional 
tables, thus the distinction.
<UL>
<LI> <B>Positional</B> tables contain data associated with specific locations 
     in the genome.  mRNA alignments, gene predictions, cross-species 
     alignments, and other annotations are stored in positional tables.  
     Each Genome Browser track is based on a positional table (but may 
     incorporate information from non-positional tables).  
<LI> <B>Non-positional</B> tables contain data not tied to genomic locations. 
     For example, a table that relates a Genethon marker name to a 
     Marshfield marker name is non-positional.  
     Many tables relate internal numeric mRNA IDs to 
     information such as author, tissue, keyword, etc.  
     Some "meta" tables contain information about the structure of the 
     database itself or describe external files containing sequence data.  
</UL>

<P>
Positional tables can be further subdivided into several categories 
based on the type of data they describe.  Some tables 
describe alignments, so they describe a block structure for each element.  
Other tables do not describe a block structure (only a start and an end).  
Some tables specify a translation start &amp; end in addition to a 
transcription start &amp; end.  Some tables specify the strand, some don't.  
Most, but not all, give a name for each element.  Based on what 
a table describes, different query and output formatting options may be 
offered.  

<P>
Descriptions of the database tables are available 
<A HREF="/goldenPath/gbdDescriptions.html">here</A>.  


</TD><TD WIDTH=15></TD></TR></TABLE>
<BR></TD></TR></TABLE>
</TD></TR></TABLE>

<!--outer table is for border purposes-->
<P>
<A NAME="Pages"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>

<FONT SIZE="4"><B>&nbsp; Table Browser Forms </B></FONT>

</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>

The following sections describe the Table Browser pages with the 
given titles.  


<A NAME="Gateway"></A>
<H4> Choose Organism &amp; Assembly </H4>

This page prompts the user to select the organism (e.g. "Human") and 
assembly (e.g. "Nov. 2002").  
This is the gateway page for the Table Browser; it will appear when 
the user clicks on the "Tables" link from the main page of the 
genome server, or whenever the Table Browser is invoked without 
specifying which genome assembly to use.  


<A NAME="ChooseTable"></A>
<H4> Choose a Table </H4>

This page is the entry point for the Table Browser when the genome 
assembly has already been specified (for example, when the user 
follows the "Tables" link from the Genome Browser).  
There are four categories of tables:
<UL>
<LI> <B>Browser tracks</B>: Tables that correspond to Genome Browser tracks. 
     </LI>
<LI> <B><A HREF="/goldenPath/help/customTrack.html">Custom tracks</A></B>:
     User custom tracks in the current session.  </LI>
<LI> <B><A HREF="#Positional">Positional</A> tables</B>: All positional tables,
     whether or not they correspond to Genome Browser tracks, both from 
     the current assembly and from the hgFixed shared database.  </LI>
<LI> <B>Non-positional tables</B>: Tables without genomic coordinates, both 
     from the current assembly and from the hgFixed shared database.  </LI>
</UL>

In web browsers that support Javascript, whenever a selection is made from 
one of the above menus, the other three menus are reset to avoid confusion.  
In older web browsers without Javascript, it is possible to select tables 
from multiple menus.  In that case, the Table Browser will use the leftmost 
selected table.  Other selected tables will be ignored.  

<P>
When a browser track, custom track or positional table is selected, 
items can be selected from the table by position or by item name (which is 
usually an accession for gene/mRNA/EST tracks).  The default position 
is <B>"genome"</B>, which selects all items in the table.  Any position or 
search term accepted by the Genome Browser will also be accepted by the 
Table Browser.  If you have chosen to select items by name/accession, 
then you must 
<A HREF="#PasteNames">paste</A> or <A HREF="#UploadNames">upload</A> 
a list of names/accessions before retrieving data.  

<P>
When a non-positional table is selected, the position and any specified 
names/accessions are ignored.  All items in the table will be returned 
by default.  To restrict non-positional table items by name or accession, 
paste values into the filter input for the appropriate field on the 
<A HREF="#AdvancedQuery">Advanced Query</A> page.  

<P>
The "Get all fields" button simply retrieves 
all elements from the table (at the specified position, if the table is 
positional), and prints them out as 
<A HREF="#TabSep">tab-separated plain text</A>.  

<P>
The "Get sequence" button works only when a browser track, custom track, 
or positional table has been selected.  It is for retrieving 
<A HREF="#FASTAOptions">FASTA</A> sequence data for selected items.  

<P>
The "Advanced Query" button leads to a richer choice of output formats and 
filtering options, described 
<A HREF="#AdvancedQuery">below</A>.  

<P>
The "Describe table" button displays
<A HREF="#DescTable">available information</A> about the selected table.  


<A NAME="PasteNames"></A>
<H4> Paste in Names/Accessions  </H4>

Here the user can enter in a list of item names (usually accessions for 
gene/mRNA/EST tracks), which may contain wildcard characters ("?" to match 
exactly one character, "*" to match 0 or more characters).  

<P>
Note: name/accession matching is not supported for non-positional 
tables.  


<A NAME="UploadNames"></A>
<H4> Upload a File of Names/Accessions  </H4>

On this page, the user can upload a file containing item names 
(usually accessions for gene/mRNA/EST tracks).  Wildcards are not 
allowed; the names must have exact matches in the table, or they will 
be ignored.  

<P>
Note: name/accession matching is not supported for non-positional 
tables.  


<A NAME="DescTable"></A>
<H4> Describe Table </H4>

This page shows descriptive information about the currently selected 
table and its fields.  Buttons at the top and bottom of the page 
offer the same choice of actions as the 
<A HREF="#ChooseTable">Choose a Table</A> page.  

<UL>
<LI><B>AutoSql definition (if available):</B> Many browser database tables 
    are generated and processed using SQL and C code generated by the tool 
    <A HREF="http://www.linuxjournal.com/article.php?sid=5949" TARGET=_BLANK>
    AutoSql</A>.  The AutoSql input format includes the name, data type 
    and descriptive comment for each field in a table and for the table itself.
    It is often the best available documentation for a table.  
    </LI>
<LI><B>Link to table description (if available):</B> 
    The <A HREF="/goldenPath/gbdDescriptions.html" TARGET=_BLANK>
    Genome Browser Database</A> document is another source of information 
    about tables.  Most database tables are described in corresponding 
    sections of the document.  
    Many sections contain just the autoSql definition, but some sections 
    contain more information about the format and other related tables. 
    </LI>
<LI><B>Field SQL types:</B> Just in case the autoSql definition is outdated 
    (this can happen - please email the 
    <A HREF="../../contacts.html">genome mailing list</A> 
    if you encounter this, thanks!) 
    or not available, we display the SQL field names and types 
    for the table as it truly exists in the current database.  
    For convenience, there are buttons which lead to histograms showing the 
    frequency of occurrences of values of text fields within the current 
    position range.  
    (The 
    <A HREF="#Stats">Summary/Statistics</A> output type, available via 
    <A HREF="#AdvancedQuery">Advanced Query</A>, provides statistics about 
    integer values of position and exon/block fields.)  
    </LI>
<LI><B>Row count and first few rows:</B> The total number of rows in the 
    table and the contents of the first few rows give an idea of 
    the data size and example values.  No filtering or batch-querying is 
    reflected in the count, and the example rows will probably not be 
    from the current position (unless it is "genome").  They are shown 
    only as a descriptive example.  
    </LI>
<LI><B>Browser Track information (if available):</B> If the table is the 
    main (positional) table for a Genome Browser track, then the data type, 
    track group name and HTML track description are shown.  
    </LI>
</UL>


<A NAME="AdvancedQuery"></A>
<H4> Advanced Query </H4>

This page offers choices for 
<A HREF="#Formats">output format</A> and 
<A HREF="#Constraints">filtering of results</A> 
in several sections:
<UL>
<LI> <B>Select Output Format for <I>table</I></B><BR>
     <A HREF="#Formats">Output format</A> choices for positional tables are 
     <UL>
     <LI> <A HREF="#BEDOptions">Bed/Custom Track...</A>
     <LI> <A HREF="#FASTAOptions">FASTA (DNA sequence)...</A>
     <LI> <A HREF="#GTF">GTF</A>
     <LI> <A HREF="#Hyperlinks">Hyperlinks to Genome Browser</A>
     <LI> <A HREF="#TabSep">Tab-separated</A>, All Fields
     <LI> <A HREF="#TabSep">Tab-separated</A>,
          <A HREF="#ChooseFields">Choose Fields...</A>
     <LI> <A HREF="#Stats">Summary/Statistics</A>
     </UL>
     For non-positional tables, the choices are fewer:
     <UL>
     <LI> <A HREF="#TabSep">Tab-separated</A>, All Fields
     <LI> <A HREF="#TabSep">Tab-separated</A>,
          <A HREF="#ChooseFields">Choose Fields...</A>
     <LI> <A HREF="#Stats">Summary/Statistics</A>
     </UL>
<LI> <B>(Optional) Filter <I>table</I> Records by Field Values</B><BR>
     This section allows the user to set 
     <A HREF="#Constraints">constraints</A> 
     on the values of most table fields.  
     If no values are modified from the defaults, then no constraining or 
     filtering (other than position for positional tables) is done.  
<LI> <B>(Optional) Intersect Results with Another Table</B><BR>
     Here the user may select a second positional table and click the 
     "Intersect Results" button, which leads to the form described 
     <A HREF="#IntersectOptions">below</A>.  
     The Table Browser will remember the output format and filtering 
     constraints from above, and apply them after the "Intersect Results" 
     form is completed.       
     This section appears only when the primary table is  positional, 
     because the 
     <A HREF="#Intersection">intersection/union</A> 
     is performed on the positions contained in the selected tables.  
</UL>

For convenience, the "Get results" button (for non-intersected results) 
appears in both the output format section and the filtering section.  
It is the same button (same action), 
offered in two places to save the user some scrolling.  


<A NAME="IntersectOptions"></A>
<H4> Intersect Results </H4>

At this point the user has specified two positional tables, as well 
as filtering constraints (if any) on the primary table.  The output 
format has also been selected -- for intersection of results, the 
output format must be 
<A HREF="#FASTA">FASTA</A>,
<A HREF="#BED">BED/Custom Track</A>,
<A HREF="#Hyperlinks">Hyperlinks</A>,
<A HREF="#GTF">GTF</A>, or
<A HREF="#Stats">Summary/Statistics</A>.  
Now, the user may specify filtering constraints on the secondary table, 
and must choose a mode for combining the results from each table:

<UL>
<LI> <B>Specify how to combine tables <I>table1</I> and <I>table2</I></B><BR>
     Here the user selects a 
     <A HREF="#Intersection">mode of combining</A> 
     primary table results and secondary table results based on position.  
     <UL>
     <LI> <A HREF="#AnyOverlap">any overlap</A>
     <LI> <A HREF="#NoOverlap">no overlap</A>
     <LI> <A HREF="#OverOverlap">overlap over a specified threshold</A>
     <LI> <A HREF="#UnderOverlap">overlap under a specified threshold</A>
     <LI> <A HREF="#BaseByBaseAND">base-by-base positional intersection (AND)</A>
     <LI> <A HREF="#BaseByBaseOR">base-by-base positional union (OR)</A>
     </UL>
     The last two combinations (base-by-base positional intersection/union) 
     can be applied to the 
     <A HREF="#BaseByBaseNOT">base-by-base positional complement</A> 
     of either or both tables.  
<LI> <B>(Optional) Filter <I>table2</I> Records by Field Values</B><BR>
     This section allows the user to set 
     <A HREF="#Constraints">constraints</A> 
     on the values of secondary table fields.  
     If no values are modified from the defaults, then no constraining or 
     filtering of the secondary table (other than position) is done.  
</UL>


<A NAME="BEDOptions"></A>
<H4> BED/Custom Track Options </H4>

When the user has selected the <A HREF="#BED">BED/Custom Track format</A>, 
the Table Browser prompts for a few additional parameters:

<UL>
<LI> <B>Include 
        <A HREF="/goldenPath/help/customTrack.html">custom track</A>
	header:</B><BR>
     If this box is checked, then a custom track 
     <A HREF="/goldenPath/help/customTrack.html#TRACK">"track" header line</A> 
     will be generated with the name, description, visibility, and url fields. 
     Defaults are supplied, so the user does not have to enter any values.
<LI> If the user has chosen to include a custom track header in the output, 
     then the user can also check 
     <B>Load this custom track into my session</B> 
     so that the results of the query will be locally stored (until 4 hours 
     after last usage).  To view the custom track in the Genome Browser, 
     click the "Get BED" button and view (or save) the resulting BED data; 
     then click your web browser's "Back" button, then click 
     the "Genome Browser" link in the blue bar at the top of the page.  
<LI> <B>Create one BED record per:</B><BR>
     Here the user has the option to retrieve BED just for certain 
     components of each record, or from upstream/downstream of each record.
     <A HREF="#FeatureBits">The options</A> are different for each type of 
     table.
</UL>


<A NAME="FASTAOptions"></A>
<H4> FASTA (DNA sequence) Options </H4>

This form prompts the user for additional parameters of 
<A HREF="#FASTA">FASTA (DNA sequence)</A> output:

<UL>
<LI> <B>Sequence Retrieval Region Options:</B><BR>
     If the table is a gene prediction or alignment table, i.e. specifying 
     exon, translation or block structure, then 
     the user can select which 
     <A HREF="#FeatureBits">subregions</A> 
     of table records are 
     used to fetch DNA sequence.  The user also has the choice to 
     concatenate the chunks of DNA sequence for each selected subregion  
     into one sequence, or to retrieve a separate sequence for each 
     selected subregion.  (Concatenating is useful when all regions are 
     selected, or when exons are selected and introns are unselected 
     to see spliced mRNA sequence.)  
<LI> <B>Sequence Formatting Options:</B><BR>
     The user can select upper or lower case (for some subregions if 
     the table specifies structures), and can mask locations of 
     repetitive sequences (RepeatMasker) if desired.  
</UL>


<A NAME="ChooseFields"></A>
<H4> Choose Fields </H4>

This form prompts the user to select which fields of the table to 
include in 
<A HREF="#TabSep">tab-separated</A> output when the user has selected 
the "Tab Separated, Choose Fields..." output format.  


</TD><TD WIDTH=15></TD></TR></TABLE>
<BR></TD></TR></TABLE>
</TD></TR></TABLE>


<!--outer table is for border purposes-->
<P>
<A NAME="Formats"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>
<FONT SIZE="4"><B>&nbsp; Output Data Formats </B></FONT>
</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>

The Table Browser returns query results in one of the following formats.  
<B>Note:</B> When querying 
non-positional 
tables, only the 
<A HREF="#TabSep">Tab Separated</A> and 
<A HREF="#Stats">Summary/Statistics</A> formats are available.  


<A NAME="BED"></A>
<H4> BED/Custom Track </H4>

BED stands for Browser Extensible Data.  It is a plain-text data format, 
and is the internal representation used for all 
<A HREF="customTrack.html">user custom tracks</A> 
and for many Genome Browser tracks.  
The BED format is described on the
<A HREF="../../FAQ/FAQformat.html#format1">Data File Formats</A> page.  

<P>
Selecting BED as the output data format allows the user to retain 
their query results as user custom tracks, which can then be viewed 
in the Genome Browser or queried again in the Table Browser to form 
compound queries.  


<A NAME="FASTA"></A>
<H4> FASTA (DNA Sequence) </H4>

FASTA is a very simple plain text format for displaying nucleotide or 
protein sequence.  
(The Table Browser only outputs DNA sequence, not protein sequence, 
at this time.)  
For each record in FASTA output, there is one 
header line which begins with "&gt;" and contains a description of 
the record, followed by one or more lines whose letters represent 
the sequence.  

<P>
Below is an example FASTA record returned by the Table Browser, 
with an explanation of the header description fields.  (The header 
line is artificially broken into two lines for better display/printing 
of this web page.)  

<PRE><TT>
>hg12_acembly_glutee_0 range=chr22:15586756-15586867 5'pad=0 3'pad=0 \
 revComp=TRUE strand=- repeatMasking=none
GGCACTGACAGGACTTGACACATTCTAATCTCCTGAGGGCCGCTAACAAC
ATAGATGACAGTCTGGACAACACAAAGGGTCAACAGGTGCTCCGAGTGTC
CGGCAGGGCTAA
</TT></PRE>

<UL>
<LI> <B><TT>hg12_acembly_glutee_0</TT></B>: The database is hg12 
     (Human June 2002),
     the table is acembly,
     the feature name is glutee,
     and the user has selected to return a separate FASTA record for 
     each exon/intron on the 
     <A HREF="#FASTAOptions">FASTA/DNA Sequence options form</A> and 
     this is the first (_0) record (the first exon).  
<LI> <B><TT>range=chr22:15586756-15586867</TT></B>: The feature 
     extends from 15586756 to 15586867 on chr22.  Note: the user may 
     choose to generate a single FASTA record for all exons, excluding 
     introns.  In that case, the record may be shorter than the length 
     implied by the range.  
<LI> <B><TT>5'pad=0 3'pad=0</TT></B>: The user has chosen to include no 
     extra bases at the 5' and 3' ends of each feature.  
<LI> <B><TT>revComp=TRUE strand=-</TT></B>: This feature is on the '-' 
     strand, so the sequence has been reverse complemented relative to 
     the genomic sequence at the given range.  
<LI> <B><TT>repeatMasking=none</TT></B>: The user has chosen not to mask 
     out (i.e. force to 'N' or lower-case) the bases that have been 
     identified as repeats by RepeatMasker.  
</UL>


<A NAME="GTF"></A>
<H4> GTF </H4>

GTF stands for Gene Transfer Format.  It is also known as 
GFF version 2.  The spec can be found 
<A HREF="http://www.sanger.ac.uk/Software/formats/GFF/GFF_Spec.shtml">
here</A>.  


<A NAME="Hyperlinks"></A>
<H4> Hyperlinks to Genome Browser </H4>

For convenience, the Table Browser can generate links to the 
Genome Browser at the location of each feature identified by 
a query on a positional table.  


<A NAME="TabSep"></A>
<H4> Tab Separated </H4>

This plain text output format simply contains the contents of a table, 
one table record per line, 
with fields (columns) separated by tab characters.  
The first line of output is a comment, starting with "#" and displaying 
the field names separated by tab characters.
This format is easy to load into spreadsheets or process with scripts, 
if the user wishes to perform additional analysis on a table.  


<A NAME="Stats"></A>
<H4> Summary/Statistics </H4>

This format summarizes the user's query (including any 
<A HREF="#Constraints">filtering constraints</A> or 
<A HREF="#Intersection">comparison of tables</A>.  
For 
positional
tables, some statistics about the features are computed 
(count, min/max/average/stdev of length, etc.) and displayed.  
For all tables, the fields of the table and their SQL database types 
are displayed.  For text fields, the user may click to obtain a 
frequency histogram of a field's values in the query results.  


</TD></TR></TABLE>
<BR></TD></TR>
</TABLE>
</TD></TR>
</TABLE>


<!--outer table is for border purposes-->
<P>
<A NAME="Constraints"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>
<FONT SIZE="4"><B>&nbsp; Filtering Results by Constraining Field Values </B></FONT>
</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>


By default, the table browser retrieves all records for the specified
coordinate range (use <B>"genome"</B> to search all chromosomes) 
or set of item names/accessions. 
To restrict the set of records retrieved from this range, you may
enter constraints on the values of fields.

The individual field constraints are combined with AND. In other
words, a record must meet all individual field constraints in order to
be retrieved. The default values will match all records, so the only
constraints that apply are the ones that you edit.


<A NAME="StringConstraints"></A>
<H4> Constraints on strings </H4>

Text fields are compared to words or patterns containing wildcard
characters. Wildcards are "*" (to match 0 or more characters) and "?"
(to match a single character). Each space-separated word/pattern in a
text field box will be matched against the value of that field in each
record. If any word/pattern matches the value, then the record meets
the constraint on that field.


<A NAME="NumericConstraints"></A>
<H4> Constraints on numbers </H4>

Numeric fields are compared to table data using an operator such as <,
>, != (not equal) followed by a number. To specify a range, enter two
numbers (start and end) separated by whitespace and/or a comma (",").


<A NAME="FreeFormConstraints"></A>
<H4> Free-Form Query </H4>

When the filters above aren't enough, free-form queries allow 
fancier constraints that typically relate two or more field names of 
the selected table. If you are familiar with 
<A HREF="http://www.w3schools.com/sql/sql_where.asp" TARGET=_blank>
SQL syntax</A>, 
queries here are just "where" clauses (with wildcards as above). 
Queries can combine simple constraints with AND, OR, and NOT using 
parentheses as needed for clarity. 
<P> A simple constraint consists of a field name listed above, 
     a comparison operator (see below), 
     and a value: a number, string, wildcard value (see below), 
     or another field name. 
     In place of a field name, you may use 
     an arithmetic expression of numeric field names.  </P>
<P> String or wildcard values for text comparisons must be quoted.  
     Single or double quotes may be used.  
     If comparing to a literal string value, 
     you may use the "=" (or "!=") operator.  
     If comparing to a wildcard 
     value, you must use the "LIKE" (or "NOT LIKE") operator.  
     </P> 
<P> Numeric comparison operators include &lt;, &lt;=, =, != (not equal), 
     &gt;=, and &gt; .  
     Arithmetic operators include +, -, *, and /.  
     Other <A HREF="http://www.w3schools.com/sql/sql_where.asp" TARGET=_blank>
     SQL</A> 
     comparison keywords may be used.  
     </P> 
</P><P>
Free-form query examples (taken from the refGene table):
<UL>
<LI> <FONT COLOR="#0000FF">txStart = cdsStart</FONT> 
     searches for gene models missing expected 5' UTR upstream sequence 
     (if strand is '+'; 3' UTR downstream if strand is '-'). </LI>
<LI> <FONT COLOR="#0000FF">chrom NOT LIKE "chr??"</FONT> 
     filters search to chromosomes 1-9 plus X and Y. </LI> 
<LI> <FONT COLOR="#0000FF">(cdsEnd - cdsStart) &gt; 10000</FONT> 
     selects genes whose coding regions span over 10 kbp. </LI> 
<LI> <FONT COLOR="#0000FF">(txStart != cdsStart) AND (txEnd != cdsEnd) 
     AND exonCount = 1</FONT> 
     finds single exon genes with both 3' and 5' flanking UTR.</LI> 
<LI> <FONT COLOR="#0000FF">((cdsEnd - cdsStart) &gt; 30000) 
     AND (exonCount=2 OR exonCount=3)</FONT> 
     finds genes with long spans but only 2-3 exons.</LI>
</UL>
</P>

</TD></TR></TABLE>
<BR></TD></TR>
</TABLE>
</TD></TR>
</TABLE>


<!--outer table is for border purposes-->
<P>
<A NAME="Intersection"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>
<FONT SIZE="4"><B>&nbsp; Filtering Results by Comparing Two Tables </B></FONT>
</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>

Sometimes it is interesting to see where the features of two 
positional 
tables do or do not cover the same positions in the genome.  
The Table Browser offers several kinds of comparisons of tables 
by feature positions.  



<A NAME="Preserving"></A>
<H4> Comparisons that preserve primary table structure </H4>

Some comparisons preserve the structure of the primary table.  
In other words, if the primary table describes exon structure 
and is filtered by one of these comparisons to any other table, 
then the results will describe exon structure (unless the user 
chooses an output format in which the structure is lost).  
Primary table features are kept or discarded based on the 
amount of positional overlap with secondary table features.  

<UL>
<A NAME="AnyOverlap"></A>
<LI> <B>Any overlap:</B> A primary table record will appear in the output 
     if any of its base positions are also covered by 
     any feature in the secondary table.  
<A NAME="NoOverlap"></A>
<LI> <B>No overlap:</B> A primary table record will appear in the output 
     only if none of its base positions are also covered by 
     any feature in the secondary table.  
<A NAME="OverOverlap"></A>
<LI> <B>Overlap over a specified threshold:</B> A primary table record will 
     appear in the output 
     if the ratio of the number of its base positions covered by 
     secondary table features to the total number 
     of its base positions is greater than the user-specified threshold.  
<A NAME="UnderOverlap"></A>
<LI> <B>Overlap under a specified threshold:</B> A primary table record will 
     appear in the output 
     if the ratio of the number of its base positions covered by 
     secondary table features to the total number 
     of its base positions is less than the user-specified threshold.  
</UL>

<B>Note:</B> If the primary table has exon/block structure, then only 
base positions in exons/blocks will be counted.  


<A NAME="BaseByBase"></A>
<H4> Base-by-base comparisons </H4>

The positions covered by features of the primary and secondary tables 
can be compared one base position at a time, or base-by-base (as opposed 
to feature-by-feature like above).  
If the user selects one of the following comparisons, then the structure 
of the primary table will not be preserved.  
In other words, even if the primary table describes exon structure, 
the results will not describe exon structure.  Only position ranges 
will be returned, with no information about exon/block structure, 
strand, or translation region.  

<UL>
<A NAME="BaseByBaseAND"></A>
<LI> <B>Base-by-base intersection (AND):</B> If a nucleotide position is 
     covered by at least one feature of both the primary table and the 
     secondary table, then it will be included in the output.  
<A NAME="BaseByBaseOR"></A>
<LI> <B>Base-by-base union (OR):</B>  If a nucleotide position is 
     covered by at least one feature of either the primary table or the 
     secondary table, then it will be included in the output.  
</UL>

<B>Note:</B> If the primary table has exon/block structure, then only 
base positions in exons/blocks will be counted.  
<P>
<A NAME="BaseByBaseNOT"></A>
<B>Base-by-base complement (NOT):</B> Before either of the above 
comparisons are applied, the set of positions covered by a table 
can be inverted.  In other words, if this is selected, then any 
position covered by the table's features will be considered 
<I>not</I> covered, and vice versa.  This gives the user more 
flexibility in comparing table positions.  

</TD></TR></TABLE>
<BR></TD></TR>
</TABLE>
</TD></TR>
</TABLE>


<!--outer table is for border purposes-->
<P>
<A NAME="FeatureBits"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>
<FONT SIZE="4"><B>&nbsp; Retrieving Subregions of Features </B></FONT>
</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>

<A NAME="FBGenePred"></A>
<H4> Subregions of gene predictions </H4>
A gene prediction table specifies both exon structure 
and translated region.  The choices of granularity are
<UL>
<LI> <B>Whole Gene:</B> the block structure and translated region 
        will be preserved in the BED data.  
<LI> <B>Upstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 5' end of the gene to <I>N</I> bases away.  
	If the strand is '+', the 5' end is the low coordinate 
	("start").  If the strand is '-', the 5' end is the 
	high coordinate ("end").  
	Block structure and translated region info will be lost.  
<LI> <B>Exons plus <I>N</I> bases at each end:</B> the BED data 
        will contain one simple record per exon, subtracting 
	<I>N</I> from the start and adding <I>N</I> to the end.  
	(<I>N</I> can be negative to make the region smaller 
	instead of larger.)
<LI> <B>Introns plus <I>N</I> bases at each end:</B> the BED data 
        will contain one simple record per intron, subtracting 
	<I>N</I> from the start and adding <I>N</I> to the end.  
	(<I>N</I> can be negative to make the region smaller 
	instead of larger.)
<LI> <B>5' UTR Exons:</B> the BED data 
        will contain one simple record per 5' UTR exon.  
	Depending on the value of the strand, 5' may start at 
	the low coordinate "start" and proceed up (+), or it may 
	start at the high coordinate "end" and proceed down (-).
<LI> <B>Coding Exons:</B> the BED data 
        will contain one simple record per coding exon.  
	(If the translated region boundaries fall in the middle 
	of exons, then the exons will be split at those boundaries.)
<LI> <B>3' UTR Exons:</B> the BED data 
        will contain one simple record per 3' UTR exon.  
	Depending on the value of the strand, 3' may start at 
	the high coordinate "end" and proceed down (+), or it may 
	start at the low coordinate "start" and proceed up (-);
<LI> <B>Downstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 3' end of the gene to <I>N</I> bases away.  
	If the strand is '+', the 3' end is the high coordinate 
	("end").  If the strand is '-', the 3' end is the 
	low coordinate ("start").  
	Block structure and translated region info will be lost.
</UL>
Note: if a feature is close to the beginning or end of a chromosome 
and upstream/downstream bases are added, they may be truncated 
in order to avoid extending past the edge of the chromosome. 

<A NAME="FBPSL"></A>
<H4> Subregions of alignments </H4>
An alignment table specifies block structure, but not translated region.
<UL>
<LI> <B>Whole Alignment:</B> the block structure 
        will be preserved in the BED data.
<LI> <B>Upstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 5' end of the alignment to <I>N</I> bases away.  
	If the strand is '+', the 5' end is the low coordinate 
	("start").  If the strand is '-', the 5' end is the 
	high coordinate ("end").  
<LI> <B>Blocks plus <I>N</I> bases at each end:</B> the BED data 
        will contain one simple record per block, subtracting 
	<I>N</I> from the start and adding <I>N</I> to the end.  
	(<I>N</I> can be negative to make the region smaller 
	instead of larger.)
<LI> <B>Regions between blocks plus <I>N</I> bases at each end:</B>
        the BED data 
        will contain one simple record per intron, subtracting 
	<I>N</I> from the start and adding <I>N</I> to the end.  
	(<I>N</I> can be negative to make the region smaller 
	instead of larger.)
<LI> <B>Downstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 3' end of the gene to <I>N</I> bases away.  
	If the strand is '+', the 3' end is the high coordinate 
	("end").  If the strand is '-', the 3' end is the 
	low coordinate ("start").  
	Block structure and translated region info will be lost.
</UL>
Note: if a feature is close to the beginning or end of a chromosome 
and upstream/downstream bases are added, they may be truncated 
in order to avoid extending past the edge of the chromosome. 


<A NAME="FBBed8"></A>
<H4> Subregions of "thick/thin" records </H4>
A thick/thin table specifies a "thick" region in 
the middle of each feature (surrounded by "thin" regions, as 
displayed in the Genome Browser).  The "thick" region usually 
refers to translated region (but note that this type of table 
does not specify block structure).  
<UL>
<LI> <B>Whole Gene:</B> the "thick" (translated) region information 
        will be preserved in the BED data.  
<LI> <B>Upstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 5' end of the record to <I>N</I> bases away.  
	If the strand is '+', the 5' end is the low coordinate 
	("start").  If the strand is '-', the 5' end is the 
	high coordinate ("end").  
<LI> <B>5' UTR:</B> the BED data 
        will contain one simple record per 5' UTR.  
	Depending on the value of the strand, 5' may start at 
	the low coordinate "start" and proceed up to the "thick" 
	region (+), or it may 
	start at the high coordinate "end" and proceed down (-) to 
	the "thick" region.
<LI> <B>CDS:</B> the BED data 
        will contain one simple record (the "thick" region).  
<LI> <B>3' UTR:</B> the BED data 
        will contain one simple record per 3' UTR.  
	Depending on the value of the strand, 3' may start at 
	the high coordinate "end" and proceed up to the "thick" 
	region (+), or it may 
	start at the low coordinate "start" and proceed down (-) to 
	the "thick" region.
<LI> <B>Downstream by <I>N</I> bases:</B> </UL> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 3' end of the record to <I>N</I> bases away.  
	If the strand is '+', the 3' end is the high coordinate 
	("end").  If the strand is '-', the 3' end is the 
	low coordinate ("start").  
Note: if a feature is close to the beginning or end of a chromosome 
and upstream/downstream bases are added, they may be truncated 
in order to avoid extending past the edge of the chromosome. 

<A NAME="FBBed4"></A>
<H4> Subregions of simple records </H4>
A simple table specifies on the the start and end position (no 
block structure or thick/thin region).  However, upstream and 
downstream regions are available.  If a strand is specified, 
then the meaning of upstream and downstream depend on the strand.  
<UL>
<LI> <B>Whole Gene:</B> simple records (with strand and score, if 
        the table specifies strand and score) are output.  
<LI> <B>Upstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 5' end of the record to <I>N</I> bases away.  
	If the strand is '+', the 5' end is the low coordinate 
	("start").  If the strand is '-', the 5' end is the 
	high coordinate ("end").  
<LI> <B>Downstream by <I>N</I> bases:</B> the BED data will contain 
        one simple (chrom/start/end/name) record per feature, 
	from the 3' end of the record to <I>N</I> bases away.  
	If the strand is '+', the 3' end is the high coordinate 
	("end").  If the strand is '-', the 3' end is the 
	low coordinate ("start").  
</UL>
Note: if a feature is close to the beginning or end of a chromosome 
and upstream/downstream bases are added, they may be truncated 
in order to avoid extending past the edge of the chromosome. 


</TD></TR></TABLE>
<BR></TD></TR>
</TABLE>
</TD></TR>
</TABLE>


<!--outer table is for border purposes-->
<P>
<A NAME="Examples"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
<TR><TD>
<TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
<TR><TD>
<TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
<TR><TD>
<FONT SIZE="4"><B>&nbsp; Example Uses of the Table Browser </B></FONT>
</TD></TR></TABLE>
<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
<TR><TD WIDTH=10>&nbsp;</TD><TD>

All of these examples assume that the user has already selected a 
genome assembly (for example, Human Nov. 2002).  


<A NAME="getDna"></A>
<H4> Get DNA sequence for RefSeq genes </H4>

<UL>
<LI> If there is a particular gene of interest, enter the gene's 
     <B>NM_??????</B> accession number
     and click on the <B>Look up</B> button.  <BR>
     Otherwise, enter a position range, e.g. <B>chr22:15500000-15750000</B>
<LI> Select <B>refGene</B> from the "Positional tables" menu.  <BR>
     Click on the <B>Advanced query...</B> button.  
<LI> Select <B>FASTA (DNA Sequence)</B> from the output format menu.  <BR>
     Click on the <B>Get results</B> button.
<LI> Review the subregion and formatting options.  <BR>
     Click on the <B>Get sequence</B> button.  
</UL>

FASTA for all RefSeq genes in the position range will be displayed.  


<A NAME="matchGeneName"></A>
<H4> Get locations of all RefSeq genes that start with HOX </H4>

The refGene table contains the location of RefSeq genes, indexed by 
accession number (NM_??????).  The refFlat table contains the same 
information, but indexed by gene name.  

<UL>
<LI> To find all RefSeq genes that start with HOX in the whole genome, 
     change the position to <B>genome</B>.  
<LI> Select <B>refFlat</B> from the "Positional tables" menu.  <BR>
     Click on the <B>Advanced query...</B> button.  
<LI> Select Hyperlinks to the  from the output format menu.  <BR>
     In the "Filter refFlat records" section, 
     enter <B>HOX*</B> in the box at the end of the line for the geneName field 
     ('geneName [does] match ...').  <BR>
     Click on the <B>Get results</B> button.  
</UL>

A page of links into the Genome Browser at the location of each HOX* 
gene will appear.  


<A NAME="blastz"></A>
<H4> Get human alignments to mouse at ESPN </H4>

<B>Note:</B> This example assumes the existence of the blastzBestMouse track 
in the current genome assembly/database, and the existence of the ESPN gene 
in the current assembly's RefSeq Genes (refGene/refFlat) track.  Before 
starting this example, make sure the current database is set to Human, 
June 2002 or later.

<UL>
<LI> Set position to <B>ESPN</B> and click the <B>Look up</B> button.  
<LI> Select <B>chrN_blastzBestMouse</B> from the "Positional tables" menu. <BR>
     Click on the <B>Get all fields</B> button.  
</UL>

The contents of the chr1_blastzBestMouse table that fall at least partly 
in the selected position range will be printed out as 
<A HREF="#TabSep">tab-separated text</A>.  
The format used by the Genome Browser for tables containing alignments 
(such as chrN_blastzBestMouse) is 
<A HREF="../../FAQ/FAQformat.html#format2">PSL</A>.  


<A NAME="repeatRefGene"></A>
<H4> Get locations of repeats identified on exons of chr21 RefSeq genes </H4>

<UL>
<LI> Set position to <B>chr21</B> and click the <B>Look up</B> button.  
<LI> Select <B>chrN_rmsk</B> from the "Positional tables" menu. <BR>
     Click on the <B>Advanced query...</B> button.  
<LI> Select <B>BED/Custom Track</B> from the output format menu. <BR>
     Scroll down to the "(Optional) Intersect Results" section.  <BR>
     Select <B>refGene</B> from the "Positional tables" menu.  <BR>
     Click the <B>Intersect Results...</B> button.
<LI> Select <B>All chrN_rmsk records that have any overlap with refGene</B>
     as the way to combine the tables.  <BR>
     Scroll down to the bottom of the page and click on the 
     <B>BED/Custom Track</B> button.  
<LI> Experiment with the BED options form by clicking the <B>Get BED</B> 
     button, looking at the results, then using your web browser's Back 
     button to return to the form and try other settings.  
</UL>

If you choose the <B>Load this custom track into my session</B> option 
when experimenting with BED/Custom Track options, then the results will 
be kept temporarily (until 8 hours after last usage) on the server.  
To view your custom track, click your web browser's Back button, 
then click the "Genome Browser" link on the blue bar near the top of 
the page.  Click the <B>Submit</B> button, and your Table Browser 
generated custom track will appear at or near the top of the browser 
tracks image.  

<A NAME="tabSep"></A>
<H4> Download the contents of a table for local processing </H4>

<UL>
<LI> Enter the <B>position</B> range (or gene name, accession, etc.) of 
     interest and click on the <B>Look up</B> button.  
<LI> Select a <B>table</B> from either the "Positional tables" menu or the 
     "Non-positional tables" menu.  <BR>
     Click the <B>Get all fields</B> button.  
</UL>

Please don't set the position to "genome" very often with this method 
unless you are also filtering the results with 
<A HREF="#Constraints">constraints on field values</A> (or 
<A HREF="#Intersection">comparison to another table</A>).  
The amount of data could be very large, could take a long time, 
and could unnecessarily and unfairly load the server.  
Instead, if you need to download the entire contents of tables, 
see the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">downloads page</A>.  


</TD></TR></TABLE>
<BR></TD></TR>
</TABLE>
</TD></TR>
</TABLE>


</TD></TR></TABLE>
</BODY></HTML>

