<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Configuring VCF Tracks</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="#FFF9D2" LINK="#0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../inc/topbar.html"--></TD></TR>
<!-----------VCF Track Help Page------------------------------>
<TR><TD ROWSPAN=2 VALIGN="top"></TD></TR>
<TR><TD>

<A NAME="Topic5"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH="100%">
			    <TR><TD><FONT SIZE="4"><A
                            NAME="TOC"></A><B>&nbsp; Configuring VCF tracks</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
Genome Browser VCF tracks may be configured in a variety of ways to
highlight different aspects of the displayed information.
Click <A HREF="vcf.html" TARGET=_BLANK>here</A> for more information
on VCF custom track creation.
</P>
<P>
The following configuration options appear only if the VCF file contains
genotype columns for at least two samples (four haplotypes):
<UL>
<LI><B>Enable Haplotype sorting display:</B>
    If checked, then each sample's phased and/or homozygous genotypes are split
    into haplotypes, clustered by similarity around a central variant, and sorted
    for display by their position in the clustering tree.  The tree (or as much of
    the tree as space allows) is drawn in the label area next to the track image.
    If unchecked, then the display is the same as for VCF without genotypes:
    a stacked bar graph of the top two alleles, showing the proportion of alleles
    if allele counts are available.
    This is checked (enabled) by default.
    The following options are applicable only when enabled.</LI>
    <UL>
    <LI><B>Haplotype sorting order:</B>
      Haplotypes are sorted using a distance function that uses a central variant;
      differences between haplotypes are penalized with weights that decrease for
      each successive variant away from the central variant.  By default, the median
      variant in the window is used.  By clicking on a variant in the display, you
      will get the option to always use that variant when it is in the current view.</LI>
    <LI><B>Color sorted haplotypes by:</B>
      By default, the reference allele is colored blue and the alternate allele is colored red.
      This can be changed to the same color scheme used when there are no genotypes:
      A is red, C is blue, G is green and T is magenta.</LI>
    <LI><B>Haplotype sorting display height:</B>Height in pixels of the haplotype sorting
      display.  If this is fewer than the number of haplotypes (2 * the number of genotype
      columns), multiple haplotypes may be compressed into a single horizontal pixel row.
      When the default red/blue coloring is in effect, pixels may be colored purple
      where the haplotypes differ.</LI>
    </UL>
</UL>
</P>
<P>
When you have finished making your configuration changes, click the
<em>Submit</em> button to return to the annotation track display
page.</P>

			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
</BODY></HTML>
