<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: Configuring VCF Tracks</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>

<BODY BGCOLOR="#FFF9D2" LINK="#0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!-----------VCF Track Help Page------------------------------>
<TR><TD ROWSPAN=2 VALIGN="top"></TD></TR>
<TR><TD>

<A NAME="Topic5"></A><BR>

	<!--outer table is for border purposes-->
	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>	
    		<TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>	
			<TABLE BGCOLOR="#D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH="100%">
			    <TR><TD><FONT SIZE="4"><A
                            NAME="TOC"></A><B>&nbsp; Configuring VCF tracks</B></FONT>
			    </TD></TR>
			</TABLE>
			<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
			    <TR><TD WIDTH=10>&nbsp;</TD><TD>
<P>
Genome Browser VCF tracks may be configured in a variety of ways to
highlight different aspects of the displayed information.
Click <A HREF="vcf.html" TARGET=_BLANK>here</A> for more information
on VCF custom track creation.
</P>
<P>
If the VCF file contains genotype columns for at least two samples (four
haplotypes), then a haplotype sorting display can be configured:
<UL>
<LI><B>Enable Haplotype sorting display:</B>
    If checked, then each sample's phased and/or homozygous genotypes are split
    into haplotypes, clustered by similarity around a central variant, and sorted
    for display by their position in the clustering tree.  The tree (or as much of
    the tree as space allows) is drawn in the label area next to the track image.
    Leaf clusters, in which all haplotypes are identical (at least for the span
    of variants used in clustering), are colored purple.
    Each variant is drawn as a vertical column, using color to distinguish between
    reference alleles and alternate alleles of the horizontally running haplotypes.
    If unchecked, then the display is the same as for VCF without genotypes:
    a stacked bar graph of the top two alleles, showing the proportion of alleles
    if allele counts are available.
    This is checked (enabled) by default.
    The following options are applicable only when enabled.</LI>
    <UL>
    <LI><B>Haplotype sorting order:</B>
      Haplotypes are sorted using a distance function that uses a central variant;
      differences between haplotypes are penalized with weights that decrease for
      each successive variant away from the central variant.  By default, the median
      variant in the window is used.  By clicking on a variant in the display, you
      will get the option to always use that variant when it is in the current view.</LI>
    <LI><B>Haplotype coloring scheme:</B>
      There are three ways that reference and alternate alleles can be colored:
      <UL>
      <LI>By default, the reference allele is invisible and the alternate allele is black.
          When multiple haplotypes must be combined into the same pixel row, grayscale is
	  used to shade according to the proportions of reference and alternate alleles.
	  The central variant has a thin purple outline.
	  Extra pixel rows at the top and bottom show the locations of variants in case
	  they are hard to see due when the invisible reference allele is the major allele.
	  Variants used in clustering have purple marks in these rows; variants outside the
	  clustered regions have black marks.</LI>
      <LI>The reference allele is blue and the alternate allele is red.
          Purple indicates a mix of reference and alternate alleles.
	  The central variant has a thick black outline.</LI>
      <LI>Both alleles are colored using the same color scheme as when there are no genotypes:
          A is red, C is blue, G is green and T is magenta.
	  Gray indicates a mix of reference and alternate alleles.
	  The central variant has a thick black outline.</LI>
      </UL>
      In all coloring modes, if some alleles in a haplotype are undefined,
      a pale yellowish color is used for those alleles.
    </LI>
    <LI><B>Haplotype clustering leaf shape:</B>
      Leaf clusters are collections of identical haplotypes.  By default, they are
      drawn as open triangles &lt; .  They can also be displayed as open rectangles [.</LI>
    <LI><B>Haplotype sorting display height:</B>Height in pixels of the haplotype sorting
      display.  If this is fewer than the number of haplotypes (2 * the number of genotype
      columns), some horizontal pixel rows must represent multiple haplotypes; differing
      haplotypes' colors will be combined according to the selected coloring scheme.
      </LI>
    </UL>
</UL>
</P>
<P>
Variants can be filtered out of the display according to several properties:
<UL>
<LI><B>Exclude items with QUAL score less than <em>N</em>:</B>
If the checkbox is checked, then all variants whose QUAL column has a
non-numeric value (e.g. &quot;.&quot;) or a value less than <em>N</em>
are excluded from display. By default, the checkbox is not checked
and <em>N</em> is 0.</LI>
<LI><B>Exclude items with these FILTER values:</B>
This option appears only if the VCF header defines at least one FILTER code.
There is a checkbox for each code defined in the header.
If checked, then all variants with that code in the FILTER
column are excluded from display. By default, no checkboxes are
checked, so all variants are displayed regardless of FILTER column values.</LI>
<LI><B>Minimum minor allele frequency (if INFO column includes AF or AC+AN):</B>
If a variant's INFO field includes AF (alternate allele frequency) or both AC and AN
(alternate allele count and total number of alleles), then its minor allele
frequency can be compared against this threshold.  If the minor allele frequency
is less than the threshold, the variant will not be displayed.
</LI>
</UL>
</P>
<P>
When you have finished making your configuration changes, click the
<em>Submit</em> button to return to the annotation track display
page.</P>

			        </TD>
			        <TD WIDTH=15></TD>
			    </TR>
			</TABLE>
		    <BR></TD></TR>
		</TABLE>
	    </TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
</BODY></HTML>
