<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<TITLE>UCSC Genome Browser: phastCons File Format</TITLE>
	<LINK REL="STYLESHEET" HREF="../../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
  <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../../images/title.jpg"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
  <TR><TD COLSPAN=2 CELLPADDING=10>	
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
      <TR><TD>	
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	  <TR><TD>	
	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../../images/hr.gif" WIDTH=100%>
	      <TR><TD>
		<FONT SIZE="4"><b>&nbsp;  phastCons File Format</b></FONT>
	      </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	      <TR><TH HEIGHT=10></TH></TR>
	      <TR><TD WIDTH=10>&nbsp;</TD><TD>
		<P>
		<em>phastCons</em> data files contain the compressed 
		conservation scores that underlie the Conservation annotation
		track and the phastCons table. For a detailed description
		of the algorithm used to produce the scores, see the
		Genome Browser description page associated with the 
		Conservation track. 
		<P>
		<B>File Format</B> (assemblies released Nov. 2004 and later)
		<P>
		When uncompressed, the file contains a declaration line and
		one column of data in wiggle table fixed-step format: 
		<FONT color="#0000FF"><PRE><TT>  fixedStep chrom=scaffold_1 start=3462 step=1
  0.0978 
  0.1588
  0.1919
  0.1948 
  0.1684 </TT></PRE></FONT>
		<P>
		<B>1. Declaration line: </B> The declaration line specifies the 
		starting point of the data in 
		the assembly. It consists of the following fields:
		<UL>
		<LI>
		<em><B>fixedStep</B></em> -- keyword indicating the wiggle
		track format used to write the data. In <em>fixed step</em>
		format, the data is single-column with a fixed
		interval between  values.
		<LI>
		<em><B>chrom</B></em> --  chromosome or scaffold on which 
		first value is located.
		<LI>
		<em><B>start</B></em> -- position of first value on 
		chromosome or scaffold specified by <em>chrom</em>. NOTE: 
		Unlike most Genome Browser coordinates, these are one-based.
		<LI>
		<em><B>step</B></em> -- size of the interval (in bases) 
		between values.
		</UL>
		A new declaration line is inserted in the file when the 
		<em>chrom</em> value changes, when a gap is encountered 
		(requiring a new <em>start</em> value), or when the 
		<em>step</em> interval changes.
		<P>
		<B>2. Data lines:</B> The first data value below the header 
		shows the score 
		corresponding to the position specified in the header. 
		Subsequent score values step along the assembly in one-base 
		intervals. The score shows the posterior probability that 
		phastCons's phylogenetic hidden Markov model (HMM) is in its 
		most-conserved state at that base position.
		<P>
		<HR>
		<B>File Format</B> (assemblies prior to Nov. 2004)
		<P>
		When uncompressed, the data file contains two columns:
		
		<FONT color="#0000FF"><PRE><TT>  294	0.0953	
  295	0.0948	
  296	0.0943	
  297	0.0936	
  298	0.0929	
  299	0.0921</TT></PRE></FONT>
		Column #1 contains a one-based position coordinate.
		Column #2 contains a score showing the posterior probability 
		that phastCons's phylogenetic hidden Markov model (HMM) is in 
		its most conserved state at that base position.
		<P>
		<HR>
		<B>References for phastCons</B>
		<P>
		Siepel A and Haussler D (2005). Phylogenetic hidden Markov 
		models. In R. Nielsen, ed., <em>Statistical Methods in Molecular
		Evolution</em>, pp. 325-351, Springer, New York. </P>
		<P>
		Siepel, A., Bejerano, G., Pedersen, J.S., Hinrichs, A., Hou, M.,
		Rosenbloom, K., Clawson, H., Spieth, J., Hillier, L.W., 
		Richards, S., Weinstock, G.M., Wilson, R. K., Gibbs, R.A., Kent,
		W.J., Miller, W., and Haussler, D.
		<A HREF="http://www.genome.org/cgi/doi/10.1101/gr.3715005"
		TARGET=_blank>Evolutionarily conserved elements in vertebrate, 
		insect, worm, and yeast genomes</A>.
		<em>Genome Res.</em> <B>15</B>, 1034-1050 (2005).</P>
		<P>
		For a discussion of the methods used to calculate the 
		phastCons scores, see the description page for the hg17 
		Conservation track in the Genome Browser.
	      </TD><TD WIDTH=15></TD></TR>
	    </TABLE>
	    <BR>
	  </TD></TR>
	</TABLE>
      </TD></TR>
    </TABLE>
  </TD></TR>
</TABLE>
</BODY></HTML>

