<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
        <meta http-equiv="Content-Type" content="text/html;CHARSET=iso-8859-1">
        <title>UCSC Genome Browser: Querying the Genome Browser</title>
        <link rel="STYLESHEET" href="../../style/HGStyle.css">
        <script type="text/javascript" src="../../js/jquery.js"></script>
        <script type="text/javascript" src="../../js/staticDoc.js"></script>
</head>
<body bgcolor="#fff9d2" link="#0000cc" vlink="#330066" alink="#330066">
<a name="top"></A>

<table border="0" cellpadding="0" cellspacing="0" width="100%">
    <tr><th colspan="2" align="left"><img src="../../images/title.jpg" alt="UCSC Genome Bioinformatics"></th></tr>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../../js/jquery.js'></script>
<script type='text/javascript' SRC='../../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../../style/nice_menu.css' TYPE='text/css' />
<tr><td colspan="2" height="40"><!--#include virtual="../../cgi-bin/hgMenubar"--></td></tr>
<!--Content Tables-->
    <tr><td colspan="2" cellpadding="10">
    <!--outer table is for border purposes-->
        <table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
            <tr><td>
                <table bgcolor="#fffee8" width="100%" border="0" cellspacing="0" cellpadding="0">
                    <tr><td>
            <table bgcolor="#d9e4f8" background="../../images/hr.gif" width="100%">
                <tr><td> <font size="4"><b>&nbsp; Querying the Genome Browser</b></font>
                            </td></tr>
                        </table>
            <table bgcolor="#fffee8" width="100%" cellpadding="5">
                <tr><td width="15">&nbsp;</td><td colspan="2">
                    <p>
                    You may browse to the default genome position of an assembly when clicking the "submit"
                    button or you may specify a genome position in a variety of formats.
                    </p>

                    <p>
                    Valid position queries can include:
                    <ul>
                      <li>Chromosome numbers</li>
                      <li>Chromosomal coordinate ranges</li>
                      <li>Gene names</li>
                      <li>Accession numbers</li>
                      <li>An mRNA, EST or STS marker</li>
                      <li>Keywords from the Genbank description of an mRNA</li>
                    </ul>
                    </p>

                    <p>
                    To specify a genome position:
                    <ol>
                      <li>Select the desired clade, genome and assembly</li>
                      <li>Enter the desired query in the &quot;search term&quot; box (see sample queries below)</li>
                      <li>Click the "submit" button</li>
                    </ol>
                    </p>

                    <p>
                    <em><b>Note:</b></em> A query may have multiple results. If this is the case, a results page will appear
                    listing each result along with the track it is associated with. Once selected, the result will
                    be displayed in the Browser with a highlighted tag, making it easier to identify.
                    </p>

                    <hr>
                    <h3>Sample Queries</h3>

                    Below is a list of examples that might be used to query the Genome Browser. Note that not every query
                    listed here will produce a result in every assembly. The list serves only to illustrate the different
                    types of queries that can be performed.

                 </td><td width="15"></td></tr>
                 <tr height="10"></tr>
                 <tr><td>&nbsp;</td>
                   <td width="200"><b>Query:</b></td>
                   <td><b>Genome Browser Response:</b></td></tr>
                 <tr height="10"></tr>
                 <tr><td>&nbsp;</td>
                    <td>chr7</td>
                    <td>Displays all of chromosome 7</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>chr3:1-1000000</td>
                    <td>Displays the first million bases of chromosome 3, counting from the p-arm telomere</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>chr3:1000000+2000</td>
                    <td>Displays a region of chromosome 3 that spans 2000 bases, starting with position 1000000</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>chrUn_gl000212</td>
                    <td>Displays all of the unplaced contig gl000212</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>chr6_GL285171_random</td>
                    <td>Displays the non-localized contig GL285171</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>GL192360</td>
                    <td>Displays all of scaffold GL192360</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>20p13</td>
                    <td>Displays the region for band p13 on chromosome 20</td></tr>
                 <tr height="15"></tr>
                 <tr><td>&nbsp;</td>
                    <td>374180</td>
                    <td>Displays the region containing Entrez Gene identifier 374180</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>AA205474</td>
                    <td>Displays the region containing the EST with Genbank accession AA205474 in the BRCA1 cancer gene on chromosome 17</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>AC008101</td>
                    <td>Displays the region containing the clone with Genbank accession AC008101</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>AF083811</td>
                    <td>Displays the region containing the mRNA with Genbank accession number AF083811</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>NM_017414</td>
                    <td>Displays the region containing RefSeq identifier NM_017414</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>NP_059110</td>
                    <td>Displays the region containing protein accession number NP_059110</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>PRNP</td>
                    <td>Displays the region containing the HUGO Gene Nomenclature Committee identifier PRNP</td></tr>
                 <tr height="25"></tr>
                 <tr><td>&nbsp;</td>
                    <td nowrap>RH18061;RH80175<br>
                        15q11;15q13<br>
                        rs1042522;rs1800370</td>
                    <td nowrap>Displays the region between genome landmarks, such as the STS markers RH18061 and RH80175, or chromosome<br>
                        bands 15q11 to 15q13, or SNPs rs1042522 and rs1800370. This syntax may also be used for other range queries,<br>
                        such as between uniquely determined ESTs, mRNAs, refSeqs, etc.</td></tr>
                <!-- <tr><td>&nbsp;</td>
                    <td>15q11;15q13</td>
                    <td>bands 15q11 to 15q13, or SNPs rs1042522 and rs1800370. This syntax may also be used for other range queries,</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>rs1042522;rs1800370</td>
                    <td>such as between uniquely determined ESTs, mRNAs, refSeqs, etc.</td></tr> -->
                 <tr height="25"></tr>
                 <tr><td>&nbsp;</td>
                    <td>pseudogene mRNA</td>
                    <td>Lists transcribed pseudogenes, but not cDNAs</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>p53</td>
                    <td>Lists mRNAs related to the p53 tumor suppressor</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>T-cell receptor</td>
                    <td>Lists mRNAs for T-cell receptor genes in Genbank</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>breast cancer</td>
                    <td>Lists mRNAs associated with breast cancer</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>homeobox caudal</td>
                    <td>Lists mRNAs for caudal homeobox genes</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>zinc finger</td>
                    <td>Lists zinc finger mRNAs</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>kruppel zinc finger</td>
                    <td>Lists only kruppel-like zinc fingers</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>huntington</td>
                    <td>Lists candidate genes associated with Huntington's disease</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>zahler</td>
                    <td>Lists mRNAs deposited by a scientist named Zahler</td></tr>
                 <tr><td>&nbsp;</td>
                    <td>Evans,J.E.</td>
                    <td>Lists mRNAs deposited by co-author J.E. Evans</td></tr>
                 <tr height="15"></tr>
                 <tr><td>&nbsp;</td><td colspan="2">
                 Use this last format for author queries. Although Genbank requires the search format <em>Evans JE</em>,
                 internally it uses the format <em>Evans,J.E.</em>
                 <p>&nbsp;</p>
                 </td></table>
                     </td></tr>
                </table>
             </td></tr>
        </table>
</table>
</body>
</html>
