<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=297>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://genome.wustl.edu/genomes/detail/pongo-abelii/"
    TARGET=_blank ALIGN=RIGHT >
    <IMG SRC="/images/Pongo_pygmaeus_abelii.jpg" WIDTH=282 HEIGHT=200 ALT="Orangutan"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Pongo pygmaeus abelii</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      Photo courtesy of NHGRI
      (<A HREF="http://www.genome.gov/dmd/img.cfm?node=Photos/Animals/Primates&id=79121" 
      TARGET=_blank>press photos</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jul. 2007 <em>Pongo pygmaeus abelii</em> draft assembly (WUSTL version 
Pongo_albelii-2.0.2) was produced by the 
<A HREF="http://genome.wustl.edu/genomes/detail/pongo-abelii/"
TARGET=_blank>Genome Sequencing Center at Washington University School of
Medicine in St. Louis</A> (WUSTL). For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/6678/" TARGET="_blank">
P_pygmaeus_2.0.2</A> in the NCBI Assembly database.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
orangutan genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chr22</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays all of chromosome 22</TD>
    </TR>
    <TR><TD VALIGN=Top>chr1:1-1000000</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays first million bases of chr 1</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr1:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 1 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>AF512949</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Displays region of mRNA with GenBank accession
		number AF512949</TD>
    </TR>
    <TR><TD VALIGN=Top>CR542506</TD>
        <TD WIDTH=14>&nbsp;</TD>
      	<TD VALIGN=Top>Displays region of EST with GenBank accession
		CR542506 on chr 22</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>paabo</TD>
        <TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by scientist named Paabo</TD>
    </TR>
    <TR><TD VALIGN=Top>Wiemann,S.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author S. Wiemann</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Wiemann S</em>, internally it
	uses the format <em>Wiemann,S.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The <em>Pongo pygmaeus abelii</em> assembly (6X coverage) consists of whole 
genome shotgun data from primary donor-derived reads obtained from 
&quot;Susie&quot;, a female Sumatran orangutan housed at the Gladys Porter Zoo 
in Brownsville, TX.
The combined sequence reads were assembled using the PCAP software (Huang, 
2003) and filtered for all known non-orangutan sequence 
contaminants using stringent parameters derived by eliminating detectable
global mis-assemblies larger than 50 kb. Mis-assemblies (interchromosomal 
cross-overs) were determined by aligning the orangutan genome against the 
human genome. 
Sequences were obtained from plasmids, fosmids, and BAC-end
sequences from CHORI-276, also obtained from &quot;Susie&quot;. A fingerprint
map with a target of 12X clone coverage is currently in progress.
</P>
<P>
The assembly data were aligned against the human genome at UCSC
(B. Raney) using BLASTZ (Schwartz, 2003) to align and score
non-repetitive orangutan regions against repeat-masked human
sequence. The alignment chains differentiated between orthologous and
paralogous alignments (Kent, 2003), and only &quot;reciprocal best&quot;
alignments were retained in the final alignment set. The orangutan AGP files
were generated from these alignments in a manner similar to that
already described (The Chimpanzee Genome Sequencing and Analysis
Consortium, 2005). Documented inversions based primarily on FISH data
(Rocchi, personal communication) as well as inversions suggested by the
assembly and supported by additional mapping data (e.g. fosmid end
sequences against the human assembly, (Chen and Eichler, personal
communication)) were introduced. The alignments to human chromosome 2 were
separated into organutan chromosomes 2A and 2B.  Finally, 78
finished BAC (CHORI-273) clones were integrated into the final
chromosomal sequences.
</P>
<P>
Of the 3.09 Gb of total sequence, 3.08 Gb are ordered and oriented along
the chromosomes. Gap sizes between supercontigs were estimated based on
their size in human, with a maximum gap size of 30 kb allowed.
Click the &quot;Sequences&quot; link above for a list of the chromosomes
available in this assembly. The mitochondrial sequence is also available as 
the virtual chromosome &quot;chrM&quot;. For more information about this
assembly, see the WUSTL 
<A HREF="http://genome.wustl.edu/genomes/detail/pongo-abelii/"
TARGET=_blank><em>Pongo abelii</em></A> web page.</P>
</P>
<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/ponAbe2/">FTP server</A> 
or the 
<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#orangutan">Downloads</A> 
page. 
These data have <A HREF="../goldenPath/credits.html#orangutan_use">specific 
conditions for use</A>. 
The orangutan browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#orangutan_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to this release. </P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>well-ordered</B>.
