<!DOCTYPE html>
<!--#set var="TITLE" value="Genome Browser Local Installation" -->
<!--#set var="ROOT" value=".." -->

<!-- Relative paths to support mirror sites with non-standard GB docs install -->
<!--#include virtual="$ROOT/inc/gbPageStart.html" -->

<h1>Installation of a UCSC Genome Browser on a local machine (&quot;mirror&quot;)</h1>

<h2>Contents</h2>
<ul>
  <li>
  <a class="toc" href="#considerations-before-installing-a-genome-browser">Considerations before 
  installing a Genome Browser</a></li>
  <li>
  <a class="toc" href="#installing-a-genome-browser-locally-with-the-gbic-installer">Installing a 
  Genome Browser locally with the GBiC installer</a></li>
  <li>
  <a class="toc" href="#manual-installation-instructions">Manual installation instructions</a></li>
  <li>
  <a class="toc" href="#using-udr-to-speed-up-downloads">Using UDR to speed up downloads</a></li>
  <li>
  <a class="toc" href="#getting-help">Getting help</a></li>
</ul>

<a name="considerations-before-installing-a-genome-browser"></a>
<h2>Considerations before installing a Genome Browser</h2>
<p>
Like most web servers, running a Genome Browser installation at your institution, even for your own 
department, requires a Unix machine, disk space (6TB for hg19), and the resources to update the site
and underlying OS regularly. You may want to consider the following alternatives before embarking on
a full UCSC Genome Browser installation directly on your server.</p>
<p> 
<strong>Embed the Genome Browser graphic in your web page:</strong> If you only want to include a 
Genome Browser view in your web page for an existing genome, you can use an &lt;iframe&gt; tag and 
point it to <a href="http://genome.ucsc.edu/cgi-bin/hgRenderTracks" 
target="blank">http://genome.ucsc.edu/cgi-bin/hgRenderTracks</a>, which will show only the main 
browser graphic without any decorations.</p>
<p>
You can then use various parameters to adapt this graphic to your use case, e.g., set the
displayed position, switch tracks on/off or highlight a range with a color. See our
<a href="https://genome.ucsc.edu/goldenpath/help/customTrack.html#SHARE" target="_blank">manual 
pages</a> for a list of the parameters.</p> 
<p>
<strong>Use assembly hubs</a>:</strong> 
<a href="https://genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Assembly"
target=_"blank">Assembly hubs</a> allow you to prepare any FASTA file, add annotations
and use the Genome Browser to visualize it. All you need is a webserver where you save the indexed 
genome sequence and files to annotate it, e.g. in BAM, bigBed or bigWig format.</p>
<p>
Pros:</p>
<ul>
  <li>
  No installation, no updates, no long-term UNIX support necessary.</li>
  <li>
  Stable for many years, the link to the assembly hub can be put into a manuscript.</li>
  <li>
  For commercial users, no license is required.</li>
</ul>
<p>
Cons:</p>
<ul>
  <li>
  You need access to a public webserver to store the files.</li>
  <li>
  Data is rendered at UCSC. Private data can be password-protected and loaded through https and 
  restricted to UCSC&#39;s IP address, but has to be located on the web and accessible by the UCSC 
  webserver.</li>
  <li>
  For BLAT searches in your genome, you have to start a BLAT server yourself on a Unix server.</li>
  <li>
  If your hub includes a large number of annotation or HAL (multiple alignment) files and is 
  located far from Santa Cruz, the display performance of assembly hubs may be slower than a local 
  mirror. This issue can be resolved by using a UCSC mirror closer to the assembly hub (e.g., use 
  genome-euro.ucsc.edu for assembly hubs located on servers in Europe or genome-asia.ucsc.edu for 
  those in Asia). Alternatively, you can improve performance by moving your hub data to a webspace 
  provider closer to Santa Cruz or by using a content distribution network, where all content is 
  mirrored and the closest location is chosen by your provider.</li>
</ul>
<p>
<strong>Use Genome Browser in a Box (GBiB):</strong> 
<a href="https://genome.ucsc.edu/goldenPath/help/gbib.html" target="_blank">Genome Browser in a 
Box</a> is a fully configured virtual machine that includes Apache and Mysql, and has behavior 
identical to the UCSC website. GBiB loads genome data from the UCSC download servers on the fly. 
Website and data updates are applied automatically every two weeks. By default, GBiB uses the 
VirtualBox virtualization software, allowing it to run on any operating system, Windows, OSX and 
Linux. It does not require VirtualBox; the virtual machine image can easily be converted to e.g., 
VMWare or HyperV. For increased privacy, you can download the genomes and annotation tracks you 
need and use GBiB offline on a laptop.</p>
<p>
Pros:</p>
<ul>
  <li>
  Relatively simple to install: download and double-click.</li>
  <li>
  By default, software and data updates are managed remotely by UCSC, via an update script run every
  week.</li>
  <li>
  Best performance when rendering local BAM/bigWig/bigBed files.</li>
  <li>
  No Unix webserver needed, runs on any OS.</li>
  <li>
  For commercial users: easier click-through licensing compared to a full multi-seat, manual 
  license.</li>
</ul>
<p>
Cons:</p>
<ul>
  <li>
  Requires the free VirtualBox software or a commercial Virtualization system.</li>
  <li>
  By default requires opening at least three outgoing ports to UCSC for MySQL, Rsync downloads and 
  BLAT in your firewall. Once all data is downloaded, no open ports are needed.</li>
  <li>
  For maximum browsing speed, can require up to 2-6TB to store all genome annotation tracks, like a 
  manual local installation.</li>
</ul>
<p>
If none of the above options fulfill your needs, consider setting up a full local mirror of the 
UCSC website. We support mirror site installations as time allows, and have many functional
mirrors of the Genome Browser worldwide.</p>
<p>
A license is required for commercial download and/or installation of the Genome Browser binaries and
source code. No license is needed for academic, nonprofit, and personal use. To purchase a
license, see our <a href="https://genome.ucsc.edu/license/index.html" target="_blank">license
instructions</a> or visit the <a href="https://genome-store.ucsc.edu/" target="_blank">Genome 
Browser store</a>.</p>

<a name="installing-a-genome-browser-locally-with-the-gbic-installer"></a>
<h2>Installing a Genome Browser locally with the GBiC installer</h2>
<p>
If you do not want to use our prepared virtual machine Genome Browser in a Box, we provide a
Genome Browser in the Cloud (GBiC) <a href="https://genome.ucsc.edu/goldenPath/help/gbic.html" 
target="_blank">installation program</a> that sets up a fully functional mirror on all major Linux 
distributions. It has been tested on Ubuntu 14/16, RedHat/CentOS 6 and 7, and Fedora 20. Preferably,
the installation should be performed on a fresh Linux installation, as it deactivates the default 
site config file in Apache and fills the MySQL directory with numerous databases. The easiest way 
to accomplish this is to run the GBiC program in a new virtual machine. The program also works on 
Docker and cloud computing virtual machines, and has been tested on those sold by Amazon, Microsoft 
and Google.</p>
<p>
Like GBiB, the mirror installed by GBiC can load the annotation data from either UCSC or a local 
database copy. If you load data from UCSC and use a cloud computing provider, it is highly advisable
to run your instances in the US West Coast / San Francisco Bay Area or San Jose data centers; 
otherwise, data-loading may be very slow.</p>
<p>
To run the installation program, please see the 
<a href="https://genome.ucsc.edu/goldenPath/help/gbic.html" target="_blank">GBiC user's 
guide</a>.</p>

<a name="manual-installation-instructions"></a>
<h2>Manual installation instructions</h2>
<p>
If the installation program does not work on your Linux distribution or you prefer to make 
adaptations to your mirror, we also provide <a href="mirrorManual.html" target="_blank">step-by-step
installation instructions</a> that cover the configuration of Apache, MySQL, the Genome Browser 
CGIs, temporary file removal data-loading through proxies, and other topics.</p>
<p>
The following external websites were not created by UCSC staff and are of varying quality, but may 
be helpful when installing on unusual platforms:</p>
<ul>
  <li>
  <a href="http://bergmanlab.genetics.uga.edu/?p=1111" target="_blank">Installation on 
  CentOS</a></li>
  <li>
  <a href='http://enotacoes.wordpress.com/2009/09/03/installing-a-minimal-ucsc-mirror-in-ubuntu-jaunty-64-bits/' 
  target="_blank">Installation on Ubuntu</a></li>
  <li>
  <a href="http://www.bioinformatics.uni-muenster.de/download/ucsc" target="_blank">Installation
  into a FreeBSD jail</a>, by Norbert Grundmann, Universitaet Muenster, Germany</li>
  <li>
  <a href="http://bergmanlab.genetics.uga.edu/?p=32" target="_blank">Compilation of the Kent 
  source tree on OSX with fink</a>, by Casey Bergman, Manchester, UK</li>
  <li>
  <a href="http://www.oliverelliott.org/article/bioinformatics/tut_genomebrowser/"
  target="_blank">Browser installation and new genome setup</a>, by Oliver Elliott, Columbia 
  University, USA</li>
  <li>
  <a href="http://genomewiki.ucsc.edu/index.php/Browser_Installation" target="_blank">Installation 
  notes in our wiki</a></li>
</ul>

<a name="using-udr-to-speed-up-downloads"></a>
<h2>Using UDR to speed up downloads</h2> 
<p>
<a href="https://github.com/LabAdvComp/UDR" target="_blank">UDR</a> (UDT Enabled Rsync) is a 
download protocol that is very efficent at sending large amounts of data over long distances. UDR 
utilizes rsync as the transport mechanism, but sends the data over the UDT protocol. UDR is not 
written or managed by UCSC. It is an open source tool created by the 
<a href="http://www.labcomputing.org/" target="_blank">Laboratory for Advanced Computing</a> at the 
University of Chicago. It has been tested under Linux, FreeBSD and Mac OSX, but may work under other
UNIX variants. The source code can be obtained through <a href="https://github.com/LabAdvComp/UDR" 
target="_blank">GitHub</a>. When using the GBiC installation program, the <code>-u</code> option 
will use UDR for all downloads.</p>
<p>
If you manually download data only occasionally, there is no need to change your method;
continue to visit our download server to download the files you need. This new protocol has been 
put in place primarily to facilitate quick downloads of huge amounts of data over long 
distances.</p> 
<p>
With typical TCP-based protocols like http, ftp and rsync, the transfer speed slows as the distance
between the download source and destination increases. Protocols like UDT/UDR allow for many UDP 
packets to be sent in batch, thus allowing for much higher transmission speeds over long distances. 
UDR will be especially useful for users who are downloading from locations distant to California. 
The East coast of the U.S. and the international community will likely see much higher download 
speeds when using UDR vs. rsync, http or ftp.</p> 
<p>
If you need help building the UDR binaries or have questions about how UDR functions, please read 
the documentation on the GitHub page and if necessary, contact the UDR authors via the GitHub page. 
We recommend reading the documentation on the UDR GitHub page to better understand how UDR works. 
UDR is written in C++. It is Open Source and is released under the Apache 2.0 License. In order for 
it to work, you must have rsync installed on your system.</p> 
<p>
For your convenience, we offer a binary distribution of UDR for Red Hat Enterprise Linux 
6.x (or variants such as CentOS 6 or Scientific Linux 6). You will find both a 64-bit and 32-bit 
rpm <a href="http://hgdownload.cse.ucsc.edu/admin/udr">here</a>.</p> 
<p>
Once you have a working UDR binary, either by building one from source or installing the rpm, you 
can download files from either of our our download servers in a fashion similar to rsync. For 
example, using rsync, all of the MySQL tables for the hg19 database can be downloaded using 
the following command:</p>
<pre><code>rsync -avP rsync://hgdownload.cse.ucsc.edu/mysql/hg19/ /my/local/hg19/
rsync -avP hgdownload.cse.ucsc.edu::mysql/hg19/ /my/local/hg19/</code></pre>
<p>
With UDR, the syntax for downloading the same data is:</p>
<pre><code>udr rsync -avP hgdownload.cse.ucsc.edu::mysql/hg19/ /my/local/hg19/ </code></pre> 
<p>
If you installed the rpm, use the <code>man udr</code> command to view a man page with more 
information. If you installed from source, please refer to the UDR GitHub page for more details on 
the capabilities of UDR and how to use it.</p> 
<p>
UDR establishes connections on TCP/9000, then transmits the data stream over UDP/9000-9100. Your 
institution may need to modify its firewall rules to allow inbound and outbound ports TCP/9000 and 
UDP/9000-9100 from either of the two download machines.</p> 

<a name="getting-help"></a>
<h2>Getting help</h2> 
<p>
If you have difficulties installing or using UDR on your system, contact the Laboratory for Advanced
Computing through their GitHub <a href="https://github.com/LabAdvComp/UDR" 
target="_blank">page</a>.</p> 
<p> 
For questions about installing and mirroring the UCSC Genome Browser, contact the UCSC mailing list 
<a href="mailto:&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;">&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;</a>. 
 <!-- above address is genome-mirror at soe.ucsc.edu --> 
<strong><span class="gbsWarnText">Messages sent to this address will be posted to the moderated genome-mirror mailing list, 
which is archived on a SEARCHABLE PUBLIC</span> 
<a href="http://groups.google.com/a/soe.ucsc.edu/group/genome-mirror" 
target="_blank">Google Groups forum</a></strong>.</p>  

<!--#include virtual="$ROOT/inc/gbPageEnd.html" -->
