<HTML>
  <HEAD>
    <META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
    <META name="description" content="UCSC Genome Browser">
    <META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, mirror, genome browser source, bioinformatics, gene prediction, SNP, EST, mRNA">
    <TITLE>UCSC Genome Browser: Mirror Instructions</TITLE>
    <LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
  </HEAD>

  <BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#6600FF">
  <A NAME="TOP"></A>

    <TABLE BORDER=0 WIDTH="100%">
      <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="/images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

      <noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
      <script type='text/javascript' SRC='../js/jquery.js'></script>
      <script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
      <LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
      <TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../cgi-bin/hgMenubar"--></TD></TR>

<!--Content Tables------------------------------------------------------->
    <TR><TD CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
	    	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; Procedure for Creating a Mirror Site for the UCSC Genome Browser</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD>
			<TD>
			    <P>
                            The <a href="http://genome-source.cse.ucsc.edu/gitweb/?p=kent.git;a=tree;f=src/product"
                            target="_blank">most complete and up-to-date instructions for setting up a mirror</a>
                            can be found in our source tree. You may choose to set up either a full mirror browser
                            or a partial one, depending on your disk space and needs. Additionally, these blog
                            posts may be helpful with setting up a mirror site on
                            <A HREF="http://bergmanlab.smith.man.ac.uk/?p=1111" TARGET=_blank>CentOS</A>
			    and <A HREF="http://enotacoes.wordpress.com/2009/09/03/installing-a-minimal-ucsc-mirror-in-ubuntu-jaunty-64-bits/" TARGET=_blank>Ubuntu</A>.
			    </P>
                            <p>
                            A <B>license</B> is required for commercial download 
                            and/or installation of the Genome Browser binaries 
                            and source code.
                            No license is needed for academic, nonprofit, and 
                            personal use.  To purchase a license, see our
                            <A HREF="../license/index.html"> License Instructions</A> or visit
			    the <A HREF="https://genome-store.ucsc.edu">Genome Browser store</A>. 
                            </P>

			    <B> Space required:</B>
			    <P>
			    The amount of data available in the Genome Browser is
			    growing constantly. To determine the size of any of
			    the download directories mentioned in these instructions,
			    use the rsync &quot;-n&quot; option on the directory prior
			    to actually transferring the data.  For instance, to find
			    the size of the /gbdb directory, run:<br>
<PRE>
rsync -navP rsync://hgdownload.cse.ucsc.edu/gbdb/
</PRE>
The rsync options used in this command are:
<PRE>
-n, --dry-run         show what would have been transferred
-a, --archive         archive mode, equivalent to -rlptgoD
-v, --verbose         increase verbosity
-P                    equivalent to --partial --progress
</PRE>
			    <P>
			    Mirror site questions may be directed to the 
			    mailing list <A HREF="mailto:&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;">&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;</A>.
			    <!-- above address is genome-mirror at soe.ucsc.edu -->
			    <b><font color="red">Messages sent to this address will be posted to the 
			    moderated genome-mirror mailing list, which is 
			    archived on a SEARCHABLE PUBLIC
			    <a HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-mirror" TARGET="_blank">Google Groups forum</a></font></b>. 
			    <P>
			    <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-mirror?hl=en"
			    TARGET=_blank>Subscribe</A> to the genome-mirror 
		 	    mailing list. </P>
			</TD><TD WIDTH=15></TD>
		    </TR>
		</TABLE>
		<BR>
	    </TD></TR>
	</TABLE>
	<BR>

	<A NAME="UDR"></A>
    <TR><TD CELLPADDING=10>	
	<TABLE BGCOLOR="fffee8" WIDTH="100%" BORDERCOLOR="888888" BORDER=1>
	    <TR><TD>
	    	<TABLE BGCOLOR="D9E4F8" BACKGROUND="/images/hr.gif" WIDTH=100%>
		    <TR><TD>
			<FONT SIZE="4"><B>&nbsp; Using UDR to Speed Up the Download Process</B></FONT>
		    </TD></TR>
		</TABLE>
		<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
		    <TR><TH HEIGHT=10></TH></TR>
		    <TR><TD WIDTH=10>&nbsp;</TD>
			<TD>
			<P><A HREF="https://github.com/LabAdvComp/UDR" TARGET="_blank">UDR</A> 
			(UDT Enabled Rsync) is a download protocol that is very efficent
			at sending large amounts of data over long distances. UDR utilizes rsync 
			as the transport mechanism, but sends the data over the UDT protocol.
			UDR is not written or managed by UCSC. It is an open source tool 
			created by the
			<A HREF="http://www.labcomputing.org/" TARGET="_blank">Laboratory for Advanced Computing</A> 
			at the University of Chicago. It has been
			tested to work under Linux, FreeBSD and Mac OSX, but may
			work under other UNIX variants. The source code can be
			obtained through 
			<A HREF="https://github.com/LabAdvComp/UDR"
			TARGET="_blank">GitHub</A>.
			</P>
			<P>If you are a casual or occasional manual downloader of data, there 
			is no need to change your method; continue to visit our download server 
			to download the files you need. This new protocol has been put in place 
			to enable huge amounts of data to be downloaded quickly over long distances.
			</P>
			<P>Typical TCP-based protocols like http, ftp and rsync
			have a problem in that the further away the download
			source is from you, the slower the speed becomes.
			Protocols like UDT/UDR allow for many UDP packets to be
			sent in batch, thus allowing for much higher transmit
			speeds over long distances. UDR will be especially
			useful for users who are downloading from places that
			are far away from California. The US East Coat and the
			international community will likely see much higher
			download speeds by using UDR rather than rsync, http or
			ftp.
			</P>
			<P>If you need help building the UDR binaries or have
			questions about how UDR functions, please read the
			documentation on the GitHub page and if necessary,
			contact the UDR authors via the GitHub 
			page. 
			We recommend reading the documentation on the UDR GitHub
			page to better understand how UDR works. UDR is written
			in C++. It is Open Source and is released under the
			Apache 2.0 License. In order for it to work, you must
			have rsync installed on your system.
			</P>
			<P>For your convenience, we are offering a binary distribution 
			of UDR for Red Hat Enterprise Linux 6.x (or variants such as 
			CentOS 6 or Scientific Linux 6). You'll find both a 64-bit and 
			32-bit rpm 
			<A HREF="http://hgdownload.cse.ucsc.edu/admin/udr">here</A>.
			</P>
			<P>Once you have a working UDR binary, either by
			building from source or by installing the rpm, you can download
			files from either of our our download servers in a very
			similar fashion to rsync. For example, using rsync, you
			may want to download all of the MySQL tables for the
			hg19 database using the following command:
<PRE>			
rsync -avP rsync://hgdownload.cse.ucsc.edu/mysql/hg19/ /my/local/hg19/
</PRE>
			Using UDR is very similar. The UDR syntax for downloading the same data would be:
<PRE>
udr rsync -avP hgdownload.cse.ucsc.edu::mysql/hg19/ /my/local/hg19/
</PRE>
			If you installed the rpm, use the <TT>man udr</TT> command for more 
			information via the man page; if you installed from source, please refer 
			to the UDR GitHub page for more details on the capabilities of UDR and 
			how to use it.
			</P>
			<P>UDR establishes connections on TCP/9000, then
			transmits the data stream over UDP/9000-9100. Your
			institution may need to modify its firewall rules to
			allow inbound and outbound ports TCP/9000 and
			UDP/9000-9100 from either of the two download machines.
			</P>
			<P>If you have difficulties installing or using UDR on your system,
			contact the Laboratory for Advanced Computing through their GitHub 
			<A HREF="https://github.com/LabAdvComp/UDR" TARGET="_blank">page</A>.
		 	</P>
			<P> If you have questions about mirroring the UCSC
			Genome Browser, direct them to the mailing list 
			<A HREF="mailto:&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;">&#103;en&#111;&#109;e&#45;&#109;&#105;&#114;&#114;&#111;&#114;&#64;&#115;&#111;&#101;.u&#99;&#115;c.e&#100;&#117;</A>.
			<!-- above address is genome-mirror at soe.ucsc.edu -->
			<b><font color="red">Messages sent to this address will be posted to the 
			moderated genome-mirror mailing list, which is 
			archived on a SEARCHABLE PUBLIC 
			<a HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-mirror" TARGET="_blank">Google Groups forum</a></font></b>. 
			</P>
			</TD><TD WIDTH=15></TD>
		    </TR>
		</TABLE>
		<BR>
	    </TD></TR>
    </TD></TR>
</TABLE>
</BODY></HTML>
