<H2>Description</H2>
<P>
This track shows mutations in the modern human lineage that rose to
fixation or near fixation since the split from the last common
ancestor with Denisovans, along with predicted functional effects from
Ensembl's
<A HREF="http://www.ensembl.org/info/docs/variation/vep/index.html"
TARGET=_BLANK>Variant Effect Predictor (VEP).</A>
</P>

<H2>Methods</H2>
<P>
Methods and analysis are described in detail in Note 19 of
<a HREF="http://www.sciencemag.org/content/suppl/2012/08/29/science.1224344.DC1/Meyer.SM.pdf"
TARGET=_BLANK>supplementary online materials</A> of (Meyer, 2012).
</P>
<P>
Whole genome
<A HREF="http://www.ensembl.org/info/docs/compara/analyses.html#epo"
TARGET=_BLANK>Enredo-Pecan-Ortheus (EPO)</A> alignments of human,
chimpanzee, gorilla and orangutan were combined with modern human genotypes
from the 1000 Genomes Project Phase 1 (1000G) to identify sites that
are fixed (&gt;99.0% frequency in 1000G) or high frequency (&gt;90.0%
frequency in 1000G) derived in modern humans and ancestral in
chimpanzee and at least one other great ape (gorilla or orangutan).
In order to avoid paralogous regions, human and chimpanzee sequences
were required to appear in only one EPO alignment block.
Some "fixed" sites are in dbSNP; these were separated out from fixed
sites not in dbSNP, so three categories of frequency are displayed:
Fixed, Fixed+dbSNP, and High Frequency.
</P>
<P>
Various quality filters were applied to Denisova genotypes:
minimum 40 PHRED genotype likelihood from the
<A HREF="http://www.broadinstitute.org/gatk/"
TARGET=_BLANK>Genome Analysis Toolkit (GATK)</A>;
minimum 30 RMS map quality score;
coverage at least 14X and at most 66X;
no sites in positions identified as systematic errors or
deemed to be of low quality due to conflicting genotype calls in a
second iteration of GATK (Note 6,
<A HREF="http://www.sciencemag.org/content/suppl/2012/08/29/science.1224344.DC1/Meyer.SM.pdf"
TARGET=_BLANK>supplementary online materials</A> of Meyer, 2012).
</P>
<P>
The derived-in-modern-human sites were intersected with the high-confidence-in-Denisova
sites and annotated using
<A HREF="http://www.ensembl.org/info/docs/variation/vep/index.html"
TARGET=_BLANK>VEP</A> to predict effects on protein structure and
transcriptional regulation.
</P>

<H2>Credits</H2>
<P>
Thanks to the
<A HREF="http://www.eva.mpg.de/"
TARGET=_BLANK>Max Planck Institute for Evolutionary Anthropology</A>
for providing the
<A HREF="http://bioinf.eva.mpg.de/download/HighCoverageDenisovaGenome/"
TARGET=_BLANK>data files</A> used for this track.
</P>

<H2>References</H2>
<P>
Meyer M, Kircher M, Gansauge MT, Li H, Racimo F, Mallick S, Schraiber JG, Jay F, Prüfer K, de Filippo C <EM>et al.</EM>
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/22936568"
TARGET=_BLANK>A High-Coverage Genome Sequence from an Archaic Denisovan Individual.</A>
<EM>Science.</EM> 2012 Aug 31. [Epub ahead of print]
<A HREF="http://www.sciencemag.org/content/suppl/2012/08/29/science.1224344.DC1/Meyer.SM.pdf"
TARGET=_BLANK>supplementary online materials</A>, Note 19</P>
