<H2>Description</H2>
<P>
This track shows over 8.8 million single nucleotide variants (SNVs)
and over 600 thousand insertions or deletions (indels)
in 61,486 unrelated individuals sequenced as part of various population genetic 
and disease-specific 
<A HREF="http://exac.broadinstitute.org/about" TARGET=_BLANK>studies</A>
collected by the
<A HREF="http://exac.broadinstitute.org/" TARGET=_BLANK>Exome Aggregation Consortium (ExAC)</A>.
Raw data from all studies have been reprocessed through a unified pipeline and jointly
variant-called to increase consistency across projects.
</P>

<H2>Display Conventions</H2>
<P>
In &quot;dense&quot; mode, a vertical line is drawn at the position of each variant.
In &quot;pack&quot; mode, ref and alt alleles are displayed to the left of a vertical
line with colored portions corresponding to allele counts.  Hovering the mouse pointer
over a variant pops up a display of alleles and counts.
</P>

<H2>Credits</H2>
<P>
Thanks to the
<A HREF="http://exac.broadinstitute.org/about" TARGET=_BLANK>Exome Aggregation Consortium</A>
for making these data available in advance of publication.  The data are released under
a <A HREF="http://www.wellcome.ac.uk/stellent/groups/corporatesite/@policy_communications/documents/web_document/wtd003207.pdf"
     TARGET=_BLANK>Fort Lauderdale Agreement</A>; please do not publish genome-wide
analyses of these data until after the ExAC flagship paper has been published (estimated 
to be in early 2015).
</P>

<H2>References</H2>

<p>
Exome Aggregation Consortium (ExAC), Cambridge, MA
(<A HREF="http://exac.broadinstitute.org" TARGET=_BLANK>http://exac.broadinstitute.org</A>),
release 0.2, downloaded Dec. 9, 2014.
</p>
