<H2>Description</H2>
<P>
The UCSC Genes track is a set of gene predictions based on data from
RefSeq, Genbank, CCDS, UniProt, Rfam, and the tRNA Genes track.  The
track includes both protein-coding genes and non-coding RNA genes.
Both types of genes can produce noncoding transcripts, but non-coding
RNA genes do not produce protein-coding transcripts. This is a
moderately conservative set of predictions.  Transcripts of
protein-coding genes require the support of one RefSeq RNA, or one
GenBank RNA sequence plus at least one additional line of evidence.
Transcripts of non-coding RNA genes require the support of one Rfam or
tRNA prediction.  Compared to RefSeq, this gene set has generally
about 10% more protein-coding genes, approximately four times as many
putative non-coding genes, and about twice as many splice
variants.</P>

<H2>Display Conventions and Configuration</H2>
<P>
This track in general follows the display conventions for
<A HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction
tracks</A>.  The exons for putative noncoding genes and untranslated regions 
are represented by relatively thin blocks, while those for coding open 
reading frames are thicker.  The following color key is used:
<UL>
<LI><b>Black</b> -- feature has a corresponding entry in the Protein Data Bank (PDB)
<LI><FONT color="#0C0C78"><b>Dark blue</b></FONT> -- transcript has been reviewed
 or validated by either the RefSeq, SwissProt or CCDS staff
<LI><FONT color="#5050A0"><b>Medium blue</b></FONT> -- other RefSeq transcripts
<LI><FONT color="#8282D2"><b>Light blue</b></FONT> -- non-RefSeq transcripts
</UL></P>
<P>
This track contains an optional codon coloring
feature that allows users to quickly validate and compare gene predictions.
To display codon colors, select the <em>genomic codons</em> option from the
<em>Color track by codons</em> pull-down menu. Click
<A HREF="../goldenPath/help/hgCodonColoring.html">here</A> for more
information about this feature. </P>

<H2>Methods</H2>
<P>
The UCSC Genes are built using a multi-step pipeline:  
<OL>
<LI>RefSeq and GenBank RNAs are aligned to the genome with BLAT, keeping 
only the best 
alignments for each RNA and discarding alignments of less than 98% identity.
<LI>Alignments are broken up at non-intronic gaps, with small isolated 
fragments thrown out.
<LI>Alignments are merged in from the hg19
<A HREF="http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=213859263&g=tRNAs">tRNA</A>
track
<LI>Alignments are also merged in for perfect alignments of sequences predicted
as human noncoding genes by 
<A HREF="http://rfam.sanger.ac.uk/" TARGET=_blank>Rfam</A>, 
in regions that are syntenic with the mm9 mouse 
genome.  Perfect alignments to regions that are not syntenic with mm9 are 
excluded because these regions are enriched for pseudogenes.
<LI>A splicing graph is created for each set of overlapping alignments. This 
graph 
has an edge for each exon or intron, and a vertex for each splice site, start, 
and end.
Each RNA that contributes to an edge is kept as evidence for that edge.
Gene models from the Consensus CDS project (CCDS) are also added to the graph.
<LI>A similar splicing graph is created in the mouse, based on mouse RNA and 
ESTs. If
the mouse graph has an edge that is orthologous to an edge in the human graph, 
that is added to the evidence for the human edge.
<LI>If an edge in the splicing graph is supported by two or more human ESTs, 
it is added as evidence for the edge.
<LI>If there is an Exoniphy prediction for an exon, that is added as evidence.
<LI>The graph is traversed to generate all unique transcripts. The traversal is 
guided by the initial RNAs to avoid a combinatorial explosion in alternative 
splicing. All refSeq transcripts are output. For other multi-exon transcripts
to be output, an edge supported by at least one additional line of evidence 
beyond the RNA is required.  Single-exon genes require either two RNAs or two 
additional lines of evidence beyond the single RNA. 
<LI>Protein predictions are generated. For non-RefSeq transcripts we use the 
txCdsPredict program to determine if the transcript is protein-coding and if so,
the locations of the start and stop codons. 
The program weighs as positive evidence the length of the protein, the 
presence of a Kozak consensus sequence at the start codon, and the length of 
the orthologous predicted protein in other species.
As negative evidence it considers nonsense-mediated decay and start codons in 
any frame upstream of the predicted start codon. For RefSeq transcripts 
the RefSeq protein prediction is used directly instead of this procedure.  For CCDS proteins 
the CCDS protein is used directly.
<LI>The corresponding UniProt protein is found, if any.
<LI>The transcript is assigned a permanent "uc" accession.  If the
transcript was not in the previous release of UCSC Genes, the
accession ends with the suffix ".1" indicating that this is the first
version of this transcript.  If the transcript was in the previous
release of UCSC Genes, the accession from the previous version is
re-used with the version number incremented.
</OL>

<H2>Related Data</H2>
<P>
The UCSC Genes transcripts are annotated in numerous tables, each of
which is also available as a downloadable file.  These include tables
that link UCSC Genes transcripts to external datasets (such
as <I>knownToLocusLink</I>, which maps UCSC Genes transcripts to Locus
Link identifiers), and tables that detail some property of UCSC Genes
transcript sequences (such as <I>knownToPfam</I>, which identifies any
Pfam domains found in the UCSC Genes protein-coding transcripts).  One
can see a full list of the associated tables in the Table Browser by
selecting UCSC Genes at the <B>track</B> menu; this list is then
available at the <B>table</B> menu.  Note that some of these tables
refer to UCSC Genes by its former name of <I>Known Genes</I>,
sometimes abbreviated as <I>known</I> or <I>kg</I>.  While the
complete set of annotation tables is too long to describe, some of the
more important tables are described below.
<UL>
<LI> <B>kgXref</B> identifies the RefSeq, SwissProt, Rfam, or tRNA
sequences (if any) on which each each transcript was based.
<LI> <B>knownGeneMrna</B> contains the mRNA sequence that represents
each UCSC Genes transcript.  If the transcript is based on a RefSeq
transcript, then this table contains the RefSeq transcript, including
any portions that do not align to the genome.
<LI> <B>knownGeneTxMrna</B> contains mRNA sequences for each UCSC Genes
transcripts.  In contrast to the sequencess in knownGeneMrna, these sequences
are derived from obtaining the sequences for each exon from the reference genome 
and concatenating these exonic sequences.
<LI> <B>knownGenePep</B> contains the protein sequences derived from
the knownGeneMrna transcript sequences.  Any protein-level annotations,
such as the contents of the knownToPfam table, are based on these sequences.
<LI> <B>knownGeneTxPep</B> contains the protein translation (if any) of
each mRNA sequence in knownGeneTxMrna.
</UL>
<P>

<H2>Credits</H2>
<P>
The UCSC Genes track was produced at UCSC using a computational pipeline
developed by Jim Kent, Chuck Sugnet, Melissa Cline and Mark Diekhans.  
It is based on data from NCBI
<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/" TARGET=_blank>RefSeq</A>,
<A HREF="http://www.uniprot.org/" TARGET=_blank>UniProt</A> 
(including TrEMBL and TrEMBL-NEW), 
<A HREF="http://www.ncbi.nlm.nih.gov/CCDS/" TARGET=_blank>CCDS</A>, and
<A HREF="http://www.ncbi.nlm.nih.gov/Genbank/index.html"
TARGET=_blank>GenBank</A> as well as data from 
<A HREF="http://rfam.sanger.ac.uk/">Rfam</A> and
<A HREF="http://lowelab.ucsc.edu/">the Todd Lowe lab</A>.
 Our thanks to the people running these databases
and to the scientists worldwide who have made contributions to them. </P>

<H2>Data Use Restrictions</H2>
<P>Copyright information from the
<A HREF="http://www.uniprot.org/help/license"
TARGET=_blank>UniProt</A> website:</P>
<P>
Copyright 2002-2009 UniProt Consortium.
We have chosen to apply the Creative Commons Attribution-NoDerivs 
<A HREF="http://creativecommons.org/licenses/by-nd/3.0/" 
TARGET=_blank>License</A> to all <A HREF="http://sciencecommons.org/"
TARGET=_blank>copyrightable parts</A> of our databases. This means that you are
free to copy, distribute, display and make commercial use of these databases,
provided you give us credit. However, if you intend to distribute a modified
version of one of our databases, you must ask us for permission first.
All databases and documents in the UniProt FTP directory may be copied and
redistributed freely, without advance permission, provided that this
copyright statement is reproduced with each copy.</P>
<P>

<H2>References</H2>
<P>
Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J,
Wheeler DL. 
<A HREF="http://nar.oupjournals.org/cgi/content/abstract/32/suppl_1/D23" 
TARGET=_blank>GenBank: update</A>. 
<em>Nucleic Acids Res.</em> 2004 Jan 1;32:D23-6.</P>
<P>
Chan PP, Lowe TM. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/gkn787" TARGET=_blank>GtRNAdb: A database 
of transfer RNA genes detected in genomic sequence.</A> 
<EM>Nucleic Acids Res</EM>. 2009 Jan;37:D93-7.
</P>
<P>
Gardner PP, Daub J, Tate J, Moore BL, Osuch IH, Griffiths-Jones S, Finn RD,
Nawrocki EP, Kolbe DL, Eddy SR, Bateman A.
<A HREF="http://nar.oxfordjournals.org/content/early/2010/11/08/nar.gkq1129">
Rfam: Wikipedia, clans and the "decimal" release</A>
<em>Nucleic Acids Res.</em>2011 Jan;39:D141-5.
</P>
<P>
Hsu F, Kent WJ, Clawson H, Kuhn RM, Diekhans M, Haussler D. 
<A HREF="http://bioinformatics.oxfordjournals.org/cgi/content/abstract/btl048?ijkey=qbTw0578H4cXuFw&keytype=ref" 
TARGET=_blank>The UCSC Known Genes</A>. 
<em>Bioinformatics</em>. 2006 May 1;22(9):1036-46.</P>
<P>
Kent WJ.
<A HREF="http://www.genome.org/cgi/content/abstract/12/4/656" 
TARGET=_blank>BLAT - the BLAST-like alignment tool</A>. 
<em>Genome Res.</em> 2002 Apr;12(4):656-64.</P>
<P>
Lowe TM, Eddy SR. 
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/9023104?dopt=Abstract" TARGET=_blank>tRNAscan-SE: A program for
improved detection of transfer RNA genes in genomic sequence.</A> 
<EM>Nucleic Acids Res</EM>. 1997 Mar 1;25(5):955-64.
</P> 
