<H2>Description</H2>
<P>
This track shows which genome regions are more or less accessible to next
generation sequencing methods that use short, paired end reads.  It summarizes whole
genome sequencing data from Phase 1 of the
<A HREF="http://1000genomes.org/"
TARGET=_BLANK>1000 Genomes Project</A>.  It shows two
levels of stringency:  &quot;pilot&quot; stringency regions (see below) cover 94% of non-N bases
in the genome and &quot;strict&quot; regions cover 72% of non-N bases.  Each site which meets
&quot;strict&quot; criteria also passes the &quot;pilot&quot; criteria.</P>
<P>
This track does <B><I>not</I></B> show a mask of regions in which variant calls can
or can not be made.
Some 1000 Genomes Phase 1 variant calls are in regions that do not meet the
&quot;strict&quot; criteria.
Phase 1 variant calls are filtered using the
<A HREF="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_variantrecalibration_VariantRecalibrator.html"
TARGET=_BLANK>Variant Quality Score Recalibrator (VQSR)</A>
method (implemented in the
<A HREF="http://www.broadinstitute.org/gatk/"
TARGET=_BLANK>Genome Analysis Toolkit (GATK)</A>)
without regard to the thresholds applied here.  VQSR assesses the evidence for
variation at sites where there is evidence, but says nothing about the remaining
sites.</P>
<P>
These regions will be useful for (a) comparing accessibility using current technologies
to accessibility in the 1000 Genomes Pilot Project, and (b) population genetic
analyses (such as estimates of mutation rate) that must focus on genomic regions
with very low false positive and false negative rates.
</P>

<H2>Methods</H2>
<P>
The total depth of mapped sequence reads, the average mapping quality score
and the fraction of reads with mapping quality zero (meaning that this read maps
equally well to more than one location in the genome) are tabulated from 1103
.bam files in the 1000 Genomes Phase 1 low coverage
<A HREF="ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/phase1/phase1.alignment.index"
TARGET=_BLANK>data release</A>.
This combines low coverage whole genome sequence information from 1094
individuals, giving a genome wide average total depth of coverage of 5132 reads.
Both &quot;pilot&quot; and &quot;strict&quot; tracks are .bed file conversions of the
&quot;pass&quot; regions from
<A HREF="ftp://ftp-trace.ncbi.nlm.nih.gov/1000genomes/ftp/phase1/analysis_results/supporting/accessible_genome_masks"
TARGET=_BLANK>.fasta mask files</A>.
See a
<A HREF="ftp://ftp-trace.ncbi.nlm.nih.gov/1000genomes/ftp/phase1/analysis_results/supporting/accessible_genome_masks/README.20120515_phase1_pop_gen_mask"
TARGET=_BLANK>README file</A> in that directory for details.
</P>
<P>
The &quot;pilot&quot; criteria require a depth of coverage between 2566 and 10264 inclusive
(between one half and twice the average depth) and that no more than 20% of
covering reads have mapping quality zero.  These are equivalent to the criteria
used for analyses in the 1000 Genomes Pilot paper (2010).  The &quot;strict&quot; criteria
require a depth of coverage between 2566 and 7698 inclusive, no more than 0.1%
of reads with mapping quality zero, and an average mapping quality of 56 or greater.
This definition is quite stringent and focuses on the most unique regions of the
genome.  In regions which pass the strict criteria, only ~2% of sites called in an
initial analysis are rejected as likely false positives by VQSR.  Since approximately
one half of 1000 Genomes Project individuals are males, the depth of coverage
is generally lower on the X chromosome.  Coverage thresholds on the X are
adjusted by a factor of 3/4 and on Y by a factor of 1/2.  The &quot;pilot&quot; criteria were
not evaluated for the Y chromosome.
</P>
<P>
1000 Genomes Phase 1 sequencing was done between 2008 and 2010 using Illumina
(86.4%), AB SOLiD (13%) and Roche LS 454 (0.6%) sequencing technologies.  45%
of the Illumina coverage is in approximately 100 bp paired end reads, 31.5% in 76 bp
reads, 15% in 51 bp reads and 8.5% in 36 bp reads.  All AB SOLiD data are 50 bp
mate paired reads.  Paired end sequence reads were mapped against the GRCh37 human
genome reference sequence using bwa version 0.5.5, bfast version 0.6.4e and ssaha
version 2.5 respectively.  Full details are in a
<A HREF="ftp://ftp-trace.ncbi.nlm.nih.gov/1000genomes/ftp/phase1/README.phase1_alignment_data"
TARGET=_BLANK>README file</A>
and in supplementary materials to the
Phase 1 paper (2012).</P>
<P>
The mapping target consists of the 22 autosomes plus X and
Y chromosomes (both pseudo-autosomal regions on the Y are masked by Ns), the
revised CRS mitochondrial sequence (NC_012920), and 59 unplaced contigs.  It does
<B><I>not</I></B> include the human herpesvirus 4 sequence (used for cell line transformation)
nor approximately 5 Mb of additional &quot;decoy&quot; sequence compiled from other human
entries in GenBank.  Those sequences were added to the mapping target in July 2011 and
will be included in the mapping during 1000 Genomes Phase 2.  Full details are in a
<A HREF="ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/technical/reference/phase2_reference_assembly_sequence/README_human_reference_20110707"
TARGET=_BLANK>README file</A>.
</P>

<H2>Credits</H2>
<P>
Mary Kate Trost and Gon&ccedil;alo Abecasis at the
<A HREF="http://csg.sph.umich.edu/"
TARGET=_BLANK>University of Michigan Center for Statistical Genetics</A>
generated the original masked fasta files.
Tom Blackwell at UM converted those to .bed format and edited the
description and methods.
Thanks to the
<A HREF="http://1000genomes.org/" TARGET=_BLANK>1000 Genomes Project</A>
for making Phase 1 data available in advance of publication.</P>

<H2>References</H2>
<P>
<em>Phase 1 of the 1000 Genomes Project:</em><BR/>
1000 Genomes Project Consortium.
An integrated map of genetic variation from 1,092 human genomes.
<em>Submitted for publication, 2012.</em>
</P>

<P>
<em>1000 Genomes Pilot Project:</em><BR/>
1000 Genomes Project Consortium.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/20981092"
TARGET=_BLANK>A map of human genome variation from population-scale sequencing</A>.
<A HREF="http://www.nature.com/nature/journal/v467/n7319/full/nature09534.html"
TARGET=_BLANK><em>Nature</em> 2010 Oct 28;467(7319):1061-73.</A></P>

<P>
<A HREF="http://www.1000genomes.org/faq"
TARGET=_BLANK>1000 Genomes Frequently Asked Questions (FAQ)</A></P>
