<h2>Description</h2>
<p>
This track shows regions
where transcription factors, proteins responsible for
modulating gene transcription, bind to DNA as assayed by ChIP-seq (chromatin 
immunoprecipitation with antibodies 
specific to the transcription factor followed by sequencing of the precipitated DNA).
The peaks of transcription factor binding underlying this track were computed using a uniform pipeline 
that uses the variation between the two replicates to guide peak thresholds. This track combines
data from 91 cell types and 189 transcription-factor targeting antibodies into a
summary display. 
Additional views of this data and additional documentation on the methods used
to generate the data underlying this track are available from the
<a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgTfbsUniform" 
target="_blank">ENCODE Uniform TFBS</a> track description.
</p>
<h2>Display Conventions</h2>
<p>
A gray box encompasses the peak clusters of transcription factor occupancy.  
The darkness of the box is proportional to the maximum signal strength observed in any cell line
contributing to the cluster.  The name to the left
of the box is the transcription factor. The letters to the right represent the cell lines where
a signal is detected.  The darkness of the letter is proportional to the signal strength in
the cell line. 
Abbreviations
starting with capital letters designate 
<a target="_blank" href="http://encodeproject.org/ENCODE/cellTypes.html">ENCODE cell types</a>
identified for intensive study - Tier 1 and Tier 2 - 
while those starting with lowercase letters designate Tier 3 cell lines.
Click on an item in the track to see the cell lines spelled out. 
</p>

<h2>Methods</h2>
<p>
<p>
Peaks of transcription factor occupancy from uniform processing of ENCODE ChIP-seq data
by the ENCODE Analysis Working Group were filtered to exclude datasets that did not pass the
integrated quality metric
(see &quot;Quality Control&quot; section of <a target="_blank" href=../cgi-bin/hgTrackUi?g=wgEncodeAwgTfbsUniform>Uniform TFBS</a>) 
and then were clustered using the UCSC hgBedsToBedExps tool.  
Scores were assigned to peaks by multiplying the input signal values by a normalization
factor calculated as the ratio of the maximum score value (1000) to the signal value at one
standard deviation from the mean, with values exceeding 1000 capped at 1000. This has the
effect of distributing scores up to mean plus one 1 standard deviation across the score range,
but assigning all above to the maximum score.
The cluster score is the highest score for any peak contributing to the cluster.  
<p>
For additional documentation on the methods used to generate the data underlying this track
please review the <a href="../cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgTfbsUniform"
target="_blank">ENCODE Uniform TFBS</a> track description.
</p>

<h2>Release Notes</h2>
<p>
Release 3 (August 2013) of this track includes more datasets (690 total,
vs. 486 in the previous release) representing all
ENCODE transcription factor ChIP-seq passing quality assessment through the ENCODE March 2012 data freeze.
The peaks used to generate these clusters were called with less stringent thresholds than 
used during the January 2011 uniform processing shown in Release 2 of this track.
The contributing datasets are displayed as individual
tracks in the ENCODE Uniform TFBS track, which is  available along with the primary data tracks
in the <a href="../cgi-bin/hgTrackUi?g=wgEncodeTfBindingSuper" 
target="_blank">ENC TF Binding Supertrack</a> page.
The clustering for V3 is based on the transcription factor target, and so differs from V2 where clustering was based on antibody.
</p>
<p>
For this release, a new track table format, 'factorSource' was used to represent the primary clusters
table and downloads file, <em>wgEncodeTfbsClusteredV3</em>.  This format consists of standard BED5
fields (see <a target="_blank" href="../FAQ/FAQformat.html#format1">File Formats</a>) followed by two
fields containing comma-separated lists.  The first list field contains numeric identifiers for
experiments, keyed to the <em>wgEncodeTfbsClusteredInputsV3</em> table, which includes such information
as the experiment's underlying Uniform TFBS table name, factor targeted, antibody used, cell type,
treatment (if any), and laboratory source.  The second list field contains the scores for the corresponding
experiments. For convenience, the 
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeRegTfbsClustered/">
file downloads directory</a>
for this track also contains a BED file,
<em>wgEncodeRegTfbsClusteredWithCellsV3</em>, that lists each cluster with the cluster score followed by a comma-separated list of cell types.
</p>

<h2>Credits</h2>
<p>
This track shows data from the <a href='http://myers.hudsonalpha.org/'
target="_blank">Myers Lab</a> at the <a href='http://www.hudsonalpha.org/'
target="_blank">HudsonAlpha Institute for Biotechnology</a> and by the labs of
<a href="http://snyderlab.stanford.edu/" target="_blank">Michael Snyder</a>,
<a href="http://bioinfo.mbb.yale.edu/" target="_blank">Mark Gerstein</a>,
<a href="http://bbs.yale.edu/people/sherman_weissman.profile" target="_blank">Sherman Weissman</a>
at Yale University,
<a href="http://www.farnhamlab.com/" target="_blank">Peggy Farnham</a> 
at the University of Southern California,
<a href="http://struhl.med.harvard.edu/" target="_blank">Kevin Struhl</a> at Harvard,
<a href="http://www.igsb.org/labs/kevin-white/" target="_blank">Kevin White</a> 
at the University of Chicago, and
<a href="http://microarray.icmb.utexas.edu/research.html" target="_blank">Vishy Iyer</a> 
at the University of Texas, Austin.
These data were processed into uniform peak calls by the ENCODE Analysis Working Group pipeline
developed by
<a href="https://sites.google.com/site/anshulkundaje" target="_blank">Anshul Kundaje</a>
The clustering of the uniform peaks was performed by UCSC.
</p>

<h2>Data Release Policy</h2>
<p>
While primary ENCODE data was subject to a restriction period as described in the 
<a href="../ENCODE/terms.html" target="_blank">
ENCODE data release policy</a>, this restriction does not apply to the integrative 
analysis results, and all primary data underlying this track have passed the restriction date. 
The data in this track are freely available.</p>
