<h2>Description</h2>
<p>
This set of data tracks represents the set of Open chromatin elements based on DNase-seq data from the Duke and UW production groups. The data tracks were generated by the ENCODE Analysis Working Group (AWG) based on a uniform processing pipeline. The datasets are based on the January 2011 internal data freeze. These datasets were used in all downstream analysis pipelines by members of the ENCODE Consortium and are one of the primary sources of data referenced in the ENCODE Integrative analyis paper (The ENCODE Project Consortium, 2012).
</p>

<h2>Methods</h2>

<p>
To reduce a source of variability between UW and Duke DNaseI processed datasets, the Stam Lab (UW) did the following with the Duke DNase-seq aligned reads, to more closely match UW processing:
<ul>
<li>
Combined all replicates for a given cell-type
</li>
<li>
Subsampled the result at a level of 30 million tags
</li>
<li>
Ran results through the Stam lab hotspot pipeline
</li>
</ul>
</p>

<p>
Results are here, alongside the UW replicate 1 calls, which are identical to what is available at <a href="http://genome.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeUwDnase" target="_blank">http://genome.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeUwDnase</a>. First, hotspots (i.e. broad, variable-sized regions of generalized chromatin accessibility) were identified using a relaxed threshold. FDR 1% peaks (narrowPeaks) were generated by first thresholding hotspots (using random simulation) at FDR 1%, and then (essentially) locating local maxima of the tag density (150 bp window, sliding every 20 bp) within the hotspots. FDR 1% peaks were set to a fixed width of 150 bp.
</p>

<p>
The combined calls are just a union of the calls for both groups, except for those 14 cell types where both groups have data. In the latter case, a collapsed set of FDR 1% peaks were generated by taking a non-overlapping selection of the calls from both centers and giving preference to the peak with the higher z-score when calls overlapped. A collapsed set of hotspots on these cell types was generated by simply merging the calls from both centers (taking the union interval of overlapping intervals).
</p>

<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="http://genome.ucsc.edu/ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

<P>There is no restriction on the use of these specific tracks.

<h2>References</h2>
<p>The ENCODE Project Consortium.
 <a href="http://dx.doi.org/10.1038/nature11247" title="http://dx.doi.org/10.1038/nature11247" rel="nofollow" TARGET = _BLANK>
An integrated encyclopedia of DNA elements in the human genome.</a>.
 <em>Nature</em> 2012.
 </p>
<p> Hoffman MM, Buske OJ, Wang J, Weng Z, Bilmes JA and Noble WS. Unsupervised pattern discovery in human chromatin structure through genomic segmentation.
 <em>In preparation</em>.
 </p>
<p> Hoffman MM, Ernst J, Wilder SP, Kundaje A, Harris RS, Libbrecht M, Bilmes JA, Giardine B, Birney E, Hardison RC, Dunham I, Kellis M and Noble WS. Integrative annotation of chromatin elements from ENCODE data. <em>Submitted</em>.




<h2>Contact</h2>
<p>
rthurman@u.washington.edu
</p>
