<h2>Description</h2>

<p>
This track represents a comprehensive set of human transcription factor binding sites based on ChIP-seq experiments generated by production groups in the ENCODE Consortium from the inception of the project in September 2007, through the March 2012 internal data freeze. The track represents peak calls (regions of enrichment) that were generated by the ENCODE Analysis Working Group (AWG) based on a uniform processing pipeline developed for the ENCODE Integrative Analysis effort and published in a set of coordinated papers in September 2012.  Peak calls from that effort, based on datasets from the January 2011 ENCODE data freeze) are available at the ENCODE Analysis Data Hub.  This track is an update that includes newer data, and slightly modified methods for the peak calling (more lenient enrichment threshold). 
<p>
This track contains 694 ChIP-seq datasets representing 177 unique regulatory factors (generic and sequence-specific factors). The datasets span 91 human cell types and some are in various treatment conditions. These datasets were generated by four main production groups : Broad, Stanford/Yale/UC-Davis/Harvard, HudsonAlpha, University of Texas(Austin) and Unversity of Washington.
</p>

<h2>Methods</h2>

<p>
All ChIP-seq experiments were performed at least in duplicate, and were scored against an appropriate control designated by the production groups (either input DNA or DNA obtained from a control immunoprecipitation). Submitted data was generally expected to meet an initial standard for inter-replicate consistency developed by the ENCODE Consortium to ensure an acceptable level of reproducibility; four fifths of the top 40% of the targets identified from one replicate (using an acceptable scoring method) should overlap the list of targets from the other replicate, or target lists scored using all available reads from each replicate should share more than 75% of targets in common. In addition, a number of quality metrics for individual replicates, including measures of library complexity and signal enrichment, were calculated, and these are available for review (Kundaje et al., 2012a, Kundaje et al., 2012b). As sequencing has become more economical, minimum standards for the number of reads required for submission of data have been established and upgraded over the course of the ENCODE project. Datasets used in the analyses presented here complied with the minimum depth requirements at the time of submission (ranging from 6M to 20M uniquely mapped reads).
</p>

<p>
A detailed description of the precise standards and considerations for evaluating the quality of ChIP-seq data and antibodies used for ChIP-seq is available (ENCODE Project Consortium, 2012).
</p>

<p>
The ENCODE Analysis Working Group built a scoring pipeline for the uniform processing of all TF ChIP-seq experiments generated by the ENCODE Consortium. This pipeline was implemented on the EBI cluster, but can readily be ported to other computers. Uniform signal was generated by processing the aligned reads using the align2rawsignal "Wiggler" software (see <a href="http://code.google.com/p/align2rawsignal" TARGET = _BLANK>http://code.google.com/p/align2rawsignal</a> for details and settings). The method accounts for the depth of sequencing, the mappability of the genome (based on read length and ambiguous bases) and different fragment length shifts for the different datasets being combined. It also differentiates between positions that showed zero signal simply because they are unmappable and positions that are mappable but have no reads.
</p>

<p>
Reads from all ENCODE ChIP-seq experiments and matching controls were mapped to a standardized version of the GRCh37 (hg19) reference human genome sequence with the following modifications:
<ul>
<li>Mitochondrial sequence was included.</li>
<li>Alternate sequences were excluded.</li>
<li>Random contigs were excluded.</li>
<li>The female version of the genome was represented by the autosomes and chrX, whereas the male genome was represented by the autosomes, chrX, and chrY with the PAR regions masked.</li> 
</ul>
Reads from experiments in cell lines labeled male or unknown were mapped to the above mentioned male genome, while experiments in cell lines labeled female were mapped to the female version of the genome.
</p>

<p>
All TF ChIP-seq datasets were processed using the standardized pipeline (Fig. S1b). Mapped reads in the form of BAM files were downloaded from the UCSC ENCODE portal (ENCODE Project Consortium, 2011). Multi-mapping reads were discarded. The SPP (Kharchenko et al., 2008) and the PeakSeq (Rozowsky et al., 2009) peak calling methods were used to identify peaks (regions of enrichment) by comparing each ChIP-seq experiment to a corresponding input DNA control experiment. For the SPP peak caller, peaks were ranked using the peak signal score which is a function of ChIP signal enrichment in each peak over background signal from a corresponding input DNA experiment, corrected for mirror correlation of positive- and negative-strand tag densities within the peak region. PeakSeq peaks were ranked using the estimated false discovery rate (q-value) for each peak, which is computed from the enrichment of ChIP-seq reads in a peak, relative the the normalized counts of matching control reads using a Binomial test.
</p>

<p>
Since every ENCODE dataset is represented by at least two biological replicate experiments, a measure of consistency of peak calling results between replicates, known as the irreproducible discovery rate (IDR), was used to determine an optimal number of reproducible peaks (Li et al., 2011, Kundaje et al., 2012b). Peak calling was performed independently on each replicate of a ChIP-seq dataset. Relaxed peak calling thresholds (FDR = 0.9 for SPP and FDR=0.05 for PeakSeq) were used to obtain a large number of peaks that span true signal as well as noise (false identifications). The IDR method analyzes a pair of replicates, and considers peaks that are present in both replicates to belong to one of two populations : a reproducible signal group or an irreproducible noise group. Peaks from the reproducible group are expected to show relatively higher ranks and stronger rank-consistency across the replicates, relative to peaks in the irreproducible groups. Based on these assumptions, a two-component probabilistic copula-mixture model is used to fit the bivariate peak rank distributions from the pairs of replicates. The method adaptively learns the degree of peak-rank consistency in the signal component and the proportion of peaks belonging to each component. The model can then be used to infer an IDR score for every peak that is found in both replicates. The IDR score of a peak represents the expected probability that the peak belongs to the noise component, and is based on its ranks in the two replicates. Hence, low IDR scores represent high-confidence peaks. An IDR score threshold was used to obtain an optimal peak rank threshold on the replicate peak sets (cross-replicate threshold). For SPP-based peak calls, the IDR threshold used was 1%, and for PeakSeq, 5%. If a dataset had more than two replicates, all pairs of replicates were analyzed using the IDR method. The maximum peak rank threshold across all pairwise analyses was used as the final cross-replicate peak rank threshold. Reads from replicate datasets were then pooled and SPP/PeakSeq was performed to call peaks on the pooled data with a relaxed FDR of 0.9/0.05. Pooled-data peaks were once again ranked by signal-score (for the SPP peak caller) or q-value (for the PeakSeq peak caller). The cross-replicate rank threshold learned from the replicates was used to threshold the ranked set of pooled-data peaks. 
</p>

<p>
Any thresholds based on reproducibility of peak calling between biological replicates are bounded by the quality and enrichment of the worst replicate. Valuable signal is lost in cases for which a dataset has one replicate that is significantly worse in data quality than another replicate. A rescue pipeline was used for such cases. In order to balance data quality between a set of replicates, mapped reads were pooled across all replicates of a dataset, and then randomly sampled (without replacement) two pseudo-replicates with equal numbers of reads. This sampling strategy tends to transfer signal from stronger replicates to the weaker replicates, thereby balancing cross-replicate data quality and sequencing depth. These pseudo-replicates were then processed using the IDR method in order to learn a rescue threshold. For datasets with comparable replicates (based on independent measures of data quality), the rescue threshold and cross-replicate thresholds were found to be very similar. However, for datasets with replicates of differing data quality, the rescue thresholds were often higher than the cross-replicate thresholds, and were able to capture true peaks that showed statistically significant and visually compelling ChIP-seq signal in one replicate but not in the other. Ultimately, for each dataset, the best of the cross-replicate and rescue thresholds were used to obtain a final consolidated optimal set of peaks. 
</p>

<p>
All peak sets were then screened against a specially curated empirical blacklist (<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeDacMapabilityConsensusExcludable.bed.gz" target="_blank">http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeDacMapabilityConsensusExcludable.bed.gz</a>) of regions in the human genome and peaks overlapping the blacklisted regions were discarded (Kundaje et al.,2012b). Briefly, these artifact regions typically show the following characteristics:
<ul>
<li>Unstructured and extreme artifactual high signal in sequenced input-DNA and control datasets as well as open chromatin datasets irrespective of cell type identity.</li>
<li>An extreme ratio of multi-mapping to unique mapping reads from sequencing experiments.</li>
<li>Overlap with pathological repeat regions such as centromeric, telomeric and satellite repeats that often have few unique mappable locations interspersed in repeats.</li>
</ul>
</p>



<h2>References</h2>

<p>
<p>ENCODE Project Consortium.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21526222" title="http://www.ncbi.nlm.nih.gov/pubmed/21526222"  rel="nofollow" TARGET = _BLANK>
A user's guide to the encyclopedia of DNA elements (ENCODE)</a>.
<em>PLoS Biol.</em> 2011 Apr;9(4):e1001046.
</p>

<p>ENCODE Project Consortium.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22955616" title="http://www.ncbi.nlm.nih.gov/pubmed/22955616"  rel="nofollow" TARGET = _BLANK>
An integrated encyclopedia of DNA elements in the human genome</a>.
<em>Nature</em> 2012 Sep 6;489(7414):57-74.</p>

<p>Kharchenko PV, Tolstorukov MY, Park PJ.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19029915" title="http://www.ncbi.nlm.nih.gov/pubmed/19029915"  rel="nofollow" TARGET = _BLANK>
Design and analysis of ChIP-seq experiments for DNA-binding proteins</a>.
<em>Nat Biotechnol.</em> 2008 Dec;26(12):1351-9.</p>

<p>Kundaje A, Jung L, Kharchenko PV, Sidow A, Batzoglou S, Park PJ.
Assessment of ChIP-seq data quality using strand cross-correlation analysis (submitted), 2012a.</p>

<p>Kundaje A, Li Q, Brown JB, Rozowsky J, Harmanci A, Wilder SP, Batzoglou S, Dunham I, Gerstein M, Birney E, <em>et al</em>.
Reproducibility measures for automatic threshold selection and quality control in ChIP-seq datasets (submitted), 2012b.</p>

<p>Li QH, Brown JB, Huang HY, Bickel PJ.
<a href="http://projecteuclid.org/euclid.aoas/1318514284" title="http://projecteuclid.org/euclid.aoas/1318514284"  rel="nofollow" TARGET = _BLANK>
Measuring reproducibility of high-throughput experiments</a>.
<em>Ann. Appl. Stat.</em> 2011; 5(3):1752-1779.</p>
</p>

<p>Rozowsky J, Euskirchen G, Auerbach RK, Zhang ZD, Gibson T, Bjornson R, Carriero N, Snyder M, Gerstein MB.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19122651" title="http://www.ncbi.nlm.nih.gov/pubmed/19122651"  rel="nofollow" TARGET = _BLANK>
PeakSeq enables systematic scoring of ChIP-seq experiments relative to controls</a>.
<em>Nat Biotechnol.</em> 2009 Jan;27(1):66-75.</p>


<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="http://genome.ucsc.edu/ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>

<P>There is no restriction on the use of these specific tracks.

<h2>Contact</h2>
<p>
<a HREF="mailto:rmyers@hudsonalpha.org"TARGET=_BLANK>Richard Myers</a>
</p>
