<HR ALIGN='bottom' style='position:relative; top:1em;'><table class='windowSize'><tr valign='top'><td rowspan=2><A NAME='TRACK_HTML'></A>
<h2>Description</h2>
<p>
Release 2 of the CSHL small RNA track depicts <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?term=%22shortTotal%22">short total RNA</a> sequencing data from <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=cellType">ENCODE tissues</a> or <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=localization">sub-cellular compartments</a> of <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=cellType">ENCODE cell lines</a>. The protocol used to generate this data produces directional reads from the 5' end of short RNAs (< 200 nt). Libraries were sequenced using an Illumina GAIIx<!-- or HiSeq-->. These data were generated by Cold Spring Harbor Laboratories as a part of the ENCODE Consortia. The ENCODE project seeks to identify and characterize all functional elements in the human genome.  In many cases, there are Cap Analysis of Gene Expression (CAGE, see the <a target="_BLANK" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRikenCage">RIKEN CAGE Loc</a> track), Long RNA-seq (>200 nucleotides, see the  <a target="_BLANK" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq">CSHL Long RNA-seq</a> track) and Pair-End di-TAG-RNA (PET-RNA, see the <a target="_BLANK" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGisRnaPet">GIS RNA PET</a> track) datasets available from the same biological replicates.</p>

<p>
Many of the datasets produced by the Hannon lab (Generation 0 datasets) in Release 1 have been replaced by newly generated data from the Gingeras lab in Release 2. Of all Generation 0 datasets, only data from K562 and Prostate tissue are still displayed. All Generation 0 datasets are still available for <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlShortRnaSeq">download</a>.</p>

<!-- OLD DESCRIPTION
<p>This track depicts RNA-Seq data from human 
<a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=rnaExtract">
RNA</a> between the sizes of 20-200 nt. 
As part of the ENCODE Consortia, Cold Spring Harbor Laboratories isolated 
<a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=localization">
tissues or sub cellular compartments</a> from
<a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgEncodeVocab?type=cellType">
ENCODE cell lines</a>.
The overall goal of the ENCODE project is to identify and characterize all functional
elements in the sequence of the human genome.
</p>

<p>This cloning protocol generates directional libraries that largely correspond to the
5&#8242; ends of mature RNAs.  The libraries were sequenced on a Solexa platform for
a total of 36, 50 or 76 cycles producing variable length reads.  Furthermore, the reads undergo post-processing
resulting in trimming of their 3&#8242; ends (Mortazavi et al., 2008).  
</p>
-->


<h2>Display Conventions and Configuration</h2>

<p>
This track is a multi-view composite track that contains multiple data types
(<em>views</em>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <a href="../../goldenPath/help/multiView.html" TARGET=_BLANK>here</a>.
To show only selected subtracks, uncheck the boxes next to the tracks that
you wish to hide.  Color differences among the views provide a visual cue for
distinguishing between the different cell types and compartments.</p>

<h4>Viewable Tracks</h4>

<dl>
<dt><i>Alignments</i></dt>
<dd>The Alignments view shows individual reads mapped to the genome and indicates where bases may mismatch. Every mapped read is displayed, i.e. uncollapsed. The alignments file follows the standard SAM format. See the <a target="_BLANK" href="http://samtools.sourceforge.net/SAM1.pdf#page=4">SAM Format Specification</a> for more information on the SAM/BAM file format.
</dd>
<dt><i>Contigs</i></dt>
<dd>The Contigs are in .BED format files representing blocks of overlapping mapped reads from pooled biological replicates. Corresponding number of mapped reads, RPKM values and non-parametric IDR (npIDR) are reported for each contig.<!--Need additional information on what parameters were used to generate contigs--></dd>
<dt><i>Raw Signal</i></dt>
<dd>The Raw Signal view shows the density of mapped reads on the plus and minus strands (wiggle format).</dd>
</dl>

<h4>Downloadable Files</h4>
The following files can be found on the <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlShortRnaSeq">Downloads Page</a>.

<dl>
<dt><i>Fastq</i></dt>
<dd>Raw sequence information in fastq format.</dd>
<dt><i>Gencode V7</i></dt>
<dd>For each Gencode v7 exon, the raw read count on the plus or minus strand, reads per million mapped reads (RPM) and the non-parametric IDR (npIDR) are reported in GFF format.</dd>
<dt><i>Transfrags</i></dt>
<dd>These data are now available for download only and have been replaced by Contigs. Small RNA reads were assembled into transcribed fragments, "Transfrags" by merging reads with one or more overlapping nucleotides. Only uniquely mapping reads were used to generate Transfrags. The BED6+ format of the transfrag files are created from "intervals-to-contigs" Galaxy tool written by Assaf Gordon in the Hannon lab at CSHL. The transfrags view is not filtered.</dd>
</dl>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>

<!-- OLD DISPLAY CONVENTIONS
<p>
<dl>
For each cell type, this track contains the following views:
<dl>
<dt><i>Contigs</i>
</dt><dd> The Contigs are in .BED format files representing blocks of overlapping mapped reads from pooled biological replicates.
</dd>
<dt><i>Transfrags</i>
</dt><dd>These data are now available for download only and have been replaced by Contigs. Small RNA reads were assembled into transcribed fragments, "Transfrags" by merging reads with one or more overlapping nucleotides.  Only uniquely mapping reads were used to generate Transfrags. The <a href="../ENCODE/transfragBed6CshlDescription.html">BED6+ format</a> of the transfrag files are created from "intervals-to-contigs" Galaxy tool written by <a href="mailto:&#103;&#111;r&#100;&#111;n&#64;&#99;&#115;h&#108;.e&#100;&#117;">Assaf Gordon</a> in the Hannon lab at CSHL.
The transfrags view is not filtered.
</dd><dt><i>Raw Signals (Plus and Minus)</i>
</dt><dd>The Raw Signal view shows the density of mapped reads on the plus and minus strands (wiggle format).
</dd>
<dt><i>Alignments</i>
</dt><dd>The Alignments view shows individual reads mapped from pooled biological replicates to the genome and indicates where
bases may mismatch. Every mapped read is displayed, i.e. uncollapsed.  

-->


<!--The track may be set so that the color of each alignment denotes the mapping frequency of the corresponding read:</br>
<font color="#FF00FF">pink</font> for reads mapping to a single genomic location, </br>
<font color="#CC00FF"> pinkish-purple</font> for reads mapping to exactly two genomic locations and </br>
<font color="#800080">purple</font> for reads mapping to more than two possible locations.</br>
-->

<!--
The alignment file follows the standard SAM format of Bowtie output with the following additions: the mapping frequency for each read is indicated in the BAM file using the tag 'YG' and the alignment color is specified in RGB space using the tag 'YC'.  The 'XA' tag is Bowtie tag that means the aligned read belongs to <a href=http://bowtie-bio.sourceforge.net/manual.shtml#strata>strata</a>. See the <a href="http://bowtie-bio.sourceforge.net/manual.shtml#sam-bowtie-output" target=_blank>Bowtie Manual</a> for more information about the SAM Bowtie output (including other tags) and the <a href="http://samtools.sourceforge.net/SAM1.pdf#page=4" target=_blank>SAM Format Specification</a> for more information on the SAM/BAM file format. Mapping quality is not available for this track and so, in accordance with the SAM Format Specification, a score of 255 is used. Also, as described in the Methods section below, the sequence quality scores are not available and so are all displayed as 40.
</dd>
</dl>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>

-->


<h2>Methods</h2>
<h3>Experimental Procedures</h3>
<p>Cells were grown according to the approved ENCODE <A href="http://genome.ucsc.edu/ENCODE/protocols/cell" target="_BLANK">cell culture protocols</a>. Short RNAs between 20-200 nt were isolated from total RNA using a Qiagen RNeasy kit (Qiagen #74204) according to manufacturer's protocol. Purified small RNAs were treated with Ribominus (Invitrogen # A10837-08) according to manufacturer protocols to remove ribosomal RNA. 5' structures were removed by treatment with Tobacco Alkaline Pyrophosphatase to allow for ligation of a 5' linker. RNA fragments were poly-adenylated using poly-A polymerase (or poly-cytidylated, in the case of Generation 0 data) and a 5' linker was ligated using T4 RNA ligase. An anchored oligo-dT was used to prime the reverse transcriptase reaction. cDNA was amplified using universal PCR primers introduced during the reverse transcriptase reaction. Resulting libraries were gel purified and used in cluster generation on an Illumina GAIIx<!-- or HiSeq-->. The libraries were sequenced as a single read from the 5' end of the inserts for a total of 36 <!--or 101--> cycles.
</p>
<p>
Complete protocols are available in the <a target="_BLANK" href="http://hgwdev.cse.ucsc.edu/cgi-bin/hgFileUi?db=hg19&g=wgEncodeCshlShortRnaSeq">Downloads Page</a>.
</p>



<h3>Data Processing and Analysis</h3>
<p>Data from the Gingeras and Guigo labs were preprocessed to remove experimentally derived poly-A tail and Illumina 3' linker from raw reads. The best alignment to the Illumina 3' linker for each read was determined. If the number of mismatches in the alignment was less than 20% of the aligned length, the read was clipped from the first aligned base. Pre-processeds reads were mapped using the STAR algorithm. For a description of STAR, the source code and mapping parameters used see <a target="_BLANK" href="http://gingeraslab.cshl.edu/STAR/">the STAR project website</a>. Reads mapping 10 times or less  are reported in the Raw Signal and Alignment files.</p>

<p>Mapped reads were discarded if they fell into one of the following categories: 1, it contained five or more consecutive A's; 2, it was less than 16 nt in length; 3, it mapped to more than one genomic position (multiply-mapped reads); 4, it mapped upstream of genomically encoded poly-A sequences. The remaining reads were used to produce expression values over Gencode v7 exons and to call contigs. Contigs were generated from overlapping reads in pooled biological replicates.</p>


<h4>Generation 0 data</h4>
(See Release Notes for more information on Generation 0 datasets). Reads were trimmed to discard any bases following a quality score less than or equal to 20 and converted into FASTA format, thereby discarding quality information for the rest of the pipeline. As a result, the sequence quality scores in the BAM output are all displayed as "40" to indicate no quality information. The read lengths may exceed the insert sizes and consequently introduce 3' adapter sequence into the 3' end of the reads. The 3' sequencing adapter was removed from the reads using a custom clipper program (available at <a href="http://hannonlab.cshl.edu/fastx_toolkit">http://hannonlab.cshl.edu/fastx_toolkit/</a>), which aligned the adapter sequence to the short-reads using up to 2 mismatches and no indels. Regions that aligned were "clipped" off from the read. Terminal C nucleotides introduced at the 3' end of the RNA via the cloning procedure are also trimmed. Reads were aligned to the human genome (version hg19, using the gender build appropriate to the sample in question -  <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/referenceSequences/female.hg19.fa.gz" target="_BLANK">female</a>/<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/referenceSequences/male.hg19.fa.gz" target="_BLANK">male</a>) using Bowtie (Langmead B. et al). Reads that mapped 20 or fewer times with 2 or less mismatches were reported.


<!-- Old Methods
<p>Small RNA between 20-200 nt was Ribominus-ed<sup>TM</sup>;
 treated according to the
manufacturer's protocol (Invitrogen).
The RNA was treated with Tobacco Alkaline Pyrophosphatase to eliminate any
5&#8242; structures that would inhibit cloning.  
The RNA is then "A-tailed" using Poly-A Polymerase.  A 5&#8242; linker is ligated using T4 RNA ligase and anchored oligo-dT is used to prime a reverse transcriptase reaction.  The libraries are gel purified and clustered for an Illumina GAIIx run. Sequence is obtained from the 5&#8242; end of the inserts for a total of 36 cycles. For a detailed protocol see: </br>###[Small RNA Cloning Protocol.pdf].
</p>

<p>The Illumina reads were initially trimmed to discard any bases following a quality score less than or equal to 20 and converted into FASTA format, thereby discarding quality information for the rest of the pipeline. As a result, the sequence quality scores in the BAM output are all displayed as "40" to indicate no quality information. The read lengths may exceed the insert sizes and consequently introduce
3&#8242; adapter sequence into the 3&#8242; end of the reads.  The 3&#8242;
sequencing adapter was removed from the reads using a custom clipper program (available at <a href="http://hannonlab.cshl.edu/fastx_toolkit">http://hannonlab.cshl.edu/fastx_toolkit/</a>), which aligned the adapter sequence to the short-reads using up to 2 mismatches and no indels.  Regions that aligned were "clipped" off from the read. Terminal C nucleotides introduced at the 3' end of the RNA via the cloning procedure are also trimmed.  The trimmed portions were collapsed into identical reads, their count noted and aligned to the human genome (version hg19, using the gender build appropriate to the sample in question - <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/referenceSequences/female.hg19.fa.gz">female</a>/<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/referenceSequences/male.hg19.fa.gz">male</a>) using Bowtie  (Langmead B. et al). The alignment parameter allowed 0, 1, or 2 mismatches iteratively. We report reads that mapped 20 or fewer times.</p>

<p>Data from the Gingeras and Guigo labs were mapped with STAR. For a description of STAR, the source code and mapping parameters used see: <a href="http://gingeraslab.cshl.edu/STAR/">http://gingeraslab.cshl.edu/STAR/<a>. Reads mapping 10 times or less are reported in the Raw Signal and Alignment files.
</p>

-->

<h2>Verification</h2>

<p> The mapped data were visually inspected to verify the majority of the reads were mapping the 5' ends of annotated small RNA classes.
</p>

<h2>Release Notes </h2>
<p>
This is release 2 (Summer 2011). This release contains both data remapped from hg18 that was produced by the Hannon lab (Generation 0) and new experiments produced by the Gingeras lab.  When there is both Generation 0 and new data available, only the new data is displayed, the older data is available for downloads only. From the original eleven experiments displayed with release 1, only two (prostate and K562 polysome) are still displayed.  In addition to improved data, there are several additional cell lines. The Contigs view is available for display and the Gencode Predicted Exons files are available for downloads.  
</p>
<p><em><strong>Discrepancies between hg18 and hg19 versions of Generation 0 CSHL small RNA data:</em></strong> The alignment pipeline for the CSHL small RNA data was updated upon the release of the human genome version hg19, resulting in a few noteworthy discrepancies with the hg18 dataset.  First, mapping was conducted with the open-source Bowtie algorithm (<a href="http://bowtie-bio.sourceforge.net/index.shtml">http://bowtie-bio.sourceforge.net/index.shtml</a>) rather than the custom NexAlign software.  As each algorithm uses different strategies to perform alignments, the mapping results may vary even in genomic regions that do not differ between builds.  The read processing pipeline also varies slightly, in that we no longer retain information regarding whether a read was 'clipped' off adapter sequence. 
</p>



<h2>Credits</h2>
Hannon lab members: Katalin Fejes-Toth, Vihra Sotirova, Gordon  Assaf, Jon Preall</p>
<p>Gingeras and Guigo laboratories: Carrie A. Davis, Lei-Hoon See, Wei Lin </p>

<h2>Contacts:</h2> 
<dl>
<dd><a href="mailto:&#106;&#112;r&#101;&#97;&#108;&#108;&#64;&#99;&#115;&#104;l.
&#101;&#100;&#117;">Jonathan Preall</a> (Generation 0 Data from Hannon Lab) 
<!-- above address is jpreall at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Carrie Davis</a> (experimental)
<!-- above address is davisc at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#100;&#111;&#98;i&#110;&#64;&#99;&#115;&#104;l.&#101;d&#117;">Alex Dobin</a> (computational)
<!-- above address is dobin at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#119;&#108;&#105;&#110;&#64;&#99;s&#104;&#108;.&#101;&#100;&#117;">Wei Lin</a> (computational)
<!--above address is wlin at cshl.edu -->
</dd>
<dd>
<a href="mailto:&#103;&#105;&#110;&#103;e&#114;a&#115;&#64;&#99;&#115;&#104;&#108;.&#101;d&#117;">Tom Gingeras</a> (primary investigator)
<!--above address is gingeras at cshl.edu -->
</p>
</dd>
</dl>

<h2>References</h2>

<p>
Fejes Toth K. <a target="_BLANK" href=""><u>Post-transcriptional processing generates a diversity of 5′-modified long and short RNAs</u></a>. <em>Nature</em>. 2009;457(7232):1028-.
</p>

<p>Langmead B, Trapnell C, Pop M, Salzberg SL.
<a target="_BLANK" href="http://genomebiology.com/2009/10/3/R25"><U>Ultrafast and memory-efficient alignment of short DNA sequences to the human genome</U></a>.
<em>Genome Biology</em>. 2009 Mar; 10:R25.</p>



<!--<p>Mortazavi A, Williams BA, McCue K, Schaeffer L, and Wold BJ. 
<a target="_BLANK" href="http://www.ncbi.nlm.nih.gov/pubmed/18516045">Mapping and quantifying mammalian transcriptomes by RNA-seq</a>.
<em>Nature Methods</em>. 2008 Jul; 5(7):621-628.</p> 
-->

<h2> Data Release Policy </h2>

<p>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <em>Restricted Until</em> column, above.  The full data release policy 
for ENCODE is available 
<a href="../ENCODE/terms.html" TARGET=_BLANK>here</a>.</p> 

