<H2>Description</H2>

<P>These tracks were generate by the 
<A TARGET=_BLANK HREF="http://www.genome.gov/10005107">ENCODE
Consortium</A>.  They contain information about human RNAs &gt; 200
nucleotides in length obtained as short reads off the Illumina GAIIx
platform.  Data is available from biological replicates of several
cell lines.  In addition to profiling Poly-A+ and Poly-A- RNA from
whole cells, we have also gather data from various subcellular
compartments.  In many cases, there are Cap Analysis of Gene
Expression (CAGE, 
<A TARGET=_BLANK HREF="http://www.riken.go.jp/engn/">RIKEN Institute</A>) and
Small RNA-Seq (&lt;200 nucleotides, 
<A TARGET=_BLANK HREF="http://www.cshl.edu/">CSHL</A>) and Pair-End
di-TAG-RNA (PET-RNA,
<A TARGET=_BLANK
HREF="http://www.gis.a-star.edu.sg/internet/site/">Genome Institute of
Singapore</A>) datasets available from the same biological replicates.   

<H2>Methods</H2>

<H3>Cell Culture</H3>
<P>Cells were grown according to the approved 
<A TARGET=_BLANK
HREF="http://genome-test.cse.ucsc.edu/ENCODE/protocols/cell/"> ENCODE
cell culture protocols</A>.  
</P>

<H3>Library Preparation</H3>
<P>We are using the published protocol
<A TARGET=_BLANK HREF="http://www.ncbi.nlm.nih.gov/pubmed/19620212">
[PMID: 19620212]</A>.  This protocol generates directional libraries
and reports the transcripts strand of origin.  Exogenous RNA spike-ins
(Round 5, pool 14), in development at National Institutes Standards
Technology were added to each endogenous RNA isolate and carried
through library construction and sequencing.  The Illumina PhiX
control library was also spiked-in at 1% to each completed human
library just prior to cluster formation. Libraries from biological
replicates are always constructed in parallel to minimize technical
variation.  The libraries were sequenced in pair-end fashion yielding
76 bases of sequence for each end. Library insert sizes are centered
on average at 200 bases.  Accompanying each RNA-Seq dataset is a
&ldquo;Production Document&rdquo;.  This document contains details
about the RNA isolations and treatments, library construction,
spike-ins as well as quality control figures for individual libraries.   
</P>

<H3>Sequencing and Mapping</H3>
<P>The libraries are sequenced on the Illumina GAIIx platform to an
average depth of &#126;200 million reads (100 million mate-pairs).
The data are mapped against hg19 using Spliced Transcript Alignment
and Reconstruction (STAR) written by Alex Dobin (CSHL).  More
information, about STAR including the parameters used for these data
can be found at: 
<A TARGET=_BLANK
HREF="http://gingeraslab.cshl.edu/STAR/">http://gingeraslab.cshl.edu/STAR/</A>. 
</P>

<H2>Verification</H2>
<P>We take several steps to ensure the primary data is high quality,
reproducible and technical variation kept to a minimum.   Further
Explanation Needed (qualitative and quantitative metrics)&hellip; 
</P>

<H3>RNA Element Files</H3>
<P>The mapped data are further mined to generate secondary files
representing transcribed &ldquo;Elements&rdquo;.  These include;
expressed contigs, splice junctions, de novo genes and transcripts,
and expression values for annotated Gencode genes, transcript and
exons.   
</P>

<H3>Experimental Validation</H3>
<P>The goals of the experimental validation are to gather Roche 454 FLX
data for transcribed elements mapping to regions lacking gene
annotations in order to:  
<OL>
  <LI>provide independent biochemical validation for the candidate
  loci identified in the primary RNA-Seq data, and
  <LI>gather longer sequence information to aid downstream annotation
  and curation.
</OL>
We selected loci that encode canonical &ldquo;GU-AG&rdquo; introns and
which appear as spliced products in the primary RNA-Seq data.  PCR
primers are designed spanning these junctions and used to amplify
oligo-dT primed cDNA.  The PCR products are pooled and sequenced on
Roche 454 FLX to check the products identity.  The sequence data is
mapped against hg19 using BLAT.   
</P>

<H3>Raw Data Files Available for Download</H3>
<P>FASTQ&#146;s and the Production Document (.pdf) explaining the
protocol used to gather the sequence data. 
</P>

<H3>Mapped Data Files Available for Download</H3>
<P>.BAM, .Wig
</P>

<H3>Element Files Available for Download</H3>
<P>Further Explanation Needed&hellip;
</P>

<UL>
  <LI>Splice Junctions
  <LI>Contigs
  <LI>De novo Genes and Transcripts
  <LI>Gencode annotation-based expression values
</UL>

<H3>Validation Files Available for Download</H3>
<P>Further Explanation Needed&hellip; (you do not have these yet
Cricket.  This will be 454 data (FASTQ and .psl files mapped with
BLAT). 
</P>

<H2>Release Notes</H2>

<P> This is Release 1 (Jan 2011) of this track for hg19. It includes data from the
hg18 releases that has been remapped to hg19 (GM12878 and K562 Total RNA whole 
cell and K562 PolyA+ RNA cytosol) as well as novel data from 13 additional cell
lines and 4 additional cell localizations, as well as some PolyA- samples.  The
singnals are now in bigWig format.  The alignments are in BAM format.  The newer
data all have Bio Replicate numbers and Protocol ID numbers that can be used to
link the samples with other experiments.
</P>

<H2>Credits</H2>
<P>These data were generated and analyzed by the transcriptome group at
Cold Spring Harbor Laboratories, and the Center for Genomic
Regulation (CRG in Barcelona), who are participants in the ENCODE Transcriptome Group.
<br/>
Carrie A. Davis, Alex Dobin, Felix Schlesinger, Wei Lin, Chris Zaleski, Sonali Jha, Jorg
Drenkow, Huaien Wang, Kim Bell, Lei-Hoon See, Megan Fastuca and Tom Gingeras
</P>

<H2>Contacts</H2>
<A HREF="mailto:&#100;a&#118;&#105;s&#99;&#64;&#99;sh&#108;.&#101;&#100;&#117;">Carrie Davis</A> (experimental)
<!-- above address is davisc at cshl.edu -->
<A HREF="mailto:&#100;&#111;&#98;i&#110;&#64;&#99;&#115;&#104;l.&#101;d&#117;">Alex Dobin</A> (computational)
<!-- above address is dobin at cshl.edu -->
<A HREF="mailto:&#115;&#99;&#104;&#108;&#101;si&#110;&#64;&#99;&#115;&#104;l.e&#100;&#117;">Felix
Schlesinger</A> (computational)
<!--above address is schlesin at cshl.edu -->
<A HREF="mailto:&#103;&#105;&#110;&#103;e&#114;a&#115;&#64;&#99;&#115;&#104;&#108;.&#101;d&#117;">Tom Gingeras</A> (primary investigator)
<!--above address is gingeras at cshl.edu -->
</P>

<P>
<A HREF="mailto:&#114;&#103;&#117;i&#103;o&#64;&#105;&#109;&#105;m.e&#115;">Roderic Guigo</A>&#146;s group at the
<A TARGET=_BLANK
HREF="http://pasteur.crg.es/portal/page/portal/Internet/">CRG</A>.
<!-- above address is rguigo at imim.es -->

</P>

<!--<H3>References</H3>
-->
