<H2>Description</H2>

<P>This track is produced as part of the ENCODE Transcriptome Project. 
It shows the starts and ends of full length mRNA transcripts determined by Gene Identification Signature (GIS) 
paired-end ditag (PET) sequencing using 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=rnaExtract">RNA extracts</A> 
from different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=localization">sub-cellular
localizations</A> 
in different 
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType">cell lines</A>.
The RNA-PET information provided in this track is composed of two different PET length versions based on how the PETs were extracted.  The cloning-based PET (18 bp and 16 bp) is an earlier version and detailed information can be found from reference (Ng et al. 2006).  The cloning-free PET (25 bp and 25 bp) is a recently modified version which uses Type II enzyme EcoP15I to generate a longer length of PET (unpublished), which results in a significant enhancement in both library construction and mapping efficiency.
Both versions of PET templates were sequenced by Illumina platform at 2 x 36 bp Paired End sequencing.
See the Methods and References sections below for more details.
</P>

<H2>Display Conventions and Configuration</H2>

<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>). For each view, there are multiple subtracks that display
individually on the browser. Instructions for configuring multi-view tracks
are <A HREF="/goldenPath/help/multiView.html" TARGET=_BLANK>here</A>.
Color differences among the views are arbitrary. They provide a 
visual cue for distinguishing between the different cell types and 
compartments.
</P>

<DL>
<DT><i>Clusters</i>
<DD>The Clusters view shows clusters built from the alignments.
In the graphical display, the ends are represented by blocks connected by a 
horizontal line. In full and packed display modes, the arrowheads on the 
horizontal line represent the direction of transcription. 
<DT><i>Plus Raw Signal</i>
<DD>The Plus Raw Signal view graphs the base-by-base density of tags on the forward strand.
<DT><i>Minus Raw Signal</i>
<DD>The Minus Raw Signal view graphs the base-by-base density of tags on the reverse strand.
<DT><i>Alignments</i>
<DD>The Alignments view shows alignment of individual PET sequences. 
The alignment file follows the standard SAM/BAM format indicated in the <a href="http://samtools.sourceforge.net/SAM1.pdf#page=4" target="_blank">SAM Format Specification</a>. Some files also use the tag XA, generated by bowtie, to represent the total number of mismatches in the tag.
</DL>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>


<H2>Methods</H2>

<P>
Cells were grown according to the approved 
<A HREF="/ENCODE/protocols/cell" TARGET=_BLANK>ENCODE cell culture
protocols</A>.
Two different GIS RNA-PET protocols were used to generate the full length transcriptome PETs: one is based on a cloning-free RNA-PET library construction and sequencing strategy
(unpublished), and the other is a cloning-based library construction (Ng <i>et al.</i>
2005) and recent Illumina paired end sequencing.
</P>
<h3>Cloning-free RNA-PET (50 bp reads, 25 bp and 25 bp tag for each of the 5' and 3' ends)</h3>
<P>
<B>Method:</B>
The cloning-free RNA-PET libraries were generated from polyA mRNA samples and constructed using a recently modified GIS protocol (unpublished). Total RNA in good quality was used as starting material and purified with a MACs polyT column to obtain full length polyA mRNAs. Approximately 5 micrograms of enriched polyA mRNA was used for reverse transcription to convert polyA mRNA to full length cDNA. Specific linker sequences were ligated to the full length cDNA. The modified cDNA was circularized by ligation generating circular cDNA molecules. The 25 bp tag from each end of the full length cDNA was extracted by type II enzyme EcoP15I digestion. The resulting PETs were ligated with sequencing adaptors at the both ends, amplified by PCR, and further purified as complex templates for paired end (PE) sequencing using Illumina platforms.
</P>
<P>
<B>Data:</B>
The sequenced RNA-PETs resulted in paired 2x25 bp reads from 5' and 3' ends of a cDNA. Redundant and noisy reads were excluded from downstream analysis. Strand specific orientation of each PET was determined using the barcode built into the sequencing template. The oriented RNA-PET was mapped onto the reference genome allowing up to two mismatches. Majority of PETs mapped to known transcripts. A small portion of misaligned PETs, defined as discordant PETs, mapped too far from each tag, with wrong orientations, or to different chromosomes. These discordant PETs indicated the existence of some transcription variants that could be caused by genomic structural variants such as: fusions, deletions, insertions, inversions, tandem repeats, translocations or RNA trans-splicing etc. 
</P>

<h3>Cloning-based RNA-PET (34 bp reads, 18 bp and 16 bp tag for each of the 5' and 3' ends)</h3>
<P>
<B>Method:</B>
The cloning-based RNA-PET (GIS-PET) libraries were generated from polyA RNA samples and constructed using the protocol described by Ng et al. (2005). Total RNA in good quality was used as starting material and further purified with a MACs polyT column to enrich polyA mRNA. Approximately 10 micrograms of polyA enriched mRNA was reverse transcribed resulting in full length cDNA. The obtained full length cDNA was modified with specific linker sequences and ligated to a GIS-developed (pGIS4) vector. The resulting plasmids form a complex full length cDNA library, which was cloned into E. coli. The plasmid DNA was then isolated from the library, followed by MmeI (a type II enzyme) digestion to generate a final length of 18 bp/16 bp ditags from each end of the full length cDNA. The single ditag (or PET) was then ligated to form a diPET structure (a concatemer with two unrelated PET linked by a linker sequence) to facilitate Illumina Paired End sequencing. 
</P>

<p><b>Data:</b>Sequencing of clone-based RNA-pets resulted in paired reads of 18bp and 16bp corresponding to the 5' and 3' end of each cDNA, respectively. The redundant reads were filtered out and unique reads were included for analysis. PET sequences 
were then mapped to (GRCh37, hg19, excluding mitochondirion, haplotypes, randoms and chromosome Y) reference genome using 
the following specific criteria (Ruan et al. 2007): </p>
<ul>
<li>A minimal continuous 16 bp match must exist for the 5' signature; the 3' signature must have a minimal continuous 14 bp match</li>
<li>Both 5' and 3' signatures must be present on the same chromosome</li>
<li>Their 5' to 3' orientation must be correct (5' signature followed by 3' signature)</li>
<li>The maximal genomic span of a PET genomic alignment must be less than one million bp</li>
</ul>
<P>
PETs mapping to 2-10 locations are also included and may represent duplicated genes or pseudogenes in the genome. 
A majority of PETs mapped on the known transcripts or splice variants. A small portion of misaligned PETs, 
defined as discordant PETs, mapped either too far from each other, in the wrong orientation, or 
to different chromosomes. The presence of discordant PETs indicates that some transcriptional variants exist. These variants could be caused by genomic structural variants such as: fusions, deletions, insertions, inversions, tandem repeats, translocation or RNA trans-splicing etc. 
</P>
<h3>Clusters</h3>
<p>PETs were clustered using the following procedure: the mapping location of the 5' and 3' tag of a given PET was extended by 100 bp in both directions creating 5' and 3' search windows. 
If the 5' and 3' tags of a second PET mapped within the 5' and 3' search window of the first PET then the two PETs were clustered 
and the search windows were adjusted so that they contained the tag extensions of the second PET. PETs which subsequently 
mapped with their 5' and 3' tags within the adjusted 5' and 3' search window, respectively, were also assigned to this cluster 
and search window was readjusted. This iterative process continued until no new PETs were fell within the search window. This process is repeated until all PETs were assigned to a cluster.</p>

<P>
The total count of PET 
sequences mapped to the same locus but with slight nucleotide differences may 
reflect the expression level of the 
transcripts. PETs that mapped to multiple locations may represent low 
complexity or repetitive sequences.
</P>

<H2>Verification</H2>
<p>To assess overall PET quality and mapping specificity, the top ten most abundant PET clusters that mapped to well-characterized 
known genes were examined. Over 99% of the PETs represented full-length transcripts, and the majority fell within 10 bp of the 
known 5' and 3' boundaries of these transcripts. The PET mapping was further verified by confirming the existence of physical 
cDNA clones represented by the ditags. PCR primers were designed based on the PET sequences and amplified the corresponding cDNA 
inserts either from full length cDNA library (cloning-based PET) or from total RNA isolate (cloning-free PET) for sequencing confirmation. 
</p>

<H2>Credits</H2>

<P>The GIS RNA-PET libraries and sequence data for transcriptome analysis were generated and analyzed by
scientists Xiaoan Ruan, Atif Shahab, Chialin Wei, and Yijun Ruan at the 
<A HREF="http://www.gis.a-star.edu.sg/internet/site" TARGET=_blank>Genome Institute of
Singapore</A>. 
</P>
<p>Contact:
<A HREF="mailto:&#114;&#117;&#97;&#110;yj&#64;&#103;&#105;&#115;.
&#97;&#45;&#115;t&#97;&#114;.
&#101;&#100;&#117;.
s&#103;">Yijun Ruan</A></p>

<H2>References</H2>

<P>Ng P, Tan JJ, Ooi HS, Lee YL, Chiu KP, Fullwood MJ, Srinivasan KG, Perbost C,
Du L, Sung WK, et al., <A HREF="http://nar.oxfordjournals.org/cgi/content/full/34/12/e84 TARGET=_blank">Multiplex sequencing of paired-end ditags (MS-PET): 
a strategy for the ultra-high-throughput analysis of transcriptomes and genomes</A>.
<EM>Nucleic Acids Res</EM>. 2006;34:e84.
</P>
<P>Ng P, Wei CL, Sung WK, Chiu KP, Lipovich L, Ang CC, Gupta S, Shahab A, 
Ridwan A, Wong CH, et al., <A HREF="http://www.nature.com/nmeth/journal/v2/n2/full/nmeth733.html" TARGET=_blank>Gene identification signature (GIS) analysis for 
transcriptome characterization and genome annotation</A>. 
<EM>Nat Methods</EM>. 2005;2:105-111.
</P>
<P>Ruan Y, Ooi HS, Choo SW, Chiu KP, Zhao XD, Srinivasan KG, Yao F, Choo CY, 
Liu J, Ariyaratne P, et al., <A HREF="http://genome.cshlp.org/content/17/6/828.full" TARGET=_blank>Fusion transcripts and transcribed retrotransposed 
loci discovered through comprehensive transcriptome analysis using Paired-End 
diTags (PETs)</A>.
<EM>Genome Res</EM>. 2007;17:828-838.
</P>
<H2> Data Release Policy </H2>

<P>Data users may freely use ENCODE data, but may not, without prior 
consent, submit publications that use an unpublished ENCODE dataset until 
nine months following the release of the dataset.  This date is listed in 
the <EM>Restricted Until</EM> column, above.  The full data release policy 
for ENCODE is available 
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P> 

