<H2>Description</H2>
<P>
This track shows $o_organism/$organism genomic alignments using
a gap scoring system that allows longer gaps than traditional
affine gap scoring systems. It can also tolerate gaps in both $o_organism 
and $organism simultaneously. These &quot;double-sided&quot;
gaps can be caused by local inversions and overlapping deletions
in both species. The $o_organism sequence is from the $o_date ($o_db)
assembly.</P>
<P>
The chain track displays boxes joined together by either single or 
double lines. The boxes represent aligning regions. 
Single lines indicate gaps that are largely due to a deletion in the 
$o_organism assembly or an insertion in the $organism assembly.
Double lines represent more complex gaps that involve substantial
sequence in both species. This may result from inversions, overlapping
deletions, an abundance of local mutation, or an unsequenced gap in one 
species.  In cases where there are multiple 
chains over a particular portion of the $organism genome, chains with 
single-lined gaps are often due to processed pseudogenes, while chains 
with double-lined gaps are more often due to paralogs and unprocessed 
pseudogenes. In the "pack" and "full" display
modes, the individual feature names indicate the chromosome, strand, and 
location (in thousands) of the match for each matching alignment.</P>


<H2>Display Conventions and Configuration</H2>
<P>By default, the chains to chromosome-based assemblies are colored
based on which chromosome they map to in the aligning organism. To turn
off the coloring, check the &quot;off&quot; button next to: Color
track based on chromosome.</P>
<P>
To display only the chains of one chromosome in the aligning
organism, enter the name of that chromosome (e.g. chr4) in box next to: 
Filter by chromosome.</P>

<H2>Methods</H2>
<P>
Transposons that have been inserted since the $o_organism/$organism
split were removed, and the resulting abbreviated genomes were
aligned with blastz.  The transposons were then put back into the
alignments.  The resulting alignments were converted into axt format
and the resulting axts fed into axtChain.  AxtChain organizes all the 
alignments between a single $o_organism and a single $organism chromosome
into a group  and makes a kd-tree out of all the gapless subsections
(blocks) of the alignments.  Next, maximally scoring chains of these
blocks were found by running a dynamic program over the kd-tree.  Chains
scoring below a threshold were discarded; the remaining chains are
displayed here.</P>

<H2>Credits</H2>
<P>
Blastz was developed at <A HREF="http://www.bx.psu.edu/miller_lab" 
TARGET=_blank>Pennsylvania State University</A> by 
Minmei Hou, Scott Schwartz, Zheng Zhang, and Webb Miller with advice from
Ross Hardison.</P>
<P>
Lineage-specific repeats were identified by Arian Smit and his
program <A HREF="http://www.repeatmasker.org" 
TARGET=_blank>RepeatMasker</A>.</P>
<P>
The axtChain program was developed at the University of California
at Santa Cruz by Jim Kent with advice from Webb Miller and David Haussler.
</P>
<P>
The browser display and database storage of the chains were generated
by Robert Baertsch and Jim Kent.</P>

<H2>References</H2>
<P>
Chiaromonte, F., Yap, V.B., Miller, W. 
<A HREF="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&db=PubMed&list_uids=11928468&dopt=Abstract" 
TARGET=_blank>Scoring pairwise genomic sequence alignments</A>. 
<em>Pac Symp Biocomput 2002</em>, 115-26 (2002).</P>
<P>
Kent, W.J., Baertsch, R., Hinrichs, A., Miller, W., and Haussler, D.
<A HREF="http://www.pnas.org/cgi/content/abstract/1932072100v1" 
TARGET=_blank>Evolution's cauldron: Duplication, deletion, and rearrangement
in the mouse and human genomes</A>.
<em>Proc Natl Acad Sci USA</em> <B>100</B>(20), 11484-11489 (2003).</P>
<P>
Schwartz, S., Kent, W.J., Smit, A., Zhang, Z., Baertsch, R., Hardison, R., 
Haussler, D., and Miller, W.
<A HREF="http://www.genome.org/cgi/content/abstract/13/1/103" 
TARGET=_blank>Human-Mouse Alignments with BLASTZ</A>. 
<em>Genome Res.</em> <B>13</B>(1), 103-7 (2003).</P>
