<H2>Description</H2>
<P>
This track provides RT-qPCR primers that were automatically designed to
measure the abundance of human and mouse transcripts using SYBR based qPCR.
Genome assemblies mm10 version (Mouse)  and hg19 version (Human) were used.
The primers were generated by an automatic design procedure, to target
all transcripts and all "possible" exon-exon and intron-exon junctions
in the human and mouse transcriptomes. 
</P>
<P>
Note that not all consecutive
exon-intron-exon triplets generate "possible" primer pairs, 
defined as satisfying a set of imposed design rules:
<OL>
<LI> We do not address the first exon-exon junction, to avoid problems
related to abortive transcription.</LI>
<LI>Intron length should be more then 800bp to avoid problems of double
products in amplification.</LI>
<LI>We design only junction primers, so that forward and reverse primers
must flank the junction.</LI>
<LI>Melting temperature of the primers should be between 60 &lt; Tm &lt; 63
(optimally 60.5 according to Breslauer et al. (1986).</LI>
<Li>Primer length should be 18-25bp.</LI>
<LI>Product size should be 60-125bp.</LI>
<LI>We first design primers for the intron-exon junction (pre-mRNA);
choose the two best primer pairs, and then for the reverse primer of
each pair we design two options for the corresponding forward primer of
the exon-exon junction (mRNA).</LI>
<LI>We use the "PRIMER MISPRIMING LIBRARY" of the 
<A HREF="http://frodo.wi.mit.edu/" TARGET=_blank>primer3</A> software:
"human" for human and "rodent" for mouse transcriptome, respectively.</LI>
</OL>
</P>
<P>
The track enables the user to easily obtain primers to almost all
transcripts in the transcriptome, without the need to go through the
tedious process that might involve human errors. After choosing the
primers the track includes a link to the primers' details, displayed
in a standard format. To meet the need to design a large number of
primers, we also provide  
<A HREF="http://www.weizmann.ac.il/complex/compphys/software/Amit/primers/batch_query_qpcr_primers.htm" TARGET=_blank>a batch query website</A>.
</P>

<H2>Methods</H2>
<p>
We use the UCSC gene model as a reference of the gene structure and the
<A HREF="http://frodo.wi.mit.edu/" TARGET=_blank>primer3</A> 
software as the design engine. The
software goes over all possible exon-exon junctions in the transcriptome
and applies our design rules/parameters to provide two primer pairs for
every "possible" intron-exon junction and four pairs for every
"possible" exon-exon junction.
</P>
<H2>Display Conventions</H2>
<P>
The resulting suggested primer pairs can be easily viewed,
the primers to amplify pre-mRNA (intro-exon junctions) are
shown  in  <font color="red"><b>red</b></font>  
and the primers to amplify mRNA (exon-exon junctions) in 
 <font color="blue"><b>blue</b></font>. 
For each pre-mRNA primer pair there are two corresponding mRNA primers 
(that use the same reverse primer, if possible). 
Each pair has a unique code which stands
for the gene name and the junction name. For example, the pair
"NR4A1_uc001rzt.2_1_1'' amplifies pre-mRNA, and the corresponding mRNA
primers are "NR4A1_uc001rzt.2_1_1_1 and NR4A1_uc001rzt.2_1_1_2''.
Click on the primer pair you select; this will take you to a new page
with details.
To view the primers' properties, click on the number next to the
instruction "Click here for primer details",
which opens a new page with the primer properties listed.

<H2>References</H2>
<P>
Zeisel A, Yitzhaky A and Domany E.
<A HREF="http://www.weizmann.ac.il/complex/compphys/software/Amit/primers/doc_qPCR_primers_track_v0.pdf" TARGET=_blank>An accessible database for mouse and human whole transcriptome qPCR primers </A>. (2013)
</P>

<P>
Breslauer, K. J., Frank, R., BlAcker, H. and Marky, L. A.
<A HREF="http://www.pnas.org/content/83/11/3746" TARGET=_blank>Predicting DNA duplex stability from the base sequence</A>.
<EM>Proc Natl Acad Sci U S A</EM>. 1986 Jun;83(11):3746-50.
</P>

<P>
Rozen, S. and Skaletsky, H. J.
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/10547847" TARGET=_blank>Primer3 on the WWW for general users and for biologist programmers</A>.
<EM>Methods Mol Biol</EM>. 2000;132:365-86.
</P>


