<H2>Description</H2>
<P>
This track displays tRNA genes predicted by using 
<A HREF="http://lowelab.ucsc.edu/tRNAscan-SE/" TARGET=_blank>tRNAscan-SE</A> v.1.23. 
</P>
<P>
tRNAscan-SE is an integrated program that uses tRNAscan (Fichant) and an A/B box motif detection 
algorithm (Pavesi) as pre-filters to obtain an initial list of tRNA candidates. 
The program then filters these candidates with a covariance model-based 
search program <A HREF="ftp://selab.janelia.org/pub/software/cove/" TARGET=_blank>
COVE</A> (Eddy) to obtain a highly specific set of primary sequence 
and secondary structure predictions that represent 99-100% of true tRNAs 
with a false positive rate of fewer than 1 per 15 gigabases.</P>
<P>
Detailed tRNA annotations for eukaryotes, bacteria, and archaea are available at
<A HREF="http://gtrnadb.ucsc.edu" TARGET=_blank>Genomic tRNA Database (GtRNAdb)</A>. 
</P>
<P>
What does the tRNAscan-SE score mean?  Anything with a score above 20 bits is likely to be
<I>derived</I> from a tRNA, although this does not indicate whether the tRNA gene still encodes a 
functional tRNA molecule (i.e. tRNA-derived SINES probably do not function in the ribosome in translation).
Vertebrate tRNAs with scores of >60.0 (bits) are likely to encode functional tRNA genes, and 
those with scores below ~45 have sequence or structural features that indicate they probably are
no longer involved in translation.  tRNAs with scores between 45-60 bits are in the "grey" zone, and may
or may not have all the required features to be functional.  In these cases, tRNAs should be inspected
carefully for loss of specific primary or secondary structure features (usually in alignments with other
genes of the same isotype), in order to make a better educated guess.  These rough score range guides 
are not exact, nor are they based on specific biochemical studies of atypical tRNA features,
so please treat them accordingly.
</P>
<P>
Please note that tRNA genes marked as "Pseudo" are low scoring predictions that are mostly pseudogenes or 
tRNA-derived elements. These genes do not usually fold into a typical cloverleaf tRNA secondary 
structure and the provided images of the predicted secondary structures may appear rotated.
</P>

<H2>Credits</H2>
<P>
Both tRNAscan-SE and GtRNAdb are maintained by the
<A HREF="http://lowelab.ucsc.edu" TARGET=_blank>Lowe Lab</A> at UCSC.
</P>
<P>
Cove-predicted tRNA secondary structures were rendered by NAVIEW (c) 1988 Robert E. Bruccoleri.
</P>

<H2>References</H2>
<P>
When making use of these data, please cite the following articles:</P>
<P>
Pavesi A, Conterio F, Bolchi A, Dieci G, Ottonello S.
<A HREF="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC523650/pdf/nar00031-0135.pdf" TARGET=_blank>
Identification of new eukaryotic tRNA genes in genomic DNA databases by a multistep
weight matrix analysis of transcriptional control regions.</A>
<EM>Nucleic Acids Res</EM>. 1994 Apr 11;22(7):1247-56.<BR>
</P>
<P>
Eddy SR, Durbin R. 
<A HREF="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC308124/pdf/nar00035-0157.pdf" TARGET=_blank>
RNA sequence analysis using covariance
models.</A>
<EM>Nucleic Acids Res</EM>. 1994 Jun 11;22(11):2079-88.<BR>
</P>
<P>
Fichant GA, Burks C. 
<A HREF="http://www.sciencedirect.com/science/article/pii/002228369190108I" TARGET=_blank>
Identifying potential tRNA genes in genomic DNA
sequences.</A>
<EM>J Mol Biol</EM>. 1991 Aug 5;220(3):659-71.<BR>
</P>

Lowe TM, Eddy SR. 
<A HREF="http://www.ncbi.nlm.nih.gov/pubmed/9023104?dopt=Abstract" TARGET=_blank>tRNAscan-SE: A program for
improved detection of transfer RNA genes in genomic sequence.</A>
<EM>Nucleic Acids Res</EM>. 1997 Mar 1;25(5):955-64.
<P>
Chan PP, Lowe TM. 
<A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/gkn787" TARGET=_blank>GtRNAdb: A database 
of transfer RNA genes detected in genomic sequence.</A>
<EM>Nucleic Acids Res</EM>. 2009 Jan;37(Database issue):D93-7.
<P>
