<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=415>
  <TR><TD ALIGN=RIGHT>
    <A HREF="http://www.ncbi.nlm.nih.gov/genome/51" TARGET=_blank>
    <IMG SRC="/images/human.jpg" WIDTH=275 HEIGHT=275 ALT="Human"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1><em>Homo sapiens</em><BR> 
    </FONT>
    <FONT SIZE=-2> 
      (<A HREF="http://www.cbse.ucsc.edu/"
      TARGET=_blank>Graphic courtesy of CBSE</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
<B>UCSC Genome Browser assembly ID:</B> hg38<BR>
<B>Sequencing/Assembly provider ID:</B> GRCh38 Genome Reference Consortium Human Reference 38 (GCA_000001405.2)<BR>
<B>Assembly date:</B> Dec. 2013<BR>
<B>GenBank accession ID:</B> GCA_000001305.2<BR>
<B>NCBI Genome information:</B> <A HREF="http://www.ncbi.nlm.nih.gov/genome/51"
TARGET="_blank">NCBI genome/51 (Homo sapiens)</A><BR>
<B>NCBI Assembly information:</B>
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/883148"
TARGET="_blank">NCBI assembly/883148 (GRCh38/GCA_000001405.2)</A><BR>
<B>BioProject information:</B></I> <A HREF="http://www.ncbi.nlm.nih.gov/bioproject/31257"
TARGET="_blank">NCBI Bioproject: 31257 TO BE CORRECTED</A>
</P>
<HR>
<P>
<B>Search the assembly:</B>
<UL>
<LI>
<B>By position or search term: </B> Use the &quot;position or search term&quot;
box to find areas of the genome associated with many different attributes, such
as a specific chromosomal coordinate range; mRNA, EST, or STS marker names; or
keywords from the GenBank description of an mRNA.
<A HREF="../goldenPath/help/query.html">More information</A>, including sample
queries.
<LI>
<B>By gene name: </B> Type a gene name into the &quot;search term&quot; box,
choose your gene from the drop-down list, then press &quot;submit&quot; to go 
directly to the assembly location associated with that gene.
<A HREF="../goldenPath/help/geneSearchBox.html">More information</A>.
<LI>
<B>By track type: </B> Click the &quot;track search&quot; button
to find Genome Browser tracks that match specific selection criteria.
<A HREF="../goldenPath/help/trackSearch.html">More information</A>.
</UL>
</P>
<HR>
<P>
<B>Download sequence and annotation data:</B>
<UL>
<LI><A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/">Using FTP</A>
(recommended)
<LI><A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#human">Using HTTP</A>
<LI><A HREF="../goldenPath/credits.html#human_credits">Data use conditions and
restrictions</A>
<LI><A HREF="../goldenPath/credits.html#human_credits">Acknowledgments</A>
</UL>
</P>
<hr>
<h3>Assembly Details</h3>

<p>
<b>Chromosome naming scheme</b><br>
With the release of hg38, there has been a slight change to the UCSC chromosome naming scheme:
<ul>
  <li>Haplotype chromosome, unplaced contig and unlocalized contig names now include
      their NCBI accession number (e.g., chr6_GL000256v2_alt)</li>
  <li>The &quot;v2&quot; at the end of the accession number indicates the NCBI version number</li>
  <li>Haplotype chromosome names consist of the chromosome number, followed by the NCBI accession
      number, followed by &quot;alt&quot;</li>
  <li>Unplaced contig names consist of the chromosome number, followed by the NCBI accession
      number, followed by &quot;random&quot;</li>
  <li>Unlocalized contig names consist of &quot;chrUn&quot; followed by the NCBI accession number</li>
</ul>
</p>

<p>
Some assembly statistics:
</p>

<TABLE BORDER=1 CELLPADDING = 3><TR><TH>number</TH><TH>haplotypes</TH><TH>unplaced contigs</TH></TR>
<TR><TD ALIGN="CENTER">chr1</TD><TD ALIGN="CENTER">12</TD><TD ALIGN="CENTER">9</TD></TR>
<TR><TD ALIGN="CENTER">chr2</TD><TD ALIGN="CENTER">15</TD><TD ALIGN="CENTER">2</TD></TR>
<TR><TD ALIGN="CENTER">chr3</TD><TD ALIGN="CENTER">16</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chr4</TD><TD ALIGN="CENTER">11</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chr5</TD><TD ALIGN="CENTER">13</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chr6</TD><TD ALIGN="CENTER">16</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr7</TD><TD ALIGN="CENTER">9</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr8</TD><TD ALIGN="CENTER">16</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr9</TD><TD ALIGN="CENTER">5</TD><TD ALIGN="CENTER">4</TD></TR>
<TR><TD ALIGN="CENTER">chr10</TD><TD ALIGN="CENTER">4</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr11</TD><TD ALIGN="CENTER">12</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chr12</TD><TD ALIGN="CENTER">13</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr13</TD><TD ALIGN="CENTER">6</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr14</TD><TD ALIGN="CENTER">4</TD><TD ALIGN="CENTER">8</TD></TR>
<TR><TD ALIGN="CENTER">chr15</TD><TD ALIGN="CENTER">9</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chr16</TD><TD ALIGN="CENTER">6</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chr17</TD><TD ALIGN="CENTER">18</TD><TD ALIGN="CENTER">3</TD></TR>
<TR><TD ALIGN="CENTER">chr18</TD><TD ALIGN="CENTER">10</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr19</TD><TD ALIGN="CENTER">43</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr20</TD><TD ALIGN="CENTER">4</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr21</TD><TD ALIGN="CENTER">7</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chr22</TD><TD ALIGN="CENTER">9</TD><TD ALIGN="CENTER">9</TD></TR>
<TR><TD ALIGN="CENTER">chrX</TD><TD ALIGN="CENTER">3</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER">chrY</TD><TD ALIGN="CENTER">0</TD><TD ALIGN="CENTER">1</TD></TR>
<TR><TD ALIGN="CENTER">chrM</TD><TD ALIGN="CENTER">0</TD><TD ALIGN="CENTER">0</TD></TR>
<TR><TD ALIGN="CENTER"></TD><TD ALIGN="CENTER"></TD><TD ALIGN="CENTER"></TD></TR>
<TR><TD ALIGN="CENTER">Total</TD><TD ALIGN="CENTER">261</TD><TD ALIGN="CENTER">42</TD></TR>
</TABLE>

<p>
There are a total of 127 unlocalized contigs.
</p>

<p>
See the <a href="../cgi-bin/hgTracks?chromInfoPage=">sequences</a> page for a complete
list of hg38 chromosome names.
</p>
<P>
Potential sections to include on this page:
<UL>
<LI>Description of new features such as centromeres</LI>
<LI>Mention of IUPAC characters and how UCSC does not use nor pass them
through to download files.  Perhaps mention where they can be found since
it is only a few locations.</LI>
<LI>Mention where the PAR regions are<BR>
<pre>
+-------+------------+-----------+------+
| chrom | chromStart | chromEnd  | name |
+-------+------------+-----------+------+
| chrX  |      10000 |   2781479 | PAR1 |
| chrX  |  155701382 | 156030895 | PAR2 |
| chrY  |      10000 |   2781479 | PAR1 |
| chrY  |   56887902 |  57217415 | PAR2 |
+-------+------------+-----------+------+
</pre></LI>
<LI>chrM fiasco AKA chrMt now == J01415.2, was: NC_001807/AF347015.1</LI>
</UL>
</P>
