<H2>Description</H2>
<P>
The GENCODE Basic track is composed of all the gene models in the <A TARGET="_blank" href="http://www.gencodegenes.org/">GENCODE v22 release.</A>
It includes both protein-coding genes and
non-coding RNA genes. 

<H2>Display Conventions and Configuration</H2>
<P>
This track in general follows the display conventions for
<A TARGET="_blank" HREF="../goldenPath/help/hgTracksHelp.html#GeneDisplay">gene prediction tracks</A>. The exons
for putative non-coding genes and untranslated regions are represented by relatively thin blocks,
while those for coding open reading frames are thicker. The following color key is used:</P>

<ul>
<li><span style="font-weight: bold;">Black</span> -- feature has a corresponding entry in the Protein
 Data Bank (PDB)</li>
<li><span style="color: #0C0C78; font-weight: bold;">Dark blue</span> -- transcript has been reviewed
 or validated by either the RefSeq or SwissProt staff</li>
<li><span style="color: #5050A0; font-weight: bold;">Medium blue</span> -- other RefSeq transcripts</li>
<li><span style="color: #8282D2; font-weight: bold;">Light blue</span> -- non-RefSeq transcripts</li>
</ul>

<P>
This track contains an optional <A TARGET="_blank" HREF="../goldenPath/help/hgCodonColoring.html">codon coloring
feature</A> that allows users to quickly validate and compare gene predictions.</P>

<H2>Methods</H2>
<P>
All the GENCODE genes in the comprehensive set are downloaded from the GENCODE 
<A href="http://www.gencodegenes.org/releases/current.html">website</A>.  
Data from other sources are correlated with the GENCODE data to build the knownTo tables.
</OL>

<H2>Related Data</H2>
<P>
The GENCODE Genes transcripts are annotated in numerous tables, each of which is also available as a
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg38/database/">downloadable file</a>.  These
include tables that link GENCODE Genes transcripts to external datasets (such as
<I>knownToLocusLink</I>, which maps GENCODE Genes transcripts to Entrez identifiers, previously known
as Locus Link identifiers), and tables that detail some property of GENCODE Genes transcript sequences
(such as <I>knownToPfam</I>, which identifies any Pfam domains found in the GENCODE Genes
protein-coding transcripts). One can see a full list of the associated tables in the
<a href="/cgi-bin/hgTables">Table Browser</a> by selecting GENCODE Genes from the <B>track</B> menu;
this list is then available on the <B>table</B> menu. Note that some of these tables refer to GENCODE
Genes by its former name of <I>Known Genes</I>, sometimes abbreviated as <I>known</I> or <I>kg</I>.
While the complete set of annotation tables is too long to describe, some of the more important
tables are described below.</P>
<UL> 
<LI> <B>kgXref</B> identifies the RefSeq, SwissProt, Rfam, or tRNA sequences (if any) which are 
associated with each transcript.
<LI> <B>knownToRefSeq</B> identifies the RefSeq transcript that each GENCODE Genes transcript is most
closely associated with. That RefSeq transcript is the RefSeq transcript that the GENCODE Genes transcript
overlaps at the most bases.</LI>  
<LI> <B>knownGeneMrna</B> contains the genomic sequence for each of the GENCODE Genes models. 
This may not be the same as the actual mRNA used to validate the gene model.</LI>
<LI> <B>knownGenePep</B> contains the protein sequences derived from the knownGeneMrna transcript
sequences. Any protein-level annotations, such as the contents of the knownToPfam table, are based
on these sequences.</LI>
<LI> <B>knownIsoforms</B> maps each transcript to a <EM>cluster ID</EM>, a cluster of isoforms of
the same gene.</LI>
<LI> <B>knownCanonical</B> identifies the canonical isoform of each cluster ID or gene using the 
ENSEMBL gene IDs to define each cluster. The canonical transcript is chosen using the APPRIS 
principal transcript when available.  
If no APPRIS tag exists for any transcript associated with the cluster, then a transcript in the 
BASIC set is chosen.  If no BASIC transcript exists, then the longest isoform is used.
</UL>

<H2>Credits</H2>
<P>
The GENCODE Genes track was produced at UCSC from the GENCODE comprehensive 
gene set using a computational pipeline developed by Jim Kent and Brian Raney.

<H2>References</H2>
<p>
Harrow J, Frankish A, Gonzalez JM, Tapanari E, Diekhans M, Kokocinski F, Aken BL, Barrell D, Zadissa
A, Searle S <em>et al</em>.
<a href="http://genome.cshlp.org/content/22/9/1760.long" target="_blank">
GENCODE: the reference human genome annotation for The ENCODE Project</a>.
<em>Genome Res</em>. 2012 Sep;22(9):1760-74.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22955987" target="_blank">22955987</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3431492/" target="_blank">PMC3431492</a>
</p>

<p>
Harrow J, Denoeud F, Frankish A, Reymond A, Chen CK, Chrast J, Lagarde J, Gilbert JG, Storey R,
Swarbreck D <em>et al</em>.
<a href="http://genomebiology.com/2006/7/S1/S4" target="_blank">
GENCODE: producing a reference annotation for ENCODE</a>.
<em>Genome Biol</em>. 2006;7 Suppl 1:S4.1-9.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16925838" target="_blank">16925838</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1810553/" target="_blank">PMC1810553</a>
</p>

<p>A full list of GENCODE publications are available
at <a href="http://www.gencodegenes.org/publications.html" target="_blank">The GENCODE Project
web site</a>.
</p>

<h2>Data Release Policy</h2>
<p>GENCODE data are available for use without restrictions.</p>
