<!DOCTYPE html>
<html>
    <head>
    <title>
        Umap Trackhub
    </title>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-48283039-1', 'pmgenomics.ca');
  ga('send', 'pageview');

</script>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-48283039-1', 'pmgenomics.ca');
  ga('send', 'pageview');

</script>
</head>

<body>
    <h1>Umap: Unique mappability of the genome</h1>
    <h2>Description</h2>
    <p>These tracks indicate regions with uniquely mappable reads of particular lengths.</p>
    <h4>Umap single-read mappability</h4>
    <ul>
        <li><i>Umap S24</i>:   Uniquely mappable regions with a read length of 24 nucleotides</li>
        <li><i>Umap S36</i>:   Uniquely mappable regions with a read length of 36 nucleotides</li>
        <li><i>Umap S50</i>:   Uniquely mappable regions with a read length of 50 nucleotides</li>
        <li><i>Umap S100</i>:  Uniquely mappable regions with a read length of 100 nucleotides</li>
        <li><i>Umap S122</i>:  Uniquely mappable regions with a read length of 122 nucleotides</li>
    </ul>
    <h4>Umap multi-read mappability</h4>
    <ul>
        <li><i>Umap M24</i>:   Multi-read mappability with a read length of 24 nucleotides</li>
        <li><i>Umap M36</i>:   Multi-read mappability with a read length of 36 nucleotides</li>
        <li><i>Umap M50</i>:   Multi-read mappability with a read length of 50 nucleotides</li>
        <li><i>Umap M100</i>:  Multi-read mappability with a read length of 100 nucleotides</li>
        <li><i>Umap M122</i>:  Multi-read mappability with a read length of 122 nucleotides</li>
    </ul>
    <p>You can use these tracks for many purposes including filtering unreliable signal from
    sequencing assays.
    <p><b>Umap single-read mappability</b> tracks mark any region of the genome that is uniquely mappable by at least one <i>k</i>-mer.
    To calculate the single-read mappability, you must find the overlap of a given region with this track.
    <b>Umap multi-read mappability</b> tracks represent the probability that
    a randomly selected <i>k</i>-mer which overlaps with a given position is uniquely mappable.
    <p>For greater detail and explanatory diagrams, see the
    <a href="http://biorxiv.org/content/early/2016/12/20/095463">preprint</a>,
    the <a href="http://bismap.hoffmanlab.org">Umap and Bismap project website</a>
    , or the <a href="https://bitbucket.org/hoffmanlab/umap">Umap and Bismap software documentation</a>.</p>
    <h2>Track format</h2>
    <p>Single-read mappability tracks: bigBed 6-column format</p>
    <p>Multi-read mappability tracks: bigWig</p>
    <p>These tracks are available for m9/GRCm37, mm10/GRCm38, hg19/GRCh37 and hg38/GRCh38 genome assemblies.</p>
    <h2>Credits</h2>
    <p><a href="https://sites.google.com/site/anshulkundaje/">Anshul Kundaje</a> (Stanford university)
    created the original Umap software in MATLAB.
    Mehran Karimzadeh (<a href="https://www.pmgenomics.ca/hoffmanlab/">Michael Hoffman lab</a>,
    Princess Margaret Cancer Centre) implemented the Python version of Umap and added features, including Bismap.</p>
    <h2>Contact, support and questions</h2>
    <p>Please ask your questions about the Umap software and this track hub using the <a href="https://groups.google.com/forum/#!forum/ubismap">Umap and Bismap mailing list</a>.
    To report a bug or request,
    please use the <a href="https://bitbucket.org/hoffmanlab/umap/issues">Umap and Bismap issue tracker</a>.</p>

</body>

</html>
