<h2>Description</h2>
<p>
These tracks contain the results of DNase I hypersensitivity experiments performed by the
<a href="http://www.gs.washington.edu/faculty/stamj.htm" target="_blank">John Stamatoyannapoulos lab</a>
at the University of Washington from September 2007 to January 2011, as part of the
<a href="../ENCODE" target="_blank">ENCODE project first production phase</a>.
</p>

<p>
Other views of this data (along with additional documentation) are available from the hg19
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase">ENCODE UW DNaseI HS track</a>.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
This track is a composite annotation track containing multiple subtracks, one for each cell type.
The display mode and filtering of each subtrack can be individually controlled. 
For more information about track configuration, see
<a target="_blank" href="../goldenPath/help/multiView.html">Configuring Multi-View Tracks</a>.
</p>

<h2>Methods</h2>
<p>
Raw sequence data files were processed by the UCSC ENCODE DNase analysis pipeline (July 2014 specification), diagrammed here:
<div align="center">
<img style="margin:8px; border:1px solid; height:216; width;483;" src="../images/encodeDnasePipeline.png" alt="ENCODE DNase Pipeline">
<font size=-1>Credit: Qian Alvin Qin, X. Liu lab</font>
</div>
<p>
Briefly, sequence files were aligned to the hg38 (GRCh38) genome assembly augmented with 'sponge'
sequence (ref).  Multi-mapped reads were removed, as were reads that aligned to 'sponge' or
mitochondiral sequence.  Results from all replicates were pooled, and further processed by
the Hotspot program to call peaks as well as broader regions of activity ('hotspots'), and to
create signal density graphs.
Signal graphs were normalized so the average value genome-wide is 1.
</p>

<h2>Credits</h2>
<p>
The processed data for this track were produced by UCSC.  Credits for the primary data 
underlying this track are included in the
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase">ENCODE UW DNaseI HS track</a>
description.
</p>

<h2>References</h2>

<p>
Miga KE, Eisenhart E, Kent WJ.
Utilizing mapping targets of sequences underrepresented in the reference assembly to reduce false
positive alignments. <em>NAR Methods</em>. Accepted.
</p>

<p>
Thurman RE, Rynes E, Humbert R, Vierstra J, Maurano MT, Haugen E, Sheffield NC, Stergachis AB, Wang
H, Vernot B <em>et al</em>.
<a href="http://www.nature.com/nature/journal/v489/n7414/abs/nature11232.html" target="_blank">
The accessible chromatin landscape of the human genome</a>.
<em>Nature</em>. 2012 Sep 6;489(7414):75-82.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22955617" target="_blank">22955617</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3721348/" target="_blank">PMC3721348</a>
</p>

<p>
See also the references in the
<a target="_blank" href="/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase">ENCODE UW DNaseI HS</a>
track.
</p>
