<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
        <title>Software and Resources for Analyzing ENCODE data</title>
        <link rel="stylesheet" href="/style/HGStyle.css" type="text/css">
        <link rel="stylesheet" href="/style/encodeProject.css" type="text/css">
    </head>

    <body>
        <!--#include virtual="/inc/encodeProjectSoftware.topbar.html"-->

        <div class="encodeHeader">
            <a href="http://www.genome.gov/10005107"><img src="/images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" title="ENCODE Project at NHGRI"></a>
            <span class="txt">Software and Resources for Analyzing ENCODE data</span>
        </div>
       <div class="wrapper">
                <div class="bar"><h4 class="title">Tools for analyzing ENCODE data</h4></div>
                <div class="content">
                      <ul>
                        <p><li>
                        <a class="thick" href="http://www.broadinstitute.org/~pouyak/motif-disc/human/" target="_blank">ENCODE-motifs</a><br>
                        A database that uncovers the molecular basis of TF binding in the human genome based on regulatory
                        motif analysis of all Transcription Factors (TFs) grouped by family. This allows browsing of all known
                        motifs for each factor, curated from TRANSFAC, Jaspar, and Protein Binding Microarray (PBM) experiments,
                        and their enrichment and instances within corresponding TF binding experiments. It also provides a list
                        of novel regulatory motifs discovered by systematic application of several motif discovery tools
                        (including MEME, MDscan, Weeder, AlignACE) and evaluated based on their enrichment relative to control
                        motifs within TF-bound regions. ENCODE-motifs also provides a genome-wide map of regulatory motif
                        instances in the human genome for both known and novel motifs.
                        </p>

                        <p>
                        Kheradpour P, Kellis M.
                        ENCODE-motifs: systematic analysis of regulatory motifs associated with transcription factor binding in the human genome. (Submitted)
                        </li></p>

                        <p><li>
                        <a class="thick" href="http://www.factorbook.org/" target="_blank">Factorbook</a><br>
                        A wiki-style resource that organizes all the information associated with each transcription factor (TF),
                        including the ChIP-seq peaks, discovered motifs, TF-TF interactions, and the chromatin features
                        (histone modification patterns, DNase I cleavage, and nucleosome positioning) around the ChIP-seq
                        peaks. Will be updated as the project proceeds. The Factorbook display of this information is
                        transcription factor anchored and dynamic.
                        </p>

                        <p>
                        Wang J, Zhuang J, Iyer S, Lin X, Whitfield TW, Greven M, Pierce BG, Dong X, Kundaje A, Cheng Y <em>et al</em>.
                        Sequence Features and Chromatin Structure around the Genomic Regions Bound by 119 Human Transcription Factors. (Submitted)
                        </li></p>

                        <p><li>
                        <a class="thick" href="http://www.broadinstitute.org/mammals/haploreg/haploreg.php" target="_blank">HaploReg</a><br>
                        Explores annotations of the noncoding genome at variants on haplotype blocks, such as candidate
                        regulatory SNPs at disease-associated loci.
                        </p>

                        <p>
                        Under Set Options tab, set Browse ENCODE button to on and select an LD threshold.
                        </p>

                        <p>
                        Under Build Query Tab, enter SNP (rsXXXXX)
                        </p>

                        <p>
                        HaploReg returns SNPs in LD, their frequency in 3 populations, and also tells you  what evidence
                        ENCODE has found for regulatory protein binding (mouse over to see the protein names), chromatin
                        structure (mouse over to see the cell types), and the chromatin state of the region (the chromatin
                        state can predict an enhancer or promoter), and putative transcription factor binding motifs that are
                        altered by the variant.  Clicking on the SNP name hyperlink reveals chromosomal position, sequence,
                        allele frequencies in three populations, and ENCODE chromatin state in 9 cell types.
                        </p>

                        <p>
                        Ward LD, Kellis M.
                        <a href="http://www.ncbi.nlm.nih.gov/pubmed/22064851" target="_blank">
                        HaploReg: a resource for exploring chromatin states, conservation, and regulatory motif alterations within sets of genetically linked variants</a>.
                        <em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D930-4.
                        </li></p>

                        <p><li>
                        <a class="thick" href="http://regulome.stanford.edu/" target="_blank">RegulomeDB</a><br>
                        Identifies DNA features and regulatory elements in non-coding regions of the human genome. One can
                        enter dbSNP IDs, BED files, VCF files, or GFF3 files.  A score is returned assessing the evidence
                        for regulatory potential.  Clicking on the score reveals the data supporting the inference, by data
                        type and cell type.  One can also click on hyperlinks to see the SNP or the region in the UCSC
                        browser, ENSEMBL browser, and dbSNP.
                        </li></p>

                        <p><li>
                        <a class="thick">Tutorial on functional annotation of non-coding sequences</a><br>
                        Briefly describes how to display the basic ENCODE tracks, using the UCSC genome browser, at
                        SNPs identified from GWAS studies.
                        </p>

                        <p>
                        Mortlock DP, Pregizer S.
                        <a href="http://www.ncbi.nlm.nih.gov/pubmed/22241655" target="_blank">
                        Identifying functional annotation for noncoding genomic sequences</a>.
                        <em>Curr Protoc Hum Genet</em>. 2012 Jan;Chapter 1:Unit1.10.
                        </li></p>

                        <p><li>
                        <a class="thick" href="http://www.openhelix.com/ENCODE/" target="_blank">Tutorials on using ENCODE data and the ENCODE portal at UCSC</a><br>
                        A suite of tutorials and PowerPoint handouts that include how to identify ENCODE data on the
                        UCSC browser, what types of ENCODE data are available, what is the data use policy, how to display
                        the data, and how to download the data.  There are also links to other tutorials on the use of the
                        UCSC genome browser.
                        </li></p>
                      </ul>
               </div><!--end content-->
       </div><!--end wrapper--> 
       <p class="date">Updated 01 May 2012</p>
    </body>
</html>
