<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
	"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=ISO-8859-1">
  <meta name="description" content="ENCODE Project at UCSC">
  <meta name="keywords"
 content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <title>ENCODE Project at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
 <script type="text/javascript" src="../js/jquery.js"></script>
 <script type="text/javascript" src="../js/staticDoc.js"></script>
</head>
<body style="font-family: arial, Geneva, Helvetica, sans-serif; background-color: #fff;" alink="#6600ff"
 link="#0000cc" vlink="#330066">
<a name="TOP"></a><!-- Header Images and links -->
<table border="0" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <th colspan="2" align="left">
<img src="../images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" height="75"> &nbsp;&nbsp;
 <font size="6" face="arial, Geneva, Helvetica, sans-serif"><font color="004D99">
 Encyclopedia of DNA Elements</font>
       </th>
    </tr>
    <tr height="100%" valign="top">
      <td rowspan="1" height="100%" valign="top"><!-- outer table is to align leftbar & content boxes at top of page -->
      <table border="0" cellpadding="0" cellspacing="0" width="100%">
        <tbody>
          <tr>
            <td><img src="../images/DOT.gif" alt="" height="7"></td>
          </tr>
          <tr height="100%">
            <td height="100%"><!--outer table is for black border-->
            <table bgcolor="#004D99" border="0" cellpadding="1" cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table border="0" cellpadding="10" cellspacing="0" width="100%">
                    <tbody>
                      <tr height="100%">
                        <td bgcolor="#0004D99" height="100%" valign="top">
                        <font color="#FFCC00"><b><em>Human</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
                        <a class="leftbar" href="analysis.html">Integrative Analysis</a>
                        <hr><a class="leftbar" href="/ENCODE/dataMatrix/encodeDataMatrixHuman.html">Experiment Matrix</a>
                        <hr><a class="leftbar" href="/ENCODE/dataSummary.html" title="formerly Data Summary">Experiment List</a>
                        <hr><a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloads.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=hg19">Genome Browser (hg19)</a>
                        <hr><a class="leftbar" href="http://genomewiki.cse.ucsc.edu/index.php/Genome_Browser_Session_Gallery"
                        TARGET=_blank>Session Gallery</a>
                        <hr><a class="leftbar" href="cellTypes.html">Cell Types</a>
                        <hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
                        <font color="#FFCC00"><b><em>Mouse</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
                        <a class="leftbar" href="/ENCODE/dataMatrix/encodeDataMatrixMouse.html">Experiment Matrix</a>
                        <hr><a class="leftbar" href="/ENCODE/dataSummaryMouse.html"title="formerly Data Summary">Experiment List</a>
                        <hr><a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloadsMouse.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=mm9">Genome Browser (mm9)</a>
                        <hr><a class="leftbar" href="cellTypesMouse.html">Cell Types</a>
                        <hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
                        <font color="FFCC00"><b><em>General</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8"><br>
                        <a class="leftbar" href="otherTerms.html">Registered Variables</a>
                        <hr><a class="leftbar" href="antibodies.html">Antibodies</a>
                        <hr><a class="leftbar" href="usageResources.html">Education and Outreach</a>
                        <hr><a class="leftbar" href="releaseLog.html">Release Log</a>
                        <hr><a class="leftbar" href="terms.html">Data Policy</a>
                        <hr><a class="leftbar" href="dataStandards.html">Data Standards</a><font color="#FF0000"></font>
                        <hr><a class="leftbar" href="/FAQ/FAQformat.html#ENCODE">File Formats</a>
                        <hr><a class="leftbar" href="softwareTools.html">Software Tools</a>
                        <hr><a class="leftbar" href="pubs.html">Publications</a>
                        <!--
                                        <font color="#FF0000"><sup><strong><em>Updated</strong></em></sup></font>
                                        -->

                        <hr><a class="leftbar" href="contributors.html">Contributors</a>
                        <hr><a class="leftbar" href="pilot.html">Pilot Project</a>
                        <!--
                        <hr><a class="leftbar" href="http://www.cbse.ucsc.edu/jobs/" target="_blank">Jobs</a>
                        -->
                        <hr><a class="leftbar" href="contacts.html">Contacts</a>
                        <hr><br>
                        <img src="../images/DOT.gif" alt="" height="970" width="1"> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
<!-- Content Tables --> <td valign="top">
      <table border="0" cellpadding="5">
        <tbody>
          <tr>
            <td><!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="About"></a><font size="4"><b>&nbsp;
About ENCODE Data</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="15"><br>
                              </td>
                              <td>
<p>The <a href="http://www.genome.gov/10005107" target="_blank">Encyclopedia of DNA Elements</a>
(ENCODE) Consortium is an international collaboration
of research groups funded by the National Human Genome Research
Institute (<a href="http://www.genome.gov/" target="_blank">NHGRI</a>).
The goal of ENCODE is to build a comprehensive parts list of
functional elements in the human genome, including elements that
act at the protein and RNA levels,
and regulatory elements that control cells and
circumstances in which a gene is active.
</p>
<!--#if expr='"$SERVER_NAME" = /hgwdev.*.cse.ucsc.edu/ || "$SERVER_NAME" = "genome-test.cse.ucsc.edu"'-->
<p>
<b>WARNING: This is a test site.</b>
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  
For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<table><tr>
<td><a href="/ENCODE/aboutScaleup.html" target="_blank"><img src="../images/EncodeNatureGraphic.png" alt="" height="200">
<em><font size=-2>Click to enlarge</em></font></td>

<td>
ENCODE data are now available for the entire human genome.
<b><em>All ENCODE data are free and available for immediate use via </em></b>:
<ul class="simple">
<li><a href="/ENCODE/search.html" target="_blank">Search</a> for displayable tracks and downloadable files
</li>
<li><a href="/ENCODE/downloads.html" target="_blank">Download</a> of data files
</li>
<li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Kate&hgS_otherUserSessionName=encodePortalSession"
target="_blank">Visualization</a> in the UCSC Genome Browser (ENCODE data marked with
the <img src="../images/nhgri.gif" alt="" height="16" width="16"> NHGRI logo)
</li>
<li><a href="/cgi-bin/hgTables?db=hg18&hgta_group=regulation&hgta_track=wgEncodeHudsonalphaChipSeq"
target="_blank">Data mining</a> with the UCSC Table Browser and other
<a href="../">UCSC Genome Bioinformatics tools</a>
</li>
</ul>
</td>

</tr></table>
</p>
<p>
To search for ENCODE data related to your area of interest and set up a browser view, use the UCSC
<a href="dataMatrix/encodeDataMatrixHuman.html" target="_blank">Experiment Matrix</a> or
<a href="/cgi-bin/hgTracks?hgt_tSearch=1" target="_blank">Track Search tool</a> (<em>Advanced</em> features).
The <a href="dataSummary.html" target="_blank">Experiment List (Human)</a> and 
 <a href="dataSummaryMouse.html" target="_blank">Experiment List (Mouse)</a> links provide
comprehensive listings of ENCODE data that is released or in preparation.
</p>
<p>
All ENCODE data is freely available for download and analysis.
However, before publishing research that uses ENCODE data,
please read the <a href="terms.html" target="_blank">ENCODE Data Release Policy</a>,
which places some restrictions on publication use of data for nine
months following data release. &nbsp;&nbsp;
<a href="/ENCODE/aboutScaleup.html">Read more</a> about ENCODE data at UCSC.</p>

                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <br>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><!--cheap trick to get background on row to be continuous-->
                              <table border="0" cellpadding="0" cellspacing="0" width="100%">
                                <tbody>
                                  <tr>
                                    <td align="left"><font size="4"><a name="TOC"></a><b>&nbsp; News</b></font> </td>
                                    <td align="right"><a href="newsarch.html"><img src="../images/news.jpg" alt="" align="right" border="0"></a> </td>
                                    <td align="right">&nbsp;</td>
                                  </tr>
                                </tbody>
                              </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0" cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="15"><br>
                              </td>
                              <td style="font-family: arial, Geneva, Helvetica, sans-serif;">

<!-- start news item -->
<font size="3"><b> 28 September 2012 - New ENCODE on-line tutorial and Quick Reference Cards</b></font>
<p>
To supplement the existing <em>ENCODE Foundations</em> tutorial, OpenHelix has developed a second
tutorial describing newer ENCODE data and access tools. Slides for this tutorial are freely available from the
website at: <a target=_blank href="http://openhelix.com/ENCODE2">http://openhelix.com/ENCODE2</a>.
Also available free (up to 30 cards shipped free in the US) is a <em>Quick Reference Card</em> that
provides a handy summary of access features.  For more information see
the <a target=_blank href="http://blog.openhelix.com/?p=13830">OpenHelix announcement</a>.
<hr>
<!-- end news item -->

<!-- start news item -->
<font size="3"><b> 27 September 2012 - ENCODE data releases: UNC/BSU ProtGenc, UNC FAIRE (Rel 2), HAIB TFBS (Rel 3), CSHL Long RNA-seq (Rel 3)</b></font>
<p>
One new track and three track updates were released on the human hg19 browser:<p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUncBsuProtGenc">
Proteogenomics Hg19 and GENCODE Mapping from ENCODE/Univ. North Carolina/Boise State Univ:</a>
This track displays mass spectrometry data that have been matched to genomic sequences
in GM12878, H1-hESC, H1-neuron, and K562 cell types.  Peptides were mapped to an in silico
translation and proteolytic digestion of the whole human genome (UCSC Hg19), and the GENCODE
translation of protein-coding transcripts database.  The track can be used to identify which parts
of the genome are translated into proteins, to verify which transcripts discovered by other ENCODE
experiments are protein-coding, to reveal new genes and/or splice variants and proteins with
post-translational modifications (PTM). Of particular interest is the possibility of uncovering the
translation of small open reading frames (ORFs), antisense transcripts, or protein-coding regions
that have been annotated as introns previously.</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromFaire">
Open Chromatin by FAIRE from ENCODE/OpenChrom(UNC Chapel Hill) (Release 2)</a>:
FAIRE (Formaldelhyde-Assisted Isolation of Regulatory Elements) is a method to isolate and identify
nucleosome-depleted regions of the genome. Release 2 of this track contains 12 new experiments,
including 11 new cell lines.</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs">
Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB (Release 3)</a>:
This release contains 110 new experiments including 3 new cell lines.
There were also data corrections (see Release Notes section for details).</p>
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlLongRnaSeq">
Long RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 3)</a>: 
This release contains additional files for many experiments including
GENCODE V10 referenced Transcripts, Genes and Exons (previous datasets
referenced GENCODE V7).  Complete details are in the track Release Notes.</p>
<hr>
<!-- end news item -->

<!-- start archived news item section -->
<!-- start news item -->
<a name="092512">
<font size="3"><b> 25 September 2012 -  Mouse ENCODE data releases: UW DNaseI DGF, UW DNaseI HS (Rel 2), LICR Histone (Rel 3), CSHL Long RNA-seq (Rel 3)
</b></font>
<p>
One new track and three track updates were released on the Mouse mm9 browser.
<a href="newsarch.html#090512">Read more</a>.</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<a name="090512">
<font size="3"><b> 5 September 2012 - ENCODE results published in Nature, Science and other journals</b></font>
<p>
The results of the ENCODE project were published today in a coordinated set of 30 papers published in multiple journals.
<a href="newsarch.html#090512">Read more</a>.</p>
<!-- end news item -->

<!-- end archived item section--> </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="Conditions"></a><font
 size="4"><b>&nbsp; Conditions of Use</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The sequence and annotation data
displayed in the Genome Browser are freely available for academic,
nonprofit, and personal use with the following conditions: </p>
                              <ul>
                                <li>ENCODE data are covered by the <a
 href="terms.html">ENCODE Consortium Data Release Policy</a>. </li>
                                <li>The general <a
 href="../#Conditions">Conditions of Use</a> for
the UCSC Genome Browser apply. </li>
                              </ul>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
</body>
</html>
