
<html>
<head>
  <meta http-equiv="Content-Type"
 content="text/html; charset=ISO-8859-1">
  <meta name="description" content="ENCODE Project at UCSC">
  <meta name="keywords"
 content="ENCODE, project, ENCODE project, genome, genome browser, human genome, Blat, UCSC, bioinformatics, gene prediction, SNP, EST, mRNA, NHGRI, NISC">
  <title>ENCODE Project at UCSC</title>
  <link rel="STYLESHEET" href="../style/HGStyle.css">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
 <script type="text/javascript" src="../js/jquery.js"></script>
 <script type="text/javascript" src="../js/staticDoc.js"></script>
</head>
<body style="font-family: arial, Geneva, Helvetica, sans-serif;" alink="#6600ff"
 link="#0000cc" vlink="#330066">
<a name="TOP"></a><!-- Header Images and links -->
<table border="0" cellpadding="0" cellspacing="0" width="100%">
  <tbody>
    <tr>
      <th colspan="2" align="left">
<img src="../images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" height="75"> &nbsp;&nbsp;
 <font size="6" face="arial, Geneva, Helvetica, sans-serif"><font color="004D99">
 Encyclopedia of DNA Elements</font>
       </th>
    </tr>
    <tr height="100%" valign="top">
      <td rowspan="1" height="100%" valign="top"><!-- outer table is to align leftbar & content boxes at top of page -->
      <table border="0" cellpadding="0" cellspacing="0" width="100%">
        <tbody>
          <tr>
            <td><img src="../images/DOT.gif" alt="" height="7"></td>
          </tr>
          <tr height="100%">
            <td height="100%"><!--outer table is for black border-->
            <table bgcolor="#004D99" border="0" cellpadding="1" cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table border="0" cellpadding="10" cellspacing="0" width="100%">
                    <tbody>
                      <tr height="100%">
                        <td bgcolor="#0004D99" height="100%" valign="top">
                        <font color="#FFCC00"><b><em>Human</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
                        <a class="leftbar" href="/ENCODE/dataMatrix/encodeDataMatrixHuman.html">Experiment Matrix</a>
                        <hr><a class="leftbar" href="/ENCODE/dataSummary.html" title="formerly Data Summary">Experiment List</a>
                        <hr><a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloads.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=hg19">Genome Browser (hg19)</a>
                        <hr><a class="leftbar" href="http://genome-preview.ucsc.edu/cgi-bin/hgTracks?db=hg19">Preview Browser (hg19)</a>
                        <hr><a class="leftbar" href="http://genomewiki.cse.ucsc.edu/index.php/Genome_Browser_Session_Gallery"
                        TARGET=_blank>Session Gallery</a>
                        <hr><a class="leftbar" href="cellTypes.html">Cell Types</a>
                        <hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
                        <font color="#FFCC00"><b><em>Mouse</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8" width="0"><br>
                        <a class="leftbar" href="/ENCODE/dataMatrix/encodeDataMatrixMouse.html">Experiment Matrix</a><font color="#FF0000"><sup><strong><em>New</strong></em></sup></font>
                        <hr><a class="leftbar" href="/ENCODE/dataSummaryMouse.html"title="formerly Data Summary">Experiment List</a>
                        <hr><a class="leftbar" href="search.html">Search</a>
                        <hr><a class="leftbar" href="downloadsMouse.html">Downloads</a>
                        <hr><a class="leftbar" href="/cgi-bin/hgTracks?db=mm9">Genome Browser (mm9)</a>
                        <hr><a class="leftbar" href="http://genome-preview.ucsc.edu/cgi-bin/hgTracks?db=mm9">Preview Browser (mm9)</a>
                        <hr><a class="leftbar" href="cellTypesMouse.html">Cell Types</a>
                        <hr>
                        <img src="../images/DOT.gif" alt="" height="5" width="0"><br>
                        <font color="FFCC00"><b><em>General</em></b></font><br>
                        <img src="../images/DOT.gif" alt="" height="8"><br>
                        <a class="leftbar" href="otherTerms.html">Registered Variables</a>
                        <hr><a class="leftbar" href="antibodies.html">Antibodies</a>
                        <hr><a class="leftbar" href="usageResources.html">Education and Outreach</a>
                        <hr><a class="leftbar" href="releaseLog.html">Release Log</a>
                        <hr><a class="leftbar" href="terms.html">Data Policy</a>
                        <hr><a class="leftbar" href="dataStandards.html">Data Standards</a>
                        <hr><a class="leftbar" href="/FAQ/FAQformat.html#ENCODE">File Formats</a>
                        <hr><a class="leftbar" href="softwareTools.html">Software Tools</a><font color="#FF0000"><sup><strong><em>New</strong></em></sup></font>
                        <hr><a class="leftbar" href="pubs.html">Publications</a>
                        <hr><a class="leftbar" href="contributors.html">Contributors</a>
                        <hr><a class="leftbar" href="pilot.html">Pilot Project</a>
                        <hr><a class="leftbar" href="http://www.cbse.ucsc.edu/jobs/" target="_blank">Jobs</a>
                        <hr><a class="leftbar" href="contacts.html">Contact Us</a>
                        <hr><br>
                        <img src="../images/DOT.gif" alt="" height="970" width="1"> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
<!-- Content Tables --> <td valign="top">
      <table border="0" cellpadding="5">
        <tbody>
          <tr>
            <td><!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="About"></a><font size="4"><b>&nbsp;
About ENCODE Data</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="15"><br>
                              </td>
                              <td>
<p>The <a href="http://www.genome.gov/10005107" target="_blank">Encyclopedia of DNA Elements</a>
(ENCODE) Consortium is an international collaboration
of research groups funded by the National Human Genome Research
Institute (<a href="http://www.genome.gov/" target="_blank">NHGRI</a>).
The goal of ENCODE is to build a comprehensive parts list of
functional elements in the human genome, including elements that
act at the protein and RNA levels,
and regulatory elements that control cells and
circumstances in which a gene is active.
</p>
<!--#if expr='"$SERVER_NAME" = /hgwdev.*.cse.ucsc.edu/ || "$SERVER_NAME" = "genome-test.cse.ucsc.edu"'-->
<p>
<b>WARNING: This is a test site.</b>
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  
For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<!--#if expr='"$SERVER_NAME" = /genome-preview.*.ucsc.edu/' -->
<p>
<b>WARNING: This is our preview site.</b>
This website is a weekly mirror of our internal 
development server for public access.  
Data and tools here are under 
construction, have not been quality reviewed, and are subject to change 
at any time.  We provide this site for early access, with the warning that it is
less available and stable than our public site. For 
high-quality reviewed annotations on our production server, visit
<a HREF="http://encodeproject.org">http://encodeproject.org</a>.</p>
<!--#endif -->

<table><tr>
<td><a href="/ENCODE/aboutScaleup.html" target="_blank"><img src="../images/EncodeNatureGraphic.png" alt="" height="200">
<em><font size=-2>Click to enlarge</em></font></td>

<td>
ENCODE data are now available for the entire human genome.
<b><em>All ENCODE data are free and available for immediate use via </em></b>:
<ul class="simple">
<li><a href="/ENCODE/search.html" target="_blank">Search</a> for displayable tracks and downloadable files
</li>
<li><a href="/ENCODE/downloads.html" target="_blank">Download</a> of data files
</li>
<li><a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Kate&hgS_otherUserSessionName=encodePortalSession"
target="_blank">Visualization</a> in the UCSC Genome Browser (ENCODE data marked with
the <img src="../images/nhgri.gif" alt="" height="16" width="16"> NHGRI logo)
</li>
<li><a href="/cgi-bin/hgTables?db=hg18&hgta_group=regulation&hgta_track=wgEncodeHudsonalphaChipSeq"
target="_blank">Data mining</a> with the UCSC Table Browser and other
<a href="../">UCSC Genome Bioinformatics tools</a>
</li>
</ul>
</td>

</tr></table>
</p>
<p>
To search for ENCODE data related to your area of interest and set up a browser view, use the UCSC
<a href="dataMatrix/encodeDataMatrixHuman.html" target="_blank">Experiment Matrix</a> or
<a href="/cgi-bin/hgTracks?hgt_tSearch=1" target="_blank">Track Search tool</a> (<em>Advanced</em> features).
The <a href="dataSummary.html" target="_blank">Experiment List (Human)</a> and 
 <a href="dataSummaryMouse.html" target="_blank">Experiment List (Mouse)</a> links provide
comprehensive listings of ENCODE data that is released or in preparation.
Early access to pre-release ENCODE data is provided at
<a href="http://genome-preview.ucsc.edu" target="_blank">http://genome-preview.ucsc.edu</a>.
If you would like to receive notifications of ENCODE data releases and related news by email,
subscribe to the
<a href="https://lists.soe.ucsc.edu/mailman/listinfo/encode-announce" target="_blank">encode-announce mailing list</a>.
For more information about how to access this data, see the free online
<a target=="_blank" href="http://www.openhelix.com/ENCODE/" target="_blank">OpenHelix ENCODE tutorial</a>.
</p>
<p>
All ENCODE data is freely available for download and analysis.
However, before publishing research that uses ENCODE data,
please read the <a href="terms.html" target="_blank">ENCODE Data Release Policy</a>,
which places some restrictions on publication use of data for nine
months following data release. &nbsp;&nbsp;
<a href="/ENCODE/aboutScaleup.html">Read more</a> about ENCODE data at UCSC.</p>

                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <br>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><!--cheap trick to get background on row to be continuous-->
                              <table border="0" cellpadding="0" cellspacing="0" width="100%">
                                <tbody>
                                  <tr>
                                    <td align="left"><font size="4"><a name="TOC"></a><b>&nbsp; News</b></font> </td>
                                    <td align="right"><a href="newsarch.html"><img src="../images/news.jpg" alt="" align="right" border="0"></a> </td>
                                    <td align="right">&nbsp;</td>
                                  </tr>
                                </tbody>
                              </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0" cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="15"><br>
                              </td>
                              <td style="font-family: arial, Geneva, Helvetica, sans-serif;">
<!-- start news item -->
<p>
<font size="3"><b> 25 May 2012 -  Mouse ENCODE data releases: Caltech RNA-seq, PSU Histone, LICR RNA-seq (Release 2)
</b></font>
<p>
Two new tracks and one track update were released on the mm9 genome browser:
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCaltechRnaSeq">
RNA-seq from ENCODE/Caltech</a>:
This track shows transcriptome measurements in 10T1/2 fibroblasts and C2C12 myoblasts, 
and these same cell types treated with differentiation media to produce fibrocytes and myocytes.
RNA-seq was performed on total cellular polyA+ RNA with 100bp paired end reads aligned to the genome.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuHistone">
Histone Modifications by ChIP-seq from ENCODE/PSU</a>:
This track displays levels of three histone modifications - H3K4me1 (indicative of active chromatin and enhancers), H3K4me3 (enriched at active promoters), and H3K27me3 (associated with some silenced genes), in 5 blood-derived primary cells and cell lines.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrRnaSeq">
RNA-seq from ENCODE/LICR (Release 2)</a>:
<p>
This release adds RNA-seq data for 12 additional primary cell types, bringing the total for this track to 22. 
</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 8 May 2012 - ENCODE Experiment Matrix web tool
</b></font>
<p>
The ENCODE DCC at UCSC is pleased to announce the release of a new
web tool for accessing ENCODE data.  
The new 
<a class="euro" target=_blank href="dataMatrix/encodeDataMatrixHuman.html"> Experiment Matrix</a> 
link on the ENCODE portal
leads to a set of three web pages that provide an up-to-date view of the
breadth of human ENCODE data available, along with an interface for
selecting experiments for viewing in the browser or downloading as
files for analysis. (Note: the related 
<a class="euro" target=_blank href="dataSummary.html"> Experiment List</a> 
(previously 'Data Summary') spreadsheets have been updated to reflect newer status).
</p>
<p>
The main Experiment Matrix page shows the number of experiments for
each cell type/assay pairing. The related
<a class="euro" target=_blank href="dataMatrix/encodeChipMatrixHuman.html">
ChIP-seq Experiment Matrix</a>
provides a view of the transcription factor and histone modification
datasets, showing experiments by cell type and antibody target. The
<a class="euro" target=_blank href="dataMatrix/encodeDataSummaryHuman.html">
Experiment Summary</a>
lists the number of experiments by assay type
alone and includes annotations that are cell-type independent
(annotations on the reference genome). Each page has a file/track
selector.  Clicking an experiment item produces a search window with
resulting files or tracks listed.
</p>
<p>
These pages are best viewed in FireFox, Chrome, and Safari browsers, and
zoomed out as far as readability on your screen allows.
<p>
Special thanks to Katrina Learned and Steve Heitner for careful QA review and testing, 
to the Genome Browser build & release group for shepherding the new program
to the public site, and to the DCC wranglers for the high quality
metadata curation that supports this new access tool.
</p>
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 20 Apr 2012 -  Human ENCODE data releases: UMass 5C, ENCODE Pilot Regions, FSU Repli-chip and 11 track updates
</b></font>
<p>
Three new tracks and 11 track updates were released on the hg19 genome browser:
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUmassDekker5C">
Chromatin Interactions by 5C from ENCODE/Univ. Mass(Dekker)</a>:
This track contains chromatin interaction data generated using the 5C 
(Chromatin Conformation Capture Carbon Copy) method by the ENCODE group 
(Dekker Lab) at the University of Massachusetts. The track shows 
significant looping interactions between transcriptional start sites 
(TSS) and distal regulatory elements in the context of the 44 ENCODE 
pilot regions spanning 1% of the human genome.
<p>
<a class="euro"href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=encodeRegions">
ENCODE Pilot Regions</a>:
This track depicts the 44 target regions covering 1% of the human genome 
defined for the ENCODE pilot project.  The hg19 coordinates for these 
regions were obtained using liftOver from the hg18 track.
<p>
<a class=euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeFsuRepliChip">
Replication Timing by Repli-chip from ENCODE/FSU</a>:
This track shows genome-wide assessment of DNA replication timing in 7 cell types (3 stem cell, HeLa, IMR90, lymphoblastoid and BG02ES) using NimbleGen tiling CGH microarrays. 
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV11">
Gene Annotations from ENCODE/GENCODE Version 11</a>:
This version features 79% coverage of genes by manual annotation.  
It is based on the Ensembl 66 (Feb 2012) 
and Vega 46 (Jan 2012) gene predictions and curation. 
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibTfbs">
Transcription Factor Binding Sites by ChIP-seq from ENCODE/HAIB (Release 2)</a>:
This release contains 68 new experiments, including 1 new cell line (HUVEC) and 20 new antibodies.
There are also data corrections and new validation datasets (shRNA knockdown). 
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs">
CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington (Release 3)</a>:
This release contains 3 additional experiments in H7ES, NHLF, and HFF cell types. 
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDnase">
DNaseI Hypersensitivity by Digital DNaseI from ENCODE/University of Washington (Release 4)</a>:
This release contains 11 new experiments across 9 new cell types. 
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeBroadHistone">
Histone Modifications by ChIP-seq from ENCODE/Broad Institute (Release 2)</a>:
This release contains 64 new experiments including 1 new cell line 
(Dnd41) and 10 new antibodies, and 3 datasets have additional data.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromDnase">
Open Chromatin by DNaseI HS from ENCODE/Duke (Release 2)</a>:
This release provides data in additional cell lines (Colo29, CD4+, 
ECC-1, H7ES, HEK293T, 3 IPS lines derived from fibroblasts, MCF-7, 
Medullo, RWPE1, T-47D), higher-quality data for the K562 cell line, and 
additional data corrections and replacements.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeDukeAffyExon">
Affymetrix Exon Array from ENCODE/Duke (Release 3)</a>:
This track displays human tissue microarray data in 56 cell lines. The 
RNA extracted from these cells were also analyzed by DNaseI 
hypersensitivity (Duke DNaseI HS track), FAIRE (UNC FAIRE track) and 
ChIP-seq (UTA TFBS) by the ENCODE OpenChromatin group.
<p>
<a class=euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq">
RNA-seq from ENCODE/Caltech (Release 3)</a>:
Signal files for all experiments have been replaced with more accurate data.  Earlier versions can still be accessed for download only.
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 29 Mar 2012 -  Mouse ENCODE data releases: PSU TFBS, CSHL Long RNA-seq, LICR Histone (Rel 2), LICR TFBS (Rel 2) 
</b></font>
<p>
Two new tracks and 2 track updates were released on the mm9 genome browser:
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodePsuTfbs">
Transcription Factor Binding Sites by ChIP-seq from ENCODE/PSU</a>:
This track shows occupancy of genomic DNA by 5 transcription factors 
(CTCF, GATA1, GATA2, PAX5, TAL1) and Pol2 as determined by ChIP-seq in 6 
erythroid and other blood cell types (CH12, MEL, G1E, megakaryocyte, 
erythroblast, G1E-ER4).
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeCshlLongRnaSeq">
Long RNA-seq from ENCODE/Cold Spring Harbor Lab</a>:
This track contains information about mouse RNAs (> 200 nucleotides in length) in 17 different tissues, as obtained by Illumina squencing.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrHistone">
Histone Modifications by ChIP-seq from ENCODE/LICR (Release 2)</a>:
<br>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm9&g=wgEncodeLicrTfbs">
Transcription Factor Binding Sites by ChIP-seq from ENCODE/LICR (Release 2)</a>:
<p>
These two companion tracks survey <em>cis</em>-regulatory elements in the mouse genome.  ChIP-seq was used to localize binding of Pol2, CTCF, and p300 factors and to profile 7 chromatin modification in 20 different mouse (C57Bl/6) tissues, primary cells, and cell lines.   Release 2 of these tracks adds 87 new experiments.
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 24 Feb 2012 -  Human ENCODE data releases: UW Repli-seq, HAIB RNA-seq, and 7 track updates
</b></font>
<p>
Two new tracks and 7 track updates were released on the hg19 genome browser:
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwRepliSeq">
Replication Timing by Repli-seq from ENCODE/University of Washington</a>:
This track shows genome-wide assessment of DNA replication timing in 15 
cell lines as identified by the sequencing-based "Repli-seq" method. 
Replication timing is known to be an important feature for  epigenetic 
control of gene expression that usually operates at a  higher-order 
level than at the level of specific genes.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibRnaSeq">
RNA-seq from ENCODE/HAIB</a>:
This track displays RNA-seq alignments and graphs of signal enrichment 
for 9 cell lines, in various treatment protocols.  Estimates of 
transcript abundance are provided for download.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeGencodeV10">
Gene Annotations from ENCODE/GENCODE Version 10</a>:
This version of the GENCODE Genes track features 77% coverage of 
genes by manual annotation.  It corresponds to the Ensembl 65 (Dec 2011) 
and Vega 45 (Oct 2011) gene predictions and curation. 
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwDgf">
DNaseI Digital Genomic Footprinting from ENCODE/University of Washington (Release 3)</a>:
Release 3 includes experiments in 19 additional cell types.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwHistone">
Histone Modifications by ChIP-seq from ENCODE/University of Washington (Release 3)</a>:
This release adds the H7-hESC cell type and another replicate for the GM12865 experiment.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeUwTfbs">
CTCF Binding Sites by ChIP-seq from ENCODE/University of Washington (Release 2)</a>:
Release 2 includes experiments in 15 additional cell lines.
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeOpenChromSynth">
DNaseI/FAIRE/ChIP Synthesis from ENCODE/OpenChrom(Duke/UNC/UTA) (Release 2)</a>:
Release 2 adds synthesis analysis for 10 cell types:
Gliobla, GM12891, GM12892, GM18507, GM19239, HeLa-S3/IFNa4h, HTR8svn, 
Medullo, PanIslets, Urothelia
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCshlShortRnaSeq">
Small RNA-seq from ENCODE/Cold Spring Harbor Lab (Release 2)</a>:
This release adds 10 new cell lines (A549, AG04450, BJ, H1-hESC, 
HeLa-S3, HepG2, HUVEC, MCF-7, NHEK, and SK-N-SH_RA), a new displayed 
view (Contigs), and two new download files (GENCODE Predicted Exons and 
a Protocol Document).
<p>
<a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeCaltechRnaSeq">
RNA-seq from ENCODE/Caltech (Release 2)</a>:
Data was reprocessed with a newer mapping algorithm (labeled V2). 
Additional download files of gene, transcript, and exon expression 
levels are now provided.
<p>
<a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeHaibGenotype">
Genotype (CNV and SNP) by Illumina 1MDuo and CBS from ENCODE/HudsonAlpha (Release 2)</a>:
This release contains data corrections to GM12878 and astrocyte experiments.
<hr>
<!-- end news item -->

<!-- start news item -->
<p>
<font size="3"><b> 5 Jan 2012 - Human ENCODE data releases: UW Histone, HAIB Methyl450, RIKEN CAGE Loc, UW Affy Exon (Release 2)
</b></font>
<p>
Three new tracks and one track update were released on the hg19 genome browser:
<a href="newsarch.html#010512">Read more</a>.</p>
<br>
<!-- end news item -->

<!-- end archived item section--> </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
<!--outer table is for border purposes-->
            <table bgcolor="#888888" border="0" cellpadding="1"
 cellspacing="0" width="100%">
              <tbody>
                <tr>
                  <td>
                  <table bgcolor="#FFFFFF" border="0" cellpadding="0"
 cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td>
                        <table background="../images/hr.gif"
 bgcolor="#FFCC00" border="0" width="100%">
                          <tbody>
                            <tr>
                              <td><a name="Conditions"></a><font
 size="4"><b>&nbsp; Conditions of Use</b></font> </td>
                            </tr>
                          </tbody>
                        </table>
                        <table bgcolor="#FFFFFF" border="0"
 cellpadding="0" width="100%">
                          <tbody>
                            <tr>
                              <th height="3"><br>
                              </th>
                              <td><br>
                              </td>
                              <td><br>
                              </td>
                            </tr>
                            <tr>
                              <td width="10"><br>
                              </td>
                              <td>
                              <p>The sequence and annotation data
displayed in the Genome Browser are freely available for academic,
nonprofit, and personal use with the following conditions: </p>
                              <ul>
                                <li>ENCODE data are covered by the <a
 href="terms.html">ENCODE Consortium Data Release Policy</a>. </li>
                                <li>The general <a
 href="../#Conditions">Conditions of Use</a> for
the UCSC Genome Browser apply. </li>
                              </ul>
                              </td>
                              <td width="15"><br>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
</body>
</html>
