<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
        <title>Publications from non-ENCODE Authors</title>
        <link rel="stylesheet" href="/style/HGStyle.css" type="text/css">
        <link rel="stylesheet" href="/style/encodeProject.css" type="text/css">
    </head>

    <body>
        <!--#include virtual="/inc/encodeProject.topbar.html"-->

        <div class="encodeHeader">
            <a href="http://www.genome.gov/10005107"><img src="/images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" title="ENCODE Project at NHGRI"></a>
            <span class="txt">Publications from non-ENCODE Authors</span>
        </div>
        <div class="wrapper">
            <div class="bar"><h4 class="title">Contents</h4></div>
            <div class="content">
            <p>
            This page lists papers that reference ENCODE data, by authors not funded by ENCODE.
            The ENCODE project tracks these papers in part to assess impact of the resource.
            Please contact 
            <a href="mailto:&#112;&#97;&#122;&#105;&#110;&#109;@&#109;&#97;&#105;&#108;.&#110;&#105;&#104;.&#103;&#111;&#118;">Mike Pazin at NHGRI</a> 
            to suggest publications to add to this list.
           <ul>
               <li><a href='#disease'>Human disease publications</a></li>
               <li><a href='#biology'>Basic biology publications</a></li>
               <li><a href='#tools'>Tools, methods, databases and review/commentary publications</a></li>
           </ul>
           </p>
           <p class="date">Updated 24 August 2012</p>
            </div><!--end content-->
        </div><!--end wrapper-->

        <div class="wrapper">
            <div class="bar">
                <a name="disease"></a> 
                <h4 class="title">Human disease publications</h4>
            </div>
            <div class="content">
                <p>
                Adrianto I, Wen F, Templeton A, Wiley G, King JB, Lessard CJ, Bates JS, Hu Y, Kelly JA, Kaufman KM
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21336280" target="_blank">
                Association of a functional variant downstream of TNFAIP3 with systemic lupus erythematosus</a>.
                <em>Nat Genet</em>. 2011 Mar;43(3):253-8.<br>
                ENCODE data presented in Figure S5, other resources also used
                </p>

                <p>
                Bekris LM, Lutz F, Yu CE.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22089642" target="_blank">
                Functional analysis of APOE locus genetic variation implicates regional enhancers in the regulation
                of both TOMM40 and APOE</a>.
                <em>J Hum Genet</em>. 2012 Jan;57(1):18-25.<br>
                ENCODE data in Figure 2a
                </p>

                <p>
                Calva D, Dahdaleh FS, Woodfield G, Weigel RJ, Carr JC, Chinnathambi S, Howe JR.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21421563" target="_blank">
                Discovery of SMAD4 promoters, transcription factor binding sites and deletions in juvenile polyposis
                patients</a>.
                <em>Nucleic Acids Res</em>. 2011 Jul;39(13):5369-78.<br>
                ENCODE data in Figure 6
                </p>

                <p>
                Carvajal-Carmona LG, Cazier JB, Jones AM, Howarth K, Broderick P, Pittman A, Dobbins S, Tenesa A,
                Farrington S, Prendergast J <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21531788" target="_blank">
                Fine-mapping of colorectal cancer susceptibility loci at 8q23.3, 16q22.1 and 19q13.11: refinement of
                association signals and use of in silico analysis to suggest functional variation and unexpected
                candidate target genes</a>.
                <em>Hum Mol Genet</em>. 2011 Jul 15;20(14):2879-88.<br>
                Some regions mapped to ENCODE annotations (Table 1)
                </p>

                <p>
                Crapoulet N, O'Brien P, Ouellette RJ, Robichaud GA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21707507" target="_blank">
                Coordinated expression of Pax-5 and FAK1 in metastasis</a>.
                <em>Anticancer Agents Med Chem</em>. 2011 Sep;11(7):643-9.<br>
                ENCODE data in Figure 1, text
                </p>

                <p>
                Davison LJ, Wallace C, Cooper JD, Cope NF, Wilson NK, Smyth DJ, Howson JM, Saleh N, Al-Jeffery A,
                Angus KL <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21989056" target="_blank">
                Long-range DNA looping and gene expression analyses identify DEXI as an autoimmune disease candidate
                gene</a>.
                <em>Hum Mol Genet</em>. 2012 Jan 15;21(2):322-33.<br>
                ENCODE data in Figure 3, and S3A; found enhancer in one gene that regulates
                distal, neighboring gene
                </p>

                <p>
                Farrell JJ, Sherva RM, Chen ZY, Luo HY, Chu BF, Ha SY, Li CK, Lee AC, Li RC, Li CK <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21385855" target="_blank">
                A 3-bp deletion in the HBS1L-MYB intergenic region on chromosome 6q23 is associated with HbF
                expression</a>.
                <em>Blood</em>. 2011 May 5;117(18):4935-45.<br>
                ENCODE data in Figure 6
                </p>

                <p>
                Fu YP, Kohaar I, Rothman N, Earl J, Figueroa JD, Ye Y, Malats N, Tang W, Liu L, Garcia-Closas M
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22416122" target="_blank">
                Common genetic variants in the PSCA gene influence gene expression and bladder cancer risk</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2012 Mar 27;109(13):4974-9.<br>
                ENCODE data in Figure 2, H3K4me1, H3K4me3
                </p>

                <p>
                Ghedira K, Hornischer K, Konovalova T, Jenhani AZ, Benkahla A, Kel A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21093613" target="_blank">
                Identification of key mechanisms controlling gene expression in Leishmania infected macrophages
                using genome-wide promoter analysis</a>.
                <em>Infect Genet Evol</em>. 2011 Jun;11(4):769-77.<br>
                ENCODE data and other resources used
                </p>

                <p>
                Glinskii AB, Ma S, Ma J, Grant D, Lim CU, Guest I, Sell S, Buttyan R, Glinsky GV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22067658" target="_blank">
                Networks of intergenic long-range enhancers and snpRNAs drive castration-resistant phenotype of
                prostate cancer and contribute to pathogenesis of multiple common human disorders</a>.
                <em>Cell Cycle</em>. 2011 Oct 15;10(20):3571-97.<br>
                ENCODE data in Fig 4A, 4D, S4
                </p>

                <p>
                Han J, Park SG, Bae JB, Choi J, Lyu JM, Park SH, Kim HS, Kim YJ, Kim S, Kim TY.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22564738" target="_blank">
                The characteristics of genome-wide DNA methylation in na&#239;ve CD4+ T cells of patients with
                psoriasis or atopic dermatitis</a>.
                <em>Biochem Biophys Res Commun</em>. 2012 May 25;422(1):157-63.<br>
                ENCODE Txn, His, DNase, and TF composite tracks, Figure 1, S1-S6
                </p>

                <p>
                Harismendy O, Bansal V, Bhatia G, Nakano M, Scott M, Wang X, Dib C, Turlotte E, Sipe JC, Murray SS
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21118518" target="_blank">
                Population sequencing of two endocannabinoid metabolic genes identifies rare and common regulatory
                variants associated with extreme obesity and metabolite level</a>.
                <em>Genome Biol</em>. 2010;11(11):R118.<br>
                ENCODE histone modifications, Fig 5C, TFs, Fig 5D
                </p>

                <p>
                Harismendy O, Notani D, Song X, Rahim NG, Tanasa B, Heintzman N, Ren B, Fu XD, Topol EJ, Rosenfeld
                MG <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21307941" target="_blank">
                9p21 DNA variants associated with coronary artery disease impair interferon-&#947; signalling
                response</a>.
                <em>Nature</em>. 2011 Feb 10;470(7333):264-8.<br>
                ENCODE histone modifications, TFs in Figure 1
                </p>

                <p>
                Higareda-Almaraz JC, Enr&#237;quez-Gasca Mdel R, Hern&#225;ndez-Ortiz M, Resendis-Antonio O,
                Encarnaci&#243;n-Guevara S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21696634" target="_blank">
                Proteomic patterns of cervical cancer cell lines, a network perspective</a>.
                <em>BMC Syst Biol</em>. 2011 Jun 22;5:96.<br>
                ENCODE data in Figure 1, Table 4
                </p>

                <p>
                Jaeger E, Leedham S, Lewis A, Segditsas S, Becker M, Cuadrado PR, Davis H, Kaur K, Heinimann K,
                Howarth K <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22561515" target="_blank">
                Hereditary mixed polyposis syndrome is caused by a 40-kb upstream duplication that leads to
                increased and ectopic expression of the BMP antagonist GREM1</a>.
                <em>Nat Genet</em>. 2012 May 6;44(6):699-703.<br>
                ENCODE data in Figure S3, Histone modifications, DNase, TFBS
                </p>

                <p>
                Lamina C, Coassin S, Illig T, Kronenberg F.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21945496" target="_blank">
                Look beyond one's own nose: combination of information from publicly available sources reveals an
                association of GATA4 polymorphisms with plasma triglycerides</a>.
                <em>Atherosclerosis</em>. 2011 Dec;219(2):698-703.<br>
                ENCODE histone modifications, TFs, DNase, Figure 2, S1, S2
                </p>

                <p>
                Lessard CJ, Adrianto I, Kelly JA, Kaufman KM, Grundahl KM, Adler A, Williams AH, Gallant CJ,
                Marta&#160;E.&#160;Alarc&#243;n-Riquelme on behalf of the BIOLUPUS and GENLES Networks, Anaya JM
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21194677" target="_blank">
                Identification of a systemic lupus erythematosus susceptibility locus at 11p13 between PDHX and CD44
                in a multiethnic study</a>.
                <em>Am J Hum Genet</em>. 2011 Jan 7;88(1):83-91.<br>
                ENCODE TFs, histone modifications, Figure 2C, text
                </p>

                <p>
                Li YJ, Minear MA, Rimmler J, Zhao B, Balajonda E, Hauser MA, Allingham RR, Eghrari AO, Riazuddin SA,
                Katsanis N <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21533127" target="_blank">
                Replication of TCF4 through association and linkage studies in late-onset Fuchs endothelial corneal
                dystrophy</a>.
                <em>PLoS One</em>. 2011 Apr 20;6(4):e18044.<br>
                ENCODE data mentioned in text
                </p>

                <p>
                Lu Y, Zhang Z, Yu H, Zheng SL, Isaacs WB, Xu J, Sun J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21541972" target="_blank">
                Functional annotation of risk loci identified through genome-wide association studies for prostate
                cancer</a>.
                <em>Prostate</em>. 2011 Jun 15;71(9):955-63.<br>
                ENCODE data in Table 1, 2, etc.
                </p>

                <p>
                Maruo S, Zhao B, Johannsen E, Kieff E, Zou J, Takada K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21245331" target="_blank">
                Epstein-Barr virus nuclear antigens 3C and 3A maintain lymphoblastoid cell growth by repressing
                p16INK4A and p14ARF expression</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Feb 1;108(5):1919-24.<br>
                ENCODE data in Figure S2
                </p>

                <p>
                Middelberg RP, Benyamin B, de Moor MH, Warrington NM, Gordon S, Henders AK, Medland SE, Nyholt DR,
                de Geus EJ, Hottenga JJ <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22010049" target="_blank">
                Loci affecting gamma-glutamyl transferase in adults and adolescents show age &#215; SNP interaction
                and cardiometabolic disease associations</a>.
                <em>Hum Mol Genet</em>. 2012 Jan 15;21(2):446-55.<br>
                ENCODE data described in text
                </p>

                <p>
                Ramagopalan SV, Heger A, Berlanga AJ, Maugeri NJ, Lincoln MR, Burrell A, Handunnetthi L, Handel AE,
                Disanto G, Orton SM <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20736230" target="_blank">
                A ChIP-seq defined genome-wide map of vitamin D receptor binding: associations with disease and
                evolution</a>.
                <em>Genome Res</em>. 2010 Oct;20(10):1352-60.<br>
                ENCODE data in Figure 5, S3
                </p>

                <p>
                Rushefski M, Aplenc R, Meyer N, Li M, Feng R, Lanken PN, Gallop R, Bellamy S, Localio AR, Feinstein
                SI <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21627785" target="_blank">
                Novel variants in the PRDX6 Gene and the risk of Acute Lung Injury following major trauma</a>.
                <em>BMC Med Genet</em>. 2011 May 31;12:77.<br>
                ENCODE data used, negative result
                </p>

                <p>
                Sankaran VG, Menne TF, &#352;&#263;epanovi&#263; D, Vergilio JA, Ji P, Kim J, Thiru P, Orkin SH,
                Lander ES, Lodish HF.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21205891" target="_blank">
                MicroRNA-15a and -16-1 act via MYB to elevate fetal hemoglobin expression in human trisomy 13</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Jan 25;108(4):1519-24.<br>
                ENCODE data in Table S2
                </p>

                <p>
                Stitzel ML, Sethupathy P, Pearson DS, Chines PS, Song L, Erdos MR, Welch R, Parker SC, Boyle AP,
                Scott LJ <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21035756" target="_blank">
                Global epigenomic analysis of primary human pancreatic islets provides insights into type 2 diabetes
                susceptibility loci</a>.
                <em>Cell Metab</em>. 2010 Nov 3;12(5):443-55.<br>
                ENCODE data used in Figure 1, 2, 4, Table S3
                </p>

                <p>
                Szymanski M, Wang R, Bassett SS, Avramopoulos D.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22832520" target="_blank">
                Alzheimer's risk variants in the clusterin gene are associated with alternative splicing</a>.
                <em>Transl Psychiatry</em>. 2011 Jul 5;1:e18.<br>
                ENCODE data in Figure 1b
                </p>

                <p>
                Teng L, Firpi HA, Tan K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21685454" target="_blank">
                Enhancers in embryonic stem cells are enriched for transposable elements and genetic variations
                associated with cancers</a>.
                <em>Nucleic Acids Res</em>. 2011 Sep 1;39(17):7371-9.<br>
                ENCODE data analyzed
                </p>

                <p>
                Thakur RK, Yadav VK, Kumar P, Chowdhury S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21556888" target="_blank">
                Mechanisms of non-metastatic 2 (NME2)-mediated control of metastasis across tumor types</a>.
                <em>Naunyn Schmiedebergs Arch Pharmacol</em>. 2011 Oct;384(4-5):397-406.<br>
                ENCODE data in Figure 3
                </p>

                <p>
                Trynka G, Hunt KA, Bockett NA, Romanos J, Mistry V, Szperl A, Bakker SF, Bardella MT, Bhaw-Rosun L,
                Castillejo G <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22057235" target="_blank">
                Dense genotyping identifies and localizes multiple common and rare variant association signals in
                celiac disease</a>.
                <em>Nat Genet</em>. 2011 Nov 6;43(12):1193-201.<br>
                ENCODE data in discussion
                </p>

                <p>
                Verlaan DJ, Berlivet S, Hunninghake GM, Madore AM, Larivi&#232;re M, Moussette S, Grundberg E, Kwan
                T, Ouimet M, Ge B <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19732864" target="_blank">
                Allele-specific chromatin remodeling in the ZPBP2/GSDMB/ORMDL3 locus associated with the risk of
                asthma and autoimmune disease</a>.
                <em>Am J Hum Genet</em>. 2009 Sep;85(3):377-93.<br>
                ENCODE data in Figure S4
                </p>

                <p>
                Visser M, Kayser M, Palstra RJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22234890" target="_blank">
                HERC2 rs12913832 modulates human pigmentation by attenuating chromatin-loop formation between a
                long-range enhancer and the OCA2 promoter</a>.
                <em>Genome Res</em>. 2012 Mar;22(3):446-55.<br>
                ENCODE data in Figure S4b
                </p>

                <p>
                Wang K, Li M, Hakonarson H.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20601685" target="_blank">
                ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data</a>.
                <em>Nucleic Acids Res</em>. 2010 Sep;38(16):e164.<br>
                Web site says GENCODE genes, DNAse I hypersensitivity sites, ENCODE H3K4Me1, H3K4Me3, H3K27Ac,
                CTCF sites, ChIP-Seq peaks, RNA-Seq peaks, or many other annotations on genomic intervals
                </p>

                <p>
                Zhang X, Cowper-Sal Lari R, Bailey SD, Moore JH, Lupien M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22665440" target="_blank">
                Integrative functional genomics identifies an enhancer looping to the SOX9 gene disrupted by the
                17q24.3 prostate cancer risk locus</a>.
                <em>Genome Res</em>. 2012 Aug;22(8):1437-46.<br>
                ENCODE data in Figure 1, histone modifications and DNase
                </p>

                <p>
                Zhao B, Zou J, Wang H, Johannsen E, Peng CW, Quackenbush J, Mar JC, Morton CC, Freedman ML, Blacklow
                SC <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21746931" target="_blank">
                Epstein-Barr virus exploits intrinsic B-lymphocyte transcription programs to achieve immortal cell
                growth</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Sep 6;108(36):14902-7.<br>
                ENCODE data in text, Figure 1, etc.
                </p>
            </div><!--end content-->
        </div><!--end wrapper-->
       
        <div class="wrapper">
            <div class="bar">
                <a name="biology"></a>
                <h4 class="title">Basic biology publications</h4>
            </div>
            <div class="content">
                <p>
                Barkess G, Postnikov Y, Campos CD, Mishra S, Mohan G, Verma S, Bustin M, West KL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22150271" target="_blank">
                The chromatin-binding protein HMGN3 stimulates histone acetylation and transcription across the
                Glyt1 gene</a>.
                <em>Biochem J</em>. 2012 Mar 15;442(3):495-505.<br>
                Mouse ENCODE data in Figure 4
                </p>

                <p>
                Ba&#249; D, Sanyal A, Lajoie BR, Capriotti E, Byron M, Lawrence JB, Dekker J, Marti-Renom MA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21131981" target="_blank">
                The three-dimensional folding of the &#945;-globin gene domain reveals formation of chromatin
                globules</a>.
                <em>Nat Struct Mol Biol</em>. 2011 Jan;18(1):107-14.<br>
                ENCODE data in Figure 1
                </p>

                <p>
                Begum G, Stevens A, Smith EB, Connor K, Challis JR, Bloomfield F, White A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22223754" target="_blank">
                Epigenetic changes in fetal hypothalamic energy regulating pathways are associated with maternal
                undernutrition and twinning</a>.
                <em>FASEB J</em>. 2012 Apr;26(4):1694-703.<br>
                ENCODE data in Figure 1, Figure S1
                </p>

                <p>
                Bell JT, Pai AA, Pickrell JK, Gaffney DJ, Pique-Regi R, Degner JF, Gilad Y, Pritchard JK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21251332" target="_blank">
                DNA methylation patterns associate with genetic and gene expression variation in HapMap cell
                lines</a>.
                <em>Genome Biol</em>. 2011;12(1):R10.<br>
                ENCODE data in Figure 5e, S5
                </p>

                <p>
                B&#246;hlig L, Friedrich M, Engeland K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20833636" target="_blank">
                p53 activates the PANK1/miRNA-107 gene leading to downregulation of CDK6 and p130 cell cycle
                proteins</a>.
                <em>Nucleic Acids Res</em>. 2011 Jan;39(2):440-53.<br>
                ENCODE data in Figure 7
                </p>

                <p>
                Cabili MN, Trapnell C, Goff L, Koziol M, Tazon-Vega B, Regev A, Rinn JL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21890647" target="_blank">
                Integrative annotation of human large intergenic noncoding RNAs reveals global properties and
                specific subclasses</a>.
                <em>Genes Dev</em>. 2011 Sep 15;25(18):1915-27.<br>
                ENCODE data in Figure S4 and supplemental material
                </p>

                <p>
                Chu C, Qu K, Zhong FL, Artandi SE, Chang HY.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21963238" target="_blank">
                Genomic maps of long noncoding RNA occupancy reveal principles of RNA-chromatin interactions</a>.
                <em>Mol Cell</em>. 2011 Nov 18;44(4):667-78.<br>
                ENCODE data in Figure 4
                </p>

                <p>
                Conley AB, Jordan IK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22371288" target="_blank">
                Epigenetic regulation of human cis-natural antisense transcripts</a>.
                <em>Nucleic Acids Res</em>. 2012 Feb;40(4):1438-45.<br>
                ENCODE data used throughout, CAGE, RNA-seq, His-ChIP-seq, TF-ChIP-seq
                </p>

                <p>
                Costessi A, Mahrour N, Tijchon E, Stunnenberg R, Stoel MA, Jansen PW, Sela D, Martin-Brown S,
                Washburn MP, Florens L <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21822215" target="_blank">
                The tumour antigen PRAME is a subunit of a Cul2 ubiquitin ligase and associates with active NFY
                promoters</a>.
                <em>EMBO J</em>. 2011 Aug 5;30(18):3786-98.<br>
                ENCODE data in Figure 5
                </p>

                <p>
                De Rijck J, Bartholomeeusen K, Ceulemans H, Debyser Z, Gijsbers R.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20484370" target="_blank">
                High-resolution profiling of the LEDGF/p75 chromatin interaction in the ENCODE region</a>.
                <em>Nucleic Acids Res</em>. 2010 Oct;38(18):6135-47.<br>
                ENCODE data in Figure 6, Table S2
                </p>

                <p>
                Dong X, Navratilova P, Fredman D, Drivenes &#216;, Becker TS, Lenhard B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19969543" target="_blank">
                Exonic remnants of whole-genome duplication reveal cis-regulatory function of coding exons</a>.
                <em>Nucleic Acids Res</em>. 2010 Mar;38(4):1071-85.<br>
                Figure S4 uses ENCODE pilot analysis
                </p>

                <p>
                Evans KJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20459624" target="_blank">
                Most transcription factor binding sites are in a few mosaic classes of the human genome</a>.
                <em>BMC Genomics</em>. 2010 May 6;11:286.<br>
                ENCODE data summary in Table 3
                </p>

                <p>
                Ferraiuolo MA, Rousseau M, Miyamoto C, Shenker S, Wang XQ, Nadler M, Blanchette M, Dostie J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20660483" target="_blank">
                The three-dimensional architecture of Hox cluster silencing</a>.
                <em>Nucleic Acids Res</em>. 2010 Nov;38(21):7472-84.<br>
                ENCODE data in Figure 5C, S6C, S7C, S8C
                </p>

                <p>
                Fredman D, Dong X, Lenhard B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22206586" target="_blank">
                Making enhancers from spare parts of the genome</a>.
                <em>Genome Biol</em>. 2011 Dec 29;12(12):138.<br>
                ENCODE data in Figure 2
                </p>

                <p>
                Friedli M, Barde I, Arcangeli M, Verp S, Quazzola A, Zakany J, Lin-Marq N, Robyr D, Attanasio C,
                Spitz F <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21206754" target="_blank">
                A systematic enhancer screen using lentivector transgenesis identifies conserved and non-conserved
                functional elements at the Olig1 and Olig2 locus</a>.
                <em>PLoS One</em>. 2010 Dec 29;5(12):e15741.<br>
                ENCODE data described in text, DNase, His-ChIP-seq
                </p>

                <p>
                Fujiwara T, O'Geen H, Keles S, Blahnik K, Linnemann AK, Kang YA, Choi K, Farnham PJ, Bresnick EH.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19941826" target="_blank">
                Discovering hematopoietic mechanisms through genome-wide analysis of GATA factor chromatin
                occupancy</a>.
                <em>Mol Cell</em>. 2009 Nov 25;36(4):667-81.<br>
                ENCODE GATA data
                </p>

                <p>
                Gotea V, Visel A, Westlund JM, Nobrega MA, Pennacchio LA, Ovcharenko I.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20363979" target="_blank">
                Homotypic clusters of transcription factor binding sites are a key component of human promoters and
                enhancers</a>.
                <em>Genome Res</em>. 2010 May;20(5):565-77.<br>
                ENCODE data in Figure 1, Table 1, etc.
                </p>

                <p>
                Guillou E, Ibarra A, Coulon V, Casado-Vela J, Rico D, Casal I, Schwob E, Losada A, M&#233;ndez J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21159821" target="_blank">
                Cohesin organizes chromatin loops at DNA replication factories</a>.
                <em>Genes Dev</em>. 2010 Dec 15;24(24):2812-22.<br>
                ENCODE data in Figure S6, 4B
                </p>

                <p>
                Gy&#246;ry I, Boller S, Nechanitzky R, Mandel E, Pott S, Liu E, Grosschedl R.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22431510" target="_blank">
                Transcription factor Ebf1 regulates differentiation stage-specific signaling, proliferation, and
                survival of B cells</a>.
                <em>Genes Dev</em>. 2012 Apr 1;26(7):668-82.<br>
                Mouse ENCODE DNase data is part of Figure 1, 5
                </p>

                <p>
                Halder K, Halder R, Chowdhury S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19587895" target="_blank">
                Genome-wide analysis predicts DNA structural motifs as nucleosome exclusion signals</a>.
                <em>Mol Biosyst</em>. 2009 Dec;5(12):1703-12.<br>
                ENCODE data
                </p>

                <p>
                Hawkins RD, Hon GC, Lee LK, Ngo Q, Lister R, Pelizzola M, Edsall LE, Kuan S, Luu Y, Klugman S <em>et
                al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20452322" target="_blank">
                Distinct epigenomic landscapes of pluripotent and lineage-committed human cells</a>.
                <em>Cell Stem Cell</em>. 2010 May 7;6(5):479-91.<br>
                ENCODE data in acknowledgments
                </p>

                <p>
                Ho ES, Gunderson SI.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21705472" target="_blank">
                Long conserved fragments upstream of Mammalian polyadenylation sites</a>.
                <em>Genome Biol Evol</em>. 2011;3:654-66.<br>
                ENCODE data used
                </p>

                <p>
                Hodges E, Molaro A, Dos Santos CO, Thekkat P, Song Q, Uren PJ, Park J, Butler J, Rafii S, McCombie
                WR <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21924933" target="_blank">
                Directional DNA methylation changes and complex intermediate states accompany lineage specificity in
                the adult hematopoietic compartment</a>.
                <em>Mol Cell</em>. 2011 Oct 7;44(1):17-28.<br>
                ENCODE data in Figure 3
                </p>

                <p>
                Huda A, Bowen NJ, Conley AB, Jordan IK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21215797" target="_blank">
                Epigenetic regulation of transposable element derived human gene promoters</a>.
                <em>Gene</em>. 2011 Apr 1;475(1):39-48.<br>
                ENCODE data used in Figure 2, 3, etc.
                </p>

                <p>
                Jia H, Osak M, Bogu GK, Stanton LW, Johnson R, Lipovich L.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20587619" target="_blank">
                Genome-wide computational identification and manual annotation of human long noncoding RNA
                genes</a>.
                <em>RNA</em>. 2010 Aug;16(8):1478-87.<br>
                ENCODE data in supplemental data sets 7, 8
                </p>

                <p>
                Jiang Y, Lucas I, Young DJ, Davis EM, Karrison T, Rest JS, Le Beau MM.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19717471" target="_blank">
                Common fragile sites are characterized by histone hypoacetylation</a>.
                <em>Hum Mol Genet</em>. 2009 Dec 1;18(23):4501-12.<br>
                ENCODE data in Figure S3A
                </p>

                <p>
                Karnani N, Taylor CM, Malhotra A, Dutta A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19955211" target="_blank">
                Genomic study of replication initiation in human chromosomes reveals the influence of transcription
                regulation and chromatin structure on origin selection</a>.
                <em>Mol Biol Cell</em>. 2010 Feb 1;21(3):393-404.<br>
                ENCODE data in Figure 5
                </p>

                <p>
                Kehayova P, Monahan K, Chen W, Maniatis T.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21949399" target="_blank">
                Regulatory elements required for the activation and repression of the protocadherin-alpha gene
                cluster</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Oct 11;108(41):17195-200.<br>
                ENCODE data in Figure 5, S3
                </p>

                <p>
                Khalil AM, Guttman M, Huarte M, Garber M, Raj A, Rivea Morales D, Thomas K, Presser A, Bernstein BE,
                van Oudenaarden A <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19571010" target="_blank">
                Many human large intergenic noncoding RNAs associate with chromatin-modifying complexes and affect
                gene expression</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2009 Jul 14;106(28):11667-72.<br>
                ENCODE data in Figure 1, 2
                </p>

                <p>
                Lindblad-Toh K, Garber M, Zuk O, Lin MF, Parker BJ, Washietl S, Kheradpour P, Ernst J, Jordan G,
                Mauceli E <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21993624" target="_blank">
                A high-resolution map of human evolutionary constraint using 29 mammals</a>.
                <em>Nature</em>. 2011 Oct 12;478(7370):476-82.<br>
                ENCODE data in Figure 4
                </p>

                <p>
                Martin MM, Ryan M, Kim R, Zakas AL, Fu H, Lin CM, Reinhold WC, Davis SR, Bilke S, Liu H <em>et
                al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21813623" target="_blank">
                Genome-wide depletion of replication initiation events in highly transcribed regions</a>.
                <em>Genome Res</em>. 2011 Nov;21(11):1822-32.<br>
                ENCODE data in Figure 5, Figure 6, Table 3, Figure S2
                </p>

                <p>
                Mikula M, Gaj P, Dzwonek K, Rubel T, Karczmarski J, Paziewska A, Dzwonek A, Bragoszewski P, Dadlez
                M, Ostrowski J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20587588" target="_blank">
                Comprehensive analysis of the palindromic motif TCTCGCGAGA: a regulatory element of the HNRNPK
                promoter</a>.
                <em>DNA Res</em>. 2010 Aug;17(4):245-60.<br>
                ENCODE data in Figure 7
                </p>

                <p>
                Raab JR, Chiu J, Zhu J, Katzman S, Kurukuti S, Wade PA, Haussler D, Kamakaka RT.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22085927" target="_blank">
                Human tRNA genes function as chromatin insulators</a>.
                <em>EMBO J</em>. 2011 Nov 15;31(2):330-50.<br>
                ENCODE data used in Figure 2A; used in findings presented in text before Figure 1, during Figure 3,
                and maybe part of Figure 8A
                </p>

                <p>
                Rackham O, Shearwood AM, Mercer TR, Davies SM, Mattick JS, Filipovska A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22028365" target="_blank">
                Long noncoding RNAs are generated from the mitochondrial genome and regulated by nuclear-encoded
                proteins</a>.
                <em>RNA</em>. 2011 Dec;17(12):2085-93.<br>
                ENCODE data: RNA-seq data analyzed, described in methods
                </p>

                <p>
                Raj B, O'Hanlon D, Vessey JP, Pan Q, Ray D, Buckley NJ, Miller FD, Blencowe BJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21884984" target="_blank">
                Cross-regulation between an alternative splicing activator and a transcription repressor controls
                neurogenesis</a>.
                <em>Mol Cell</em>. 2011 Sep 2;43(5):843-50.<br>
                ENCODE data in Figure 3C
                </p>

                <p>
                Ramos PS, Williams AH, Ziegler JT, Comeau ME, Guy RT, Lessard CJ, Li H, Edberg JC, Zidovetzki R,
                Criswell LA <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21437871" target="_blank">
                Genetic analyses of interferon pathway-related genes reveal multiple new loci associated with
                systemic lupus erythematosus</a>.
                <em>Arthritis Rheum</em>. 2011 Jul;63(7):2049-57.<br>
                ENCODE data used in text, Figure S1
                </p>

                <p>
                Ren L, Shi M, Wang Y, Yang Z, Wang X, Zhao Z.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21948239" target="_blank">
                CTCF and cohesin cooperatively mediate the cell-type specific interchromatin interaction between
                Bcl11b and Arhgap6 loci</a>.
                <em>Mol Cell Biochem</em>. 2012 Jan;360(1-2):243-51.<br>
                ENCODE results used in text
                </p>

                <p>
                Restivo G, Nguyen BC, Dziunycz P, Ristorcelli E, Ryan RJ, &#214;zuysal &#214;Y, Di Piazza M, Radtke
                F, Dixon MJ, Hofbauer GF <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21909072" target="_blank">
                IRF6 is a mediator of Notch pro-differentiation and tumour suppressive function in
                keratinocytes</a>.
                <em>EMBO J</em>. 2011 Nov 16;30(22):4571-85.<br>
                ENCODE data in Figure 4a
                </p>

                <p>
                Rose D, Stadler PF.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22303364" target="_blank">
                Molecular evolution of the non-coding eosinophil granule ontogeny transcript</a>.
                <em>Front Genet</em>. 2011;2:69.<br>
                ENCODE data in Figure 5, A3
                </p>

                <p>
                Smith AM, Calero-Nieto FJ, Sch&#252;tte J, Kinston S, Timms RT, Wilson NK, Hannah RL, Landry JR,
                G&#246;ttgens B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22158964" target="_blank">
                Integration of Elf-4 into stem/progenitor and erythroid regulatory networks through locus-wide
                chromatin studies coupled with in vivo functional validation</a>.
                <em>Mol Cell Biol</em>. 2012 Feb;32(4):763-73.<br>
                In text, referred to Figure S3, S4, S5 at
                <a href="http://hscl.cimr.cam.ac.uk/genomic_supplementary.html" target="_blank">
                http://hscl.cimr.cam.ac.uk/genomic_supplementary.html</a> which links to the data at
                <a href="http://hscl.cimr.cam.ac.uk/Supplementary_Data/Supplementary_data_Smith.pdf" target="_blank">
                http://hscl.cimr.cam.ac.uk/Supplementary_Data/Supplementary_data_Smith.pdf</a>
                </p>

                <p>
                Stadler MB, Murr R, Burger L, Ivanek R, Lienert F, Sch&#246;ler A, van Nimwegen E, Wirbelauer C,
                Oakeley EJ, Gaidatzis D <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22170606" target="_blank">
                DNA-binding factors shape the mouse methylome at distal regulatory regions</a>.
                <em>Nature</em>. 2011 Dec 14;480(7378):490-5.<br>
                ENCODE mouse and human DNase data used
                </p>

                <p>
                Taft RJ, Hawkins PG, Mattick JS, Morris KV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21813016" target="_blank">
                The relationship between transcription initiation RNAs and CCCTC-binding factor (CTCF)
                localization</a>.
                <em>Epigenetics Chromatin</em>. 2011 Aug 3;4:13.<br>
                ENCODE data in Figure 2
                </p>

                <p>
                Terrenoire E, McRonald F, Halsall JA, Page P, Illingworth RS, Taylor AM, Davison V, O'Neill LP,
                Turner BM.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21078160" target="_blank">
                Immunostaining of modified histones defines high-level features of the human metaphase
                epigenome</a>.
                <em>Genome Biol</em>. 2010;11(11):R110.<br>
                ENCODE data used in Figure 5
                </p>

                <p>
                Woodfine K, Huddleston JE, Murrell A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21281512" target="_blank">
                Quantitative analysis of DNA methylation at all human imprinted regions reveals preservation of
                epigenetic stability in adult somatic tissue</a>.
                <em>Epigenetics Chromatin</em>. 2011 Jan 31;4(1):1.<br>
                ENCODE data in Table 1, Figure S3, CTCF binding
                </p>
            </div><!--end content-->
        </div><!--end wrapper-->
       
        <div class="wrapper">
            <div class="bar">
                <a name="tools"></a>
                <h4 class="title">Tools, methods, databases and review/commentary publications</h4>
            </div>
            <div class="content">
                <p>
                Bedo J, Kowalczyk A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21558156" target="_blank">
                Genome annotation test with validation on transcription start site and ChIP-Seq for Pol-II binding
                data</a>.
                <em>Bioinformatics</em>. 2011 Jun 15;27(12):1610-7.<br>
                ENCODE data in Table 1, throughout; TF-ChIP-seq, STAT1, Pol II, methods development
                </p>

                <p>
                Berlanga-Taylor AJ, Disanto G, Ebers GC, Ramagopalan SV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22000399" target="_blank">
                Vitamin D-gene interactions in multiple sclerosis</a>.
                <em>J Neurol Sci</em>. 2011 Dec 15;311(1-2):32-6.<br>
                In text, comparison of ENCODE His-ChIP-seq and VDR described
                </p>

                <p>
                Bhattacharyya S, Tian J, Bouhassira EE, Locker J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21858039" target="_blank">
                Systematic targeted integration to study Albumin gene control elements</a>.
                <em>PLoS One</em>. 2011;6(8):e23234.<br>
                ENCODE data in Figure 8
                </p>

                <p>
                Boeva V, Surdez D, Guillon N, Tirode F, Fejes AP, Delattre O, Barillot E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20375099" target="_blank">
                De novo motif identification improves the accuracy of predicting transcription factor binding sites
                in ChIP-Seq data analysis</a>.
                <em>Nucleic Acids Res</em>. 2010 Jun;38(11):e126.<br>
                ENCODE data used in Figure 2, 3
                </p>

                <p>
                Carstensen L, Sandelin A, Winther O, Hansen NR.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20828413" target="_blank">
                Multivariate Hawkes process models of the occurrence of regulatory elements</a>.
                <em>BMC Bioinformatics</em>. 2010 Sep 9;11:456.<br>
                ENCODE data in Figure 5, 6, 7, 8, 9
                </p>

                <p>
                Chang HW, Cheng YH, Chuang LY, Yang CH.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20377871" target="_blank">
                SNP-RFLPing 2: an updated and integrated PCR-RFLP tool for SNP genotyping</a>.
                <em>BMC Bioinformatics</em>. 2010 Apr 8;11:173.<br>
                Java application for annotating SNPs, includes ENCODE annotations
                </p>

                <p>
                Chen Y, J&#248;rgensen M, Kolde R, Zhao X, Parker B, Valen E, Wen J, Sandelin A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22047616" target="_blank">
                Prediction of RNA Polymerase II recruitment, elongation and stalling from histone modification
                data</a>.
                <em>BMC Genomics</em>. 2011 Nov 3;12:544.<br>
                ENCODE data used throughout, see Figure 1
                </p>

                <p>
                Dozmorov MG, Cara LR, Giles CB, Wren JD.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22155868" target="_blank">
                GenomeRunner: automating genome exploration</a>.
                <em>Bioinformatics</em>. 2012 Feb 1;28(3):419-20.<br>
                Tool, uses ENCODE data in analyses, picture on their website, supplement
                </p>

                <p>
                Enroth S, Andersson CR, Andersson R, Wadelius C, Gustafsson MG, Komorowski J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22248020" target="_blank">
                A strand specific high resolution normalization method for chip-sequencing data employing multiple
                experimental control measurements</a>.
                <em>Algorithms Mol Biol</em>. 2012 Jan 16;7(1):2.<br>
                ENCODE (and other) data used for method development; Table 1, Figure 2
                </p>

                <p>
                Firpi HA, Ucar D, Tan K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20453004" target="_blank">
                Discover regulatory DNA elements using chromatin signatures and artificial neural network</a>.
                <em>Bioinformatics</em>. 2010 Jul 1;26(13):1579-86.<br>
                ENCODE data in Table 2
                </p>

                <p>
                H&#229;ndstad T, Rye MB, Drabl&#248;s F, S&#230;trom P.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21533218" target="_blank">
                A ChIP-Seq benchmark shows that sequence conservation mainly improves detection of strong
                transcription factor binding sites</a>.
                <em>PLoS One</em>. 2011 Apr 14;6(4):e18430.<br>
                ENCODE data used
                </p>

                <p>
                Huda A, Tyagi E, Mari&#241;o-Ram&#237;rez L, Bowen NJ, Jjingo D, Jordan IK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22087331" target="_blank">
                Prediction of transposable element derived enhancers using chromatin modification profiles</a>.
                <em>PLoS One</em>. 2011;6(11):e27513.<br>
                ENCODE data analyzed throughout
                </p>

                <p>
                Lee D, Karchin R, Beer MA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21875935" target="_blank">
                Discriminative prediction of mammalian enhancers from DNA sequence</a>.
                <em>Genome Res</em>. 2011 Dec;21(12):2167-80.<br>
                Mouse ENCODE DHS data used in Figure 5, Table 2
                </p>

                <p>
                Levenstien MA, Klein RJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21247465" target="_blank">
                Predicting functionally important SNP classes based on negative selection</a>.
                <em>BMC Bioinformatics</em>. 2011 Jan 19;12:26.<br>
                ENCODE data downloaded through Ensembl, used in Tables 2-4
                </p>

                <p>
                Maeder ML, Thibodeau-Beganny S, Osiak A, Wright DA, Anthony RM, Eichtinger M, Jiang T, Foley JE,
                Winfrey RJ, Townsend JA <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18657511" target="_blank">
                Rapid "open-source" engineering of customized zinc-finger nucleases for highly efficient gene
                modification</a>.
                <em>Mol Cell</em>. 2008 Jul 25;31(2):294-301.<br>
                ENCODE data acknowledged in discussion
                </p>

                <p>
                Martin P, Barton A, Eyre S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21177990" target="_blank">
                ASSIMILATOR: a new tool to inform selection of associated genetic variants for functional
                studies</a>.
                <em>Bioinformatics</em>. 2011 Jan 1;27(1):144-6.<br>
                ENCODE example in Figure 1A
                </p>

                <p>
                McLeay RC, Leat CJ, Bailey TL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21724591" target="_blank">
                Tissue-specific prediction of directly regulated genes</a>.
                <em>Bioinformatics</em>. 2011 Sep 1;27(17):2354-60.<br>
                ENCODE data in Table 1, Table 2
                </p>

                <p>
                Mokry M, Hatzis P, Schuijers J, Lansu N, Ruzius FP, Clevers H, Cuppen E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21914722" target="_blank">
                Integrated genome-wide analysis of transcription factor occupancy, RNA polymerase II binding and
                steady-state RNA levels identify differentially regulated functional gene classes</a>.
                <em>Nucleic Acids Res</em>. 2012 Jan;40(1):148-58.<br>
                ENCODE data in Supplementary Data SII, Supplementary Figure S4
                </p>

                <p>
                Mortlock DP, Pregizer S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22241655" target="_blank">
                Identifying functional annotation for noncoding genomic sequences</a>.
                <em>Curr Protoc Hum Genet</em>. 2012 Jan;Chapter 1:Unit1.10.<br>
                ENCODE data used throughout; Figure 1, 2
                </p>

                <p>
                Nelms BL, Labosky PA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22355623" target="_blank">
                A predicted hairpin cluster correlates with barriers to PCR, sequencing and possibly BAC
                recombineering</a>.
                <em>Sci Rep</em>. 2011;1:106.<br>
                ENCODE data in Figure 2
                </p>

                <p>
                Oh YM, Kim JK, Choi S, Yoo JY.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22187154" target="_blank">
                Identification of co-occurring transcription factor binding sites from DNA sequence using clustered
                position weight matrices</a>.
                <em>Nucleic Acids Res</em>. 2012 Mar;40(5):e38.<br>
                ENCODE TF-ChIP-seq data used throughout, described in Table S2D
                </p>

                <p>
                Pickrell JK, Gaffney DJ, Gilad Y, Pritchard JK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21690102" target="_blank">
                False positive peaks in ChIP-seq and other sequencing-based functional assays caused by unannotated
                high copy number regions</a>.
                <em>Bioinformatics</em>. 2011 Aug 1;27(15):2144-6.<br>
                ENCODE data used in Figure 1
                </p>

                <p>
                Pinello L, Lo Bosco G, Hanlon B, Yuan GC.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22017798" target="_blank">
                A motif-independent metric for DNA sequence specificity</a>.
                <em>BMC Bioinformatics</em>. 2011 Oct 21;12:408.<br>
                ENCODE H3K4me1 data (and other data) used to test method
                </p>

                <p>
                Pique-Regi R, Degner JF, Pai AA, Gaffney DJ, Gilad Y, Pritchard JK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21106904" target="_blank">
                Accurate inference of transcription factor binding from DNA sequence and chromatin accessibility
                data</a>.
                <em>Genome Res</em>. 2011 Mar;21(3):447-55.<br>
                ENCODE data in Figure 1, 2, 3, 6, etc.
                </p>

                <p>
                Rye M, S&#230;trom P, H&#229;ndstad T, Drabl&#248;s F.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22115494" target="_blank">
                Clustered ChIP-Seq-defined transcription factor binding sites and histone modifications map distinct
                classes of regulatory elements</a>.
                <em>BMC Biol</em>. 2011 Nov 24;9:80.<br>
                ENCODE data throughout, summarized in Table S7
                </p>

                <p>
                Sadri J, Diallo AB, Blanchette M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21685080" target="_blank">
                Predicting site-specific human selective pressure using evolutionary signatures</a>.
                <em>Bioinformatics</em>. 2011 Jul 1;27(13):i266-74.<br>
                ENCODE data in Figure 4
                </p>

                <p>
                Su J, Teichmann SA, Down TA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21152003" target="_blank">
                Assessing computational methods of cis-regulatory module prediction</a>.
                <em>PLoS Comput Biol</em>. 2010 Dec 2;6(12):e1001020.<br>
                ENCODE data used in Figure 9, 10
                </p>

                <p>
                Valen E, Sandelin A, Winther O, Krogh A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19911049" target="_blank">
                Discovery of regulatory elements is improved by a discriminatory approach</a>.
                <em>PLoS Comput Biol</em>. 2009 Nov;5(11):e1000562.<br>
                ENCODE data used in Table S7
                </p>

                <p>
                Veiga DF, Deus HF, Akdemir C, Vasconcelos AT, Almeida JS.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19919683" target="_blank">
                DASMiner: discovering and integrating data from DAS sources</a>.
                <em>BMC Syst Biol</em>. 2009 Nov 17;3:109.<br>
                ENCODE data displayed in Figure 4
                </p>

                <p>
                Whitington T, Frith MC, Johnson J, Bailey TL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21602262" target="_blank">
                Inferring transcription factor complexes from ChIP-seq data</a>.
                <em>Nucleic Acids Res</em>. 2011 Aug;39(15):e98.<br>
                ENCODE data used
                </p>

                <p>
                Yaffe E, Tanay A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22001755" target="_blank">
                Probabilistic modeling of Hi-C contact maps eliminates systematic biases to characterize global
                chromosomal architecture</a>.
                <em>Nat Genet</em>. 2011 Oct 16;43(11):1059-65.<br>
                ENCODE data in Figure 3
                </p>

                <p>
                Zambelli F, Prazzoli GM, Pesole G, Pavesi G.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22669907" target="_blank">
                Cscan: finding common regulators of a set of genes by using a collection of genome-wide ChIP-seq
                datasets</a>.
                <em>Nucleic Acids Res</em>. 2012 Jul;40(Web Server issue):W510-5.<br>
                RNA polymerases, chrom-HMM tracks produced by the ENCODE project
                </p>

                <p>
                Zhang Z, Zhang MQ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21569556" target="_blank">
                Histone modification profiles are predictive for tissue/cell-type specific expression of both
                protein-coding and microRNA genes</a>.
                <em>BMC Bioinformatics</em>. 2011 May 14;12:155.<br>
                ENCODE data used in Figure 3
                </p>
            </div><!--end content-->
        </div><!--end wrapper-->
    </body>
</html>
