<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
        <title>Publications from non-ENCODE Authors</title>
        <link rel="stylesheet" href="/style/HGStyle.css" type="text/css">
        <link rel="stylesheet" href="/style/encodeProject.css" type="text/css">
    </head>

    <body>
        <!--#include virtual="/inc/encodeProject.topbar.html"-->

        <div class="encodeHeader">
            <a href="http://www.genome.gov/10005107"><img src="/images/ENCODE_scaleup_logo.png" alt="ENCODE Project at NHGRI" title="ENCODE Project at NHGRI"></a>
            <span class="txt">Publications from non-ENCODE Authors</span>
        </div>
        <div class="wrapper">
            <div class="bar"><h4 class="title">Contents</h4></div>
            <div class="content">
            <p>
            This page lists papers that use ENCODE data, by authors not funded by ENCODE.
            The ENCODE project tracks these papers in part to assess impact of the resource.
            Please contact 
            <a href="mailto:&#112;&#97;&#122;&#105;&#110;&#109;@&#109;&#97;&#105;&#108;.&#110;&#105;&#104;.&#103;&#111;&#118;">Mike Pazin at NHGRI</a> 
            to suggest publications to add to this list. The lists below are non-overlapping;
            each paper is only assigned to one list.
           <ul>
               <li><a href='#disease'>Human disease publications</a></li>
               <li><a href='#biology'>Basic biology publications</a></li>
               <li><a href='#tools'>Tools, methods, databases and review/commentary publications</a></li>
               <li><a href='#modencode'>Publications using modENCODE data, without modENCODE authors</a></li>
           </ul>
           </p>
           <p class="date">Updated 19 October 2012</p>
            </div><!--end content-->
        </div><!--end wrapper-->

        <div class="wrapper">
            <div class="bar">
                <a name="disease"></a> 
                <h4 class="title">Human disease publications (44)</h4>
            </div>
            <div class="content">
                <p>
                Adrianto I, Wen F, Templeton A, Wiley G, King JB, Lessard CJ, Bates JS, Hu Y, Kelly JA, Kaufman KM
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21336280" target="_blank">
                Association of a functional variant downstream of TNFAIP3 with systemic lupus erythematosus</a>.
                <em>Nat Genet</em>. 2011 Mar;43(3):253-8.<br>
                ENCODE DNase, TF ChIP, histone modification ChIP presented in Figure S5; other resources also used
                </p>

                <p>
                Akhtar-Zaidi B, Cowper-Sal-lari R, Corradin O, Saiakhova A, Bartels CF, Balasubramanian D, Myeroff L,
                Lutterbaugh J, Jarrar A, Kalady MF <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22499810" target="_blank">
                Epigenomic enhancer profiling defines a signature of colon cancer</a>.
                <em>Science</em>. 2012 May 11;336(6082):736-9.<br>
                ENCODE TF ChIP, Histone modification ChIP; Fig 2a, S2, S6, S7
                </p>

                <p>
                Bekris LM, Lutz F, Yu CE.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22089642" target="_blank">
                Functional analysis of APOE locus genetic variation implicates regional enhancers in the regulation
                of both TOMM40 and APOE</a>.
                <em>J Hum Genet</em>. 2012 Jan;57(1):18-25.<br>
                ENCODE histone modification ChIP, TF ChIP in Figure 2a
                </p>

                <p>
                Berlivet S, Moussette S, Ouimet M, Verlaan DJ, Koka V, Al Tuwaijri A, Kwan T, Sinnett D, Pastinen T,
                Naumova AK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22271045" target="_blank">
                Interaction between genetic and epigenetic variation defines gene expression patterns at the
                asthma-associated locus 17q12-q21 in lymphoblastoid cell lines</a>.
                <em>Hum Genet</em>. 2012 Jul;131(7):1161-71.<br>
                ENCODE TF ChIP, histone modification ChIP, Fig 1S
                </p>

                <p>
                Berman BP, Weisenberger DJ, Aman JF, Hinoue T, Ramjan Z, Liu Y, Noushmehr H, Lange CP, van Dijk CM,
                Tollenaar RA <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22120008" target="_blank">
                Regions of focal DNA hypermethylation and long-range hypomethylation in colorectal cancer coincide with
                nuclear lamina-associated domains</a>.
                <em>Nat Genet</em>. 2011 Nov 27;44(1):40-6.<br>
                ENCODE TF ChIP, histone modification ChIP, DNase, RNA, chromatin states, Fig 2B, S6, S8, S16
                </p>

                <p>
                Burd CE, Jeck WR, Liu Y, Sanoff HK, Wang Z, Sharpless NE.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21151960" target="_blank">
                Expression of linear and novel circular forms of an INK4/ARF-associated non-coding RNA correlates with
                atherosclerosis risk</a>.
                <em>PLoS Genet</em>. 2010 Dec 2;6(12):e1001233.<br>
                ENCODE RNA, Fig 2B and results
                </p>

                <p>
                Calva D, Dahdaleh FS, Woodfield G, Weigel RJ, Carr JC, Chinnathambi S, Howe JR.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21421563" target="_blank">
                Discovery of SMAD4 promoters, transcription factor binding sites and deletions in juvenile polyposis
                patients</a>.
                <em>Nucleic Acids Res</em>. 2011 Jul;39(13):5369-78.<br>
                ENCODE TF ChIP, DNase, Figure 6
                </p>

                <p>
                Carvajal-Carmona LG, Cazier JB, Jones AM, Howarth K, Broderick P, Pittman A, Dobbins S, Tenesa A,
                Farrington S, Prendergast J <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21531788" target="_blank">
                Fine-mapping of colorectal cancer susceptibility loci at 8q23.3, 16q22.1 and 19q13.11: refinement of
                association signals and use of in silico analysis to suggest functional variation and unexpected
                candidate target genes</a>.
                <em>Hum Mol Genet</em>. 2011 Jul 15;20(14):2879-88.<br>
                ENCODE DNase, histone modification ChIP, TF ChIP (Table 1)
                </p>

                <p>
                Crapoulet N, O'Brien P, Ouellette RJ, Robichaud GA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21707507" target="_blank">
                Coordinated expression of Pax-5 and FAK1 in metastasis</a>.
                <em>Anticancer Agents Med Chem</em>. 2011 Sep;11(7):643-9.<br>
                ENCODE data in Figure 1, text
                </p>

                <p>
                Davison LJ, Wallace C, Cooper JD, Cope NF, Wilson NK, Smyth DJ, Howson JM, Saleh N, Al-Jeffery A,
                Angus KL <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21989056" target="_blank">
                Long-range DNA looping and gene expression analyses identify DEXI as an autoimmune disease candidate
                gene</a>.
                <em>Hum Mol Genet</em>. 2012 Jan 15;21(2):322-33.<br>
                ENCODE histone modification ChIP, DNase, TF ChIP, data in Figure 3, and S3A; found enhancer in one
                gene that regulates distal, neighboring gene
                </p>

                <p>
                Delgado-Vega AM, Dozmorov MG, Quir&#243;s MB, Wu YY, Mart&#237;nez-Garc&#237;a B, Kozyrev SV,
                Frosteg&#229;rd J, Truedsson L, de Ram&#243;n E, Gonz&#225;lez-Escribano MF <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22696686" target="_blank">
                Fine mapping and conditional analysis identify a new mutation in the autoimmunity susceptibility gene
                BLK that leads to reduced half-life of the BLK protein</a>.
                <em>Ann Rheum Dis</em>. 2012 Jul;71(7):1219-26.<br>
                ENCODE TF ChIP, Histone modification ChIP, Fig 1, S2, Table S3
                </p>

                <p>
                Farrell JJ, Sherva RM, Chen ZY, Luo HY, Chu BF, Ha SY, Li CK, Lee AC, Li RC, Li CK <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21385855" target="_blank">
                A 3-bp deletion in the HBS1L-MYB intergenic region on chromosome 6q23 is associated with HbF
                expression</a>.
                <em>Blood</em>. 2011 May 5;117(18):4935-45.<br>
                ENCODE TF ChIP, Histone modification ChIP, DNase, RNA in Figure 6
                </p>

                <p>
                Feenstra B, Geller F, Krogh C, Hollegaard MV, G&#248;rtz S, Boyd HA, Murray JC, Hougaard DM, Melbye M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22306654" target="_blank">
                Common variants near MBNL1 and NKX2-5 are associated with infantile hypertrophic pyloric stenosis</a>.
                <em>Nat Genet</em>. 2012 Feb 5;44(3):334-7.<br>
                ENCODE histone modification ChIP, described in results
                </p>

                <p>
                Fu YP, Kohaar I, Rothman N, Earl J, Figueroa JD, Ye Y, Malats N, Tang W, Liu L, Garcia-Closas M
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22416122" target="_blank">
                Common genetic variants in the PSCA gene influence gene expression and bladder cancer risk</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2012 Mar 27;109(13):4974-9.<br>
                ENCODE histone modification ChIP, Figure 2
                </p>

                <p>
                Ghedira K, Hornischer K, Konovalova T, Jenhani AZ, Benkahla A, Kel A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21093613" target="_blank">
                Identification of key mechanisms controlling gene expression in Leishmania infected macrophages
                using genome-wide promoter analysis</a>.
                <em>Infect Genet Evol</em>. 2011 Jun;11(4):769-77.<br>
                ENCODE TF ChIP, Fig 3B, Table S3, S4
                </p>

                <p>
                Gijselinck I, Van Langenhove T, van der Zee J, Sleegers K, Philtjens S, Kleinberger G, Janssens J,
                Bettens K, Van Cauwenberghe C, Pereson S <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22154785" target="_blank">
                A C9orf72 promoter repeat expansion in a Flanders-Belgian cohort with disorders of the frontotemporal
                lobar degeneration-amyotrophic lateral sclerosis spectrum: a gene identification study</a>.
                <em>Lancet Neurol</em>. 2012 Jan;11(1):54-65.<br>
                ENCODE Histone modification ChIP, TF ChIP, DNase, Figure S4
                </p>

                <p>
                Glinskii AB, Ma S, Ma J, Grant D, Lim CU, Guest I, Sell S, Buttyan R, Glinsky GV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22067658" target="_blank">
                Networks of intergenic long-range enhancers and snpRNAs drive castration-resistant phenotype of
                prostate cancer and contribute to pathogenesis of multiple common human disorders</a>.
                <em>Cell Cycle</em>. 2011 Oct 15;10(20):3571-97.<br>
                ENCODE TF ChIP, histone modification ChIP, DNase, Fig 4A, 4D, S4, S10
                </p>

                <p>
                Han J, Park SG, Bae JB, Choi J, Lyu JM, Park SH, Kim HS, Kim YJ, Kim S, Kim TY.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22564738" target="_blank">
                The characteristics of genome-wide DNA methylation in na&#239;ve CD4+ T cells of patients with
                psoriasis or atopic dermatitis</a>.
                <em>Biochem Biophys Res Commun</em>. 2012 May 25;422(1):157-63.<br>
                ENCODE Txn, His, DNase, and TF composite tracks, Figure 1, S1-S6
                </p>

                <p>
                Harismendy O, Bansal V, Bhatia G, Nakano M, Scott M, Wang X, Dib C, Turlotte E, Sipe JC, Murray SS
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21118518" target="_blank">
                Population sequencing of two endocannabinoid metabolic genes identifies rare and common regulatory
                variants associated with extreme obesity and metabolite level</a>.
                <em>Genome Biol</em>. 2010;11(11):R118.<br>
                ENCODE histone modifications, Fig 5C, TFs, Fig 5D
                </p>

                <p>
                Harismendy O, Notani D, Song X, Rahim NG, Tanasa B, Heintzman N, Ren B, Fu XD, Topol EJ, Rosenfeld
                MG <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21307941" target="_blank">
                9p21 DNA variants associated with coronary artery disease impair interferon-&#947; signalling
                response</a>.
                <em>Nature</em>. 2011 Feb 10;470(7333):264-8.<br>
                ENCODE histone modification ChIP, TF ChIP, Figure 1
                </p>

                <p>
                Higareda-Almaraz JC, Enr&#237;quez-Gasca Mdel R, Hern&#225;ndez-Ortiz M, Resendis-Antonio O,
                Encarnaci&#243;n-Guevara S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21696634" target="_blank">
                Proteomic patterns of cervical cancer cell lines, a network perspective</a>.
                <em>BMC Syst Biol</em>. 2011 Jun 22;5:96.<br>
                ENCODE TF ChIP, GENCODE, Figure 1, table 4
                </p>

                <p>
                Hofer T, Foll M, Excoffier L.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22439654" target="_blank">
                Evolutionary forces shaping genomic islands of population differentiation in humans</a>.
                <em>BMC Genomics</em>. 2012 Mar 22;13:107.<br>
                ENCODE TF ChIP, described in text and methods
                </p>

                <p>
                Jaeger E, Leedham S, Lewis A, Segditsas S, Becker M, Cuadrado PR, Davis H, Kaur K, Heinimann K,
                Howarth K <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22561515" target="_blank">
                Hereditary mixed polyposis syndrome is caused by a 40-kb upstream duplication that leads to
                increased and ectopic expression of the BMP antagonist GREM1</a>.
                <em>Nat Genet</em>. 2012 May 6;44(6):699-703.<br>
                ENCODE Histone modification ChIP, TF ChIP, DNase, Fig S3
                </p>

                <p>
                Jones AM, Beggs AD, Carvajal-Carmona L, Farrington S, Tenesa A, Walker M, Howarth K, Ballereau S,
                Hodgson SV, Zauber A <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21708826" target="_blank">
                TERC polymorphisms are associated both with susceptibility to colorectal cancer and with longer
                telomeres</a>.
                <em>Gut</em>. 2012 Feb;61(2):248-54.<br>
                ENCODE DNase, Histone modification ChIP, TF ChIP, described in results
                </p>

                <p>
                Lamina C, Coassin S, Illig T, Kronenberg F.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21945496" target="_blank">
                Look beyond one's own nose: combination of information from publicly available sources reveals an
                association of GATA4 polymorphisms with plasma triglycerides</a>.
                <em>Atherosclerosis</em>. 2011 Dec;219(2):698-703.<br>
                ENCODE histone modification ChIP, TF ChIP, DNase, figure 2, S1, S2
                </p>

                <p>
                Lessard CJ, Adrianto I, Kelly JA, Kaufman KM, Grundahl KM, Adler A, Williams AH, Gallant CJ,
                Marta&#160;E.&#160;Alarc&#243;n-Riquelme on behalf of the BIOLUPUS and GENLES Networks, Anaya JM
                <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21194677" target="_blank">
                Identification of a systemic lupus erythematosus susceptibility locus at 11p13 between PDHX and CD44
                in a multiethnic study</a>.
                <em>Am J Hum Genet</em>. 2011 Jan 7;88(1):83-91.<br>
                ENCODE TF ChIP, histone modification ChIP, Figure 2C, text
                </p>

                <p>
                Li YJ, Minear MA, Rimmler J, Zhao B, Balajonda E, Hauser MA, Allingham RR, Eghrari AO, Riazuddin SA,
                Katsanis N <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21533127" target="_blank">
                Replication of TCF4 through association and linkage studies in late-onset Fuchs endothelial corneal
                dystrophy</a>.
                <em>PLoS One</em>. 2011 Apr 20;6(4):e18044.<br>
                ENCODE data mentioned in text
                </p>

                <p>
                Liu JZ, Almarri MA, Gaffney DJ, Mells GF, Jostins L, Cordell HJ, Ducker SJ, Day DB, Heneghan MA,
                Neuberger JM <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22961000" target="_blank">
                Dense fine-mapping study identifies new susceptibility loci for primary biliary cirrhosis</a>.
                <em>Nat Genet</em>. 2012 Sep 9;44(10):1137-1141.<br>
                ENCODE DNase, FAIRE, Table 1, S7, S9, Figure 2, S9
                </p>

                <p>
                Lu Y, Zhang Z, Yu H, Zheng SL, Isaacs WB, Xu J, Sun J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21541972" target="_blank">
                Functional annotation of risk loci identified through genome-wide association studies for prostate
                cancer</a>.
                <em>Prostate</em>. 2011 Jun 15;71(9):955-63.<br>
                ENCODE TF ChIP, histone modification ChIP, Table 1, 2, 3
                </p>

                <p>
                Lubbe SJ, Pittman AM, Olver B, Lloyd A, Vijayakrishnan J, Naranjo S, Dobbins S, Broderick P,
                G&#243;mez-Skarmeta JL, Houlston RS.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22158048" target="_blank">
                The 14q22.2 colorectal cancer variant rs4444235 shows cis-acting regulation of BMP4</a>.
                <em>Oncogene</em>. 2012 Aug 16;31(33):3777-84.<br>
                ENCODE histone modification ChIP, DNase, Fig 2
                </p>

                <p>
                Maruo S, Zhao B, Johannsen E, Kieff E, Zou J, Takada K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21245331" target="_blank">
                Epstein-Barr virus nuclear antigens 3C and 3A maintain lymphoblastoid cell growth by repressing
                p16INK4A and p14ARF expression</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Feb 1;108(5):1919-24.<br>
                ENCODE histone modification ChIP, TF ChIP, figure S2
                </p>

                <p>
                Middelberg RP, Benyamin B, de Moor MH, Warrington NM, Gordon S, Henders AK, Medland SE, Nyholt DR,
                de Geus EJ, Hottenga JJ <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22010049" target="_blank">
                Loci affecting gamma-glutamyl transferase in adults and adolescents show age &#215; SNP interaction
                and cardiometabolic disease associations</a>.
                <em>Hum Mol Genet</em>. 2012 Jan 15;21(2):446-55.<br>
                ENCODE TF ChIP, histone modification ChIP, described in discussion
                </p>

                <p>
                Ramagopalan SV, Heger A, Berlanga AJ, Maugeri NJ, Lincoln MR, Burrell A, Handunnetthi L, Handel AE,
                Disanto G, Orton SM <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20736230" target="_blank">
                A ChIP-seq defined genome-wide map of vitamin D receptor binding: associations with disease and
                evolution</a>.
                <em>Genome Res</em>. 2010 Oct;20(10):1352-60.<br>
                ENCODE DNase, Histone modification ChIP, TF ChIP, Fig 5, S1, S3
                </p>

                <p>
                Rushefski M, Aplenc R, Meyer N, Li M, Feng R, Lanken PN, Gallop R, Bellamy S, Localio AR, Feinstein
                SI <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21627785" target="_blank">
                Novel variants in the PRDX6 Gene and the risk of Acute Lung Injury following major trauma</a>.
                <em>BMC Med Genet</em>. 2011 May 31;12:77.<br>
                ENCODE TF ChIP used, negative result
                </p>

                <p>
                Sankaran VG, Menne TF, &#352;&#263;epanovi&#263; D, Vergilio JA, Ji P, Kim J, Thiru P, Orkin SH,
                Lander ES, Lodish HF.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21205891" target="_blank">
                MicroRNA-15a and -16-1 act via MYB to elevate fetal hemoglobin expression in human trisomy 13</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Jan 25;108(4):1519-24.<br>
                ENCODE histone modification ChIP in Table S2, S3
                </p>

                <p>
                Spain SL, Carvajal-Carmona LG, Howarth KM, Jones AM, Su Z, Cazier JB, Williams J, Aaltonen LA, Pharoah
                P, Kerr DJ <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22076443" target="_blank">
                Refinement of the associations between risk of colorectal cancer and polymorphisms on chromosomes 1q41
                and 12q13.13</a>.
                <em>Hum Mol Genet</em>. 2012 Feb 15;21(4):934-46.<br>
                ENCODE DNase, Histone modification ChIP, TF ChIP, in results and discussion
                </p>

                <p>
                Szymanski M, Wang R, Bassett SS, Avramopoulos D.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22832520" target="_blank">
                Alzheimer's risk variants in the clusterin gene are associated with alternative splicing</a>.
                <em>Transl Psychiatry</em>. 2011 Jul 5;1:e18.<br>
                ENCODE DNase, TF ChIP, Methyl-seq, Figure 1b
                </p>

                <p>
                Thakur RK, Yadav VK, Kumar P, Chowdhury S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21556888" target="_blank">
                Mechanisms of non-metastatic 2 (NME2)-mediated control of metastasis across tumor types</a>.
                <em>Naunyn Schmiedebergs Arch Pharmacol</em>. 2011 Oct;384(4-5):397-406.<br>
                ENCODE TF ChIP, Figure 3
                </p>

                <p>
                Tomlinson IP, Carvajal-Carmona LG, Dobbins SE, Tenesa A, Jones AM, Howarth K, Palles C, Broderick P,
                Jaeger EE, Farrington S <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21655089" target="_blank">
                Multiple common susceptibility variants near BMP pathway loci GREM1, BMP4, and BMP2 explain part of
                the missing heritability of colorectal cancer</a>.
                <em>PLoS Genet</em>. 2011 Jun;7(6):e1002105.<br>
                ENCODE DNase, Histone modification ChIP, results
                </p>

                <p>
                Trynka G, Hunt KA, Bockett NA, Romanos J, Mistry V, Szperl A, Bakker SF, Bardella MT, Bhaw-Rosun L,
                Castillejo G <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22057235" target="_blank">
                Dense genotyping identifies and localizes multiple common and rare variant association signals in
                celiac disease</a>.
                <em>Nat Genet</em>. 2011 Nov 6;43(12):1193-201.<br>
                ENCODE TF ChIP, Table S2, discussion
                </p>

                <p>
                Verlaan DJ, Berlivet S, Hunninghake GM, Madore AM, Larivi&#232;re M, Moussette S, Grundberg E, Kwan
                T, Ouimet M, Ge B <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19732864" target="_blank">
                Allele-specific chromatin remodeling in the ZPBP2/GSDMB/ORMDL3 locus associated with the risk of
                asthma and autoimmune disease</a>.
                <em>Am J Hum Genet</em>. 2009 Sep;85(3):377-93.<br>
                ENCODE TF ChIP, figure S4
                </p>

                <p>
                Visser M, Kayser M, Palstra RJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22234890" target="_blank">
                HERC2 rs12913832 modulates human pigmentation by attenuating chromatin-loop formation between a
                long-range enhancer and the OCA2 promoter</a>.
                <em>Genome Res</em>. 2012 Mar;22(3):446-55.<br>
                ENCODE histone modification ChIP, DNase, Figure S4b
                </p>

                <p>
                Zhang X, Cowper-Sal Lari R, Bailey SD, Moore JH, Lupien M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22665440" target="_blank">
                Integrative functional genomics identifies an enhancer looping to the SOX9 gene disrupted by the
                17q24.3 prostate cancer risk locus</a>.
                <em>Genome Res</em>. 2012 Aug;22(8):1437-46.<br>
                ENCODE Histone modification ChIP, DNase, Fig 1
                </p>

                <p>
                Zhu Q, Ge D, Maia JM, Zhu M, Petrovski S, Dickson SP, Heinzen EL, Shianna KV, Goldstein DB.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21457907" target="_blank">
                A genome-wide comparison of the functional properties of rare and common genetic variants in humans</a>.
                <em>Am J Hum Genet</em>. 2011 Apr 8;88(4):458-68.<br>
                ENCODE histone modification ChIP, TF ChIP, FAIRE, Figure 3B, 4B, Table 3, 4
                </p>

            </div><!--end content-->
        </div><!--end wrapper-->
       
        <div class="wrapper">
            <div class="bar">
                <a name="biology"></a>
                <h4 class="title">Basic biology publications (82)</h4>
            </div>
            <div class="content">
                <p>
                Abeel T, Saeys Y, Bonnet E, Rouz&#233; P, Van de Peer Y.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18096745" target="_blank">
                Generic eukaryotic core promoter prediction using structural features of DNA</a>.
                <em>Genome Res</em>. 2008 Feb;18(2):310-23.<br>
                ENCODE Pilot DNase, GENCODE genes, described in results
                </p>

                <p>
                Bailey TL, Machanick P.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22610855" target="_blank">
                Inferring direct DNA binding from ChIP-seq</a>.
                <em>Nucleic Acids Res</em>. 2012 Sep 1;40(17):e128.<br>
                ENCODE TF ChIP, Fig S7, Table S3
                </p>

                <p>
                Baker A, Audit B, Chen CL, Moindrot B, Leleu A, Guilbaud G, Rappailles A, Vaillant C, Goldar A, Mongelard F <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22496629" target="_blank">
                Replication fork polarity gradients revealed by megabase-sized U-shaped replication timing domains in human cell lines</a>.
                <em>PLoS Comput Biol</em>. 2012;8(4):e1002443.<br>
                ENCODE DNase, TF ChIP, HiC, Fig 3, 4, 5, S10, S11, S13
                </p>

                <p>
                Barkess G, Postnikov Y, Campos CD, Mishra S, Mohan G, Verma S, Bustin M, West KL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22150271" target="_blank">
                The chromatin-binding protein HMGN3 stimulates histone acetylation and transcription across the
                Glyt1 gene</a>.
                <em>Biochem J</em>. 2012 Mar 15;442(3):495-505.<br>
                Mouse ENCODE DNase, TF ChIP, histone modification ChIP in figure 4
                </p>

                <p>
                Begum G, Stevens A, Smith EB, Connor K, Challis JR, Bloomfield F, White A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22223754" target="_blank">
                Epigenetic changes in fetal hypothalamic energy regulating pathways are associated with maternal
                undernutrition and twinning</a>.
                <em>FASEB J</em>. 2012 Apr;26(4):1694-703.<br>
                ENCODE histone modification ChIP, TF ChIP in Figure 1, Figure S1
                </p>

                <p>
                Bell JT, Pai AA, Pickrell JK, Gaffney DJ, Pique-Regi R, Degner JF, Gilad Y, Pritchard JK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21251332" target="_blank">
                DNA methylation patterns associate with genetic and gene expression variation in HapMap cell
                lines</a>.
                <em>Genome Biol</em>. 2011;12(1):R10.<br>
                ENCODE histone modification ChIP,  figure 5e, S5
                </p>

                <p>
                Bieberstein NI, Oesterreich FC, Straube K, Neugebauer KM.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22840397" target="_blank">
                First exon length controls active chromatin signatures and transcription</a>.
                <em>Cell Rep</em>. 2012 Jul 26;2(1):62-8.<br>
                ENCODE RNA, histone modification ChIP, TF ChIP, nucleosomes, mouse TF ChIP, Fig 1, 4, S1, S2, S3, S5, Table S3
                </p>

                <p>
                Birnbaum RY, Clowney EJ, Agamy O, Kim MJ, Zhao J, Yamanaka T, Pappalardo Z, Clarke SL, Wenger AM, Nguyen L <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22442009" target="_blank">
                Coding exons function as tissue-specific enhancers of nearby genes</a>.
                <em>Genome Res</em>. 2012 Jun;22(6):1059-68.<br>
                ENCODE histone modification ChIP, TF ChIP, Table S1
                </p>

                <p>
                Bock C, Halachev K, B&#252;ch J, Lengauer T.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19208250" target="_blank">
                EpiGRAPH: user-friendly software for statistical analysis and prediction of (epi)genomic data</a>.
                <em>Genome Biol</em>. 2009 Feb 10;10(2):R14.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), text, supplement
                </p>

                <p>
                B&#246;hlig L, Friedrich M, Engeland K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20833636" target="_blank">
                p53 activates the PANK1/miRNA-107 gene leading to downregulation of CDK6 and p130 cell cycle
                proteins</a>.
                <em>Nucleic Acids Res</em>. 2011 Jan;39(2):440-53.<br>
                ENCODE histone modification ChIP, DNase, TF ChIP in Figure 7
                </p>

                <p>
                Bolotin E, Chellappa K, Hwang-Verslues W, Schnabl JM, Yang C, Sladek FM.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22085832" target="_blank">
                Nuclear receptor HNF4&#945; binding sequences are widespread in Alu repeats</a>.
                <em>BMC Genomics</em>. 2011 Nov 15;12:560.<br>
                ENCODE DNase, TF ChIP, used in text, Fig 7
                </p>

                <p>
                Cabili MN, Trapnell C, Goff L, Koziol M, Tazon-Vega B, Regev A, Rinn JL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21890647" target="_blank">
                Integrative annotation of human large intergenic noncoding RNAs reveals global properties and
                specific subclasses</a>.
                <em>Genes Dev</em>. 2011 Sep 15;25(18):1915-27.<br>
                ENCODE histone modification ChIP, GENCODE, Figure 1, S2, S4, S5, S6
                </p>

                <p>
                Chen H, Tian Y, Shu W, Bo X, Wang S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22829947" target="_blank">
                Comprehensive identification and annotation of cell type-specific and ubiquitous CTCF-binding sites in the human genome</a>.
                <em>PLoS One</em>. 2012;7(7):e41374.<br>
                ENCODE TF ChIP, DNase, FAIRE, histone modification ChIP used throughout
                </p>

                <p>
                Chen W, Feng P, Lin H.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22449982" target="_blank">
                Prediction of replication origins by calculating DNA structural properties</a>.
                <em>FEBS Lett</em>. 2012 Mar 23;586(6):934-8.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), Fig 2, 3B
                </p>

                <p>
                Chorley BN, Campbell MR, Wang X, Karaca M, Sambandan D, Bangura F, Xue P, Pi J, Kleeberger SR, Bell DA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22581777" target="_blank">
                Identification of novel NRF2-regulated genes by ChIP-Seq: influence on retinoid X receptor alpha</a>.
                <em>Nucleic Acids Res</em>. 2012 Aug 1;40(15):7416-29.<br>
                ENCODE TF ChIP, Histone modification ChIP, DNase, Fig 4A, S4, S6, Table S2, described in results
                </p>

                <p>
                Conley AB, Jordan IK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22371288" target="_blank">
                Epigenetic regulation of human cis-natural antisense transcripts</a>.
                <em>Nucleic Acids Res</em>. 2012 Feb;40(4):1438-45.<br>
                ENCODE CAGE, RNA, Histone modification ChIP, TF ChIP, used thoughout
                </p>

                <p>
                Costessi A, Mahrour N, Tijchon E, Stunnenberg R, Stoel MA, Jansen PW, Sela D, Martin-Brown S,
                Washburn MP, Florens L <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21822215" target="_blank">
                The tumour antigen PRAME is a subunit of a Cul2 ubiquitin ligase and associates with active NFY
                promoters</a>.
                <em>EMBO J</em>. 2011 Aug 5;30(18):3786-98.<br>
                ENCODE histone modification ChIP, TF ChIP, Figure 5
                </p>

                <p>
                Dai Z, Dai X.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22125484" target="_blank">
                Gene expression divergence is coupled to evolution of DNA structure in coding regions</a>.
                <em>PLoS Comput Biol</em>. 2011 Nov;7(11):e1002275.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), Fig 3A
                </p>

                <p>
                De Rijck J, Bartholomeeusen K, Ceulemans H, Debyser Z, Gijsbers R.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20484370" target="_blank">
                High-resolution profiling of the LEDGF/p75 chromatin interaction in the ENCODE region</a>.
                <em>Nucleic Acids Res</em>. 2010 Oct;38(18):6135-47.<br>
                ENCODE histone modification ChIP, TF ChIP, RNA, figure 6, 7, Table S2
                </p>

                <p>
                Desai SS, Achrekar SK, Pathak BR, Desai SK, Mangoli VS, Mangoli RV, Mahale SD.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21752882" target="_blank">
                Follicle-stimulating hormone receptor polymorphism (G-29A) is associated with altered level of receptor expression in Granulosa cells</a>.
                <em>J Clin Endocrinol Metab</em>. 2011 Sep;96(9):2805-12.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), Fig 1A, methods
                </p>

                <p>
                Dong X, Navratilova P, Fredman D, Drivenes &#216;, Becker TS, Lenhard B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19969543" target="_blank">
                Exonic remnants of whole-genome duplication reveal cis-regulatory function of coding exons</a>.
                <em>Nucleic Acids Res</em>. 2010 Mar;38(4):1071-85.<br>
                ENCODE pilot analysis (neutrally evolving sites),  Figure S4
                </p>

                <p>
                Evans KJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20459624" target="_blank">
                Most transcription factor binding sites are in a few mosaic classes of the human genome</a>.
                <em>BMC Genomics</em>. 2010 May 6;11:286.<br>
                ENCODE TF CHIP used throughout, summary in table 3
                </p>

                <p>
                Fairfax BP, Makino S, Radhakrishnan J, Plant K, Leslie S, Dilthey A, Ellis P, Langford C, Vannberg FO, Knight JC.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22446964" target="_blank">
                Genetics of gene expression in primary immune cells identifies cell type-specific master regulators and roles of HLA alleles</a>.
                <em>Nat Genet</em>. 2012 Mar 25;44(5):502-10.<br>
                ENCODE RNA, DNase, Histone modification ChIP, TF ChIP, Fig 4c, S12
                </p>

                <p>
                Ferraiuolo MA, Rousseau M, Miyamoto C, Shenker S, Wang XQ, Nadler M, Blanchette M, Dostie J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20660483" target="_blank">
                The three-dimensional architecture of Hox cluster silencing</a>.
                <em>Nucleic Acids Res</em>. 2010 Nov;38(21):7472-84.<br>
                ENCODE TF ChIP, Figure 5C, S6C, S7C, S8C
                </p>

                <p>
                Fredman D, Dong X, Lenhard B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22206586" target="_blank">
                Making enhancers from spare parts of the genome</a>.
                <em>Genome Biol</em>. 2011 Dec 29;12(12):138.<br>
                ENCODE histone modification ChIP, Figure 2
                </p>

                <p>
                Friedli M, Barde I, Arcangeli M, Verp S, Quazzola A, Zakany J, Lin-Marq N, Robyr D, Attanasio C,
                Spitz F <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21206754" target="_blank">
                A systematic enhancer screen using lentivector transgenesis identifies conserved and non-conserved
                functional elements at the Olig1 and Olig2 locus</a>.
                <em>PLoS One</em>. 2010 Dec 29;5(12):e15741.<br>
                ENCODE DNase, histone modification ChIP, described in text
                </p>

                <p>
                Gewurz BE, Mar JC, Padi M, Zhao B, Shinners NP, Takasaki K, Bedoya E, Zou JY, Cahir-McFarland E, Quackenbush J <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21543491" target="_blank">
                Canonical NF-kappaB activation is essential for Epstein-Barr virus latent membrane protein 1 TES2/CTAR2 gene regulation</a>.
                <em>J Virol</em>. 2011 Jul;85(13):6764-73.<br>
                ENCODE TF ChIP, described in methods
                </p>

                <p>
                Gorkin DU, Lee D, Reed X, Fletez-Brant C, Bessling SL, Loftus SK, Beer MA, Pavan WJ, McCallion AS.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/23019145" target="_blank">
                Integration of ChIP-seq and machine learning reveals enhancers and a predictive regulatory sequence vocabulary in melanocytes</a>.
                <em>Genome Res</em>. 2012 Sep 27;.<br>
                ENCODE and mouse ENCODE, TF ChIP, histone modification ChIP, DNase, Fig 3B, 3C, 7E, 7F
                </p>

                <p>
                Guilbaud G, Rappailles A, Baker A, Chen CL, Arneodo A, Goldar A, d'Aubenton-Carafa Y, Thermes C, Audit B, Hyrien O.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22219720" target="_blank">
                Evidence for sequential and increasing activation of replication origins along replication timing gradients in the human genome</a>.
                <em>PLoS Comput Biol</em>. 2011 Dec;7(12):e1002322.<br>
                ENCODE DNase, origin mapping, Fig 11, table 1
                </p>

                <p>
                Guillou E, Ibarra A, Coulon V, Casado-Vela J, Rico D, Casal I, Schwob E, Losada A, M&#233;ndez J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21159821" target="_blank">
                Cohesin organizes chromatin loops at DNA replication factories</a>.
                <em>Genes Dev</em>. 2010 Dec 15;24(24):2812-22.<br>
                ENCODE TF ChIP, replication origins, Fig S6, 4B
                </p>

                <p>
                Gy&#246;ry I, Boller S, Nechanitzky R, Mandel E, Pott S, Liu E, Grosschedl R.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22431510" target="_blank">
                Transcription factor Ebf1 regulates differentiation stage-specific signaling, proliferation, and
                survival of B cells</a>.
                <em>Genes Dev</em>. 2012 Apr 1;26(7):668-82.<br>
                Mouse ENCODE DNase data is part of Figure 1, 5
                </p>

                <p>
                Ho ES, Gunderson SI.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21705472" target="_blank">
                Long conserved fragments upstream of Mammalian polyadenylation sites</a>.
                <em>Genome Biol Evol</em>. 2011;3:654-66.<br>
                ENCODE DNase, Fig 6
                </p>

                <p>
                Hodges E, Molaro A, Dos Santos CO, Thekkat P, Song Q, Uren PJ, Park J, Butler J, Rafii S, McCombie
                WR <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21924933" target="_blank">
                Directional DNA methylation changes and complex intermediate states accompany lineage specificity in
                the adult hematopoietic compartment</a>.
                <em>Mol Cell</em>. 2011 Oct 7;44(1):17-28.<br>
                ENCODE histone modification ChIP, figure 3
                </p>

                <p>
                Horakova AH, Moseley SC, McLaughlin CR, Tremblay DC, Chadwick BP.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22791747" target="_blank">
                The macrosatellite DXZ4 mediates CTCF-dependent long-range intrachromosomal interactions on the human inactive X chromosome</a>.
                <em>Hum Mol Genet</em>. 2012 Oct 15;21(20):4367-77.<br>
                ENCODE TF ChIP, Fig 3
                </p>

                <p>
                Huda A, Bowen NJ, Conley AB, Jordan IK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21215797" target="_blank">
                Epigenetic regulation of transposable element derived human gene promoters</a>.
                <em>Gene</em>. 2011 Apr 1;475(1):39-48.<br>
                ENCODE histone modification ChIP, figures 2, 3, etc
                </p>

                <p>
                Jeffries CD, Perkins DO, Guan X.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21167075" target="_blank">
                Gene processing control loops suggested by sequencing, splicing, and RNA folding</a>.
                <em>BMC Bioinformatics</em>. 2010 Dec 20;11:602.<br>
                ENCODE Gencode, described in text
                </p>

                <p>
                Jia H, Osak M, Bogu GK, Stanton LW, Johnson R, Lipovich L.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20587619" target="_blank">
                Genome-wide computational identification and manual annotation of human long noncoding RNA
                genes</a>.
                <em>RNA</em>. 2010 Aug;16(8):1478-87.<br>
                ENCODE TF ChIP, histone modification ChIP, in supplemental data sets 7, 8
                </p>

                <p>
                Jiang Y, Lucas I, Young DJ, Davis EM, Karrison T, Rest JS, Le Beau MM.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19717471" target="_blank">
                Common fragile sites are characterized by histone hypoacetylation</a>.
                <em>Hum Mol Genet</em>. 2009 Dec 1;18(23):4501-12.<br>
                ENCODE histone modification ChIP, figure S3A
                </p>

                <p>
                Kehayova P, Monahan K, Chen W, Maniatis T.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21949399" target="_blank">
                Regulatory elements required for the activation and repression of the protocadherin-alpha gene
                cluster</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Oct 11;108(41):17195-200.<br>
                ENCODE TF ChIP, Figures 5, S3
                </p>

                <p>
                Kosti I, Radivojac P, Mandel-Gutfreund Y.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22844237" target="_blank">
                An integrated regulatory network reveals pervasive cross-regulation among transcription and splicing factors</a>.
                <em>PLoS Comput Biol</em>. 2012 Jul;8(7):e1002603.<br>
                ENCODE TF ChIP, described in methods
                </p>

                <p>
                Levenstien MA, Klein RJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21247465" target="_blank">
                Predicting functionally important SNP classes based on negative selection</a>.
                <em>BMC Bioinformatics</em>. 2011 Jan 19;12:26.<br>
                ENCODE histone modification ChIP, DNase, TF ChIP, data downloaded through Ensembl, used in Tables 2-4, etc.
                </p>

                <p>
                Managadze D, Rogozin IB, Chernikova D, Shabalina SA, Koonin EV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22071789" target="_blank">
                Negative correlation between expression level and evolutionary rate of long intergenic noncoding RNAs</a>.
                <em>Genome Biol Evol</em>. 2011;3:1390-404.<br>
                ENCODE mouse RNA, described in methods
                </p>

                <p>
                Martin DI, Singer M, Dhahbi J, Mao G, Zhang L, Schroth GP, Pachter L, Boffelli D.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21908772" target="_blank">
                Phyloepigenomic comparison of great apes reveals a correlation between somatic and germline methylation states</a>.
                <em>Genome Res</em>. 2011 Dec;21(12):2049-57.<br>
                ENCODE Histone modification ChIP, FAIRE, table 1
                </p>

                <p>
                Martin MM, Ryan M, Kim R, Zakas AL, Fu H, Lin CM, Reinhold WC, Davis SR, Bilke S, Liu H <em>et
                al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21813623" target="_blank">
                Genome-wide depletion of replication initiation events in highly transcribed regions</a>.
                <em>Genome Res</em>. 2011 Nov;21(11):1822-32.<br>
                ENCODE histone modification ChIP, TF ChIP, DNase, methyl-RRBS, Figure 5, Figure 6, Table 3, Figure S2
                </p>

                <p>
                Mikula M, Gaj P, Dzwonek K, Rubel T, Karczmarski J, Paziewska A, Dzwonek A, Bragoszewski P, Dadlez
                M, Ostrowski J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20587588" target="_blank">
                Comprehensive analysis of the palindromic motif TCTCGCGAGA: a regulatory element of the HNRNPK
                promoter</a>.
                <em>DNA Res</em>. 2010 Aug;17(4):245-60.<br>
                ENCODE DNase, Figure 7
                </p>

                <p>
                Mokry M, Hatzis P, Schuijers J, Lansu N, Ruzius FP, Clevers H, Cuppen E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21914722" target="_blank">
                Integrated genome-wide analysis of transcription factor occupancy, RNA polymerase II binding and steady-state RNA levels identify differentially regulated functional gene classes</a>.
                <em>Nucleic Acids Res</em>. 2012 Jan;40(1):148-58.<br>
                ENCODE RNA, TF ChIP, Supplementary Figure S4, supplementary material 2
                </p>

                <p>
                Morikawa M, Koinuma D, Tsutsumi S, Vasilaki E, Kanki Y, Heldin CH, Aburatani H, Miyazono K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21764776" target="_blank">
                ChIP-seq reveals cell type-specific binding patterns of BMP-specific Smads and a novel binding motif</a>.
                <em>Nucleic Acids Res</em>. 2011 Nov 1;39(20):8712-27.<br>
                ENCODE histone modification ChIP, Fig 3D, S3C, results, methods
                </p>

                <p>
                Morrissy AS, Griffith M, Marra MA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21719572" target="_blank">
                Extensive relationship between antisense transcription and alternative splicing in the human genome</a>.
                <em>Genome Res</em>. 2011 Aug;21(8):1203-12.<br>
                ENCODE TF ChIP, Fig 4b and text
                </p>

                <p>
                Moseley SC, Rizkallah R, Tremblay DC, Anderson BR, Hurt MM, Chadwick BP.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22064860" target="_blank">
                YY1 associates with the macrosatellite DXZ4 on the inactive X chromosome and binds with CTCF to a hypomethylated form in some male carcinomas</a>.
                <em>Nucleic Acids Res</em>. 2012 Feb;40(4):1596-608.<br>
                ENCODE histone modification ChIP, TF ChIP, Fig 5a
                </p>

                <p>
                Nakken S, R&#248;dland EA, Hovig E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20886615" target="_blank">
                Impact of DNA physical properties on local sequence bias of human mutation</a>.
                <em>Hum Mutat</em>. 2010 Dec;31(12):1316-25.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), Table 1, methods
                </p>

                <p>
                Nelms BL, Labosky PA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22355623" target="_blank">
                A predicted hairpin cluster correlates with barriers to PCR, sequencing and possibly BAC recombineering</a>.
                <em>Sci Rep</em>. 2011;1:106.<br>
                ENCODE histone modification ChIP, DNase, Figure 2
                </p>

                <p>
                Nozaki T, Yachie N, Ogawa R, Saito R, Tomita M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21338662" target="_blank">
                Computational analysis suggests a highly bendable, fragile structure for nucleosomal DNA</a>.
                <em>Gene</em>. 2011 May 1;476(1-2):10-4.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), Fig 2, 3B, methods
                </p>

                <p>
                Oh YM, Kim JK, Choi S, Yoo JY.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22187154" target="_blank">
                Identification of co-occurring transcription factor binding sites from DNA sequence using clustered position weight matrices</a>.
                <em>Nucleic Acids Res</em>. 2012 Mar;40(5):e38.<br>
                ENCODE TF ChIP data used throughout, described in Table S2D
                </p>

                <p>
                Ott CJ, Bischof JM, Unti KM, Gillen AE, Leir SH, Harris A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21948798" target="_blank">
                Nucleosome occupancy reveals regulatory elements of the CFTR promoter</a>.
                <em>Nucleic Acids Res</em>. 2012 Jan;40(2):625-37.<br>
                ENCODE DNase, 28 mammals conservation, Fig 3C, 6B, 6C, 6D, S3
                </p>

                <p>
                Prendergast JG, Tong P, Hay DC, Farrington SM, Semple CA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22607690" target="_blank">
                A genome-wide screen in human embryonic stem cells reveals novel sites of allele-specific histone modification associated with known disease loci</a>.
                <em>Epigenetics Chromatin</em>. 2012 May 19;5(1):6.<br>
                ENCODE RNA, histone modification ChIP, TF ChIP, used throughout, listed in additional file 6
                (also Roadmap Epigenomics RNA, DNAme, histone modification CHIP)
                </p>

                <p>
                Previti C, Harari O, Zwir I, del Val C.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19383127" target="_blank">
                Profile analysis and prediction of tissue-specific CpG island methylation classes</a>.
                <em>BMC Bioinformatics</em>. 2009 Apr 21;10:116.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), Table 4, methods
                </p>

                <p>
                Raab JR, Chiu J, Zhu J, Katzman S, Kurukuti S, Wade PA, Haussler D, Kamakaka RT.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22085927" target="_blank">
                Human tRNA genes function as chromatin insulators</a>.
                <em>EMBO J</em>. 2011 Nov 15;31(2):330-50.<br>
                ENCODE data used in Figure 2A; used in findings presented in text before Figure 1, during Figure 3,
                and maybe part of Figure 8A
                </p>

                <p>
                Rackham O, Shearwood AM, Mercer TR, Davies SM, Mattick JS, Filipovska A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22028365" target="_blank">
                Long noncoding RNAs are generated from the mitochondrial genome and regulated by nuclear-encoded
                proteins</a>.
                <em>RNA</em>. 2011 Dec;17(12):2085-93.<br>
                ENCODE RNA analyzed, described in methods
                </p>

                <p>
                Radwan A, Younis A, Luykx P, Khuri S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18430246" target="_blank">
                Prediction and analysis of nucleosome exclusion regions in the human genome</a>.
                <em>BMC Genomics</em>. 2008 Apr 22;9:186.<br>
                ENCODE Pilot DNase, hydroxyl radical cleavage (ORChiD), Fig 7, 8, methods
                </p>

                <p>
                Raj B, O'Hanlon D, Vessey JP, Pan Q, Ray D, Buckley NJ, Miller FD, Blencowe BJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21884984" target="_blank">
                Cross-regulation between an alternative splicing activator and a transcription repressor controls
                neurogenesis</a>.
                <em>Mol Cell</em>. 2011 Sep 2;43(5):843-50.<br>
                ENCODE TF ChIP, Figure 3C
                </p>

                <p>
                Ren L, Shi M, Wang Y, Yang Z, Wang X, Zhao Z.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21948239" target="_blank">
                CTCF and cohesin cooperatively mediate the cell-type specific interchromatin interaction between
                Bcl11b and Arhgap6 loci</a>.
                <em>Mol Cell Biochem</em>. 2012 Jan;360(1-2):243-51.<br>
                ENCODE TF ChIP, Fig 1B
                </p>

                <p>
                Restivo G, Nguyen BC, Dziunycz P, Ristorcelli E, Ryan RJ, &#214;zuysal &#214;Y, Di Piazza M, Radtke
                F, Dixon MJ, Hofbauer GF <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21909072" target="_blank">
                IRF6 is a mediator of Notch pro-differentiation and tumour suppressive function in
                keratinocytes</a>.
                <em>EMBO J</em>. 2011 Nov 16;30(22):4571-85.<br>
                ENCODE histone modification ChIP, TF ChIP, DNase, Figure 4a
                </p>

                <p>
                Rose D, Stadler PF.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22303364" target="_blank">
                Molecular evolution of the non-coding eosinophil granule ontogeny transcript</a>.
                <em>Front Genet</em>. 2011;2:69.<br>
                ENCODE DNase, histone modification ChIP, RNA, Figure 5, A3, results
                </p>

                <p>
                Ruppert SM, Chehtane M, Zhang G, Hu H, Li X, Khaled AR.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22384197" target="_blank">
                JunD/AP-1-mediated gene expression promotes lymphocyte growth dependent on interleukin-7 signal transduction</a>.
                <em>PLoS One</em>. 2012;7(2):e32262.<br>
                ENCODE TF ChIP, methods, Figure S2, Data S1
                </p>

                <p>
                Schmidt D, Schwalie PC, Wilson MD, Ballester B, Gon&#231;alves A, Kutter C, Brown GD, Marshall A, Flicek P, Odom DT.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22244452" target="_blank">
                Waves of retrotransposon expansion remodel genome organization and CTCF binding in multiple mammalian lineages</a>.
                <em>Cell</em>. 2012 Jan 20;148(1-2):335-48.<br>
                ENCODE TF ChIP, related to Fig 1, S1
                </p>

                <p>
                Sch&#246;del J, Oikonomopoulos S, Ragoussis J, Pugh CW, Ratcliffe PJ, Mole DR.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21447827" target="_blank">
                High-resolution genome-wide mapping of HIF-binding sites by ChIP-seq</a>.
                <em>Blood</em>. 2011 Jun 9;117(23):e207-17.<br>
                ENCODE DNase, Fig 5, S10
                </p>

                <p>
                Shu W, Chen H, Bo X, Wang S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21685456" target="_blank">
                Genome-wide analysis of the relationships between DNaseI HS, histone modifications and gene expression reveals distinct modes of chromatin domains</a>.
                <em>Nucleic Acids Res</em>. 2011 Sep 1;39(17):7428-43.<br>
                ENCODE DNase, Histone modification ChIP, TF ChIP, GENCODE genes, used throughout
                </p>

                <p>
                Smith AM, Calero-Nieto FJ, Sch&#252;tte J, Kinston S, Timms RT, Wilson NK, Hannah RL, Landry JR,
                G&#246;ttgens B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22158964" target="_blank">
                Integration of Elf-4 into stem/progenitor and erythroid regulatory networks through locus-wide
                chromatin studies coupled with in vivo functional validation</a>.
                <em>Mol Cell Biol</em>. 2012 Feb;32(4):763-73.<br>
                ENCODE histone modification ChIP, DNase, TF ChIP, mouse and human, referred to in text as Figure S3, S4; S5 at
                <a href="http://hscl.cimr.cam.ac.uk/genomic_supplementary.html" target="_blank">
                http://hscl.cimr.cam.ac.uk/genomic_supplementary.html</a> which links to the data at
                <a href="http://hscl.cimr.cam.ac.uk/Supplementary_Data/Supplementary_data_Smith.pdf" target="_blank">
                http://hscl.cimr.cam.ac.uk/Supplementary_Data/Supplementary_data_Smith.pdf</a>
                </p>

                <p>
                Somel M, Liu X, Tang L, Yan Z, Hu H, Guo S, Jiang X, Zhang X, Xu G, Xie G <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22162950" target="_blank">
                MicroRNA-driven developmental remodeling in the brain distinguishes humans from other primates</a>.
                <em>PLoS Biol</em>. 2011 Dec;9(12):e1001214.<br>
                ENCODE histone modication ChIP, DNase, Fig S9
                </p>

                <p>
                Stadler MB, Murr R, Burger L, Ivanek R, Lienert F, Sch&#246;ler A, van Nimwegen E, Wirbelauer C,
                Oakeley EJ, Gaidatzis D <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22170606" target="_blank">
                DNA-binding factors shape the mouse methylome at distal regulatory regions</a>.
                <em>Nature</em>. 2011 Dec 14;480(7378):490-5.<br>
                ENCODE mouse and human DNase, Fig 2, S2, S6
                </p>

                <p>
                Stoltzfus A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18597793" target="_blank">
                Evidence for a predominant role of oxidative damage in germline mutation in mammals</a>.
                <em>Mutat Res</em>. 2008 Sep 26;644(1-2):71-3.<br>
                ENCODE hydroxyl radical cleavage (ORChiD), Fig 1, 2
                </p>

                <p>
                Taft RJ, Hawkins PG, Mattick JS, Morris KV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21813016" target="_blank">
                The relationship between transcription initiation RNAs and CCCTC-binding factor (CTCF)
                localization</a>.
                <em>Epigenetics Chromatin</em>. 2011 Aug 3;4:13.<br>
                ENCODE TF ChIP, Figure 2
                </p>

                <p>
                Tanaka Y, Nakamura A, Morioka MS, Inoue S, Tamamori-Adachi M, Yamada K, Taketani K, Kawauchi J, Tanaka-Okamoto M, Miyoshi J <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22046379" target="_blank">
                Systems analysis of ATF3 in stress response and cancer reveals opposing effects on pro-apoptotic genes in p53 pathway</a>.
                <em>PLoS One</em>. 2011;6(10):e26848.<br>
                ENCODE TF ChIP, used in discussion
                </p>

                <p>
                Taylor J, Schenck I, Blankenberg D, Nekrutenko A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18428782" target="_blank">
                Using galaxy to perform large-scale interactive data analyses</a>.
                <em>Curr Protoc Bioinformatics</em>. 2007 Sep;Chapter 10:Unit 10.5.<br>
                ENCODE TF ChIP, used in basic protocol 2, 3
                </p>

                <p>
                Teng L, Firpi HA, Tan K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21685454" target="_blank">
                Enhancers in embryonic stem cells are enriched for transposable elements and genetic variations associated with cancers</a>.
                <em>Nucleic Acids Res</em>. 2011 Sep 1;39(17):7371-9.<br>
                ENCODE histone modification ChIP, TF ChIP, DNase, data analyzed
                </p>

                <p>
                Terrenoire E, McRonald F, Halsall JA, Page P, Illingworth RS, Taylor AM, Davison V, O'Neill LP,
                Turner BM.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21078160" target="_blank">
                Immunostaining of modified histones defines high-level features of the human metaphase
                epigenome</a>.
                <em>Genome Biol</em>. 2010;11(11):R110.<br>
                ENCODE histone modification ChIP, Figure 5
                </p>

                <p>
                Tiana M, Villar D, P&#233;rez-Guijarro E, G&#243;mez-Maldonado L, Molt&#243; E, Fern&#225;ndez-Mi&#241;&#225;n A, G&#243;mez-Skarmeta JL, Montoliu L, del Peso L.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22067454" target="_blank">
                A role for insulator elements in the regulation of gene expression response to hypoxia</a>.
                <em>Nucleic Acids Res</em>. 2012 Mar;40(5):1916-27.<br>
                ENCODE TF ChIP, Fig S1, S4
                </p>

                <p>
                Tsai KN, Wang D.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22323516" target="_blank">
                Identification of activated cryptic 5' splice sites using structure profiles and odds measure</a>.
                <em>Nucleic Acids Res</em>. 2012 May;40(10):e73.<br>
                ENCODE Pilot hydroxyl radical cleavage (ORChiD), results, methods
                </p>

                <p>
                Wang YM, Zhou P, Wang LY, Li ZH, Zhang YN, Zhang YX.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22900019" target="_blank">
                Correlation between DNase I hypersensitive site distribution and gene expression in HeLa S3 cells</a>.
                <em>PLoS One</em>. 2012;7(8):e42414.<br>
                ENCODE DNase, TF ChIP, Fig S2, Tables S7, S8
                </p>

                <p>
                Woodfine K, Huddleston JE, Murrell A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21281512" target="_blank">
                Quantitative analysis of DNA methylation at all human imprinted regions reveals preservation of
                epigenetic stability in adult somatic tissue</a>.
                <em>Epigenetics Chromatin</em>. 2011 Jan 31;4(1):1.<br>
                ENCODE TF ChIP, Table 1, Figure S3
                </p>

                <p>
                You JS, Kelly TK, De Carvalho DD, Taberlay PC, Liang G, Jones PA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21844352" target="_blank">
                OCT4 establishes and maintains nucleosome-depleted regions that provide additional layers of epigenetic regulation of its target genes</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Aug 30;108(35):14497-502.<br>
                ENCODE TF ChIP, Methyl-Seq, Fig 4C
                </p>

                <p>
                Zemojtel T, Kielbasa SM, Arndt PF, Behrens S, Bourque G, Vingron M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22016335" target="_blank">
                CpG deamination creates transcription factor-binding sites with high efficiency</a>.
                <em>Genome Biol Evol</em>. 2011;3:1304-11.<br>
                ENCODE TF ChIP, described in text, used in Table S2, and Supplementary material
                </p>

            </div><!--end content-->
        </div><!--end wrapper-->
       
        <div class="wrapper">
            <div class="bar">
                <a name="tools"></a>
                <h4 class="title">Tools, methods, databases and review/commentary publications (46)</h4>
            </div>
            <div class="content">
                <p>
                Abeel T, Saeys Y, Bonnet E, Rouz&#233; P, Van de Peer Y.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18096745" target="_blank">
                Generic eukaryotic core promoter prediction using structural features of DNA</a>.
                <em>Genome Res</em>. 2008 Feb;18(2):310-23.<br>
                ENCODE Pilot DNase, GENCODE genes, described in results
                </p>

                <p>
                Althammer S, Gonz&#225;lez-Vallinas J, Ballar&#233; C, Beato M, Eyras E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21994224" target="_blank">
                Pyicos: a versatile toolkit for the analysis of high-throughput sequencing data</a>.
                <em>Bioinformatics</em>. 2011 Dec 15;27(24):3333-40.<br>
                ENCODE TF ChIP, histone modification ChIP, RNA, Fig 1, Table 1, S6
                </p>

                <p>
                Althammer S, Pag&#232;s A, Eyras E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22924024" target="_blank">
                Predictive models of gene regulation from high-throughput epigenomics data</a>.
                <em>Comp Funct Genomics</em>. 2012;2012:284786.<br>
                ENCODE RNA, DNase, Histone modification ChIP, TF ChIP, Methyl-RRBS, GENCODE, used throughout, Table 2
                </p>

                <p>
                Anders S, Reyes A, Huber W.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22722343" target="_blank">
                Detecting differential usage of exons from RNA-seq data</a>.
                <em>Genome Res</em>. 2012 Oct;22(10):2008-17.<br>
                ENCODE RNA, Fig S5, S6
                </p>

                <p>
                Bedo J, Kowalczyk A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21558156" target="_blank">
                Genome annotation test with validation on transcription start site and ChIP-Seq for Pol-II binding
                data</a>.
                <em>Bioinformatics</em>. 2011 Jun 15;27(12):1610-7.<br>
                ENCODE TF CHIP, Table 1, used throughout
                </p>

                <p>
                Berlanga-Taylor AJ, Disanto G, Ebers GC, Ramagopalan SV.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22000399" target="_blank">
                Vitamin D-gene interactions in multiple sclerosis</a>.
                <em>J Neurol Sci</em>. 2011 Dec 15;311(1-2):32-6.<br>
                ENCODE histone modifcation ChIP in text, comparison of ENCODE His-ChIP-seq and VDR described
                </p>

                <p>
                Bhattacharyya S, Tian J, Bouhassira EE, Locker J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21858039" target="_blank">
                Systematic targeted integration to study Albumin gene control elements</a>.
                <em>PLoS One</em>. 2011;6(8):e23234.<br>
                ENCODE TF ChIP, figure 8 legend
                </p>

                <p>
                Boeva V, Surdez D, Guillon N, Tirode F, Fejes AP, Delattre O, Barillot E.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20375099" target="_blank">
                De novo motif identification improves the accuracy of predicting transcription factor binding sites
                in ChIP-Seq data analysis</a>.
                <em>Nucleic Acids Res</em>. 2010 Jun;38(11):e126.<br>
                ENCODE TF ChIP, Figure 2, 3
                </p>

                <p>
                Cai X, Hou L, Su N, Hu H, Deng M, Li X.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20946653" target="_blank">
                Systematic identification of conserved motif modules in the human genome</a>.
                <em>BMC Genomics</em>. 2010 Oct 14;11:567.<br>
                ENCODE TF ChIP, methods, used throughout, additional file 1
                </p>

                <p>
                Carstensen L, Sandelin A, Winther O, Hansen NR.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20828413" target="_blank">
                Multivariate Hawkes process models of the occurrence of regulatory elements</a>.
                <em>BMC Bioinformatics</em>. 2010 Sep 9;11:456.<br>
                ENCODE TF ChIP, Histone modification ChIP, Figures 5, 6, 7, 8, 9
                </p>

                <p>
                Chang HW, Cheng YH, Chuang LY, Yang CH.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20377871" target="_blank">
                SNP-RFLPing 2: an updated and integrated PCR-RFLP tool for SNP genotyping</a>.
                <em>BMC Bioinformatics</em>. 2010 Apr 8;11:173.<br>
                Java application for annotating SNPs, includes ENCODE annotations
                </p>

                <p>
                Chen Y, J&#248;rgensen M, Kolde R, Zhao X, Parker B, Valen E, Wen J, Sandelin A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22047616" target="_blank">
                Prediction of RNA Polymerase II recruitment, elongation and stalling from histone modification
                data</a>.
                <em>BMC Genomics</em>. 2011 Nov 3;12:544.<br>
                ENCODE histone modification ChIP, TF ChIP, RNA, used throughout; scheme in figure 1
                </p>

                <p>
                Coetzee SG, Rhie SK, Berman BP, Coetzee GA, Noushmehr H.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22684628" target="_blank">
                FunciSNP: an R/bioconductor tool integrating functional non-coding data sets with genetic association studies to identify candidate regulatory SNPs</a>.
                <em>Nucleic Acids Res</em>. 2012 Oct 1;40(18):e139.<br>
                ENCODE TF ChIP, DNase, FAIRE, Histone modification ChIP, Fig 3, 4, S1, S2, Supplement
                </p>

                <p>
                Ding J, Bashashati A, Roth A, Oloumi A, Tse K, Zeng T, Haffari G, Hirst M, Marra MA, Condon A <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22084253" target="_blank">
                Feature-based classifiers for somatic mutation detection in tumour-normal paired sequencing data</a>.
                <em>Bioinformatics</em>. 2012 Jan 15;28(2):167-75.<br>
                ENCODE mappability, results and Fig S8
                </p>

                <p>
                Dozmorov MG, Cara LR, Giles CB, Wren JD.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22155868" target="_blank">
                GenomeRunner: automating genome exploration</a>.
                <em>Bioinformatics</em>. 2012 Feb 1;28(3):419-20.<br>
                ENCODE data used in analyses, Picture on their website, supplement
                </p>

                <p>
                Enroth S, Andersson CR, Andersson R, Wadelius C, Gustafsson MG, Komorowski J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22248020" target="_blank">
                A strand specific high resolution normalization method for chip-sequencing data employing multiple
                experimental control measurements</a>.
                <em>Algorithms Mol Biol</em>. 2012 Jan 16;7(1):2.<br>
                ENCODE TF ChIP; Table 1, Figure 2
                </p>

                <p>
                Firpi HA, Ucar D, Tan K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20453004" target="_blank">
                Discover regulatory DNA elements using chromatin signatures and artificial neural network</a>.
                <em>Bioinformatics</em>. 2010 Jul 1;26(13):1579-86.<br>
                ENCODE histone modification ChIP, Table 2
                </p>

                <p>
                Halder K, Halder R, Chowdhury S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19587895" target="_blank">
                Genome-wide analysis predicts DNA structural motifs as nucleosome exclusion signals</a>.
                <em>Mol Biosyst</em>. 2009 Dec;5(12):1703-12.<br>
                ENCODE DNase, FAIRE
                </p>

                <p>
                H&#229;ndstad T, Rye MB, Drabl&#248;s F, S&#230;trom P.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21533218" target="_blank">
                A ChIP-Seq benchmark shows that sequence conservation mainly improves detection of strong
                transcription factor binding sites</a>.
                <em>PLoS One</em>. 2011 Apr 14;6(4):e18430.<br>
                ENCODE TF-ChIP
                </p>

                <p>
                Huda A, Tyagi E, Mari&#241;o-Ram&#237;rez L, Bowen NJ, Jjingo D, Jordan IK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22087331" target="_blank">
                Prediction of transposable element derived enhancers using chromatin modification profiles</a>.
                <em>PLoS One</em>. 2011;6(11):e27513.<br>
                ENCODE histone modification ChIP, DNase, RNA, analyzed throughout
                </p>

                <p>
                Lee D, Karchin R, Beer MA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21875935" target="_blank">
                Discriminative prediction of mammalian enhancers from DNA sequence</a>.
                <em>Genome Res</em>. 2011 Dec;21(12):2167-80.<br>
                mouse ENCODE DNase, Figure 5, Table 2
                </p>

                <p>
                Maeder ML, Thibodeau-Beganny S, Osiak A, Wright DA, Anthony RM, Eichtinger M, Jiang T, Foley JE,
                Winfrey RJ, Townsend JA <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18657511" target="_blank">
                Rapid "open-source" engineering of customized zinc-finger nucleases for highly efficient gene
                modification</a>.
                <em>Mol Cell</em>. 2008 Jul 25;31(2):294-301.<br>
                ENCODE histone modification ChIP, discussion
                </p>

                <p>
                Martin P, Barton A, Eyre S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21177990" target="_blank">
                ASSIMILATOR: a new tool to inform selection of associated genetic variants for functional
                studies</a>.
                <em>Bioinformatics</em>. 2011 Jan 1;27(1):144-6.<br>
                ENCODE Histone modification ChIP, TF ChIP, RNA, DNase, GENCODE Figure 1A
                </p>

                <p>
                McLeay RC, Leat CJ, Bailey TL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21724591" target="_blank">
                Tissue-specific prediction of directly regulated genes</a>.
                <em>Bioinformatics</em>. 2011 Sep 1;27(17):2354-60.<br>
                ENCODE TF ChIP, Histone modification ChIP, Table 1, Table 2
                </p>

                <p>
                Mittal VK, McDonald JF.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22287631" target="_blank">
                R-SAP: a multi-threading computational pipeline for the characterization of high-throughput RNA-sequencing data</a>.
                <em>Nucleic Acids Res</em>. 2012 May;40(9):e67.<br>
                ENCODE RNA, used in Table 5, Fig 6, 7
                </p>

                <p>
                Mortlock DP, Pregizer S.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22241655" target="_blank">
                Identifying functional annotation for noncoding genomic sequences</a>.
                <em>Curr Protoc Hum Genet</em>. 2012 Jan;Chapter 1:Unit1.10.<br>
                ENCODE RNA, histone modification ChIP, DNase, TF ChIP, Figure 1, 2
                </p>

                <p>
                Nix DA, Di Sera TL, Dalley BK, Milash BA, Cundick RM, Quinn KS, Courdy SJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20828407" target="_blank">
                Next generation tools for genomic data generation, distribution, and visualization</a>.
                <em>BMC Bioinformatics</em>. 2010 Sep 9;11:455.<br>
                ENCODE Histone modification ChIP, Fig 11
                </p>

                <p>
                Pickrell JK, Gaffney DJ, Gilad Y, Pritchard JK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21690102" target="_blank">
                False positive peaks in ChIP-seq and other sequencing-based functional assays caused by unannotated
                high copy number regions</a>.
                <em>Bioinformatics</em>. 2011 Aug 1;27(15):2144-6.<br>
                ENCODE TF ChIP, Figure 1
                </p>

                <p>
                Pinello L, Lo Bosco G, Hanlon B, Yuan GC.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22017798" target="_blank">
                A motif-independent metric for DNA sequence specificity</a>.
                <em>BMC Bioinformatics</em>. 2011 Oct 21;12:408.<br>
                ENCODE TF ChIP, histone modification ChIP, Fig 2, 3, Table 3, 4, 5
                </p>

                <p>
                Pique-Regi R, Degner JF, Pai AA, Gaffney DJ, Gilad Y, Pritchard JK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21106904" target="_blank">
                Accurate inference of transcription factor binding from DNA sequence and chromatin accessibility
                data</a>.
                <em>Genome Res</em>. 2011 Mar;21(3):447-55.<br>
                ENCODE DNase, histone modification ChIP, TF ChIP, Fig 1, 2, 3, 6 etc
                </p>

                <p>
                Rolfe PA, Gifford DK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21736741" target="_blank">
                ReadDB provides efficient storage for mapped short reads</a>.
                <em>BMC Bioinformatics</em>. 2011 Jul 7;12:278.<br>
                ENCODE DNase, CTCF ChIP, described in text and used throughout
                </p>

                <p>
                Rye M, S&#230;trom P, H&#229;ndstad T, Drabl&#248;s F.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22115494" target="_blank">
                Clustered ChIP-Seq-defined transcription factor binding sites and histone modifications map distinct
                classes of regulatory elements</a>.
                <em>BMC Biol</em>. 2011 Nov 24;9:80.<br>
                ENCODE TF ChIP, DNase, FAIRE, histone modification ChIP, RNA, used throughout; summarized in Table S7
                </p>

                <p>
                Sadri J, Diallo AB, Blanchette M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21685080" target="_blank">
                Predicting site-specific human selective pressure using evolutionary signatures</a>.
                <em>Bioinformatics</em>. 2011 Jul 1;27(13):i266-74.<br>
                ENCODE DNase, histone modification ChIP, figure 4
                </p>

                <p>
                Singer M, Boffelli D, Dhahbi J, Sch&#246;nhuth A, Schroth GP, Martin DI, Pachter L.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20856582" target="_blank">
                MetMap enables genome-scale Methyltyping for determining methylation states in populations</a>.
                <em>PLoS Comput Biol</em>. 2010 Aug 19;6(8):e1000888.<br>
                ENCODE FAIRE data, Table 3
                </p>

                <p>
                Smith AJ, Howard P, Shah S, Eriksson P, Stender S, Giambartolomei C, Folkersen L, Tybj&#230;rg-Hansen A, Kumari M, Palmen J <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22916038" target="_blank">
                Use of allele-specific FAIRE to determine functional regulatory polymorphism using large-scale genotyping arrays</a>.
                <em>PLoS Genet</em>. 2012 Aug;8(8):e1002908.<br>
                ENCODE TF ChIP, DNase, Histone modification ChIP, FAIRE, Figure 4
                </p>

                <p>
                Su J, Teichmann SA, Down TA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21152003" target="_blank">
                Assessing computational methods of cis-regulatory module prediction</a>.
                <em>PLoS Comput Biol</em>. 2010 Dec 2;6(12):e1001020.<br>
                ENCODE DNase, ENCODE regions, figures 9,10
                </p>

                <p>
                Sulonen AM, Ellonen P, Almusa H, Lepist&#246; M, Eldfors S, Hannula S, Miettinen T, Tyynismaa H,
                Salo P, Heckman C <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21955854" target="_blank">
                Comparison of solution-based exome capture methods for next generation sequencing</a>.
                <em>Genome Biol</em>. 2011 Sep 28;12(9):R94.<br>
                ENCODE mappability, described in text

                <p>
                Taylor J, Schenck I, Blankenberg D, Nekrutenko A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/18428782" target="_blank">
                Using galaxy to perform large-scale interactive data analyses</a>.
                <em>Curr Protoc Bioinformatics</em>. 2007 Sep;Chapter 10:Unit 10.5.<br>
                ENCODE TF ChIP, used in basic protocol 2, 3
                </p>
                </p>

                <p>
                Valen E, Sandelin A, Winther O, Krogh A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19911049" target="_blank">
                Discovery of regulatory elements is improved by a discriminatory approach</a>.
                <em>PLoS Comput Biol</em>. 2009 Nov;5(11):e1000562.<br>
                ENCODE TF ChIP, table S7
                </p>

                <p>
                Veiga DF, Deus HF, Akdemir C, Vasconcelos AT, Almeida JS.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19919683" target="_blank">
                DASMiner: discovering and integrating data from DAS sources</a>.
                <em>BMC Syst Biol</em>. 2009 Nov 17;3:109.<br>
                ENCODE histone modification ChIP, figure 4
                </p>

                <p>
                Wang K, Li M, Hakonarson H.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20601685" target="_blank">
                ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data</a>.
                <em>Nucleic Acids Res</em>. 2010 Sep;38(16):e164.<br>
                ENCODE DNase, Histone modification ChIP, TF ChIP, GENCODE genes, ChIP-Seq peaks, RNA-Seq peaks,
                or many other annotations on genomic intervals, reported on web site
                </p>

                <p>
                Whitington T, Frith MC, Johnson J, Bailey TL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21602262" target="_blank">
                Inferring transcription factor complexes from ChIP-seq data</a>.
                <em>Nucleic Acids Res</em>. 2011 Aug;39(15):e98.<br>
                ENCODE TF ChIP, Table 1, 2, 3, S1, S3, S4
                </p>

                <p>
                Yaffe E, Tanay A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22001755" target="_blank">
                Probabilistic modeling of Hi-C contact maps eliminates systematic biases to characterize global
                chromosomal architecture</a>.
                <em>Nat Genet</em>. 2011 Oct 16;43(11):1059-65.<br>
                ENCODE DNase, histone modification ChIP, TF ChIP, Figure 3
                </p>

                <p>
                Zambelli F, Prazzoli GM, Pesole G, Pavesi G.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22669907" target="_blank">
                Cscan: finding common regulators of a set of genes by using a collection of genome-wide ChIP-seq
                datasets</a>.
                <em>Nucleic Acids Res</em>. 2012 Jul;40(Web Server issue):W510-5.<br>
                ENCODE TF ChIP, histone modification ChIP, chromatin states
                </p>

                <p>
                Zhang Y, Lv J, Liu H, Zhu J, Su J, Wu Q, Qi Y, Wang F, Li X.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19892823" target="_blank">
                HHMD: the human histone modification database</a>.
                <em>Nucleic Acids Res</em>. 2010 Jan;38(Database issue):D149-54.<br>
                ENCODE histone modification ChIP, Table 2 and database
                </p>

                <p>
                Zhang Z, Zhang MQ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21569556" target="_blank">
                Histone modification profiles are predictive for tissue/cell-type specific expression of both
                protein-coding and microRNA genes</a>.
                <em>BMC Bioinformatics</em>. 2011 May 14;12:155.<br>
                ENCODE RNA, histone modification ChIP, methyl-seq, figure 3, table 3
                </p>
            </div><!--end content-->
        </div><!--end wrapper-->

        <div class="wrapper">
            <div class="bar">
                <a name="modencode"></a>
                <h4 class="title">Publications using modENCODE data, without modENCODE authors (36)</h4>
            </div>
            <div class="content">
                <p>
                Ahn SJ, Vogel H, Heckel DG.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22155036" target="_blank">
                Comparative analysis of the UDP-glycosyltransferase multigene family in insects</a>.
                <em>Insect Biochem Mol Biol</em>. 2012 Feb;42(2):133-47.<br>
                modENCODE fly RNA-seq data were used
                </p>

                <p>
                Bushey AM, Ramos E, Corces VG.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19443682" target="_blank">
                Three subclasses of a Drosophila insulator show distinct and cell type-specific genomic distributions</a>.
                <em>Genes Dev</em>. 2009 Jun 1;23(11):1338-50.<br>
                modENCODE Fly S2 ChIP, CP190 and dCTCF
                </p>

                <p>
                Cardoso-Moreira M, Emerson JJ, Clark AG, Long M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22072977" target="_blank">
                Drosophila duplication hotspots are associated with late-replicating regions of the genome</a>.
                <em>PLoS Genet</em>. 2011 Nov;7(11):e1002340.<br>
                modENCODE Fly Replication timing maps, Kc, S2, Bg3, were used
                </p>

                <p>
                Cernilogar FM, Onorati MC, Kothe GO, Burroughs AM, Parsi KM, Breiling A, Lo Sardo F, Saxena A, Miyoshi K, Siomi H <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22056986" target="_blank">
                Chromatin-associated RNA interference components contribute to transcriptional regulation in Drosophila</a>.
                <em>Nature</em>. 2011 Nov 6;480(7377):391-5.<br>
                modENCODE Fly CAGE data used to determine antisense transcription frequent at heat shock loci
                </p>

                <p>
                Cline TW, Dorsett M, Sun S, Harrison MM, Dines J, Sefton L, Megna L.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20837995" target="_blank">
                Evolution of the Drosophila feminizing switch gene Sex-lethal</a>.
                <em>Genetics</em>. 2010 Dec;186(4):1321-36.<br>
                modENCODE Fly RNA-seq splice junction data used
                </p>

                <p>
                Cryderman DE, Vitalini MW, Wallrath LL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21468237" target="_blank">
                Heterochromatin protein 1a is required for an open chromatin structure</a>.
                <em>Transcription</em>. 2011 Mar;2(2):95-99.<br>
                modENCODE fly chromatin states, histone modifications, expression
                </p>

                <p>
                DeMaso CR, Kovacevic I, Uzun A, Cram EJ.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21799850" target="_blank">
                Structural and functional evaluation of C. elegans filamins FLN-1 and FLN-2</a>.
                <em>PLoS One</em>. 2011;6(7):e22428.<br>
                modENCODE worm data used, SL1 binding, RNA-seq data used for expression and exon/isoform determination
                </p>

                <p>
                Feller C, Prestel M, Hartmann H, Straub T, S&#246;ding J, Becker PB.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22039099" target="_blank">
                The MOF-containing NSL complex associates globally with housekeeping genes, but activates only a defined subset</a>.
                <em>Nucleic Acids Res</em>. 2012 Feb;40(4):1509-22.<br>
                modENCODE Fly data used; Figure 1, Figure 2, Figure 5; Pol II ChIP, chromatin factors and histone
                modifications, and "all available modENCODE chromatin ChIP-chip data sets were screened"
                </p>

                <p>
                Fischer SE, Montgomery TA, Zhang C, Fahlgren N, Breen PC, Hwang A, Sullivan CM, Carrington JC, Ruvkun G.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22102828" target="_blank">
                The ERI-6/7 helicase acts at the first stage of an siRNA amplification pathway that targets recent gene duplications</a>.
                <em>PLoS Genet</em>. 2011 Nov;7(11):e1002369.<br>
                modENCODE Worm data; intron/exon structures of genes from RNA-seq
                </p>

                <p>
                Gangishetti U, Veerkamp J, Bezdan D, Schwarz H, Lohmann I, Moussian B.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22458773" target="_blank">
                The transcription factor Grainy head and the steroid hormone ecdysone cooperate during differentiation of the skin of Drosophila melanogaster</a>.
                <em>Insect Mol Biol</em>. 2012 Jun;21(3):283-95.<br>
                modENCODE fly data on bFtzF1 binding, used in Fig 7
                </p>

                <p>
                Gilchrist DA, Adelman K.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22406341" target="_blank">
                Coupling polymerase pausing and chromatin landscapes for precise regulation of transcription</a>.
                <em>Biochim Biophys Acta</em>. 2012 Jul;1819(7):700-6.<br>
                modENCODE fly histone modifications, chromatin states, Pol II binding used in Figure 3B
                </p>

                <p>
                Greer EL, Maures TJ, Ucar D, Hauswirth AG, Mancini E, Lim JP, Benayoun BA, Shi Y, Brunet A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22012258" target="_blank">
                Transgenerational epigenetic inheritance of longevity in Caenorhabditis elegans</a>.
                <em>Nature</em>. 2011 Oct 19;479(7373):365-71.<br>
                modENCODE Worm H3K4me3 data used
                </p>

                <p>
                Ihuegbu NE, Stormo GD, Buhler J.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22300316" target="_blank">
                Fast, sensitive discovery of conserved genome-wide motifs</a>.
                <em>J Comput Biol</em>. 2012 Feb;19(2):139-47.<br>
                modENCODE worm TF HOT regions; data used for Table 3
                </p>

                <p>
                Katzemich A, Long JY, Jani K, Lee BR, Sch&#246;ck F.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21867777" target="_blank">
                Muscle type-specific expression of Zasp52 isoforms in Drosophila</a>.
                <em>Gene Expr Patterns</em>. 2011 Dec;11(8):484-90.<br>
                modENCODE fly data mentioned in discussion, used to confirm exons
                </p>

                <p>
                Kellner WA, Ramos E, Van Bortle K, Takenaka N, Corces VG.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22508764" target="_blank">
                Genome-wide phosphoacetylation of histone H3 at Drosophila enhancers and promoters</a>.
                <em>Genome Res</em>. 2012 Jun;22(6):1081-8.<br>
                modENCODE data: Fig 1A, Pol II ChIP-chip, Fig S4, ChIP-chip, RNA used in Figure 6 analysis; refer
                to antibodies as validated in modENCODE
                </p>

                <p>
                Kvon EZ, Stampfel G, Y&#225;&#241;ez-Cuna JO, Dickson BJ, Stark A.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22499593" target="_blank">
                HOT regions function as patterned developmental enhancers and have a distinct cis-regulatory signature</a>.
                <em>Genes Dev</em>. 2012 May 1;26(9):908-13.<br>
                modENCODE fly HOT regions were analyzed in this work; modENCODE RNA-seq used; modENCODE tracks in Fig 2B
                </p>

                <p>
                Majumdar A, Cesario WC, White-Grindley E, Jiang H, Ren F, Khan MR, Li L, Choi EM, Kannan K, Guo F <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22284910" target="_blank">
                Critical role of amyloid-like oligomers of Drosophila Orb2 in the persistence of memory</a>.
                <em>Cell</em>. 2012 Feb 3;148(3):515-29.<br>
                modENCODE Fly data used to determine Orb2 encodes 6 isoforms (S1B)
                </p>

                <p>
                McQuilton P, St Pierre SE, Thurmond J, FlyBase Consortium.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22127867" target="_blank">
                FlyBase 101--the basics of navigating FlyBase</a>.
                <em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D706-14.<br>
                modENCODE fly genome-wide data available
                </p>

                <p>
                Morton JJ, Blumenthal T.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21156961" target="_blank">
                Identification of transcription start sites of trans-spliced genes: uncovering unusual operon arrangements</a>.
                <em>RNA</em>. 2011 Feb;17(2):327-37.<br>
                modENCODE Worm Htz and Pol II ChIP, 3. end formation sites, and RNA-seq used
                </p>

                <p>
                Nowak SJ, Aihara H, Gonzalez K, Nibu Y, Baylies MK.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22396663" target="_blank">
                Akirin links twist-regulated transcription with the Brahma chromatin remodeling complex during embryogenesis</a>.
                <em>PLoS Genet</em>. 2012;8(3):e1002547.<br>
                modENCODE fly histone modification data
                </p>

                <p>
                Noyes A, Stefaniuk C, Cheng Y, Kennison JA, Kassis JA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22384357" target="_blank">
                Modulation of the activity of a polycomb-group response element in Drosophila by a mutation in the transcriptional activator woc</a>.
                <em>G3 (Bethesda)</em>. 2011 Nov;1(6):471-8.<br>
                modENCODE fly HP1c ChIP data were used
                </p>

                <p>
                Paro S, Li X, O'Connell MA, Keegan LP.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21761288" target="_blank">
                Regulation and functions of ADAR in drosophila</a>.
                <em>Curr Top Microbiol Immunol</em>. 2012;353:221-36.<br>
                modENCODE fly data to supplement published data on Adar gene transcription, and splicing
                </p>

                <p>
                Pearson JC, Watson JD, Crews ST.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22537497" target="_blank">
                Drosophila melanogaster Zelda and Single-minded collaborate to regulate an evolutionarily dynamic CNS midline cell enhancer</a>.
                <em>Dev Biol</em>. 2012 Jun 15;366(2):420-32.<br>
                modENCODE fly RNA-Seq data, timecourse, analyzed to determine transcription start, stop,
                splice sites, and expression pattern
                </p>

                <p>
                Peng HW, Slattery M, Mann RS.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/19762509" target="_blank">
                Transcription factor choice in the Hippo signaling pathway: homothorax and yorkie regulation of the microRNA bantam in the progenitor domain of the Drosophila eye imaginal disc</a>.
                <em>Genes Dev</em>. 2009 Oct 1;23(19):2307-19.<br>
                modENCODE Fly data used to find TSS for bantam; Figure 6C
                </p>

                <p>
                Rodriguez J, Menet JS, Rosbash M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22658416" target="_blank">
                Nascent-seq indicates widespread cotranscriptional RNA editing in Drosophila</a>.
                <em>Mol Cell</em>. 2012 Jul 13;47(1):27-37.<br>
                modENCODE fly editing data used, fig 3A, text
                </p>

                <p>
                Rogers RL, Hartl DL.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21771717" target="_blank">
                Chimeric genes as a source of rapid evolution in Drosophila melanogaster</a>.
                <em>Mol Biol Evol</em>. 2012 Feb;29(2):517-29.<br>
                modENCODE Fly RNA-seq data used
                </p>

                <p>
                Rotstein B, Molnar D, Adryan B, Llimargas M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22216153" target="_blank">
                Tramtrack is genetically upstream of genes controlling tracheal tube size in Drosophila</a>.
                <em>PLoS One</em>. 2011;6(12):e28985.<br>
                modENCODE Fly data used, ChIP for Tramtrack, Figure 4, Figure S4
                </p>

                <p>
                Sakoparnig T, Kockmann T, Paro R, Beisel C, Beerenwinkel N.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22300315" target="_blank">
                Binding profiles of chromatin-modifying proteins are predictive for transcriptional activity and promoter-proximal pausing</a>.
                <em>J Comput Biol</em>. 2012 Feb;19(2):126-38.<br>
                modENCODE Fly S2 TF, chromatin organizers, insulator protein data used, also RNA-seq
                </p>

                <p>
                Schrider DR, Stevens K, Carde&#241;o CM, Langley CH, Hahn MW.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22135405" target="_blank">
                Genome-wide analysis of retrogene polymorphisms in Drosophila melanogaster</a>.
                <em>Genome Res</em>. 2011 Dec;21(12):2087-95.<br>
                modENCODE Fly data, cDNA, EST, and RNA-seq data, used for intron/exon structure of some genes
                </p>

                <p>
                Spradling AC, Bellen HJ, Hoskins RA.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/21896744" target="_blank">
                Drosophila P elements preferentially transpose to replication origins</a>.
                <em>Proc Natl Acad Sci U S A</em>. 2011 Sep 20;108(38):15948-53.<br>
                modENCODE fly ORC data used to identify replication origins
                </p>

                <p>
                Vatolina TY, Boldyreva LV, Demakova OV, Demakov SA, Kokoza EB, Semeshin VF, Babenko VN, Goncharov FP, Belyaeva ES, Zhimulev IF.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22022482" target="_blank">
                Identical functional organization of nonpolytene and polytene chromosomes in Drosophila melanogaster</a>.
                <em>PLoS One</em>. 2011;6(10):e25960.<br>
                modENCODE Fly ChIP data used in Figure 2, 6, chromatin states also used
                </p>

                <p>
                Vujatovic O, Zaragoza K, Vaquero A, Reina O, Bernu&#233;s J, Azor&#237;n F.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22406835" target="_blank">
                Drosophila melanogaster linker histone dH1 is required for transposon silencing and to preserve genome integrity</a>.
                <em>Nucleic Acids Res</em>. 2012 Jul;40(12):5402-14.<br>
                modENCODE Fly H1 ChIP data, S2 and BG3 cells
                </p>

                <p>
                Wallace HA, Plata MP, Kang HJ, Ross M, Labrador M.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/20033198" target="_blank">
                Chromatin insulators specifically associate with different levels of higher-order chromatin organization in Drosophila</a>.
                <em>Chromosoma</em>. 2010 Apr;119(2):177-94.<br>
                modENCODE FLY data were used; ChIP-chip Su(Hw), CP190, BEAF-32, CTCF
                </p>

                <p>
                Xiao T, Wang Y, Luo H, Liu L, Wei G, Chen X, Sun Y, Chen X, Skogerb&#248; G, Chen R.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22345127" target="_blank">
                A differential sequencing-based analysis of the C. elegans noncoding transcriptome</a>.
                <em>RNA</em>. 2012 Apr;18(4):626-39.<br>
                modENCODE Worm data used; RNA-seq, H3K4me3, 22 TFs
                </p>

                <p>
                Yook K, Harris TW, Bieri T, Cabunoc A, Chan J, Chen WJ, Davis P, de la Cruz N, Duong A, Fang R <em>et al</em>.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22067452" target="_blank">
                WormBase 2012: more genomes, more data, new website</a>.
                <em>Nucleic Acids Res</em>. 2012 Jan;40(Database issue):D735-41.<br>
                modENCODE worm genome-wide data available
                </p>

                <p>
                Young RS, Marques AC, Tibbit C, Haerty W, Bassett AR, Liu JL, Ponting CP.
                <a href="http://www.ncbi.nlm.nih.gov/pubmed/22403033" target="_blank">
                Identification and properties of 1,119 candidate lincRNA loci in the Drosophila melanogaster genome</a>.
                <em>Genome Biol Evol</em>. 2012;4(4):427-42.<br>
                modENCODE fly whole transcriptome (RNA-seq) data
                </p>

            </div><!--end content-->
        </div><!--end wrapper-->
    </body>
</html>
