<h2>Description</h2>

<div class="warn-note" style="border: 2px solid #9e5900; padding: 5px 20px; background-color: #ffe9cc;">
<p><span style="font-weight: bold; color: #c70000;">NOTE:</span><br> 
ClinVar is intended for use primarily by physicians and other
professionals concerned with genetic disorders, by genetics researchers, and
by advanced students in science and medicine. While the ClinVar database is
open to all academic users, users seeking information about a personal medical
or genetic condition are urged to consult with a qualified physician for
diagnosis and for answers to personal questions.</p>
</div>

<p>
This track shows the genomic positions of variants in the
<a href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar database</a>. 
ClinVar is a free, public archive of reports
of the relationships among human variations and phenotypes, with supporting
evidence. The track is updated after every ClinVar release, once a month.</p>

<p>
<b>Note:</b> The data in the track are obtained directly from ClinVar's FTP site.
We display the data obtained from ClinVar as-is to avoid discrepancies between UCSC and NCBI. 
However, be aware that the ClinVar conventions are different from the VCF standard. 
Variants may be right-aligned or may contain additional context, e.g. for
inserts. ExAC/gnomAD make available <a href="https://github.com/macarthur-lab/clinvar"
target="_blank">a converter</a>
to make ClinVar more comparable to VCF files.</p>

<h2>Display Conventions and Configuration</h2>

<p>
Genomic locations of ClinVar variants are labeled with the ClinVar variant
descriptions. All information related to each is variant is shown on that
variant's details page.  Leave the mouse over a feature for more than 2 seconds
to show the clinical significance of a variant.
</p>

<p>The track is divided into two subtracks, one for copy number variants (CNVs), which are all 
variants longer than 100bp, and a second track for shorter substitutions and indels.</p>

<p>
Until October 2017, all variants with the ClinVar types copy number gain/loss
and DbVar "nsv" accessions were put in the CNV category.  Due
to the ClinVar type not capturing this information anymore, anything longer
than 100bp is now considered a CNV.</p>

<p>
Entries in the ClinVar CNV track are colored
<b><font color="red">red for loss</font></b> and
<B><font color="blue">blue for gain</font></b>.
</p>

<p>
Entries in the ClinVar short variants track are shaded by clinical annotation:
<b><font color="black">black for pathogenic</font></b>,
<B><font color="#888">dark grey for uncertain significance or not provided</font></b> and
<B><font color="#c8c8c8">light grey for benign</font></b>.
</p>

<p>
The score of the variants is the number of "stars" in ClinVar. On the track configuration page (above), you can filter the track to show only variants with more than a certain number of stars. For more information on the star rating, see the <a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/review_status/"
target="_blank">ClinVar documentation</a>.
</p>


<H2>Data access</H2>
<p>
The raw data can be explored interactively with the <a href="../cgi-bin/hgTables">Table Browser</a>
or the <a href="../cgi-bin/hgIntegrator">Data Integrator</a>.

<p>
For automated download and analysis, the genome annotation is stored in a bigBed file that
can be downloaded from
<a href="http://hgdownload.soe.ucsc.edu/gbdb/$db/bbi/" target="_blank">our download server</a>.
The files for this track are called <tt>clinVarMain.bb</tt> and <tt>clinVarCnv.bb</tt>. Individual
regions or the whole genome annotation can be obtained using our tool <tt>bigBedToBed</tt>
which can be compiled from the source code or downloaded as a precompiled
binary for your system. Instructions for downloading source code and binaries can be found
<a href="http://hgdownload.soe.ucsc.edu/downloads.html#utilities_downloads">here</a>.
The tool
can also be used to obtain only features within a given range, e.g. 
<tt>bigBedToBed http://hgdownload.soe.ucsc.edu/gbdb/hg19/bbi/clinvar/clinvarMain.bb -chrom=chr21 -start=0 -end=100000000 stdout</tt></p>
</p>

<p>
You can find previous versions of the ClinVar tracks organized by month on our
download server <a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/archive/clinvar/" target="_blank">here</a>.
</p>

<h2>Methods</h2>

<p>
ClinVar files were reformatted at UCSC to the <a href="../goldenPath/help/bigBed.html">bigBed</a> format.
The data is updated every month, one week after the ClinVar release date.
The program that performs the update is available on
<a href="https://github.com/ucscGenomeBrowser/kent/blob/master/src/hg/utils/otto/clinvar/clinVarToBed"
target="_blank">Github</a>.
</p>

<h2>Credits</h2>
<p>
Thanks to NCBI for making the ClinVar data available on their FTP site as a tab-separated file.
</p>

<h2>References</h2>
<p>
Landrum MJ, Lee JM, Benson M, Brown G, Chao C, Chitipiralla S, Gu B, Hart J, Hoffman D, Hoover J
<em>et al</em>.
<a href="https://academic.oup.com/nar/article/44/D1/D862/2502702/ClinVar-public-archive-of-interpretations-of" target="_blank">
ClinVar: public archive of interpretations of clinically relevant variants</a>.
<em>Nucleic Acids Res</em>. 2016 Jan 4;44(D1):D862-8.
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26582918" target="_blank">26582918</a>; PMC: <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4702865/" target="_blank">PMC4702865</a>
</p>
