<H2>Description</H2>
<P>
The Coriell Cell Line Deletion and Duplication track shows
genotyping of chromosomal aberration and inherited disorder samples
from an NIGMS study  
(http://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000269.v1.p1 ,
dbGaP Study Accession: phs000269.v1.p1).
</P>

<p>
716 samples derived from 697 individuals 
from the Chromosomal Aberrations and Inherited Disorders collections of the 
NIGMS Human Genetic Cell Repository were genotyped and analyzed for CNV 
determination by the Microarray Center at the Coriell Institute for Medical 
Research. 
</p>

<p>
Karyotyping was performed on all cell cultures in the Repository with 
reported chromosome abnormalities. The samples chosen for genotyping in 
this study were intended to represent a diverse set of copy number variants, 
but the selection was also weighted to over-sample commonly manifested types 
of aberrations. When available, the ISCN description of the sample based 
on G-banding and FISH analysis is included in the phenotypic data. 
Karyotypes for these cells can be viewed in the online Repository 
catalog (http://ccr.coriell.org/Sections/Collections/NIGMS/?SsId=8).
</P>

<P>
Some field definitions from within an item description:
<UL><LI><B>CN State:</B> ...</LI>
<LI><B>Type: </B> ...</LI></UL>
<P>
The track displays CNVs labeled with type as <B>Gain</B> in black  [this
should be blue - to conform to convention in OMIM, ISCA and DECIPHER], and
those with type <B>Loss</B> in light gray[this should be red]. The 
score field also corresponds
to the type: all <B>Gain</B> CNVs have a score of 1000, and <B>Loss</B>
CNVs have a score of 0.
</P>
<H2>Methods</H2>
<P>A description of how the data were created</P>
<H2>Credits</H2>
<P>
We thank Dorit Berlin and Zhenya Tang at the NIGMS Human Genetic Cell Repository
and CHD GENES Biorepository at the 
<A HREF="http://www.coriell.org/" TARGET=_blank>Coriell Institute for Medical Research</A>
for these data.
</P>
<H2>References</H2>
<P>
http://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000269.v1.p1
</p>
