<H2>Description</H2>

<div class="warn-note" style="border: 2px solid #9e5900; padding: 5px 20px; background-color: #ffe9cc;">
 <P><span style="font-weight: bold; color: #c70000;">NOTE:</span><br> 
 While the DECIPHER database is 
 open to the public, users seeking information about a personal medical or
 genetic condition are urged to consult with a qualified physician for
 diagnosis and for answers to personal questions.
 </P>
 <P>Because the UCSC Genes mappings are based on associations from RefSeq and
 UniProt, they are dependent on any interpretations from those sources.
 Furthermore, because many DECIPHER records refer to multiple gene names,
 or syndromes not tightly mapped to individual genes, the associations
 in this track should be treated with skepticism and any conclusions
 based on them should be carefully scrutinized using independent
 resources.
 </P>
 <P>UCSC is authorized to present these data only via display in the
 Browser, and not for bulk download.  Access to bulk data may be obtained
 directly from DECIPHER and is subject to a Data Access Agreement, in which
 the user certifies that no attempt to identify individual patients will
 be undertaken.  The same restrictions apply to the public data displayed
 at UCSC:  No one is authorized to attempt to identify patients by any
 means.
 </P>
</div>

<P>
The 
<A HREF="https://decipher.sanger.ac.uk" TARGET=_BLANK>DECIPHER</A>
database of submicroscopic chromosomal imbalance 
collects clinical information about chromosomal 
microdeletions/duplications/insertions, translocations and inversions, 
and displays this information on the human genome map.
<p>
This track shows genomic regions of reported cases and their 
associated phenotype information.  All data have passed the strict
consent requirements of the DECIPHER project and are approved for
unrestricted public release.  Clicking the Patient View ID link
brings up a more detailed informational page on the patient at the 
DECIPHER web site. 
<H2>Method</H2>
<P>
Data provided by the DECIPHER project group are imported and processed 
to create a simple BED track to annotate the genomic regions associated 
with individual patients.  
</P>
<P>
The entries are colored <B><FONT COLOR = RED>red</FONT></B> for deletions 
(mean log ratio &lt; 0) and <B><FONT COLOR = BLUE>blue</FONT></B> for duplications 
(mean log ratio &gt; 0).
<BR>Note that the color scheme changed in March 2011.

<H2>Contact</H2>
<P>
For more information on DECIPHER, please contact
<A HREF="mailto:&#100;&#101;c&#105;&#112;h&#101;&#114;&#64;&#115;&#97;&#110;&#103;&#101;&#114;.
a&#99;.
&#117;&#107;">
&#100;&#101;c&#105;&#112;h&#101;&#114;&#64;&#115;&#97;&#110;&#103;&#101;&#114;.a&#99;.&#117;&#107;</A>.
</P>

<H2>References</H2>
<p>
Firth HV, Richards SM, Bevan AP, Clayton S, Corpas M, Rajan D, Van Vooren S, Moreau Y, Pettett RM,
Carter NP.
<a href="http://www.cell.com/ajhg/abstract/S0002-9297(09)00107-4" target="_blank">
DECIPHER: Database of Chromosomal Imbalance and Phenotype in Humans Using Ensembl Resources</a>.
<em>Am J Hum Genet</em>. 2009 Apr;84(4):524-33.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19344873" target="_blank">19344873</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667985/" target="_blank">PMC2667985</a>
</p>
