<H2>Description</H2>

<P>
This track displays single nucleotide polymorphisms (SNPs) identified by published 
Genome-Wide Association Studies (GWAS), collected in the 
<A HREF="http://www.genome.gov/gwastudies/" TARGET=_BLANK>Catalog of Published
Genome-Wide Association Studies</A> at the 
<A HREF="http://www.genome.gov/" TARGET=_BLANK>National Human Genome Research 
Institute</A> (NHGRI).
Some <A HREF="http://www.genome.gov/27026047" TARGET=_BLANK>abbreviations</A>
are used above.
</P>
<P>
From <A HREF="http://www.genome.gov/gwastudies/" TARGET=_BLANK>http://www.genome.gov/gwastudies/</A>:
<blockquote>
The genome-wide association study (GWAS) publications listed here
include only those attempting to assay at least 100,000 single
nucleotide polymorphisms (SNPs) in the initial stage. Publications are
organized from most to least recent date of publication, indexing from
online publication if available. Studies focusing only on candidate
genes are excluded from this catalog. Studies are identified through
weekly PubMed literature searches, daily NIH-distributed compilations
of news and media reports and occasional comparisons with an existing
database of GWAS literature 
(<A HREF="http://hugenavigator.net/HuGENavigator/home.do" TARGET=_BLANK>HuGE Navigator</A>).
</blockquote>
</P>

<H2>Methods</H2>

<P>
From <A HREF="http://www.genome.gov/gwastudies/" TARGET=_BLANK>http://www.genome.gov/gwastudies/</A>:
<blockquote>
SNP-trait associations listed here are limited to those with p-values
&lt; 1.0 x 10<sup>-5</sup> (see <A HREF="http://www.genome.gov/27529028"
TARGET=_BLANK>full methods</A> for additional details). Multipliers of
powers of 10 in p-values are rounded to the nearest single digit; odds
ratios and allele frequencies are rounded to two decimals. Standard
errors are converted to 95 percent confidence intervals where
applicable. Allele frequencies, p-values and odds ratios derived from
the largest sample size, typically a combined analysis (initial plus
replication studies), are recorded below if reported; otherwise,
statistics from the initial study sample are recorded. For
quantitative traits, information on % variance explained, SD
increment, or unit difference is reported where available. Odds ratios (OR)
&lt; 1 in the original paper are converted to OR &gt; 1 for the alternate
allele. Where results from multiple genetic models are available, we
prioritized effect sizes (ORs or beta-coefficients) as follows: 1)
genotypic model, per-allele estimate; 2) genotypic model, heterozygote
estimate; 3) allelic model, allelic estimate.
<BR><BR>
Gene regions corresponding to SNPs were identified from the UCSC
Genome Browser. Gene names and risk alleles are those reported by the
authors in the original paper. Only one SNP within a gene or region of
high linkage disequilibrium is recorded unless there was evidence of
independent association.
<BR><BR>
Occasionally, the term &quot;pending&quot; is used to denote one or more studies
that were identified as an eligible GWAS, but for which SNP information
has not yet been extracted; studies of CNVs are also noted as pending.
</blockquote>
</P>

<H2>References</H2>
<p>
Hindorff LA, Sethupathy P, Junkins HA, Ramos EM, Mehta JP, Collins FS, Manolio TA.
<a href="http://www.pnas.org/content/106/23/9362.long" target="_blank">
Potential etiologic and functional implications of genome-wide association loci for human diseases
and traits</a>.
<em>Proc Natl Acad Sci U S A</em>. 2009 Jun 9;106(23):9362-7.
PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19474294" target="_blank">19474294</a>; PMC: <a
href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2687147/" target="_blank">PMC2687147</a>
</p>
