<H2>Description</H2>
<P>

<div class="warn-note" style="border: 2px solid #9e5900; padding: 5px 20px; background-color: #ffe9cc;">
<p><span style="font-weight: bold; color: #c70000;">NOTE:</span><br>
 <B>These data are for research purposes only.  While the ISCA data are
 open to the public, users seeking information about a personal medical or
 genetic condition are urged to consult with a qualified physician for
 diagnosis and for answers to personal medical questions.
 </P>
 
 <P>UCSC presents these data for use by qualified professionals, and even
 such professionals should use caution in interpreting the significance of 
 information found here.  No single data point should be taken at face 
 value and such data should always be used in conjunction with as much 
 corroborating data as possible.  No treatment protocols should be 
 developed or patient advice given on the basis of these data without 
 careful consideration of all possible sources of information.
 </P>

 <P>No attempt to identify individual patients should
 be undertaken.  No one is authorized to attempt to identify patients 
 by any means.
 </P>
 </B>
</div>

</P>

<P>  
The <A HREF = "https://iscaconsortium.org/" TARGET = _BLANK>International 
Standards for Cytogenomic Arrays</A> (ISCA) Consortium is 
group of clinical cytogenetics and molecular genetics laboratories 
that has been organized to improve the quality of patient care in clinical 
genetic testing using new molecular cytogenetic technologies. 
These technologies include array comparative genomic hybridization 
(aCGH) and quantitative SNP analysis by microarrays or bead chips.
Membership in the ISCA Consortium is open to all individuals and 
laboratories involved in clinical array testing who are committed 
to free data sharing and to participation in a process to develop 
evidence-based standards and guidelines to improve patient care.
</P>

<P>  
The ISCA dataset displays clinical microarray data submitted to dbGaP/dbVar 
at NCBI by ISCA Consortium member laboratories
(dbVar study <A HREF="http://www.ncbi.nlm.nih.gov/dbvar/studies/nstd37/"
TARGET=_BLANK>nstd37</A>). This track shows copy
number variants (CNVs) found in patients referred for genetic testing 
for indications such as intellectual disability, developmental 
delay, autism and congenital anomalies.  

<P>The CNVs in this study have been reviewed for their clinical significance by
the submitting ISCA laboratory. Some of the deletions and duplications in the track
have been reported as causative for a phenotype by the submitting clinical 
laboratory; this information was based on current knowledge at the time of submission.
However, it should be noted that phenotype information is often vague and imprecise and
should be used with caution.  While all samples were submitted because of a phenotype in 
a patient, only 15% of patients had variants determined to be causal, 
and most patients will have additional variants that are not causal.
</P>

<P>CNVs are separated into subtracks and are labeled as:
<ul>
  <li>Pathogenic</li>
  <li>Uncertain: Likely Pathogenic</li>
  <li>Uncertain</li>
  <li>Uncertain: Likely Benign</li>
  <li>Benign</li>
</ul>
The user should be aware that some of the data were submitted using a 3-class
system, with the two &quot;Likely&quot; categories omitted.  
</P>

<P>  Two subtracks, ISCA Path Gain and ISCA Path Loss, are aggregate tracks
showing graphically the accumulated level of gains and losses in the 
Pathogenic subtrack across the genome.
</P>

<P>  Many samples have multiple variants, not all of which are causative 
of the phenotype.  The CNVs in these samples have been decoupled, so it is not
possible to connect multiple imbalances as coming from a single patient.
It is therefore not possible to identify individuals via their genotype.  
</P>


<H2>Methods and Color Convention</H2>
<P>
The samples were analyzed by arrays from patients referred for 
cytogeneic testing due to clinical phenotypes.  Samples were analyzed with a 
probe spacing of 20-75 kb. The minimum CNV breakpoints are shown; if available,
the maximum CNV breakpoints are provided in the details page, but are not shown 
graphically on the Browser image.
</P>

<P>Data were submitted to 
<A HREF = "http://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000205.v2.p1"
TARGET = _BLANK>dbGaP</A> at NCBI and thence decoupled as described into 
<A HREF = "http://www.ncbi.nlm.nih.gov/dbvar/studies/nstd37/" 
TARGET = _BLANK>dbVar</A> for unrestricted release. 
</P>

<P>
The entries are colored <B><FONT COLOR = RED>red for loss</FONT></B>  and 
<B><FONT COLOR = BLUE>blue for gain</FONT></B>. 
</P>

<H2>Verification</H2>
<P>
Most data were validated by the submitting laboratory using various methods, 
including FISH, G-banded karyotype, MLPA and qPCR.
</P>

<H2>Credits</H2>
<P>
Thanks to the ISCA Consortium, to Viren Patel at Emory University for technical
coordination with ISCA on track details, to Deanna Church and 
Justin Paschall at NCBI for consultation and integration into dbGaP and dbVar, 
and to Angie Hinrichs and Robert Kuhn at UCSC for engineering.
</P>

<H2>References</H2>
<P>Miller DT, Adam MP, Aradhya S, Biesecker LG, Brothman AR, Carter NP, Church DM, Crolla JA, 
Eichler EE, Epstein CJ <EM>et al</EM>.
<A HREF = "http://www.ncbi.nlm.nih.gov/pubmed/20466091" 
TARGET = _BLANK>Consensus statement: chromosomal microarray is a first-tier 
clinical diagnostic test for individuals with developmental disabilities 
or congenital anomalies</A>.
<EM>Am J Hum Genet</EM>. 2010 May 14;86(5):749-64.
</P>

<P>
Kaminsky EB, Kaul V, Paschall J, Church DM, Bunke B, Kunig D, Moreno-De-Luca D, 
Moreno-De-Luca A, Mulle JG, Warren ST <EM>et al</EM>.
<A HREF = "http://www.ncbi.nlm.nih.gov/pubmed/21844811" 
TARGET = _BLANK>
An evidence-based approach to establish the functional and clinical significance 
of copy number variants in intellectual and developmental disabilities</A>.
<EM>Genet Med</EM>. 2011 Sep;13(9):777-84.
</P>
