<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=315>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.cas.vanderbilt.edu/bioimages/biohires/animals/mammalia/hhetglajuvenile43838.JPG" TARGET=_blank>
      <IMG SRC="/images/Heterocephalus_glaber.jpg" WIDTH=300 HEIGHT=200 ALT="Naked Mole-rat "></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT>
    <FONT SIZE=-1><em>Heterocephalus glaber</em><BR>
    </FONT>
    <FONT SIZE=-2>
    Images <A HREF="http://www.cas.vanderbilt.edu/bioimages/contact/baskauf-contact.htm" TARGET=_blank>&copy; 2003 Steve Baskauf</A>. <BR>
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Jul. 2011 <em>Heterocephalus glaber</em> (Naked mole-rat) draft assembly 
(NCBI project <A HREF="http://www.ncbi.nlm.nih.gov/bioproject/68323"
TARGET=_blank>NCBI project 68323</A>, <A HREF="ftp://ftp.ncbi.nlm.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Heterocephalus_glaber/HetGla_1.0/"
TARGET=_blank>GCA_00000230455.1</A>))
was produced by the 
<A HREF="http://en.genomics.cn/"
TARGET=_blank>Beijing Genomics Institute, Shenzhen</A>.
For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/genome/assembly/310228/" TARGET="_blank">
HetGla_1.0</A> in the NCBI Assembly database.


<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Naked mole-rat genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TH ALIGN=LEFT>Request:</TH>
        <TH WIDTH=14>&nbsp;</TH>
       	<TH ALIGN=LEFT>Genome Browser Response:</TH></TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD VALIGN=Top>chrM</TD>
        <TD WIDTH=14>&nbsp;</TD>
        <TD VALIGN=Top>Displays complete mitochondrion genome</TD>
    </TR>

    <TR><TD VALIGN=Top>JH169211</TD>
        <TD WIDTH=14>&nbsp;</TD>
        <TD VALIGN=Top>Displays all of scaffold JH169211</TD>
    </TR>
    <TR><TD VALIGN=Top>JH169211:1-400,000</TD>
        <TD WIDTH=14>&nbsp;</TD>
        <TD VALIGN=Top>Displays first 400,000 bases of scaffold JH169211</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>JH169211:30,000+2000</TD>
        <TD WIDTH=14></TD>
        <TD VALIGN=Top>Displays a region of scaffold JH169211 that spans 2000 bases, starting with position 300000 </TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
   <TR><TD VALIGN=Top>HQ326637</TD>
        <TD WIDTH=14>&nbsp;</TD>
        <TD VALIGN=Top>Displays region of mRNA with GenBank accession
                number HQ326637</TD>
    <TR><TD VALIGN=Top NOWRAP>
            HQ326686; HQ326637 <BR>
        <TD WIDTH=14></TD>
        <TD VALIGN=Top>Displays region between genome landmarks, such as
                      the mRNA markers HQ326686 and HQ326637.
                      This syntax may also be used for other range queries,
                      such as between uniquely determined ESTs, STSs,
                      refSeqs, etc.</TD>

    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;<br></TD></TR>
    <TR><TD VALIGN=Top>pseudogene mRNA</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists transcribed pseudogenes, but not cDNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>homeobox caudal</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs for caudal homeobox genes</TD>
    </TR>
    <TR><TD VALIGN=Top>zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>
    <TR><TD VALIGN=Top>kruppel zinc finger</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>
    <TR><TD VALIGN=Top>Taguchi,T.</TD>
        <TD WIDTH=14>&nbsp;</TD>
       	<TD VALIGN=Top>Lists mRNAs deposited by co-author T. Taguchi</TD>
    </TR>
    <TR><TD VALIGN=Top COLSPAN=3>&nbsp;</TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although
	GenBank requires the search format <em>Taguchi T</em>, internally it
	uses the format <em>Taguchi,T.</em>.</TD>
    </TR>
</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
The male naked mole-rat genome has been sequenced using Illumina reads, with a sequencing depth of >20x. 
There are 39,266 scaffolds with a total size of 2,643,961,837 bases. The N50 size of the scaffolds is 1,603,177 bases with 502 scaffolds larger, and 38,764 scaffolds smaller. 
For more information about this assembly, see also:
<A HREF="http://www.ncbi.nlm.nih.gov/Traces/wgs/?val=AFSB01"

TARGET=_blank>WGS record</A> at NCBI.

The mitochondrial sequence is available as the virtual chromosome
&quot;chrM&quot;, see  
<A HREF="http://www.ncbi.nlm.nih.gov/nuccore/JN242813.1"
TARGET=_blank>mitochondrion record</A> at NCBI for details.
</P>


<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hetGla1/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#naked_mole-rat">Downloads</A> page. 
These data have
<A HREF="../goldenPath/credits.html#naked_mole-rat_use">specific 
conditions for use</A>. 

The Naked mole-rat browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#naked_mole-rat_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
<HR>
<H3>Genbank Pipeline Details</H3>
For the purposes of the Genbank alignment pipeline, this assembly is considered to be: <B>low-coverage</B>.
