<H2>Description</H2>

<p>This track is produced as part of the mouse ENCODE Project.
This track shows RNA-seq measured genome-wide in mouse
<A TARGET=_BLANK HREF="/cgi-bin/hgEncodeVocab?type=cellType"> tissues and cell lines</A>
using the RNA-seq methodology &#40;see below&#41;, and DNaseI hypersensitive
sites.
</p>

<H2>Display Conventions and Configuration</H2>
<p>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>). For each view, there are multiple subtracks that display
individually on the browser. Instructions for configuring multi-view tracks
are <a href="/goldenPath/help/multiView.html" target="_blank">here</a>.
This track contains the following views:
<DL>
<DT><I>Plus</I> and <I>Minus</I> <I>Signals</I>
<DD>These views display clusters of overlapping read mappings on the
forward and reverse genomic strands
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based
on processed data.</DD>
<DT><I>Alignments</I></DT><DD>Mappings of short reads to the genome. <DD>
</DL>

<H2>Methods</H2>
<p>
Cells were grown according to the approved
<a href="/ENCODE/protocols/cell/mouse" target="_blank">ENCODE cell culture protocols</a>.
Fresh tissues were harvested from mice and the nuclei prepared according 
to the tissue appropriate
<a href="/ENCODE/protocols/cell/mouse" target="_blank">protocol</a>.
</p>

<p>
Reads were aligned to mm9 reference using ABI BioScope software version 1.2.1.
Colorspace FASTQ format files were created using Heng Li's
solid2fastq.pl script version 0.1.4, representing 0,1,2,3 color codes
with the letters A,C,G,T respectively. 
Signal files were created from the BAM alignments using BEDTools.
</p>

<H2>Verification</H2>

<H2>Credits</H2>

<p>These data were generated by the UW ENCODE group. </p>
<p>Contact: <A HREF="mailto:&#115;&#117;l&#108;&#64;&#117;.w&#97;&#115;h&#105;&#110;&#103;&#116;&#111;n.&#101;&#100;&#117;">
Richard Sandstrom</A>
<!-- above address is sull at u.washington.edu -->
</p>

<H2>References</H2>


<H2> Data Release Policy </H2>

<p>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column, above.  The full data release policy
for ENCODE is available
<A TARGET=_BLANK HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</p>

