<H2>Description</H2>

<P><DT><B><I>Rationale for the Mouse ENCODE project</I></B></DT>
Knowledge of the function of genomic DNA sequences comes from three 
basic approaches. Genetics uses changes in behavior or structure of a cell or organism
in response to changes in DNA sequence to infer function of the altered sequence. 
Biochemical approaches monitor states of histone modification, binding of specific 
transcription factors, accessibility to DNases and other epigenetic features along 
genomic DNA. In general, these are associated with gene activity, but the precise 
relationships remain to be established. The third approach is evolutionary, using 
comparisons among homologous DNA sequences to find segments that are evolving 
more slowly or more rapidly than expected given the local rate of neutral change. These 
are inferred to be under negative or positive selection, respectively, and interpreted 
as DNA sequences needed for a preserved (negative selection) or adaptive 
(positive selection) function. </P>

<P>The ENCODE project aims to discover all the DNA sequences associated with 
various epigenetic features, with the reasonable expectation that these will also be 
functional &#40;best tested by genetic methods&#41;. However, it is not clear how to relate these 
results with those from evolutionary analyses. The mouse ENCODE project aims to 
make this connection explicitly and with a moderate breadth. Assays identical to those 
being used in the ENCODE project are performed in cell types in mouse that are similar 
or homologous to those studied in the human project. The comparison will be used to discover 
which epigenetic features are conserved between mouse and human, and  
examine the extent to which these overlap with the DNA sequences under negative 
selection. The contribution of functional DNA preserved in mammals versus 
function with in only one species will be discovered. The results will have a 
significant impact on the understanding of the evolution of gene regulation.
</P>

<P><DT><B><I>Maps of Occupancy by Transcription Factors</I></B></DT>
Maps of occupancy of genomic DNA by transcription factors &#40;TFs&#41; are determined by 
ChIP-seq. This consists of two basic steps: chromatin immunoprecipitation &#40;ChIP&#41; is 
used to highly enrich genomic DNA for the segments bound by specific proteins &#40;the 
antigens recognized by the antibodies&#41; followed by massively parallel short read 
sequencing to tag the enriched DNA segments. Sequencing is done on the Illumina 
GAIIx and HiSeq. The sequence tags are mapped back to the mouse genome 
&#40;Langmead <EM>et al.</EM> 2009&#41;, and a graph of the enrichment for TF binding are displayed as 
the &#34;Signal&#34; track &#40;essentially the counts of mapped reads per interval&#41; and the deduced 
probable binding sites from the MACS program &#40;Zhang <EM>et al.</EM> 2008&#41; are shown in the 
&#34;Peaks&#34; track. Each experiment is associated with an input signal, which represents the 
control condition where immunoprecipitation with non-specific immunoglobulin was 
performed in the same cell type. The sequence reads, quality scores, and alignment 
coordinates from these experiments are available for download. </P>

<H2>Display Conventions and Configuration</H2>

<P>
This track is a multi-view composite track that contains multiple data types
(<EM>views</EM>).  For each view, there are multiple subtracks that
display individually on the browser.  Instructions for configuring multi-view
tracks are <a href="/goldenPath/help/multiView.html" target="_blank">here</a>.
This track contains the following views:
<DL>
<DT><I>Peaks</I></DT><DD> Regions of signal enrichment based on processed data
(usually normalized data from pooled replicates). Intensity is represented in 
grayscale; the darker shading shows higher intensity (a solid vertical line
in the peak region represents the point with the highest signal).
ENCODE Peaks tables contain fields for statistical significance, including FDR
(<a href="/goldenPath/help/qValue.html" target="_blank">qValue</a>).</DD>
<DT><I>Signal</I></DT><DD>Density graph (wiggle) of signal enrichment based on
processed data.</DD>
</DL>
<p>Metadata for a particular subtrack can be found by clicking the down arrow in the list of subtracks.</p>

<H2>Methods</H2>

<P>
Cells were grown according to the approved
<a href="/ENCODE/protocols/cell/mouse" target="_blank">ENCODE cell culture protocols</a>.
</P>
<P>
The chromatin immunoprecipitation followed published methods &#40;Welch <EM>et al.</EM> 2004&#41;.
Information on antibodies used is available via the hyperlinks in the &#34;Select
subtracks&#34; menu. Samples passing initial quality thresholds &#40;enrichment and depletion 
for positive and negative controls &#45; if available &#45; by quantitative PCR of ChIP material&#41;
are processed for library construction for Illumina sequencing, using the ChIP-seq 
Sample Preparation Kit purchased from Illumina. Starting with a 10 ng sample of ChIP 
DNA, DNA fragments were repaired to generate blunt ends and a single A nucleotide 
was added to each end. Double-stranded Illumina adaptors were ligated to the 
fragments. Ligation products were amplified by 18 cycles of PCR, and the DNA between 
250-350 bp was gel purified. Completed libraries were quantified with Quant-iT dsDNA 
HS Assay Kit. The DNA library was sequenced on the Illumina Genome Analyzer II 
sequencing system, and more recently on the HiSeq. Cluster generation, linearization, 
blocking and sequencing primer reagents were provided in the Illumina Cluster 
Amplification kits. All samples were determined as biological replicates except time 
course samples. The data displayed are from the pooled reads for all replicates, but 
individual replicates are available by download.
</P>

<P>
The resulting 36-nucleotide sequence reads &#40;fastq files&#41; were moved to a data 
library in Galaxy, and the tools implemented in Galaxy were used for further processing 
via workflows &#40;Blankenberg <EM>et al.</EM> 2010&#41;. The reads were mapped to the mouse genome 
&#40;mm9 assembly&#41; using the program bowtie &#40;Langmead <EM>et al.</EM> 2009&#41;, and the files of 
mapped reads for the ChIP sample and from the &#34;input&#34; control &#40;no antibody&#41; were 
processed by MACs &#40;Zhang <EM>et al.</EM> 2008&#41; to call peaks for occupancy by transcription 
factors, using the parameters mfold=15, bandwidth=125.  Per-replicate
alignments and sequences are available for download at
<A HREF="../cgi-bin/hgFileUi?db=mm9&amp;g=wgEncodePsuTfbs"
 TARGET=_BLANK>downloads page</A>.
</P>

<H2>Credits</H2>

<p>
Cell growth, ChIP, and Illumina library construction were done in the laboratories 
of Ross Hardison, Mitchell J. Weiss and Gerd A. Blobel &#40;PSU and Children&#39;s Hospital of 
Philadelpia&#41;. Sequencing on the Illumina was done by the laboratories of Stephan Schuster 
and Ross Hardison. Data processing and analysis were overseen by James Taylor 
&#40;Emory University&#41; and Anton Nekrutenko and Hardison &#40;PSU&#41;.
</p>
<p>
Contact:
<A HREF="mailto:&#114;&#99;h&#56;&#64;&#112;&#115;&#117;.
&#101;&#100;u"
title="mailto:&#114;&#99;h&#56;&#64;&#112;&#115;&#117;.
&#101;&#100;u" rel="nofollow"
 TARGET=_BLANK>Ross Hardison</A>
<!-- above address is Ross Hardison at psu.edu -->
</p>

<H2>References</H2>

<P>
Aplan PD, Nakahara K, Orkin SH, Kirsch IR.
 <A HREF="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC556917/"
         class="external text"
	 title="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC556917/"
	 TARGET = _BLANK>The SCL gene product: a positive regulator of erythroid
 differentiation</A>.
<EM>EMBO J</EM>. 1992 November;11&#40;11&#41;:4073&#45;4081.
</P>

<P>
Blankenberg D, Gordon A, Von Kuster G, Coraor N, Taylor J, Nekrutenko A.
 <A HREF="http://bioinformatics.oxfordjournals.org/content/26/14/1783.short"
          class="external text"
	  title="http://bioinformatics.oxfordjournals.org/content/26/14/1783.short"
	  TARGET = _BLANK>Manipulation of FASTQ data with Galaxy</A>.
<EM>Bioinformatics</EM>. 2010 June 18;26&#40;14&#41;:1783&#45;1785.
</P>

<P>
Green A, DeLuca E, Begley C.
 <A HREF="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC453166/"
          class="external text"
	  title="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC453166/"
          TARGET = _BLANK>Antisense SCL suppresses self-renewal and enhances 
spontaneous erythroid differentiation of the human leukemic cell line K562</A>.
<EM>EMBO J</EM>. 1991 December;10&#40;13&#41;:4153&#45;4158. 
</P>

<P>
Huang S, Brandt SJ.
 <A HREF="http://mcb.asm.org/cgi/content/full/20/6/2248"
          class="external text"
	  title="http://mcb.asm.org/cgi/content/full/20/6/2248"
	  TARGET = _BLANK>mSin3A regulates murine erythroleukemia cell 
differentiation through association with the TAL1 (or SCL) transcription factor</A>.
<EM>Mol Cell Biol</EM>. 2010 March;20&#40;6&#41;:2248&#45;2259.
</P>

<P>
Huang S, Qiu Y, Shi Y, Xu Z, Brandt SJ.
 <A HREF="http://www.nature.com/emboj/journal/v19/n24/abs/7593492a.html"
          class="external text"
	  title="http://www.nature.com/emboj/journal/v19/n24/abs/7593492a.html"
	  TARGET = _BLANK>P/CAF-mediated acetylation regulates the function 
of the basic helix- loop-helix transcription factor TAL1/SCL</A>.
<EM>EMBO J</EM>. 2000 Dec;19&#40;24&#41;:6792&#45;6803.
</P>

<P>
Langmead B, Trapnell C, Pop M, Salzberg SL.
 <A HREF="http://genomebiology.com/2009/10/3/R25"
          class="external text"
	  title="http://genomebiology.com/2009/10/3/R25"
	  TARGET = _BLANK>Ultrafast and memory-efficient alignment of 
short DNA sequences to the human genome</A>.
<EM>Genome Biol</EM>. 2009 March;10&#40;3&#41;:R25.
</P>

<P>
Shivdasani RA, Mayer EL, Orkin SH.
 <A HREF="http://www.nature.com/nature/journal/v373/n6513/abs/373432a0.html"
          class="external text"
	  title="http://www.nature.com/nature/journal/v373/n6513/abs/373432a0.html"
	  TARGET = _BLANK>Absence of blood formation in mice lacking the 
T-cell leukaemia oncoprotein tal-1/SCL</A>.
<EM>Nature</EM>. 1995 Feb;373:432-434.
</P>

<P>
Tripic T, Deng W, Cheng Y, Zhang Y, Vakoc CR, Gregory GD, Hardison RC, Blobel GA.
 <A HREF="http://bloodjournal.hematologylibrary.org/cgi/content/short/113/10/2191"
          class="external text"
	  title="http://bloodjournal.hematologylibrary.org/cgi/content/short/113/10/2191"
	  TARGET = _BLANK>SCL and associated proteins distinguish 
active from repressive GATA transcription factor complexes</A>.
<EM>Blood</EM>. 2009 March;13&#40;10&#41;:2191&#45;2201. 
</P>

<P>
Wadman IA, Osada H, Grutz G, Agulnick AD, Westphal H, Forster A, Rabbitts TH.
 <A HREF="http://www.ncbi.nlm.nih.gov/pubmed/9214632?dopt=Abstract"
          class="external text"
	  title="http://www.ncbi.nlm.nih.gov/pubmed/9214632?dopt=Abstract"
	  TARGET = _BLANK>The LIM-only protein Lmo2 is a bridging 
molecule assembling an erythroid, DNA-binding complex which includes the TAL1, E47, GATA-1 and Ldb1/NL1 proteins</A>.
<EM>EMBO J</EM>. 1997 Jun;16:3145&#45;3157. 
</P>

<P>
Weiss MJ, Yu C, Orkin SH.
 <A HREF="http://mcb.asm.org/content/17/3/1642.abstract"
          class="external text"
	  title="http://mcb.asm.org/content/17/3/1642.abstract"
	  TARGET = _BLANK>Erythroid-cell-specific properties of transcription factor 
GATA-1 revealed by phenotypic rescue of a gene-targeted cell line</A>.
<EM>Mol Cell Biol</EM>. 1997 March; 16:3145&#45;3157.
</P>

<P>
Welch JJ, Watts JA, Vakoc CR, Yao Y, Wang H, Hardison RC, Blobel GA, Chodosh LA, Weiss MJ.
 <A HREF="http://bloodjournal.hematologylibrary.org/content/104/10/3136.short"
          class="external text"
	  title="http://bloodjournal.hematologylibrary.org/content/104/10/3136.short"
	  TARGET = _BLANK>Global regulation of erythroid gene expression by
 transcription factor GATA-1</A>.
<EM>Blood</EM>. 2004 Nov;104&#40;10&#41;:3136&#45;3147.
</P>

<P>
Wold B, Myers RM.
 <A HREF="http://www.nature.com/nmeth/journal/v5/n1/full/nmeth1157.html"
          class="external text"
	  title="http://www.nature.com/nmeth/journal/v5/n1/full/nmeth1157.html"
	  TARGET = _BLANK>Sequence census methods for functional genomics</A>.
<EM> Nat Methods </EM>. 2008 Jan;5&#40;1&#41;:19&#45;21.
</P>

<P>
Zhang Y, Liu T, Meyer CA, Eeckhoute J, Johnson DS, Bernstein BE, Nussbaum C, Myers RM, Brown M, Li W <em>et al.</em>
 <A HREF="http://genomebiology.com/2008/9/9/R137"
          class="external text"
	  title="http://genomebiology.com/2008/9/9/R137"
	  TARGET = _BLANK>Model-based analysis of ChIP-Seq (MACS)</A>.
<EM>Genome Biol</EM>. 2008 September;9&#40;9&#41;:R137.
</P>

<h2>Publications</h2>

<p>
Wu W, Cheng Y, Keller CA, Ernst J, Kumar SA, Mishra T, Morrissey C, Dorman CM, Chen KB, Drautz D <em>et al</em>.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/21795386" target="_blank">
Dynamics of the epigenetic landscape during erythroid differentiation after GATA1 restoration</a>.
<em>Genome Res</em>. 2011 Oct;21(10):1659-71.
</p>

<H2>Data Release Policy</H2>

<P>Data users may freely use ENCODE data, but may not, without prior
consent, submit publications that use an unpublished ENCODE dataset until
nine months following the release of the dataset.  This date is listed in
the <EM>Restricted Until</EM> column on the track configuration page and
the download page.  The full data release policy for ENCODE is available
<A HREF="../ENCODE/terms.html" TARGET=_BLANK>here</A>.</P>
