<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/FAQ.topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Blat</b></FONT>
                </TD></TR></TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
		<UL>
                <LI><A HREF="#blat1">Blat vs. Blast</A>
                <LI><A HREF="#blat2">Blat use restrictions</A>
                <LI><A HREF="#blat3">Downloading Blat source and documentation</A>
                <LI><A HREF="#blat5">Replicating web-based Blat parameters in command-line version</A>
                <LI><A HREF="#blat6">Using the <em>-ooc</em> flag</A>
                <LI><A HREF="#blat4">Replicating web-based Blat percent identity and score calculations</A>
                <LI><A HREF="#blat7">Replicating web-based Blat &quot;I'm feeling lucky&quot; search results</A> 
                <LI><A HREF="#blat8">Using Blat for short sequences with maximum sensitivity</A> 
		</UL>
		<P>
		<A HREF="index.html">Return to FAQ Table of Contents</A>
		<BR>
            </TD><TD WIDTH=15></TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat1"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Blat vs. Blast</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What are the differences between Blat and Blast?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Blat is an alignment tool like BLAST, but it is structured differently.  On 
			DNA, Blat works by keeping an index of an entire genome in memory. 
			Thus, the target database of BLAT is not a set of GenBank sequences, but
                        instead an index derived from the assembly of the entire genome. By default,
                        the index consists of all non-overlapping 11-mers except for those heavily
                        involved in repeats, and it uses less than a gigabyte of RAM. This smaller
                        size means that Blat is far more easily <a href="../admin/mirror.html">mirrored</a>
                        than BLAST. Blat of DNA is designed to quickly find sequences of 95% and
                        greater similarity of length 40 bases or more. It may miss more divergent or
                        shorter sequence alignments. (The default settings and expected behavior of
                        standalone Blat are slightly different from those on the
                        <a href="../cgi-bin/hgBlat">graphical version of Blat</a>.)
			<P>
			On proteins, Blat uses 4-mers rather than 11-mers, finding protein sequences 
			of 80% and greater similarity to the query of length 20+ amino acids. The
                        protein index requires slightly more than 2 gigabytes of RAM. 
			In practice -- due to sequence divergence rates over evolutionary time -- DNA
			Blat works well within humans and primates, while protein Blat 
			continues to find good matches within terrestrial vertebrates and even earlier 
			organisms for conserved proteins. Within humans, protein Blat gives a much better 
			picture of gene families (paralogs) than DNA Blat. However, BLAST and 
			psi-BLAST at NCBI can find much more remote matches.
			<P>
			From a practical standpoint, Blat has several advantages over BLAST: 
			<UL>
			<LI>speed (no queues, response in seconds) at the price of lesser homology depth
			<LI>the ability to submit a long list of simultaneous queries in fasta format
			<LI>five convenient output sort options
			<LI>a direct link into the UCSC browser
			<LI>alignment block details in natural genomic order
			<LI>an option to launch the alignment later as part of a custom track
			</UL>
			Blat is commonly used to look up the location of a 
			sequence in the genome or determine the exon structure of an mRNA, but expert 
			users can run large batch jobs and make internal parameter sensitivity 
			changes by installing command line Blat on their own Linux server.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Blat use restrictions</b></FONT></TD>
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I received a high-volume traffic warning from your Blat
			server informing me that I had exceeded the server use
			limitations. Can you give me information on the UCSC
			Blat server use parameters?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Due to the high demand on our Blat servers, we restrict 
			service for users who programatically query Blat or do 
			large batch queries. Program-driven use of Blat is 
			limited to a maximum of one hit every 15 
			seconds and no more than 5,000 hits per day. Please limit 
			batch queries to 25 sequences or less. 
			<P>
			For users with high-volume Blat demands, we recommend
			downloading Blat for local use. For more information, 
			see <A HREF="#blat3">Downloading Blat source and 
			documentation</A>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Downloading Blat source and documentation</b></FONT></TD>
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Is the Blat source available for download? Is there
			documentation available?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Blat source and executables are freely available for
			academic, nonprofit and personal use. Commercial licensing
			information is available on the 
			<A HREF="http://www.kentinformatics.com" 
			TARGET=_blank>Kent Informatics website</A>.
			<P>
			Blat source may be downloaded from 
			<A HREF="http://www.soe.ucsc.edu/~kent/src">http://www.soe.ucsc.edu/~kent</A> 
			(look for the blatSrc* zip file with the most recent 
			date). For 
			Blat executables, go to 
			<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/">http://hgdownload.cse.ucsc.edu/admin/exe/</A>; and choose your machine type. 
			<P>
			Documentation on Blat program specifications is available
			<A HREF="../goldenPath/help/blatSpec.html">here</A>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Replicating web-based Blat 
			parameters in command-line version</A>
</b></FONT></TD>
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I'm setting up my own Blat server and would like to use 
			the same parameter values that the UCSC web-based Blat 
			server uses."
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Use the following settings to replicate 
			the search results of the UCSC Blat server. Note that
			you may still observe some slight differences between
			command line results and web-based results, depending
			on the search being performed.
			<P>
			<em>faToTwoBit</em>: 
			<UL>
			<LI>Use soft masking.
			</UL>
			<P>
			<em>gfServer</em> (<B>this is how the UCSC web-based blat servers are configured</B>):</BR>
			<UL>
			<LI>blat server (capable of PCR):</BR>
    			&nbsp;&nbsp;&nbsp;<TT>gfServer start blatMachine portX -stepSize=5 -log=untrans.log database.2bit</TT>
			<LI>translated blat server:<BR>
    			&nbsp;&nbsp;&nbsp;<TT>gfServer start blatMachine portY -trans -mask -log=trans.log database.2bit</TT>
			</UL>
			For enabling DNA/DNA and DNA/RNA
			matches, only the host, port and twoBit files are needed.
			The same port is used for both untranslated blat (gfClient)
			and PCR (webPcr). You'll need a separate blat server on a separate
			port to enable translated blat (protein searches or translated searches in protein-space). 
			</P>
			<P>
			<em>gfClient</em>: 
			<UL>
			<LI>Set <em>-minScore=0</em> and 
			<em>-minIdentity=0</em>. This will result in some 
			low-scoring, generally spurious hits, but for 
			interactive use it's sufficiently easy to ignore them 
			(because results are sorted by score) and sometimes 
			the low-scoring hits come in handy. 
			</UL></P>
			<P>
			<em>standalone blat</em>: 
			<UL>
			<LI>blat search:<BR>
			&nbsp;&nbsp;&nbsp;<TT>blat -stepSize=5 -repMatch=2253 -minScore=0 -minIdentity=0 database.2bit query.fa output.psl </TT><br>
			</UL></P>
			<br>Notes on repMatch: 
			<br>The default setting for gfServer dna matches is: repMatch = 1024 * (tileSize/stepSize).
			<br>The default setting for blat dna matches is: repMatch = 1024 (if tileSize=11).
			<br>To get command-line results that are equivalent to web-based results, repMatch must
			    be specified when using blat.
			<P>
			For more information on the parameters available for
			blat, gfServer, and gfClient, see the 
			<A HREF="../goldenPath/help/blatSpec.html">blat
			specifications</A>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>


<!--Content Table------------------------------------------------------------>
<A NAME="blat6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Using the <em>-ooc</em> flag</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What does the <em>-ooc</em> flag do?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Using any <em>-ooc</em> option in blat, such
			as <em>-ooc=11.ooc</em>, simply serves to speed up 
			searches similar to repeat-masking sequence. The
			<em>11.ooc</em> file contains sequences 
			determined to be over-represented in the genome 
			sequence. To speed up searches, these sequences are not 
			used when seeding an alignment against the genome. For 
		 	reasonably-sized sequences, this will not create a 
			problem and will significantly reduce processing time.
			<P>
			By not using the <em>11.ooc</em> file, you will increase 
			alignment time, but will also slightly increase 
			sensitivity. This may be important if you are aligning 
			shorter sequences or sequences of poor quality. For example,
			if a particular sequence consists primarily of 
			sequences in the <em>11.ooc</em> file, it will 
			never be seeded correctly for an alignment if the 
			<em>-ooc</em> flag is used.  
			<P>
			In summary,
			if you are not finding certain sequences and can afford 
			the extra processing time, you may want to run blat 
			without the <em>11.ooc</em> file if your particular
			situation warrants its use.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Replicating web-based Blat percent identity and score calculations</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Using my own command-line Blat server, how can I 
			replicate the percent identity and score calculations
			produced by web-based Blat?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			There isn't an option to command-line Blat that gives 
			you the percent ID and the score. Instead, you will
			have to write your own program to produce the 
			calculations, incorporating some of the functions from 
			the Genome Browser source code.
			<P>
			To calculate the percent ID, incorporate the following
			code and function into a program that processes your 
			Blat PSL output. The parameter <em>isMrna</em> should
			be set to TRUE, regardless of whether the input
			sequence is mRNA or protein. 
			<P>
			The percent identity score is calculated like this:
			<P>
			<BLOCKQUOTE>
			<TT>
    			100.0 - pslCalcMilliBad(psl, TRUE) * 0.1
			</TT>
			</BLOCKQUOTE>
			<P>
			Here is the source for <TT>pslCalcMilliBad</TT>:
			<P>
			<BLOCKQUOTE>
			<PRE>
int pslCalcMilliBad(struct psl *psl, boolean isMrna)
/* Calculate badness in parts per thousand. */
{
int sizeMul = pslIsProtein(psl) ? 3 : 1;
int qAliSize, tAliSize, aliSize;
int milliBad = 0;
int sizeDif;
int insertFactor;
int total;

qAliSize = sizeMul * (psl->qEnd - psl->qStart);
tAliSize = psl->tEnd - psl->tStart;
aliSize = min(qAliSize, tAliSize);
if (aliSize <= 0)
    return 0;
sizeDif = qAliSize - tAliSize;
if (sizeDif < 0)
    {
    if (isMrna)
        sizeDif = 0;
    else
        sizeDif = -sizeDif;
    }
insertFactor = psl->qNumInsert;
if (!isMrna)
    insertFactor += psl->tNumInsert;

total = (sizeMul * (psl->match + psl->repMatch + psl->misMatch));
if (total != 0)
    milliBad = (1000 * (psl->misMatch*sizeMul + insertFactor + 
	round(3*log(1+sizeDif)))) / total;
return milliBad;
}
</PRE>
</BLOCKQUOTE>
			<P>	
			The complexity in milliBad arises primarily from how it
			handles inserts. Ignoring the inserts, the calculation
			is simply mismatches expressed as parts per thousand. 
			However, the algorithm factors in insertion penalties as
			well, which are relatively weak compared to say blasts 
			but still present. When huge inserts are allowed (which 
			is necessary to accommodate introns), it is typically 
			necessary to resort to logarithms like this calculation 
			does.
			<P>
			The <TT>pslIsProtein</TT> function called by 
			<TT>pslCalcMilliBad</TT> is:
			<BLOCKQUOTE>
			<PRE>
boolean pslIsProtein(const struct psl *psl)
/* is psl a protein psl (are it's blockSizes and scores in protein space) 
*/
{
int lastBlock = psl->blockCount - 1;

return  (((psl->strand[1] == '+' ) &&
     (psl->tEnd == psl->tStarts[lastBlock] + 3*psl->blockSizes[lastBlock])) 
||
    ((psl->strand[1] == '-') &&
     (psl->tStart == (psl->tSize-(psl->tStarts[lastBlock] + 
3*psl->blockSizes[lastBlock])))));
}
</PRE>
</BLOCKQUOTE>
			<P>
			This function automatically determines whether or not 
			the PSL output file contains alignment information for 
			a protein query. Alternatively, you could write the 
			program such that the user specifies if the query is a 
			protein or not.
			<P>
			The score calculation is generated by the following
			function:
<BLOCKQUOTE>
<PRE>
int pslScore(const struct psl *psl)
/* Return score for psl. */
{
int sizeMul = pslIsProtein(psl) ? 3 : 1;

return sizeMul * (psl->match + ( psl->repMatch>>1)) -
         sizeMul * psl->misMatch - psl->qNumInsert - psl->tNumInsert;
}
</PRE>
</BLOCKQUOTE>
			<P>
			For help with creating a C program to do perform these
			calculations, you may want to use the libraries from 
			the Genome Browser source code. See our 
			<A HREF="FAQlicense.html">FAQ</A> on source code 
			licensing and downloads for information on obtaining
			the source. The file <em>kent/src/lib/psl.c</em> 
			contains the <TT>pslCalcMilliBad</TT>, 
			<TT>pslIsProtein</TT> and <TT>pslScore</TT> 
			functions and also a useful function called 
			<TT>pslLoadAll</TT> that loads the psl file into a 
			linked list structure. The definition of the psl struct
			can be found in <em>kent/src/inc/psl.h</em>. 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat7"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Replicating web-based Blat &quot;I'm feeling lucky&quot; search results</b></FONT></TD>
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I generate the same search results as web-based
			Blat's &quot;I'm feeling lucky&quot; option using 
			command-line blat?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The code for the &quot;I'm feeling lucky&quot; Blat
			search orders the results based on the sort output 
			option that you selected on the query page. It then 
			returns the highest-scoring alignment of the first 
			query sequence. 
			<P>
			If you are sorting results by &quot;query, start&quot; 
			or &quot;chrom, start&quot;, generating the &quot;I'm
			feeling lucky&quot; result is straightforward:
			sort the output file by these columns, then select the 
			top result. 
			<P>
			To replicate any of the sort options involving score, 
			you first must calculate the score for each result in 
			your PSL output file, then sort the results by score or 
			other combination (<em>e.g.</em> &quot;query, 
			score&quot; and &quot;chrom, score&quot;).
			See the section on <A HREF="#blat4">Replicating 
			web-based Blat percent identity and score 
			calculations</A> for information on calculating the
			score.
			<P>
			Alternatively, you can try filtering your Blat PSL 
			output using either the <TT>pslReps</TT> or 
			<TT>pslCDnaFilter</TT> program available in the Genome
			Browser source code. For information on obtaining the
			source code, see our <A HREF="FAQlicense.html">FAQ</A> 
			on source code licensing and downloads. 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>

    <BR>

<!--Content Table------------------------------------------------------------>
<A NAME="blat8"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Using Blat for short sequences with maximum sensitivity </b></FONT></TD>
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I configure blat for short sequences with 
		   	 maximum sensitivity?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Here are some guidelines for configuring standalone 
			blat and gfServer/gfClient for these conditions:</P>
		    	<P>
			<UL>
			<LI>
			The formula to find the shortest query size that will
			guarantee a match (if matching tiles are not marked as
			overused) is: 2 * <em>stepSize</em> + <em>tileSize</em> - 1<br>
			For example, with <em>stepSize</em> set to 5 and
			<em>tileSize</em> set to 11, matches of query size
                        2 * 5 + 11 - 1 = 20 bp will be found if the query matches the target exactly.<br>
                        The <em>stepSize</em> parameter can range from 1 to <em>tileSize</em>.<br>
                        The <em>tileSize</em> parameter can range from 6 to 15. For protein, the
                        range starts lower.<br>
			For <em>minMatch</em>=1 (e.g., protein), the minimum guaranteed match length is:
                        1 * <em>stepSize</em> + <em>tileSize</em> - 1<br>
			<LI>
		 	Try using <em>-fine</em>.
			<LI>
			Use a large value for <em>repMatch</em> (e.g. 
			<em>-repMatch</em> = 1000000) 
			to reduce the chance of a tile being marked as 
			over-used.
			<LI>
			Do not use an <em>.ooc</em> file.
			<LI>
		 	Do not use <em>-fastMap</em>.
                        <LI>
			Do not use masking command-line options.
			</UL>
			<P>
			The above changes will make BLAT more sensitive, but 
			will also slow the speed and increase the memory usage. 
			It may be necessary to process one chromosome
			at a time to reduce the memory requirements. 
			<P>
			A note on filtering output: increasing the
			<em>-minScore</em> parameter value beyond one-half of
			the query size has no further effect.  Therefore, use
			either the <TT>pslReps</TT> or <TT>pslCDnaFilter</TT>
			program available in the Genome Browser source code to
			filter for the size, score, coverage, or quality
			desired.  For information on obtaining the
			source code, see our <A HREF="FAQlicense.html">FAQ</A> 
			on source code licensing and downloads. </P>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    </TD><TD WIDTH=15></TD></TR>


</TABLE>
</BODY></HTML>
