<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Browser: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../js/jquery.js'></script>
<script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Data and Downloads</b></FONT>
                </TD></TR></TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
		<UL>
                <LI><A HREF="#download1">Downloading sequence and annotation data</A>
                <LI><A HREF="#download32">Extracting sequence in batch from an assembly</A>
                <LI><A HREF="#download23">Downloading data from the UCSC DAS server</A>
                <LI><A HREF="#download27">Downloading the UCSC Genome Browser source</A>
                <LI><A HREF="#download2">Download restrictions</A>
                <LI><A HREF="#download3">Opening .fa files</A>
                <LI><A HREF="#download4">Data differences between downloaded data and browser display</A>
                <LI><A HREF="#download5">Strange characters in FASTA file</A>
                <LI><A HREF="#download6">Selection of GenBank ESTs</A>
                <LI><A HREF="#download7">EST strand direction</A>
                <LI><A HREF="#download8">Missing RefSeq ID</A>
                <LI><A HREF="#download9">Finished vs. draft segments</A>
                <LI><A HREF="#download10">chrN_random tables</A>
                <LI><A HREF="#download11">Chromosome Un</A>
                <LI><A HREF="#download12">Chromosome M</A>
                <LI><A HREF="#download13">N characters at beginning of human chr22</A>
                <LI><A HREF="#download30">Erroneous duplicated chrY_random region on Mouse Build 34 (mm6)</A>
                <LI><A HREF="#download26">Problems with Mouse Build 32 (mm4)</A>
                <LI><A HREF="#download25">Mapping chimp chromosome numbers to human chromsomes numbers</A>
                <LI><A HREF="#download28">Converting genome coordinates between assemblies</A>
                <LI><A HREF="#download33">Linking gene name with accession number</A>
                <LI><A HREF="#download31">Obtaining a list of Known Genes</A>
                <LI><A HREF="#download16">Repeat-masking data</A>
                <LI><A HREF="#download17">Availability of repeat-masked data</A>
                <LI><A HREF="#download24">RepeatMasker version differences - UCSC vs. Repeatmasker website</A>
                <LI><A HREF="#download18">Obtaining promoter sequence</A>
                <LI><A HREF="#download19">Data from Evolutionary Conservation Score tracks</A>
                <LI><A HREF="#download20">Minus strand coordinates - axtNet files</A>
                <LI><A HREF="#download21">Mapping UCSC STS marker IDS to those of other groups</A>
                <LI><A HREF="#download22">deCODE map data</A>
                <LI><A HREF="#download29">Direct MySQL access to data</A>
		<LI><A HREF="#download34">Name of fourth column in BED output</A>

		</UL>
		<A HREF="index.html">Return to FAQ Table of Contents</A>
		<BR>
            </TD><TD WIDTH=15></TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download1"></A>
<TABLE BORDER=0 WIDTH="100%">
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Downloading sequence and annotation data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I obtain the sequence and/or annotation data for a release?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			<p>Sequence and annotation data downloads are usually made available within 
			the first week of the release of a new assembly. The download directories
			are automatically updated nightly to incorporate additions and modifications
			to the data.</p>
			
			<p>You can download sequence and annotation data <a href="../goldenPath/help/ftp.html">
			using our FTP server</a>, but we recommend using rsync, which has the
                        advantage of starting up where it left off after a failure, when run again.
                        Please see the previous link for examples.
			
			<P>You can also download data from our 
			<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads</A>
			page or our <A HREF="../cgi-bin/das/dsn" TARGET=_blank>DAS 
		 	server</A>. To download a specific subset of the data or to configure the output
			format of the data, use the <A HREF="../cgi-bin/hgTables">Table 
			Browser</A>. For information on extracting a large set 
			of sequences from an assembly, see 	
			<A HREF="#download32">Extracting sequence in batch from
		    	an assembly</A>. 
			<P>
			For more information on using the UCSC DAS server, see 
			<A HREF="#downloads">Downloading data from the UCSC DAS server</A>.
                        <p>
                        <b>To quickly download large volumes of data you can use
                        UDR (UDT Enabled Rysnc):</b> UDR provides users much faster download rates.
                        Here is an example using UDR, once installed, to download all the mouse mm9
                        ENCODE information that amounts to several terabytes:</p>
                        <pre class="code">$ udr rsync -avP hgdownload.cse.ucsc.edu::goldenPath/mm9/encodeDCC/ /my/local/mm9/</pre>
                        Please read more about the new UDR method
                        <a href="../../goldenPath/newsarch.html#030315" target="_blank">here</a>.
                        </p>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download32"></A>
<TABLE BORDER=0 WIDTH="100%">
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Extracting sequence in batch from an assembly</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I have a lot of coordinates for an assembly and want to
			extract the corresponding sequences. What is the best 
		   	way to proceed? 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			<P>
			There are two ways to extract genomic sequence in batch 
			from an assembly:</P>
			<P>
			A. Download the appropriate fasta files from our 
			<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/">ftp server</A>
			and extract sequence data using your own tools or the
		 	tools from our source tree. This is the recommended
		 	method when you have very large sequence datasets or 
			will be extracting data frequently.
			Sequence data for most assemblies is located in the 
			assembly's &quot;chromosomes&quot; subdirectory on the
			downloads server. For example, the sequence for human
			assembly hg17 can be found in 
			<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17/chromosomes/">ftp://hgdownload.cse.ucsc.edu/goldenPath/hg17/chromosomes/</A>.
			You'll find instructions for obtaining our source 
			programs and utilities 
			<A HREF="FAQlicense.html#license3">here</A>. Some programs 
		   	that you may find useful are nibFrag and twoBitToFa,
			as well as other fa* programs. To obtain
			usage information about most programs, execute it 
			without arguments.</P>
			<P>
			B. Use the Table browser to extract sequence. This is a 
			convenient way to obtain small amounts of sequence. 
			<OL>
			<LI>Create a 
			<A HREF="../goldenPath/help/hgTracksHelp.html#CustomTracks">custom 
			track</A> of the genomic coordinates in 
			<A HREF="FAQformat.html#format1">BED format</A> and upload 
			into the Genome Browser. 
			<LI>Select the custom track in the Table browser, then 
			select the &quot;sequence&quot; output format to 
			retrieve data. We recommend that you save the file 
			locally as gzip. 
			</OL></P>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ========================================================== -->
<A NAME="download23"></A>
<TABLE BORDER=0 WIDTH="100%">
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Downloading data from the UCSC DAS server</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I download data using the UCSC DAS server?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The UCSC DAS server provides access to genome annotation data for all current assemblies
			featured in the Genome Browser. To view a list of the assemblies available from the
			DAS server and their base URLs, see 
			<A HREF="../cgi-bin/das/dsn">http://genome.ucsc.edu/cgi-bin/das/dsn</A>.
			<P>
			To construct a DAS query, combine an assembly's base URL with the 
			sequence entry point and type specifiers available for that assembly. The entry point 
			specifies chromosome position, and the type indicates the annotation table 
			requested. You can view the lists of entry points and types available for an assembly 
			with requests of the form:
<PRE>
	http://genome.ucsc.edu/cgi-bin/das/[db_name]/entry_points
	http://genome.ucsc.edu/cgi-bin/das/[db_name]/types
</PRE>
			where [db_name] is the UCSC name for the assembly, e.g. hg16, mm4.
			<P>
			For example, here is a query that returns all the records in the refGene table for the
			chromosome position chr1:1-100000 on the hg16 assembly:
<PRE>
	<A HREF="../cgi-bin/das/hg16/features?segment=1:1,100000;type=refGene">http://genome.ucsc.edu/cgi-bin/das/hg16/features?segment=1:1,100000;type=refGene</A>
</PRE>
			For more information on DAS, see the 
			<A HREF="http://www.biodas.org" TARGET=_blank>Biodas website</A> and the
			<A HREF="http://www.biodas.org/documents/spec.html" TARGET=_blank>DAS specification</A>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download27"></A>
<TABLE BORDER=0 WIDTH="100%">
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Downloading the UCSC Genome Browser source</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<P>
			<B><FONT COLOR="#006666">Question:</FONT></B><BR> 
			"Where can I download the Genome Browser source code and 
			executables?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The Genome Browser source code and executables are freely
			available for academic, nonprofit, and personal 
			use (see <A HREF="FAQlicense.html#license2">Licensing the Genome Browser
			or Blat</A> for commerical licensing requirements). 
			The latest version of the source code may be downloaded 
			<A HREF="http://hgdownload.cse.ucsc.edu/admin/jksrc.zip">here</A>.
			<P>
			See <A HREF="FAQblat.html#blat3">Downloading Blat source
			and documentation</A> for information on Blat downloads.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp;Download restrictions</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Do you have restrictions on the amount of downloads one can do?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Generally, we'd prefer that you not hit our interactive site with programs, 
			unless they are themselves front ends for interactive sites. We can handle 
			the traffic from all the clicks that biologists are likely to generate, 
			but not from programs. Program-driven use is limited to a maximum of one 
			hit every 15 seconds and no more than 5,000 hits per day. 
			<P>
			If you need to run batch Blat jobs, see
			<A HREF="FAQblat.html#blat3">Downloading Blat source
			and documentation</A> for a copy of Blat you can run
			locally.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Opening .fa files</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I am trying to look at the final decoding of the human genome. How can I
			open the *.fa files?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Microsoft Word or any program that can handle large text files will do. 
			Some of the chromosomes begin with long blocks of <em>N</em>'s. You may want 
			to search for an <em>A</em> to get past them. 
			<P>
			Unless you have a particular need 
			to view or use the raw data files, you might find it more interesting to 
			look at the data using the Genome Browser. Type the name of a gene in which
			you're interested into the position box (or use the default position), 
			then click the submit button. In the resulting Genome Browser  
			display, click the DNA link on the menu bar at the top of the page. 
			Select the Extended case/color options button at the bottom of the 
			next page. Now you can color the DNA sequence to display which portions are 
			repeats, known genes, genetic markers, etc.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Data differences between downloaded data and browser display</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I downloaded the genome annotations from your MySQL database tables, but the 
			mRNA locations didn't match what was showing in the Genome Browser. Shouldn't they 
			be in synch?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Yes. The Genome Browser and Table Browser are both driven by the same 
			underlying MySQL database. Check that your downloaded tables are from
			the same assembly version as the one you are viewing in the Genome Browser. If the 
			assembly
			dates don't match, the coordinates of the data within the tables may differ. 
			In a very rare instance, you could also be affected 
			by the brief lag time between the update of the live databases underlying the Genome
			Browser and the time it takes for text dumps of these databases to become available 
			in the downloads directory.  
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Strange characters in FASTA file</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I noticed several characters other than <em>A</em>, <em>C</em>, <em>G</em>, 
			<em>T</em>, and <em>N</em> in my fasta file, for example <em>y</em>, <em>k</em>, 
			<em>s</em>, etc. Is the file corrupted or are these characters valid?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The characters most commonly seen in sequence are <em>A</em>, <em>C</em>, <em>G</em>, 
			<em>T</em>, and <em>N</em>, but there are 
			several other valid characters that are used in clones to indicate
			ambiguity about the identity of certain bases in the sequence. It's not uncommon to 
			see these &quot;wobble&quot; codes at polymorphic positions in DNA sequences. The
			following chart (IUPAC-IUB Symbols for Nucleotide Nomenclature: Cornish-Bowden
			(1985). <em>Nucl. Acids Res.</em> 13:3021-3030) lists nucleotide symbols, including those
			used for ambiguity:
			<PRE>
          		--------------------------------------
          		Symbol    Meaning      Nucleic Acid
          		--------------------------------------
           		A            A           Adenine
           		C            C           Cytosine
           		G            G           Guanine
           		T            T           Thymine
           		U            U           Uracil
           		M          A or C
           		R          A or G        Purine
           		W          A or T
           		S          C or G
           		Y          C or T        Pyrimidine
           		K          G or T
           		V        A or C or G
           		H        A or C or T
           		D        A or G or T
           		B        C or G or T
           		X      G or A or T or C
           		N      G or A or T or C
			</PRE>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Selection of GenBank ESTs</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I am interested in ESTs. How do you select which ones from GenBank to display in the 
			Genome Browser?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			All ESTs in GenBank on the date of the track data freeze for the given organism are 
			used - none are discarded. When two ESTs have identical sequences, both are 
			retained because this can be significant corroboration of a splice site. </P>
			<P>
			ESTs are aligned against the genome using the Blat program. When a single EST aligns 
			in multiple places, the alignment having the highest base identity is found. Only 
			alignments that have a base identity level within a selected percentage of 
			the best are kept. Alignments 
			must also have a minimum base identity to be kept. For more information on the 
			selection criteria specific to each organism, consult the description page accompanying
			the EST track for that organism.</P>
			<P>
			The maximum intron length allowed by Blat is 500,000 bases, which may 
			eliminate some ESTs with very long introns that might otherwise align. If an EST 
			aligns non-contiguously (i.e. an intron has been spliced out), it is also a candidate 
			for the Spliced EST track, provided it meets various quality controls for intron and 
			exon length and match quality.  Start and stop coordinates of each alignment block are 
			available from the appropriate table within the 
			<A HREF="../cgi-bin/hgTables">Table Browser</A>.</P>
			<P>
			Note that only 250 EST tracks can be viewed at a time within the browser. If more 
			than 250 tracks exist for the selected region, the display defaults to a denser display
			mode to 
			prevent the user's web browser from being overloaded. You can restore the EST track 
			display to a fuller display mode by zooming in on the chromosomal range or by using 
			the EST track filter to restrict the number of tracks displayed.</P>
			<P>
			For tracks such as Non[Organism] ESTs and Non[Organism] mRNAs, some selection is done
			on the full set at GenBank. If a sequence is too divergent from the organism's genome 
			to generate a significant Blat hit, it is not included in the track.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download7"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; EST strand direction</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Could you help me with my interpretation of EST data? If the EST is taken
			from the minus (-) strand, does this always mean that the transcript is generated 
			on the minus strand? Are two corresponding ESTs that are assigned 
			- and + always complementary? 
			<P>
			I want to confirm the strand assignment for two 
			human ESTs: 
			<UL>
			<LI>BQ016549 (chr22:22,310,674-22,332,143 on hg18): + strand in text and - strand in graphical 
			display 
			<LI>AA928010 (chr22:20,345,264-20,354,528 on hg18): - strand in text and + strand in graphical 
			display. 
			</UL>
			The graphical display goes with the orientation of the gene in that location."
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR>
			From the examples above, it can be seen that the strand to which an EST aligns is not 
                        necessarily reflected in the direction of transcription shown by the arrows in the 
                        display. When UCSC downloads mRNAs and ESTs from GenBank and aligns them to a genome 
			assembly using Blat, each EST aligns to the + or - strand (forward or reverse direction)
			of the genome, which we record as + or - in the strand field of the corresponding database 
			table, e.g. all_ests or chrN_est. The strand information (+/-) therefore 
                        indicates the direction of the match between the EST and the matching genomic 
                        sequence. It bears no relationship to the direction of transcription of the RNA with 
                        which it might be associated. Determining the direction of transcription for ESTs is 
                        not an easy task so we do some calculations to make the best guess for the 
                        transcription direction. 
			<P>
			ESTs are sequenced from either the 5' or the 3' end. When sequenced from the 5' end, the resulting
			sequence is the same as that of the mRNA which it represents. With a 3' end read, the resulting 
		 	sequence matches the opposite strand of the cDNA clone. Therefore, it is the reverse complement of 
			the actual mRNA sequence.  A problem occurs if the EST contributor reverse-complements
		        the 3'-read sequence before depositing it into GenBank, with the idea that people will want 
			the mRNA (transcription-direction) sequence. It is not always possible to determine if this has 
			been done. Therefore, we do some calculations to try to determine the correct direction of 
			transcription for the EST sequence. 
 			<P>
			If an EST alignment produces canonical introns (with gt-ag splice-site pairs), this is used 
			to determine the transcription direction. For example when an EST is aligned to the genome, a 
			canonical intron would look like this:
			<BLOCKQUOTE>
			<B>NNNNexonNNNNgtnnnnintronnnnnnnnagNNNNexon</B></P>
			</BLOCKQUOTE>
			<P>
			Here, the two nucleotides on either end of the intron show the canonical gt-ag splice site pairs. 
			To find transcription direction, we use a method that relies on finding gt-ag canonical pairs in one 
			direction more often than in the opposite direction. The calculation is:
			<BLOCKQUOTE>
			<B>gt/ag introns minus ct/ac introns = intronOrientation</B> </P>
			</BLOCKQUOTE>
 			<P>	
			The sign of this calculated intronOrientation field (stored in the estOrientInfo table) shows the 
			orientation of the transcript relative to the EST. Therefore, if intronOrientation is positive, 
			then the EST appears in the display with the arrows pointing in the same direction as the EST 
			alignment. If intronOrientation is negative, then the arrows point in the opposite direction. If 
			no introns exist or all of the introns are non-canonical, then intronOrientation is set to zero.  
			In both BQ016549 and AA928010 (in the example above), the intronOrientation is negative; therefore, 
			the arrows on the Genome Browser display point in the opposite direction to that indicated by the 
			alignment on the EST details page. <B>Note:</B> A low intronOrientation number can cause an incorrect
			assignment of transcription direction when calculated in this way.
			<P>
			The alignment details pages and the Table Browser do not take the intron orientation
			into account. They show only the alignment of the 
			GenBank sequence (as given) to the genome. If the alignment is used to 
			retrieve DNA sequence from the genome, the DNA sequence will look 
			similar to the GenBank sequence (not its complement).
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download8"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Missing RefSeq ID</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Why isn't my refseq ID in your database?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			It may have been added after we last downloaded data from Genbank, or it may have 
			been replaced or removed. You can check the submission date and status of an accession
			on the 
			<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/" TARGET=_blank>NCBI 
			Entrez Nucleotide site</A>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download9"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Finished vs. draft segments</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Do chr<em>N</em>.fa tables contain both finished and draft segments? If so, 
			how do you determine which segments are finished?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Yes, these tables contain both finished and draft segments. Use the 
			corresponding chr<em>N</em>_gold table to look them up. The quality of the draft 
			varies. In 
			general, the larger the contig it is in, the better the quality. The 
			quality of the last 500 bases on either end of a contig tends to be 
			lower than the rest of the contig. 
			<P>
			How do you determine the accuracy? The 
			base-calling program <A HREF="http://www.phrap.org/" TARGET=_blank>Phred</A> analyzes 
			the traces from the sequencing machines 
			and assigns a quality score to these. These quality scores are used by the 
			<A HREF="http://www.phrap.org/" TARGET=_blank>Phrap</A> assembly program, which gives 
			quality scores for the bases on the assembly as well.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download10"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; chrN_random tables</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What are the chr<em>N</em>_random_[table] files in the human assembly? Why are they 
			called random? Is there something biologically random about the sequence in 
			these tables or are they just not placed within their given chromosomes?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			In the past, these tables contained data related to sequence that is 
			known to be in a particular chromosome, but could not be reliably ordered 
			within the current sequence. 
			<P>
			Starting with the April 2003 human assembly, these tables also include data for 
			sequence that is not in a finished state, but whose location in the chromosome is 
			known, in addition to the unordered sequence.  
			Because this sequence is not quite finished, it could not be included in the 
			main &quot;finished&quot; ordered and oriented section of the chromosome.  
			<P>
			Also, in 
			a very few cases in the April 2003 assembly, the random files contain data related to sequence for alternative 
			haplotypes.  
			This is present primarily in chr6, where we have included two alternative
			versions of the MHC region in chr6_random. There are a few clones in 
			other chromosomes that also correspond to a different haplotype.  Because the 
			primary reference sequence can only display a single haplotype, these 
			alternatives were included in random files.  In subsequent assemblies, 
			these regions have been moved into separate files (<em>e.g.</em> chr6_hla_hap1).
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download11"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Chromosome Un</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What is ChrUn?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			ChrUn contains clone contigs that can't be confidently placed on a 
			specific chromosome. For the chr<em>N</em>_random and chrUn_random files, we 
			essentially just concatenate together all the contigs into short 
			pseudo-chromosomes. The coordinates of these are fairly arbitrary, 
			although the relative positions of the coordinates 
			are good within a contig. You can find more information about the data organization 
			and format on the <A HREF="../goldenPath/datorg.html">Data
			Organization and Format</A> page.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download12"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Chromosome M </b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What is chromosome M (chrM)?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Mitochondrial DNA.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download13"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; N characters at beginning of human chr22</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"When I download human chr22 from your web site, the unzipped file contains only 
			<em>N</em>'s."
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			There is a large block of <em>N</em>'s at the beginning and end of chr22. Search 
			for an <em>A</em> to bypass the initial group of <em>N</em>'s.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download30"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Erroneous duplicated chrY_random region on Mouse Build 34 (mm6)</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"On the mm6 assembly, I've found duplicate contigs 
			that are placed on both chrY and chrY_random. Is this 
			intentional?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			On the mm6 assembly, chrY_random erroneously contains 
			a region duplicated from chrY. Because NCBI 
			discovered this assembly problem after the UCSC 
			Genome Browser was processed, we were not able to 
			remove it from mm6 prior to the browser's release. 
			The duplicated section occupies chrY:1-696,521 and 
			chrY_random:29,615,053-30,311,573 (the end of the
			chromosome) and includes the following repeated 
			fragments: 
			<UL>
			<LI>AC139318.5 
			<LI>AC134433.3
			<LI>AC145392.2
			<LI>AC148319.2
			<LI>AC145571.3 
			<LI>AC145393.4 
			</UL>
			The fragments
			are assembled into the contig NT_111995 for 
			chrY_random and also appear (under different names) 
			as regions on contigs MmY_110865_34, MmY_78990_34 
			and NT_078925.</P>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Content Table ============================================================================= -->
<A NAME="download26"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Problems with Mouse Build 32 (mm4)</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I have heard that the Build 32 mouse assembly isn't 
			as good as the Build 30 assembly. Can you clarify?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Unfortunately, there appear to be some problems with 
			the Build 32 assembly. Ensembl has conducted an analysis 
			of the assembly and has attributed the
			problems to incorrect mapping information that led to 
			the generation of artificial duplications and some 
			incorrect flips in orientation. You can read more 
			information about the problems Ensembl identified and 
			review a list of the chromosomes and genes most likely 
			to be affected by these issues on the  Ensembl
			<A HREF="http://www.ensembl.org/Mus_musculus/"
			TARGET=_blank><em>Mus musculus</em></A> web page.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download25"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Mapping chimp chromosome numbers to human chromsomes numbers</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			How do the chimp and human chromosome numbering
			schemes compare?
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The following table shows the mapping of chromosomes in 
			the chimp draft assemblies to human chromosomes.
		 	Starting with the panTro2 assembly, the numbering scheme
			has been changed to reflect a new standard that 
			preserves orthology with human chromosomes. Initially 
			proposed by E.H. McConkey in 2004, the new numbering 
			convention was subsequently endorsed by the 
			International Chimpanzee Sequencing and Analysis 
			Consortium. This standard assigns the identifiers 
			"2a" and "2b" to the two chimp chromosomes that fused in
			the human genome to form chromosome 2 and renumbers the
			other chromosomes to more closely match their human 
			counterparts. As a result, chromosomes 2 and
			23 (present in the panTro1 assembly) do not exist in
			later versions.

        		<!--chrom table-->
			<BR>
			<BLOCKQUOTE>
            		<TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 >            
			    <TR>
			    <TD nowrap><FONT color="#006666"><B>Human Chr</B></FONT></TD>
                	    <TD nowrap><FONT color="#006666"><B>Chimp Chr (panTro1)</B></FONT></TD>
                	    <TD nowrap><FONT color="#006666"><B>Chimp Chr (panTro2)</B></FONT></TD>
                	    </TR>
  			    <TR><TD ALIGN=center>1</TD><TD ALIGN=center>1</TD><TD ALIGN=center>1</TD></TR>
  			    <TR><TD ALIGN=center>2 (part)</TD><TD ALIGN=center>12</TD><TD ALIGN=center>2a</TD></TR>
  			    <TR><TD ALIGN=center>2 (part)</TD><TD ALIGN=center>13</TD><TD ALIGN=center>2b</TD></TR>
  			    <TR><TD ALIGN=center>3</TD><TD ALIGN=center>2</TD><TD ALIGN=center>3</TD></TR>
  			    <TR><TD ALIGN=center>4</TD><TD ALIGN=center>3</TD><TD ALIGN=center>4</TD></TR>
  			    <TR><TD ALIGN=center>5</TD><TD ALIGN=center>4</TD><TD ALIGN=center>5</TD></TR>
  			    <TR><TD ALIGN=center>6</TD><TD ALIGN=center>5</TD><TD ALIGN=center>6</TD></TR>
  			    <TR><TD ALIGN=center>7</TD><TD ALIGN=center>6</TD><TD ALIGN=center>7</TD></TR>
  			    <TR><TD ALIGN=center>8</TD><TD ALIGN=center>7</TD><TD ALIGN=center>8</TD></TR>
  			    <TR><TD ALIGN=center>9</TD><TD ALIGN=center>11</TD><TD ALIGN=center>9</TD></TR>
  			    <TR><TD ALIGN=center>10</TD><TD ALIGN=center>8</TD><TD ALIGN=center>10</TD></TR>
  			    <TR><TD ALIGN=center>11</TD><TD ALIGN=center>9</TD><TD ALIGN=center>11</TD></TR>
  			    <TR><TD ALIGN=center>12</TD><TD ALIGN=center>10</TD><TD ALIGN=center>12</TD></TR>
  			    <TR><TD ALIGN=center>13</TD><TD ALIGN=center>14</TD><TD ALIGN=center>13</TD></TR>
  			    <TR><TD ALIGN=center>14</TD><TD ALIGN=center>15</TD><TD ALIGN=center>14</TD></TR>
  			    <TR><TD ALIGN=center>15</TD><TD ALIGN=center>16</TD><TD ALIGN=center>15</TD></TR>
  			    <TR><TD ALIGN=center>16</TD><TD ALIGN=center>18</TD><TD ALIGN=center>16</TD></TR>
  			    <TR><TD ALIGN=center>17</TD><TD ALIGN=center>19</TD><TD ALIGN=center>17</TD></TR>
  			    <TR><TD ALIGN=center>18</TD><TD ALIGN=center>17</TD><TD ALIGN=center>18</TD></TR>
  			    <TR><TD ALIGN=center>19</TD><TD ALIGN=center>20</TD><TD ALIGN=center>19</TD></TR>
  			    <TR><TD ALIGN=center>20</TD><TD ALIGN=center>21</TD><TD ALIGN=center>20</TD></TR>
  			    <TR><TD ALIGN=center>21</TD><TD ALIGN=center>22</TD><TD ALIGN=center>21</TD></TR>
  			    <TR><TD ALIGN=center>22</TD><TD ALIGN=center>23</TD><TD ALIGN=center>22</TD></TR>
  			    <TR><TD ALIGN=center>X</TD><TD ALIGN=center>X</TD><TD ALIGN=center>X</TD></TR>
  			    <TR><TD ALIGN=center>Y</TD><TD ALIGN=center>Y</TD><TD ALIGN=center>Y</TD></TR>
	    	    </TABLE>
		    </BLOCKQUOTE>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download28"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Converting genome coordinates between assemblies</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I've been researching a specific area of the human genome on the current assembly, 
			and now you've just released a new version. Is there an easy way to locate 
			my area of interest on the new assembly?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			You can migrate data from one assembly to another by using the 
			<A HREF="../cgi-bin/hgBlat">blat</A> alignment tool 
			or by converting assembly coordinates</A>.  There are two conversion 
			tools available on the Genome Browser web site: the
			Convert utility and the LiftOver tool.
			The Convert utility, 
			which is accessed from the menu on the Genome Browser 
			annotation tracks page, supports forward, reverse, and 
			cross-species conversions, but does not accept batch 
			input. 
			The <A HREF="../cgi-bin/hgLiftOver">LiftOver</A> tool,
			accessed via the Utilities link on the Genome Browser 
			home page, also supports forward, reverse, and 
			cross-species conversions, as well as batch conversions.
			<P>
			If you wish to update a large number of coordinates	
			to a different assembly and have access to a Linux 
			platform, you may find it useful to try the command-line
			version of the LiftOver tool. The executable file for 
			this utility can be downloaded 
			<A HREF="http://hgdownload.cse.ucsc.edu/admin/exe/" 
			TARGET=_blank>here</A>. 
			LiftOver requires a UCSC-generated <em>over.chain</em> 
			file as input. Pre-generated files are available for 
			selected assemblies from the 
			<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#liftover">Downloads</A> page.
			If the desired file is not available, send a request to
			the 
			<A HREF="../contacts.html">genome mailing list</A> 
			and we may be able to  provide you with one.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>


<!-- Content Table ============================================================================= -->
<A NAME="download33"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Linking gene name with accession number</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I have the accession number for a gene and would like to
			link it to the gene name. Is there a table that shows both
			pieces of information?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			If you are looking at the RefSeq Genes, the 
			<em>refFlat</em> 
			table contains both the gene name (usually a 
			HUGO Gene Nomenclature Committee ID) and its accession 
		    	number. For the Known Genes, 
			use the <em>kgAlias</em> table.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Content Table ============================================================================= -->
<A NAME="download31"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Obtaining a list of Known Genes</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How can I obtain a complete list of all the genes in
			the UCSC Known Genes table for a particular organism?
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
		    	To obtain a complete copy of the entire Known Genes 
			data set for an organism, open the Genome Browser
			<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Downloads page</A>, 
			jump to the section specific to the organism, click the
			Annotation database link in that section, then click the
			link for the <em>knownGene.txt.gz</em> table.</P>
			<P>
			Data for a specific region or chromosome may be
			obtained from the Table Browser by selecting the 
			&quot;Genes and Gene Prediction Tracks&quot; group, the
			&quot;UCSC Genes&quot; track and the 
			&quot;knownGene&quot; table. Set the position to the 
			region of interest, then click the &quot;get 
			output&quot; button.
			</P>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>


<!-- Content Table ============================================================================= -->
<A NAME="download16"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Repeat-masking data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What version of RepeatMasker do you use on your data?
			Which flags do you use?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			UCSC uses the latest versions of RepeatMasker and
			repeat libraries available on the date when the 
			assembly data is processed. RepeatMasker version 
			information can usually be found in the README text for
			the assembly's bigZips 
			<A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">downloads</A> directory. 
			<P>Masking is done using the RepeatMasker <em>-s</em> 
			flag. For mouse repeats, we also use <em>-m</em>. 
			In addition to RepeatMasker, we use the Tandem Repeat 
			Finder (trf) program, masking out repeats of period 12 
			or less. The repeats are just &quot;soft&quot; masked. 
			Alignments are allowed to extend through repeats, but 
			not initiate in them.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download17"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Availability of repeat-masked data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Are the repeat annotation files available for every chromosome?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Yes, you can obtain the repeat-masked files via the Table Browser or from the 
			organism's annotation database downloads directory. The RepeatMasker annotation 
			tables are named 
			chr<em>N</em>_rmsk (where <em>N</em> represents the chromosome number) and the 
			Tandem Repeat Finder (TRF) tables are named simpleRepeat.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download24"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; RepeatMasker version differences - UCSC vs. RepeatMasker website</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"When I run RepeatMasker independently from the 
			RepeatMasker web server, my results vary from those of 
			UCSC.  What's the cause?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			UCSC occasionally uses updated versions of the 
			RepeatMasker software and repeat libraries that are not 
			yet available on the RepeatMasker website (see 
			<A HREF="#download16">Repeat-masking data</A> for more
			information).
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>
<!-- Content Table ============================================================================= -->
<A NAME="download18"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Obtaining promoter sequence</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How can I fetch promoter sequence upstream of a gene?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The UCSC Genome Browser offers several ways to obtain this information, 
			depending on your requirements.
			<P>
			The Genome Browser <a href="http://hgdownload.cse.ucsc.edu/downloads.html">downloads 
			site</a> provides prepackaged downloads of 1000 bp, 2000 bp, and 5000 bp upstream 
			sequence for RefSeq genes that have a coding portion and annotated 5' and 3' UTRs. You 
			can obtain these from the bigZips downloads directory for the assembly of interest. 
			<P>
			To fetch the upstream sequence for a specific gene, use the 
			<A HREF="../cgi-bin/hgTables">Table Browser</A>.
			Enter the genome, assembly, and select the knownGene table. Paste the gene name 
			or accession number in the identifier field. Choose sequence for the output format 
			type, then click the get output button. On the next page, select genomic.  On the 
			final page, you will have the opportunity to configure the amount of upstream 
			promoter sequence to fetch, along with several other options. Click Get Sequence 
			when you've finished configuring the output. 
			<P>
			You can also use the Genome Browser to obtain sequence for a specific gene. 
			Open the Genome Browser window to display the gene in which you're
			interested. Click the entry for the gene in the RefSeq or Known Genes track, then
			click the Genomic Sequence link. Alternatively, you can click the DNA link in 
			the top menu bar of the Genome Browser tracks window to access options for displaying 
			the sequence. 
			<P>
			The Stanford Human Promoters track on the 
			<A HREF="../goldenPath/customTracks/custTracks.html">UCSC 
			Custom Annotation Tracks page</A> shows promoters for some of the human assemblies.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download19"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Data from Evolutionary Conservation Score tracks</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Where can I download the conservation score data from the Human/Mouse 
			Evolutionary Conservation Score track?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The conservation score data are stored in a group of tables in the annotation 
			database downloads directory. 
			The naming conventions of the tables vary among releases. In earlier
			assemblies, table names are of the form chr<em>N</em>_humMusL, chr<em>N</em>_zoom1_humMusL, and
			or chr<em>N</em>_zoom2500_humMusL. In later releases, the tables are named using 
			specific release numbers, such as chr<em>N</em>_hg16Mm3. The tables within a given set
			differ by the number of bases/score interval and are used to generate the browser
			displays at different zooming levels. 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download20"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Minus strand coordinates - axtNet
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I downloaded the axtNet alignments between the latest human and mouse assemblies. 
			I found that some of the alignments listed in the axtNet
			files do not agree with what is shown in the browser." 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR>
			Is this alignment on the minus strand? Minus strand coordinates in axt files 
			are handled differently from how they are handled in the Genome Browser. To convert 
			axt minus strand coordinates to Genome Browser coordinates, use:
			<PRE>
      	start = chromSize + 1 - axtEnd
      	end = chromSize + 1 - axtStart</PRE>
			See an explanation of coordinate transforms in the <a
			href="http://genomewiki.ucsc.edu/index.php/Coordinate_Transforms"
			target=blank>genomeWiki</a>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download21"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Mapping UCSC STS marker IDs to those of other groups</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do I map the STS genetic marker IDs in the genome browser to the 
			IDs assigned by other groups? "
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			We assign our own IDs to each of the STS markers, but we also track 
			the UniSTS IDs for each marker in the downloadable stsInfo2 table.  
			To determine the location of a specific marker, look up the marker's name
			in the stsAlias table to determine the UCSC ID assigned to the 
			marker, and then use this ID to look it up in the stsMap table where the marker 
			is located. For example, D10S249 has UCSC ID 2880 and is located at chr10:240791-241019.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!-- Content Table ============================================================================= -->
<A NAME="download22"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; deCODE map data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Where can I get more information about the deCODE map?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			You can obtain this information from the combination of a couple of tables. 
			The stsMap table contains the physical position of all STS markers, 
			including those on the deCODE map. This file also contains information about 
			the position on the genome-wide maps, including the deCODE map. A second file, 
			stsInfo2, contains additional information about each marker, including aliases, 
			primer sequence information, etc. This table is related to the first table by an 
			ID (the identNo field in both files). 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Content Table ========================================================== -->
<A NAME="download29"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Direct MySQL access to data</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Is it possible to run SQL queries directly on the
			database rather than using the Table Browser interface?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			<p>Yes. See our documentation on <a href="../goldenPath/help/mysql.html">Downloading Data
			 using MySQL</a>. </P>

                         <p>Connect to the MySQL server using the command:</p>
	<pre class="code">
mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A</pre>

		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    <BR>

<!-- Content Table ============================================================================= -->
<A NAME="download34"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
            <TR><TD>                
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD>                
                <!--cheap trick to get background on row to be continuous-->
                <TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
                    <TR><TD ALIGN="left">
                    <FONT SIZE="4"><b>&nbsp; Name of fourth column in BED output</b></FONT></TD>
                    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR>
                </TABLE>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
                <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
                    <TD>
                    <P>
                        <B><FONT COLOR="#006666">Question: </FONT></B><BR>
                        "When using the Table Browser to extract exons from a Gene track, what does the 'Name' column (fourth BED column) refer to?"
                        <P>
                        <B><FONT COLOR="#006666">Response:</FONT></B><BR>
			<BR> The fourth column of the BED output contains a lot of information separated by underscores. For example:<BR>
                        <PRE>
     uc009vjk.2_cds_1_0_chr1_324343_f</PRE>
                        
                        This information is represented as follows:<BR>
                        <PRE>
     ucscId_sequenceType_sequenceTypeNumber_basesAdded_chromosome_positionOfFirstBaseOfItem_strand</PRE>
			<UL>
			<LI>
			UCSC ID: our identification for the transcripts in the UCSC Genes track.
			<LI>
			Sequence Type: exons, introns, cds, utr5, etc.
			<LI>
			Sequence Type Number: for every transcript, there will be a row for 
			each sequence type (cds or intron) and this identifies which is represented in this 
			row;  the first is denoted with 0. So, if you requested exons, and a particular 
			transcript has 10 exons, you will see a row for each one and in this position 
			they will be numbered 0-9.
			<LI>
			Bases Added: number of bases added to the regions requested.
			<LI>
			Chromosome: chromosome number the item is on.
			<LI>
			Position of First Base of Item: if you have specified bases added to the 
			requested features (for example, Exons plus 10 bases on each end), then 
			columns 2 and 3 of the output wouldn't be the exact coordinates of the exon, 
			they would start and end 10 bases before/after the exon. So, this part of 
			the information is an easy way to see where the actual feature starts as 
			displayed in the browser. It is "as displayed in the browser" because the 
			coordinates in our tables almost always have 0-based starts (as they do 
			in columns 2 and 3 of this output) but display as 1-based in the browser 
			(for more info see this <A HREF=FAQtracks.html#tracks1>FAQ</A>), 
			but this start position listed in this section of the 4th column is actually 1 based. 
			It will be the exact coordinate the feature starts on as displayed in the browser.
			<LI>
			Strand: forward(f) or reverse(-) strand.
			</UL>
		    </TD><TD WIDTH=15></TD>
                </TR>
            </TABLE>
            <br></TD></TR>
        </TABLE>
        </TD></TR>
    </TABLE>

    </TD><TD WIDTH=15></TD></TR>
</TABLE>
</BODY></HTML>
