<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
	"http://www.w3.org/TR/html4/loose.dtd">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
	<script type="text/javascript" src="/js/jquery.js"></script>
	<script type="text/javascript" src="/js/staticDoc.js"></script>
</HEAD>
<BODY BGCOLOR="#FFF9D2" LINK="#0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
    <TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../js/jquery.js'></script>
<script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../cgi-bin/hgMenubar"--></TD></TR>



<!--Content Tables-->
    <TR><TD COLSPAN=2>
    <!--outer table is for border purposes-->
   	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
	    <TR><TD>
           	<TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
		    <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD> <FONT SIZE="4"><b>&nbsp; Frequently Asked Questions: Data File Formats</b></FONT> 
			    </TD></TR>
			</TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
			    <TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
                    <b> General formats:</b>
                    <UL>
                        <LI><A HREF="../goldenPath/help/axt.html">Axt format</A>
                        <LI><A HREF="#format5.1">BAM format</A>
                        <LI><A HREF="#format1">BED format</A>
                        <LI><A HREF="#format1.7">BED detail format</A>
                        <LI><A HREF="#format1.8">bedGraph format</A>
                        <LI><A HREF="#format1.5">bigBed format</A>
			<LI><A HREF="#format9.1">bigGenePred table format</A>
                        <LI><A HREF="#format6.1">bigWig format</A>
                        <LI><A HREF="../goldenPath/help/chain.html">Chain format</A>
                        <LI><A HREF="#format9">GenePred table format</A>
                        <LI><A HREF="#format3">GFF format</A>
                        <LI><A HREF="#format4">GTF format</A>
			<LI><A HREF="#format20">HAL format</A>
                        <LI><A HREF="#format5">MAF format</A>
                        <LI><A HREF="#format6.5">Microarray format</A>
                        <LI><A HREF="../goldenPath/help/net.html">Net format</A>
                        <LI><A HREF="#format10">Personal Genome SNP format</A>
                        <LI><A HREF="#format2">PSL format</A>
                        <LI><A HREF="#format10.1">VCF format</A>
                        <LI><A HREF="#format6">WIG format</A>
                    </UL>
                    <a name="ENCODE"></a><b> ENCODE-specific formats:</b>
                    <UL>
                        <LI><A HREF="#format13">ENCODE broadPeak format</A>
                        <LI><A HREF="#format14">ENCODE gappedPeak format</A>
                        <LI><A HREF="#format12">ENCODE narrowPeak format</A>
                        <LI><A HREF="#format16">ENCODE pairedTagAlign format</A>
                        <LI><A HREF="#format17">ENCODE peptideMapping format</A>
                        <LI><A HREF="#format11">ENCODE RNA elements format</A>
                        <LI><A HREF="#format15">ENCODE tagAlign format</A>
                    </UL>
                    <b> Download only formats:</b>
                    <UL>
                        <LI><A HREF="#format7">.2bit format</A>
                        <LI><A HREF="#format18">.fasta format</A>
                        <LI><A HREF="#format19">.fastQ format</A>
                        <LI><A HREF="#format8">.nib format</A>
                    </UL>
                    <P>
                    <A HREF="index.html">Return to FAQ Table of Contents</A>
                    <BR>
                 </TD><TD WIDTH=15></TD></TR>
			</TABLE><BR>
        	     </TD></TR>
		</TABLE>
    	     </TD></TR>
	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format1"></A>
    <!--outer table is for border purposes-->        
    	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
             <TR><TD>            
        	<TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    	     <TR><TD>                
  	    		<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        	     <TR><TD>                
	        	     <!--cheap trick to get background on row to be continuous-->
	   			<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    		     <TR><TD ALIGN="left"><FONT SIZE="4"><b>&nbsp; BED format</b></FONT></TD> 
		    		     <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            		     </TD></TR>
  	        		</TABLE>                
	        	     </TD></TR>
  	    		</TABLE>                
   	    		<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        	    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD WIDTH=10>&nbsp;</TD>
		    	    <TD>
	            	    <P>
<P>BED format provides a flexible way to define the data lines that
are displayed in an annotation track. BED lines have three required fields and
nine additional optional fields. The number of fields per line must be consistent
throughout any single set of data in an annotation track. The order of the
optional fields is binding: lower-numbered fields must always be populated if
higher-numbered fields are used.

<P>If your data set is BED-like, but it is very large (over 50MB) and you would like to keep
it on your own server, you should use the
<A HREF="../goldenPath/help/bigBed.html">bigBed</A> data format.

<P>The first three required BED fields are:</P>
<OL>
        <LI><B>chrom</B> - The name of the chromosome (e.g. chr3, chrY,
        chr2_random) or scaffold (e.g. scaffold10671).
        <LI><B>chromStart</B> - The starting position of the feature in the
        chromosome or scaffold. The first base in a chromosome is numbered 0.
        <LI><B>chromEnd</B> - The ending position of the feature in the
        chromosome or scaffold. The <I>chromEnd</I> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <I>chromStart=0, chromEnd=100</I>, and span
        the bases numbered 0-99.
</OL>

<P>The 9 additional optional BED fields are:</P>
<OL START=4>
        <LI><B>name</B> - Defines the name of the BED line.  This label is
        displayed to the left of the BED line in the Genome Browser
        window when the track is open to full display mode or directly to the
        left of the item in pack mode.
        <LI><B>score</B> - A score between 0 and 1000. If the track line
        <I>useScore</I> attribute is set to 1 for this annotation data set, the
        <I>score</I> value will determine the level of gray in which
        this feature is displayed (higher numbers = darker gray).
        This table shows the Genome Browser's translation of BED score values into
        shades of gray:
<TABLE CELLPADDING=0 BORDER=1>
<TR><TD style="background-color: white">
    <TABLE CELLSPACING=5>
    <TR><TD>shade</TD>
        <TD style="background-color: #e2e2e2">&nbsp;</TD>
        <TD style="background-color: #c6c6c6">&nbsp;</TD>
        <TD style="background-color: #aaaaaa">&nbsp;</TD>
        <TD style="background-color: #8d8d8d">&nbsp;</TD>
        <TD style="background-color: #717171">&nbsp;</TD>
        <TD style="background-color: #555555">&nbsp;</TD>
        <TD style="background-color: #383838">&nbsp;</TD>
        <TD style="background-color: #1c1c1c">&nbsp;</TD>
        <TD style="background-color: #000000">&nbsp;</TD></TR>
    <TR><TD>score in range&nbsp;&nbsp;</TD>
        <TD>&le; 166</TD>
        <TD>167-277</TD>
        <TD>278-388</TD>
        <TD>389-499</TD>
        <TD>500-611</TD>
        <TD>612-722</TD>
        <TD>723-833</TD>
        <TD>834-944</TD>
        <TD>&ge; 945</TD></TR>
    </TABLE>
</TD></TR></TABLE>
        <LI><B>strand</B> - Defines the strand - either '+' or '-'.
        <LI><B>thickStart</B> - The starting position at which the feature
        is drawn thickly (for example, the start codon in gene
        displays). When there is no thick part, thickStart and thickEnd are
        usually set to the chromStart position.
        <LI><B>thickEnd</B> - The ending position at which the feature is
        drawn thickly (for example, the stop codon in gene displays).
        <LI><B>itemRgb</B> - An RGB value of the form R,G,B (e.g. 255,0,0).
        If the track line <em>itemRgb</em> attribute is set to &quot;On&quot;,
        this RBG value will determine the display color of the data contained
        in this BED line. NOTE: It is recommended that a simple color scheme
        (eight colors or less) be used with this attribute to avoid 
	overwhelming the color resources of the Genome Browser and your 
	Internet browser.
        <LI><B>blockCount</B> - The number of blocks (exons) in the
        BED line.
        <LI><B>blockSizes</B> - A comma-separated list of the block
        sizes. The number of items in this list should correspond to
        <I>blockCount</I>.
        <LI><B>blockStarts</B> - A comma-separated list of block starts.
        All of the <I>blockStart</I> positions should be calculated relative to
        <I>chromStart</I>. The number of items in
        this list should correspond to <I>blockCount</I>.
</OL>

<P>
In BED files with block definitions, the first <i>blockStart</i> value must
be 0, so that the first block begins at <i>chromStart</i>.  Similarly, the final
<i>blockStart</i> position plus the final <i>blockSize</i> value must equal
<i>chromEnd</i>.  Blocks may not overlap.

<P><B><I>Example:</I></B><BR>
Here's an example of an annotation track that uses a complete BED definition:</P>
<PRE>track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1
chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512
chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</PRE>

<P><B><I>Example:</I></B><BR>
This example shows an annotation track that uses the itemRgb attribute to
individually color each data line. In this track, the color scheme distinguishes
between items named &quot;Pos*&quot; and those named &quot;Neg*&quot;. See the
usage note in the <em>itemRgb</em> description above for color palette
restrictions.
NOTE: The track and data lines in this example have been reformatted for
documentation purposes. This <A HREF="../goldenPath/help/ItemRGBDemo.txt"
TARGET="_blank">example</A> can be pasted into the browser without editing.
<BR>
<PRE>browser position chr7:127471196-127495720
browser hide all
track name="ItemRGBDemo" description="Item RGB demonstration" visibility=2
itemRgb="On"
chr7    127471196  127472363  Pos1  0  +  127471196  127472363  255,0,0
chr7    127472363  127473530  Pos2  0  +  127472363  127473530  255,0,0
chr7    127473530  127474697  Pos3  0  +  127473530  127474697  255,0,0
chr7    127474697  127475864  Pos4  0  +  127474697  127475864  255,0,0
chr7    127475864  127477031  Neg1  0  -  127475864  127477031  0,0,255
chr7    127477031  127478198  Neg2  0  -  127477031  127478198  0,0,255
chr7    127478198  127479365  Neg3  0  -  127478198  127479365  0,0,255
chr7    127479365  127480532  Pos5  0  +  127479365  127480532  255,0,0
chr7    127480532  127481699  Neg4  0  -  127480532  127481699  0,0,255
</PRE>
Click
<a class="insideLink" href="../cgi-bin/hgTracks?org=human&amp;position=chr7&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/ItemRGBDemo.txt"
target="_blank">here</a> to display this track in the Genome Browser.

<P><B><I>Example:</I></B><BR>
It is also possible to color items by strand in a BED track
using the <em>colorByStrand</em> attribute in the
<A HREF="../goldenPath/help/customTrack.html#TRACK">track line</A> as shown
below. For BED tracks, this attribute functions only for custom tracks
with 6 to 8 fields (i.e. BED6 through BED8).
NOTE: The track and data lines in this example have been reformatted for
documentation purposes. This <A HREF="../goldenPath/help/ColorByStrandDemo.txt"
TARGET="_blank">example</A> can be pasted into the browser without editing.
<BR>
<PRE>browser position chr7:127471196-127495720
browser hide all
track name="ColorByStrandDemo" description="Color by strand demonstration"
visibility=2 colorByStrand="255,0,0 0,0,255"
chr7    127471196  127472363  Pos1  0  +
chr7    127472363  127473530  Pos2  0  +
chr7    127473530  127474697  Pos3  0  +
chr7    127474697  127475864  Pos4  0  +
chr7    127475864  127477031  Neg1  0  -
chr7    127477031  127478198  Neg2  0  -
chr7    127478198  127479365  Neg3  0  -
chr7    127479365  127480532  Pos5  0  +
chr7    127480532  127481699  Neg4  0  -
</PRE>
Click
<a class="insideLink" href="../cgi-bin/hgTracks?org=human&amp;position=chr7&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/ColorByStrandDemo.txt"
target="_blank">here</a> to display this track in the Genome Browser.

			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format1.5"></A>
    <!--outer table is for border purposes-->        
    	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
             <TR><TD>            
        	<TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    	     <TR><TD>                
  	    		<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        	     <TR><TD>                
	        	     <!--cheap trick to get background on row to be continuous-->
	   			<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    		     <TR><TD ALIGN="left"><FONT SIZE="4"><b>&nbsp; bigBed format</b></FONT></TD> 
		    		     <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            		     </TD></TR>
  	        		</TABLE>                
	        	     </TD></TR>
  	    		</TABLE>                
   	    		<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        	    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD WIDTH=10>&nbsp;</TD>
		    	    <TD>
	            	    <P>
<P>The bigBed format stores annotation items that can either be simple, or a
linked collection of exons, much as <A HREF="#format1">bed</A> files do.  
BigBed files are created initially from bed type files, 
using the program <TT>bedToBigBed</TT>.   The
resulting bigBed files are in an indexed binary format.  The main advantage of
the bigBed files is that only the portions of the files needed to display a
particular region are transferred to UCSC, so for large data sets bigBed is
considerably faster than regular bed files.  The bigBed file remains on 
your web accessible server (http, https, or ftp), not on the UCSC server. 
</P>
<P>Click <A HREF="../goldenPath/help/bigBed.html">here</A> for more information
on the bigBed format.</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format1.7"></A>
    <!--outer table is for border purposes-->
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
             <TR><TD>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                     <TR><TD>
                        <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                             <TR><TD>
                             <!--cheap trick to get background on row to be continuous-->
                                <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                                     <TR><TD ALIGN="left"><FONT SIZE="4"><b>&nbsp; BED detail format</b></FONT></TD>
                                     <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
                                     </TD></TR>
                                </TABLE>
                             </TD></TR>
                        </TABLE>
                        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                            <TR><TH HEIGHT=10></TH></TR>
                            <TR><TD WIDTH=10>&nbsp;</TD>
                            <TD>
                            <P>
This is an extension of BED format. BED detail uses the first 4 to
12 columns of BED format, plus 2 additional fields that are used to enhance
the track details pages.  The first additional field is an ID, which can be
used in place of the name field for creating links from the details pages.  The
second additional field is a description of the item, which can be a long
description and can consist of html, including tables and lists.
<P><b>Requirements</b> for BED detail custom tracks are: fields must be
tab-separated, &quot;type=bedDetail&quot; must be included in the
<A HREF="../goldenPath/help/customTrack.html#TRACK" target="_blank">track line</A>,
and the name and position fields should uniquely describe items so that the
correct ID and description will be displayed on the details pages.
<P><B><I>Example:</I></B><BR>
This example uses the first 4 columns of BED format, but up to 12 may be used.
Click <a class="insideLink" 
href="../cgi-bin/hgTracks?db=hg19&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/examples/bedDetailExample.txt" 
target="_blank">here</a> to view this track in the Genome Browser.</P>
<PRE style="width:800px;overflow:auto;background-color:#fffffa;color:#111;padding:5px;">
track name=HbVar type=bedDetail description="HbVar custom track" db=hg19 visibility=3 url="http://globin.bx.psu.edu/cgi-bin/hbvar/query_vars3?display_format=page&amp;mode=output&amp;id=$$"
chr11&#09;5246919&#09;5246920&#09;Hb_North_York&#09;2619&#09;Hemoglobin variant
chr11&#09;5255660&#09;5255661&#09;HBD c.1 G>A&#09;2659&#09;delta0 thalassemia
chr11&#09;5247945&#09;5247946&#09;Hb Sheffield&#09;2672&#09;Hemoglobin variant
chr11&#09;5255415&#09;5255416&#09;Hb A2-Lyon&#09;2676&#09;Hemoglobin variant
chr11&#09;5248234&#09;5248235&#09;Hb Aix-les-Bains&#09;2677&#09;Hemoglobin variant
</PRE>
                            </TD><TD WIDTH=15></TD></TR>
                        </TABLE><BR>
                     </TD></TR>
                </TABLE>
             </TD></TR>
        </TABLE>
<BR>

<!--content table-->
<a name="format1.8"></a>
    <!--outer table is for border purposes-->
        <table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>
                <table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
                     <tr><td>
                        <table BGCOLOR="#D9e4f8" width="100%">
                             <tr><td>
                                <table cellpadding=0 cellspacing=0 width="100%">
                                     <tr><td align="left"><font size="4"><b>&nbsp; bedGraph format</b></font></td>
                                     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
                                     </td></tr>
                                </table>
                             </td></tr>
                        </table>
                        <table BGCOLOR="#fffee8" width="100%" cellpadding=0>
                            <tr><th height=10></th></tr>
                            <tr><td width=10>&nbsp;</td>
                            <td>
                            <!--body here-->
                        <P>
                        The bedGraph format allows display of continuous-valued data in track
                        format. This display type is useful for probability scores and transcriptome
                        data. This track type is similar to the
                        <A HREF="#format6">WIG</A> format, but unlike the WIG format, data
                        exported in the bedGraph format are preserved in their original state. This
                        can be seen on export using the table browser. For more information about
                        the bedGraph format, please see the
                        <A HREF="../goldenPath/help/bedgraph.html">bedGraph</A> details page.
                        </P>
                        <P>
                        If you have a very large data set and you would like to keep it on your own
                        server, you should use the
                        <A HREF="#format6.1">bigWig</A> format.
                        </P>
                        </table><br>
                     </td></tr>
                </table>
             </td></tr>
        </table>
        <br>

<!--Content Table-->
<A NAME="format2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; PSL format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>PSL lines represent alignments, and are typically taken from files generated by BLAT
or psLayout. See the
<a
href="../goldenPath/help/hgTracksHelp.html#BLATAlign"
target="_blank">BLAT
documentation</a> for more details. All of the following fields are
required on each data line within a PSL file:
<OL>
<LI><B>matches</B> - Number of bases that match that aren't repeats
<LI><B>misMatches</B> - Number of bases that don't match
<LI><B>repMatches</B> - Number of bases that match but are part of repeats
<LI><B>nCount</B> - Number of 'N' bases
<LI><B>qNumInsert</B> - Number of inserts in query
<LI><B>qBaseInsert</B> - Number of bases inserted in query
<LI><B>tNumInsert</B> - Number of inserts in target
<LI><B>tBaseInsert</B> - Number of bases inserted in target
<LI><B>strand</B> - '+' or '-' for query strand. For translated alignments, second '+'or '-' is for genomic strand
<LI><B>qName</B> - Query sequence name
<LI><B>qSize</B> - Query sequence size
<LI><B>qStart</B> - Alignment start position in query
<LI><B>qEnd</B> - Alignment end position in query
<LI><B>tName</B> - Target sequence name
<LI><B>tSize</B> - Target sequence size
<LI><B>tStart</B> - Alignment start position in target
<LI><B>tEnd</B> - Alignment end position in target
<LI><B>blockCount</B> - Number of blocks in the alignment (a block contains no gaps)
<LI><B>blockSizes</B> - Comma-separated list of sizes of each block
<LI><B>qStarts</B> - Comma-separated list of starting positions of each block in query
<LI><B>tStarts</B> - Comma-separated list of starting positions of each block in target
</OL>
<P><B><I>Example:</I></B><BR>
Here is an example of an annotation track in PSL format. Note that line
breaks have been inserted into the PSL lines in this example for
documentation display purposes. This <A HREF="../goldenPath/help/fishBlats.txt"
TARGET="blank">example</A> can be pasted into the browser without editing.
<PRE>
browser position chr22:13073000-13074000
browser hide all
track name=fishBlats description=&quot;Fish BLAT&quot; visibility=2
useScore=1
59 9 0 0 1 823 1 96 +- FS_CONTIG_48080_1 1955 171 1062 chr22
    47748585 13073589 13073753 2 48,20,  171,1042,  34674832,34674976,
59 7 0 0 1 55 1 55 +- FS_CONTIG_26780_1 2825 2456 2577 chr22
    47748585 13073626 13073747 2 21,45,  2456,2532,  34674838,34674914,
59 7 0 0 1 55 1 55 -+ FS_CONTIG_26780_1 2825 2455 2676 chr22
    47748585 13073727 13073848 2 45,21,  249,349,  13073727,13073827,
</PRE>
Click
<a class="insideLink" href="../cgi-bin/hgTracks?org=human&amp;position=chr7&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/fishBlats.txt"
target="_blank">here</a> to display this track in the Genome Browser.

<P>Be aware that the coordinates for a negative strand in a PSL line are handled in a special way. In the
<I>qStart</I> and <I>qEnd</I> fields, the coordinates indicate the position where the query matches from
the point of view of the forward strand, even when the match is on the reverse strand.
However, in the <I>qStarts</I> list, the coordinates are reversed.</P>
<P>
<B><I>Example:</I></B><BR>
Here is a 61-mer containing 2 blocks that align on the minus strand and
2 blocks that align on the plus strand (this sometimes happens due to assembly errors): 
<BR><BR>
<PRE>
0         1         2         3         4         5         6 tens position in query  
0123456789012345678901234567890123456789012345678901234567890 ones position in query   
                      ++++++++++++++                    +++++ plus strand alignment on query   
    ------------------              --------------------      minus strand alignment on query   
0987654321098765432109876543210987654321098765432109876543210 ones position in query negative strand coordinates
6         5         4         3         2         1         0 tens position in query negative strand coordinates

Plus strand:   
     qStart=22
     qEnd=61 
     blockSizes=14,5 
     qStarts=22,56 
                  
Minus strand:   
     qStart=4 
     qEnd=56 
     blockSizes=20,18 
     qStarts=5,39   
</PRE>
Essentially, the minus strand <I>blockSizes</I> and <I>qStarts</I> are
what you would get if you reverse-complemented the query.
However, the <I>qStart</I> and <I>qEnd</I> are not reversed. Use the following formulas to convert one to the other:
<PRE>
     Negative-strand-coordinate-qStart = qSize - qEnd   = 61 - 56 =  5
     Negative-strand-coordinate-qEnd   = qSize - qStart = 61 -  4 = 57
</PRE>
<br>
BLAT this actual sequence against hg19 for a real-world example:<br>
<br>


CCCC<br>
GGGTAAAATGAGTTTTTT<br>
GGTCCAATCTTTTA<br>
ATCCACTCCCTACCCTCCTA<br>
GCAAG<br>



<br>
Look for the alignment on the negative strand (-) of chr21, which
conveniently aligns to the window chr21:10,000,001-10,000,061.<br>

<br>
Browser window coordinates are 1-based [start,end] while psl coordinates are 0-based [start,end), so a start of 10,000,001
in the browser corresponds to a start of 10,000,000 in the psl. Subtracting 10,000,000 from the target (chromosome) position
in psl gives the query negative strand coordinate above.<br>

<br>
The 4, 14, and 5 bases at beginning, middle, and end were
chosen to not match with the genome at the corresponding position.<br>


			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GFF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<p>
GFF (General Feature Format) lines are based on the Sanger
<a href="http://www.sanger.ac.uk/resources/software/gff/spec.html" target="_blank">
GFF2 specification</a>. GFF lines have nine required fields that <I>must</I> be
tab-separated. If the fields are separated by spaces instead of tabs, the track will not
display correctly. For more information on GFF format, refer to Sanger's
<a href="http://www.sanger.ac.uk/resources/software/gff/" target="_blank">
GFF page</a>.
</p>

<p>
Note that there is also a GFF3 specification that is not currently supported by the Browser.
All GFF tracks must be formatted according to Sanger's GFF2 specification.
</p>

<p>
If you would like to obtain browser data in GFF (GTF) format, please refer to
<A HREF="http://genomewiki.ucsc.edu/index.php/Genes_in_gtf_or_gff_format" TARGET="_blank">
Genes in gtf or gff format</A> on the Wiki.</P>
<P>
Here is a brief description of the GFF fields:</P>

<OL>
        <LI><B>seqname</B> - The name of the sequence.  Must be a chromosome or scaffold.
        <LI><B>source</B> - The program that generated this feature.
        <LI><B>feature</B> - The name of this type of feature. Some examples of
        standard feature types are "CDS", "start_codon", "stop_codon", and
        "exon".
        <LI><B>start</B> - The starting position of the feature in the sequence.  The first base is numbered 1.
        <LI><B>end</B> - The ending position of the feature (inclusive).
        <LI><B>score</B> - A score between 0 and 1000. If the track line
        <I>useScore</I> attribute is set to 1 for this annotation data set, the
        <I>score</I> value will determine the level of gray in which
        this feature is displayed (higher numbers = darker gray). If there is no
        score value, enter ".".        <LI><B>strand</B> - Valid entries include '+', '-', or '.' (for don't know/don't care).        <LI><B>frame</B> - If the feature is a coding exon, <I>frame</I> should be a number between 0-2 that represents the reading frame of the
        first base.  If the feature is not a coding exon, the value should be '.'.        <LI><B>group</B> - All lines with the same group are linked together into a single item.
</OL>
<P><B><I>Example:</I></B><BR>
Here's an example of a GFF-based track.
This <A HREF="../goldenPath/help/regulatory.txt" TARGET="blank">example</A>
can be pasted into the browser without editing. NOTE: Paste operations on some 
operating systems will replace tabs with spaces, which will result in an error 
when the GFF track is uploaded. You can circumvent this problem by pasting the 
URL of the above example
(http://genome.ucsc.edu/goldenPath/help/regulatory.txt) instead of the text 
itself into the custom annotation track text box. If you encounter an error
when loading a GFF track, check that the data lines contain tabs rather than
spaces.

<PRE><TT>
browser position chr22:10000000-10025000
browser hide all
track name=regulatory description=&quot;TeleGene(tm) Regulatory Regions&quot;
visibility=2
chr22  TeleGene enhancer  10000000  10001000  500 +  .  touch1
chr22  TeleGene promoter  10010000  10010100  900 +  .  touch1
chr22  TeleGene promoter  10020000  10025000  800 -  .  touch2
</TT></PRE>
Click
<a class="insideLink" href="../cgi-bin/hgTracks?org=human&amp;position=chr22&amp;hgt.customText=http://genome.ucsc.edu/goldenPath/help/regulatory.txt"
TARGET="_blank">here</a> to display this track in the Genome Browser.

			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GTF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<!--#include virtual="../goldenPath/help/GTF.html" -->
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>


<!--Content Table-->
<A NAME="format20"></A>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
            <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD>
                <!--cheap trick to get background on row to be continuous-->
                <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                    <TR><TD ALIGN="left">
                    <FONT SIZE="4"><b>&nbsp; HAL format</b></FONT></TD>
                    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR>
                </TABLE>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
                    <TD>
                    	<P>
			HAL is a graph-based structure to efficiently store and index multiple genome alignments and ancestral 
			reconstructions. HAL files are represented in <A HREF="http://www.hdfgroup.org/HDF5/" TARGET="_BLANK">HDF5 format</A>, 
			an open standard for storing and indexing large, compressed scientific data sets. Genomes within HAL are organized according 
			to the phylogenetic tree that relate them: each genome is segmented into pairwise DNA alignment blocks 
			with respect to its parent and children (if present) in the tree. Note that if the phylogeny is unknown, 
			a star tree can be used. The modularity provided by this tree-based decomposition allows for efficient querying 
			of sub-alignments, as well as the ability to add, remove and update genomes within the alignment 
			with only local modifications to the structure. Another important feature of HAL is reference 
			independence: alignments in this format can be queried with respect to the coordinates of any genome they contain.
			</P>
			<P>
			HAL files can be created or read with a comprehensive C++ API (click <A HREF="https://github.com/glennhickey/hal" TARGET="_BLANK">here</A> 
			for source code and manual). A set of command line tools is included to perform basic operations, such as importing and exporting data, 
			identifying mutations, coordinate mapping (liftOver), and comparative assembly hub generation.
			</P>
			<P>
			HAL is the native output format of the Progressive Cactus alignment pipeline, and is 
			included in the <A HREF="https://github.com/glennhickey/progressiveCactus" TARGET="_BLANK">Progressive Cactus</A> installation package.
			</P>
                            </TD><TD WIDTH=15></TD></TR>
                        </TABLE><BR>
                     </TD></TR>
                </TABLE>
             </TD></TR>
        </TABLE>
        <BR>



<!--Content Table-->
<A NAME="format5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; MAF format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
                    The multiple alignment format stores a series of
                    multiple alignments in a format that is easy to
                    parse and relatively easy to read.  This format
                    stores multiple alignments at the DNA level
                    between entire genomes. Previously used formats 
		    are suitable for multiple alignments of single proteins or 
		    regions of DNA without rearrangements, but would require 
		    considerable extension to cope with genomic issues such as
                    forward and reverse strand directions, multiple pieces 
		    to the alignment, and so forth.
                    <P>
		    <B>General Structure</B>
		    <P> The <em>.maf</em> format is
                    line-oriented. Each multiple alignment ends with a
                    blank line.  Each sequence in an alignment is on a single 
		    line, which can get quite long, but there is no length limit.
		    Words in a line are delimited by any white space. Lines
                    starting with # are considered to be comments.
                    Lines starting with ## can be ignored by most
                    programs, but contain meta-data of one form or
                    another.
                    <P>
                    The file is divided into paragraphs that terminate
                    in a blank line.  Within a paragraph, the first
                    word of a line indicates its type.  Each multiple
                    alignment is in a separate paragraph that begins
                    with an &quot;a&quot; line and contains an &quot;s&quot; line 
		    for each sequence in the multiple alignment.  
		    Some MAF files may contain other  optional line types: 
		    <UL>
		    <LI>
		    an "i" line containing information about what is in the 
		    aligned species DNA before and after the immediately 
		    preceding "s" line
		    <LI>
		    an "e" line containing information about the size of the 
		    gap between the alignments that span the current 
		    block 
		    <LI>
		    a "q" line indicating the quality of each aligned base
		    for the species
		    </UL>
		    <P>
		    Parsers may ignore any other types of paragraphs and 
		    other types of lines within an alignment paragraph. 
                    <P>
		    <B>Custom Tracks</B>
		    <P>
                    The first line of a custom MAF track must be a "track" line
		    that contains a name=value pair specifying the track name.
		    Here is an example of a minimal track line:
		    
		    <PRE style="color:#0000ff;"> track name=sample</PRE>
                    The following variables can be specified in the track line
		    of a custom MAF:
                    <UL>
                    <LI><B>name=sample</B> - Required. Name the track </LI>
                    <LI><B>description="Sample Track"</B> - Optional. 
		    Gives a long name for the track</LI>
                    <LI><B>frames=multiz28wayFrames</B> - Optional. 
		    Tells the browser which table to 
		    grab the gene frames from.  This is usually associated with
		    an N-way alignment where the name ends in the string 
		    "Frames" </LI>
                    <LI><B>mafDot=on</B> - Optional. 
		    Use dots instead of bases when bases are identical</LI>
                    <LI><B>visibility=dense|pack|full</B> - Optional. 
		    Sets the default visibility mode for this track.</LI>
                    <LI><B>speciesOrder="hg18 panTro2"</B> - Optional. 
		    White-space separated list specifying the order in which
		    the sequences in the maf should be displayed.
		    </LI>
		    </UL>
		    The second line of a custom MAF track must be a header
	            line as described below.
		    <P>
		    <B>Header Line</B>
		    <P>
                    The first line of a <em>.maf</em> file begins with ##maf. 
		    This word is followed by white-space-separated
                    variable=value pairs.  There should be <em>no</em> white
                    space surrounding the &quot;=&quot;.  
		    <PRE style="color:#0000ff;"> ##maf version=1 scoring=tba.v8 </PRE>
		    The currently defined variables are:
                    <UL>
                    <LI><B>version</B> - Required.  Currently set to one.  </LI>
                    <LI><B>scoring</B> - Optional. A name for the scoring
                                   scheme used for the alignments.
                                   The current scoring schemes are:
                        <UL>
			<LI><em>bit</em> - roughly corresponds to blast bit
                                   values (roughly 2 points per
                                   aligning base minus penalties for
                                   mismatches and inserts).  </LI>
                        <LI><em>blastz</em> - blastz scoring scheme -- roughly
                                   100 points per aligning base.</LI>
                        <LI><em>probability</em> - some score normalized
                                   between 0 and 1.  </LI>
                        </UL>
                    <LI><B>program</B> - Optional. Name of the program
                                   generating the alignment.</LI>
                    </UL>
                    Undefined variables are ignored by the parser.
		    <P>
                    The header line is usually followed by a comment line (it begins with a #) 
                    that describes the parameters that were used to run the alignment program.
		    <PRE style="color:#0000ff;"> # tba.v8 (((human chimp) baboon) (mouse rat))</PRE>
                    <P>
		    <B>Alignment Block Lines</B> (lines starting with 'a' -- parameters for a new alignment block)
		    <P>                    
		    <PRE style="color:#0000ff;"> a score=23262.0</PRE>
                    Each alignment begins with an 'a' line that set 
		    variables for the entire alignment block.  The 'a'
                    is followed by name=value pairs.  There are no
                    required name=value pairs.  The currently defined
                    variables are: 
		    <UL>
                    <LI> score -- Optional. Floating point score. If
                                 this is present, it is good practice
                                 to also define scoring in the first line.  </LI>

                    <LI> pass --  Optional.  Positive integer value.
                                 For programs that do multiple pass
                                 alignments such as blastz, this shows
                                 which pass this alignment came from.
                                 Typically, pass 1 will find the
                                 strongest alignments genome-wide, and
                                 pass 2 will find weaker alignments
                                 between two first-pass alignments.</LI>
                    </UL>
                    <P>
		    <B>Lines starting with 's' -- a sequence within an alignment block</B>
		    <P>
<PRE style="color:#0000ff;"> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA
</PRE>
                    The 's' lines together with the 'a' lines define a
                    multiple alignment.  The 's' lines have the
                    following fields which are defined by position
                    rather than name=value pairs.
		    <UL>
                    <LI> src -- The name of one of the source sequences
                               	for the alignment.  For sequences that
                               	are resident in a browser assembly, the
			 	form 'database.chromosome' allows
                               	automatic creation of links to other 
				assemblies.  Non-browser sequences are
                               	typically reference by the species name
                               	alone.</LI>
                    <LI> start -- The start of the aligning region in the 
				source sequence. This is a
                               	zero-based number. If the strand field 
				is '-' then this is the start
                               	relative to the reverse-complemented
                                source sequence (see
                                <A HREF="http://genomewiki.ucsc.edu/index.php/Coordinate_Transforms"
                                TARGET=BLANK>Coordinate Transforms</A>). </LI>
                    <LI> size -- The size of the aligning region in the
                               	source sequence.  This number is equal
                                to the number of non-dash characters in
                               	the alignment text field below.  </LI>          
		    <LI> strand -- Either '+' or '-'.  If '-', then the           
                               	alignment is to the
                               	reverse-complemented source.  </LI>
                    <LI> srcSize -- The size of the entire source sequence, 
				not just the parts involved in the alignment.</LI>
		    <LI> text -- The nucleotides (or amino acids) in
				the alignment and any insertions
                               (dashes) as well.  </LI> 
                    </UL>          
                                   
  		    <P>
		    <B>Lines starting with 'i' -- information about what's 
		    happening before and after this block in the aligning 
		    species</B> 
                    <P>
<PRE style="color:#0000ff;"> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
 i panTro1.chr6 N 0 C 0
 s baboon         249182 13 +   4622798 gcagctgaaaaca
 i baboon       I 234 n 19</PRE>
		    <P>
		    The 'i' lines contain information about the context of the 
		    sequence lines immediately preceeding them. The following 
		    fields are defined by position rather than name=value pairs:
                    <UL>
                    <LI>src -- The name of the source sequence for the 
		    alignment. Should be the same as the 's' line immediately 
		    above this line.             
		    <LI>leftStatus -- A character that specifies the 
		    relationship between the sequence in this block and the 
		    sequence that appears in the previous block.
		    <LI>leftCount -- Usually the number of bases in the aligning
                    species between the start of this alignment and the end of 
		    the previous one.                          
		    <LI>rightStatus -- A character that specifies the 
		    relationship between the sequence in this block and the 
		    sequence that appears in the subsequent block.
	            <LI>rightCount -- Usually the number of bases in the 
		    aligning species between the end of this alignment and the 
		    start of the next one. 
		    </UL>
		    <P>
		    The status characters can be one of the following values:
                    <UL>                          
		    <LI>C -- the sequence before or after is contiguous with 
		    this block.
		    <LI>I -- there are bases between the bases in this block 
		    and the one before or after it.                          
		    <LI>N -- this is the first sequence from this src chrom or 
		    scaffold.
		    <LI>n -- this is the first sequence from this src chrom or 
		    scaffold but it is bridged by another alignment from a 
		    different chrom or scaffold.                          
		    <LI>M -- there is missing data before or after this block 
		    (Ns in the sequence).
		    <LI>T -- the sequence in this block has been used before
		    in a previous block (likely a tandem duplication)
		    </UL>                        
		    <P>
		    <B>Lines starting with 'e' -- information about empty parts 
		    of the alignment block</B>                        
		    <P>
<PRE style="color:#0000ff;"> s hg16.chr7    27707221 13 + 158545518 gcagctgaaaaca
 e mm4.chr6     53310102 13 + 151104725 I</PRE>
		    <P>
		    The 'e' lines indicate that there isn't aligning DNA for a 
		    species but that the current block is bridged by a chain 
		    that connects blocks before and after this block. The 
		    following fields are defined by position rather than 
		    name=value pairs.
                    <UL>                          
		    <LI>src -- The name of one of the source sequences for the 
		    alignment.                          
		    <LI>start -- The start of the non-aligning region in the
                    source sequence. This is a zero-based number. If the
		    strand field is '-' then this is the start relative to 
                    the reverse-complemented source sequence (see
                    <A HREF="http://genomewiki.ucsc.edu/index.php/Coordinate_Transforms"
                    TARGET=BLANK>Coordinate Transforms</A>). </LI>
		    <LI>size -- The size in base pairs of the non-aligning 
		    region in the source sequence.                          
		    <LI>strand -- Either '+' or '-'. If '-', then the
                    alignment is to the reverse-complemented source.
                    <LI>srcSize -- The size of the entire source sequence, 
                    not just the parts involved in the alignment.
                    alignment and any insertions (dashes) as well.
		    <LI>status -- A character that specifies the 
		    relationship between the non-aligning sequence in this 
		    block and the sequence that appears in the previous and 
		    subsequent blocks.
		    </UL>
		    <P>
		    The status character can be one of the following values:
                    <UL>                          
		    <LI>C -- the sequence before and after is contiguous 
		    implying that this region was either deleted in the source 
		    or inserted in the reference sequence. The browser draws a 
		    single line or a '-' in base mode in these blocks.
		    <LI>I -- there are non-aligning bases in the source species 
		    between chained alignment blocks before and after this 
		    block. The browser shows a double line or '=' in base mode.
		    <LI>M -- there are non-aligning bases in the source and 
		    more than 90% of them are Ns in the source. The browser 
		    shows a pale yellow bar.                         
		    <LI>n -- there are non-aligning bases in the source and the 
		    next aligning block starts in a new chromosome or scaffold 
		    that is bridged by a chain between still other blocks. The 
		    browser shows either a single line or a double line based 
		    on how many bases are in the gap between the bridging 
		    alignments.
		    </UL>
		    <B>Lines starting with 'q' -- information about the quality
		    of each aligned base for the species</B> 
                    <P>
<PRE style="color:#0000ff;"> s hg18.chr1                  32741 26 + 247249719 TTTTTGAAAAACAAACAACAAGTTGG
 s panTro2.chrUn            9697231 26 +  58616431 TTTTTGAAAAACAAACAACAAGTTGG
 q panTro2.chrUn                                   99999999999999999999999999
 s dasNov1.scaffold_179265     1474  7 +      4584 TT----------AAGCA---------
 q dasNov1.scaffold_179265                         99----------32239--------- </PRE>
		    <P>
		    The 'q' lines contain a compressed version of the actual
		    raw quality data, representing the quality of each 
		    aligned base for the species with a single character of 0-9 
		    or F. The following fields are defined by position rather 
		    than name=value pairs:
		    <UL>
                    <LI>src -- The name of the source sequence for the 
		    alignment. Should be the same as the 's' line immediately 
		    preceding this line.             
		    <LI>value -- A MAF quality value corresponding to 
		    the aligning nucleotide acid in the preceding 
		    's' line.  Insertions (dashes) in the preceding 's' line
	       	    are represented by dashes in the 'q' line as well. 
		    The quality value can be 'F' (finished
		    sequence) or a number derived from the actual quality
		    scores (which range from 0-97) or the manually assigned
		    score of 98. These numeric values are calculated as: 
<BLOCKQUOTE><TT>MAF quality value = min( floor(actual quality value/5), 9 )</TT></BLOCKQUOTE>
		    This results in the following mapping: 
		    <P>
            	    <TABLE BORDER=1> 
		    <TR>
                    <TH nowrap><FONT color="#006666"><B>MAF quality value</B></FONT></TH>
		    <TH nowrap><FONT color="#006666"><B>Raw quality score range</B></FONT></TH>
                    <TH nowrap><FONT color="#006666"><B>Quality level</B></FONT></TH>
                    </TR>
		    <TR>
                    <TD ALIGN="center">0-8</TD>
		    <TD ALIGN="center">0-44</TD>
                    <TD ALIGN="center">Low</TD>
                    </TR>
		    <TR>
                    <TD ALIGN="center">9</TD>
		    <TD ALIGN="center">45-97</TD>
                    <TD ALIGN="center">High</TD>
                    </TR>
		    <TR>
                    <TD ALIGN="center">0</TD>
		    <TD ALIGN="center">98</TD>
                    <TD ALIGN="center">Manually assigned</TD>
                    </TR>
		    <TR>
                    <TD ALIGN="center">F</TD>
		    <TD ALIGN="center">99</TD>
                    <TD ALIGN="center">Finished</TD>
                    </TR>
		    </TABLE>
		    </UL><BR>
                    <P>
		    <B>A Simple Example</B>
		    <P>                     
		    Here is a simple example of a three alignment
                    blocks derived from five starting sequences.
		    The first <B>track</B> line is necessary for custom
		    tracks, but should be removed otherwise.
		    Repeats are shown as lowercase, and each block may
                    have a subset of the input sequences.  All
		    sequence columns and rows must contain at least one nucleotide
                    (no columns or rows that contain only insertions).

<PRE style="color:#0000ff;"> 
track name=euArc visibility=pack
##maf version=1 scoring=tba.v8 
# tba.v8 (((human chimp) baboon) (mouse rat)) 
                   
a score=23262.0     
s hg18.chr7    27578828 38 + 158545518 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
s panTro1.chr6 28741140 38 + 161576975 AAA-GGGAATGTTAACCAAATGA---ATTGTCTCTTACGGTG
s baboon         116834 38 +   4622798 AAA-GGGAATGTTAACCAAATGA---GTTGTCTCTTATGGTG
s mm4.chr6     53215344 38 + 151104725 -AATGGGAATGTTAAGCAAACGA---ATTGTCTCTCAGTGTG
s rn3.chr4     81344243 40 + 187371129 -AA-GGGGATGCTAAGCCAATGAGTTGTTGTCTCTCAATGTG
                   
a score=5062.0                    
s hg18.chr7    27699739 6 + 158545518 TAAAGA
s panTro1.chr6 28862317 6 + 161576975 TAAAGA
s baboon         241163 6 +   4622798 TAAAGA 
s mm4.chr6     53303881 6 + 151104725 TAAAGA
s rn3.chr4     81444246 6 + 187371129 taagga

a score=6636.0
s hg18.chr7    27707221 13 + 158545518 gcagctgaaaaca
s panTro1.chr6 28869787 13 + 161576975 gcagctgaaaaca
s baboon         249182 13 +   4622798 gcagctgaaaaca
s mm4.chr6     53310102 13 + 151104725 ACAGCTGAAAATA
&nbsp
</PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format5.1"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; BAM format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
BAM is the compressed binary version of the 
<A HREF="http://samtools.sourceforge.net/" 
target="_blank">Sequence Alignment/Map (SAM)</A> format,
a compact and index-able representation of nucleotide sequence alignments.  
Many 
<A HREF="http://samtools.sourceforge.net/swlist.shtml" 
target="_blank">next-generation sequencing and analysis tools</A> 
work with SAM/BAM.
For custom track display, the main advantage of indexed BAM over PSL
and other human-readable alignment formats is that only the portions
of the files needed to display a particular region are transferred to
UCSC.  This makes it possible to display alignments from files that
are so large that the connection to UCSC would time out when
attempting to upload the whole file to UCSC.
Both the BAM file and its associated index file remain on your
web-accessible server (http or ftp), not on the UCSC server.
UCSC temporarily caches the accessed portions of the files to speed up
interactive display.
</P><P>
Click <A HREF="../goldenPath/help/bam.html">here</A> for more information
about BAM custom tracks.</P>
<P></P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; WIG format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
Wiggle format (WIG) allows the display of continuous-valued data in a track
format. Click <A HREF="../goldenPath/help/wiggle.html">here</A> for more 
information.</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format6.1"></A>
    <!--outer table is for border purposes-->        
    	<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
             <TR><TD>            
        	<TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    	     <TR><TD>                
  	    		<TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        	     <TR><TD>                
	        	     <!--cheap trick to get background on row to be continuous-->
	   			<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    		     <TR><TD ALIGN="left"><FONT SIZE="4"><b>&nbsp; bigWig format</b></FONT></TD> 
		    		     <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="/images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            		     </TD></TR>
  	        		</TABLE>                
	        	     </TD></TR>
  	    		</TABLE>                
   	    		<TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        	    <TR><TH HEIGHT=10></TH></TR>
                	    <TR><TD WIDTH=10>&nbsp;</TD>
		    	    <TD>
	            	    <P>
The bigWig format is for display of dense, continuous data that will be
displayed in the Genome Browser as a graph.  BigWig files are
created initially from <A HREF="#format6">wiggle</A> (wig) type  
files, using the program <TT>wigToBigWig</TT>.  Alternatively, bigWig
files can be created from <A HREF="../goldenPath/help/bedgraph.html">bedGraph</A> 
files, using the program <TT>bedGraphToBigWig</TT>.  
In either case, the resulting bigWig files are 
in an indexed binary format.  The main advantage of the bigWig files 
is that only the portions of the files needed to display a particular 
region are transferred to UCSC, so for large data sets bigWig is 
considerably faster than regular wiggle files.
The bigWig file remains on your web accessible server (http, 
https, or ftp), not on the UCSC server. Only the portion that is needed
for the chromosomal position you are currently viewing is locally cached as a 
&quot;sparse file&quot;.</P>
<P>Click <A HREF="../goldenPath/help/bigWig.html">here</A> for more information
on the bigWig format.</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format6.5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Microarray format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
The datasets for the built-in microarray tracks in the Genome Browser are stored
in BED15 format, an extension of <A HREF="#format1">BED</A> format that includes
three additional fields: expCount, expIds, and expScores. 
To display correctly in the Genome Browser, microarray tracks 
require the setting of several attributes in the trackDb file associated with 
the track's genome assembly. Each microarray track set must also have an 
associated microarrayGroups.ra configuration file that contains additional 
information about the data in each of the arrays.</P>
<P>
User-created microarray custom tracks are similar in format to BED 
custom tracks with the addition of three required track line parameters in the 
header--expNames, expScale, and expStep--that mimic the trackDb and 
microarrayGroups.ra settings of built-in microarray tracks. </P>
<P>
For a complete description of the microarray track format and an explanation
of how to construct a microarray custom track, see the 
<A HREF="http://genomewiki.ucsc.edu/index.php/Microarray_track"
target="_blank">Genome Browser Wiki</A>.
</P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format7"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; .2bit format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
A .2bit file stores multiple DNA sequences (up to 4 Gb total) in a compact 
randomly-accessible format. The file contains masking information as well 
as the DNA itself.</P>
<P>
The file begins with a 16-byte header containing the 
following fields:
<UL>
<LI>signature - the number 0x1A412743 in the architecture of the machine that 
created the file
<LI>version - zero for now. Readers should abort if they see a version number 
higher than 0.
<LI>sequenceCount - the number of sequences in the file.
<LI>reserved - always zero for now
</UL>
<P>
All fields are 32 bits unless noted. If the signature value is not as given, 
the reader program should byte-swap the signature and check if the swapped 
version matches. If so, all multiple-byte entities in the file will have to be 
byte-swapped. This enables these binary files to be used unchanged on 
different architectures.</P>
<P>
The header is followed by a file index, which contains one entry for 
each sequence. Each index entry contains three fields:
<UL>
<LI>nameSize - a byte containing the length of the name field
<LI>name - the sequence name itself, of variable length depending on nameSize
<LI>offset - the 32-bit offset of the sequence data relative to the start of the
file
</UL>
<P>
The index is followed by the sequence records, which contain nine fields:
<UL>
<LI>dnaSize - number of bases of DNA in the sequence
<LI>nBlockCount - the number of blocks of Ns in the file (representing unknown 
sequence)
<LI>nBlockStarts - an array of length nBlockCount of 32 bit integers
indicating the starting position of a block of Ns 
<LI>nBlockSizes - an array of length nBlockCount of 32 bit integers
indicating the length of a block of Ns
<LI>maskBlockCount - the number of masked (lower-case) blocks
<LI>maskBlockStarts - an array of length maskBlockCount of 32 bit integers
indicating the starting position of a masked block
<LI>maskBlockSizes - an array of length maskBlockCount of 32 bit integers
indicating the length of a masked block
<LI>reserved - always zero for now
<LI>packedDna - the DNA packed to two bits per base, represented as so: 
T - 00, C - 01, A - 10, G - 11.  The first base is in the most significant 
2-bit byte; the last base is in the least significant 2 bits. For example, the
sequence TCAG is represented as 00011011. 
</UL>
<P>For a complete definition of all fields in the twoBit format, see 
<A HREF="http://genome-source.cse.ucsc.edu/gitweb/?p=kent.git;a=blob_plain;f=src/inc/twoBit.h;hb=HEAD">this</A> 
description in the source code.
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format8"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; .nib format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
<P>
The .nib format pre-dates the .2bit format and is less compact. It 
describes a DNA sequence by packing two bases into each byte. 
Each .nib file contains only a single sequence.  The file begins with a 32-bit 
signature that is 0x6BE93D3A in the architecture of the machine that created 
the file (or possibly a byte-swapped version of the same number on another 
machine). This is followed by a 32-bit number in the same format that describes 
the number of bases in the file. Next, the bases themselves are listed, packed 
two bases to the byte. The first base is packed in the high-order 4 bits 
(nibble); the second base is packed in the low-order four bits:</P>

<BLOCKQUOTE><TT>
byte = (base1<<4) + base2
</TT></BLOCKQUOTE>

<P>
The numerical representations for the bases are:</P>
<UL>
<LI>0 - T
<LI>1 - C
<LI>2 - A
<LI>3 - G
<LI>4 - N (unknown)
</UL>
<P>
The most significant bit in a nibble is set if the base is masked. </P>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format9"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; GenePred table format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
genePred is a table format commonly used for gene prediction tracks in the
Genome Browser. Variations of the genePred format are listed below.</P>
<P>
If you would like to obtain browser data in GFF (GTF) format, please refer to
<A HREF="http://genomewiki.ucsc.edu/index.php/Genes_in_gtf_or_gff_format" TARGET="_blank">
Genes in gtf or gff format</A> on the Wiki.</P>

<H4><A name=GenePredictions></A>Gene Predictions</H4>
The following definition is used for gene prediction tables. In 
alternative-splicing situations, each transcript has a row in this table.
<PRE>table genePred
"A gene prediction."
    (
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>

<H4><A name=GenePredExt></A>Gene Predictions (Extended)</H4>
The following definition is used for extended gene prediction tables. In 
alternative-splicing situations, each transcript has a row in this table.
The refGene table is an example of the genePredExt format.
<PRE>table genePredExt
"A gene prediction with some additional info."
    (
    string name;        	"Name of gene (usually transcript_id from GTF)"
    string chrom;       	"Chromosome name"
    char[1] strand;     	"+ or - for strand"
    uint txStart;       	"Transcription start position"
    uint txEnd;         	"Transcription end position"
    uint cdsStart;      	"Coding region start"
    uint cdsEnd;        	"Coding region end"
    uint exonCount;     	"Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    uint id;            	"Unique identifier"
    string name2;       	"Alternate name (e.g. gene_id from GTF)"
    string cdsStartStat; 	"enum('none','unk','incmpl','cmpl')"
    string cdsEndStat;   	"enum('none','unk','incmpl','cmpl')"
    lstring exonFrames; 	"Exon frame offsets {0,1,2}"
    )
</PRE>

<H4><A name=RefFlat></A>Gene Predictions and RefSeq Genes with Gene Names</H4>
A version of genePred that associates the gene name with the gene prediction 
information. In alternative-splicing situations, each transcript has a row in
this table.
<PRE>table refFlat
"A gene prediction with additional geneName field."
    (
    string  geneName;           "Name of gene as it appears in Genome Browser."
    string  name;               "Name of gene"
    string  chrom;              "Chromosome name"
    char[1] strand;             "+ or - for strand"
    uint    txStart;            "Transcription start position"
    uint    txEnd;              "Transcription end position"
    uint    cdsStart;           "Coding region start"
    uint    cdsEnd;             "Coding region end"
    uint    exonCount;          "Number of exons"
    uint[exonCount] exonStarts; "Exon start positions"
    uint[exonCount] exonEnds;   "Exon end positions"
    )
</PRE>
			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format9.1"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; bigGenePred table format</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
bigGenePred is a table format commonly used for gene prediction tracks in the
Genome Browser. bigGenePred format is a superset of the 
<A HREF="FAQformat.html#format9">genePred</A> text-based format supported using 
the <A HREF="FAQformat.html#format1.5">bigBed</A> format, so it can be efficiently 
accessed over a network. 
<P>Click <A HREF="../goldenPath/help/bigGenePred.html">here</A> for more information
on the bigGenePred format.</P>
<P>


			    </TD><TD WIDTH=15></TD></TR>
	    		</TABLE><BR>
		     </TD></TR>
		</TABLE>
     	     </TD></TR>
    	</TABLE>
	<BR>

<!--Content Table-->
<A NAME="format10"></A>
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
            <TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD>
                <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                    <TR><TD ALIGN="left">
                    <FONT SIZE="4"><b>&nbsp; Personal Genome SNP format</b></FONT></TD>
                    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR>
                </TABLE>
                </TD></TR>
            </TABLE>
            <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
                    <TD>
<P>
This format is for displaying SNPs from personal genomes.  It is the same as
is used for the Genome Variants and Population Variants tracks.</P>

<OL>
    <LI><B>chrom</B> - The name of the chromosome (e.g. chr3, chrY, chr2_random)
     or scaffold (e.g. scaffold10671). 
    <LI><B>chromStart</B> - The starting position of the feature in the
     chromosome or scaffold. The first base in a chromosome is numbered 0.
    <LI><B>chromEnd</B> - The ending position of the feature in the
     chromosome or scaffold. The <I>chromEnd</I> base is not included in the
     display of the feature. For example, the first 100 bases of a
     chromosome are defined as <I>chromStart=0, chromEnd=100</I>, and span
     the bases numbered 0-99.
    <LI><B>name</B> - The allele or alleles, consisting of one or more A, C, T,
     or G, optionally followed by one or more '/' and another allele (there can
     be more than 2 alleles). A '-'
     can be used in place of a base to denote an insertion or deletion; if the
     position given is zero bases wide, it is an insertion. The alleles are
     expected to be for the plus strand.
    <LI><B>alleleCount</B> - The number of alleles listed in the name field.
    <LI><B>alleleFreq</B> - A comma-separated list of the frequency of each
     allele, given in the same order as the name field.  If unknown, a list of
     zeroes (matching the alleleCount) should be used.
    <LI><B>alleleScores</B> - A comma-separated list of the quality score of
     each allele, given in the same order as the name field.  If unknown, a list
     of zeroes (matching the alleleCount) should be used.
</OL>
<P>
In the Genome Browser, when viewing the forward strand of the reference genome
(the normal case), the displayed alleles are relative to the forward strand.
When viewing the reverse strand of the reference genome (via the &quot;<--&quot;
or &quot;reverse&quot; button), the displayed alleles are reverse-complemented
to match the reverse strand. If the allele frequencies are given, the coloring
of the box will reflect the frequency for each allele.</P>
<P>
The details pages for this track type will automatically compute amino acid
changes for coding SNPs as well as give a chart of amino acid properties if
there is a non-synonymous change. (The Sift and PolyPhen predictions that are
in some of the Genome Variants subtracks are not available.)</P>
<P><B><I>Example:</I></B><BR>
Here is an example of an annotation track in Personal Genome SNP format. The
first SNP using a '-' is an insertion; the second is a deletion. The last 4
SNPs are in a coding region.</P>
<PRE>track type=pgSnp visibility=3 db=hg19 name="pgSnp" description="Personal Genome SNP example"
browser position chr21:31811924-31812937
chr21	31812007	31812008	T/G	2	21,70	90,70
chr21	31812031	31812032	T/G/A	3	9,60,7	80,80,30
chr21	31812035	31812035	-/CGG	2	20,80	0,0
chr21	31812088	31812093	-/CTCGG	2	30,70	0,0
chr21	31812277	31812278	T	1	15	90
chr21	31812771	31812772	A	1	36	80
chr21	31812827	31812828	A/T	2	15,5	0,0
chr21	31812879	31812880	C	1	0	0
chr21   31812915	31812916	-	1	0	0
</PRE>
                            </TD><TD WIDTH=15></TD></TR>
                        </TABLE><BR>
                     </TD></TR>
                </TABLE>
             </TD></TR>
        </TABLE>
        <BR>
    </TD></TR>
</TABLE>

<!--Content Table-->
<A NAME="format10.1"></A>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>
	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>
	        <!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
		    <FONT SIZE="4"><b>&nbsp; VCF format</b></FONT></TD>
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
	        </TABLE>
	        </TD></TR>
	    </TABLE>
	    <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
<P>
<A HREF="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41"
target="_blank">Variant Call Format (VCF)</A> is a flexible and extendable
format for variation data such as single nucleotide variants,
insertions/deletions, copy number variants and structural variants.
When a VCF file is compressed and indexed using
<A HREF="http://samtools.sourceforge.net/tabix.shtml"
target="_blank">tabix</A>, and made web-accesible,
the Genome Browser can fetch only the portions of the file necessary
to display items in the viewed region.
This makes it possible to display alignments from files that
are so large that the connection to UCSC would time out when
attempting to upload the whole file to UCSC.
Both the compressed VCF file and its tabix index file remain on your
web-accessible server (http or ftp), not on the UCSC server.
UCSC temporarily caches the accessed portions of the files to speed up
interactive display.
</P>
<P>
Go to the <A HREF="../goldenPath/help/vcf.html">VCF Track Format</A>
page for more information about VCF custom tracks.</P>
			    
                        </TD><TD WIDTH=15></TD></TR>
			</TABLE><BR>
		     </TD></TR>
		</TABLE>
	     </TD></TR>
	</TABLE>
	<BR>

<!--content table-->
<a name="format11"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; ENCODE RNA elements: BED6 + 3 scores format</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <p>
<ol>
        <li><b>chrom</b> - Name of the chromosome (or contig, scaffold, etc.).
        <li><b>chromStart</b> - The starting position of the feature in the
        chromosome or scaffold. The first base in a chromosome is numbered 0.
        <li><b>chromEnd</b> - The ending position of the feature in the
        chromosome or scaffold. The <i>chromEnd</i> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <i>chromStart=0, chromEnd=100</i>, and span
        the bases numbered 0-99.
        <li><b>name</b> - Name given to a region (preferably unique). 
        Use '.' if no name is assigned.
        <li><b>score</b> - Indicates how dark the peak will be displayed in the 
        browser (0-1000). If all scores were '0' when the data were submitted to the DCC, the DCC assigned scores 1-1000 based on signal value. Ideally the average signalValue per base spread is between 100-1000.
        <li><b>strand</b> - +/- to denote strand or orientation (whenever 
        applicable). Use '.' if no orientation is assigned.
        <li><b>level</b> - Expression level such as RPKM or FPKM.
        <li><b>signif</b> - Statistical significance such as IDR.
        <li><b>score2</b> - Additional measurement/count e.g. number of reads.
</ol>			    </td><td width=15></td></tr>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>

<!--content table-->
<a name="format12"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; ENCODE narrowPeak: Narrow (or Point-Source) Peaks format</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <p>
<P>This format is used to provide called peaks of signal enrichment based on pooled, normalized (interpreted) data. It is a BED6+4 format.
</P>
<ol>
        <li><b>chrom</b> - Name of the chromosome (or contig, scaffold, etc.).
        <li><b>chromStart</b> - The starting position of the feature in the
        chromosome or scaffold. The first base in a chromosome is numbered 0.
        <li><b>chromEnd</b> - The ending position of the feature in the
        chromosome or scaffold. The <i>chromEnd</i> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <i>chromStart=0, chromEnd=100</i>, and span
        the bases numbered 0-99.
        <li><b>name</b> - Name given to a region (preferably unique). 
        Use '.' if no name is assigned.
        <li><b>score</b> - Indicates how dark the peak will be displayed in the 
        browser (0-1000). If all scores were '0' when the data were submitted to the DCC, the DCC assigned scores 1-1000 based on signal value. Ideally the average signalValue per base spread is between 100-1000.
        <li><b>strand</b> - +/- to denote strand or orientation (whenever 
        applicable). Use '.' if no orientation is assigned.
	<li><b>signalValue</b> - Measurement of overall (usually, average) 
        enrichment for the region.
        <li><b>pValue</b> - Measurement of statistical significance (-log10). 
        Use -1 if no pValue is assigned.
        <li><b>qValue</b> - Measurement of statistical significance using 
        false discovery rate (-log10). Use -1 if no qValue is assigned.
        <li><b>peak</b> - Point-source called for this peak; 0-based offset 
        from chromStart. Use -1 if no point-source called.
</ol>
<P>
Here is an example of narrowPeak format:</P>
<PRE>
track type=narrowPeak visibility=3 db=hg19 name="nPk" description="ENCODE narrowPeak Example"
browser position chr1:9356000-9365000
chr1    9356548 9356648 .       0       .       182     5.0945  -1  50
chr1    9358722 9358822 .       0       .       91      4.6052  -1  40
chr1    9361082 9361182 .       0       .       182     9.2103  -1  75
</PRE>

        		     </td><td width=15></td></tr>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>


<!--content table-->
<a name="format13"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; ENCODE broadPeak: Broad Peaks (or Regions) format</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <p>
This format is used to provide called regions of signal enrichment based on pooled, normalized (interpreted) data. It is a BED 6+3 format.</P>
<ol>
        <li><b>chrom</b> - Name of the chromosome (or contig, scaffold, etc.).
        <li><b>chromStart</b> - The starting position of the feature in the
        chromosome or scaffold. The first base in a chromosome is numbered 0.
        <li><b>chromEnd</b> - The ending position of the feature in the
        chromosome or scaffold. The <i>chromEnd</i> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <i>chromStart=0, chromEnd=100</i>, and span
        the bases numbered 0-99. If all scores were '0' when the data were submitted to the DCC, the DCC assigned scores 1-1000 based on signal value. Ideally the average signalValue per base spread is between 100-1000.
        <li><b>name</b> - Name given to a region (preferably unique). 
        Use '.' if no name is assigned.
        <li><b>score</b> - Indicates how dark the peak will be displayed in the 
        browser (0-1000).
        <li><b>strand</b> - +/- to denote strand or orientation (whenever 
        applicable). Use '.' if no orientation is assigned.
	<li><b>signalValue</b> - Measurement of overall (usually, average) 
        enrichment for the region.
        <li><b>pValue</b> - Measurement of statistical significance (-log10). 
        Use -1 if no pValue is assigned.
        <li><b>qValue</b> - Measurement of statistical significance using 
        false discovery rate (-log10). Use -1 if no qValue is assigned.
</ol>
<P>
Here is an example of broadPeak format:</P>
<PRE>
track type=broadPeak visibility=3 db=hg19 name="bPk" description="ENCODE broadPeak Example"
browser position chr1:798200-800700
chr1     798256 798454 .       116      .       4.89716 3.70716 -1
chr1     799435 799507 .       103      .       2.46426 1.54117 -1
chr1     800141 800596 .       107      .       3.22803 2.12614 -1
</Pre>

        		     </Td><td width=15></td></tr>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>


<!--content table-->
<a name="format14"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; ENCODE gappedPeak: Gapped Peaks (or Regions) format</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>

<P>
This format is used to provide called regions of signal enrichment based on pooled, normalized (interpreted) data where the regions may be spliced or incorporate gaps in the genomic sequence. It is a BED12+3 format.
</P>
<ol>
        <li><b>chrom</b> - Name of the chromosome (or contig, scaffold, etc.).
        <li><b>chromStart</b> - The starting position of the feature in the
        chromosome or scaffold. The first base in a chromosome is numbered 0.
        <li><b>chromEnd</b> - The ending position of the feature in the
        chromosome or scaffold. The <i>chromEnd</i> base is not included in the
        display of the feature. For example, the first 100 bases of a
        chromosome are defined as <i>chromStart=0, chromEnd=100</i>, and span
        the bases numbered 0-99.
        <li><b>name</b> - Name given to a region (preferably unique). 
        Use '.' if no name is assigned.
        <li><b>score</b> - Indicates how dark the peak will be displayed in the 
        browser (0-1000). If all scores were '0' when the data were submitted to the DCC, the DCC assigned scores 1-1000 based on signal value. Ideally the average signalValue per base spread is between 100-1000.
        <li><b>strand</b> - +/- to denote strand or orientation (whenever 
        applicable). Use '.' if no orientation is assigned.
	<li><b>thickStart</b> - The starting position at which the feature is drawn thickly.
          Not used in gappedPeak type, set to 0.
	<li><b>thickEnd</b> - The ending position at which the feature is drawn thickly.
          Not used in gappedPeak type, set to 0.
	<li><b>itemRgb</b> - An RGB value of the form R,G,B (e.g. 255,0,0).
          Not used in gappedPeak type, set to 0. 
	<li><b>blockCount</b> - The number of blocks (exons) in the BED line.
	<li><b>blockSizes</b> - A comma-separated list of the block sizes. The number of items
          in this list should correspond to <I>blockCount</I>.
	<li><b>blockStarts</b> - A comma-separated list of block starts. The first value must be 0
        and all of the <I>blockStart</I> positions should be calculated relative to
        <I>chromStart</I>. The number of items in this list should correspond to <I>blockCount</I>.
	<li><b>signalValue</b> - Measurement of overall (usually, average) 
        enrichment for the region.
        <li><b>pValue</b> - Measurement of statistical significance (-log10). 
        Use -1 if no pValue is assigned.
        <li><b>qValue</b> - Measurement of statistical significance using 
        false discovery rate (-log10). Use -1 if no qValue is assigned.
</ol>
<p>
Here is an example of gappedPeak format:</p>
<pre>
track name=gappedPeakExample type=gappedPeak
chr1 171000 171600 Anon_peak_1 55 . 0 0 0 2 400,100 0,500 4.04761 7.53255 5.52807
</pre>
                              </td><td width=15></td></tr>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>

<!--content table-->
<a name="format15"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; ENCODE tagAlign: BED3+3 format (historical)</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <p>
<P><b>tagAlign was used with hg18, but is no longer in use with hg19.</b>
Tag Alignment provided genomic mapping of short sequence tags. 
It is a BED3+3 format.
</P>
<ol>
	<li><b>chrom</b> - Name of the chromosome.
        <li><b>chromStart</b> - The starting position of the feature in the chromosome. The first base in a chromosome is numbered 0.
        <li><b>chromEnd</b> - The ending position of the feature in the chromosome or scaffold. The chromEnd base is not included in the display of the feature. For example, the first 100 bases of a chromosome are defined as chromStart=0, chromEnd=100, and span the bases numbered 0-99.
        <li><b>sequence</b> - Sequence of this read.
        <li><b>score</b> - Indicates uniqueness or quality (preferably 1000/alignmentCount).
        <li><b>strand</b> - Orientation of this read (+ or -).
</ol>
<P>
Here is an example of tagAlign format:
</P>
<PRE>
chrX 8823384 8823409 AGAAGGAAAATGATGTGAAGACATA 1000 +
chrX 8823387 8823412 TCTTATGTCTTCACATCATTTTCCT 500  -
</PRE>
        		     </td><td width=15></td></tr>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>


<!--content table-->
<a name="format16"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; ENCODE pairedTagAlign: BED6+2 format (historical)</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <p>
<P><b>pairedTagAlign was used with hg18, but is no longer in use with hg19.</b>
Tag Alignment Format for Paired Reads was used to provide genomic mapping of paired-read short sequence tags. It is a BED6+2 format.
</P>
<ol>
        <li><b>chrom</b> - Name of the chromosome.
        <li><b>chromStart</b> - The starting position of the feature in the chromosome. The first base in a chromosome is numbered 0.
        <li><b>chromEnd</b> - The ending position of the feature in the chromosome or scaffold. The chromEnd base is not included in the display of the feature. For example, the first 100 bases of a chromosome are defined as chromStart=0, chromEnd=100, and span the bases numbered 0-99.
        <li><b>name</b> - Identifier of paired-read.
        <li><b>score</b> - Indicates uniqueness or quality (preferably 1000/alignment-count).
        <li><b>strand</b> - Orientation of this read (+ or -).
        <li><b>seq1</b> - Sequence of first read.
        <li><b>seq2</b> - Sequence of second read.
</ol>			    </td><td width=15></td></tr>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>

<a name="format17"></a><br>
<table bgcolor="#888888" border="0" cellpadding="1" cellspacing="0"                      
 width="100%">   
  <tbody>
    <tr>
      <td>
      <table bgcolor="#fffee8" border="0" cellpadding="0"
 cellspacing="0" width="100%">
        <tbody>
          <tr>
            <td>
            <table bgcolor="#d9e4f8"  width="100%">
              <tbody>
                <tr>
                  <td>
                  <table cellpadding="0" cellspacing="0" width="100%">
                    <tbody>
                      <tr>
                        <td align="left"><font size="4"><b>&nbsp;
ENCODE peptideMapping: BED6+4 format</b></font></td>
                        <td align="right"><a href="#top"><img
 src="/images/index.gif" alt="" align="right" border="0"></a> </td>
                      </tr>
                    </tbody>
                  </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <table bgcolor="#fffee8" cellpadding="0" width="100%">
              <tbody>
                <tr>
                  <th height="10"><br>
                  </th>
                </tr>
                <tr>
                  <td width="10">&nbsp;</td>
                  <td>
                  <p></p>
                  <p><b>PeptideMapping. </b>The peptide mapping format was used
		    to provide genomic mapping
		    of proteogenomic mappings of peptides to the genome, with information
		    that is appropriate for assessing the confidence of the mapping.&nbsp; <br>
		  </p>
		  <ol>
		    <li><b>chrom</b> - Name of the chromosome. </li>
		    <li><b>chromStart</b> - The starting position of the feature
		      in the chromosome. The first base in a chromosome is numbered 0. </li>
		    <li><b>chromEnd</b>
		      - The ending position of the feature in the chromosome or scaffold. The
		      chromEnd base is not included in the display of the feature. For
		      example, the first 100 bases of a chromosome are defined as
		      chromStart=0, chromEnd=100, and span the bases numbered 0-99. </li>
		    <li><b>name</b> - The peptide sequence. </li>
		    <li><b>score</b> - Indicates uniqueness or quality
		      (preferably 1000/alignment-count). </li>
		    <li><b>strand</b> - Orientation of this read (+ or -). </li>
		    <li><b>rawScore</b> - Raw score for this hit, as estimated through HMM analysis. </li>
		    <li><b>spectrumId</b> - Non-unique identifier for the spectrum file </li>
		    <li><b>peptideRank</b> - Rank of this hit, for peptides with multiple genomic hits></li>
		    <li><b>peptideRepeatCount</b> - Indicates how many times this same hit was observed</li>
		  </ol>
                  </td>
                  <td width="15"><br>
                  </td>
                </tr>
              </tbody>
            </table>
            <br>
            </td>
          </tr>
        </tbody>
      </table>
      </td>
    </tr>
  </tbody>
</table>
<br>

<!--content table-->
<a name="format18"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; Fasta Format</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <!--body here-->
                        <A HREF="http://genetics.bwh.harvard.edu/pph/FASTA.html">http://genetics.bwh.harvard.edu/pph/FASTA.html</A>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>

<!--content table-->
<a name="format19"></a>
    <!--outer table is for border purposes-->        
    	<table width="100%" bgcolor="#888888" border="0" cellspacing="0" cellpadding="1">
             <tr><td>            
        	<table BGCOLOR="#fffee8" width="100%"  border="0" cellspacing="0" cellpadding="0">
	    	     <tr><td>                
  	    		<table BGCOLOR="#D9e4f8" width="100%">
	        	     <tr><td>                
	   			<table cellpadding=0 cellspacing=0 width="100%">
		    		     <tr><td align="left"><font size="4"><b>&nbsp; FastQ Format</b></font></td> 
		    		     <td align="right"><a href="#top"><img src="/images/index.gif" alt="" align="right" border=0></a>
	            		     </td></tr>
  	        		</table>                
	        	     </td></tr>
  	    		</table>                
   	    		<table BGCOLOR="#fffee8" width="100%" cellpadding=0>
	        	    <tr><th height=10></th></tr>
                	    <tr><td width=10>&nbsp;</td>
		    	    <td>
	            	    <!--body here-->
                        <A HREF="http://maq.sourceforge.net/fastq.shtml">http://maq.sourceforge.net/fastq.shtml</A>
	    		</table><br>
		     </td></tr>
		</table>
     	     </td></tr>
    	</table>
	<br>

</BODY></HTML>
