<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../inc/FAQ.topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Mirroring or Licensing the Genome Browser</b></FONT>
                </TD></TR></TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
		<UL>
                <LI><A HREF="#license1">Licensing the Genome Browser or Blat</A>
                <LI><A HREF="#license3">Downloading the Genome Browser source</A>
                <LI><A HREF="#license2">Mirroring the Genome Browser</A>
                <LI><A HREF="#license4">Setting up a mirror with a customized data set</A>
		</UL>
		<A HREF="index.html">Return to FAQ Table of Contents</A>
		<BR>
            </TD><TD WIDTH=15>
	   </TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>


<!--Content Table------------------------------------------------------------>
<A NAME="license1"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Licensing the Genome Browser or Blat</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Do I need a license to install the Genome Browser or its databases on my
			own machine?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			A license is required for commercial use of the Genome Browser
			or Blat alignment tool.
			No license is needed for academic, nonprofit, and personal 
			use. The data displayed by the Genome Browser is freely 
			available for both public and commercial use with a few 
			exceptions. Check the README.txt file in the assembly download 
			directory to view the use restrictions specific to that 
			release. 
			<P>
			For information on licensing the Genome Browser or Blat tool,
			see the <A HREF="../license/">licensing
			page</A>.
            </TD><TD WIDTH=15>
	   </TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="license3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Downloading the Genome Browser source</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Where can I download the source code and executables for the Genome Browser?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The Genome Browser source code and executables are freely
			available for academic, nonprofit, and personal 
			use (see <A HREF="#license1">Licensing the Genome Browser
			or Blat</A> for commercial licensing requirements). 
			The latest version of the source code may be downloaded 
			<A HREF="http://hgdownload.cse.ucsc.edu/admin/jksrc.zip">here</A>.
			<P>
			See <A HREF="FAQblat.html#blat3">Downloading Blat source
			and documentation</A> for information on Blat downloads.
            </TD><TD WIDTH=15>
	   </TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!--Content Table------------------------------------------------------------>
<A NAME="license2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Mirroring the Genome Browser</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Our academic institution would like to install and run the 
			Genome Browser and its databases on our local server. How do
			we do this? Is there a procedure for updating the data when
			new tables and assemblies are released?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Non-commercial organizations are welcome to become a Genome
			Browser mirror site. A license is required for commercial 
			mirroring of the Genome Browser. For detailed procedures on
			creating a full or partial mirror browser, see the 
			<A HREF="../admin/mirror.html">mirror site procedures page</A>.
            </TD><TD WIDTH=15>
	   </TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!-- Content Table ========================================================== -->
<A NAME="license4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="D9E4F8" BACKGROUND="../images/hr.gif" WIDTH=100%>
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Setting up a mirror with a customized data set</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Is it possible to set up a local version of the Genome
			Browser that uses my own database rather than UCSC's?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The default Genome Browser installation described on
			the 
			<A HREF="../admin/mirror.html">mirror
			page</A> includes all the databases and annotation 
			tracks found on the UCSC Genome Browser website. It is
			possible to download a smaller data set to conserve
			space on your server. Or, if you prefer, you can load
			your local version of the Genome Browser with your own
			data rather than using the data supplied by UCSC. 
			<P>
			Here are the basic steps to follow to add a new
			genome assembly to the Genome Browser (replace 
			references to <em>newGenome</em> with the name of your
			genome assembly):
			<OL>
			<LI>
			Download your sequences in fasta format. This format
			is usually organized as one file per chromosome, 
			although unfinished assemblies may be grouped into
			scaffolds rather than chromosomes. In some cases, the
			assembly may consist only of unplaced contigs, 
			<em>eg.</em> the <em>C. briggsae</em> assembly. In 
			these cases, we arbitrarily group the contigs 
			together&mdash;separated by some sort of gap&mdash;into 
			a single chromosome chrUn (this works only for small 
			organisms).
			<LI>
			Repeatmask the sequence, then concatenate the masked
			fasta files into a single two-bit 
			file (.2bit) using the utility <TT>faToTwoBit</TT>:
			<BLOCKQUOTE>
			faToTwoBit <em>newGenome</em>.fa /gbdb/<em>newGenome</em>/<em>newGenome</em>.2bit
			</BLOCKQUOTE>
			For information on the usage options for the
			faToTwoBit utility, execute the command with no 
			arguments (this is true for all utilities in the Genome 
			Browser source tree). The faToTwoBit utility will place 
			the resulting .2bit file in the directory
			/gbdb/<em>newGenome</em>/<em>newGenome</em>.2bit.
			<LI>
			Create a database for the new assembly using the 
			command: 
			<BLOCKQUOTE>
			hgsql -e "create database <em>newGenome</em>;" mysql
			</BLOCKQUOTE>
			<LI>
			Create a group table for the new database, using any of 
			the existing UCSC grp.sql SQL dumps:
			<BLOCKQUOTE>
     			hgsql <em>newGenome</em> < grp.sql<BR>
     			hgsql <em>newGenome</em> -e 'load data local infile "grp.sql" into table grp;'
			</BLOCKQUOTE>
			<LI>
			Create a chromInfo table from the .2bit file:
			<BLOCKQUOTE>
     			twoBitInfo <em>newGenome</em>.2bit stdout | \<BR>
     			awk '{printf "%s\t%s\t/gbdb/<em>newGenome</em>/<em>newGenome</em>.2bit\n", $1,$2}' > chromInfo.tab<BR>
     			hgsql <em>newGenome</em> < $HOME/kent/src/hg/lib/chromInfo.sql<BR>
     			hgsql <em>newGenome</em> -e 'load data local infile "chromInfo.tab" into table chromInfo;'<BR>
			</BLOCKQUOTE>
			<LI>		
			Add an entry for the new assembly to the dbDb and 
			defaultDb tables in the hgcentral database. For example, the entry for the
			monDom1 opossum assembly was created like this:
			<BLOCKQUOTE>
     			# Enter monDom1 into dbDb and defaultDb so test browser knows about it:<BR>
     			hgsql -e 'INSERT INTO dbDb (name, description, nibPath, organism, \<BR>
         		defaultPos, active, orderKey, genome, scientificName, \<BR>
         		htmlPath, hgNearOk, hgPbOk, sourceName) \<BR>
         		VALUES("monDom1", "Oct 2004", "/gbdb/monDom1", "M. domestica", \<BR>
         		"scaffold_13303:1000000-11000000", 1, 33, "Opossum", \<BR>
         		"Monodelphis domestica", \<BR>
         		"/gbdb/monDom1/html/description.html", 0, 0, \<BR>
         		"Broad Inst. Prelim Oct04");' \<BR>
         		-h localhost hgcentraltest<BR><BR>
     			hgsql -e 'INSERT INTO defaultDb (name, genome) \<BR>
         		VALUES("monDom1", "Opossum")' \<BR>
         		-h localhost hgcentraltest<BR>
			</BLOCKQUOTE>
			<LI>
			Add the new database name to the DBS variable in the 
			makefile in 
			src/hg/makeDb/trackDb/makefile. 
			You can configure the
			list of genome databases into which you want to load
			trackDb tables by editing the list in this makefile.  
                        You may also want to comment out the 'git pull' 
                        step for the 'alpha' make target in that file.
			For the makefile to work correctly,
			add a subdirectory in 
			src/hg/makeDb/trackDb/ corresponding
			to the organism name (<em>e.g.</em> opossum), and then a
			<em>newGenome</em> subdirectory in the organism 
			directory. Using the opossum example shown in the 
			previous step, this would result in the directory
			hierarchy 
			src/hg/makeDb/trackDb/opossum/monDom1/. 
			<LI>
			Run "make alpha" in src/hg/makeDb/trackDb/.  This 
			creates the tables trackDb and hgFindSpec that contain
			information about the Genome Browser tracks that will 
			be displayed for the new assembly. For more information
			about creating these tables, see 
			src/product/README.trackDb in the source tree. 
			Do not load 
			the trackDb table manually; instead, use the
			trackDb.ra file format found in the source tree at
			src/hg/makeDb/trackDb/ and load it with the 
			command <TT>'make alpha'</TT>. 
			</OL>
			<P>
			The previous steps will create an empty browser for the
			new assembly, which can then be populated with 
			annotation tracks, as desired. For examples of how
			tracks are created for the standard Genome Browser
			assemblies, see the files src/hg/doc/*.txt
			in the Genome Browser source tree (these are plain text
			files, not MS Word documents). The Gold and Gap 
			tracks can be created from the genome assembly's AGP 
			file.  
			<P>
			If you encounter problems or have questions about this
			procedure, email the 
			<A HREF="mailto:genome-mirror@soe.ucsc.edu">genome-mirror</A> mailing list.
			<B>Messages sent to this address will be posted to the 
			moderated genome-mirror mailing list, which is archived
			on a public Web-accessible pipermail archive.</B>  This
			archive may be indexed by non-UCSC sites such as Google.
			</P>
		    </TD><TD WIDTH=15></TD></TR></TABLE>
	    <br></TD></TR></TABLE>
	</TD></TR></TABLE>
    </TD><TD WIDTH=15></TD></TR></TABLE>
</BODY>
</HTML>

