<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
        <META name="description" content="UCSC Genome Browser FAQ">
        <META name="keywords" content="genome, genome browser, genome assembly, UCSC, bioinformatics, sequence, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Bioinformatics: FAQ</TITLE>
	<LINK REL="STYLESHEET" HREF="../style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>

<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="../images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<noscript><div class='noscript'><div class='noscript-inner'><p><b>JavaScript is disabled in your web browser</b></p><p>You must have JavaScript enabled in your web browser to use the Genome Browser</p></div></div></noscript>
<script type='text/javascript' SRC='../js/jquery.js'></script>
<script type='text/javascript' SRC='../js/jquery.plugins.js'></script>
<LINK rel='STYLESHEET' href='../style/nice_menu.css' TYPE='text/css' />
<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="../cgi-bin/hgMenubar"--></TD></TR>
<!--Content Tables==================================================== -->
<TR><TD COLSPAN=2>
    <!--outer table is for border purposes-->
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>
            <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
                <TR><TD> <FONT SIZE="4"><b>&nbsp;   Frequently Asked Questions: Assembly Releases and Versions</b></FONT>
                </TD></TR></TABLE>
            <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TR><TH HEIGHT=10></TH></TR>
                <TR><TD>&nbsp;</TD><TD>
		<UL>
                <LI><A HREF="#release1">List of UCSC genome releases</A>
                <LI><A HREF="#release2">Initial assembly release dates</A>
                <LI><A HREF="#release3">UCSC assemblies</A>
                <LI><A HREF="#release4">Comparison of UCSC and NCBI human assemblies</A>
                <LI><A HREF="#release12">Differences between UCSC and NCBI mouse assemblies</A>
                <LI><A HREF="#release5">Accessing older assembly versions</A>
                <LI><A HREF="#release6">Frequency of GenBank data updates</A>
                <LI><A HREF="#release7">Coordinate changes between assemblies</A>
                <LI><A HREF="#release8">Converting positions between assembly versions</A>
                <LI><A HREF="#release9">Missing annotation tracks</A>
                <LI><A HREF="#release10">What next with the human genome?</A>
                <LI><A HREF="#release11">Mouse strain used for mouse genome sequence</A>
                <LI><A HREF="#release13">UniProt (Swiss-Prot/TrEMBL) display changes</A>
		</UL>
		<A HREF="index.html">Return to FAQ Table of Contents</A>
		<BR>
            </TD><TD WIDTH=15></TD></TR></TABLE><BR>
        </TD></TR></TABLE>
    </TD></TR></TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release1"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; List of UCSC genome releases</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do UCSC's release numbers correspond to those of other organizations, such as NCBI?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR>

                        The first release of an assembly is given a name using the first three characters of the organism's
                        genus and species classification in the format gggSss#, with subsequent assemblies incrementing
                        the number.  Assemblies predating the 2003 introduction of the six-letter naming system were
                        given two-letter names in a similar gs# format and human assemblies are named hg# for human genome.
                        <BR>
                        <P>
<!--
    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH>        
	<TR><TD WIDTH=10></TD><TD>
-->
        <!--track table-->
<!--
            <TABLE BORDER=1 BORDERCOLOR="#aaaaaa" CELLPADDING=4 WIDTH="100%">            
-->
            <TABLE class="descTbl" WIDTH="100%">            
		<TR>
		<TD nowrap><FONT color="#006666"><B>SPECIES</B></FONT></TD>
                <TD nowrap><FONT color="#006666"><B>UCSC VERSION</B></FONT></TD>
                <TD nowrap><FONT color="#006666"><B>RELEASE DATE</B></FONT></TD>
                <TD nowrap><FONT color="#006666"><B>RELEASE NAME</B></FONT></TD>
                <TD nowrap><FONT color="#006666"><B>STATUS</B></FONT></TD>
            </TR>
  			<TR><TD><B>MAMMALS</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD>Human</TD><TD>hg38</TD><TD>Dec. 2013</TD><TD>Genome Reference Consortium GRCh38</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg19</TD><TD>Feb. 2009</TD><TD>Genome Reference Consortium GRCh37</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg18</TD><TD>Mar. 2006</TD><TD>NCBI Build 36.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg17</TD><TD>May 2004</TD><TD>NCBI Build 35</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg16</TD><TD>Jul. 2003</TD><TD>NCBI Build 34</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg15</TD><TD>Apr. 2003</TD><TD>NCBI Build 33</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg13</TD><TD>Nov. 2002</TD><TD>NCBI Build 31</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg12</TD><TD>Jun. 2002</TD><TD>NCBI Build 30</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg11</TD><TD>Apr. 2002</TD><TD>NCBI Build 29</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg10</TD><TD>Dec. 2001</TD><TD>NCBI Build 28</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg8</TD><TD>Aug. 2001</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg7</TD><TD>Apr. 2001</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg6</TD><TD>Dec. 2000</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg5</TD><TD>Oct. 2000</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg4</TD><TD>Sep. 2000</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg3</TD><TD>Jul. 2000</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg2</TD><TD>Jun. 2000</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>hg1</TD><TD>May 2000</TD><TD>UCSC-assembled</TD><TD>Archived (data only)</TD></TR>
			<TR><TD>Alpaca</TD><TD>vicPac2</TD><TD>Mar. 2013</TD><TD>Broad Institute Vicugna_pacos-2.0.1</TD><TD>Available</TD></TR>
			<TR><TD>&nbsp;</TD><TD>vicPac1</TD><TD>Jul. 2008</TD><TD>Broad Institute VicPac1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Armadillo</TD><TD>dasNov3</TD><TD>Dec. 2011</TD><TD>Broad Institute DasNov3</TD><TD>Available</TD></TR>
  			<TR><TD>Bushbaby</TD><TD>otoGar3</TD><TD>Mar. 2011</TD><TD>Broad Institute OtoGar3</TD><TD>Available</TD></TR>
  			<TR><TD>Baboon</TD><TD>papHam1</TD><TD>Nov. 2008</TD><TD>Baylor College of Medicine HGSC Pham_1.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>papAnu2</TD><TD>Mar. 2012</TD><TD>Baylor College of Medicine Panu_2.0</TD><TD>Available</TD></TR>
			<TR><TD>Cat</TD><TD>felCat5</TD><TD>Sep. 2011</TD><TD>ICGSC Felis_catus-6.2</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>felCat4</TD><TD>Dec. 2008</TD><TD>NHGRI catChrV17e</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>felCat3</TD><TD>Mar. 2006</TD><TD>Broad Institute Release 3</TD><TD>Available</TD></TR>
  			<TR><TD>Chimp</TD><TD>panTro4</TD><TD>Feb. 2011</TD><TD>CGSC Build 2.1.4</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>panTro3</TD><TD>Oct. 2010</TD><TD>CGSC Build 2.1.3</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>panTro2</TD><TD>Mar. 2006</TD><TD>CGSC Build 2.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>panTro1</TD><TD>Nov. 2003</TD><TD>CGSC Build 1.1</TD><TD>Available</TD></TR>
                        <TR><TD>Chinese hamster</TD><TD>criGri1</TD><TD>Jul. 2013</TD><TD>Beijing Genomics Institution-Shenzhen C_griseus_v1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Cow</TD><TD>bosTau8</TD><TD>Jun. 2014</TD><TD>University of Maryland v3.1.1</TD><TD>Available</TD></TR>
			<TR><TD>&nbsp;</TD><TD>bosTau7</TD><TD>Oct. 2011</TD><TD>Baylor College of Medicine HGSC Btau_4.6.1</TD><TD>Available</TD<TR>
  			<TR><TD>&nbsp;</TD><TD>bosTau6</TD><TD>Nov. 2009</TD><TD>University of Maryland v3.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>bosTau4</TD><TD>Oct. 2007</TD><TD>Baylor College of Medicine HGSC Btau_4.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>bosTau3</TD><TD>Aug. 2006</TD><TD>Baylor College of Medicine HGSC Btau_3.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>bosTau2</TD><TD>Mar. 2005</TD><TD>Baylor College of Medicine HGSC Btau_2.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>bosTau1</TD><TD>Sep. 2004</TD><TD>Baylor College of Medicine HGSC Btau_1.0</TD><TD>Archived</TD></TR>
			<TR><TD>Dog</TD><TD>canFam3</TD><TD>Sep. 2011</TD><TD>Broad Institute v3.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>canFam2</TD><TD>May 2005</TD><TD>Broad Institute v2.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>canFam1</TD><TD>Jul. 2004</TD><TD>Broad Institute v1.0</TD><TD>Available</TD></TR>
			<TR><TD>Dolphin</TD><TD>turTru2</TD><TD>Oct. 2011</TD><TD>Baylor College of Medicine Ttru_1.4</TD><TD>Available</TD></TR>
  			<TR><TD>Elephant</TD><TD>loxAfr3</TD><TD>Jul. 2009</TD><TD>Broad Institute LoxAfr3</TD><TD>Available</TD></TR>
                        <TR><TD>Ferret</TD><TD>musFur1</TD><TD>Apr. 2011</TD><TD>Ferret Genome Sequencing Consortium MusPutFur1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Gibbon</TD><TD>nomLeu3</TD><TD>Oct. 2012</TD><TD>Gibbon Genome Sequencing Consortium Nleu3.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>nomLeu2</TD><TD>Jun. 2011</TD><TD>Gibbon Genome Sequencing Consortium Nleu1.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>nomLeu1</TD><TD>Jan. 2010</TD><TD>Gibbon Genome Sequencing Consortium Nleu1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Gorilla</TD><TD>gorGor3</TD><TD>May 2011</TD><TD>Wellcome Trust Sanger Institute gorGor3.1</TD><TD>Available</TD></TR>
  			<TR><TD>Guinea pig</TD><TD>cavPor3</TD><TD>Feb. 2008</TD><TD>Broad Institute cavPor3</TD><TD>Available</TD></TR>
  			<TR><TD>Hedgehog</TD><TD>eriEur2</TD><TD>May 2012</TD><TD>Broad Institute EriEur2.0</TD><TD>Available</TD></TR>
                        <TR><TD>&nbsp;</TD><TD>eriEur1</TD><TD>Jun. 2006</TD><TD>Broad Institute Draft_v1</TD><TD>Available</TD></TR>
  			<TR><TD>Horse</TD><TD>equCab2</TD><TD>Sep. 2007</TD><TD>Broad Institute EquCab2</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>equCab1</TD><TD>Jan. 2007</TD><TD>Broad Institute EquCab1</TD><TD>Available</TD></TR>
  			<TR><TD>Kangaroo rat</TD><TD>dipOrd1</TD><TD>Jul. 2008</TD><TD>Baylor/Broad Institute DipOrd1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Manatee</TD><TD>triMan1</TD><TD>Oct. 2011</TD><TD>Broad Institute TriManLat1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Marmoset</TD><TD>calJac3</TD><TD>Mar. 2009</TD><TD>WUSTL Callithrix_jacchus-v3.2</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>calJac1</TD><TD>Jun. 2007</TD><TD>WUSTL Callithrix_jacchus-v2.0.2</TD><TD>Available</TD></TR>
  			<TR><TD>Megabat</TD><TD>pteVam1</TD><TD>Jul. 2008</TD><TD>Broad Institute Ptevap1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Microbat</TD><TD>myoLuc2</TD><TD>Jul. 2010</TD><TD>Broad Institute MyoLuc2.0</TD><TD>Available</TD></TR>
  			<TR><TD>Minke whale</TD><TD>balAcu1</TD><TD>Oct. 2013</TD><TD>KORDI BalAcu1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Mouse</TD><TD>mm10</TD><TD>Dec. 2011</TD><TD>Genome Reference Consortium GRCm38</TD><TD>Available</TD></TR>
			<TR><TD>&nbsp;</TD><TD>mm9</TD><TD>Jul. 2007</TD><TD>NCBI Build 37</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm8</TD><TD>Feb. 2006</TD><TD>NCBI Build 36</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm7</TD><TD>Aug. 2005</TD><TD>NCBI Build 35</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm6</TD><TD>Mar. 2005</TD><TD>NCBI Build 34</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm5</TD><TD>May 2004</TD><TD>NCBI Build 33</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm4</TD><TD>Oct. 2003</TD><TD>NCBI Build 32</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm3</TD><TD>Feb. 2003</TD><TD>NCBI Build 30</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm2</TD><TD>Feb. 2002</TD><TD>MGSCv3</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>mm1</TD><TD>Nov. 2001</TD><TD>MGSCv2</TD><TD>Archived (data only)</TD></TR>
  			<TR><TD>Mouse lemur</TD><TD>micMur1</TD><TD>Jul. 2007</TD><TD>Broad Institute MicMur1.0</TD><TD>Available</TD></TR>
		        <TR><TD>Naked mole-rat</TD><TD>hetGla2</TD><TD>Jan. 2012</TD><TD>Broad Institute HetGla_female_1.0</TD><TD>Available</TD></TR>
		        <TR><TD>&nbsp;</TD><TD>hetGla1</TD><TD>Jul. 2011</TD><TD>Beijing Genomics Institute HetGla_1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Opossum</TD><TD>monDom5</TD><TD>Oct. 2006</TD><TD>Broad Institute release MonDom5</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>monDom4</TD><TD>Jan. 2006</TD><TD>Broad Institute release MonDom4</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>monDom1</TD><TD>Oct. 2004</TD><TD>Broad Institute release MonDom1</TD><TD>Available</TD></TR>
  			<TR><TD>Orangutan</TD><TD>ponAbe2</TD><TD>Jul. 2007</TD><TD>WUSTL Pongo_albelii-2.0.2</TD><TD>Available</TD></TR>
  			<TR><TD>Panda</TD><TD>ailMel1</TD><TD>Dec. 2009</TD><TD>BGI-Shenzhen AilMel 1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Pig</TD><TD>susScr3</TD><TD>Aug. 2011</TD><TD>Swine Genome Sequencing Consortium Sscrofa10.2</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>susScr2</TD><TD>Nov. 2009</TD><TD>Swine Genome Sequencing Consortium Sscrofa9.2</TD><TD>Available</TD></TR>
  			<TR><TD>Pika</TD><TD>ochPri3</TD><TD>May 2012</TD><TD>Broad Institute OchPri3.0</TD><TD>Available</TD></TR>
                        <TR><TD>&nbsp;</TD><TD>ochPri2</TD><TD>Jul. 2008</TD><TD>Broad Institute OchPri2</TD><TD>Available</TD></TR>
  			<TR><TD>Platypus</TD><TD>ornAna1</TD><TD>Mar. 2007</TD><TD>WUSTL v5.0.1</TD><TD>Available</TD></TR>
  			<TR><TD>Rabbit</TD><TD>oryCun2</TD><TD>Apr. 2009</TD><TD>Broad Institute release OryCun2</TD><TD>Available</TD></TR>
			<TR><TD>Rat</TD><TD>rn6</TD><TD>Jul. 2014</TD><TD>RGSC Rnor_6.0</TD><TD>Available</TD></TR>
			<TR><TD>&nbsp;</TD><TD>rn5</TD><TD>Mar. 2012</TD><TD>RGSC Rnor_5.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>rn4</TD><TD>Nov. 2004</TD><TD>Baylor College of Medicine HGSC v3.4</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>rn3</TD><TD>Jun. 2003</TD><TD>Baylor College of Medicine HGSC v3.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>rn2</TD><TD>Jan. 2003</TD><TD>Baylor College of Medicine HGSC v2.1</TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>rn1</TD><TD>Nov. 2002</TD><TD>Baylor College of Medicine HGSC v1.0</TD><TD>Archived</TD></TR>
  			<TR><TD>Rhesus</TD><TD>rheMac3</TD><TD>Oct. 2010</TD><TD>Beijing Genomics Institute CR_1.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>rheMac2</TD><TD>Jan. 2006</TD><TD>Baylor College of Medicine HGSC v1.0 Mmul_051212</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>rheMac1</TD><TD>Jan. 2005</TD><TD>Baylor College of Medicine HGSC Mmul_0.1</TD><TD>Archived</TD></TR>
  			<TR><TD>Rock hyrax</TD><TD>proCap1</TD><TD>Jul. 2008</TD><TD>Baylor College of Medicine HGSC Procap1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Sheep</TD><TD>oviAri3</TD><TD>Aug. 2012</TD><TD>ISGC Oar_v3.1</TD><TD>Available</TD></TR>	
   			<TR><TD>&nbsp;</TD><TD>oviAri1</TD><TD>Feb. 2010</TD><TD>ISGC Ovis aries 1.0</TD><TD>Available</TD></TR>
                        <TR><TD>Shrew</TD><TD>sorAra2</TD><TD>Aug. 2008</TD><TD>Broad Institute SorAra2.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>sorAra1</TD><TD>Jun. 2006</TD><TD>Broad Institute SorAra1.0</TD><TD>Available</TD></TR>	
  			<TR><TD>Sloth</TD><TD>choHof1</TD><TD>Jul. 2008</TD><TD>Broad Institute ChoHof1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Squirrel</TD><TD>speTri2</TD><TD>Nov. 2011</TD><TD>Broad Institute SpeTri2.0</TD><TD>Available</TD></TR>
  			<TR><TD>Squirrel monkey</TD><TD>saiBol1</TD><TD>Oct. 2011</TD><TD>Broad Institute SaiBol1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Tarsier</TD><TD>tarSyr2</TD><TD>Sep. 2013</TD><TD>WashU Tarsius_syrichta-2.0.1</TD><TD>Available</TD></TR>	
  			<TR><TD>&nbsp;</TD><TD>tarSyr1</TD><TD>Aug. 2008</TD><TD>WUSTL/Broad Institute Tarsyr1.0</TD><TD>Available</TD></TR>	
			<TR><TD>Tasmanian devil</TD><TD>sarHar1</TD><TD>Feb. 2011</TD><TD>Wellcome Trust Sanger Institute Devil_refv7.0</TD><TD>Available</TD></TR>
  			<TR><TD>Tenrec</TD><TD>echTel2</TD><TD>Nov. 2012</TD><TD>Broad Institute EchTel2.0</TD><TD>Available</TD></TR>
                        <TR><TD></TD><TD>echTel1</TD><TD>Jul. 2005</TD><TD>Broad Institute echTel1</TD><TD>Available</TD></TR>
  			<TR><TD>Tree shrew</TD><TD>tupBel1</TD><TD>Dec. 2006</TD><TD>Broad Institute Tupbel1.0</TD><TD>Available</TD></TR>	
			<TR><TD>Wallaby</TD><TD>macEug2</TD><TD>Sep. 2009</TD><TD>Tammar Wallaby Genome Sequencing Consortium Meug_1.1</TD><TD>Available</TD></TR> 
			<TR><TD>White rhinoceros</TD><TD>cerSim1</TD><TD>May 2012</TD><TD>Broad Institute CerSimSim1.0</TD><TD>Available</TD></TR> 
  			<TR><TD><B>&nbsp;</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><B>VERTEBRATES</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
                        <TR><TD>American alligator</TD><TD>allMis1</TD><TD>Aug. 2012</TD><TD>Int. Crocodilian Genomes Working Group allMis0.2</TD><TD>Available</TD></TR>
  			<TR><TD>Atlantic cod</TD><TD>gadMor1</TD><TD>May 2010</TD><TD>Genofisk GadMor_May2010</TD><TD>Available</TD></TR>
  			<TR><TD>Budgerigar</TD><TD>melUnd1</TD><TD>Sep. 2011</TD><TD>WUSTL v6.3</TD><TD>Available</TD></TR>
  			<TR><TD>Chicken</TD><TD>galGal4</TD><TD>Nov.
2011</TD><TD>ICGC Gallus-gallus-4.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>galGal3</TD><TD>May 2006</TD><TD>WUSTL Gallus-gallus-2.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>galGal2</TD><TD>Feb. 2004</TD><TD>WUSTL Gallus-gallus-1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Coelacanth</TD><TD>latCha1</TD><TD>Aug. 2011</TD><TD>Broad Institute LatCha1</TD><TD>Available</TD></TR>
  			<TR><TD>Elephant shark</TD><TD>calMil1</TD><TD>Dec. 2013</TD><TD>IMCB Callorhinchus_milli_6.1.3</TD><TD>Available</TD></TR>
  			<TR><TD>Fugu</TD><TD>fr3</TD><TD>Oct. 2011</TD><TD>JGI v5.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>fr2</TD><TD>Oct. 2004</TD><TD>JGI v4.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>fr1</TD><TD>Aug. 2002</TD><TD>JGI v3.0</TD><TD>Available</TD></TR>
                        <TR><TD>Lamprey</TD><TD>petMar2</TD><TD>Sep. 2010</TD><TD>WUGSC 7.0</TD><TD>Available</TD></TR>
  			<TR><TD></TD><TD>petMar1</TD><TD>Mar. 2007</TD><TD>WUSTL v3.0</TD><TD>Available</TD></TR>
  			<TR><TD>Lizard</TD><TD>anoCar2</TD><TD>May 2010</TD><TD>Broad Institute AnoCar2</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>anoCar1</TD><TD>Feb. 2007</TD><TD>Broad Institute AnoCar1</TD><TD>Available</TD></TR>
  			<TR><TD>Medaka</TD><TD>oryLat2</TD><TD>Oct. 2005</TD><TD>NIG v1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Medium ground finch</TD><TD>geoFor1</TD><TD>Apr. 2012</TD><TD>BGI GeoFor_1.0 / NCBI 13302</TD><TD>Available</TD></TR>
  			<TR><TD>Nile tilapia</TD><TD>oreNil2</TD><TD>Jan. 2011</TD><TD>Broad Institute Release OreNil1.1</TD><TD>Available</TD></TR>
  			<TR><TD>Painted turtle</TD><TD>chrPic1</TD><TD>Dec. 2011</TD><TD>IPTGSC Chrysemys_picta_bellii-3.0.1</TD><TD>Available</TD></TR>
  			<TR><TD>Stickleback</TD><TD>gasAcu1</TD><TD>Feb. 2006</TD><TD>Broad Institute Release 1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Tetraodon</TD><TD>tetNig2</TD><TD>Mar. 2007</TD><TD>Genoscope v7</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>tetNig1</TD><TD>Feb. 2004</TD><TD>Genoscope v7</TD><TD>Available</TD></TR>
  			<TR><TD>Turkey</TD><TD>melGal1</TD><TD>Dec. 2009</TD><TD>Turkey Genome Consortium v2.01</TD><TD>Available</TD></TR>
  			<TR><TD><em>X. tropicalis</em></TD><TD>xenTro3</TD><TD>Nov. 2009</TD><TD>JGI v.4.2</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>xenTro2</TD><TD>Aug. 2005</TD><TD>JGI v.4.1</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>xenTro1</TD><TD>Oct. 2004</TD><TD>JGI v.3.0</TD><TD>Available</TD></TR>
  			<TR><TD>Zebra finch</TD><TD>taeGut2</TD><TD>Feb. 2013</TD><TD>WashU taeGut324</TD><TD>Available</TD></TR>
                        <TR><TD>&nbsp;</TD><TD>taeGut1</TD><TD>Jul. 2008</TD><TD>WUSTL v3.2.4</TD><TD>Available</TD></TR>
                        <TR><TD>Zebrafish</TD><TD>danRer10</TD><TD>Sep. 2014</TD><TD>Genome Reference Consortium GRCz10 </TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer7</TD><TD>Jul. 2010</TD><TD>Sanger Institute Zv9 </TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer6</TD><TD>Dec. 2008</TD><TD>Sanger Institute Zv8 </TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer5</TD><TD>Jul. 2007</TD><TD>Sanger Institute Zv7 </TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer4</TD><TD>Mar. 2006</TD><TD>Sanger Institute Zv6 </TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer3</TD><TD>May 2005</TD><TD>Sanger Institute Zv5 </TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer2</TD><TD>Jun. 2004</TD><TD>Sanger Institute Zv4 </TD><TD>Archived</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>danRer1</TD><TD>Nov. 2003</TD><TD>Sanger Institute Zv3 </TD><TD>Archived</TD></TR>
  			<TR><TD><B>&nbsp;</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><B>DEUTEROSTOMES</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><em>C. intestinalis</em></TD><TD>ci2</TD><TD>Mar. 2005</TD><TD>JGI v2.0</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>ci1</TD><TD>Dec. 2002</TD><TD>JGI v1.0</TD><TD>Available</TD></TR>
  			<TR><TD>Lancelet</TD><TD>braFlo1</TD><TD>Mar. 2006</TD><TD>JGI v1.0</TD><TD>Available</TD></TR>
  			<TR><TD><em>S. purpuratus</em></TD><TD>strPur2</TD><TD>Sep. 2006</TD><TD>Baylor College of Medicine HGSC v. Spur 2.1</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>strPur1</TD><TD>Apr. 2005</TD><TD>Baylor College of Medicine HGSC v. Spur_0.5</TD><TD>Available</TD></TR>
  			<TR><TD><B>&nbsp;</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><B>INSECTS</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><em>A. mellifera</em></TD><TD>apiMel2</TD><TD>Jan. 2005</TD><TD>Baylor College of Medicine HGSC v.Amel_2.0 </TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>apiMel1</TD><TD>Jul. 2004</TD><TD>Baylor College of Medicine HGSC v.Amel_1.2 </TD><TD>Available</TD></TR>
  			<TR><TD><em>A. gambiae</em></TD><TD>anoGam1</TD><TD>Feb. 2003</TD><TD>IAGP v.MOZ2</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. ananassae</em></TD><TD>droAna2</TD><TD>Aug. 2005</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>droAna1</TD><TD>Jul. 2004</TD><TD>TIGR Celera release</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. erecta</em></TD><TD>droEre1</TD><TD>Aug. 2005</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. grimshawi</em></TD><TD>droGri1</TD><TD>Aug. 2005</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. melanogaster</em></TD><TD>dm6</TD><TD>Aug. 2014</TD><TD>BDGP Release 6 + ISO1 MT</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>dm3</TD><TD>Apr. 2006</TD><TD>BDGP Release 5</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>dm2</TD><TD>Apr. 2004</TD><TD>BDGP Release 4</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>dm1</TD><TD>Jan. 2003</TD><TD>BDGP Release 3</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. mojavensis</em></TD><TD>droMoj2</TD><TD>Aug. 2005</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>droMoj1</TD><TD>Aug. 2004</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. persimilis</em></TD><TD>droPer1</TD><TD>Oct. 2005</TD><TD>Broad Institute release</TD><TD>Available</TD></TR>
  		<TR><TD><em>D. pseudoobscura</em></TD><TD>dp3</TD><TD>Nov. 2004</TD><TD>Flybase Release 1.0</TD><TD>Available</TD></TR>	
                        <TR><TD><em>&nbsp;</em></TD><TD>dp2</TD><TD>Aug. 2003</TD><TD>Baylor College of Medicine HGSC Freeze 1</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. sechellia</em></TD><TD>droSec1</TD><TD>Oct. 2005</TD><TD>Broad Institute Release 1.0</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. simulans</em></TD><TD>droSim1</TD><TD>Apr. 2005</TD><TD>WUSTL Release 1.0</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. virilis</em></TD><TD>droVir2</TD><TD>Aug. 2005</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>droVir1</TD><TD>Jul. 2004</TD><TD>Agencourt Arachne release</TD><TD>Available</TD></TR>
  			<TR><TD><em>D. yakuba</em></TD><TD>droYak2</TD><TD>Nov. 2005</TD><TD>WUSTL Release 2.0</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>droYak1</TD><TD>Apr. 2004</TD><TD>WUSTL Release 1.0</TD><TD>Available</TD></TR>
  			<TR><TD><B>&nbsp;</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><B>NEMATODES</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><em>C. brenneri</em></TD><TD>caePb2</TD><TD>Feb. 2008</TD><TD>WUSTL 6.0.1</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>caePb1</TD><TD>Jan. 2007</TD><TD>WUSTL 4.0</TD><TD>Available</TD></TR>
  			<TR><TD><em>C. briggsae</em></TD><TD>cb3</TD><TD>Jan. 2007</TD><TD>WUSTL Cb3</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>cb1</TD><TD>Jul. 2002</TD><TD>WormBase v. cb25.agp8</TD><TD>Available</TD></TR>
  			<TR><TD><em>C. elegans</em></TD><TD>ce10</TD><TD>Oct. 2010</TD><TD>WormBase v. WS220</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>ce6</TD><TD>May 2008</TD><TD>WormBase v. WS190</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>ce4</TD><TD>Jan. 2007</TD><TD>WormBase v. WS170</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>ce2</TD><TD>Mar. 2004</TD><TD>WormBase v. WS120</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>ce1</TD><TD>May 2003</TD><TD>WormBase v. WS100</TD><TD>Archived</TD></TR>
  			<TR><TD><em>C. japonica</em></TD><TD>caeJap1</TD><TD>Mar. 2008</TD><TD>WUSTL 3.0.2</TD><TD>Available</TD></TR>
  			<TR><TD><em>C. remanei</em></TD><TD>caeRem3</TD><TD>May 2007</TD><TD>WUSTL 15.0.1</TD><TD>Available</TD></TR>
  			<TR><TD><em>&nbsp;</em></TD><TD>caeRem2</TD><TD>Mar. 2006</TD><TD>WUSTL 1.0</TD><TD>Available</TD></TR>
  			<TR><TD><em>P. pacificus</em></TD><TD>priPac1</TD><TD>Feb. 2007</TD><TD>WUSTL 5.0</TD><TD>Available</TD></TR>
  			<TR><TD><B>&nbsp;</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><B>OTHER</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD>Sea Hare</TD><TD>aplCal1</TD><TD>Sep. 2008</TD><TD>Broad Release Aplcal2.0</TD><TD>Available</TD></TR>
  			<TR><TD>Yeast</TD><TD>sacCer3</TD><TD>April 2011</TD><TD>SGD April 2011 sequence</TD><TD>Available</TD></TR>
			<TR><TD></TD><TD>sacCer2</TD><TD>June 2008</TD><TD>SGD June 2008 sequence</TD><TD>Available</TD></TR>
  			<TR><TD>&nbsp;</TD><TD>sacCer1</TD><TD>Oct. 2003</TD><TD>SGD 1 Oct 2003 sequence</TD><TD>Available</TD></TR>
  			<TR><TD><B>&nbsp;</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD><B>VIRUSES</B></TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD><TD>&nbsp;</TD></TR>
  			<TR><TD>Ebola Virus</TD><TD>eboVir3</TD><TD>June 2014</TD><TD>Sierra Leone 2014 (G3683/KM034562.1)</TD><TD>Available</TD></TR>

			</TABLE>
<!--
		    </TD></TR>
    	   	    </TABLE>
-->

		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
        </TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release2"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Initial assembly release dates</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"When will the next assembly be out?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			UCSC does not produce its own genome assemblies, but instead obtains them from 
			standard sources. For example, the human assembly is obtained
			from NCBI. Because of this, you can expect us to release a new version of a 
			genome soon after the assembling organization has released the version. 
			A new assembly release initially consists of the genome sequence and a
			small set of aligned annotation tracks. Additional annotation tracks are added 
			as they are obtained or generated. Bulk downloads of the data are typically
			available in the first week after the assembly is released in the browser.</P> 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release3"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Data sources - UCSC assemblies</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Where does UCSC obtain the assembly and annotation data
			 displayed in the Genome Browser?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			All the assembly data displayed in the UCSC Genome
			Browser are obtained from external sequencing centers.
			To determine the data source and version for a given 
			assembly, see the assembly's description on the Genome
			Browser <A HREF="../cgi-bin/hgGateway">Gateway</A> page or
			the <A HREF="#release1">List of UCSC Genome Releases</A>. 
			<P>
			The annotations accompanying an assembly are obtained
			from a variety of sources. The UCSC Genome Bioinformatics 
			Group generates several of the tracks; the remainder are
			contributed by collaborators at other sites. Each track
			has an associated description page that credits the
			authors of the annotation. 
			<P>
			For detailed information about the individuals and 
			organizations who contributed to a specific assembly,
			see the <A HREF="../goldenPath/credits.html">Credits</A>
			page.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release4"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Comparison of UCSC and NCBI human assemblies</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How do the human assemblies displayed in the UCSC 
			 Genome Browser differ from the NCBI human assemblies?
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Recent human assemblies displayed in the Genome
		  	Browser (hg10 and higher) are identical to the NCBI
		  	assemblies.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release12"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Differences between UCSC and NCBI mouse assemblies</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Is the mouse genome assembly displayed in the UCSC
			Genome Browser the same as the one on the NCBI website?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The mouse genome assemblies featured in the UCSC 
			Genome Browser are the same as 
			those on the NCBI web site with one difference: the UCSC 
			versions contain
			only the reference strain data (C57BL/6J). NCBI provides
			data for several additional strains in their builds.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release5"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Accessing older assembly versions</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I need to access an older version of a genome assembly that's no
			longer listed in the Genome Browser menu. What should I do?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			In addition to the assembly versions currently available in the Genome Browser, 
			you can access the data for older assemblies of the browser through our <A
			HREF="http://hgdownload.soe.ucsc.edu/downloads.html" TARGET="_blank">Downloads</A> page.</P>
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release6"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Frequency of GenBank data updates</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"How frequently does UCSC update its databases with new 
			data from GenBank?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Daily and weekly incremental updates of mRNA, RefSeq, 
			and EST data are in place for several of the 
			more recent Genome Browser assemblies. 
			Assemblies that are not on an incremental update 
			schedule are updated whenever we load a new assembly or
			make a major revision to a table.
			<P>
			Data are updated on the following schedule:
			<UL>
			<LI>
			Native and xeno mRNA and refSeq tracks: updated 
			daily for human and mouse assemblies; updated
			approximately weekly for all other organisms.
			<LI>
			EST data: updated weekly on Saturday morning
			<LI>
			Downloadable data files: updated weekly on Saturday 
			morning
			<LI>
			Outdated sequences - removed once per quarter
			</UL>
			<P>Mirror sites are not required to use an 
			incremental update process, and should not experience 
			problems as a result of these updates. 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>


<!--Content Table========================================================= -->
<A NAME="release7"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Coordinate changes between assemblies </b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I noticed that the chromosomal coordinates for a particular gene that I'm
			looking at have changed since the last time I used your browser. What happened?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			A common source of confusion for users arises from 
			mixing up different assemblies. It is very 
			important to be aware of which assembly you are looking at. Within the Genome
			Browser display, assemblies are labeled by organism and date. To look up the 
			corresponding UCSC database name or NCBI build number, use the <A HREF="#release1">
			release table</A>.
			<P>
			UCSC database labels are of the form hg<em>#</em>, 
			panTro<em>#</em>, etc. The letters designate the organism,
			e.g. <em>hg</em> for human genome or <em>panTro</em> for
			<em>Pan troglodytes</em>. The number denotes the UCSC 
		   	assembly version for that organism.  For example, ce1 
			refers to the first UCSC assembly of the 
			<em>C. elegans</em> genome.
			<P>
			The coordinates of your favorite gene in one assembly may 
			not be the same as those in the next release of the 
			assembly unless the gene happens to lie on a completely 
			sequenced and unrevised chromosome. For information on 
			integrating data from one assembly into another, see the 
			<A HREF="#release8">Converting positions between
			assembly versions</A> section. 
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release8"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Converting positions between assembly versions</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"I've been researching a specific area of the human genome
			on the current assembly, and now you've just released a 
			new version. Is there an easy way to locate 
			my area of interest on the new assembly?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			See the section on <A HREF="FAQdownloads.html#download28">converting coordinates</A> for information on assembly migration tools.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release9"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Missing annotation tracks</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Why is my favorite annotation track missing from your latest release?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			The initial release of a new genome assembly typically contains a small subset
			of core annotation tracks. New tracks are added as they are generated. In many
			cases, our annotation tracks are contributed by scientists not affiliated with
			UCSC who must first obtain the sequence, repeatmasked data, etc. before they
			can produce their tracks. If you have need of an annotation that has not appeared
			on an assembly within a month or so of its release, feel free to send an inquiry
			to <A HREF="mailto:&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;">&#103;&#101;&#110;&#111;&#109;e&#64;&#115;&#111;&#101;.&#117;&#99;sc.e&#100;&#117;</A>.
<!-- above address is genome at soe.ucsc.edu -->
			<b><font color="red">Messages sent to this address will be posted to the 
			moderated genome mailing list, which is archived on a SEARCHABLE PUBLIC 
		    	<a HREF="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome">
			Google Groups forum</a></font></b>.
			</TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release10"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; What next with the human genome?</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"Now that the human genome is &quot;finished&quot;, will there be any more releases?" 
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Rest assured that work will continue. There will be updates to the assembly over the next 
			several years. This has been the case for all other finished (i.e. essentially complete) genome
			assemblies as gaps are closed. For example, the <em>C. elegans</em> genome has been 
			&quot;finished&quot; for several years, but small bits of sequence are still being 
			added and corrections are being made. NCBI will continue to coordinate the human
			genome assemblies in collaboration with the individual chromosome coordinators, and 
			UCSC will continue to QC the assembly in conjunction with NCBI (and, to a lesser extent,
			Ensembl). UCSC, NCBI, Ensembl, and others will display the new releases on their
			sites as they become available.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>
	
<!--Content Table========================================================= -->
<A NAME="release11"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; Mouse strain used for  mouse genome sequence</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What strain of mouse was used for the Mus musculus genome?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			C57BL/6J.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
<BR>

<!--Content Table========================================================= -->
<A NAME="release13"></A>
    <!--outer table is for border purposes-->        
    <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
        <TR><TD>            
        <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
	    <TR><TD>                
  	    <TABLE BGCOLOR="#D9E4F8" WIDTH="100%">
	        <TR><TD>                
	        <!--cheap trick to get background on row to be continuous-->
	   	<TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
		    <TR><TD ALIGN="left">
	   	    <FONT SIZE="4"><b>&nbsp; UniProt (Swiss-Prot/TrEMBL) display changes</b></FONT></TD> 
		    <TD ALIGN="right"><A HREF="#TOP"><IMG SRC="../images/index.gif" ALT="" ALIGN="right" BORDER=0></A>
	            </TD></TR>
  	        </TABLE>                
	        </TD></TR>
  	    </TABLE>                
   	    <TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0>
	        <TR><TH HEIGHT=10></TH></TR>
                <TR><TD WIDTH=10>&nbsp;</TD>
		    <TD>
	            <P>
			<B><FONT COLOR="#006666">Question: </FONT></B><BR>
			"What has UCSC done to accommodate the changes to 
			 display IDs recently introduced by UniProt (aka
		 	 Swiss-Prot/TrEMBL)?"
			<P>
			<B><FONT COLOR="#006666">Response:</FONT></B><BR> 
			Here is a detailed description of the database changes
			we have made to accommodate the UniProt changes. If 
			you are using the <em>proteinID</em> field in our 
			knownGene 
			table or the Swiss-Prot/TrEMBL display ID for indexing
			or cross-referencing other data, we strongly suggest 
			you transition to the UniProt accession number. 
		  	These changes will also affect anyone who is 
			mirroring our site.
			<OL>
			<LI>
			The latest UniProt Knowledgebase (Release 46.0, 
			Feb. 1st, 2005) was parsed and the results were 
			stored in a newly created database <em>sp050201</em>.
			<LI>
			A corresponding database, <em>proteins050201</em>, 
			was constructed based on data in <em>sp050201</em> 
			and other protein data sources.
			<LI>
			Two new symbolic database pointers, <em>uniProt</em> 
			and <em>proteome</em>, have been created to point to 
			the two new databases mentioned above. Some parts of 
			our programs use the data in these two 
			DBs.<PRE><TT>   uniProt  ---> sp050201
   proteome ---> proteins050201</TT></PRE>
			<LI>
			The existing protein symbolic database pointers, 
			<em>swissProt</em> and <em>proteins</em> remain 
			unchanged. Some parts of our programs still use these 
			two pointers and the data in their associated protein 
			databases.<PRE><TT>   swissProt ---> sp041115
   proteins  ---> proteins041115</TT></PRE>
			<LI>
			Two new tables, <em>spOldNew</em> and 
			<em>uniProtAlias</em>, have been added to the proteome
			database.<BR><BR>
    			The <em>spOldNew</em> table contains three columns:
			<UL>
			<LI><em>acc</em>  -- primary accession number
    			<LI><em>oldDisplayId</em> -- old display ID
    			<LI><em>newDisplayId</em> -- new display ID
			</UL>
			<BR>
    			The uniProtAlias table contains four columns:
			<UL>
    			<LI><em>acc</em> -- UniProt accession number
    			<LI><em>alias</em> -- alias (could be acc, old and new
			display IDs, etc.)
    			<LI><em>aliasSrc</em> -- source of the alias type
    			<LI><em>aliasSrcDate</em> -- date of the source data
			</UL>
			<P>
    			The aliases include primary accessions, secondary 
			accessions new display IDs, old display IDs, and old 
			display IDs corresponding to new secondary accessions.
			</P>
			<LI>
			Three new functions have been added to 
			<em>kent/src/hg/spDb.c</em>: <PRE><TT>   char *oldSpDisplayId(char *newSpDisplayId);
   /* Convert from new Swiss-Prot display ID to old display ID */
      
   char *newSpDisplayId(char *oldSpDisplayId); 
   /* Convert from old Swiss-Prot display ID to new display ID */

   char *uniProtFindPrimAcc(char *id);
   /* Return primary accession given an alias. */</TT></PRE>
			<P>
   			The <em>uniProtFindPrimAcc()</em> function is enabled 
			by the new <em>uniProtAlias</em> table.</P>
			</OL>
			<P>
			We anticipate additional changes down the road and 
			may eventually merge the two sets of protein DB 
			pointers into one set.
			<P>
			Currently, the <em>proteinID</em> field of the 
			knownGene table 
			for existing genome releases (hg15, hg16, 
			hg17, mm3, mm4, mm5, rn2, and rn3) uses old 
			Swiss-Prot/TrEMBL display IDs (pre-1 Feb. '05). In 
			the future, we may change this field to show the 
			UniProt accession number. Should we choose not to 
			change the content of the <em>proteinID</em> field, 
			we may consider adding a new field, 
			<em>uniProtAcc</em>.
			<P>
			If you have any questions about these changes and 
			their impact on your work, please email us at 
			<A HREF="mailto:genome@soe.ucsc.edu">genome@soe.ucsc.edu</A>. 
			Mirror sites may send questions to 
			<A HREF="mailto:genome-mirror@soe.ucsc.edu">genome-mirror@soe.ucsc.edu</A>.
			<b><font color="red">Messages sent to these addresses will be posted to the 
			moderated mailing lists, which are archived on SEARCHABLE PUBLIC 
                            <a HREF="https://groups.google.com/a/soe.ucsc.edu/forum/#!forum/genome-mirror">
                        Google Groups forum</a></font></b>.
		    </TD><TD WIDTH=15></TD>
		</TR>
	    </TABLE>
	    <br></TD></TR>
	</TABLE>
	</TD></TR>
    </TABLE>
    </TD><TD WIDTH=15></TD></TR>

</TABLE>
</BODY></HTML>

