<!-- Display image in righthand corner -->
<TABLE ALIGN=RIGHT BORDER=0 WIDTH=223 HEIGHT=185>
  <TR><TD ALIGN=RIGHT><A HREF="http://www.ncbi.nlm.nih.gov/genome/52" 
    TARGET=_blank>
    <IMG SRC="/images/Mus_musculus.jpg" WIDTH=208 HEIGHT=160 ALT="Mouse"></A>
  </TD></TR>
  <TR><TD ALIGN=RIGHT> 
    <FONT SIZE=-1> <em>Mus musculus</em><BR>
    </FONT>
    <FONT SIZE=-2> (Photo courtesy of
      <A HREF="http://www.jax.org/" TARGET=_blank>The Jackson Laboratory</A>)
    </FONT>
  </TD></TR>
</TABLE>

<P>
The Dec. 2011 <em>Mus musculus</em> assembly 
(Genome Reference Consortium Mouse Build 38 (GCA_000001635.2))
was produced by the <A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/"
TARGET=_blank>Mouse Genome Reference Consortium</A>.
For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/327618/" TARGET="_blank">
GRCm38</A> in the NCBI Assembly database.

<H3>Sample position queries</H3>
<P>
A genome position can be specified by the accession number of a
sequenced genomic region, an mRNA or EST, a chromosomal coordinate
range, or keywords from the GenBank description of an mRNA. The
following list shows examples of valid position queries for the
Mouse genome.  See the
<A HREF="../goldenPath/help/hgTracksHelp.html"
    TARGET=_blank>User's Guide</A> for more information.
<P>
<TABLE  border=0 CELLPADDING=0 CELLSPACING=0>
    <TR><TD VALIGN=Top NOWRAP><B>Request:</B><BR></TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top><B>Genome Browser Response:</B><BR></TD>
    </TR>
	
    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD VALIGN=Top NOWRAP>chr16</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays all of chromosome 16</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>chr16:1-5000000</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays first 5 million bases of chr 16</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>chr16:1000000+2000</TD>
       	<TD WIDTH=14></TD>
       	<TD VALIGN=Top>Displays a region of chr 16 that spans 2000 bases, starting with position 1000000 </TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>GL456161.2</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region of contig GL456161.2</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>BC012683</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays alignment location of mRNA with GenBank accession BC012683</TD>
    </TR>
    <TR><TD VALIGN=Top NOWRAP>AW045217</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays alignment location of EST with GenBank accession AW045217</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Ncam2</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Displays region of genome with official MGI mouse genetic nomenclature Ncam2</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>

    <TR><TD VALIGN=Top NOWRAP>pseudogene mRNA</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists transcribed pseudogenes but not cDNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists many zinc finger mRNAs</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>kruppel zinc finger</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists only kruppel-like zinc fingers</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>huntington</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists candidate genes associated with Huntington's disease</TD>
    </TR>

    <TR><TD VALIGN=Top NOWRAP>Smith,D.</TD>
	<TD WIDTH=14>&nbsp;</TD>
	<TD VALIGN=Top>Lists mRNAs deposited by co-author D. Smith</TD>
    </TR>

    <TR><TD VALIGN=Top><BR></TD></TR>
    <TR><TD COLSPAN="3"> Use this last format for author queries. Although GenBank 
			requires the search format <em>Smith D</em>, internally it uses 
			the format <em>Smith,D.</em>.</TD>
    </TR>

</TABLE>

<HR>
<H3>Assembly details</H3>
<P>
Genome Reference Consortium GRCm38, which includes approximately 2.6 Gb of sequence, is considered 
to be &quot;essentially complete&quot;. The assembly includes 
chromosomes 1-19, X, Y, M (mitochondrial DNA) and chr*_random (unlocalized)
and chrUn_* (unplaced clone contigs).
For information about the process used to assemble this version, see the 
<A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/mouse/" 
TARGET=_blank>GRC website</A>. For a glossary of assembly-related
terms, please see the GRC
<A HREF="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/info/definitions.shtml"
TARGET="_blank">Assembly Terminology</A> page.
Note that the UCSC mm10 database contains only the reference strain C57BL/6J.</P>

<P>
Bulk downloads of the sequence and annotation data are available via 
the Genome Browser <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/mm10/">FTP
server</A> or the <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#mouse">Downloads</A> page. 
We recommend that you use <A HREF="../goldenPath/help/ftp.html">rsync</A> for downloading
large or multiple files. 
</P>

<P>
The Mouse browser annotation tracks were generated by UCSC and 
collaborators worldwide. See the 
<A HREF="/goldenPath/credits.html#mouse_credits">Credits</A> 
page for a detailed list of the organizations and individuals who contributed 
to the success of this release. 
</P>
