<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
	<p>
        <font face="courier" size="3"><b>24 October 2013 -
        Job Opening: UCSC Genome Browser Trainer</b></font>
        <p>
	The <a href="http://cbse.ucsc.edu/" target="_blank">Center for Biomolecular Science
	and Engineering</a> (CBSE) at University of California Santa Cruz
	seeks an articulate, self-motivated educator for the two-year position of UCSC Genome
	Browser trainer. The trainer develops curriculum and provides in-person training on the UCSC
	Genome Browser at universities, hospitals, institutes, and professional
	meetings in the United States and internationally. Typical audiences include graduate 
	and post-graduate biologists and doctors, with Genome Browser experience ranging from novice
	users to bioinformatics specialists. Presentations include formal talks, problem-solving 
	sessions, and two-day workshops. </p>
 	<p>
	This position requires a Master's degree in a biological science, depth in molecular
	biology, experience in a research environment, working knowledge of the UCSC Genome Browser,
	understanding of its role in research methodology, and experience teaching or training in a
	scientific environment. Preferred qualifications include a PhD in a relevant field,
	experience with video production, and experience with HTML or web content management
	systems. </p>
	<p>
	For more information and to apply for this position, see
	<a href="http://bit.ly/browserJob" target="_blank">Job #1304619</a> on the UCSC Staff 
	Employment website.  </p>
	
	<hr>

	<p>
        <font face="courier" size="3"><b>23 October 2013 -
        dbSNP Build 138 Available for hg19</b></font>
        <p>
	We are pleased to announce the release of four tracks derived from NCBI
 	<a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 138
 	data, available on the human assembly (GRCh37/hg19). 
	The new tracks contain additional annotation data not included in previous dbSNP 
	tracks, with corresponding coloring and filtering options in the Genome Browser.
	</p>
	<p>
	As was the case for the annotations based on the previous dbSNP build 137, there are 
	four tracks in this release.  One is a track 
	containing all mappings of reference SNPs to the human assembly, labeled 
	"All SNPs (138)". The other three tracks are subsets of this track and show 
	interesting and easily defined subsets of dbSNP:
	</p>
	<ul>
	<li>Common SNPs (138): uniquely mapped variants that appear in at least 1% 
	of the population or are 100% non-reference
	<li>Flagged SNPs (138): uniquely mapped variants, excluding Common SNPs, 
	that have been flagged by dbSNP as "clinically associated"
	<li>Mult. SNPs (138): variants that have been mapped to more than one genomic location
	</ul>
	<p>
	By default, only the Common SNPs (138) are visible; other tracks must be 
	made visible using the track controls.
	</p>
	<p>
	You will find the four SNPs (138) tracks on the Human Feb. 2009 (GRCh37/hg19) 
	browser in the "Variation and Repeats" group.
	</p>
	<p>
	The tracks were produced at UCSC by Angie Hinrichs and Luvina Guruvadoo. We'd like
	to thank the dbSNP group at NCBI for providing access to these data.
        </p>

        <!-- staged for archive use
	<p>
        <font face="courier" size="3"><b>24 October 2013 -
        Job Opening: UCSC Genome Browser Trainer</b></font>
        <p>
	The <a href="http://cbse.ucsc.edu/" target="_blank">Center for Biomolecular Science
	and Engineering</a> (CBSE) at University of California Santa Cruz
	seeks an articulate, self-motivated educator for the two-year position of UCSC Genome
	Browser trainer. 
        <a href="goldenPath/newsarch.html#102413">Read more</a>.
        </p>
        
        <p>
        <font face="courier" size="3"><b>23 October 2013 -
        dbSNP 138 Available for hg19: </b></font>
	We are pleased to announce the release of four tracks derived from NCBI
 	<a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 138
 	data, available on the human assembly (GRCh37/hg19). 
        <a href="goldenPath/newsarch.html#102313">Read more</a>.
        </p>

        -->

        <!-- start archives -->
	<hr>
	<p>
        <font face="courier" size="3"><b>11 October 2013 -
        Updated DGV Structural Variation Track: </b></font>
        The UCSC Genome Browser team is pleased to announce an updated DGV Structural Variation
        track to reflect the official launch of the new <a href="http://dgv.tcag.ca/dgv/app/home"
        target="_blank">Database of Genomic Variants</a>.
        <a href="goldenPath/newsarch.html#101113">Read more</a>.
        </p>
        
	<p>
        <font face="courier" size="3"><b>26 September 2013 -
        New American Alligator (allMis1) Assembly Now Available in the Genome
        Browser: </b></font>
        A Genome Browser is now available for the American alligator
        (<em>Alligator mississippiensis</em>) assembly released
        August 2012 by the International Crocodilian Genomes Working Group
        (version allMis0.2, UCSC version allMis1).
        <a href="goldenPath/newsarch.html#092613">Read more</a>.
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
