<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        <p>
        <font face="courier" size="3"><b>27 November 2013 -
        100 Species Conservation Track now available on hg19</b></font>
        </p>
        <p>
        After 15.4 years of CPU run-time in 9,905,594 individual 'jobs' and 99 cluster runs for lastz 
        pair-wise alignment...we are excited to announce the release of a 100 species alignment on the 
        hg19/GRCh37 human Genome Browser.
        </p>
        <p>
        This new Conservation track shows multiple alignments of 100 species and measurements of 
        evolutionary conservation using two methods (phastCons and phyloP) from the PHAST package. 
        This adds 40 more species to the existing 60way on the mm10 mouse browser. For more information 
        about the 100 species Conservation track, see its description page.
        </p>
        <p>
        We'd also like to acknowledge the hard work of the UCSC Genome Browser staff who pulled together 
        the information for this track: Hiram Clawson and Pauline Fujita.
        </p>
        <hr>

	<p>
        <font face="courier" size="3"><b>24 October 2013 -
        Job Opening: UCSC Genome Browser Trainer</b></font>
        <p>
	The <a href="http://cbse.ucsc.edu/" target="_blank">Center for Biomolecular Science
	and Engineering</a> (CBSE) at University of California Santa Cruz
	seeks an articulate, self-motivated educator for the two-year position of UCSC Genome
	Browser trainer. The trainer develops curriculum and provides in-person training on the UCSC
	Genome Browser at universities, hospitals, institutes, and professional
	meetings in the United States and internationally. Typical audiences include graduate 
	and post-graduate biologists and doctors, with Genome Browser experience ranging from novice
	users to bioinformatics specialists. Presentations include formal talks, problem-solving 
	sessions, and two-day workshops. </p>
 	<p>
	This position requires a Master's degree in a biological science, depth in molecular
	biology, experience in a research environment, working knowledge of the UCSC Genome Browser,
	understanding of its role in research methodology, and experience teaching or training in a
	scientific environment. Preferred qualifications include a PhD in a relevant field,
	experience with video production, and experience with HTML or web content management
	systems. </p>
	<p>
	For more information and to apply for this position, see
	<a href="http://bit.ly/browserJob" target="_blank">Job #1304619</a> on the UCSC Staff 
	Employment website.  </p>
	
        <!-- staged for archive use
        <p>
        <font face="courier" size="3"><b>27 November 2013 -
        100 Species Conservation Track now available on hg19</b></font>
        </p>
       
        <p>
        We're happy to announce the release of a new track on the hg19/GRCh37 human Genome Browser. 
        The new Conservation track shows multiple alignments of 100 species and measurements of 
        evolutionary conservation using two methods (phastCons and phyloP) from the PHAST package. 
        </p> 

	<p>
        <font face="courier" size="3"><b>24 October 2013 -
        Job Opening: UCSC Genome Browser Trainer</b></font>
        <p>
	The <a href="http://cbse.ucsc.edu/" target="_blank">Center for Biomolecular Science
	and Engineering</a> (CBSE) at University of California Santa Cruz
	seeks an articulate, self-motivated educator for the two-year position of UCSC Genome
	Browser trainer. 
        <a href="goldenPath/newsarch.html#102413">Read more</a>.
        </p>
        -->

        <!-- start archives -->
        <hr>
 
        <p>
        <font face="courier" size="3"><b>23 October 2013 -
        dbSNP 138 Available for hg19: </b></font>
	We are pleased to announce the release of four tracks derived from NCBI
 	<a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 138
 	data, available on the human assembly (GRCh37/hg19). 
        <a href="goldenPath/newsarch.html#102313">Read more</a>.
        </p>

	<p>
        <font face="courier" size="3"><b>11 October 2013 -
        Updated DGV Structural Variation Track: </b></font>
        The UCSC Genome Browser team is pleased to announce an updated DGV Structural Variation
        track to reflect the official launch of the new <a href="http://dgv.tcag.ca/dgv/app/home"
        target="_blank">Database of Genomic Variants</a>.
        <a href="goldenPath/newsarch.html#101113">Read more</a>.
        </p>
        
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
