<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        <p>
        <font face="courier" size="3"><b>22 August 2013
        - Three New Human Variation Tracks Released</b></font>
        </p>
        <p>
        We are pleased to announce the release of three tracks that describe
        human genetic variation. They are based on the
        <a href="http://lovd.nl/2.0/index_list.php" target="_blank">Leiden Open
        Variation Database (LOVD)</a>, the
        <a href="http://www.hgmd.cf.ac.uk/ac/index.php" target="_blank">Human Gene
        Mutation Database Public (HGMD)</a>, and amino acid mutations in the
        <a href="http://www.uniprot.org/" target="_blank">UniProt database</a>. The
        majority of the data were collected from publications by curators in the three
        databases.
        </p>
        <p>
        The three databases have varying license restrictions. UniProt provides complete
        information about the mutation amino acid change, the disease and a link to the
        publications that mention it. LOVD provides the mutation and a link to a website
        with disease and publication details. HGMD provides some coordinates and a link to
        its own website which requires registration.
        </p>
        <p>
        These tracks are intended for use primarily by physicians and other professionals
        concerned with genetic disorders, by genetics researchers, and by advanced students
        in science and medicine. While the Genome Browser database is open to the public, users
        seeking information about a personal medical or genetic condition are urged to consult
        with a qualified physician for diagnosis and for answers to personal questions.
        </p>
        <p>
        We'd like to thank all three databases for their help in putting the data together. The
        tracks were produced at UCSC by Maximilian Haeussler and Steve Heitner.
        </p>
        <p>
        <font face="courier" size="3"><b>22 July 2013
	- Variant Annotation Integrator</b></font>
        </p>
	<p>
        In order to assist researchers in annotating and prioritizing thousands
        of variant calls from sequencing projects, we have developed the Variant
        Annotation Integrator (VAI). Given a set of variants uploaded as a
        custom track (in either <a href="FAQ/FAQformat.html#format10">pgSnp</a>
        or <a href="goldenPath/help/vcf.html">VCF</a> format), the VAI will
        return the predicted functional effect (e.g., synonymous, missense,
        frameshift, intronic) for each variant. The VAI can optionally add
        several other types of relevant information, including: the
        <a href="https://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a>
        identifier if the variant is found in dbSNP, protein damage scores for
        missense variants from the <a href="https://sites.google.com/site/jpopgen/dbNSFP"
        target="_blank">Database of Non-synonymous Functional Predictions (dbNSFP)</a>,
        and conservation scores computed from multi-species alignments. The VAI
        also offers filters to help narrow down results to the most interesting
        variants.
        </p>
	<p>
        Future releases of the VAI will include more input/upload options,
        output formats, and annotation options, and a way to add information
        from any track in the Genome Browser, including custom tracks.
        </p>
	<p>
        There are two ways to navigate to the VAI: (1) From the &quot;Tools&quot;
        menu, follow the &quot;Variant Annotation Integrator&quot; link. (2)
        After uploading a custom track, hit the &quot;go to variant annotation
        integrator&quot; button. The user's guide is at the bottom of the page,
        under &quot;Using the Variant Annotation Integrator.&quot;
        </p>
	<p>
        Thank you to Angie Hinrichs, Bob Kuhn, Brian Raney, Brooke Rhead, and
        Jonathan Casper for their input and work on this feature.
        </p>

        <!-- start archives -->
	<hr>
        <p>

        <!-- staged for archive use
        <p>
        <font face="courier" size="3"><b>22 August 2013
        - Three New Human Variation Tracks Released:</b></font> We are
        pleased to announce the release of three tracks that describe human
        genetic variation.
        <a href="goldenPath/newsarch.html#082213">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>22 July 2013
        - Variant Annotation Integrator:</b></font> In order to assist
        researchers in annotating and prioritizing thousands of variant
        calls from sequencing projects, we have developed the Variant
        Annotation Integrator (VAI).
        <a href="goldenPath/newsarch.html#072213">Read more</a>.
        </p>
        -->

        <p>
        <font face="courier" size="3"><b>2 July 2013
        - Assembly Data Hubs:</b></font> Assembly Data Hubs extend the
        functionality of Track Data Hubs to assemblies that are not hosted
        natively on the Browser.
        <a href="goldenPath/newsarch.html#070213">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>2 July 2013
        - New Gene Alleles feature on the details pages of UCSC genes:</b></font>
        For each protein-coding gene in the UCSC Genes track, variant data from
        the 2,184 (per autosome) phased chromosomes from Phase 1 of the 1000
        Genomes Project have been distilled into &quot;gene haplotype
        alleles.&quot;
        <a href="goldenPath/newsarch.html#070213b">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>28 June 2013
        - New UCSC Genes Track Released for GRCh37/hg19:</b></font> We're happy
        to announce the release of an updated UCSC Genes track for the
        GRCh37/hg19 human Genome Browser.
        <a href="goldenPath/newsarch.html#062813">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>27 June 2013
        - New Official European Mirror Released: </b></font> Announcing
        an official mirror for European users, with automatic
        redirection.
        <a href="goldenPath/newsarch.html#062713">Read more</a>.
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
