<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->

  <hr>
  <p>
  <font color="#006666" size="4"><b>14 September 2015 &mdash;
  Human Genome Browser default changed to GRCh38/hg38 </b></font>
  </p>
  <p>
  In conjunction with the release of the new 100-species Conservation track on the 
  hg38/GRCh38 human assembly, we have changed the default human browser on our website from
  hg19 to hg38. In most cases, if you are currently using the hg19 (or older) browser, the
  Genome Browser will continue to display that assembly for you when you start it up.
  However, there are circumstances in which the assembly may switch to the newer version;
  for instance, if you reset your browser defaults. If you find yourself in a situation
  where some of your favorite browser tracks have &quot;disappeared&quot;, you may want to check 
  that you're viewing the right assembly. 
  </p>
  <p>
  In upcoming months, we will continue our efforts to expand the annotation track set
  on the hg38 browser to include many of the tracks present on previous human assemblies. In cases
  where it makes sense, data may be simply &quot;lifted&quot; from hg19 using migration
  tools. However, in many instances, we must rely on our data providers to generate new versions of 
  their data on the latest assembly. We'll publish these data sets as they become available.
  </p>
  <p>
  For a summary of the new features in the GRCh38 assembly, see the 
  <a href="goldenPath/newsarch.html#030614">overview</a> we published in March 2014.
  </p>
  <hr>


  <p>
  <font color="#006666" size="4"><b>10 September 2015 &mdash;
  100 Species Conservation Track now available on GRCh38/hg38 </b></font>
  </p>
  <p>
  We are pleased to announce the release of a new Conservation track based on the human 
  (GRCh38/hg38) assembly. This track shows multiple alignments of 100 vertebrate species and
  measurements of evolutionary conservation using two methods (phastCons and phyloP) from the
  PHAST package for all species. The multiple alignments were generated using multiz and other
  tools in the UCSC/Penn State Bioinformatics comparative genomics alignment pipeline.
  Conserved elements identified by phastCons are also displayed in this track. For more
  details, visit the 
  <a href="../cgi-bin/hgTrackUi?db=hg38&g=cons100way">track description page</a>.
  </p>
  <p>
  We would like to thank Hiram Clawson and Jonathan Casper for their efforts in creating this track.
  </p>
	<hr>

        <!-- start archives -->
<!--
  	<p>
  	<font color="#006666" size="4"><b>14 September 2015 &mdash;
  	Human Genome Browser default changed to GRCh38/hg38 </b>:</font>
  	In conjunction with the release of the new 100-species Conservation track on the 
  	hg38/GRCh38 human assembly, we have changed the default human browser on our website from
  	hg19 to hg38. 
        <a href="goldenPath/newsarch.html#091415">Read more</a>.
  	</p>
  	<hr>
  	<p>
    	<font color="#006666" size="4"><b>10 September 2015 &mdash;
    	100 Species Conservation Track now available on GRCh38/hg38 </b>:
    	</font>
    	We are pleased to announce the release of a new Conservation track based on the human
   	(GRCh38/hg38) assembly.
    	<a href="goldenPath/newsarch.html#091015">Read more</a>.
  	</p>
-->

	<p>
        <font color="#006666" size="4"><b>12 August 2015 &mdash;
       	New blog post:  How to share your UCSC screenthoughts</b>:
        </font>
        We talk about saving static images and dynamic Browser sessions in a new
        <A HREF = "">blog</A> post,
        and discuss some of the advantages and pitfalls of each.
        <a href="goldenPath/newsarch.html#081215">Read more</a>.
	</p>

        <p>
        <font color="#006666" size="4"><b>29 June 2015 &mdash;
        GENCODE Genes Now Default Gene Set on the Human (GRCh38/hg38) Assembly</b>:
        </font>
        In a move towards standardizing on a common gene set within the bioinformatics 
        community, UCSC has made the decision to adopt the GENCODE set of gene models 
        as our default gene set on the human genome assembly.
        <a href="goldenPath/newsarch.html#062915">Read more</a>.
        </p>
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
