<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            <!-- start news -->
		<font face="courier" size="3"><b>16 August 2012 -
Announcing a Genome Browser for the Medium ground finch </b></font>
		<p>
		We have released a browser for the Medium ground finch, 
		<em>Geospiza fortis</em>, reknowned as one of naturalist Charles
		Darwin's Galapagos finches. This species, which has been the 
		subject of many evolutionary studies, is one of a group of 
		birds that evolved over a few million years from a single 
		ancestral species into multiple species whose beak sizes and
		shapes are specialized for using different food resources. The 
		phenotypic diversity of these birds contributed to Darwin's 
		theory of evolution. The significance of this genome assembly is
		described in the August 16, 2012, 
		<a href="https://cbse.soe.ucsc.edu/news/article/1939" 
		target=_blank>press release</a> issued by the UCSC Center for 
		Biomolecular Science and Engineering (CBSE). 
		</p>
		<p>
		The initial Medium ground finch genome assembly (GeoFor_1.0, 
		UCSC version geoFor1) is the  product of a collaboration between
		the Genome 10K project and Beijing Genomics Institute (BGI) to 
		sequence 100 vertebrate species, and is the first to be 
		released in the UCSC Genome Browser. For more information about 
		the <em>G. fortis</em> genome, see the 
		<a href="http://www.ncbi.nlm.nih.gov/genome/13302"
		target=_blank>NCBI website</a>. 
		</p>
		<p>
		Bulk downloads of the sequence and annotation data are available
		via the Genome Browser 
		<a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/geoFor1/">FTP server</a> or the 
		<a href="http://hgdownload.cse.ucsc.edu/downloads.html#mediumgroundfinch">Downloads</a> page. 
		The browser annotation tracks were generated by UCSC and 
		collaborators worldwide. See the 
		<a href="/goldenPath/credits.html#mediumgroundfinch_credits">Credits</a> 
		page for a detailed list of the organizations and individuals 
		who contributed to this release. We'd like to thank BGI for
		contributing the data for this assembly and acknowledge the
		UCSC staff members who released this browser: Hiram Clawson and 
		Greg Roe.
		</p>
		<HR>

<A NAME="070612"></A>
		<font face="courier" size="3"><b>14 August 2012 - Changes to our website</b></font>
		<p>
		We are pleased to announce a few changes to the look and feel of the Genome
		Browser website. In addition to a new font throughout the site, the first thing you might notice is
		a new menu bar across the top of many of the pages. We have reorganized the navigation on our web
		applications with you in mind; we hope you find it more intuitive!</p>
		<p>
		Additionally, we greatly simplified the search mechanism by 
		collapsing all search methods into a single box.</p>
		<a href="http://genome.ucsc.edu/images/autoselect.jpg" style="border:none; display: block;
		width:700px; height: 129px; margin:0 auto;"><img src="http://genome.ucsc.edu/images/autoselect.jpg"
		style="width:700px; height: 129px; border: 1px solid #000;"></a>

		<p>
		In this box, you can search for chromosomal coordinate ranges, gene names, accession numbers,
		keywords, etc. For a complete list of ways to query the browser data, see 
		<a href="http://genome.ucsc.edu/goldenPath/help/query.html">Querying the Genome Browser</a>.</p>

		<p>
		As always, if you have questions about the website or the data, please 
		<a href="http://genome.ucsc.edu/contacts.html">Contact Us</a>.</p>
                
                <hr>

	   <!-- start archives -->
        	<p>        
           	<font face="courier" size="3"><b>16 July 2012 - Tasmanian Devil
		assembly now available in Genome Browser: </b></font>
		We are pleased to announce the release of a Genome Browser 
		for the February 2011 Tasmanian devil assembly, <em>Sarcophilus harrisii</em> 
		(WTSI version Devil_refv7.0, UCSC version sarHar1). 
		<A HREF="goldenPath/newsarch.html#071612">Read more</A>.

		<P>
                <FONT FACE="courier" SIZE="3"><B>06 July 2012 - Retirement of 
		hgText CGI: </B></FONT>
		We are announcing the imminent retirement of the hgText
                CGI.  
		<A HREF="goldenPath/newsarch.html#070612">Read more</A>.
		</P>
		<P>
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
