<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            <!-- start news -->
                <p> 
                <font face="courier" size="3"><b>11 February 2013 - 
                Denisova tracks released on hg19 browser</b></font>
                <br>
		In conjunction with the publication of the paper by Meyer <em>et al.</em>
		<a href="http://www.ncbi.nlm.nih.gov/pubmed/22936568">
                A High-Coverage Genome Sequence from an Archaic Denisovan Individual</a>
                the UCSC Genome Browser is hosting a set of new tracks. Briefly, Meyer <em>et al.</em>
		applied a novel single-stranded DNA library preparation method to DNA previously 
		extracted from 40mg of a phalanx bone excavated from Denisova Cave in the 
		Altai Mountains in southern Siberia.
		</p>
		<p>
		Two supporting datasets and one analysis dataset are now available in the Genome Browser:
		</p>
		<ul>
		<li>Modern Human Variants: variant calls made from sequence reads of eleven individuals 
		mapped to the human genome, used in statistical analyses to put the divergence of the 
		Denisova genome into perspective with regard to present-day humans.
                <li>Denisova Sequence: high-coverage (30X) sequence reads from an archaic Denisovan 
		individual mapped to the human genome reference assembly.
                <li>Modern Human Derived, Denisova Ancestral: mutations in the modern human lineage 
		that rose to fixation or near fixation since the split from the last common 
		ancestor with Denisovans, along with predicted functional effects from Ensembl's 
		Variant Effect Predictor (VEP).
		</ul>
                <p>
		We'd like to thank Richard Green and the many authors of the Science paper 
		for providing the UCSC Genome Browser with these data. 
		We'd also like to acknowledge the hard work of the UCSC Genome Browser staff 
		who pulled together the annotations and information for these tracks:
		Angie Hinrichs and Pauline Fujita. 
	        </p>
                <hr>
                <p>
                <font face="courier" size="3"><b>25 January 2013 -
                Southern White Rhinoceros Genome Browser Release: </b></font> 
	        We are pleased to announce the release of a Genome Browser for the May 2012
                assembly of the Southern White Rhinoceros, <em>Ceratotherium simum simum</em> 
                (Broad Institute version cerSimSim1.0, UCSC version cerSim1). 
                <a href="goldenPath/newsarch.html#012513">Read more</a>.
                </p>
  
		<p>
                <font face="courier" size="3"><b>22 January 2013 -
                New Baboon (papAnu2) Assembly Now Available in the Genome Browser: </b></font>
                We are pleased to announce the release of a Genome Browser for the March 2012
                assembly of the Olive Baboon, <em>Papio anubis</em> (Baylor Panu_2.0, UCSC version 
		papAnu2).
		<a href="goldenPath/newsarch.html#012213">Read more</a>.
                </p>
		<p>
	        <font face="courier" size="3"><b>15 January 2013 -
                New Lamprey (petMar2) Assembly Now Available in the Genome Browser: </b></font>
                We are pleased to announce the release of a Genome Browser for the September 2010
                assembly of the Lamprey, <em>Petromyzon marinus</em> (WUGSC 7.0, UCSC version 
		petMar2). 
		<a href="goldenPath/newsarch.html#011513">Read more</a>.
                </p>

              

	    <!-- start archives -->
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
