<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        
	<hr>
	<p>
        <font face="courier" size="3"><b>06 March 2014 - 
	The new GRCh38 Human Genome Browser is here!</b></font>
	<p>
	In the final days of 2013, the 
	<a href="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/" 
	target="_blank">Genome Reference Consortium</a> (GRC) released the eagerly awaited GRCh38 
	human genome assembly, the first major revision of the human genome in more than four 
	years. During the past two months, the UCSC team has been 
	hard at work building a browser that will let our users explore the new
	assembly using their favorite Genome Browser features and tools. 
	Today we're announcing the release of a preliminary browser on the GRCh38
	assembly. Although we still have plenty of work ahead of us in constructing the rich
	feature set that our users have come to expect, this early release will allow you to
	take a peek at what's new. 
	<p>
	Starting with this release, the UCSC Genome Browser version numbers for human 
	assemblies will match those of the GRC to minimize version confusion. Hence, the GRCh38 
	assembly is referred to as hg38 in Genome Browser datasets and documentation. We've
	also made some slight changes to our chromosome naming scheme that affect primarily the 
	names of haplotype chromosomes, unplaced contigs and unlocalized contigs. For more details 
	about this, see the 
	<a href="../cgi-bin/hgGateway?db=hg38">hg38 gateway page</a>.  
	<p>
	<b>What's new in GRCh38?</b> 
	<ul>
	<li>
	<b>Alternate sequences</b> - Several human chromosomal regions exhibit sufficient 
	variability to prevent adequate representation by a single sequence. To address this, the 
	GRCh38 assembly provides alternate sequence for selected variant regions through the 
	inclusion of  <em>alternate loci scaffolds</em> (or <em>alt loci</em>). Alt loci are 
	separate accessioned sequences that are aligned to reference chromosomes. This assembly 
	contains 261 alt loci, many of which are associated with the LRC/KIR area of chr19 and the 
	MHC region on chr6. (See the 
	<a href="../cgi-bin/hgTracks?chromInfoPage=">sequences</a> page for a complete list of the 
	reference chromosomes and alternate sequences in GRCh38.)
	<li>
	<b>Centromere representation</b> - Debuting in this release, the large megabase-sized gaps 
	that were previously used to represent centromeric regions in human assemblies have been 
	replaced by sequences from centromere models created by 
	<a href="http://genome.cshlp.org/content/early/2014/02/05/gr.159624.113.abstract"
	target="_blank">Karen Miga et al.</a>, using centromere databases developed during her 
	work in the <a href="http://www.genome.duke.edu/labs/WillardLab/" 
	target="_blank">Willard lab</a> at 
	Duke University and analysis software developed while working in the 
	<a href="http://cbse.soe.ucsc.edu/people/kent" target="_blank">Kent lab</a> at UCSC.
	The models, which provide the 
	approximate repeat number and order for each centromere, will be useful for read mapping 
	and variation studies. 
	<li>
	<b>Mitochondrial genome</b> - The mitochondrial reference sequence included in the GRCh38 
	assembly and hg38 Genome Browser (termed &quot;chrM&quot; in the browser) is the 
	<a href="http://mitomap.org/bin/view.pl/MITOMAP/HumanMitoSeq" 
	target="_blank">Revised Cambridge Reference Sequence</a> (rCRS) from 
	<a href="http://mitomap.org/MITOMAP" target="_blank">MITOMAP</a> with GenBank accession 
	number J01415.2 and RefSeq accession number NC_012920.1. This differs from the chrM 
	sequence (RefSeq accession number NC_001907) used by the previous hg19 Genome Browser, 
	which was not updated when the GRCh37 assembly later transitioned to the new version.
	<li>
	<b>Sequence updates</b> - Several erroneous bases and misassembled regions in GRCh37 have 
	been corrected in the GRCh38 assembly, and more than 100 gaps have been filled or reduced. 
	Much of the data used to improve the reference sequence was obtained from other genome 
	sequencing and analysis projects, such as the <a href="http://www.1000genomes.org/" 
	target="_blank">1000 Genomes Project</a>.  
	<li>
	<b>Analysis set</b> - The GRCh38 assembly offers an &quot;analysis set&quot; that was 
	created to accommodate next generation sequencing read alignment pipelines. Several
	GRCh38 regions have been eliminated from this set to improve read mapping. 
	The analysis set may be downloaded from the Genome Browser 
	<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/analysisSet/"
	target="_blank">downloads page</a>.
	</ul>
	<p>
	For more information about the files included in the GRCh38 GenBank submission, 
	see the
	<a href="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh38/README"
	target="_blank">GRCh38 README</a>. The GRCh38 GenBank record provides a detailed
	array of <a href="http://www.ncbi.nlm.nih.gov/assembly/883148/#/st_Primary-Assembly"
	target="_blank">statistics about this assembly</a>.
        Bulk downloads of the sequence and annotation data may be obtained from the Genome
        Browser <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg38/">FTP server</a>
        or the
        <a href="http://hgdownload.cse.ucsc.edu/downloads.html#human">Downloads</a>
        page.  The annotation tracks for this browser were generated by UCSC and collaborators 
	worldwide. 
	</p>
	<p>
	<b>Much more to come!</b> This initial release of the hg38 Genome Browser provides a 
	rudimentary set of annotations.  Many of our annotations rely on data sets from external 
	contributors (such as our popular SNPs tracks) or require massive computational effort 
	(our comparative genomics tracks). In the upcoming months/years, we will release many more 
	annotation tracks as they become available. To stay abreast of new datasets, join our 
        <a href="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
	target="_blank">genome-announce</a> mailing list or follow us on
	<a href="http://www.twitter.com/GenomeBrowser" target="_blank">twitter</a>.
	</p>
	We'd like to thank our GRC and NCBI collaborators who worked closely with
	us in producing the hg38 browser. Their quick responses and helpful feedback were a
	key factor in expediting this release. The production of the hg38 Genome Browser was
	a team effort, but in particular we'd like to acknowledge the engineering efforts of Hiram 
	Clawson and Brian Raney, the QA work done by Steve Heitner, project guidance
	provided by Ann Zweig, Robert Kuhn, and Jim Kent, and documentation work by Donna
	Karolchik.
	See the <a href="goldenPath/credits.html#human_credits">Credits</a> page for a detailed
        list of the organizations and individuals who contributed to this release.
        </p>
	<p>
	<hr>
        <font face="courier" size="3"><b>04 March 2014 - 
	Introducing new Genome Browser highlight feature</b></font>
        </p>
        We are excited to announce a new highlight feature in the UCSC
        Genome Browser. Using drag-and-select, you can now highlight a
        region or gene of interest.
        <br><br>
        <img style="border: 2px solid black; margin-left:auto; 
                    margin-right:auto; display: block;" 
                    src="../images/highlight_screenshot.png"
                    alt="Highlight screenshot">
        <br>
        <p>
        To highlight a region: Click and hold the mouse button on one edge
        of the desired area to be highlighted in the Base Position track,
        drag the mouse right or left to highlight the selection area, then
        release the mouse button. Click the &quot;Highlight&quot; button on the
        &quot;drag-and-select&quot; popup. More details about this new feature
        can be found on our
        <a href="goldenPath/help/hgTracksHelp.html#Highlight" target="_blank">help</a> page.
	</p>
        <p>
        Credit goes to Tim Dreszer, Larry Meyer, Robert Kuhn and Luvina Guruvadoo
        for the design, development and testing of this feature. Additional
        testing was also provided by several members of the QA team.
	</p>
	<p>
        <hr>

        <!-- staged for archive use

        <p>
        <font face="courier" size="3"><b>04 March 2014 - 
        Introducing Genome Browser highlight feature</b></font>
        </p>
        We are excited to announce a new highlight feature in the UCSC
        Genome Browser. Using drag-and-select, you can now highlight a
        region or gene of interest.
 	<a href="goldenPath/newsarch.html#030414">Read more</a>.
	</p>

        -->
        <!-- start archives -->

        <font face="courier" size="3"><b>28 February 2014 -
        New!  Expanded onsite workshop program!: </b> </font>
        Explore the full power of the UCSC Genome Browser! 
        Thanks to the funding support of NHGRI, we can now offer hands-on Genome Browser 
        training onsite at your institution. 
        <a href="goldenPath/newsarch.html#022814">Read more</a>. 
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
