<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->

        <hr>
        <p>
        <font face="courier" size="3"><b>15 April 2014 -
        New Minke Whale (balAcu1) Assembly Now Available in the Genome Browser</b></font>
        <p>
        A Genome Browser is now available for the minke whale (Balaenoptera acutorostrata scammoni)
        assembly released October 2013 by the <A HREF="http://eng.kiost.ac/kordi_eng/main/" TARGET="_BLANK">
        Korea Ocean Research &amp; Development Institute</A> (KORDI version BalAcu1.0, UCSC version balAcu1).
        For more information and statistics about this assembly, see the NCBI assembly record for
        <A HREF="http://www.ncbi.nlm.nih.gov/assembly/78761" TARGET="_BLANK">BalAcu1.0</A>. There are
        10,776 scaffolds with a total size of 2,431,687,698 bases.
        </p>

        <p>
        Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
        <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/balAcu1/">FTP server</A> or the
        <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#minke_whale">Downloads</A> page. Please
        observe the <A HREF="goldenPath/credits.html#minke_whale_use">conditions for use</A> when accessing and
        using these data sets. The annotation tracks for this browser were generated by UCSC and
        collaborators worldwide. See the <A HREF="goldenPath/credits.html#minke_whale_credits">Credits</A>
        page for a detailed list of the organizations and individuals who contributed to this release.
        </p>

        <hr>
        <p>
        <font face="courier" size="3"><b>11 April 2014 -
        New UCSC Genes Track Released for GRCm38/mm10</b></font>
        <p>
        We're happy to announce the release of an updated UCSC Genes track for the GRCm38/mm10
        mouse Genome Browser.  The new release has 61,642 total transcripts, compared with 59,121
        in the previous version.  The total number of canonical genes has increased from 31,469
        to 32,408.  Comparing the new gene set with the previous version:
        <ul>
          <li>54,714 transcripts did not change between versions.</li>
          <li>19 transcripts were not carried forward to the new version.</li>
          <li>3,866 transcripts are &quot;compatible&quot; with those in the previous set, meaning
              that the two transcripts show consistent splicing.  In most cases, the old and new
              transcripts differ in the lengths of their UTRs.</li>
          <li>522 transcripts overlap with those in the previous set, but do not show consistent
              splicing (i.e., they contain overlapping introns with differing splice sites).</li>
        </ul>
        </p>
        <p>
        More details about the new UCSC Genes track can be found on the
        <a href="../cgi-bin/hgTrackUi?db=mm10&g=knownGene">UCSC Genes track
        description page</a>.
        </p>
	<hr>

        <!-- staged for archive use

        <p>
        <font face="courier" size="3"><b>15 April 2014 -
        New Minke Whale (balAcu1) Assembly Now Available in the
        Genome Browser</b></font>
        A Genome Browser is now available for the minke whale
        (Balaenoptera acutorostrata scammoni) assembly released 
        October 2013 by the Korea Ocean Research &amp; Development Institute.
        <a href="goldenPath/newsarch.html#041514">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>11 April 2014 -
        New UCSC Genes Track Released for GRCm38/mm10</b></font>
        </p>
        We're happy to announce the release of an updated UCSC Genes
        track for the GRCm38/mm10 mouse Genome Browser.
        <a href="goldenPath/newsarch.html#041114">Read more</a>.
        </p>

                -->
        <!-- start archives -->
	<p>
        <font face="courier" size="3"><b>06 March 2014 -
        The new GRCh38 Human Genome Browser is here!</b></font>
        In the final days of 2013, the Genome Reference Consortium (GRC)
        released the eagerly awaited GRCh38 human genome assembly, the
        first major revision of the human genome in more than four years.
        <a href="goldenPath/newsarch.html#030614">Read more</a>.
        </p>

        <p>
        <font face="courier" size="3"><b>04 March 2014 - 
        Introducing Genome Browser highlight feature:</b></font>
        We are excited to announce a new highlight feature in the UCSC
        Genome Browser. Using drag-and-select, you can now highlight a
        region or gene of interest.
 	<a href="goldenPath/newsarch.html#030414">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
