<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->

        <hr>
        <p>
        <font face="courier" size="3"><b> 29 Apr 2014 - 
        100-species Conservation track re-release</b></font>
        <p>
        We released the initial version of the 100-species Conservation track
        for the hg19 human assembly in November 2013. Over the past few months,
        we found a couple of inconsistencies and decided, for the integrity of
        the data, that we should rerun the computation pipeline and re-release
        the data. The fixes required minimal changes, and resulted in a more
        precise data set. We published the new version of the hg19 100-species
        Conservation track on 15 April 2014.
        </p>
        <p>
        Here are more details about the two fixes that we made:
        </p>
        <p>
        1. We tracked down the cause of some unusual gaps in the alignments to a 
        bug in the multiz program used in our alignment pipeline. The Penn State 
        Bioinformatics Group provided us with a fixed version of multiz that we 
        used to rerun the pipeline.
        </p>
        <p>
        2. We inadvertently used the pre-release assembly for one of the species 
        in the 100-way alignment: the panda. We replaced this assembly with the 
        correct version and verified that the data sets for all other species 
        were correct before rerunning the pipeline.
        </p>
        <hr>

        <p>
        <font face="courier" size="3"><b> 18 April 2014 - 
        New Motif Displays for Transcription Factor ChIP-seq Track and New Genome Segmentations from ENCODE</b></font>
        <p>
        The Browser's default displayed Transcription Factor ChIP-seq track is updating to the
        <a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTfbsClusteredV3">latest data release</a>,
        which has been enhanced with the display of Factorbook motifs.  Within a cluster, a green
        highlight indicates the highest scoring site of a Factorbook-identified canonical motif for the
        corresponding factor. Upon clicking a transcription factor's cluster with a motif, the details page
        now displays the motif's sequence logo, alignment and underlying Positional Weight Matrix. Also,
        the track configuration page now enables the filtering of factors.
        </p>
        <p>
        The newly added <a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgSegmentation">
        Genome Segmentations from ENCODE</a> tracks display multivariate genome-segmentation performed on
        <a target=ucscVocab href="http://genome.ucsc.edu/cgi-bin/hgEncodeVocab?term=GM12878,H1-hESC,K562,HeLa-S3,HepG2,HUVEC">six human cell types</a>
        (GM12878, K562, H1-hESC, HeLa-S3, HepG2, and HUVEC), integrating ChIP-seq data for
        <a target=ucscVocab href="http://genome.ucsc.edu/cgi-bin/hgEncodeVocab?term=CTCF,H3K4me1,H3K4me2,H3K4me3,H3K27ac,H3K9ac,H3K36me3,H4K20me1,H3K27me3,Pol2,Input">
        eight chromatin marks, RNA Polymerase II, the CTCF transcription factor and input data</a>. In total,
        twenty-five states were used to segment the genome, and these states were then grouped and colored to
        highlight predicted functional elements. These Genome Segmentations are the same data as
        found in the Analysis Working Group Hub, but are now hosted natively in the Browser with enhanced
        filtering capability where desired segmented states can be selected using the
        'Filter by Segment Type' control on the track configuration page.
        </p>

        <hr>

        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b></font>: 

        <p>
        <font face="courier" size="3"><b>18 April 2014 - 
        New Motif Displays and New Genome Segmentations from ENCODE</b></font>:
        The Browser's default displayed 
        <a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeRegTfbsClusteredV3">Transcription Factor ChIP-seq track</a>
        updated to the latest release, enhanced with the display of Factorbook motifs. And the Browser added
        <a class="euro" href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=wgEncodeAwgSegmentation">Genome
        Segmentations from ENCODE</a> tracks, which segment the genome into twenty-five states that were then grouped
        and colored to highlight predicted functional elements.
        <a href="goldenPath/newsarch.html#041814">Read more</a>.
        </p>

        <font face="courier" size="3"><b> 29 Apr 2014 - 
        100-species Conservation track re-release</b></font>
        <p>
        We released the initial version of the 100-species Conservation track
        for the hg19 human assembly in November 2013. Over the past few months,
        we found a couple of inconsistencies and decided, for the integrity of
        the data, that we should rerun the computation pipeline and re-release
        the data. <a href="goldenPath/newsarch.html#042914">Read more</a> 

                -->
        <!-- start archives -->

        <p>
        <font face="courier" size="3"><b>15 April 2014 -
        New Minke Whale (balAcu1) Assembly Now Available in the
        Genome Browser</b></font>:
        A Genome Browser is now available for the minke whale
        (Balaenoptera acutorostrata scammoni) assembly released 
        October 2013 by the Korea Ocean Research &amp; Development Institute.
        <a href="goldenPath/newsarch.html#041514">Read more</a>.
        </p>


        <p>
        <font face="courier" size="3"><b>11 April 2014 -
        New UCSC Genes Track Released for GRCm38/mm10</b></font>:
         We're happy to announce the release of an updated UCSC Genes
        track for the GRCm38/mm10 mouse Genome Browser.
        <a href="goldenPath/newsarch.html#041114">Read more</a>.
        </p>


            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
