<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        <p>
        <font face="courier" size="3"><b>14 January 2014 -
        Two E. coli comparative assembly hubs now available</b></font>
        </p>
        <p>
        We're pleased to add two new assembly hubs produced by the UCSC
        <A HREF="https://cbse.soe.ucsc.edu/hausslerlab" TARGET="_BLANK">David Haussler lab</A> 
        to our collection of publicly available hubs. The new hubs feature over 60 bacterial assemblies, 
        including more than 55 different E. coli strains. The assembly annotations include genes, pathogenic
        genes, conservation, GC percent, repetitive elements and much more.
        </p>
        <p>
        These hubs focus on comparative genomics and showcase the new "snake" track type. Snakes, which
        visualize alignments from
        <A HREF="http://genome.ucsc.edu/FAQ/FAQformat.html#format20">Hierarchical Alignment</A>
        (HAL) files, provide a way to view sets of pairwise gapless alignments that may overlap on both
        the chosen genome (reference) and the query genome, and show various types of genomic variations
        such as insertions, substitutions, and duplications. More details about the new snake track display
        and its configuration options can be found on our Genome Browser
        <a href="http://genome.ucsc.edu/goldenPath/help/hgTracksHelp.html" target="_blank">help</a> page.
        </p>
        <p>
        To access and view these hubs, navigate to the
        <A HREF="http://genome.ucsc.edu/cgi-bin/hgHubConnect">Track Hub gateway page</A>
        and select one of the two E. coli comparative assembly hubs from the Public Hubs list.
        </p>
        <p>
        We would like to acknowledge Ngan Nguyen, Glenn Hickey, Brian Raney, Joel Armstrong, Benedict Paten,
        Matt Speir, and Luvina Guruvadoo for their hard work in creating these hubs. 
        </p>
        <hr>

       <p>
        <font face="courier" size="3"><b>27 November 2013 -
        100 Species Conservation Track now available on hg19</b></font>
        </p>
        <p>
        After 15.4 years of CPU run-time in 9,905,594 individual 'jobs' and 99 cluster runs for lastz 
        pair-wise alignment...we are excited to announce the release of a 100 species alignment on the 
        hg19/GRCh37 human Genome Browser.
        </p>
        <p>
        This new Conservation track shows multiple alignments of 100 species and measurements of 
        evolutionary conservation using two methods (phastCons and phyloP) from the PHAST package. 
        This adds 40 more species to the existing 60-way on the mm10 mouse browser. For more information 
        about the 100 species Conservation track, see its 
        <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=hg19&g=cons100way" target="_blank">description page</a>.
        </p>
        <p>
        We'd also like to acknowledge the hard work of the UCSC Genome Browser staff who pulled together 
        the information for this track: Hiram Clawson and Pauline Fujita.
        </p>

        <!-- staged for archive use
        <p>
        <font face="courier" size="3"><b>14 January 2014 -
        Two E. coli comparative assembly hubs now available</b></font>
        We're pleased to add two new assembly hubs produced by the UCSC
        <A HREF="https://cbse.soe.ucsc.edu/hausslerlab" TARGET="_BLANK">David Haussler lab</A> 
        to our collection of publicly available hubs.
        <a href="goldenPath/newsarch.html#011414">Read more</a> 
        </p>

        <p>
        <font face="courier" size="3"><b>27 November 2013 -
        100 Species Conservation Track now available on hg19</b></font>
        </p>
       
        <p>
        We're happy to announce the release of a new track on the hg19/GRCh37 human Genome Browser. 
        The new Conservation track shows multiple alignments of 100 species and measurements of 
        evolutionary conservation using two methods (phastCons and phyloP) from the PHAST package. 
        </p> 
        -->

        <!-- start archives -->
        <hr>
        <p>
        <font face="courier" size="3"><b>24 October 2013 -
        Job Opening: UCSC Genome Browser Trainer</b></font>
	The <a href="http://cbse.ucsc.edu/" target="_blank">Center for Biomolecular Science
	and Engineering</a> (CBSE) at University of California Santa Cruz
	seeks an articulate, self-motivated educator for the two-year position of UCSC Genome
	Browser trainer. 
        <a href="goldenPath/newsarch.html#102413">Read more</a>.
        </p>
        
        <p>
        <font face="courier" size="3"><b>23 October 2013 -
        dbSNP 138 Available for hg19: </b></font>
	We are pleased to announce the release of four tracks derived from NCBI
 	<a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 138
 	data, available on the human assembly (GRCh37/hg19). 
        <a href="goldenPath/newsarch.html#102313">Read more</a>.
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
