<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->

    <!-- staged announcemnt for YouTube 
	<hr>
        <p>
        <font face="courier" size="3"><b>19 January 2015 - Genome Browser YouTube Channel</b></font>

        <p>We are pleased to announce the release of the UCSC Genome Browser
        <A HREF = "https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos">YouTube
        channel</A>.  The channel contains short videos showing
        how to use the Genome Browser and associated tools to solve selected problems.  The
        videos were produced by Robert Kuhn and Pauline Fujita, with assistance from Luvina
        Guruvadoo, Jonathan Caspar and the rest of the Browser staff.
        </p>

        <p>See also the other materials on our <a href = "training/">training page</a>.

        </p>
    -->
        <hr>
        <p>
        <font face="courier" size="3"><b>19 December 2014 -
        New Cow (bosTau8) Assembly Now Available in the Genome Browser</b></font>
        <p>
        We are pleased to announce the release of a Genome Browser for the June 2014 assembly
        of cow, <i>Bos taurus</i> (Bos_taurus_UMD 3.1.1, UCSC version bosTau8). This updated cow assembly was 
	provided by the  <a href="http://www.cbcb.umd.edu/"
        target="_blank">UMD Center for Bioinformatics and Computational Biology</a> (CBCB). This
	assembly is an update to the previous UMD 3.1 (bosTau6) assembly.  UMD 3.1 contained 
	138 unlocalized contigs that were found to be contaminants. 
	These have been suppressed in UMD 3.1.1.
        <p>
        Bulk downloads of the sequence and annotation data are available via the Genome
        Browser
        <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/bosTau8/">FTP server</a>
        or the
        <a href="http://hgdownload.cse.ucsc.edu/downloads.html#cow">Downloads</a>
        page. These data have
        <a href="/goldenPath/credits.html#cow_credits">specific conditions for use</a>.
        The cow browser annotation tracks were generated by UCSC and
        collaborators worldwide. See the
        <a href="/goldenPath/credits.html#cow_credits">Credits</a> page for a detailed
        list of the organizations and individuals who contributed to this release.
        </p>

	<hr>
        <p>
        <font face="courier" size="3"><b>12 December 2014 -
        New <i>D. melanogaster</i> (dm6) Assembly Now Available in the Genome Browser</b></font>
        <p>
	We are pleased to announce the release of a Genome Browser for the August 2014 assembly
        of <i>Drosophila melanogaster</i> (BDGP Release 6, UCSC version dm6). This updated <i>D.
	melanogaster</i> assembly was provided by the  <a href="http://www.fruitfly.org/"
        target="_blank">Berkeley Drosophila Genome Project</a> (BDGP). Please see
	<a href="http://flybase.org/static_pages/feature/previous/articles/2014_07/FB2014_04.html"
	target="_blank">this announcement</a> on the FlyBase website for more details about this
	updated <i>D. melanogaster</i> assembly.
	<p>
	Bulk downloads of the sequence and annotation data are available via the Genome
        Browser
        <a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/dm6/">FTP server</a>
        or the
        <a href="http://hgdownload.cse.ucsc.edu/downloads.html#fruitfly">Downloads</a>
        page. These data have
        <a href="/goldenPath/credits.html#dm_credits">specific conditions for use</a>.
        The <i>D. melanogaster</i> (dm6) browser annotation tracks were generated by UCSC and
        collaborators worldwide. See the
        <a href="/goldenPath/credits.html#dm_credits">Credits</a> page for a detailed
        list of the organizations and individuals who contributed to this release.
        </p>

	<hr>

        <!-- start archives -->

 	<p>
        <font face="courier" size="3"><b>29 October 2014 -
        Genome Browser in a Box (GBiB)</b>:</font>
        Sometimes you just want to keep your genomics data to yourself. Have you ever
        hesitated when uploading your data set into the UCSC Genome Browser? If so,
        you'll be happy to know that we have created a stand-alone personal version:
        Genome Browser in a Box (GBiB). <a href="goldenPath/newsarch.html#102914">Read more</a>.

	<p>
        <font face="courier" size="3"><b>20 October 2014 -
        dbSNP 141 Available for hg19 and hg38</b>: </font>
        We are pleased to announce the release of four tracks derived from NCBI
        <a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 141
        data, available on the two most recent human assemblies GRCh37/hg19 and GRCh38/hg38.
        <a href="goldenPath/newsarch.html#102014">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
