<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
  <hr>
  <p>
  <font color="#006666" size="4"><b>
  Happy Holidays from the UCSC Genome Browser! &mdash;</b></font>
  Here's a <A
  HREF="../cgi-bin/hgTracks?db=hg38&hgct_customText=http://hgwdev.cse.ucsc.edu/~brianlee/customTracks/testBigGenePred/testFun/customTrackLines">card</A> 
  for you!
  <hr>
  <p>
  <font color="#006666" size="4"><b>16 December 2015 &mdash;
  Output from Related Tables Now Available in the Data Integrator</b></font>
  <p>
  In the UCSC Genome Browser database, detailed information on the annotations for many tracks
  is stored in extra tables. This information can include things such as identifiers
  in other databases, transcript status, or other descriptive information. You can now obtain
  data from these related tables in your output from the
  <a href="cgi-bin/hgIntegrator">Data Integrator</a>.
  </p>

  <p>
  To add related tables to your Data Integrator output, first, select and add at least one
  track to your data sources. Then, scroll down to the "Output Options" section and click the
  <button type="button">Choose fields</button> button.
  Lastly, at the bottom the section for each selected track, you will see a drop-down menu
  labeled "Related tables" which allows you to select and add various tables to your output.
  <p><img style="border: 1px solid black; display: block;"
                src="images/hgIntegratorRelatedTablesAnnouncement.png">
  <p>
  For more information about using the Data Integrator, please see the
  <a href="goldenPath/help/hgIntegratorHelp.html">User Guide</a>.
  </p>

  <p>
  Thank you to Angie Hinrichs and Matthew Speir for their work on this feature.
  </p>

  <hr>
  <p>
  <font color="#006666" size="4"><b>9 December 2015 &mdash;
  Data from 1000 Genomes Project Phase 3 Now Available</b></font>
  <p>
  We have made data from Phase 3 of the
  <a href="http://www.1000genomes.org/" target="_blank">1000 Genomes Project</a>
  available for the hg19 version of the human assembly.
  The data includes almost 90 million variants in the form of single nucleotide variants (SNVs),
  insertions/deletions (InDels), and structural variants (SVs). Each variant includes a
  detail page that incorporates links to the variant in the dbSNP database, quality scores,
  and allele frequency information for various populations. More information is available
  on the <a href="../cgi-bin/hgTrackUi?db=hg19&g=tgpPhase3">track description page</a>.
  </p>

  <p>
  In addition to variant calls, the data includes two tracks that highlight regions of the
  genome that are accessible to next generation sequencing technologies that utilize
  paired-end reads. One track contains regions that meet the  1000 Genomes Project
  &quot;pilot&quot; criteria while the other contains regions that meet a more
  &quot;strict&quot; set of guidelines. More information about how these accessible regions
  were identified can be found on the
  <a href="../cgi-bin/hgTrackUi?db=hg19&g=tgpPhase3Accessibility">track description page</a>.
  </p>

  <p>
  Thank you to the 1000 Genomes Project and Tom Blackwell and Mary Kate Wing at the
  University of Michigan for making these data available. Thank you to Angie Hinrichs
  and Matthew Speir at UCSC for their work on these tracks.
  </p>

  <hr>
  <p>
  <font color="#006666" size="4"><b>25 November 2015 &mdash;
  New &quot;View - In External Tools&quot; Menu Sends Data to Prediction Websites!</b></font>
  <p>
  Do you often get the current sequence with &quot;View - DNA&quot; just to copy
  and paste it into another website? With the new menu &quot;View - In External Tools&quot;
  you can now send the current region to remote tools with two clicks.
  </p>
  <p>
  For example, when browsing hg38 and zoomed in on the start of a gene like ABO at
  <a href="/cgi-bin/hgTracks?db=hg38%26position=chr9%3A133255000-133255555"
  target="_blank">chr9:133,255,000-133,255,555</a>, 
  click the View menu and select &quot;In External Tools&quot;,
  or simply type the keyboard shortcut &quot;s&quot; then &quot;t&quot;,
  to send the region to a list of different prediction tools.</p>
  <p>
  Some of the available tools (not available for all assemblies or all zoom levels)
  are primer design and restriction enzyme predictions (Primer3Plus, Primer-BLAST,
  NEBCutter), mRNA and protein predictions (RegRNA 2, Riboswitches, RNAfold, PFAM),
  cis-regulatory predictions (Promoter 2, RSAT Metazoa), and CRISPR/Cas9 predictions
  (CRISPOR, CHOPCHOP), as well as links to view the region at NCBI or Ensembl.
  You can suggest other tools via our
  <a href="/cgi-bin/hgUserSuggestion">Suggestion Box</a>.</p>
  <p>
  Credit goes to Max Haeussler for implementing this new feature.
  </p>
  <hr>
  
        <!-- start archives -->
<!--staged for future release
	<p>
	<font color="#006666" size="4"><b>16 December 2015 &mdash;
	Output from Related Tables Now Available in the Data Integrator</b>: </font>
	You can now obtain data from these related tables in your output from the
	<a href="cgi-bin/hgIntegaror">Data Integrator</a>.
	<a href="goldenPath/newsarch.html#121615">Read more</a>.
	</p>

	<p>
	<font color="#006666" size="4"><b>9 December 2015 &mdash;
	Data from 1000 Genomes Project Phase 3 Now Available</b>: </font>
	We have made data from Phase 3 of the
	<a href="http://www.1000genomes.org/" target="_blank">1000 Genomes Project</a>
	available for the hg19 version of the human assembly.
        <a href="goldenPath/newsarch.html#120915">Read more</a>.
	</p>

        <p>
        <font color="#006666" size="4"><b>25 November 2015 &mdash;
        New &quot;View - In External Tools&quot; Menu Sends Data to Prediction Websites!</b></font>
        Do you often get the current sequence with &quot;View - DNA&quot; just to copy
        and paste it into another website? With the new menu &quot;View - In External Tools&quot;
        you can now send the current region to remote tools with two clicks.
        <a href="goldenPath/newsarch.html#112515">Read more</a>.
        </p>
-->
  	<p>
        <font color="#006666" size="4"><b>3 November 2015 &mdash;
        New quick links for the Genome Browser's top menu bar </b>:</font>
        The "Genomes" menu now features direct links to jump to browsing the human
        (hg38 and hg19) and mouse (mm10 and mm9) assemblies!
        <a href="goldenPath/newsarch.html#110315">Read more</a>.
  	</p>

        <p>
        <font color="#006666" size="4"><b>23 September 2015 &mdash;
        New keyboard shortcuts on the Genome Browser </b>:</font>
        It's now possible to zoom in and out, navigate left
        and right, hide all tracks, configure the browser display and much more using
        only your keyboard. Use the "?" key to view the entire list of shortcuts.
        <a href="goldenPath/newsarch.html#092315">Read more</a>. 
  	</p>
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
