<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list.</P>

    <!-- start news -->
        <hr>
        <p>
        <font face="courier" size="3"><b> 19 June 2014 -
	Conservation Track Released for Rat</b>
        </font>
        </p>

	<p>
	We are pleased to announce the release of a new Conservation track based on the rat 
	(RGSC 5.0/rn5) assembly. This track shows multiple alignments of 13 vertebrate species and 
	measurements of evolutionary conservation using two methods (phastCons and phyloP) from the 
	PHAST package for all species. The multiple alignments were generated using multiz and other
	tools in the UCSC/Penn State Bioinformatics comparative genomics alignment pipeline. 
	Conserved elements identified by phastCons are also displayed in this track. For more 
	details, visit the 
	<a href="../cgi-bin/hgTrackUi?db=rn5&g=cons13way">track description page</a>.
	</p>

	<p>
	With the release of this new conservation track, we have changed the default rat browser on 
	our website from rn4 to rn5. In general, the Genome Browser will continue to display 
	whichever rat assembly you have already been viewing, and will default to the rn5 assembly 
	only for those users who haven't recently viewed a rat assembly. However, there are 
	circumstances in which the browser may update the view to the newer version, for instance 
	if you reset your browser settings to the default values. If you find yourself
	in a situation where some of your favorite browser tracks have &quot;disappeared&quot;, 
	you may want to check that you're viewing the expected assembly.
	</p>

	<p>
	We would like to thank Pauline Fujita, Matthew Speir, and Hiram Clawson for their efforts in
	creating this track.
	</p>

	<hr>
        <p>
        <font face="courier" size="3"><b> 30 May 2014 -
	New Human Variation Track Now Available</b>
        </font>
        </p>

	<p>
	We would like to announce the release of the new Exome Variant Server (EVS) Variants track. 
	The
	<a href="http://evs.gs.washington.edu/EVS/" target="_blank">EVS</a>, maintained by the
	<a href="http://chum.gs.washington.edu/" target="_blank">Nickerson lab</a> at the University
	of Washington, contains variant annotations for over 6,500 exomes that were sequenced as 
	part of the <a href="https://esp.gs.washington.edu/drupal/" 
	target="_blank">NHLBI GO Exome Sequencing Project</a>
	(ESP). This annotation includes information such as function, clinical association, dbSNP 
	rsID,
	average sample read depth, and much more. You can find more information on how the data for 
	this track was generated on the 
	<a href="../cgi-bin/hgTrackUi?db=hg19&g=evsEsp6500">track description page</a>.

	<p>
	We would like to acknowledge Tim Dreszer and Matthew Speir for their hard work in creating 
	this track.
	</p>

        <hr>

        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b>: </font>

	<p>
        <font face="courier" size="3"><b> 19 June 2014 -
        Conservation Track Released for Rat (rn5)</b>: </font>
        We have released a new Conservation track based on the rat (rn5) assembly.
        <a href="goldenPath/newsarch.html#061914">Read more</a>.
	</p>

	<p>
	<font face="courier" size="3"><b> 30 May 2014 -
	New Human Variation Track Now Available</b>: </font>
        We are pleased to announce the release of the new Exome Variant Server (EVS) Variants track.
        <a href="goldenPath/newsarch.html#053014">Read more</a>.
	</p>
                -->

        <!-- start archives -->

	<p>
        <font face="courier" size="3"><b> 22 May 2014 -
        New Roadmap Epigenomics Integrative Analysis Hub now available</b>: </font>
        We are pleased to announce the addition of the
        <a href="../cgi-bin/hgTracks?db=hg19&hubUrl=http://vizhub.wustl.edu/VizHub/RoadmapIntegrative.txt">Roadmap Epigenomics Integrative Analysis Hub</a> 
	to our list of publicly available track hubs.
	<a href="goldenPath/newsarch.html#052714">Read more</a>.
	</p>

	<p>
	<font face="courier" size="3"><b> 15 May 2014 -
        New UCSC Gene Track Released for GRCh38/hg38</b>: </font>
	We're happy to announce the release of a new UCSC Genes track for the
        GRCh38/hg38 human Genome Browser.
	<a href="goldenPath/newsarch.html#051514">Read more</a>.
	</p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
