<!-- News Section ============================================- -->  
<TABLE style="background-color:#888888; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
    <TR><TD>        
    <TABLE style="background-color:#fffee8; width: 100%; border-width: 0px; border-spacing: 0px; padding: 0px">
        <TR><TD>        
        <TABLE style="background-color:#D9E4F8; background-image: url('images/hr.gif'); width: 100%; border: 0px">
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE style="background-color:#fffee8; width: 100%; padding: 0px">
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
  <hr>
  <font color="#006666" size="4"><b>8 Mar 2016
  - Combine Multiple Regions of the Genome Browser into a Single Visualization!</b></font>
  <p>
  Have you ever wished you could remove all of the intronic or intergenic regions from the
  Genome Browser display? Have you ever dreamed of being able to visualize multiple
  far-flung regions of a genome? Well, now you can with the new &quot;multi-region&quot; option
  in the Genome Browser!
  </p>

  <p>
  The multi-region modes provide four different ways to manipulate the display.
  First, we have an &quot;exon-only&quot; mode that allows you to
  remove all of the intergenic and intronic sequences from the display, leaving you with
  only the exonic regions. Next, there is the &quot;gene-only&quot; mode allowing you to remove only
  the intergenic regions from the display, leaving the genes with exons and introns intact.
  Then there is the &quot;custom regions&quot; mode that allows you to use a remote BED file to
  specify the regions you are interested in viewing alongside each other. Lastly, for human
  assemblies hg17 and newer, there is the &quot;alternative haplotype&quot; mode that allows you to
  view a haplotype sequence inserted into its position in the reference genome.
  </p>

  <p>
  To access these exciting, new multi-region modes, first select your organism and assembly
  of interest and navigate to the Genome Browser visualization. Below the tracks display,
  there is a button labeled &quot;multi-region&quot;. Clicking this button will bring up
  a configuration box for the different multi-region modes. For example:
  </p>

  <p>
  <img style="border: 1px solid black; display: block;"
                src="images/MultiRegionButton.png">
  </p>

  <p>
  From this pop-up, you will be able to switch between the different modes or exit these
  multi-region modes if desired. For instance:
  </p>

  <p>
  <img style="border: 1px solid black; display: block;"
                src="images/MultiRegionConfigurationWindow.png">
  </p>

  <p>
  Additionally, you can enter the &quot;Exon-only&quot; mode by typing the keyboard shortcut
  &quot;e&quot; then &quot;v&quot;. You can exit any of the multi-region modes and return to the
  default Genome Browser display by typing the keyboard shortcut &quot;d&quot; then &quot;v&quot;.
  </p>

  <p>
  For more information about each individual multi-region mode, please see the
  <a href="goldenPath/help/multiRegionHelp.html">User Guide</a>.
  </p>

  <p>
  Thank you to Galt Barber, Matthew Speir, and the entire UCSC Genome Browser quality assurance team for
  all of their efforts in creating these exciting new display modes.
  </p>

  <hr>
<p>
  <font color="#006666" size="4"><b> 10 February 2016 -
  Two New Assemblies Now Available in the Genome Browser</b></font>
  <p>
  A Genome Browser is now available for the mouse lemur (<em>Microcebus murinus</em>)
  assembly released May 2015 by the
<A HREF="http://www.broadinstitute.org/scientific-community/science/projects/mammals-models/29-mammals-project"
TARGET="_blank">The Broad Institute</A> and the
<A HREF="https://www.hgsc.bcm.edu/non-human-primates/mouse-lemur-genome-project"
TARGET="_blank">Baylor College of Medicine Human Genome Sequencing Center</A>.
  For more information and statistics about this assembly, see the NCBI assembly record
  for <a href="http://www.ncbi.nlm.nih.gov/assembly/329621" target="_blank"> Mmur_2.0</a>.
There are 10,311 scaffolds with a total size of 2,438,804,424 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/micMur2/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#mouse_lemur">Downloads</a> page.
  Please observe the <a href="./goldenPath/credits.html#mouse_lemur_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="./goldenPath/credits.html#mouse_lemur_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
  ---
  <p>
  A Genome Browser is now available for the platypus (<em>Ornithorhynchus anatinus</em>)
  assembly released February 2007 by the
<A HREF="http://genome.wustl.edu/genomes/detail/ornithorhynchus-anatinus/"
TARGET=_blank>Genome Sequencing Center at Washington University, St. Louis</A>.
For more information about this assembly, see
<A HREF="http://www.ncbi.nlm.nih.gov/assembly/237598/" TARGET="_blank">
Ornithorhynchus_anatinus-5.0.1</A> in the NCBI Assembly database. There are
201,525 scaffolds with a total size of 1,996,826,513 bases.
  </p>

  <p>
  Bulk downloads of the sequence and annotation data may be obtained from the Genome Browser
  <a href="ftp://hgdownload.soe.ucsc.edu/goldenPath/ornAna2/">FTP server</a> or the
  <a href="http://hgdownload.soe.ucsc.edu/downloads.html#platypus">Downloads</a> page.
  Please observe the <a href="./goldenPath/credits.html#platypus_use">conditions for use</a> when
  accessing and using these data sets.  The annotation tracks for this browser were
  generated by UCSC and collaborators worldwide.  See the <a
  href="./goldenPath/credits.html#platypus_credits">Credits</a> page for a detailed list of the
  organizations and individuals who contributed to this release.
  </p>
<hr>

<!-- start archives -->
<!--staged for future release

	<p>
	<font color="#006666" size="4"><b>8 Mar 2016
	- Combine Multiple Regions of the Genome Browser into a Single Visualization!</b></font>
	<a href="goldenPath/newsarch.html#030816">Read more</a>.
        </p>

        <p>
        <font color="#006666" size="4"><b> 10 February 2016 -
        Two New Assemblies Now Available in the Genome Browser</b>: </font>
        Genome Browsers are now available for the mouse lemur 
	(<em>Microcebus murinus</em>)
        and platypus (<i>Ornithorhynchus anatinus</i>) assemblies.
	<a href="goldenPath/newsarch.html#021016">Read more</a>. 
        </p>

-->
	<font color="#006666" size="4"><b>20 Jan 2016 &mdash;
	dbSNP 142 Available for mm10</b></font>
        <a href="goldenPath/newsarch.html#012016">Read more</a>.
        </p>
        <p>
        <font color="#006666" size="4"><b>08 January 2016 &mdash; 
        dbSNP 144 Available for hg19 and hg38 </font></b>
        <a href="goldenPath/newsarch.html#010816">Read more</a>.
        </p>

            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
