<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                          <FONT SIZE="4"><A NAME="TOC"></A><B>&nbsp; News</B></FONT></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                    <TR><TH HEIGHT=3></TH></TR>
                    <TR><TD WIDTH=10></TD>
                    <TD>
                    <P>
                    To receive announcements of new genome 
                    assembly releases, new software features, updates and 
                    training seminars by email, subscribe to the
                    <A HREF="https://lists.soe.ucsc.edu/mailman/listinfo/genome-announce" 
                    TARGET=_blank>genome-announce</A> mailing list.</P>

                    <!-- start news -->
		    <P>
                    <FONT FACE="courier" SIZE="3"><B>19 January 2011 -
Browser Released for Zebrafish Zv9 Assembly</B></FONT>
		    <P>
                    The most recent zebrafish assembly -- Zv9 (UCSC version
                    danRer7, Jul. 2010) -- is now available in the UCSC
                    Genome Browser. This assembly was produced by
                    <A HREF="http://www.sanger.ac.uk/"
		    TARGET=_blank>The Wellcome Trust Sanger Institute</A>, UK.
                    </P>
                    <P>
		    The Zv9 assembly comprises a sequence length of 1.4 Gb in 
		    26 chromosomes and 1,107 scaffolds. This assembly is based 
		    on a clone path sorted with 
		    the high-density meiotic map SATMAP (Clark et al., in 
		    preparation). The data freeze was taken on 1 April 2010. 
		    The remaining gaps were filled with sequence from WGS31, a 
		    combined Illumina and capillary assembly. The assembly 
		    integration process involved sequence alignemnts as well as 
		    cDNA, marker and BAC/Fosmid end sequence placements. 
		    The zebrafish mitochondrial sequence is also available in 
		    the Genome Browser as the virtual chromosome, chrM. For 
		    more details about the Zv9 assembly, see the Sanger 
		    Institute page for the 
		    <A HREF="http://www.sanger.ac.uk/Projects/D_rerio/Zv9_assembly_information.shtml" 
		    TARGET=_blank><em>Danio rerio</em> Sequencing Project</A>. 
		    </P>

                    <P>
                    The danRer7 sequence and annotation data can be
                    downloaded from the UCSC Genome Browser
                    <A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/danRer7/">FTP server</A>
                    or
                    <A HREF="http://hgdownload.cse.ucsc.edu/downloads.html#zebrafish">downloads</A>
                    page. Please review the Sanger Institute
                    <A HREF="http://www.sanger.ac.uk/legal/"
                    TARGET=_blank>guidelines</A> for using these data.
                    <P>
                    We'd like to thank the Wellcome Trust Sanger Institute
                    for providing this assembly.
                    The UCSC zebrafish Genome Browser
                    was produced by Hiram Clawson, Greg Roe, Mary Goldman,
                    Brian Raney and Donna Karolchik. See the
                    <A HREF="goldenPath/credits.html#zebrafish_credits">credits</A>
                    page for a detailed list of the organizations and
                    individuals who contributed to this release.</P>
		    <P>
		    <HR>
                    <FONT FACE="courier" SIZE="3"><B>10 January 2011 - Introducing Genome Browser Right-click Navigation</B></FONT>
		    <P>
		    Several of the common display and navigation operations 
		    offered on the Genome Browser tracks page may now be quickly
		    accessed by right-clicking on a feature on the tracks image 
		    and selecting an option from the displayed menu. Depending 
		    on context, the right-click feature will let you 
		    change the track display mode, zoom in or out to the exact 
		    position coordinates of the feature, open the &quot;Get 
		    DNA&quot; window at the feature's coordinates, display 
		    details about the feature, open a popup window to configure 
		    the track's display, or display the entire tracks image in 
		    a separate window for inclusion in spreadsheets or other 
		    documents. (Note that the &quot;PDF/PS&quot; option in the 
		    Genome Browser top navigation bar can also be used to 
		    generate a high-quality annotation tracks image suitable 
		    for printing.)
		    <BR><BR>
                    <img style="border: 2px solid black; margin-left:auto; 
		    margin-right:auto; display: block;" 
		    src="/images/rightclick.png"
                    alt="Right-click screenshot">
		    <BR>
		    <P>
		    To use the right-click feature, make sure the &quot;Enable 
		    advanced javascript features&quot; option on the 
		    <A
HREF="../cgi-bin/hgTracks?hgTracksConfigPage=configure+tracks+and+display">tracks 
		    configuration page</A> is checked, and configure your 
		    internet browser to allow the display of popup windows from 
		    <em>genome.ucsc.edu</em>. When enabled, the right-click 
		    navigation feature replaces the default contextual menu 
		    typically displayed by your internet browser when you 
		    right-click on the tracks image. A few combinations of the 
		    Mozilla Firefox browser on Mac OS do not support the 
		    right-click menu functionality using secondary click; in 
		    these instances, you must use the explicit ctrl-click action
		    to display the menu. 
		    <P>
	 	    Credit goes to Larry Meyer and Brooke Rhead for doing the
		    lion's share of the design, development and testing of this 
		    feature, with engineering support from Tim Dreszer and 
		    additional testing by several others on the QA team.
		    <HR>
                    <FONT FACE="courier" SIZE="3"><B>2 December 2010 -
Searching for Tracks Just Got Easier!</B></FONT>
		    <P>
		    The number of genome assemblies and annotation
		    tracks in the UCSC Genome Browser has increased 
		    exponentially over the past few years. Along with this 
		    growth, it has become more difficult for browser users to 
		    sort through the many tracks to find the data they're 
		    interested in.
		    <P>
		    With our latest Genome Browser release, we've introduced a 
		    new track search feature that will help you to quickly and 
		    easily find and display specific annotation data sets. The 
		    search feature is available via the "track search" button 
		    on the browser gateway and tracks display pages.
		    <A HREF="goldenPath/newsarch.html#120210">Read
more</A>.
		    </P>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
