<!--News Section----------------------------------------------->  
        <TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
            <TR><TD>        
            <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
                <TR><TD>        
                <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
                    <TR><TD>
                    <!--cheap trick to get background on row to be continuous-->
                    <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                        <TR><TD ALIGN="left">
                         <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" alt="Follow GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a></TD>
                        <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                        SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
                    </TD></TR></TABLE>
                </TD></TR></TABLE>
                <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
                  <TR><TH HEIGHT=3></TH></TR>
                  <TR><TD WIDTH=10></TD>
                  <TD>
                  <P>
                  To receive announcements of new genome 
                  assembly releases, new software features, updates and 
                  training seminars by email, subscribe to the
                  <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
                  TARGET=_blank>genome-announce</A> mailing list.</P>

            <!-- start news -->
                <font face="courier" size="3"><b>07 December 2012 -
                dbSNP 137 Available for hg19</b></font>

                <p>
                We are pleased to announce the release of four tracks derived
                from dbSNP build 137, available on the human assembly (GRCh37/hg19).
                dbSNP build 137 is available at NCBI. The new tracks contain
                additional annotation data not included in previous dbSNP tracks,
                with corresponding coloring and filtering options in the Genome
                Browser.
                </p>

                <p>
                As for dbSNP build 137, there are four tracks in this release.
                One is a track containing all mappings of reference SNPs to the
                human assembly, labeled "All SNPs (137)". The other three tracks
                are subsets of this track and show interesting and easily defined
                subsets of dbSNP:
                </p>

                <ul>
                <li>Common SNPs (137): uniquely mapped variants that appear in
                at least 1% of the population
                <li>Flagged SNPs (137): uniquely mapped variants, excluding
                Common SNPs, that have been flagged by dbSNP as "clinically
                associated"
                <li>Mult. SNPs (137): variants that have been mapped to more
                than one genomic location
                </ul>

                <p>
                By default, only the Common SNPs (137) are visible; other
                tracks must be made visible using the track controls.
                </p>

                <p>
                You will find the four SNPs (137) tracks on the Human Feb.
                2009 (GRCh37/hg19) browser in the "Variation and Repeats" group.
                </p>

                </p>
                The tracks were produced at UCSC by Angie Hinrichs and Steve Heitner.
                </p>
                <hr>

	    	<font face="courier" size="3"><b>27 November 2012 -
		New UCSC Genes Track Released for Mouse</b></font>
		<p>
		We're happy to announce the release of a new UCSC Genes 
		track for the GRCm38/mm10 mouse Genome Browser.
		</p>
		<p>
		The new release has 59,121 transcripts, compared with 
		55,419 in the previous version on mm9. The total number 
		of canonical genes has increased from 28,661 to 31,227. 
		Comparing the new gene set with the previous version:
		<p>
		<ul>
		<li>48,739 transcripts did not change between versions.
		<li>423 transcripts were not carried forward to the new version.
		<li>5,108 transcripts are "compatible" with those in the previous 
		set, meaning that the two transcripts show consistent splicing.
		<li>1,037 transcripts overlap with those in the previous set.
		</ul>
		<p>
		More details about the new UCSC Genes track can be found on 
		the <a href="http://genome.ucsc.edu/cgi-bin/hgTrackUi?db=mm10&g=knownGene"
		>UCSC Genes track description page</a>. Also, please note that GRCm38/mm10 
		is now the default mouse assembly on the Genome Browser Gateway.
		<p>
		Many thanks to Brian Raney, Jim Kent, and Luvina Guruvadoo 
		for their work on this track!
		</p>

	    <!-- start archives -->
==> <a href="goldenPath/newsarch.html">News Archives</a>

                    </TD><TD WIDTH=15>
                 </TD></TR></TABLE>
            <BR></TD></TR></TABLE>
        </TD></TR></TABLE>
