<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
	<hr>
        <p>
        <font face="courier" size="3"><b>29 October 2014 -
        Genome Browser in a Box (GBiB)</b></font>
        <p>
	Sometimes you just want to keep your genomics data to yourself. Have you ever 
	hesitated when uploading your data set into the UCSC Genome Browser? If so, 
	you'll be happy to know that we have created a stand-alone personal version: 
	Genome Browser in a Box (GBiB). If you have sensitive genomics data that you 
	would like to view securely on your own laptop in the context of the UCSC 
	Genome Browser, GBiB is for you.
	<p>
	GBiB is an easy-to-install personal copy of the Genome Browser that comes 
	preloaded with the most popular annotation tracks for human. It is highly 
	configurable &mdash; you can access or download other annotation tracks of 
	interest, or view any of the other 90+ organisms featured in the public 
	Genome Browser. GBiB runs inside of Oracle's free VirtualBox virtual machine. 
	It has the same core functionality as the UCSC Genome Browser, but keeps your 
	data private and local to your own computer.
	<p>
	GBiB is free for non-commercial use by non-profit organizations, academic 
	institutions, and for personal use. Commercial use requires purchase of a 
	license with setup fee and annual payment. Download or purchase GBiB in
	our secure online <a href="https://genome-store.ucsc.edu/">store</a>.
	<p>
	You can read more about GBiB on our 
	<a href="http://genome.ucsc.edu/blog/genome-browser-in-a-box-gbib-origins/">
	blog</a>, or in the <a href="/goldenPath/help/gbib.html">
	help doc</a>.
        <p>
	<hr>
        <font face="courier" size="3"><b>20 October 2014 -
        dbSNP 141 Available for hg19 and hg38</b></font>
        <p>
        We are pleased to announce the release of four tracks derived from NCBI
        <a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 141
        data, available on the two most recent human assemblies GRCh37/hg19 and GRCh38/hg38.
	The new tracks contain additional annotation data not included in previous dbSNP
        tracks, with corresponding coloring and filtering options in the Genome Browser.
	</p>
	<p>
	There are three SNP tracks available for the GRCh37/hg19 assembly. One is a track
        containing all mappings of reference SNPs to the human assembly, labeled
        "All SNPs (141)". The other two tracks are subsets of this track and show
        interesting and easily defined subsets of dbSNP:
        </p>
        <ul>
        <li>Common SNPs (141): uniquely mapped variants that appear in at least 1%
        of the population or are 100% non-reference
        <li>Flagged SNPs (141): uniquely mapped variants, excluding Common SNPs,
        that have been flagged by dbSNP as "clinically associated"
        </ul>
	<p>
	Unlike previous dbSNP releases for GRCh37/hg19, Build 141 does not include
	mappings to alternate haplotype sequences. In addition, variants that map to
	multiple genomic locations (e.g. to different chromosomes, or more than one
	position on the same chromosome) have been removed.
	</p>
	<p>
	This release also marks the first set of dbSNP data available for the newest
	human assembly, GRCh38/hg38. In addition to the three tracks described for
	hg19, there is one additional track available for hg38:
	<ul>
        <li>Mult. SNPs (141): variants that have been mapped to more than one genomic location
	</ul>
        <p>
        By default, only the Common SNPs (141) are visible; other tracks must be
        made visible using the track controls. You will find the other SNPs (141) tracks on
	both of GRCh37/hg19 and GRCh38/hg38 browsers in the "Variation" group.
        </p>
        <p>
        The tracks were produced at UCSC by Angie Hinrichs and Matthew Speir. We'd like
        to thank the dbSNP group at NCBI for providing access to these data.
        </p>

        <hr>
        <font face="courier" size="3"><b>07 October 2014 -
        Introducing the UCSC Genome Browser Blog</b>
        </font>
        <p>
        We're excited to introduce a new
        <a href="http://genome.ucsc.edu/blog" target="_blank">blog</a> that will feature posts by
        Genome Browser staff and guests. On the blog we'll be publishing in-depth information about
        UCSC Genome Browser features, tools, projects and related topics that we hope people will
        find both useful and interesting. We just published our inaugural post written by Jim Kent,
        the UCSC Genome Browser Architect and PI, on the
        <a href="http://genome.ucsc.edu/blog/2014-ebola-epidemic/"
        target="_blank">2014 Ebola Epidemic</a>.
        </p>
        <p>
        If you have a suggestion for a blog post topic, please let us know via our
        <a href="/cgi-bin/hgUserSuggestion">suggestion box</a>.
        </p>

        <hr>
        <!-- staged for archive use.  NOTE: condense header line from .... </b></font><p> to </b>: </font>

	<p>
        <font face="courier" size="3"><b>20 October 2014 -
        dbSNP 141 Available for hg19 and hg38</b>: </font>
        We are pleased to announce the release of four tracks derived from NCBI
        <a href="http://www.ncbi.nlm.nih.gov/SNP/" target="_blank">dbSNP</a> Build 141
        data, available on the two most recent human assemblies GRCh37/hg19 and GRCh38/hg38.
        <a href="goldenPath/newsarch.html#102014">Read more</a>.
	</p>

        <p>
        <font face="courier" size="3"><b>07 October 2014 -
        Introducing the UCSC Genome Browser Blog</b>: </font>
        We're excited to introduce a new
        <a href="http://genome.ucsc.edu/blog" target="_blank">blog</a> that will feature posts by
        Genome Browser staff and guests.
        <a href="goldenPath/newsarch.html#100714">Read more</a>.
        </p> -->

        <!-- start archives -->
        <p>
	<font face="courier" size="3"><b>30 September 2014 -
        Genome Browser for the Ebola Virus Now Available</b>: </font>
        We are excited to announce the release of a Genome Browser for the June 2014 assembly
        of Ebola Virus (Sierra Leone 2014 (G3683/KM034562.1), UCSC version eboVir3).
        <a href="goldenPath/newsarch.html#093014">Read more</a>.
	</p>

	<p>
        <font face="courier" size="3"><b>23 September 2014 -
        Ensembl Regulatory Build Now Available in the Genome Browser</b>: </font>
        We are pleased to announce that data from the
        <a href="http://www.ensembl.org/info/genome/funcgen/regulatory_build.html"
        target="_blank">Ensembl Regulatory Build</a> is now available in the UCSC Genome Browser
        as a public track hub for both hg19 and hg38.
	<a href="goldenPath/newsarch.html#092314">Read more</a>.
	</p>
	
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
