<!-- News Section ============================================- -->  
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1">
    <TR><TD>        
    <TABLE BGCOLOR="#fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0">
        <TR><TD>        
        <TABLE BGCOLOR="#D9E4F8" BACKGROUND="images/hr.gif" WIDTH="100%" BORDER=0>
            <TR><TD>
            <!--cheap trick to get background on row to be continuous-->
            <TABLE CELLPADDING=0 CELLSPACING=0 WIDTH="100%">
                <TR><TD ALIGN="left">
                 <font size="4"><a name="TOC"></a><b>&nbsp; News</b></font><a href="http://www.twitter.com/GenomeBrowser" target="_blank"><img src="http://twitter-badges.s3.amazonaws.com/t_small-a.png" title="Follow &#64;GenomeBrowser on Twitter" alt="Follow &#64;GenomeBrowser on Twitter" style="vertical-align:text-bottom; margin-left: 15px;"></a>
                 <a title="Genome Browser Facebook page"
                 href="http://www.facebook.com/ucscGenomeBrowser"
                 target="_blank"><img
                 src="/images/FB-f-Logo__blue_22.jpg"
                 style="vertical-align:text-bottom; margin-left: 15px;"
                 alt="Genome Browser Facebook page"></a>
                 <a href="https://plus.google.com/107643636314131661344" rel="publisher"></a>
               </TD>
                <TD ALIGN="right"><A HREF="goldenPath/newsarch.html"><IMG 
                SRC="images/news.jpg" ALT="" ALIGN="right" BORDER=0></A>
            </TD></TR></TABLE>
        </TD></TR></TABLE>
        <TABLE BGCOLOR="#fffee8" WIDTH="100%" CELLPADDING=0>
          <TR><TH HEIGHT=3></TH></TR>
          <TR><TD WIDTH=10></TD>
          <TD>
          <P>
          To receive announcements of new genome 
          assembly releases, new software features, updates and 
          training seminars by email, subscribe to the
          <A HREF="http://groups.google.com/a/soe.ucsc.edu/group/genome-announce?hl=en"
          TARGET=_blank>genome-announce</A> mailing list. Please see our
          <a href="http://genome.ucsc.edu/blog"
          target=_blank>blog</a> for posts about Genome Browser tools, features, projects and more.
          </p>

    <!-- start news -->
	<hr>
	<p>
	<font face="courier" size="3"><b>08 April 2015 -
        bigGenePred Format Now Supported in the Genome Browser</b></font>
        <p>
	We are pleased to announce that the UCSC Genome Browser now supports data in 
	bigGenePred format. bigGenePred format stores annotation items that are a linked collection 
	of exons, much as BED files do, but bigGenePred has additional information 
	about the coding frames and other gene specific information. bigGenePred files are created initially 
	from BED-type files with some extra fields, using the program bedToBigBed with a special 
	AutoSQL file that defines the fields of the bigGenePred. The resulting bigBed files are in an 
	indexed binary format. The advantage of these bigBed files is that only portions of the files 
	needed to display a particular region are transferred to UCSC. So for large data sets, 
	bigBed is considerably faster than regular BED files.
	<p>
	This new format is available for use in custom tracks and data hubs. For more information 
	about bigGenePred, please see our 
	<a href="/goldenPath/help/bigGenePred.html">bigGenePred Track Format help page</a>.
	</p>

	<hr>
	<p>
	<font face="courier" size="3"><b>02 April 2015 -
	New Tarsier (tarSyr2) Assembly Now Available in the Genome Browser</b></font>
	<p>
	We are pleased to announce the release of a Genome Browser for the September 2013
	assembly of tarsier, <em>Tarsius syrichta</em> (WashU Tarsius_syrichta-2.0.1,
	UCSC version tarSyr2). The whole genome shotgun assembly was provided by
	<a href="http://genome.wustl.edu/genomes/detail/tarsius-syrichta/"
	target="_blank">Washington University</a>.
	There are 337,189 scaffolds with a total size of 3,453,864,774 bases.
	</p>   
	<p>
	Bulk downloads of the sequence and annotation data are available via the Genome
	Browser
	<a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/tarSyr2/">FTP server</a>
	or the
	<a href="http://hgdownload.cse.ucsc.edu/downloads.html#tarsier">Downloads</a>
	page. These data have
	<a href="/goldenPath/credits.html#tarsier_credits">specific conditions for use</a>.
	The tarsier (tarSyr2) browser annotation tracks were generated by UCSC and
	collaborators worldwide. See the
	<a href="/goldenPath/credits.html#tarsier_credits">Credits</a> page for a detailed
	list of the organizations and individuals who contributed to this release.
	</p>

        <hr>
        <font face="courier" size="3"><b>02 April 2015 - New Training Video!</b></font>
        <p>
        Going crazy counting amino acids? Learn an easier way to find the codon number in our new training video. See our
        <a href="../training/index.html" target="_blank">training page</a>, and our
        <a href="http://bit.ly/genomebrowserYoutube" target=_"blank">YouTube channel</a> for details.
        </p>

        <hr>
        <p>
        <font face="courier" size="3"><b>01 April 2015 - New GRCz10 Zebrafish Assembly Now Available</b></font>
        <p>
        The most recent zebrafish assembly -- GRCz10 (UCSC version danRer10, Sep. 2014) -- is now
        available in the UCSC Genome Browser. This is the first zebrafish release by the
        <a href="http://www.ncbi.nlm.nih.gov/projects/genome/assembly/grc/zebrafish/"
         target="_blank">Genome Reference Consortium</a>, who took over the improvement and
        maintenance of the zebrafish genome assembly.
        </p>

        <p>
        The GRCz10 assembly improves upon Zv9 by utilizing optical mapping and additional meiotic
        mapping to fill gaps with new clone sequence and improve the positioning of assembly
        components.  This has also been supplemented with a new whole genome shotgun assembly
        WGS31. The result is an assembly with a sequence length of 1.3 Gb in 26 chromosomes and
        1,035 scaffolds.  For more details about the GRCz10 assembly, see the
        <a href="http://www.ncbi.nlm.nih.gov/assembly/210611"
         target="_blank">GRCz10 assembly page</a> or the official
        <a href="http://www.ncbi.nlm.nih.gov/news/10-08-2014-zebrafish-grcz10-annotated/"
         target="_blank">GRCz10 announcement</a>.
        </p>
        <hr>

        <!-- start archives -->
<!--
	<font face="courier" size="3"><b>08 April 2015 -
        bigGenePred Format Now Supported in the Genome Browser</b>: </font>
        We are pleased to announce that the UCSC Genome Browser now supports data in
        bigGenePred format.
	<a href="goldenPath/newsarch.html#040815">Read more</a>.
        </p>

	<p>
	<font face="courier" size="3"><b>02 April 2015 -
        New Tarsier (tarSyr2) Assembly Now Available in the Genome Browser</b>: </font>
        We are pleased to announce the release of a Genome Browser for the September 2013
        assembly of tarsier, <em>Tarsius syrichta</em> (WashU Tarsius_syrichta-2.0.1,
        UCSC version tarSyr2).
	<a href="goldenPath/newsarch.html#040215_2">Read more</a>.
	</p>

        <p>
        <font face="courier" size="3"><b>02 April 2015 - New Training Video!</b></font>
        View it now on our
        <a href="http://bit.ly/genomebrowserYoutube" target=_"blank">YouTube channel</a>.
        </p>
-->
	<p>
        <font face="courier" size="3"><b>01 April 2015 - New GRCz10 Zebrafish Assembly Now Available</b>: </font>
        The most recent zebrafish assembly -- GRCz10 (UCSC version danRer10, Sep. 2014) -- is now
        available in the UCSC Genome Browser.
        <a href="goldenPath/newsarch.html#040115">Read more</a>.
	</p>

	<p>
        <font face="courier" size="3"><b>20 March 2015 - New Visualization Option for RepeatMasker</b>: </font>
	We are excited to announce the release of the new RepeatMasker Visualization track for the
	hg38 and mm10 assemblies.
        <a href="goldenPath/newsarch.html#032015">Read more</a>.
	</p>
            </TD><TD WIDTH=15>
         </TD></TR></TABLE>
    <BR></TD></TR></TABLE>
</TD></TR></TABLE>
